{
 "metadata": {
  "name": "",
  "signature": "sha256:367aee7ec4f549f70e46df5bb1f61c5ad2af41f467e277d8e1fb14171c65bd63"
 },
 "nbformat": 3,
 "nbformat_minor": 0,
 "worksheets": [
  {
   "cells": [
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "!date"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "Mon Apr 18 11:00:57 PDT 2016\r\n"
       ]
      }
     ],
     "prompt_number": 1
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "%%bash\n",
      "system_profiler SPSoftwareDataType"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "Software:\n",
        "\n",
        "    System Software Overview:\n",
        "\n",
        "      System Version: OS X 10.8.5 (12F45)\n",
        "      Kernel Version: Darwin 12.6.0\n",
        "      Boot Volume: Macintosh HD\n",
        "      Boot Mode: Normal\n",
        "      Computer Name: genefish\n",
        "      User Name: Sam (Sam)\n",
        "      Secure Virtual Memory: Enabled\n",
        "      Time since boot: 68 days 1:00\n",
        "\n"
       ]
      }
     ],
     "prompt_number": 2
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "%%bash\n",
      "system_profiler SPHardwareDataType | grep -v [SH][ea]"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n",
        "\n",
        "      Model Name: Mac mini\n",
        "      Model Identifier: Macmini6,2\n",
        "      Processor Name: Intel Core i7\n",
        "      Processor Speed: 2.3 GHz\n",
        "      Number of Processors: 1\n",
        "      Total Number of Cores: 4\n",
        "      L2 Cache (per Core): 256 KB\n",
        "      L3 Cache: 6 MB\n",
        "      Memory: 16 GB\n",
        "      Boot ROM Version: MM61.0106.B03\n",
        "      SMC Version (system): 2.8f0\n",
        "\n"
       ]
      }
     ],
     "prompt_number": 3
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "%%bash\n",
      "wget https://github.com/dereneaton/pyrad/archive/3.0.66.tar.gz"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stderr",
       "text": [
        "--2016-04-18 11:06:56--  https://github.com/dereneaton/pyrad/archive/3.0.66.tar.gz\n",
        "Resolving github.com... 192.30.252.128\n",
        "Connecting to github.com|192.30.252.128|:443... connected.\n",
        "HTTP request sent, awaiting response... 302 Found\n",
        "Location: https://codeload.github.com/dereneaton/pyrad/tar.gz/3.0.66 [following]\n",
        "--2016-04-18 11:06:57--  https://codeload.github.com/dereneaton/pyrad/tar.gz/3.0.66\n",
        "Resolving codeload.github.com... 192.30.252.160\n",
        "Connecting to codeload.github.com|192.30.252.160|:443... connected.\n",
        "HTTP request sent, awaiting response... 200 OK\n",
        "Length: unspecified [application/x-gzip]\n",
        "Saving to: '3.0.66.tar.gz'\n",
        "\n",
        "     0K .......... .......... .......... .......... ..........  316K\n",
        "    50K .......... .......... ....                             87.9M=0.2s\n",
        "\n",
        "2016-04-18 11:06:57 (469 KB/s) - '3.0.66.tar.gz' saved [76100]\n",
        "\n"
       ]
      }
     ],
     "prompt_number": 4
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "%%bash\n",
      "tar -xzf 3.0.66.tar.gz"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 5
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "ls"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "3.0.66.tar.gz           \u001b[34mDocuments\u001b[m\u001b[m/              \u001b[34mLibrary\u001b[m\u001b[m/                \u001b[34mMusic\u001b[m\u001b[m/                  \u001b[34mPictures\u001b[m\u001b[m/               Untitled0.ipynb\r\n",
        "\u001b[34mDesktop\u001b[m\u001b[m/                \u001b[34mDownloads\u001b[m\u001b[m/              \u001b[34mMovies\u001b[m\u001b[m/                 PE-GBS_empirical.ipynb  \u001b[34mPublic\u001b[m\u001b[m/                 \u001b[34mpyrad-3.0.66\u001b[m\u001b[m/\r\n"
       ]
      }
     ],
     "prompt_number": 6
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "cd pyrad-3.0.66/"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/Users/Sam/pyrad-3.0.66\n"
       ]
      }
     ],
     "prompt_number": 7
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "%%bash\n",
      "python setup.py install"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "running install\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stderr",
       "text": [
        "error: can't create or remove files in install directory\n",
        "\n",
        "The following error occurred while trying to add or remove files in the\n",
        "installation directory:\n",
        "\n",
        "    [Errno 13] Permission denied: '/usr/local/bioinformatics/anaconda/lib/python2.7/site-packages/test-easy-install-83059.write-test'\n",
        "\n",
        "The installation directory you specified (via --install-dir, --prefix, or\n",
        "the distutils default setting) was:\n",
        "\n",
        "    /usr/local/bioinformatics/anaconda/lib/python2.7/site-packages/\n",
        "\n",
        "Perhaps your account does not have write access to this directory?  If the\n",
        "installation directory is a system-owned directory, you may need to sign in\n",
        "as the administrator or \"root\" account.  If you do not have administrative\n",
        "access to this machine, you may wish to choose a different installation\n",
        "directory, preferably one that is listed in your PYTHONPATH environment\n",
        "variable.\n",
        "\n",
        "For information on other options, you may wish to consult the\n",
        "documentation at:\n",
        "\n",
        "  https://pythonhosted.org/setuptools/easy_install.html\n",
        "\n",
        "Please make the appropriate changes for your system and try again.\n",
        "\n"
       ]
      }
     ],
     "prompt_number": 9
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "pyrad -h"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "ename": "NameError",
       "evalue": "name 'pyrad' is not defined",
       "output_type": "pyerr",
       "traceback": [
        "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m\n\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
        "\u001b[0;32m<ipython-input-10-f75d644a6021>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mpyrad\u001b[0m \u001b[0;34m-\u001b[0m\u001b[0mh\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
        "\u001b[0;31mNameError\u001b[0m: name 'pyrad' is not defined"
       ]
      }
     ],
     "prompt_number": 10
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "cd /usr/local/bin"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/usr/local/bin\n"
       ]
      }
     ],
     "prompt_number": 1
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "%%bash\n",
      "wget http://www.drive5.com/muscle/downloads3.8.31/muscle3.8.31_i86darwin64.tar.gz"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stderr",
       "text": [
        "--2016-04-18 11:18:20--  http://www.drive5.com/muscle/downloads3.8.31/muscle3.8.31_i86darwin64.tar.gz\n",
        "Resolving www.drive5.com... 205.196.220.130\n",
        "Connecting to www.drive5.com|205.196.220.130|:80... connected.\n",
        "HTTP request sent, awaiting response... 200 OK\n",
        "Length: 166130 (162K) [application/x-tar]\n",
        "Saving to: 'muscle3.8.31_i86darwin64.tar.gz'\n",
        "\n",
        "     0K .......... .......... .......... .......... .......... 30%  391K 0s\n",
        "    50K .......... .......... .......... .......... .......... 61%  386K 0s\n",
        "   100K .......... .......... .......... .......... .......... 92%  777K 0s\n",
        "   150K .......... ..                                         100% 5.86M=0.3s\n",
        "\n",
        "2016-04-18 11:18:22 (501 KB/s) - 'muscle3.8.31_i86darwin64.tar.gz' saved [166130/166130]\n",
        "\n"
       ]
      }
     ],
     "prompt_number": 2
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "%%bash\n",
      "tar -xzf muscle3.8.31_i86darwin64.tar.gz"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 3
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "%%bash\n",
      "wget https://github.com/torognes/vsearch/releases/download/v1.11.1/vsearch-1.11.1-osx-x86_64.tar.gz"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stderr",
       "text": [
        "--2016-04-18 11:22:27--  https://github.com/torognes/vsearch/releases/download/v1.11.1/vsearch-1.11.1-osx-x86_64.tar.gz\n",
        "Resolving github.com... 192.30.252.131\n",
        "Connecting to github.com|192.30.252.131|:443... connected.\n",
        "HTTP request sent, awaiting response... 302 Found\n",
        "Location: https://github-cloud.s3.amazonaws.com/releases/19848353/5a974bd0-018b-11e6-810c-daa981b4e335.gz?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=AKIAISTNZFOVBIJMK3TQ%2F20160418%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20160418T182228Z&X-Amz-Expires=300&X-Amz-Signature=85098bc4bc106568a61bbc8d43517659e99f54919d3ca6ac414e43d772434528&X-Amz-SignedHeaders=host&actor_id=0&response-content-disposition=attachment%3B%20filename%3Dvsearch-1.11.1-osx-x86_64.tar.gz&response-content-type=application%2Foctet-stream [following]\n",
        "--2016-04-18 11:22:28--  https://github-cloud.s3.amazonaws.com/releases/19848353/5a974bd0-018b-11e6-810c-daa981b4e335.gz?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=AKIAISTNZFOVBIJMK3TQ%2F20160418%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20160418T182228Z&X-Amz-Expires=300&X-Amz-Signature=85098bc4bc106568a61bbc8d43517659e99f54919d3ca6ac414e43d772434528&X-Amz-SignedHeaders=host&actor_id=0&response-content-disposition=attachment%3B%20filename%3Dvsearch-1.11.1-osx-x86_64.tar.gz&response-content-type=application%2Foctet-stream\n",
        "Resolving github-cloud.s3.amazonaws.com... 54.231.81.72\n",
        "Connecting to github-cloud.s3.amazonaws.com|54.231.81.72|:443... connected.\n",
        "HTTP request sent, awaiting response... 200 OK\n",
        "Length: 274585 (268K) [application/octet-stream]\n",
        "Saving to: 'vsearch-1.11.1-osx-x86_64.tar.gz'\n",
        "\n",
        "     0K .......... .......... .......... .......... .......... 18%  367K 1s\n",
        "    50K .......... .......... .......... .......... .......... 37%  729K 0s\n",
        "   100K .......... .......... .......... .......... .......... 55%  368K 0s\n",
        "   150K .......... .......... .......... .......... .......... 74% 54.4M 0s\n",
        "   200K .......... .......... .......... .......... .......... 93%  744K 0s\n",
        "   250K .......... ........                                   100% 79.1M=0.4s\n",
        "\n",
        "2016-04-18 11:22:29 (656 KB/s) - 'vsearch-1.11.1-osx-x86_64.tar.gz' saved [274585/274585]\n",
        "\n"
       ]
      }
     ],
     "prompt_number": 12
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "%%bash\n",
      "tar -xzf vsearch-1.11.1-osx-x86_64.tar.gz"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 13
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "cd .."
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/usr/local/bin\n"
       ]
      }
     ],
     "prompt_number": 15
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "ls"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\u001b[35maclocal\u001b[m\u001b[m@                         \u001b[31mgenOSAuth\u001b[m\u001b[m*                       \u001b[31mimkdir\u001b[m\u001b[m*                          \u001b[31mitrim\u001b[m\u001b[m*                           runQuota.r\r\n",
        "\u001b[35maclocal-1.15\u001b[m\u001b[m@                    \u001b[31mgenomeCoverageBed\u001b[m\u001b[m*               \u001b[31mimv\u001b[m\u001b[m*                             \u001b[31miuserinfo\u001b[m\u001b[m*                       showCore.ir\r\n",
        "\u001b[31mannotateBed\u001b[m\u001b[m*                     \u001b[31mgetOverlap\u001b[m\u001b[m*                      \u001b[31mintersectBed\u001b[m\u001b[m*                    \u001b[31mixmsg\u001b[m\u001b[m*                           \u001b[31mshuffleBed\u001b[m\u001b[m*\r\n",
        "\u001b[35mautoconf\u001b[m\u001b[m@                        \u001b[31mgroupBy\u001b[m\u001b[m*                         \u001b[31mipasswd\u001b[m\u001b[m*                         \u001b[31mlinksBed\u001b[m\u001b[m*                        \u001b[31mslopBed\u001b[m\u001b[m*\r\n",
        "\u001b[35mautoheader\u001b[m\u001b[m@                      \u001b[31mhg\u001b[m\u001b[m*                              \u001b[31mipcluster\u001b[m\u001b[m*                       \u001b[35mlzcat\u001b[m\u001b[m@                           \u001b[31msortBed\u001b[m\u001b[m*\r\n",
        "\u001b[35mautom4te\u001b[m\u001b[m@                        \u001b[31miadmin\u001b[m\u001b[m*                          \u001b[31mipcontroller\u001b[m\u001b[m*                    \u001b[35mlzcmp\u001b[m\u001b[m@                           \u001b[31msubtractBed\u001b[m\u001b[m*\r\n",
        "\u001b[35mautomake\u001b[m\u001b[m@                        \u001b[31mibun\u001b[m\u001b[m*                            \u001b[31mipengine\u001b[m\u001b[m*                        \u001b[35mlzdiff\u001b[m\u001b[m@                          \u001b[31mtagBam\u001b[m\u001b[m*\r\n",
        "\u001b[35mautomake-1.15\u001b[m\u001b[m@                   \u001b[31micd\u001b[m\u001b[m*                             \u001b[31miphybun\u001b[m\u001b[m*                         \u001b[35mlzegrep\u001b[m\u001b[m@                         \u001b[31mtclsh8.6\u001b[m\u001b[m*\r\n",
        "\u001b[35mautoreconf\u001b[m\u001b[m@                      \u001b[31michksum\u001b[m\u001b[m*                         \u001b[31miphymv\u001b[m\u001b[m*                          \u001b[35mlzfgrep\u001b[m\u001b[m@                         \u001b[35mtwdiff\u001b[m\u001b[m@\r\n",
        "\u001b[35mautoscan\u001b[m\u001b[m@                        \u001b[31michmod\u001b[m\u001b[m*                          \u001b[31miplogger\u001b[m\u001b[m*                        \u001b[35mlzgrep\u001b[m\u001b[m@                          \u001b[35mtwfind\u001b[m\u001b[m@\r\n",
        "\u001b[35mautoupdate\u001b[m\u001b[m@                      \u001b[31micp\u001b[m\u001b[m*                             \u001b[31mips\u001b[m\u001b[m*                             \u001b[35mlzless\u001b[m\u001b[m@                          \u001b[31munionBedGraphs\u001b[m\u001b[m*\r\n",
        "\u001b[31mbamToBed\u001b[m\u001b[m*                        \u001b[31midbo\u001b[m\u001b[m*                            \u001b[31miptest\u001b[m\u001b[m*                          \u001b[35mlzma\u001b[m\u001b[m@                            \u001b[35munlzma\u001b[m\u001b[m@\r\n",
        "\u001b[31mbamToFastq\u001b[m\u001b[m*                      \u001b[31midbug\u001b[m\u001b[m*                           \u001b[31miput\u001b[m\u001b[m*                            \u001b[35mlzmadec\u001b[m\u001b[m@                         \u001b[35munxz\u001b[m\u001b[m@\r\n",
        "\u001b[31mbed12ToBed6\u001b[m\u001b[m*                     \u001b[31mienv\u001b[m\u001b[m*                            \u001b[31mipwd\u001b[m\u001b[m*                            \u001b[35mlzmainfo\u001b[m\u001b[m@                        v1.11.1.tar.gz\r\n",
        "\u001b[31mbedToBam\u001b[m\u001b[m*                        \u001b[31mierror\u001b[m\u001b[m*                          \u001b[31mipython\u001b[m\u001b[m*                         \u001b[35mlzmore\u001b[m\u001b[m@                          \u001b[31mvsearch\u001b[m\u001b[m*\r\n",
        "\u001b[31mbedToIgv\u001b[m\u001b[m*                        \u001b[31miexecmd\u001b[m\u001b[m*                         \u001b[31miqdel\u001b[m\u001b[m*                           \u001b[35mmakedepend\u001b[m\u001b[m@                      \u001b[34mvsearch-1.11.1\u001b[m\u001b[m/\r\n",
        "\u001b[31mbedpeToBam\u001b[m\u001b[m*                      \u001b[31miexit\u001b[m\u001b[m*                           \u001b[31miqmod\u001b[m\u001b[m*                           \u001b[31mmapBed\u001b[m\u001b[m*                          \u001b[35mwget\u001b[m\u001b[m@\r\n",
        "\u001b[31mbedtools\u001b[m\u001b[m*                        \u001b[35mifnames\u001b[m\u001b[m@                         \u001b[31miqstat\u001b[m\u001b[m*                          \u001b[31mmaskFastaFromBed\u001b[m\u001b[m*                \u001b[31mwindowBed\u001b[m\u001b[m*\r\n",
        "\u001b[31mbrew\u001b[m\u001b[m*                            \u001b[31mifsck\u001b[m\u001b[m*                           \u001b[31miquest\u001b[m\u001b[m*                          \u001b[31mmergeBed\u001b[m\u001b[m*                        \u001b[31mwindowMaker\u001b[m\u001b[m*\r\n",
        "chgCoreToCore1.ir                \u001b[31miget\u001b[m\u001b[m*                            \u001b[31miquota\u001b[m\u001b[m*                          \u001b[31mmultiBamCov\u001b[m\u001b[m*                     \u001b[31mwish8.6\u001b[m\u001b[m*\r\n",
        "chgCoreToCore2.ir                \u001b[31migetbyticket.pl\u001b[m\u001b[m*                 \u001b[31mireg\u001b[m\u001b[m*                            \u001b[31mmultiIntersectBed\u001b[m\u001b[m*               \u001b[35mxz\u001b[m\u001b[m@\r\n",
        "chgCoreToOrig.ir                 \u001b[31migetwild.sh\u001b[m\u001b[m*                     \u001b[31mirepl\u001b[m\u001b[m*                           \u001b[31mmuscle3.8.31_i86darwin64\u001b[m\u001b[m*        \u001b[35mxzcat\u001b[m\u001b[m@\r\n",
        "\u001b[31mclosestBed\u001b[m\u001b[m*                      \u001b[31migroupadmin\u001b[m\u001b[m*                     \u001b[31mirm\u001b[m\u001b[m*                             muscle3.8.31_i86darwin64.tar.gz  \u001b[35mxzcmp\u001b[m\u001b[m@\r\n",
        "\u001b[31mclusterBed\u001b[m\u001b[m*                      \u001b[31mihelp\u001b[m\u001b[m*                           \u001b[31mirmtrash\u001b[m\u001b[m*                        \u001b[31mnucBed\u001b[m\u001b[m*                          \u001b[35mxzdec\u001b[m\u001b[m@\r\n",
        "\u001b[31mcomplementBed\u001b[m\u001b[m*                   \u001b[31miinit\u001b[m\u001b[m*                           \u001b[34mirods3.2.icmds.mac.intel\u001b[m\u001b[m/        \u001b[31mpairToBed\u001b[m\u001b[m*                       \u001b[35mxzdiff\u001b[m\u001b[m@\r\n",
        "\u001b[31mcoverageBed\u001b[m\u001b[m*                     \u001b[31milocate\u001b[m\u001b[m*                         \u001b[31mirsync\u001b[m\u001b[m*                          \u001b[31mpairToPair\u001b[m\u001b[m*                      \u001b[35mxzegrep\u001b[m\u001b[m@\r\n",
        "delUnusedAVUs.ir                 \u001b[31mils\u001b[m\u001b[m*                             \u001b[31mirule\u001b[m\u001b[m*                           \u001b[35mpkg-config\u001b[m\u001b[m@                      \u001b[35mxzfgrep\u001b[m\u001b[m@\r\n",
        "\u001b[35medit\u001b[m\u001b[m@                            \u001b[31milsresc\u001b[m\u001b[m*                         \u001b[31mirunner\u001b[m\u001b[m*                         \u001b[31mpycolor\u001b[m\u001b[m*                         \u001b[35mxzgrep\u001b[m\u001b[m@\r\n",
        "\u001b[31mexpandCols\u001b[m\u001b[m*                      \u001b[31mimcoll\u001b[m\u001b[m*                          \u001b[31miscan\u001b[m\u001b[m*                           \u001b[34mpyrad-3.0.66\u001b[m\u001b[m/                    \u001b[35mxzless\u001b[m\u001b[m@\r\n",
        "\u001b[31mfastaFromBed\u001b[m\u001b[m*                    \u001b[31mimeta\u001b[m\u001b[m*                           \u001b[31misysmeta\u001b[m\u001b[m*                        \u001b[31mrandomBed\u001b[m\u001b[m*                       \u001b[35mxzmore\u001b[m\u001b[m@\r\n",
        "\u001b[31mflankBed\u001b[m\u001b[m*                        \u001b[31mimiscsvrinfo\u001b[m\u001b[m*                    \u001b[31miticket\u001b[m\u001b[m*                         runQuota.ir\r\n"
       ]
      }
     ],
     "prompt_number": 16
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "%%bash\n",
      "rm muscle3.8.31_i86darwin64.tar.gz v1.11.1.tar.gz"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 17
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "%%bash\n",
      "git clone https://github.com/xflouris/PEAR.git"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "Cloning into 'PEAR'...\n"
       ]
      }
     ],
     "prompt_number": 18
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "cd PEAR/"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/usr/local/bin/PEAR\n"
       ]
      }
     ],
     "prompt_number": 19
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "%%bash\n",
      "./autogen.sh"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stderr",
       "text": [
        "configure.ac:15: installing './compile'\n",
        "configure.ac:6: installing './install-sh'\n",
        "configure.ac:6: installing './missing'\n",
        "src/Makefile.am: installing './depcomp'\n"
       ]
      }
     ],
     "prompt_number": 20
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "%%bash\n",
      "./configure"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "checking for a BSD-compatible install... /usr/bin/install -c\n",
        "checking whether build environment is sane... yes\n",
        "checking for a thread-safe mkdir -p... ./install-sh -c -d\n",
        "checking for gawk... no\n",
        "checking for mawk... no\n",
        "checking for nawk... no\n",
        "checking for awk... awk\n",
        "checking whether make sets $(MAKE)... yes\n",
        "checking whether make supports nested variables... yes\n",
        "checking for gcc... gcc\n",
        "checking whether the C compiler works... yes\n",
        "checking for C compiler default output file name... a.out\n",
        "checking for suffix of executables... \n",
        "checking whether we are cross compiling... no\n",
        "checking for suffix of object files... o\n",
        "checking whether we are using the GNU C compiler... yes\n",
        "checking whether gcc accepts -g... yes\n",
        "checking for gcc option to accept ISO C89... none needed\n",
        "checking whether gcc understands -c and -o together... yes\n",
        "checking for style of include used by make... GNU\n",
        "checking dependency style of gcc... gcc3\n",
        "checking for log in -lm... yes\n",
        "checking for main in -lpthread... yes\n",
        "checking how to run the C preprocessor... gcc -E\n",
        "checking for grep that handles long lines and -e... /usr/bin/grep\n",
        "checking for egrep... /usr/bin/grep -E\n",
        "checking for ANSI C header files... yes\n",
        "checking for sys/types.h... yes\n",
        "checking for sys/stat.h... yes\n",
        "checking for stdlib.h... yes\n",
        "checking for string.h... yes\n",
        "checking for memory.h... yes\n",
        "checking for strings.h... yes\n",
        "checking for inttypes.h... yes\n",
        "checking for stdint.h... yes\n",
        "checking for unistd.h... yes\n",
        "checking for stdint.h... (cached) yes\n",
        "checking for stdlib.h... (cached) yes\n",
        "checking for string.h... (cached) yes\n",
        "checking for inline... inline\n",
        "checking for size_t... yes\n",
        "checking for uint32_t... yes\n",
        "checking for stdlib.h... (cached) yes\n",
        "checking for GNU libc compatible malloc... yes\n",
        "checking for working strtod... yes\n",
        "checking for memmove... yes\n",
        "checking for pow... yes\n",
        "checking for strtol... yes\n",
        "checking for library containing pthread_create... none required\n",
        "checking for library containing BZ2_bzCompress... -lbz2\n",
        "checking for library containing zlibVersion... -lz\n",
        "checking pthread.h usability... yes\n",
        "checking pthread.h presence... yes\n",
        "checking for pthread.h... yes\n",
        "checking bzlib.h usability... yes\n",
        "checking bzlib.h presence... yes\n",
        "checking for bzlib.h... yes\n",
        "checking zlib.h usability... yes\n",
        "checking zlib.h presence... yes\n",
        "checking for zlib.h... yes\n",
        "checking that generated files are newer than configure... done\n",
        "configure: creating ./config.status\n",
        "config.status: creating Makefile\n",
        "config.status: creating src/Makefile\n",
        "config.status: creating man/Makefile\n",
        "config.status: creating src/config.h\n",
        "config.status: executing depfiles commands\n"
       ]
      }
     ],
     "prompt_number": 21
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "%%bash\n",
      "make"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "Making all in src\n",
        "make  all-am\n",
        "gcc -DHAVE_CONFIG_H -I.    -O3 -fomit-frame-pointer -funroll-loops -Wall -Wredundant-decls  -Wreturn-type  -Wswitch-default -Wimplicit  -Wimplicit-function-declaration  -Wimplicit-int -Wimport -Wunused-label -Wno-int-to-pointer-cast -Wbad-function-cast  -Wmissing-declarations -Wmissing-prototypes  -Wnested-externs  -Wold-style-definition -Wstrict-prototypes -Wpointer-sign -Wextra -Wredundant-decls -Wunused -Wunused-function -Wunused-parameter -Wunused-value  -Wunused-variable -Wformat  -Wformat-nonliteral -Wparentheses -Wsequence-point -Wuninitialized -Wundef -O3 -MT pear-pear-pt.o -MD -MP -MF .deps/pear-pear-pt.Tpo -c -o pear-pear-pt.o `test -f 'pear-pt.c' || echo './'`pear-pt.c\n",
        "mv -f .deps/pear-pear-pt.Tpo .deps/pear-pear-pt.Po\n",
        "gcc -DHAVE_CONFIG_H -I.    -O3 -fomit-frame-pointer -funroll-loops -Wall -Wredundant-decls  -Wreturn-type  -Wswitch-default -Wimplicit  -Wimplicit-function-declaration  -Wimplicit-int -Wimport -Wunused-label -Wno-int-to-pointer-cast -Wbad-function-cast  -Wmissing-declarations -Wmissing-prototypes  -Wnested-externs  -Wold-style-definition -Wstrict-prototypes -Wpointer-sign -Wextra -Wredundant-decls -Wunused -Wunused-function -Wunused-parameter -Wunused-value  -Wunused-variable -Wformat  -Wformat-nonliteral -Wparentheses -Wsequence-point -Wuninitialized -Wundef -O3 -MT pear-args.o -MD -MP -MF .deps/pear-args.Tpo -c -o pear-args.o `test -f 'args.c' || echo './'`args.c\n",
        "mv -f .deps/pear-args.Tpo .deps/pear-args.Po\n",
        "gcc -DHAVE_CONFIG_H -I.    -O3 -fomit-frame-pointer -funroll-loops -Wall -Wredundant-decls  -Wreturn-type  -Wswitch-default -Wimplicit  -Wimplicit-function-declaration  -Wimplicit-int -Wimport -Wunused-label -Wno-int-to-pointer-cast -Wbad-function-cast  -Wmissing-declarations -Wmissing-prototypes  -Wnested-externs  -Wold-style-definition -Wstrict-prototypes -Wpointer-sign -Wextra -Wredundant-decls -Wunused -Wunused-function -Wunused-parameter -Wunused-value  -Wunused-variable -Wformat  -Wformat-nonliteral -Wparentheses -Wsequence-point -Wuninitialized -Wundef -O3 -MT pear-statistics.o -MD -MP -MF .deps/pear-statistics.Tpo -c -o pear-statistics.o `test -f 'statistics.c' || echo './'`statistics.c\n",
        "mv -f .deps/pear-statistics.Tpo .deps/pear-statistics.Po\n",
        "gcc -DHAVE_CONFIG_H -I.    -O3 -fomit-frame-pointer -funroll-loops -Wall -Wredundant-decls  -Wreturn-type  -Wswitch-default -Wimplicit  -Wimplicit-function-declaration  -Wimplicit-int -Wimport -Wunused-label -Wno-int-to-pointer-cast -Wbad-function-cast  -Wmissing-declarations -Wmissing-prototypes  -Wnested-externs  -Wold-style-definition -Wstrict-prototypes -Wpointer-sign -Wextra -Wredundant-decls -Wunused -Wunused-function -Wunused-parameter -Wunused-value  -Wunused-variable -Wformat  -Wformat-nonliteral -Wparentheses -Wsequence-point -Wuninitialized -Wundef -O3 -MT pear-reader.o -MD -MP -MF .deps/pear-reader.Tpo -c -o pear-reader.o `test -f 'reader.c' || echo './'`reader.c\n",
        "mv -f .deps/pear-reader.Tpo .deps/pear-reader.Po\n",
        "gcc -O3 -fomit-frame-pointer -funroll-loops -Wall -Wredundant-decls  -Wreturn-type  -Wswitch-default -Wimplicit  -Wimplicit-function-declaration  -Wimplicit-int -Wimport -Wunused-label -Wno-int-to-pointer-cast -Wbad-function-cast  -Wmissing-declarations -Wmissing-prototypes  -Wnested-externs  -Wold-style-definition -Wstrict-prototypes -Wpointer-sign -Wextra -Wredundant-decls -Wunused -Wunused-function -Wunused-parameter -Wunused-value  -Wunused-variable -Wformat  -Wformat-nonliteral -Wparentheses -Wsequence-point -Wuninitialized -Wundef -O3   -o pear pear-pear-pt.o pear-args.o pear-statistics.o pear-reader.o  -lz -lbz2 -lpthread -lm \n",
        "Making all in man\n",
        "make[1]: Nothing to be done for `all'.\n",
        "make[1]: Nothing to be done for `all-am'.\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stderr",
       "text": [
        "reader.c:967: warning: \u2018do_cpuid\u2019 defined but not used\n"
       ]
      }
     ],
     "prompt_number": 22
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "%%bash\n",
      "make install"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "Making install in src\n",
        " .././install-sh -c -d '/usr/local/bin'\n",
        "  /usr/bin/install -c pear '/usr/local/bin'\n",
        "make[2]: Nothing to be done for `install-data-am'.\n",
        "Making install in man\n",
        "make[2]: Nothing to be done for `install-exec-am'.\n",
        " .././install-sh -c -d '/usr/local/share/man/man1'\n",
        " /usr/bin/install -c -m 644 pear.1 '/usr/local/share/man/man1'\n",
        "make[2]: Nothing to be done for `install-exec-am'.\n",
        "make[2]: Nothing to be done for `install-data-am'.\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stderr",
       "text": [
        "install: /usr/local/bin/pear: Inappropriate file type or format\n"
       ]
      }
     ],
     "prompt_number": 23
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "cd"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/Users/Sam\n"
       ]
      }
     ],
     "prompt_number": 24
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "%%bash\n",
      "mkdir -p analysis_pyrad"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 25
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "%%bash\n",
      "pyrad -n"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stderr",
       "text": [
        "\tnew params.txt file created\n"
       ]
      }
     ],
     "prompt_number": 26
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "cat params.txt"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "==** parameter inputs for pyRAD version 3.0.66  **======================== affected step ==\r\n",
        "./                        ## 1. Working directory                                 (all)\r\n",
        "./*.fastq.gz              ## 2. Loc. of non-demultiplexed files (if not line 18)  (s1)\r\n",
        "./*.barcodes              ## 3. Loc. of barcode file (if not line 18)             (s1)\r\n",
        "vsearch                   ## 4. command (or path) to call vsearch (or usearch)    (s3,s6)\r\n",
        "muscle                    ## 5. command (or path) to call muscle                  (s3,s7)\r\n",
        "TGCAG                     ## 6. Restriction overhang (e.g., C|TGCAG -> TGCAG)     (s1,s2)\r\n",
        "2                         ## 7. N processors (parallel)                           (all)\r\n",
        "6                         ## 8. Mindepth: min coverage for a cluster              (s4,s5)\r\n",
        "4                         ## 9. NQual: max # sites with qual < 20 (or see line 20)(s2)\r\n",
        ".88                       ## 10. Wclust: clustering threshold as a decimal        (s3,s6)\r\n",
        "rad                       ## 11. Datatype: rad,gbs,pairgbs,pairddrad,(others:see docs)(all)\r\n",
        "4                         ## 12. MinCov: min samples in a final locus             (s7)\r\n",
        "3                         ## 13. MaxSH: max inds with shared hetero site          (s7)\r\n",
        "c88d6m4p3                 ## 14. Prefix name for final output (no spaces)         (s7)\r\n",
        "==== optional params below this line ===================================  affected step ==\r\n",
        "                       ## 15.opt.: select subset (prefix* only selector)            (s2-s7)\r\n",
        "                       ## 16.opt.: add-on (outgroup) taxa (list or prefix*)         (s6,s7)\r\n",
        "                       ## 17.opt.: exclude taxa (list or prefix*)                   (s7)\r\n",
        "                       ## 18.opt.: loc. of de-multiplexed data                      (s2)\r\n",
        "                       ## 19.opt.: maxM: N mismatches in barcodes (def= 1)          (s1)\r\n",
        "                       ## 20.opt.: phred Qscore offset (def= 33)                    (s2)\r\n",
        "                       ## 21.opt.: filter: def=0=NQual 1=NQual+adapters. 2=strict   (s2)\r\n",
        "                       ## 22.opt.: a priori E,H (def= 0.001,0.01, if not estimated) (s5)\r\n",
        "                       ## 23.opt.: maxN: max Ns in a cons seq (def=5)               (s5)\r\n",
        "                       ## 24.opt.: maxH: max heterozyg. sites in cons seq (def=5)   (s5)\r\n",
        "                       ## 25.opt.: ploidy: max alleles in cons seq (def=2;see docs) (s4,s5)\r\n",
        "                       ## 26.opt.: maxSNPs: (def=100). Paired (def=100,100)         (s7)\r\n",
        "                       ## 27.opt.: maxIndels: within-clust,across-clust (def. 3,99) (s3,s7)\r\n",
        "                       ## 28.opt.: random number seed (def. 112233)              (s3,s6,s7)\r\n",
        "                       ## 29.opt.: trim overhang left,right on final loci, def(0,0) (s7)\r\n",
        "                       ## 30.opt.: output formats: p,n,a,s,v,u,t,m,k,g,* (see docs) (s7)\r\n",
        "                       ## 31.opt.: maj. base call at depth>x<mindepth (def.x=mindepth) (s5)\r\n",
        "                       ## 32.opt.: keep trimmed reads (def=0). Enter min length.    (s2)\r\n",
        "                       ## 33.opt.: max stack size (int), def= max(500,mean+2*SD)    (s3)\r\n",
        "                       ## 34.opt.: minDerep: exclude dereps with <= N copies, def=1 (s3)\r\n",
        "                       ## 35.opt.: use hierarchical clustering (def.=0, 1=yes)      (s6)\r\n",
        "                       ## 36.opt.: repeat masking (def.=1='dust' method, 0=no)      (s3,s6)\r\n",
        "                       ## 37.opt.: vsearch max threads per job (def.=6; see docs)   (s3,s6)\r\n",
        "==== optional: list group/clade assignments below this line (see docs) ==================\r\n"
       ]
      }
     ],
     "prompt_number": 27
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "ls"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "3.0.66.tar.gz           \u001b[34mDocuments\u001b[m\u001b[m/              \u001b[34mLibrary\u001b[m\u001b[m/                \u001b[34mMusic\u001b[m\u001b[m/                  \u001b[34mPictures\u001b[m\u001b[m/               Untitled0.ipynb         params.txt\r\n",
        "\u001b[34mDesktop\u001b[m\u001b[m/                \u001b[34mDownloads\u001b[m\u001b[m/              \u001b[34mMovies\u001b[m\u001b[m/                 PE-GBS_empirical.ipynb  \u001b[34mPublic\u001b[m\u001b[m/                 \u001b[34manalysis_pyrad\u001b[m\u001b[m/\r\n"
       ]
      }
     ],
     "prompt_number": 28
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "%%bash\n",
      "mkdir oly_pyrad"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 29
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "cp params.txt oly_pyrad/"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 30
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "cp -r analysis_pyrad/ oly_pyrad/"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 32
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "cd oly_pyrad/"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/Users/Sam/oly_pyrad\n"
       ]
      }
     ],
     "prompt_number": 33
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "%%bash\n",
      "mkdir fastq"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 34
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "cd fastq/"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/Users/Sam/oly_pyrad/fastq\n"
       ]
      }
     ],
     "prompt_number": 35
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "%%bash\n",
      "time cp /Volumes/web/nightingales/O_lurida/20160223_gbs/*_{1..10}A_*.fq.gz ."
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stderr",
       "text": [
        "\n",
        "real\t4m33.626s\n",
        "user\t0m0.024s\n",
        "sys\t0m14.946s\n"
       ]
      }
     ],
     "prompt_number": 37
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "ls "
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\u001b[31m1HL_10A_1.fq.gz\u001b[m\u001b[m* \u001b[31m1HL_3A_1.fq.gz\u001b[m\u001b[m*  \u001b[31m1HL_6A_1.fq.gz\u001b[m\u001b[m*  \u001b[31m1HL_9A_1.fq.gz\u001b[m\u001b[m*  \u001b[31m1NF_2A_1.fq.gz\u001b[m\u001b[m*  \u001b[31m1NF_6A_1.fq.gz\u001b[m\u001b[m*  \u001b[31m1NF_9A_1.fq.gz\u001b[m\u001b[m*  \u001b[31m1SN_2A_1.fq.gz\u001b[m\u001b[m*  \u001b[31m1SN_5A_1.fq.gz\u001b[m\u001b[m*  \u001b[31m1SN_8A_1.fq.gz\u001b[m\u001b[m*\r\n",
        "\u001b[31m1HL_10A_2.fq.gz\u001b[m\u001b[m* \u001b[31m1HL_3A_2.fq.gz\u001b[m\u001b[m*  \u001b[31m1HL_6A_2.fq.gz\u001b[m\u001b[m*  \u001b[31m1HL_9A_2.fq.gz\u001b[m\u001b[m*  \u001b[31m1NF_2A_2.fq.gz\u001b[m\u001b[m*  \u001b[31m1NF_6A_2.fq.gz\u001b[m\u001b[m*  \u001b[31m1NF_9A_2.fq.gz\u001b[m\u001b[m*  \u001b[31m1SN_2A_2.fq.gz\u001b[m\u001b[m*  \u001b[31m1SN_5A_2.fq.gz\u001b[m\u001b[m*  \u001b[31m1SN_8A_2.fq.gz\u001b[m\u001b[m*\r\n",
        "\u001b[31m1HL_1A_1.fq.gz\u001b[m\u001b[m*  \u001b[31m1HL_4A_1.fq.gz\u001b[m\u001b[m*  \u001b[31m1HL_7A_1.fq.gz\u001b[m\u001b[m*  \u001b[31m1NF_10A_1.fq.gz\u001b[m\u001b[m* \u001b[31m1NF_4A_1.fq.gz\u001b[m\u001b[m*  \u001b[31m1NF_7A_1.fq.gz\u001b[m\u001b[m*  \u001b[31m1SN_10A_1.fq.gz\u001b[m\u001b[m* \u001b[31m1SN_3A_1.fq.gz\u001b[m\u001b[m*  \u001b[31m1SN_6A_1.fq.gz\u001b[m\u001b[m*  \u001b[31m1SN_9A_1.fq.gz\u001b[m\u001b[m*\r\n",
        "\u001b[31m1HL_1A_2.fq.gz\u001b[m\u001b[m*  \u001b[31m1HL_4A_2.fq.gz\u001b[m\u001b[m*  \u001b[31m1HL_7A_2.fq.gz\u001b[m\u001b[m*  \u001b[31m1NF_10A_2.fq.gz\u001b[m\u001b[m* \u001b[31m1NF_4A_2.fq.gz\u001b[m\u001b[m*  \u001b[31m1NF_7A_2.fq.gz\u001b[m\u001b[m*  \u001b[31m1SN_10A_2.fq.gz\u001b[m\u001b[m* \u001b[31m1SN_3A_2.fq.gz\u001b[m\u001b[m*  \u001b[31m1SN_6A_2.fq.gz\u001b[m\u001b[m*  \u001b[31m1SN_9A_2.fq.gz\u001b[m\u001b[m*\r\n",
        "\u001b[31m1HL_2A_1.fq.gz\u001b[m\u001b[m*  \u001b[31m1HL_5A_1.fq.gz\u001b[m\u001b[m*  \u001b[31m1HL_8A_1.fq.gz\u001b[m\u001b[m*  \u001b[31m1NF_1A_1.fq.gz\u001b[m\u001b[m*  \u001b[31m1NF_5A_1.fq.gz\u001b[m\u001b[m*  \u001b[31m1NF_8A_1.fq.gz\u001b[m\u001b[m*  \u001b[31m1SN_1A_1.fq.gz\u001b[m\u001b[m*  \u001b[31m1SN_4A_1.fq.gz\u001b[m\u001b[m*  \u001b[31m1SN_7A_1.fq.gz\u001b[m\u001b[m*\r\n",
        "\u001b[31m1HL_2A_2.fq.gz\u001b[m\u001b[m*  \u001b[31m1HL_5A_2.fq.gz\u001b[m\u001b[m*  \u001b[31m1HL_8A_2.fq.gz\u001b[m\u001b[m*  \u001b[31m1NF_1A_2.fq.gz\u001b[m\u001b[m*  \u001b[31m1NF_5A_2.fq.gz\u001b[m\u001b[m*  \u001b[31m1NF_8A_2.fq.gz\u001b[m\u001b[m*  \u001b[31m1SN_1A_2.fq.gz\u001b[m\u001b[m*  \u001b[31m1SN_4A_2.fq.gz\u001b[m\u001b[m*  \u001b[31m1SN_7A_2.fq.gz\u001b[m\u001b[m*\r\n"
       ]
      }
     ],
     "prompt_number": 38
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "cd .."
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/Users/Sam/oly_pyrad\n"
       ]
      }
     ],
     "prompt_number": 39
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "cp /Users/Sam/Downloads/params.txt ."
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 40
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "cat params.txt"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "==** parameter inputs for pyRAD version 3.0.66  **======================== affected step ==\r\n",
        "./                     ## 1. Working directory                                 (all)\r\n",
        "             \t       ## 2. Loc. of non-demultiplexed files (if not line 18)  (s1)\r\n",
        "\t               ## 3. Loc. of barcode file (if not line 18)             (s1)\r\n",
        "/usr/local/bin/vsearch-1.11.1/bin/vsearch                ## 4. command (or path) to call vsearch (or usearch)    (s3,s6)\r\n",
        "/usr/local/bin/muscle3.8.31_i86darwin64                    ## 5. command (or path) to call muscle                  (s3,s7)\r\n",
        "CWGC                   ## 6. Restriction overhang (e.g., C|TGCAG -> TGCAG)     (s1,s2)\r\n",
        "16                     ## 7. N processors (parallel)                           (all)\r\n",
        "6                      ## 8. Mindepth: min coverage for a cluster              (s4,s5)\r\n",
        "4                      ## 9. NQual: max # sites with qual < 20 (or see line 20)(s2)\r\n",
        ".88                    ## 10. Wclust: clustering threshold as a decimal        (s3,s6)\r\n",
        "merged                 ## 11. Datatype: rad,gbs,pairgbs,pairddrad,(others:see docs)(all)\r\n",
        "4                      ## 12. MinCov: min samples in a final locus             (s7)\r\n",
        "3                      ## 13. MaxSH: max inds with shared hetero site          (s7)\r\n",
        "oly_gbs_pyrad          ## 14. Prefix name for final output (no spaces)         (s7)\r\n",
        "==== optional params below this line ===================================  affected step ==\r\n",
        "                       ## 15.opt.: select subset (prefix* only selector)            (s2-s7)\r\n",
        "                       ## 16.opt.: add-on (outgroup) taxa (list or prefix*)         (s6,s7)\r\n",
        "                       ## 17.opt.: exclude taxa (list or prefix*)                   (s7)\r\n",
        "/Users/Sam/oly_pyrad/analysis_pyrad/fastq/*.assembled.fastq                       ## 18.opt.: loc. of de-multiplexed data                      (s2)\r\n",
        "                       ## 19.opt.: maxM: N mismatches in barcodes (def= 1)          (s1)\r\n",
        "                       ## 20.opt.: phred Qscore offset (def= 33)                    (s2)\r\n",
        "2                      ## 21.opt.: filter: def=0=NQual 1=NQual+adapters. 2=strict   (s2)\r\n",
        "                       ## 22.opt.: a priori E,H (def= 0.001,0.01, if not estimated) (s5)\r\n",
        "4                      ## 23.opt.: maxN: max Ns in a cons seq (def=5)               (s5)\r\n",
        "8                      ## 24.opt.: maxH: max heterozyg. sites in cons seq (def=5)   (s5)\r\n",
        "                       ## 25.opt.: ploidy: max alleles in cons seq (def=2;see docs) (s4,s5)\r\n",
        "                       ## 26.opt.: maxSNPs: (def=100). Paired (def=100,100)         (s7)\r\n",
        "                       ## 27.opt.: maxIndels: within-clust,across-clust (def. 3,99) (s3,s7)\r\n",
        "                       ## 28.opt.: random number seed (def. 112233)              (s3,s6,s7)\r\n",
        "                       ## 29.opt.: trim overhang left,right on final loci, def(0,0) (s7)\r\n",
        "*                      ## 30.opt.: output formats: p,n,a,s,v,u,t,m,k,g,* (see docs) (s7)\r\n",
        "                       ## 31.opt.: maj. base call at depth>x<mindepth (def.x=mindepth) (s5)\r\n",
        "50                     ## 32.opt.: keep trimmed reads (def=0). Enter min length.    (s2)\r\n",
        "                       ## 33.opt.: max stack size (int), def= max(500,mean+2*SD)    (s3)\r\n",
        "                       ## 34.opt.: minDerep: exclude dereps with <= N copies, def=1 (s3)\r\n",
        "                       ## 35.opt.: use hierarchical clustering (def.=0, 1=yes)      (s6)\r\n",
        "                       ## 36.opt.: repeat masking (def.=1='dust' method, 0=no)      (s3,s6)\r\n",
        "16                     ## 37.opt.: vsearch max threads per job (def.=6; see docs)   (s3,s6)\r\n",
        "==== optional: list group/clade assignments below this line (see docs) ==================\r\n"
       ]
      }
     ],
     "prompt_number": 12
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "%%bash\n",
      "time gunzip analysis_pyrad/fastq/*.gz"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stderr",
       "text": [
        "\n",
        "real\t21m41.187s\n",
        "user\t4m30.902s\n",
        "sys\t0m28.895s\n"
       ]
      }
     ],
     "prompt_number": 44
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "ls analysis_pyrad/fastq/"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\u001b[31m1HL_10A_1.fq\u001b[m\u001b[m* \u001b[31m1HL_2A_2.fq\u001b[m\u001b[m*  \u001b[31m1HL_5A_1.fq\u001b[m\u001b[m*  \u001b[31m1HL_7A_2.fq\u001b[m\u001b[m*  \u001b[31m1NF_10A_1.fq\u001b[m\u001b[m* \u001b[31m1NF_2A_2.fq\u001b[m\u001b[m*  \u001b[31m1NF_6A_1.fq\u001b[m\u001b[m*  \u001b[31m1NF_8A_2.fq\u001b[m\u001b[m*  \u001b[31m1SN_1A_1.fq\u001b[m\u001b[m*  \u001b[31m1SN_3A_2.fq\u001b[m\u001b[m*  \u001b[31m1SN_6A_1.fq\u001b[m\u001b[m*  \u001b[31m1SN_8A_2.fq\u001b[m\u001b[m*\r\n",
        "\u001b[31m1HL_10A_2.fq\u001b[m\u001b[m* \u001b[31m1HL_3A_1.fq\u001b[m\u001b[m*  \u001b[31m1HL_5A_2.fq\u001b[m\u001b[m*  \u001b[31m1HL_8A_1.fq\u001b[m\u001b[m*  \u001b[31m1NF_10A_2.fq\u001b[m\u001b[m* \u001b[31m1NF_4A_1.fq\u001b[m\u001b[m*  \u001b[31m1NF_6A_2.fq\u001b[m\u001b[m*  \u001b[31m1NF_9A_1.fq\u001b[m\u001b[m*  \u001b[31m1SN_1A_2.fq\u001b[m\u001b[m*  \u001b[31m1SN_4A_1.fq\u001b[m\u001b[m*  \u001b[31m1SN_6A_2.fq\u001b[m\u001b[m*  \u001b[31m1SN_9A_1.fq\u001b[m\u001b[m*\r\n",
        "\u001b[31m1HL_1A_1.fq\u001b[m\u001b[m*  \u001b[31m1HL_3A_2.fq\u001b[m\u001b[m*  \u001b[31m1HL_6A_1.fq\u001b[m\u001b[m*  \u001b[31m1HL_8A_2.fq\u001b[m\u001b[m*  \u001b[31m1NF_1A_1.fq\u001b[m\u001b[m*  \u001b[31m1NF_4A_2.fq\u001b[m\u001b[m*  \u001b[31m1NF_7A_1.fq\u001b[m\u001b[m*  \u001b[31m1NF_9A_2.fq\u001b[m\u001b[m*  \u001b[31m1SN_2A_1.fq\u001b[m\u001b[m*  \u001b[31m1SN_4A_2.fq\u001b[m\u001b[m*  \u001b[31m1SN_7A_1.fq\u001b[m\u001b[m*  \u001b[31m1SN_9A_2.fq\u001b[m\u001b[m*\r\n",
        "\u001b[31m1HL_1A_2.fq\u001b[m\u001b[m*  \u001b[31m1HL_4A_1.fq\u001b[m\u001b[m*  \u001b[31m1HL_6A_2.fq\u001b[m\u001b[m*  \u001b[31m1HL_9A_1.fq\u001b[m\u001b[m*  \u001b[31m1NF_1A_2.fq\u001b[m\u001b[m*  \u001b[31m1NF_5A_1.fq\u001b[m\u001b[m*  \u001b[31m1NF_7A_2.fq\u001b[m\u001b[m*  \u001b[31m1SN_10A_1.fq\u001b[m\u001b[m* \u001b[31m1SN_2A_2.fq\u001b[m\u001b[m*  \u001b[31m1SN_5A_1.fq\u001b[m\u001b[m*  \u001b[31m1SN_7A_2.fq\u001b[m\u001b[m*\r\n",
        "\u001b[31m1HL_2A_1.fq\u001b[m\u001b[m*  \u001b[31m1HL_4A_2.fq\u001b[m\u001b[m*  \u001b[31m1HL_7A_1.fq\u001b[m\u001b[m*  \u001b[31m1HL_9A_2.fq\u001b[m\u001b[m*  \u001b[31m1NF_2A_1.fq\u001b[m\u001b[m*  \u001b[31m1NF_5A_2.fq\u001b[m\u001b[m*  \u001b[31m1NF_8A_1.fq\u001b[m\u001b[m*  \u001b[31m1SN_10A_2.fq\u001b[m\u001b[m* \u001b[31m1SN_3A_1.fq\u001b[m\u001b[m*  \u001b[31m1SN_5A_2.fq\u001b[m\u001b[m*  \u001b[31m1SN_8A_1.fq\u001b[m\u001b[m*\r\n"
       ]
      }
     ],
     "prompt_number": 45
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "cd ./oly_pyrad/"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/Users/Sam/oly_pyrad\n"
       ]
      }
     ],
     "prompt_number": 1
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "ls"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\u001b[34manalysis_pyrad\u001b[m\u001b[m/ params.txt      pear.log\r\n"
       ]
      }
     ],
     "prompt_number": 2
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "%%bash\n",
      "rm pear.log"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 4
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "ls"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\u001b[34manalysis_pyrad\u001b[m\u001b[m/ params.txt\r\n"
       ]
      }
     ],
     "prompt_number": 5
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "%%bash\n",
      "time for gfile in analysis_pyrad/fastq/*_1.fq;\n",
      "    do pear -f $gfile \\\n",
      "            -r ${gfile/_1.fq/_2.fq} \\\n",
      "            -o ${gfile/_1.fq/} \\\n",
      "            -n 33 \\\n",
      "            -t 33 \\\n",
      "            -q 10 \\\n",
      "            -j 20  >> pear.log 2>&1;\n",
      "done"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stderr",
       "text": [
        "\n",
        "real\t51m28.614s\n",
        "user\t310m59.181s\n",
        "sys\t1m40.621s\n"
       ]
      }
     ],
     "prompt_number": 7
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "%%bash\n",
      "time pyrad -p params.txt -s 2"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stderr",
       "text": [
        "\n",
        "\n",
        "     ------------------------------------------------------------\n",
        "      pyRAD : RADseq for phylogenetics & introgression analyses\n",
        "     ------------------------------------------------------------\n",
        "\n",
        "\tsorted .fastq from /Users/Sam/oly_pyrad/analysis_pyrad/fastq/*.assembled.fastq being used\n",
        "\tstep 2: editing raw reads \n",
        "\t.............................\n",
        "real\t25m24.605s\n",
        "user\t194m20.913s\n",
        "sys\t0m48.644s\n"
       ]
      }
     ],
     "prompt_number": 11
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "pwd"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 14,
       "text": [
        "u'/Users/Sam/oly_pyrad'"
       ]
      }
     ],
     "prompt_number": 14
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "%%bash\n",
      "cat stats/s2.rawedit.txt"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "sample \tNreads\tpassed\tpassed.w.trim\tpassed.total\n",
        "1HL_10A.assembled\t1820770\t1391336\t199641\t1590977\n",
        "1HL_1A.assembled\t1272730\t983255\t135282\t1118537\n",
        "1HL_2A.assembled\t1601871\t1230522\t175952\t1406474\n",
        "1HL_3A.assembled\t1249829\t959096\t136975\t1096071\n",
        "1HL_4A.assembled\t1467079\t1127114\t160413\t1287527\n",
        "1HL_5A.assembled\t1108322\t854805\t118424\t973229\n",
        "1HL_6A.assembled\t1120919\t862095\t123224\t985319\n",
        "1HL_7A.assembled\t1555328\t1126212\t243812\t1370024\n",
        "1HL_8A.assembled\t1390494\t1065503\t151199\t1216702\n",
        "1HL_9A.assembled\t1602480\t1226829\t176983\t1403812\n",
        "1NF_10A.assembled\t1187014\t907067\t134206\t1041273\n",
        "1NF_1A.assembled\t727077\t554075\t84411\t638486\n",
        "1NF_2A.assembled\t1946720\t1482437\t222755\t1705192\n",
        "1NF_4A.assembled\t2025316\t1544728\t231490\t1776218\n",
        "1NF_5A.assembled\t1517299\t1153759\t177977\t1331736\n",
        "1NF_6A.assembled\t1356495\t1038280\t152972\t1191252\n",
        "1NF_7A.assembled\t1458739\t1118102\t163618\t1281720\n",
        "1NF_8A.assembled\t1526495\t1163697\t175955\t1339652\n",
        "1NF_9A.assembled\t1896089\t1448815\t216321\t1665136\n",
        "1SN_10A.assembled\t1461469\t1127065\t157990\t1285055\n",
        "1SN_1A.assembled\t1689935\t1295083\t190364\t1485447\n",
        "1SN_2A.assembled\t1748906\t1352061\t188942\t1541003\n",
        "1SN_3A.assembled\t1340465\t1029190\t148242\t1177432\n",
        "1SN_4A.assembled\t1918135\t1462860\t217735\t1680595\n",
        "1SN_5A.assembled\t1726528\t1321611\t191401\t1513012\n",
        "1SN_6A.assembled\t1920845\t1478193\t211244\t1689437\n",
        "1SN_7A.assembled\t1694825\t1293688\t189713\t1483401\n",
        "1SN_8A.assembled\t1855879\t1421110\t206644\t1627754\n",
        "1SN_9A.assembled\t1829438\t1396438\t206920\t1603358\n",
        "\n",
        "    Nreads = total number of reads for a sample\n",
        "    passed = retained reads that passed quality filtering at full length\n",
        "    passed.w.trim= retained reads that were trimmed due to detection of adapters\n",
        "    passed.total  = total kept reads of sufficient length\n",
        "    note: you can set the option in params file to include trimmed reads of xx length. \n"
       ]
      }
     ],
     "prompt_number": 15
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "cat params.txt"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "==** parameter inputs for pyRAD version 3.0.66  **======================== affected step ==\r\n",
        "./                     ## 1. Working directory                                 (all)\r\n",
        "             \t       ## 2. Loc. of non-demultiplexed files (if not line 18)  (s1)\r\n",
        "\t               ## 3. Loc. of barcode file (if not line 18)             (s1)\r\n",
        "/usr/local/bin/vsearch-1.11.1/bin/vsearch                ## 4. command (or path) to call vsearch (or usearch)    (s3,s6)\r\n",
        "/usr/local/bin/muscle3.8.31_i86darwin64                    ## 5. command (or path) to call muscle                  (s3,s7)\r\n",
        "CWGC                   ## 6. Restriction overhang (e.g., C|TGCAG -> TGCAG)     (s1,s2)\r\n",
        "16                     ## 7. N processors (parallel)                           (all)\r\n",
        "6                      ## 8. Mindepth: min coverage for a cluster              (s4,s5)\r\n",
        "4                      ## 9. NQual: max # sites with qual < 20 (or see line 20)(s2)\r\n",
        ".88                    ## 10. Wclust: clustering threshold as a decimal        (s3,s6)\r\n",
        "merged                 ## 11. Datatype: rad,gbs,pairgbs,pairddrad,(others:see docs)(all)\r\n",
        "4                      ## 12. MinCov: min samples in a final locus             (s7)\r\n",
        "3                      ## 13. MaxSH: max inds with shared hetero site          (s7)\r\n",
        "oly_gbs_pyrad          ## 14. Prefix name for final output (no spaces)         (s7)\r\n",
        "==== optional params below this line ===================================  affected step ==\r\n",
        "                       ## 15.opt.: select subset (prefix* only selector)            (s2-s7)\r\n",
        "                       ## 16.opt.: add-on (outgroup) taxa (list or prefix*)         (s6,s7)\r\n",
        "                       ## 17.opt.: exclude taxa (list or prefix*)                   (s7)\r\n",
        "/Users/Sam/oly_pyrad/analysis_pyrad/fastq/*.assembled.fastq                       ## 18.opt.: loc. of de-multiplexed data                      (s2)\r\n",
        "                       ## 19.opt.: maxM: N mismatches in barcodes (def= 1)          (s1)\r\n",
        "                       ## 20.opt.: phred Qscore offset (def= 33)                    (s2)\r\n",
        "2                      ## 21.opt.: filter: def=0=NQual 1=NQual+adapters. 2=strict   (s2)\r\n",
        "                       ## 22.opt.: a priori E,H (def= 0.001,0.01, if not estimated) (s5)\r\n",
        "4                      ## 23.opt.: maxN: max Ns in a cons seq (def=5)               (s5)\r\n",
        "8                      ## 24.opt.: maxH: max heterozyg. sites in cons seq (def=5)   (s5)\r\n",
        "                       ## 25.opt.: ploidy: max alleles in cons seq (def=2;see docs) (s4,s5)\r\n",
        "                       ## 26.opt.: maxSNPs: (def=100). Paired (def=100,100)         (s7)\r\n",
        "                       ## 27.opt.: maxIndels: within-clust,across-clust (def. 3,99) (s3,s7)\r\n",
        "                       ## 28.opt.: random number seed (def. 112233)              (s3,s6,s7)\r\n",
        "                       ## 29.opt.: trim overhang left,right on final loci, def(0,0) (s7)\r\n",
        "*                      ## 30.opt.: output formats: p,n,a,s,v,u,t,m,k,g,* (see docs) (s7)\r\n",
        "                       ## 31.opt.: maj. base call at depth>x<mindepth (def.x=mindepth) (s5)\r\n",
        "50                     ## 32.opt.: keep trimmed reads (def=0). Enter min length.    (s2)\r\n",
        "                       ## 33.opt.: max stack size (int), def= max(500,mean+2*SD)    (s3)\r\n",
        "                       ## 34.opt.: minDerep: exclude dereps with <= N copies, def=1 (s3)\r\n",
        "                       ## 35.opt.: use hierarchical clustering (def.=0, 1=yes)      (s6)\r\n",
        "                       ## 36.opt.: repeat masking (def.=1='dust' method, 0=no)      (s3,s6)\r\n",
        "16                     ## 37.opt.: vsearch max threads per job (def.=6; see docs)   (s3,s6)\r\n",
        "==== optional: list group/clade assignments below this line (see docs) ==================\r\n",
        "1HL\t5\t1HL*\r\n",
        "1NF\t5\t1NF*\r\n",
        "1SN\t5\t1SN*\r\n"
       ]
      }
     ],
     "prompt_number": 17
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "%%bash\n",
      "time pyrad -p params.txt -s 3"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stderr",
       "text": [
        "\n",
        "\n",
        "     ------------------------------------------------------------\n",
        "      pyRAD : RADseq for phylogenetics & introgression analyses\n",
        "     ------------------------------------------------------------\n",
        "\n",
        "\n",
        "\tde-replicating files for clustering...\n",
        "\n",
        "\tstep 3: within-sample clustering of 29 samples at \n",
        "\t        '.88' similarity. Running 16 parallel jobs\n",
        "\t \twith up to 16 threads per job. If needed, \n",
        "\t\tadjust to avoid CPU and MEM limits\n",
        "\n",
        "\tsample 1HL_7A finished, 93536 loci\n",
        "\tsample 1HL_9A finished, 99179 loci\n",
        "\tsample 1HL_2A finished, 84916 loci\n",
        "\tsample 1NF_8A finished, 88468 loci\n",
        "\tsample 1SN_5A finished, 106967 loci\n",
        "\tsample 1SN_7A finished, 96438 loci\n",
        "\tsample 1SN_1A finished, 86975 loci\n",
        "\tsample 1HL_10A finished, 96319 loci\n",
        "\tsample 1SN_8A finished, 93143 loci\n",
        "\tsample 1SN_9A finished, 110001 loci\n",
        "\tsample 1SN_2A finished, 95605 loci\n",
        "\tsample 1SN_4A finished, 99963 loci\n",
        "\tsample 1NF_9A finished, 94576 loci\n",
        "\tsample 1SN_6A finished, 97635 loci\n",
        "\tsample 1NF_2A finished, 93430 loci\n",
        "\tsample 1NF_4A finished, 100841 loci\n",
        "\tsample 1NF_1A finished, 54628 loci\n",
        "\tsample 1NF_5A finished, 88013 loci\n",
        "Process Worker-7:\n",
        "Traceback (most recent call last):\n",
        "  File \"/usr/local/bioinformatics/anaconda/lib/python2.7/multiprocessing/process.py\", line 258, in _bootstrap\n",
        "    self.run()\n",
        "  File \"build/bdist.macosx-10.5-x86_64/egg/pyrad/potpour.py\", line 31, in run\n",
        "    res = self.func(*job)\n",
        "  File \"build/bdist.macosx-10.5-x86_64/egg/pyrad/cluster7dp.py\", line 545, in final\n",
        "    alignwrap(outfolder+\"/\"+handle.split(\"/\")[-1].replace(\".edit\",\".clust.gz\"), mindepth, muscle, w1)\n",
        "  File \"build/bdist.macosx-10.5-x86_64/egg/pyrad/cluster7dp.py\", line 432, in alignwrap\n",
        "    keys.sort(key=lambda x:int(x.split(\";\")[1].replace(\"size=\",\"\")), reverse=True)\n",
        "  File \"build/bdist.macosx-10.5-x86_64/egg/pyrad/cluster7dp.py\", line 432, in <lambda>\n",
        "    keys.sort(key=lambda x:int(x.split(\";\")[1].replace(\"size=\",\"\")), reverse=True)\n",
        "IndexError: list index out of range\n",
        "\tsample 1HL_5A finished, 78327 loci\n",
        "\tsample 1HL_3A finished, 96685 loci\n",
        "\tsample 1HL_6A finished, 77553 loci\n",
        "\tsample 1NF_7A finished, 85566 loci\n",
        "\tsample 1SN_10A finished, 94779 loci\n",
        "\tsample 1HL_1A finished, 81103 loci\n",
        "\tsample 1HL_8A finished, 80377 loci\n",
        "\tsample 1NF_6A finished, 83802 loci\n",
        "\tsample 1HL_4A finished, 84504 loci\n",
        "\tsample 1SN_3A finished, 106541 loci\n",
        "\n",
        "real\t250m0.701s\n",
        "user\t1679m50.366s\n",
        "sys\t194m22.084s\n"
       ]
      }
     ],
     "prompt_number": 18
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "cat stats/s3.clusters.txt"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\r\n",
        "taxa\ttotal\tdpt.me\tdpt.sd\td>5.tot\td>5.me\td>5.sd\tbadpairs\r\n",
        "\r\n",
        "    ## total = total number of clusters, including singletons\r\n",
        "    ## dpt.me = mean depth of clusters\r\n",
        "    ## dpt.sd = standard deviation of cluster depth\r\n",
        "    ## >N.tot = number of clusters with depth greater than N\r\n",
        "    ## >N.me = mean depth of clusters with depth greater than N\r\n",
        "    ## >N.sd = standard deviation of cluster depth for clusters with depth greater than N\r\n",
        "    ## badpairs = mismatched 1st & 2nd reads (only for paired ddRAD data)\r\n",
        "\r\n",
        "HISTOGRAMS\r\n",
        "\r\n",
        "    \r\n",
        "\r\n",
        "taxa\ttotal\tdpt.me\tdpt.sd\td>5.tot\td>5.me\td>5.sd\tbadpairs\r\n",
        "1HL_10A.assembled\t96319\t8.175\t26.643\t27546\t24.034\t46.113\t0\r\n",
        "1HL_1A.assembled\t81103\t6.915\t32.77\t21685\t20.681\t61.266\t0\r\n",
        "1HL_2A.assembled\t84916\t8.135\t26.405\t25225\t22.924\t45.083\t0\r\n",
        "1HL_3A.assembled\t96685\t5.794\t21.024\t21927\t19.527\t41.239\t0\r\n",
        "1HL_4A.assembled\t84504\t7.699\t36.772\t24572\t21.916\t66.042\t0\r\n",
        "1HL_5A.assembled\t78327\t6.242\t24.341\t19956\t18.949\t45.874\t0\r\n",
        "1HL_6A.assembled\t77553\t6.448\t23.051\t19925\t19.647\t42.773\t0\r\n",
        "1HL_7A.assembled\t93536\t7.511\t35.323\t26658\t21.716\t63.97\t0\r\n",
        "1HL_8A.assembled\t80377\t7.702\t29.826\t22318\t22.847\t53.688\t0\r\n",
        "1HL_9A.assembled\t99179\t7.095\t27.517\t25142\t22.827\t51.495\t0\r\n",
        "1NF_1A.assembled\t54628\t5.986\t16.562\t13499\t18.349\t30.043\t0\r\n",
        "1NF_2A.assembled\t93430\t8.834\t32.408\t28560\t24.709\t55.405\t0\r\n",
        "1NF_4A.assembled\t100841\t8.704\t35.853\t30880\t24.282\t62.006\t0\r\n",
        "1NF_5A.assembled\t88013\t7.512\t25.075\t25169\t21.616\t43.778\t0\r\n",
        "1NF_6A.assembled\t83802\t7.107\t30.616\t22252\t21.568\t56.934\t0\r\n",
        "1NF_7A.assembled\t85566\t7.348\t25.729\t23063\t22.143\t46.393\t0\r\n",
        "1NF_8A.assembled\t88468\t7.492\t24.994\t24030\t22.604\t44.525\t0\r\n",
        "1NF_9A.assembled\t94576\t8.75\t42.972\t28263\t24.948\t76.169\t0\r\n",
        "1SN_10A.assembled\t94779\t6.814\t26.835\t23903\t21.522\t50.616\t0\r\n",
        "1SN_1A.assembled\t86975\t8.342\t31.219\t25283\t24.13\t54.753\t0\r\n",
        "1SN_2A.assembled\t95605\t7.934\t37.881\t26414\t23.834\t69.577\t0\r\n",
        "1SN_3A.assembled\t106541\t5.684\t24.182\t22272\t20.421\t50.179\t0\r\n",
        "1SN_4A.assembled\t99963\t8.253\t30.238\t29839\t23.322\t52.308\t0\r\n",
        "1SN_5A.assembled\t106967\t6.996\t22.104\t28188\t21.451\t39.58\t0\r\n",
        "1SN_6A.assembled\t97635\t8.526\t47.897\t29140\t24.224\t85.627\t0\r\n",
        "1SN_7A.assembled\t96438\t7.718\t29.465\t27143\t22.727\t52.608\t0\r\n",
        "1SN_8A.assembled\t93143\t8.68\t89.174\t28977\t23.765\t158.831\t0\r\n",
        "1SN_9A.assembled\t110001\t7.246\t27.658\t28911\t22.59\t50.867\t0\r\n",
        "\r\n",
        "    ## total = total number of clusters, including singletons\r\n",
        "    ## dpt.me = mean depth of clusters\r\n",
        "    ## dpt.sd = standard deviation of cluster depth\r\n",
        "    ## >N.tot = number of clusters with depth greater than N\r\n",
        "    ## >N.me = mean depth of clusters with depth greater than N\r\n",
        "    ## >N.sd = standard deviation of cluster depth for clusters with depth greater than N\r\n",
        "    ## badpairs = mismatched 1st & 2nd reads (only for paired ddRAD data)\r\n",
        "\r\n",
        "HISTOGRAMS\r\n",
        "\r\n",
        "    \r\n",
        "sample: 1HL_10A.assembled\r\n",
        "bins\tdepth_histogram\tcnts\r\n",
        "   :\t0------------50-------------100%\r\n",
        "0 \t*********************               65252\r\n",
        "5 \t*****                               12231\r\n",
        "10 \t***                                 5823\r\n",
        "15 \t**                                  3322\r\n",
        "20 \t**                                  2177\r\n",
        "25 \t*                                   1510\r\n",
        "30 \t*                                   1156\r\n",
        "35 \t*                                   841\r\n",
        "40 \t*                                   1380\r\n",
        "50 \t**                                  2101\r\n",
        "100 \t*                                   401\r\n",
        "250 \t*                                   83\r\n",
        "500 \t*                                   42\r\n",
        "\r\n",
        "sample: 1HL_1A.assembled\r\n",
        "bins\tdepth_histogram\tcnts\r\n",
        "   :\t0------------50-------------100%\r\n",
        "0 \t**********************              56029\r\n",
        "5 \t*****                               11364\r\n",
        "10 \t***                                 4582\r\n",
        "15 \t**                                  2526\r\n",
        "20 \t**                                  1641\r\n",
        "25 \t*                                   1178\r\n",
        "30 \t*                                   969\r\n",
        "35 \t*                                   746\r\n",
        "40 \t*                                   932\r\n",
        "50 \t*                                   850\r\n",
        "100 \t*                                   216\r\n",
        "250 \t*                                   49\r\n",
        "500 \t*                                   21\r\n",
        "\r\n",
        "sample: 1HL_2A.assembled\r\n",
        "bins\tdepth_histogram\tcnts\r\n",
        "   :\t0------------50-------------100%\r\n",
        "0 \t*********************               56361\r\n",
        "5 \t*****                               11825\r\n",
        "10 \t***                                 5506\r\n",
        "15 \t**                                  2800\r\n",
        "20 \t**                                  1841\r\n",
        "25 \t*                                   1380\r\n",
        "30 \t*                                   995\r\n",
        "35 \t*                                   841\r\n",
        "40 \t*                                   1265\r\n",
        "50 \t**                                  1652\r\n",
        "100 \t*                                   344\r\n",
        "250 \t*                                   69\r\n",
        "500 \t*                                   37\r\n",
        "\r\n",
        "sample: 1HL_3A.assembled\r\n",
        "bins\tdepth_histogram\tcnts\r\n",
        "   :\t0------------50-------------100%\r\n",
        "0 \t***********************             71269\r\n",
        "5 \t*****                               11612\r\n",
        "10 \t***                                 4992\r\n",
        "15 \t**                                  2525\r\n",
        "20 \t**                                  1697\r\n",
        "25 \t*                                   1190\r\n",
        "30 \t*                                   926\r\n",
        "35 \t*                                   666\r\n",
        "40 \t*                                   860\r\n",
        "50 \t*                                   679\r\n",
        "100 \t*                                   202\r\n",
        "250 \t*                                   46\r\n",
        "500 \t*                                   21\r\n",
        "\r\n",
        "sample: 1HL_4A.assembled\r\n",
        "bins\tdepth_histogram\tcnts\r\n",
        "   :\t0------------50-------------100%\r\n",
        "0 \t*********************               56535\r\n",
        "5 \t*****                               11765\r\n",
        "10 \t***                                 5419\r\n",
        "15 \t**                                  2958\r\n",
        "20 \t**                                  1960\r\n",
        "25 \t*                                   1304\r\n",
        "30 \t*                                   996\r\n",
        "35 \t*                                   843\r\n",
        "40 \t*                                   1153\r\n",
        "50 \t*                                   1210\r\n",
        "100 \t*                                   265\r\n",
        "250 \t*                                   67\r\n",
        "500 \t*                                   29\r\n",
        "\r\n",
        "sample: 1HL_5A.assembled\r\n",
        "bins\tdepth_histogram\tcnts\r\n",
        "   :\t0------------50-------------100%\r\n",
        "0 \t**********************              55099\r\n",
        "5 \t*****                               11043\r\n",
        "10 \t***                                 4350\r\n",
        "15 \t**                                  2286\r\n",
        "20 \t**                                  1510\r\n",
        "25 \t*                                   1202\r\n",
        "30 \t*                                   855\r\n",
        "35 \t*                                   582\r\n",
        "40 \t*                                   626\r\n",
        "50 \t*                                   557\r\n",
        "100 \t*                                   165\r\n",
        "250 \t*                                   35\r\n",
        "500 \t*                                   17\r\n",
        "\r\n",
        "sample: 1HL_6A.assembled\r\n",
        "bins\tdepth_histogram\tcnts\r\n",
        "   :\t0------------50-------------100%\r\n",
        "0 \t**********************              54404\r\n",
        "5 \t*****                               10719\r\n",
        "10 \t***                                 4149\r\n",
        "15 \t**                                  2356\r\n",
        "20 \t**                                  1592\r\n",
        "25 \t*                                   1164\r\n",
        "30 \t*                                   940\r\n",
        "35 \t*                                   717\r\n",
        "40 \t*                                   706\r\n",
        "50 \t*                                   562\r\n",
        "100 \t*                                   187\r\n",
        "250 \t*                                   40\r\n",
        "500 \t*                                   17\r\n",
        "\r\n",
        "sample: 1HL_7A.assembled\r\n",
        "bins\tdepth_histogram\tcnts\r\n",
        "   :\t0------------50-------------100%\r\n",
        "0 \t*********************               63246\r\n",
        "5 \t*****                               12604\r\n",
        "10 \t***                                 6017\r\n",
        "15 \t**                                  3462\r\n",
        "20 \t**                                  2115\r\n",
        "25 \t*                                   1426\r\n",
        "30 \t*                                   1123\r\n",
        "35 \t*                                   763\r\n",
        "40 \t*                                   1168\r\n",
        "50 \t*                                   1221\r\n",
        "100 \t*                                   291\r\n",
        "250 \t*                                   66\r\n",
        "500 \t*                                   34\r\n",
        "\r\n",
        "sample: 1HL_8A.assembled\r\n",
        "bins\tdepth_histogram\tcnts\r\n",
        "   :\t0------------50-------------100%\r\n",
        "0 \t*********************               54804\r\n",
        "5 \t*****                               11048\r\n",
        "10 \t***                                 4511\r\n",
        "15 \t**                                  2450\r\n",
        "20 \t**                                  1718\r\n",
        "25 \t*                                   1250\r\n",
        "30 \t*                                   981\r\n",
        "35 \t*                                   767\r\n",
        "40 \t*                                   1114\r\n",
        "50 \t**                                  1356\r\n",
        "100 \t*                                   293\r\n",
        "250 \t*                                   52\r\n",
        "500 \t*                                   33\r\n",
        "\r\n",
        "sample: 1HL_9A.assembled\r\n",
        "bins\tdepth_histogram\tcnts\r\n",
        "   :\t0------------50-------------100%\r\n",
        "0 \t**********************              70776\r\n",
        "5 \t****                                11563\r\n",
        "10 \t***                                 5531\r\n",
        "15 \t**                                  2890\r\n",
        "20 \t**                                  1852\r\n",
        "25 \t*                                   1415\r\n",
        "30 \t*                                   1103\r\n",
        "35 \t*                                   793\r\n",
        "40 \t*                                   1225\r\n",
        "50 \t*                                   1596\r\n",
        "100 \t*                                   340\r\n",
        "250 \t*                                   68\r\n",
        "500 \t*                                   27\r\n",
        "\r\n",
        "sample: 1NF_1A.assembled\r\n",
        "bins\tdepth_histogram\tcnts\r\n",
        "   :\t0------------50-------------100%\r\n",
        "0 \t**********************              38814\r\n",
        "5 \t*****                               7149\r\n",
        "10 \t***                                 3021\r\n",
        "15 \t**                                  1863\r\n",
        "20 \t**                                  1303\r\n",
        "25 \t*                                   877\r\n",
        "30 \t*                                   527\r\n",
        "35 \t*                                   316\r\n",
        "40 \t*                                   301\r\n",
        "50 \t*                                   308\r\n",
        "100 \t*                                   119\r\n",
        "250 \t*                                   21\r\n",
        "500 \t*                                   9\r\n",
        "\r\n",
        "sample: 1NF_2A.assembled\r\n",
        "bins\tdepth_histogram\tcnts\r\n",
        "   :\t0------------50-------------100%\r\n",
        "0 \t*********************               61356\r\n",
        "5 \t*****                               12215\r\n",
        "10 \t***                                 6132\r\n",
        "15 \t**                                  3496\r\n",
        "20 \t**                                  2281\r\n",
        "25 \t*                                   1523\r\n",
        "30 \t*                                   1188\r\n",
        "35 \t*                                   938\r\n",
        "40 \t*                                   1415\r\n",
        "50 \t**                                  2323\r\n",
        "100 \t*                                   425\r\n",
        "250 \t*                                   97\r\n",
        "500 \t*                                   41\r\n",
        "\r\n",
        "sample: 1NF_4A.assembled\r\n",
        "bins\tdepth_histogram\tcnts\r\n",
        "   :\t0------------50-------------100%\r\n",
        "0 \t*********************               66278\r\n",
        "5 \t*****                               12991\r\n",
        "10 \t***                                 6516\r\n",
        "15 \t**                                  4062\r\n",
        "20 \t**                                  2727\r\n",
        "25 \t**                                  1821\r\n",
        "30 \t*                                   1243\r\n",
        "35 \t*                                   946\r\n",
        "40 \t*                                   1459\r\n",
        "50 \t**                                  2239\r\n",
        "100 \t*                                   421\r\n",
        "250 \t*                                   85\r\n",
        "500 \t*                                   53\r\n",
        "\r\n",
        "sample: 1NF_5A.assembled\r\n",
        "bins\tdepth_histogram\tcnts\r\n",
        "   :\t0------------50-------------100%\r\n",
        "0 \t*********************               59287\r\n",
        "5 \t*****                               12241\r\n",
        "10 \t***                                 5483\r\n",
        "15 \t**                                  3067\r\n",
        "20 \t**                                  1895\r\n",
        "25 \t*                                   1339\r\n",
        "30 \t*                                   1045\r\n",
        "35 \t*                                   846\r\n",
        "40 \t*                                   1214\r\n",
        "50 \t*                                   1220\r\n",
        "100 \t*                                   281\r\n",
        "250 \t*                                   63\r\n",
        "500 \t*                                   32\r\n",
        "\r\n",
        "sample: 1NF_6A.assembled\r\n",
        "bins\tdepth_histogram\tcnts\r\n",
        "   :\t0------------50-------------100%\r\n",
        "0 \t**********************              58114\r\n",
        "5 \t*****                               11484\r\n",
        "10 \t***                                 4719\r\n",
        "15 \t**                                  2537\r\n",
        "20 \t**                                  1638\r\n",
        "25 \t*                                   1212\r\n",
        "30 \t*                                   948\r\n",
        "35 \t*                                   778\r\n",
        "40 \t*                                   1065\r\n",
        "50 \t*                                   992\r\n",
        "100 \t*                                   227\r\n",
        "250 \t*                                   55\r\n",
        "500 \t*                                   33\r\n",
        "\r\n",
        "sample: 1NF_7A.assembled\r\n",
        "bins\tdepth_histogram\tcnts\r\n",
        "   :\t0------------50-------------100%\r\n",
        "0 \t**********************              58873\r\n",
        "5 \t*****                               11822\r\n",
        "10 \t***                                 4784\r\n",
        "15 \t**                                  2579\r\n",
        "20 \t**                                  1680\r\n",
        "25 \t*                                   1282\r\n",
        "30 \t*                                   971\r\n",
        "35 \t*                                   756\r\n",
        "40 \t*                                   1172\r\n",
        "50 \t*                                   1287\r\n",
        "100 \t*                                   267\r\n",
        "250 \t*                                   59\r\n",
        "500 \t*                                   34\r\n",
        "\r\n",
        "sample: 1NF_8A.assembled\r\n",
        "bins\tdepth_histogram\tcnts\r\n",
        "   :\t0------------50-------------100%\r\n",
        "0 \t**********************              60973\r\n",
        "5 \t*****                               11764\r\n",
        "10 \t***                                 5081\r\n",
        "15 \t**                                  2592\r\n",
        "20 \t**                                  1713\r\n",
        "25 \t*                                   1323\r\n",
        "30 \t*                                   1059\r\n",
        "35 \t*                                   811\r\n",
        "40 \t*                                   1250\r\n",
        "50 \t**                                  1510\r\n",
        "100 \t*                                   296\r\n",
        "250 \t*                                   66\r\n",
        "500 \t*                                   30\r\n",
        "\r\n",
        "sample: 1NF_9A.assembled\r\n",
        "bins\tdepth_histogram\tcnts\r\n",
        "   :\t0------------50-------------100%\r\n",
        "0 \t*********************               62779\r\n",
        "5 \t*****                               12604\r\n",
        "10 \t***                                 5809\r\n",
        "15 \t**                                  3317\r\n",
        "20 \t**                                  2191\r\n",
        "25 \t*                                   1554\r\n",
        "30 \t*                                   1170\r\n",
        "35 \t*                                   873\r\n",
        "40 \t*                                   1362\r\n",
        "50 \t**                                  2357\r\n",
        "100 \t*                                   433\r\n",
        "250 \t*                                   75\r\n",
        "500 \t*                                   52\r\n",
        "\r\n",
        "sample: 1SN_10A.assembled\r\n",
        "bins\tdepth_histogram\tcnts\r\n",
        "   :\t0------------50-------------100%\r\n",
        "0 \t**********************              67109\r\n",
        "5 \t*****                               12369\r\n",
        "10 \t***                                 5164\r\n",
        "15 \t**                                  2708\r\n",
        "20 \t**                                  1690\r\n",
        "25 \t*                                   1295\r\n",
        "30 \t*                                   976\r\n",
        "35 \t*                                   800\r\n",
        "40 \t*                                   1131\r\n",
        "50 \t*                                   1186\r\n",
        "100 \t*                                   266\r\n",
        "250 \t*                                   59\r\n",
        "500 \t*                                   26\r\n",
        "\r\n",
        "sample: 1SN_1A.assembled\r\n",
        "bins\tdepth_histogram\tcnts\r\n",
        "   :\t0------------50-------------100%\r\n",
        "0 \t*********************               58247\r\n",
        "5 \t*****                               11888\r\n",
        "10 \t***                                 5228\r\n",
        "15 \t**                                  2764\r\n",
        "20 \t**                                  1725\r\n",
        "25 \t*                                   1419\r\n",
        "30 \t*                                   1100\r\n",
        "35 \t*                                   829\r\n",
        "40 \t*                                   1331\r\n",
        "50 \t**                                  2002\r\n",
        "100 \t*                                   326\r\n",
        "250 \t*                                   78\r\n",
        "500 \t*                                   38\r\n",
        "\r\n",
        "sample: 1SN_2A.assembled\r\n",
        "bins\tdepth_histogram\tcnts\r\n",
        "   :\t0------------50-------------100%\r\n",
        "0 \t**********************              65452\r\n",
        "5 \t*****                               12751\r\n",
        "10 \t***                                 5595\r\n",
        "15 \t**                                  2993\r\n",
        "20 \t**                                  1862\r\n",
        "25 \t*                                   1320\r\n",
        "30 \t*                                   1024\r\n",
        "35 \t*                                   878\r\n",
        "40 \t*                                   1292\r\n",
        "50 \t**                                  1962\r\n",
        "100 \t*                                   342\r\n",
        "250 \t*                                   95\r\n",
        "500 \t*                                   39\r\n",
        "\r\n",
        "sample: 1SN_3A.assembled\r\n",
        "bins\tdepth_histogram\tcnts\r\n",
        "   :\t0------------50-------------100%\r\n",
        "0 \t************************            80441\r\n",
        "5 \t****                                12267\r\n",
        "10 \t**                                  4617\r\n",
        "15 \t**                                  2503\r\n",
        "20 \t*                                   1672\r\n",
        "25 \t*                                   1247\r\n",
        "30 \t*                                   979\r\n",
        "35 \t*                                   755\r\n",
        "40 \t*                                   922\r\n",
        "50 \t*                                   841\r\n",
        "100 \t*                                   226\r\n",
        "250 \t*                                   53\r\n",
        "500 \t*                                   18\r\n",
        "\r\n",
        "sample: 1SN_4A.assembled\r\n",
        "bins\tdepth_histogram\tcnts\r\n",
        "   :\t0------------50-------------100%\r\n",
        "0 \t*********************               66393\r\n",
        "5 \t*****                               13090\r\n",
        "10 \t***                                 6546\r\n",
        "15 \t**                                  3852\r\n",
        "20 \t**                                  2395\r\n",
        "25 \t*                                   1627\r\n",
        "30 \t*                                   1198\r\n",
        "35 \t*                                   941\r\n",
        "40 \t*                                   1400\r\n",
        "50 \t**                                  1996\r\n",
        "100 \t*                                   399\r\n",
        "250 \t*                                   80\r\n",
        "500 \t*                                   46\r\n",
        "\r\n",
        "sample: 1SN_5A.assembled\r\n",
        "bins\tdepth_histogram\tcnts\r\n",
        "   :\t0------------50-------------100%\r\n",
        "0 \t**********************              74908\r\n",
        "5 \t*****                               13523\r\n",
        "10 \t***                                 6269\r\n",
        "15 \t**                                  3448\r\n",
        "20 \t**                                  2200\r\n",
        "25 \t*                                   1485\r\n",
        "30 \t*                                   1106\r\n",
        "35 \t*                                   839\r\n",
        "40 \t*                                   1279\r\n",
        "50 \t*                                   1488\r\n",
        "100 \t*                                   310\r\n",
        "250 \t*                                   82\r\n",
        "500 \t*                                   30\r\n",
        "\r\n",
        "sample: 1SN_6A.assembled\r\n",
        "bins\tdepth_histogram\tcnts\r\n",
        "   :\t0------------50-------------100%\r\n",
        "0 \t*********************               64836\r\n",
        "5 \t*****                               12907\r\n",
        "10 \t***                                 6379\r\n",
        "15 \t**                                  3500\r\n",
        "20 \t**                                  2255\r\n",
        "25 \t*                                   1567\r\n",
        "30 \t*                                   1095\r\n",
        "35 \t*                                   1008\r\n",
        "40 \t*                                   1394\r\n",
        "50 \t**                                  2180\r\n",
        "100 \t*                                   379\r\n",
        "250 \t*                                   92\r\n",
        "500 \t*                                   43\r\n",
        "\r\n",
        "sample: 1SN_7A.assembled\r\n",
        "bins\tdepth_histogram\tcnts\r\n",
        "   :\t0------------50-------------100%\r\n",
        "0 \t*********************               65621\r\n",
        "5 \t*****                               12628\r\n",
        "10 \t***                                 6010\r\n",
        "15 \t**                                  3312\r\n",
        "20 \t**                                  2150\r\n",
        "25 \t*                                   1473\r\n",
        "30 \t*                                   1084\r\n",
        "35 \t*                                   875\r\n",
        "40 \t*                                   1268\r\n",
        "50 \t*                                   1563\r\n",
        "100 \t*                                   339\r\n",
        "250 \t*                                   73\r\n",
        "500 \t*                                   42\r\n",
        "\r\n",
        "sample: 1SN_8A.assembled\r\n",
        "bins\tdepth_histogram\tcnts\r\n",
        "   :\t0------------50-------------100%\r\n",
        "0 \t*********************               60561\r\n",
        "5 \t*****                               12830\r\n",
        "10 \t***                                 6261\r\n",
        "15 \t**                                  3761\r\n",
        "20 \t**                                  2392\r\n",
        "25 \t**                                  1617\r\n",
        "30 \t*                                   1149\r\n",
        "35 \t*                                   979\r\n",
        "40 \t*                                   1374\r\n",
        "50 \t**                                  1745\r\n",
        "100 \t*                                   364\r\n",
        "250 \t*                                   70\r\n",
        "500 \t*                                   40\r\n",
        "\r\n",
        "sample: 1SN_9A.assembled\r\n",
        "bins\tdepth_histogram\tcnts\r\n",
        "   :\t0------------50-------------100%\r\n",
        "0 \t**********************              77269\r\n",
        "5 \t*****                               13128\r\n",
        "10 \t***                                 6562\r\n",
        "15 \t**                                  3704\r\n",
        "20 \t**                                  2318\r\n",
        "25 \t*                                   1550\r\n",
        "30 \t*                                   1102\r\n",
        "35 \t*                                   878\r\n",
        "40 \t*                                   1302\r\n",
        "50 \t*                                   1715\r\n",
        "100 \t*                                   357\r\n",
        "250 \t*                                   72\r\n",
        "500 \t*                                   44\r\n",
        "\r\n"
       ]
      }
     ],
     "prompt_number": 19
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "%%bash\n",
      "time pyrad -p params.txt -s 4"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stderr",
       "text": [
        "\n",
        "\n",
        "     ------------------------------------------------------------\n",
        "      pyRAD : RADseq for phylogenetics & introgression analyses\n",
        "     ------------------------------------------------------------\n",
        "\n",
        "\n",
        "\tstep 4: estimating error rate and heterozygosity\n",
        "\t.............................\n",
        "real\t471m18.216s\n",
        "user\t3176m23.286s\n",
        "sys\t3m33.770s\n"
       ]
      }
     ],
     "prompt_number": 20
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "%%bash\n",
      "time pyRAD -p params.txt -s 5"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stderr",
       "text": [
        "\n",
        "\n",
        "     ------------------------------------------------------------\n",
        "      pyRAD : RADseq for phylogenetics & introgression analyses\n",
        "     ------------------------------------------------------------\n",
        "\n",
        "\n",
        "\tstep 5: creating consensus seqs for 29 samples, using H=0.01108 E=0.00184\n",
        "\t.............................\n",
        "real\t150m47.055s\n",
        "user\t1150m47.513s\n",
        "sys\t1m8.497s\n"
       ]
      }
     ],
     "prompt_number": 21
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "%%bash\n",
      "time pyRAD -p params.txt -s 6"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "vsearch v1.11.1_osx_x86_64, 16.0GB RAM, 8 cores\n",
        "https://github.com/torognes/vsearch\n",
        "\n",
        "\n",
        "\tfinished clustering\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stderr",
       "text": [
        "\n",
        "\n",
        "     ------------------------------------------------------------\n",
        "      pyRAD : RADseq for phylogenetics & introgression analyses\n",
        "     ------------------------------------------------------------\n",
        "\n",
        "\n",
        "\tstep 6: clustering across 29 samples at '.88' similarity \n",
        "\n",
        "Reading file /Users/Sam/oly_pyrad/clust.88/cat.haplos_ 100%\n",
        "62229589 nt in 556070 seqs, min 32, max 245, avg 112\n",
        "Counting unique k-mers 100%\n",
        "Clustering 100%\n",
        "Sorting clusters 100%\n",
        "Writing clusters 100%\n",
        "Clusters: 71397 Size min 1, max 65, avg 7.8\n",
        "Singletons: 18100, 3.3% of seqs, 25.4% of clusters\n",
        "\n",
        "real\t11m41.582s\n",
        "user\t15m13.421s\n",
        "sys\t0m10.647s\n"
       ]
      }
     ],
     "prompt_number": 22
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "%%bash\n",
      "time pyRAD -p params.txt -s 7"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\tingroup 1HL_10A.assembled,1HL_1A.assembled,1HL_2A.assembled,1HL_3A.assembled,1HL_4A.assembled,1HL_5A.assembled,1HL_6A.assembled,1HL_7A.assembled,1HL_8A.assembled,1HL_9A.assembled,1NF_10A.assembled,1NF_1A.assembled,1NF_2A.assembled,1NF_4A.assembled,1NF_5A.assembled,1NF_6A.assembled,1NF_7A.assembled,1NF_8A.assembled,1NF_9A.assembled,1SN_10A.assembled,1SN_1A.assembled,1SN_2A.assembled,1SN_3A.assembled,1SN_4A.assembled,1SN_5A.assembled,1SN_6A.assembled,1SN_7A.assembled,1SN_8A.assembled,1SN_9A.assembled\n",
        "\taddon \n",
        "\texclude \n",
        "\t\n",
        "\tfinal stats written to:\n",
        "\t /Users/Sam/oly_pyrad/stats/oly_gbs_pyrad.stats\n",
        "\toutput files being written to:\n",
        "\t /Users/Sam/oly_pyrad/outfiles/ directory\n",
        "\n",
        "\tfiltering & writing to phylip file\n",
        "\twriting nexus file\n",
        "\tWriting gphocs file\n",
        "\t  + writing full SNPs file\n",
        "\t  + writing unlinked bi-allelic SNPs file\n",
        "\t  + writing STRUCTURE file\n",
        "\t  + writing geno file\n",
        "\t  + writing treemix file\n",
        "\t    data set reduced for group coverage minimums\n",
        "\t    1HL ['1HL_10A.assembled', '1HL_1A.assembled', '1HL_2A.assembled', '1HL_3A.assembled', '1HL_4A.assembled', '1HL_5A.assembled', '1HL_6A.assembled', '1HL_7A.assembled', '1HL_8A.assembled', '1HL_9A.assembled'] minimum= 5\n",
        "\t    1NF ['1NF_10A.assembled', '1NF_1A.assembled', '1NF_2A.assembled', '1NF_4A.assembled', '1NF_5A.assembled', '1NF_6A.assembled', '1NF_7A.assembled', '1NF_8A.assembled', '1NF_9A.assembled'] minimum= 5\n",
        "\t    1SN ['1SN_10A.assembled', '1SN_1A.assembled', '1SN_2A.assembled', '1SN_3A.assembled', '1SN_4A.assembled', '1SN_5A.assembled', '1SN_6A.assembled', '1SN_7A.assembled', '1SN_8A.assembled', '1SN_9A.assembled'] minimum= 5\n",
        "\twriting vcf file\n",
        "\twriting migrate-n file\n",
        "\t    data set reduced for group coverage minimums\n",
        "\t    1HL ['1HL_10A.assembled', '1HL_1A.assembled', '1HL_2A.assembled', '1HL_3A.assembled', '1HL_4A.assembled', '1HL_5A.assembled', '1HL_6A.assembled', '1HL_7A.assembled', '1HL_8A.assembled', '1HL_9A.assembled'] minimum= 5\n",
        "\t    1NF ['1NF_10A.assembled', '1NF_1A.assembled', '1NF_2A.assembled', '1NF_4A.assembled', '1NF_5A.assembled', '1NF_6A.assembled', '1NF_7A.assembled', '1NF_8A.assembled', '1NF_9A.assembled'] minimum= 5\n",
        "\t    1SN ['1SN_10A.assembled', '1SN_1A.assembled', '1SN_2A.assembled', '1SN_3A.assembled', '1SN_4A.assembled', '1SN_5A.assembled', '1SN_6A.assembled', '1SN_7A.assembled', '1SN_8A.assembled', '1SN_9A.assembled'] minimum= 5\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stderr",
       "text": [
        "\n",
        "\n",
        "     ------------------------------------------------------------\n",
        "      pyRAD : RADseq for phylogenetics & introgression analyses\n",
        "     ------------------------------------------------------------\n",
        "\n",
        "\n",
        "\tCluster input file: using \n",
        "\t/Users/Sam/oly_pyrad/clust.88/cat.clust_.gz\n",
        "\n",
        "........Traceback (most recent call last):\n",
        "  File \"/usr/local/bioinformatics/anaconda/bin/pyRAD\", line 9, in <module>\n",
        "    load_entry_point('pyrad==3.0.66', 'console_scripts', 'pyrad')()\n",
        "  File \"build/bdist.macosx-10.5-x86_64/egg/pyrad/pyRAD.py\", line 527, in main\n",
        "  File \"build/bdist.macosx-10.5-x86_64/egg/pyrad/alignable.py\", line 1013, in main\n",
        "  File \"build/bdist.macosx-10.5-x86_64/egg/pyrad/loci2mig.py\", line 75, in make\n",
        "IndexError: list index out of range\n",
        "\n",
        "real\t8m51.355s\n",
        "user\t31m44.363s\n",
        "sys\t4m53.344s\n"
       ]
      }
     ],
     "prompt_number": 23
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "cat stats/oly_gbs_pyrad.stats"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\r\n",
        "\r\n",
        "35373       ## loci with > minsp containing data\r\n",
        "26281       ## loci with > minsp containing data & paralogs removed\r\n",
        "26281       ## loci with > minsp containing data & paralogs removed & final filtering\r\n",
        "\r\n",
        "## number of loci recovered in final data set for each taxon.\r\n",
        "taxon\tnloci\r\n",
        "1HL_10A.assembled\t10633\r\n",
        "1HL_1A.assembled \t8091\r\n",
        "1HL_2A.assembled \t9825\r\n",
        "1HL_3A.assembled \t8641\r\n",
        "1HL_4A.assembled \t9753\r\n",
        "1HL_5A.assembled \t7550\r\n",
        "1HL_6A.assembled \t7099\r\n",
        "1HL_7A.assembled \t10707\r\n",
        "1HL_8A.assembled \t8555\r\n",
        "1HL_9A.assembled \t9876\r\n",
        "1NF_10A.assembled\t6648\r\n",
        "1NF_1A.assembled \t5081\r\n",
        "1NF_2A.assembled \t10995\r\n",
        "1NF_4A.assembled \t12148\r\n",
        "1NF_5A.assembled \t9897\r\n",
        "1NF_6A.assembled \t8795\r\n",
        "1NF_7A.assembled \t9020\r\n",
        "1NF_8A.assembled \t8517\r\n",
        "1NF_9A.assembled \t11097\r\n",
        "1SN_10A.assembled\t9426\r\n",
        "1SN_1A.assembled \t9878\r\n",
        "1SN_2A.assembled \t10092\r\n",
        "1SN_3A.assembled \t8229\r\n",
        "1SN_4A.assembled \t12028\r\n",
        "1SN_5A.assembled \t11164\r\n",
        "1SN_6A.assembled \t11243\r\n",
        "1SN_7A.assembled \t10620\r\n",
        "1SN_8A.assembled \t11504\r\n",
        "1SN_9A.assembled \t11687\r\n",
        "\r\n",
        "\r\n",
        "## nloci = number of loci with data for exactly ntaxa\r\n",
        "## ntotal = number of loci for which at least ntaxa have data\r\n",
        "ntaxa\tnloci\tsaved\tntotal\r\n",
        "1\t-\r\n",
        "2\t-\t\t-\r\n",
        "3\t-\t\t-\r\n",
        "4\t4023\t*\t26281\r\n",
        "5\t3130\t*\t22258\r\n",
        "6\t2433\t*\t19128\r\n",
        "7\t2093\t*\t16695\r\n",
        "8\t1756\t*\t14602\r\n",
        "9\t1461\t*\t12846\r\n",
        "10\t1392\t*\t11385\r\n",
        "11\t1170\t*\t9993\r\n",
        "12\t957\t*\t8823\r\n",
        "13\t817\t*\t7866\r\n",
        "14\t812\t*\t7049\r\n",
        "15\t651\t*\t6237\r\n",
        "16\t624\t*\t5586\r\n",
        "17\t569\t*\t4962\r\n",
        "18\t534\t*\t4393\r\n",
        "19\t499\t*\t3859\r\n",
        "20\t430\t*\t3360\r\n",
        "21\t379\t*\t2930\r\n",
        "22\t328\t*\t2551\r\n",
        "23\t327\t*\t2223\r\n",
        "24\t309\t*\t1896\r\n",
        "25\t300\t*\t1587\r\n",
        "26\t284\t*\t1287\r\n",
        "27\t266\t*\t1003\r\n",
        "28\t318\t*\t737\r\n",
        "29\t419\t*\t419\r\n",
        "\r\n",
        "\r\n",
        "## nvar = number of loci containing n variable sites (pis+autapomorphies).\r\n",
        "## sumvar = sum of variable sites (SNPs).\r\n",
        "## pis = number of loci containing n parsimony informative sites.\r\n",
        "## sumpis = sum of parsimony informative sites.\r\n",
        "\tnvar\tsumvar\tPIS\tsumPIS\r\n",
        "0\t11292\t0\t15943\t0\r\n",
        "1\t3989\t3989\t3613\t3613\r\n",
        "2\t2426\t8841\t2024\t7661\r\n",
        "3\t1748\t14085\t1404\t11873\r\n",
        "4\t1285\t19225\t974\t15769\r\n",
        "5\t1104\t24745\t694\t19239\r\n",
        "6\t909\t30199\t531\t22425\r\n",
        "7\t734\t35337\t352\t24889\r\n",
        "8\t599\t40129\t230\t26729\r\n",
        "9\t487\t44512\t164\t28205\r\n",
        "10\t360\t48112\t123\t29435\r\n",
        "11\t292\t51324\t62\t30117\r\n",
        "12\t199\t53712\t60\t30837\r\n",
        "13\t199\t56299\t45\t31422\r\n",
        "14\t154\t58455\t21\t31716\r\n",
        "15\t91\t59820\t17\t31971\r\n",
        "16\t101\t61436\t9\t32115\r\n",
        "17\t71\t62643\t6\t32217\r\n",
        "18\t49\t63525\t2\t32253\r\n",
        "19\t43\t64342\t0\t32253\r\n",
        "20\t34\t65022\t1\t32273\r\n",
        "21\t22\t65484\t0\t32273\r\n",
        "22\t28\t66100\t1\t32295\r\n",
        "23\t10\t66330\t1\t32318\r\n",
        "24\t19\t66786\t1\t32342\r\n",
        "25\t2\t66836\t2\t32392\r\n",
        "26\t2\t66888\t0\t32392\r\n",
        "27\t6\t67050\t1\t32419\r\n",
        "28\t3\t67134\t0\t32419\r\n",
        "29\t3\t67221\t0\t32419\r\n",
        "30\t3\t67311\t0\t32419\r\n",
        "31\t4\t67435\t0\t32419\r\n",
        "32\t1\t67467\t0\t32419\r\n",
        "33\t4\t67599\t0\t32419\r\n",
        "34\t1\t67633\t0\t32419\r\n",
        "35\t1\t67668\t0\t32419\r\n",
        "36\t1\t67704\t0\t32419\r\n",
        "37\t0\t67704\t0\t32419\r\n",
        "38\t1\t67742\t0\t32419\r\n",
        "39\t3\t67859\t0\t32419\r\n",
        "40\t0\t67859\t0\t32419\r\n",
        "41\t0\t67859\t0\t32419\r\n",
        "42\t0\t67859\t0\t32419\r\n",
        "43\t1\t67902\t0\t32419\r\n",
        "total var= 67902\r\n",
        "total pis= 32419\r\n",
        "sampled unlinked SNPs= 14989\r\n",
        "sampled unlinked bi-allelic SNPs= 14881\r\n"
       ]
      }
     ],
     "prompt_number": 24
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "ls outfiles/"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "oly_gbs_pyrad.alleles         oly_gbs_pyrad.migrate         oly_gbs_pyrad.snps            oly_gbs_pyrad.unlinked_snps\r\n",
        "oly_gbs_pyrad.excluded_loci   oly_gbs_pyrad.nex             oly_gbs_pyrad.snps.geno       oly_gbs_pyrad.usnps.geno\r\n",
        "oly_gbs_pyrad.gphocs          oly_gbs_pyrad.phy             oly_gbs_pyrad.str             oly_gbs_pyrad.vcf\r\n",
        "oly_gbs_pyrad.loci            oly_gbs_pyrad.phy.partitions  oly_gbs_pyrad.treemix.gz\r\n"
       ]
      }
     ],
     "prompt_number": 25
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "%%bash \n",
      "head -n 39 outfiles/oly_gbs_pyrad.loci | cut -c 1-75"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        ">1HL_1A.assembled     ATACAGTCAGATTGGTGCAAAGCCTTCCTGACTGCCGGGAAGTCACAACGTGA\n",
        ">1HL_2A.assembled     ATACAGTCAGATTGGTGCAAAGCCTTCCTGACTGCCGGGAAGYCACAACGTGA\n",
        ">1HL_5A.assembled     ATACAGTCAGATTGGTGCAAAGCCTTCCTGACTGCCGGGAAGTCACAACGTGA\n",
        ">1HL_6A.assembled     ATACAGTCAGATTGGTGCAAAGCCTTCCTGACTGCCGGGAAGTCACAACGTGA\n",
        ">1HL_7A.assembled     ATACAGTCAGATTGGTGCAAAGCCTTCCTGACTGCCGGGAAGYCACAACGTGA\n",
        ">1HL_9A.assembled     ATACAGTCAGATTGGTGCAAAGCCTTCCTGACTGCCGGGAAGTCACAACGTGA\n",
        ">1NF_10A.assembled    ATACAGTCAGATTGGTGCAAAGCCTTCCTGACTGCCGGGAAGTCACAACGTGA\n",
        ">1NF_1A.assembled     ATACAGTCAGATTGGTGCAAAGCCTTCCTGACTGCCGGGAAGTCACAACGTGA\n",
        ">1NF_6A.assembled     ATACAGTCAGATTGGTGCAAAGCCTTCCTGACTGCCGGGAAGTCACAACGTGA\n",
        ">1NF_7A.assembled     ATACAGTCAGATTGGTGCAAAGCCTTCCTGACTGCCGGGAAGCCACAACGTGA\n",
        ">1SN_10A.assembled    ATACAGTCAGATTGGTGCAAAGCCTTCCTGACTGCCGGGAAGTCACAACGTGA\n",
        ">1SN_2A.assembled     ATACAGTCAGATTGGTGCAAAGCCTTCCTGACTGCCGGGAAGTCACAACGTGA\n",
        ">1SN_3A.assembled     ATACAGTCAGATTGGTGCAAAGCCTTCCTGACTGCCGGGAAGTCACAACGTGA\n",
        ">1SN_6A.assembled     ATACAGTCAGATTGGTGCAAAGCCTTCCTGACTGCCGGGAAGTCACAACGTGA\n",
        ">1SN_7A.assembled     ATACAGTCAGATTGGTGCAAAGCCTTCCTGACTGCCGGGAAGTCACAACGTGA\n",
        ">1SN_9A.assembled     ATACAGTCAGATTGGTGCAAAGCCTTCCTGACTGCCGGGAAGTCACAACGTGA\n",
        "//                                                              *          \n",
        ">1HL_10A.assembled    AAGCCAAAATATCTTGAATCTTGATGCGACCAGTCCTCATGGGATCCAGGAAA\n",
        ">1HL_3A.assembled     AAGCCAAAATATCTTGAATCTTGATGCGACCAGTCCTCATGGGATCCAGGAAA\n",
        ">1HL_4A.assembled     AAGCCAAAATATCTTGAATCTTGATGCGACCAGTCCTCATGGGATCCAGGAAA\n",
        ">1HL_7A.assembled     AAGCCAAAATATCTTGAATCTTGATGCGACCAGTCCTCATGGGATCCAGGAAA\n",
        ">1HL_8A.assembled     AAGCCAAAATATCTTGAATCTTGATGCGACCAGTCCTCATGGGATCCAGGAAA\n",
        ">1NF_4A.assembled     AAGCCAAAATATCTTGAATCTTGATGCGACCAGTCCTCATGGGATCCAGGAAA\n",
        ">1NF_7A.assembled     AAGCCAAAATATCTTGAATCTTGATGCGACCAGTCCTCATGGGATCCAGGAAA\n",
        ">1NF_9A.assembled     AAGCCAAAATATCTTGAATCTTGATGCGACCAGTCCTCATGGGATCCAGGAAA\n",
        ">1SN_10A.assembled    AAGCCAAAATATCTTGAATCTTGATGCGACCAGTCCTCATGGGATCCAGGAAA\n",
        ">1SN_1A.assembled     AAGCCAAAATATCTTGAATCTTGATGCGACCAGTNNTCATGGGATCCAGGAAA\n",
        ">1SN_2A.assembled     AAGCCAAAATATCTTGNATCTTGATGCGACCAGTCCTCATGGGATCCAGGAAA\n",
        ">1SN_4A.assembled     AAGCCAAAATATCTTGAATCTTGATGCGACCAGTCCTCATGGGATCCAGGAAA\n",
        ">1SN_5A.assembled     AAGCCAAAATATCTTGAATCTTGATGCGACCAGTCCTCATGGGATCCAGGAAA\n",
        ">1SN_8A.assembled     AAGCCAAAATATCTTGAATCTTGATGCGACCAGTCCTCATGGGATCCAGGAAA\n",
        "//                                                                         \n",
        ">1HL_2A.assembled     NAATCGGAGGCATTTTCCCATGCAGGTTTGTCGAGTGGACAATTGTCGGAGAC\n",
        ">1HL_5A.assembled     GAATCGGAGGCATTTTCCCATGCAGGTTTGTCGAGTGGACAATTGTCGGAGAC\n",
        ">1HL_7A.assembled     GAATCGGAGGCATTTTCCCATGCAGGTTTGTCGAGTGGACAATTGTCGGRGAC\n",
        ">1NF_2A.assembled     GAATCGGAGGCATTTTCCCATGCAGGTTTGTCGAGTGGACAATTGTCGGRGAC\n",
        ">1NF_4A.assembled     GAATCGGAGGCATTTTCCCATGCAGGTTTGTCGAGTGGACAATTGTCGGAGAC\n",
        ">1NF_5A.assembled     GAATCGGAGGCATTTTCCCATGCAGGTTTGTCGAGTGGACAATTGTCGGAGAC\n",
        ">1NF_6A.assembled     GAATCGGAGGCATTTTCCCATGCAGGTTTGTCGAGTGGACAATTGTCGGAGAC\n"
       ]
      }
     ],
     "prompt_number": 26
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "%%bash\n",
      "head -1 outfiles/oly_gbs_pyrad.snps"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "## 29 taxa, 52562 loci, 105475 snps\n"
       ]
      }
     ],
     "prompt_number": 27
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "%%bash\n",
      "pwd"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/Users/Sam\n"
       ]
      }
     ],
     "prompt_number": 1
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "cd /Users/Sam/oly_pyrad/outfiles/"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/Users/Sam/oly_pyrad/outfiles\n"
       ]
      }
     ],
     "prompt_number": 2
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "%%bash\n",
      "zcat < oly_gbs_pyrad.treemix.gz | head"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "1HL 1NF 1SN\n",
        "20,0 15,3 18,0\n",
        "13,1 8,2 7,3\n",
        "12,0 10,0 12,2\n",
        "10,0 4,6 14,0\n",
        "14,0 9,1 12,0\n",
        "9,1 12,0 14,0\n",
        "17,1 16,0 16,0\n",
        "20,0 17,1 19,1\n",
        "14,0 12,0 15,1\n"
       ]
      }
     ],
     "prompt_number": 4
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [],
     "language": "python",
     "metadata": {},
     "outputs": []
    }
   ],
   "metadata": {}
  }
 ]
}