{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Mon Jun 13 13:53:23 UTC 2016\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "date"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "4bc848ecaf48\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "# Running Docker on roadrunner. Prints container ID, instead of host computer name (roaderunner)\n",
    "hostname"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Architecture:          x86_64\n",
      "CPU op-mode(s):        32-bit, 64-bit\n",
      "Byte Order:            Little Endian\n",
      "CPU(s):                2\n",
      "On-line CPU(s) list:   0,1\n",
      "Thread(s) per core:    1\n",
      "Core(s) per socket:    2\n",
      "Socket(s):             1\n",
      "Vendor ID:             GenuineIntel\n",
      "CPU family:            6\n",
      "Model:                 26\n",
      "Model name:            Intel(R) Xeon(R) CPU           E5520  @ 2.27GHz\n",
      "Stepping:              5\n",
      "CPU MHz:               2260.998\n",
      "BogoMIPS:              4521.99\n",
      "Hypervisor vendor:     KVM\n",
      "Virtualization type:   full\n",
      "L1d cache:             32K\n",
      "L1i cache:             32K\n",
      "L2 cache:              256K\n",
      "L3 cache:              8192K\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "lscpu"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "u'/notebooks'"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/data/oly_gbs\n"
     ]
    }
   ],
   "source": [
    "cd /data/oly_gbs/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "160123_I132_FCH3YHMBBXX_L4_OYSzenG1AAD96FAAPEI-109_R1.fq.gz\r\n",
      "160123_I132_FCH3YHMBBXX_L4_OYSzenG1AAD96FAAPEI-109_R2.fq.gz\r\n"
     ]
    }
   ],
   "source": [
    "ls"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Rename FASTQ files to match R1 and R2 requirements for pyrad demultiplexing"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "mv 160123_I132_FCH3YHMBBXX_L4_OYSzenG1AAD96FAAPEI-109_R1.fq.gz 160123_I132_FCH3YHMBBXX_L4_OYSzenG1AAD96FAAPEI-109_R1_.fq.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "mv 160123_I132_FCH3YHMBBXX_L4_OYSzenG1AAD96FAAPEI-109_R2.fq.gz 160123_I132_FCH3YHMBBXX_L4_OYSzenG1AAD96FAAPEI-109_R2_.fq.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "160123_I132_FCH3YHMBBXX_L4_OYSzenG1AAD96FAAPEI-109_R1_.fq.gz\r\n",
      "160123_I132_FCH3YHMBBXX_L4_OYSzenG1AAD96FAAPEI-109_R2_.fq.gz\r\n"
     ]
    }
   ],
   "source": [
    "ls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/analysis/20160609_pyrad\n"
     ]
    }
   ],
   "source": [
    "cd /analysis/20160609_pyrad/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "barcodes.txt  params.txt\r\n"
     ]
    }
   ],
   "source": [
    "ls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "==** parameter inputs for pyRAD version 3.0.66  **======================== affected step ==\n",
      "./                     ## 1. Working directory                                 (all)\n",
      "/data/oly_gbs/*.gz             \t       ## 2. Loc. of non-demultiplexed files (if not line 18)  (s1)\n",
      "/analysis/20160609_pyrad/barcodes.txt\t               ## 3. Loc. of barcode file (if not line 18)             (s1)\n",
      "/usr/local/bioinformatics/vsearch-1.11.1/bin/vsearch                ## 4. command (or path) to call vsearch (or usearch)    (s3,s6)\n",
      "/usr/local/bioinformatics/muscle3.8.31_i86darwin64                    ## 5. command (or path) to call muscle                  (s3,s7)\n",
      "CWGC                   ## 6. Restriction overhang (e.g., C|TGCAG -> TGCAG)     (s1,s2)\n",
      "16                     ## 7. N processors (parallel)                           (all)\n",
      "6                      ## 8. Mindepth: min coverage for a cluster              (s4,s5)\n",
      "4                      ## 9. NQual: max # sites with qual < 20 (or see line 20)(s2)\n",
      ".88                    ## 10. Wclust: clustering threshold as a decimal        (s3,s6)\n",
      "pairgbs                 ## 11. Datatype: rad,gbs,pairgbs,pairddrad,(others:see docs)(all)\n",
      "4                      ## 12. MinCov: min samples in a final locus             (s7)\n",
      "3                      ## 13. MaxSH: max inds with shared hetero site          (s7)\n",
      "oly_gbs_pyrad          ## 14. Prefix name for final output (no spaces)         (s7)\n",
      "==== optional params below this line ===================================  affected step ==\n",
      "                       ## 15.opt.: select subset (prefix* only selector)            (s2-s7)\n",
      "                       ## 16.opt.: add-on (outgroup) taxa (list or prefix*)         (s6,s7)\n",
      "                       ## 17.opt.: exclude taxa (list or prefix*)                   (s7)\n",
      "                       ## 18.opt.: loc. of de-multiplexed data                      (s2)\n",
      "                       ## 19.opt.: maxM: N mismatches in barcodes (def= 1)          (s1)\n",
      "                       ## 20.opt.: phred Qscore offset (def= 33)                    (s2)\n",
      "2                      ## 21.opt.: filter: def=0=NQual 1=NQual+adapters. 2=strict   (s2)\n",
      "                       ## 22.opt.: a priori E,H (def= 0.001,0.01, if not estimated) (s5)\n",
      "4                      ## 23.opt.: maxN: max Ns in a cons seq (def=5)               (s5)\n",
      "8                      ## 24.opt.: maxH: max heterozyg. sites in cons seq (def=5)   (s5)\n",
      "                       ## 25.opt.: ploidy: max alleles in cons seq (def=2;see docs) (s4,s5)\n",
      "                       ## 26.opt.: maxSNPs: (def=100). Paired (def=100,100)         (s7)\n",
      "                       ## 27.opt.: maxIndels: within-clust,across-clust (def. 3,99) (s3,s7)\n",
      "                       ## 28.opt.: random number seed (def. 112233)              (s3,s6,s7)\n",
      "                       ## 29.opt.: trim overhang left,right on final loci, def(0,0) (s7)\n",
      "*                      ## 30.opt.: output formats: p,n,a,s,v,u,t,m,k,g,* (see docs) (s7)\n",
      "                       ## 31.opt.: maj. base call at depth>x<mindepth (def.x=mindepth) (s5)\n",
      "50                     ## 32.opt.: keep trimmed reads (def=0). Enter min length.    (s2)\n",
      "                       ## 33.opt.: max stack size (int), def= max(500,mean+2*SD)    (s3)\n",
      "                       ## 34.opt.: minDerep: exclude dereps with <= N copies, def=1 (s3)\n",
      "                       ## 35.opt.: use hierarchical clustering (def.=0, 1=yes)      (s6)\n",
      "                       ## 36.opt.: repeat masking (def.=1='dust' method, 0=no)      (s3,s6)\n",
      "16                     ## 37.opt.: vsearch max threads per job (def.=6; see docs)   (s3,s6)\n",
      "==== optional: list group/clade assignments below this line (see docs) ==================\n",
      "1HL\t5\t1HL*\n",
      "1NF\t5\t1NF*\n",
      "1SN\t5\t1SN*\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "cat params.txt"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### View barcodes.txt file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1NF_1A\tCTCC\n",
      "1NF_2A\tTGCA\n",
      "1NF_4A\tACTA\n",
      "1NF_5A\tCAGA\n",
      "1NF_6A\tAACT\n",
      "1NF_7A\tGCGT\n",
      "1NF_8A\tCGAT\n",
      "1NF_9A\tGTAA\n",
      "1NF_10A\tAGGC\n",
      "1NF_11A\tGATC\n",
      "1NF_12A\tTCAC\n",
      "1NF_13A\tTGCGA\n",
      "1NF_14A\tCGCTT\n",
      "1NF_15A\tTCACC\n",
      "1NF_16A\tCTAGC\n",
      "1NF_17A\tACAAA\n",
      "1NF_18A\tTTCTC\n",
      "1NF_19A\tAGCCC\n",
      "1NF_20A\tGTATT\n",
      "1NF_21A\tCTGTA\n",
      "1NF_22A\tACCGT\n",
      "1NF_23A\tGCTTA\n",
      "1NF_24A\tGGTGT\n",
      "1NF_25A\tAGGAT\n",
      "1NF_26A\tATTGA\n",
      "1NF_27A\tCATCT\n",
      "1NF_28A\tCCTAC\n",
      "1NF_29A\tGAGGA\n",
      "1NF_30A\tGGAAC\n",
      "1NF_31A\tGTCAA\n",
      "1NF_32A\tTAATA\n",
      "1NF_33A\tTACAT\n",
      "1SN_1A\tTCGTT\n",
      "1SN_2A\tGGTTGT\n",
      "1SN_3A\tCCAGCT\n",
      "1SN_4A\tTTCAGA\n",
      "1SN_5A\tTAGGAA\n",
      "1SN_6A\tGCTCTA\n",
      "1SN_7A\tCCACAA\n",
      "1SN_8A\tCTTCCA\n",
      "1SN_9A\tGAGATA\n",
      "1SN_10A\tATGCCT\n",
      "1SN_11A\tAGTGGA\n",
      "1SN_12A\tACCTAA\n",
      "1SN_13A\tATATGT\n",
      "1SN_14A\tATCGTA\n",
      "1SN_15A\tCATCGT\n",
      "1SN_16A\tCGCGGT\n",
      "1SN_17A\tCTATTA\n",
      "1SN_18A\tGCCAGT\n",
      "1SN_19A\tGGAAGA\n",
      "1SN_20A\tGTACTT\n",
      "1SN_21A\tGTTGAA\n",
      "1SN_22A\tTAACGA\n",
      "1SN_23A\tTGGCTA\n",
      "1SN_24A\tTATTTTT\n",
      "1SN_25A\tCTTGCTT\n",
      "1SN_26A\tATGAAAC\n",
      "1SN_27A\tAAAAGTT\n",
      "1SN_28A\tGAATTCA\n",
      "1SN_29A\tGAACTTC\n",
      "1SN_30A\tGGACCTA\n",
      "1SN_31A\tGTCGATT\n",
      "1SN_32A\tAACGCCT\n",
      "1HL_1A\tAATATGC\n",
      "1HL_2A\tACGTGTT\n",
      "1HL_3A\tATTAATT\n",
      "1HL_4A\tATTGGAT\n",
      "1HL_5A\tCATAAGT\n",
      "1HL_6A\tCGCTGAT\n",
      "1HL_7A\tCGGTAGA\n",
      "1HL_8A\tCTACGGA\n",
      "1HL_9A\tGCGGAAT\n",
      "1HL_10A\tTAGCGGA\n",
      "1HL_11A\tTCGAAGA\n",
      "1HL_12A\tTCTGTGA\n",
      "1HL_13A\tTGCTGGA\n",
      "1HL_14A\tACGACTAC\n",
      "1HL_15A\tTAGCATGC\n",
      "1HL_16A\tTAGGCCAT\n",
      "1HL_17A\tTGCAAGGA\n",
      "1HL_19A\tTGGTACGT\n",
      "1HL_20A\tTCTCAGTC\n",
      "1HL_21A\tCCGGATAT\n",
      "1HL_22A\tCGCCTTAT\n",
      "1HL_23A\tAACCGAGA\n",
      "1HL_24A\tACAGGGAA\n",
      "1HL_25A\tACGTGGTA\n",
      "1HL_26A\tCCATGGGT\n",
      "1HL_27A\tCGCGGAGA\n",
      "1HL_28A\tCGTGTGGT\n",
      "1HL_29A\tGCTGTGGA\n",
      "1HL_31A\tGGATTGGT\n",
      "1HL_33A\tGTGAGGGT\n",
      "1HL_34A\tTATCGGGA\n",
      "1HL_35A\tTTCCTGGA\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "cat barcodes.txt"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Step 1: De-multiplex reads"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\tcannot find vsearch (or usearch), edit path in param file\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "     ------------------------------------------------------------\n",
      "      pyRAD : RADseq for phylogenetics & introgression analyses\n",
      "     ------------------------------------------------------------\n",
      "\n",
      "\n",
      "real\t0m2.121s\n",
      "user\t0m0.340s\n",
      "sys\t0m0.120s\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "time pyrad -p params.txt -s 1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/usr/local/bioinformatics/vsearch-1.11.1-linux-x86_64/bin/vsearch\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "which vsearch"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Fixed vsearch path in params.txt file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "==** parameter inputs for pyRAD version 3.0.66  **======================== affected step ==\n",
      "./                     ## 1. Working directory                                 (all)\n",
      "/data/oly_gbs/*.gz             \t       ## 2. Loc. of non-demultiplexed files (if not line 18)  (s1)\n",
      "/analysis/20160609_pyrad/barcodes.txt\t               ## 3. Loc. of barcode file (if not line 18)             (s1)\n",
      "/usr/local/bioinformatics/vsearch-1.11.1-linux-x86_64/bin/vsearch                ## 4. command (or path) to call vsearch (or usearch)    (s3,s6)\n",
      "/usr/local/bioinformatics/muscle3.8.31_i86darwin64                    ## 5. command (or path) to call muscle                  (s3,s7)\n",
      "CWGC                   ## 6. Restriction overhang (e.g., C|TGCAG -> TGCAG)     (s1,s2)\n",
      "16                     ## 7. N processors (parallel)                           (all)\n",
      "6                      ## 8. Mindepth: min coverage for a cluster              (s4,s5)\n",
      "4                      ## 9. NQual: max # sites with qual < 20 (or see line 20)(s2)\n",
      ".88                    ## 10. Wclust: clustering threshold as a decimal        (s3,s6)\n",
      "pairgbs                 ## 11. Datatype: rad,gbs,pairgbs,pairddrad,(others:see docs)(all)\n",
      "4                      ## 12. MinCov: min samples in a final locus             (s7)\n",
      "3                      ## 13. MaxSH: max inds with shared hetero site          (s7)\n",
      "oly_gbs_pyrad          ## 14. Prefix name for final output (no spaces)         (s7)\n",
      "==== optional params below this line ===================================  affected step ==\n",
      "                       ## 15.opt.: select subset (prefix* only selector)            (s2-s7)\n",
      "                       ## 16.opt.: add-on (outgroup) taxa (list or prefix*)         (s6,s7)\n",
      "                       ## 17.opt.: exclude taxa (list or prefix*)                   (s7)\n",
      "                       ## 18.opt.: loc. of de-multiplexed data                      (s2)\n",
      "                       ## 19.opt.: maxM: N mismatches in barcodes (def= 1)          (s1)\n",
      "                       ## 20.opt.: phred Qscore offset (def= 33)                    (s2)\n",
      "2                      ## 21.opt.: filter: def=0=NQual 1=NQual+adapters. 2=strict   (s2)\n",
      "                       ## 22.opt.: a priori E,H (def= 0.001,0.01, if not estimated) (s5)\n",
      "4                      ## 23.opt.: maxN: max Ns in a cons seq (def=5)               (s5)\n",
      "8                      ## 24.opt.: maxH: max heterozyg. sites in cons seq (def=5)   (s5)\n",
      "                       ## 25.opt.: ploidy: max alleles in cons seq (def=2;see docs) (s4,s5)\n",
      "                       ## 26.opt.: maxSNPs: (def=100). Paired (def=100,100)         (s7)\n",
      "                       ## 27.opt.: maxIndels: within-clust,across-clust (def. 3,99) (s3,s7)\n",
      "                       ## 28.opt.: random number seed (def. 112233)              (s3,s6,s7)\n",
      "                       ## 29.opt.: trim overhang left,right on final loci, def(0,0) (s7)\n",
      "*                      ## 30.opt.: output formats: p,n,a,s,v,u,t,m,k,g,* (see docs) (s7)\n",
      "                       ## 31.opt.: maj. base call at depth>x<mindepth (def.x=mindepth) (s5)\n",
      "50                     ## 32.opt.: keep trimmed reads (def=0). Enter min length.    (s2)\n",
      "                       ## 33.opt.: max stack size (int), def= max(500,mean+2*SD)    (s3)\n",
      "                       ## 34.opt.: minDerep: exclude dereps with <= N copies, def=1 (s3)\n",
      "                       ## 35.opt.: use hierarchical clustering (def.=0, 1=yes)      (s6)\n",
      "                       ## 36.opt.: repeat masking (def.=1='dust' method, 0=no)      (s3,s6)\n",
      "16                     ## 37.opt.: vsearch max threads per job (def.=6; see docs)   (s3,s6)\n",
      "==== optional: list group/clade assignments below this line (see docs) ==================\n",
      "1HL\t5\t1HL*\n",
      "1NF\t5\t1NF*\n",
      "1SN\t5\t1SN*\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "cat params.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\tcannot find muscle, edit path in input file\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "     ------------------------------------------------------------\n",
      "      pyRAD : RADseq for phylogenetics & introgression analyses\n",
      "     ------------------------------------------------------------\n",
      "\n",
      "\n",
      "real\t0m0.354s\n",
      "user\t0m0.280s\n",
      "sys\t0m0.060s\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "time pyrad -p params.txt -s 1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/usr/local/bioinformatics/muscle3.8.31_i86linux64\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "ls /usr/local/bioinformatics/muscle3.8.31_i86linux64"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Fixed muscle path in params.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "==** parameter inputs for pyRAD version 3.0.66  **======================== affected step ==\n",
      "./                     ## 1. Working directory                                 (all)\n",
      "/data/oly_gbs/*.gz             \t       ## 2. Loc. of non-demultiplexed files (if not line 18)  (s1)\n",
      "/analysis/20160609_pyrad/barcodes.txt\t               ## 3. Loc. of barcode file (if not line 18)             (s1)\n",
      "/usr/local/bioinformatics/vsearch-1.11.1-linux-x86_64/bin/vsearch                ## 4. command (or path) to call vsearch (or usearch)    (s3,s6)\n",
      "/usr/local/bioinformatics/muscle3.8.31_i86linux64/                    ## 5. command (or path) to call muscle                  (s3,s7)\n",
      "CWGC                   ## 6. Restriction overhang (e.g., C|TGCAG -> TGCAG)     (s1,s2)\n",
      "16                     ## 7. N processors (parallel)                           (all)\n",
      "6                      ## 8. Mindepth: min coverage for a cluster              (s4,s5)\n",
      "4                      ## 9. NQual: max # sites with qual < 20 (or see line 20)(s2)\n",
      ".88                    ## 10. Wclust: clustering threshold as a decimal        (s3,s6)\n",
      "pairgbs                 ## 11. Datatype: rad,gbs,pairgbs,pairddrad,(others:see docs)(all)\n",
      "4                      ## 12. MinCov: min samples in a final locus             (s7)\n",
      "3                      ## 13. MaxSH: max inds with shared hetero site          (s7)\n",
      "oly_gbs_pyrad          ## 14. Prefix name for final output (no spaces)         (s7)\n",
      "==== optional params below this line ===================================  affected step ==\n",
      "                       ## 15.opt.: select subset (prefix* only selector)            (s2-s7)\n",
      "                       ## 16.opt.: add-on (outgroup) taxa (list or prefix*)         (s6,s7)\n",
      "                       ## 17.opt.: exclude taxa (list or prefix*)                   (s7)\n",
      "                       ## 18.opt.: loc. of de-multiplexed data                      (s2)\n",
      "                       ## 19.opt.: maxM: N mismatches in barcodes (def= 1)          (s1)\n",
      "                       ## 20.opt.: phred Qscore offset (def= 33)                    (s2)\n",
      "2                      ## 21.opt.: filter: def=0=NQual 1=NQual+adapters. 2=strict   (s2)\n",
      "                       ## 22.opt.: a priori E,H (def= 0.001,0.01, if not estimated) (s5)\n",
      "4                      ## 23.opt.: maxN: max Ns in a cons seq (def=5)               (s5)\n",
      "8                      ## 24.opt.: maxH: max heterozyg. sites in cons seq (def=5)   (s5)\n",
      "                       ## 25.opt.: ploidy: max alleles in cons seq (def=2;see docs) (s4,s5)\n",
      "                       ## 26.opt.: maxSNPs: (def=100). Paired (def=100,100)         (s7)\n",
      "                       ## 27.opt.: maxIndels: within-clust,across-clust (def. 3,99) (s3,s7)\n",
      "                       ## 28.opt.: random number seed (def. 112233)              (s3,s6,s7)\n",
      "                       ## 29.opt.: trim overhang left,right on final loci, def(0,0) (s7)\n",
      "*                      ## 30.opt.: output formats: p,n,a,s,v,u,t,m,k,g,* (see docs) (s7)\n",
      "                       ## 31.opt.: maj. base call at depth>x<mindepth (def.x=mindepth) (s5)\n",
      "50                     ## 32.opt.: keep trimmed reads (def=0). Enter min length.    (s2)\n",
      "                       ## 33.opt.: max stack size (int), def= max(500,mean+2*SD)    (s3)\n",
      "                       ## 34.opt.: minDerep: exclude dereps with <= N copies, def=1 (s3)\n",
      "                       ## 35.opt.: use hierarchical clustering (def.=0, 1=yes)      (s6)\n",
      "                       ## 36.opt.: repeat masking (def.=1='dust' method, 0=no)      (s3,s6)\n",
      "16                     ## 37.opt.: vsearch max threads per job (def.=6; see docs)   (s3,s6)\n",
      "==== optional: list group/clade assignments below this line (see docs) ==================\n",
      "1HL\t5\t1HL*\n",
      "1NF\t5\t1NF*\n",
      "1SN\t5\t1SN*\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "cat params.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\tcannot find muscle, edit path in input file\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "     ------------------------------------------------------------\n",
      "      pyRAD : RADseq for phylogenetics & introgression analyses\n",
      "     ------------------------------------------------------------\n",
      "\n",
      "\n",
      "real\t0m0.358s\n",
      "user\t0m0.260s\n",
      "sys\t0m0.080s\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "time pyrad -p params.txt -s 1"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Fixed muscle path in params.txt (removed trailing slash)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\twarning: CUT site matches within one of the barcodes, I suggest double \n",
      "\tchecking the file to make sure it properly demultiplexes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "     ------------------------------------------------------------\n",
      "      pyRAD : RADseq for phylogenetics & introgression analyses\n",
      "     ------------------------------------------------------------\n",
      "\n",
      "\n",
      "\tstep 1: sorting reads by barcode\n",
      "\t .\n",
      "real\t1118m28.900s\n",
      "user\t945m13.910s\n",
      "sys\t42m11.240s\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "time pyrad -p params.txt -s 1"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Look at the first 100 lines of the stats output for demultiplexing"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "file    \tNreads\tcut_found\tbar_matched\n",
      "160123_I132_FCH3YHMBBXX_L4_OYSzenG1AAD96FAAPEI-109_.fq.gz\t364261046\t355811046\t354725421\n",
      "\n",
      "\n",
      "sample\ttrue_bar\tobs_bars\tN_obs\n",
      "1SN_27A    \tAAAAGTT    \tAAAAGTT\t3325108   \n",
      "1SN_27A    \tAAAAGTT    \tNAAAGTT\t8330      \n",
      "1SN_27A    \tAAAAGTT    \tAAAAATT\t4748      \n",
      "1SN_27A    \tAAAAGTT    \tANAAGTT\t1455      \n",
      "1SN_27A    \tAAAAGTT    \tAAAATTT\t782       \n",
      "1SN_27A    \tAAAAGTT    \tGAAAGTT\t739       \n",
      "1SN_27A    \tAAAAGTT    \tAAACGTT\t665       \n",
      "1SN_27A    \tAAAAGTT    \tCAAAGTT\t617       \n",
      "1SN_27A    \tAAAAGTT    \tAACAGTT\t592       \n",
      "1SN_27A    \tAAAAGTT    \tTAAAGTT\t558       \n",
      "1SN_27A    \tAAAAGTT    \tAAAAGCT\t351       \n",
      "1SN_27A    \tAAAAGTT    \tACAAGTT\t320       \n",
      "1SN_27A    \tAAAAGTT    \tAAAGGTT\t284       \n",
      "1SN_27A    \tAAAAGTT    \tAGAAGTT\t210       \n",
      "1SN_27A    \tAAAAGTT    \tAAAAGTC\t201       \n",
      "1SN_27A    \tAAAAGTT    \tAAGAGTT\t176       \n",
      "1SN_27A    \tAAAAGTT    \tAAANGTT\t127       \n",
      "1SN_27A    \tAAAAGTT    \tAAAAGGT\t106       \n",
      "1SN_27A    \tAAAAGTT    \tAAAAGAT\t94        \n",
      "1SN_27A    \tAAAAGTT    \tATAAGTT\t94        \n",
      "1SN_27A    \tAAAAGTT    \tAATAGTT\t92        \n",
      "1SN_27A    \tAAAAGTT    \tAAATGTT\t80        \n",
      "1SN_27A    \tAAAAGTT    \tAAAACTT\t42        \n",
      "1SN_27A    \tAAAAGTT    \tAANAGTT\t37        \n",
      "1SN_27A    \tAAAAGTT    \tAAAAGTG\t29        \n",
      "1SN_27A    \tAAAAGTT    \tAAAAGTA\t27        \n",
      "1SN_27A    \tAAAAGTT    \tAAAAGNT\t11        \n",
      "1HL_23A    \tAACCGAGA    \tAACCGAGA\t4532975   \n",
      "1HL_23A    \tAACCGAGA    \tNACCGAGA\t11670     \n",
      "1HL_23A    \tAACCGAGA    \tAGCCGAGA\t3232      \n",
      "1HL_23A    \tAACCGAGA    \tTACCGAGA\t2503      \n",
      "1HL_23A    \tAACCGAGA    \tANCCGAGA\t2022      \n",
      "1HL_23A    \tAACCGAGA    \tAACCAAGA\t1624      \n",
      "1HL_23A    \tAACCGAGA    \tGACCGAGA\t1595      \n",
      "1HL_23A    \tAACCGAGA    \tAACCTAGA\t1463      \n",
      "1HL_23A    \tAACCGAGA    \tAATCGAGA\t1328      \n",
      "1HL_23A    \tAACCGAGA    \tCACCGAGA\t1183      \n",
      "1HL_23A    \tAACCGAGA    \tAACTGAGA\t1147      \n",
      "1HL_23A    \tAACCGAGA    \tAACCGCGA\t1040      \n",
      "1HL_23A    \tAACCGAGA    \tACCCGAGA\t721       \n",
      "1HL_23A    \tAACCGAGA    \tAACCGAGC\t679       \n",
      "1HL_23A    \tAACCGAGA    \tAACCGGGA\t486       \n",
      "1HL_23A    \tAACCGAGA    \tAACCGAAA\t427       \n",
      "1HL_23A    \tAACCGAGA    \tAACCGAGT\t415       \n",
      "1HL_23A    \tAACCGAGA    \tAACAGAGA\t398       \n",
      "1HL_23A    \tAACCGAGA    \tAACCGAGG\t375       \n",
      "1HL_23A    \tAACCGAGA    \tAACCGTGA\t342       \n",
      "1HL_23A    \tAACCGAGA    \tAACCGATA\t293       \n",
      "1HL_23A    \tAACCGAGA    \tAAACGAGA\t232       \n",
      "1HL_23A    \tAACCGAGA    \tAACNGAGA\t214       \n",
      "1HL_23A    \tAACCGAGA    \tATCCGAGA\t187       \n",
      "1HL_23A    \tAACCGAGA    \tAACGGAGA\t137       \n",
      "1HL_23A    \tAACCGAGA    \tAACCCAGA\t95        \n",
      "1HL_23A    \tAACCGAGA    \tAAGCGAGA\t92        \n",
      "1HL_23A    \tAACCGAGA    \tAANCGAGA\t54        \n",
      "1HL_23A    \tAACCGAGA    \tAACCGACA\t40        \n",
      "1HL_23A    \tAACCGAGA    \tAACCGNGA\t20        \n",
      "1SN_32A    \tAACGCCT    \tAACGCCT\t3274859   \n",
      "1SN_32A    \tAACGCCT    \tNACGCCT\t8278      \n",
      "1SN_32A    \tAACGCCT    \tTACGCCT\t2642      \n",
      "1SN_32A    \tAACGCCT    \tGACGCCT\t2577      \n",
      "1SN_32A    \tAACGCCT    \tANCGCCT\t1463      \n",
      "1SN_32A    \tAACGCCT    \tAACTCCT\t1421      \n",
      "1SN_32A    \tAACGCCT    \tAACACCT\t1005      \n",
      "1SN_32A    \tAACGCCT    \tATCGCCT\t958       \n",
      "1SN_32A    \tAACGCCT    \tAATGCCT\t860       \n",
      "1SN_32A    \tAACGCCT    \tACCGCCT\t856       \n",
      "1SN_32A    \tAACGCCT    \tAGCGCCT\t732       \n",
      "1SN_32A    \tAACGCCT    \tCACGCCT\t604       \n",
      "1SN_32A    \tAACGCCT    \tAACGTCT\t434       \n",
      "1SN_32A    \tAACGCCT    \tAACGCTT\t430       \n",
      "1SN_32A    \tAACGCCT    \tAACGCCC\t381       \n",
      "1SN_32A    \tAACGCCT    \tAACGCAT\t290       \n",
      "1SN_32A    \tAACGCCT    \tAACGACT\t229       \n",
      "1SN_32A    \tAACGCCT    \tAACGGCT\t170       \n",
      "1SN_32A    \tAACGCCT    \tAACNCCT\t160       \n",
      "1SN_32A    \tAACGCCT    \tAAAGCCT\t141       \n",
      "1SN_32A    \tAACGCCT    \tAACCCCT\t140       \n",
      "1SN_32A    \tAACGCCT    \tAACGCGT\t107       \n",
      "1SN_32A    \tAACGCCT    \tAAGGCCT\t94        \n",
      "1SN_32A    \tAACGCCT    \tAACGCCA\t79        \n",
      "1SN_32A    \tAACGCCT    \tAACGCCG\t45        \n",
      "1SN_32A    \tAACGCCT    \tAANGCCT\t28        \n",
      "1SN_32A    \tAACGCCT    \tAACGCNT\t13        \n",
      "1NF_6A    \tAACT    \tAACT\t3192263   \n",
      "1NF_6A    \tAACT    \tNACT\t7691      \n",
      "1NF_6A    \tAACT    \tANCT\t1424      \n",
      "1NF_6A    \tAACT    \tAGCT\t963       \n",
      "1NF_6A    \tAACT    \tTACT\t873       \n",
      "1NF_6A    \tAACT    \tGACT\t791       \n",
      "1NF_6A    \tAACT    \tCACT\t717       \n",
      "1NF_6A    \tAACT    \tATCT\t460       \n",
      "1NF_6A    \tAACT    \tAACC\t417       \n",
      "1NF_6A    \tAACT    \tACCT\t390       \n",
      "1NF_6A    \tAACT    \tAAGT\t289       \n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "head -n 100 stats/s1.sorting.txt"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Merge paired reads"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "For this we will use the program PEAR. This step is very important. I can pretty much assure you that you will have many paired end reads in your library that overlap, and if you do not merge these reads they can really mess up your analysis. In this case the vast majority of paired reads are overlapping (~80%), and in fact, the merged reads are the only portion of the data that we will be using for analysis."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The first thing we have to do is unzip the reads."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "real\t73m48.470s\n",
      "user\t18m41.580s\n",
      "sys\t20m13.890s\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "time gunzip ./fastq/*.gz"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": true
   },
   "source": [
    "#### Restarted VirtualBox/Docker with updated CPU/RAM settings"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Architecture:          x86_64\n",
      "CPU op-mode(s):        32-bit, 64-bit\n",
      "Byte Order:            Little Endian\n",
      "CPU(s):                8\n",
      "On-line CPU(s) list:   0-7\n",
      "Thread(s) per core:    1\n",
      "Core(s) per socket:    8\n",
      "Socket(s):             1\n",
      "Vendor ID:             GenuineIntel\n",
      "CPU family:            6\n",
      "Model:                 26\n",
      "Model name:            Intel(R) Xeon(R) CPU           E5520  @ 2.27GHz\n",
      "Stepping:              5\n",
      "CPU MHz:               2260.998\n",
      "BogoMIPS:              4521.99\n",
      "Hypervisor vendor:     KVM\n",
      "Virtualization type:   full\n",
      "L1d cache:             32K\n",
      "L1i cache:             32K\n",
      "L2 cache:              256K\n",
      "L3 cache:              8192K\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "lscpu"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/analysis/20160609_pyrad\n"
     ]
    }
   ],
   "source": [
    "cd /analysis/20160609_pyrad/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Analyze merged reads"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1HL_10A_R1.fq  1HL_29A_R2.fq  1NF_17A_R1.fq  1NF_4A_R2.fq   1SN_23A_R1.fq\r\n",
      "1HL_10A_R2.fq  1HL_2A_R1.fq   1NF_17A_R2.fq  1NF_5A_R1.fq   1SN_23A_R2.fq\r\n",
      "1HL_11A_R1.fq  1HL_2A_R2.fq   1NF_18A_R1.fq  1NF_5A_R2.fq   1SN_24A_R1.fq\r\n",
      "1HL_11A_R2.fq  1HL_31A_R1.fq  1NF_18A_R2.fq  1NF_6A_R1.fq   1SN_24A_R2.fq\r\n",
      "1HL_12A_R1.fq  1HL_31A_R2.fq  1NF_19A_R1.fq  1NF_6A_R2.fq   1SN_25A_R1.fq\r\n",
      "1HL_12A_R2.fq  1HL_33A_R1.fq  1NF_19A_R2.fq  1NF_7A_R1.fq   1SN_25A_R2.fq\r\n",
      "1HL_13A_R1.fq  1HL_33A_R2.fq  1NF_1A_R1.fq   1NF_7A_R2.fq   1SN_26A_R1.fq\r\n",
      "1HL_13A_R2.fq  1HL_34A_R1.fq  1NF_1A_R2.fq   1NF_8A_R1.fq   1SN_26A_R2.fq\r\n",
      "1HL_14A_R1.fq  1HL_34A_R2.fq  1NF_20A_R1.fq  1NF_8A_R2.fq   1SN_27A_R1.fq\r\n",
      "1HL_14A_R2.fq  1HL_35A_R1.fq  1NF_20A_R2.fq  1NF_9A_R1.fq   1SN_27A_R2.fq\r\n",
      "1HL_15A_R1.fq  1HL_35A_R2.fq  1NF_21A_R1.fq  1NF_9A_R2.fq   1SN_28A_R1.fq\r\n",
      "1HL_15A_R2.fq  1HL_3A_R1.fq   1NF_21A_R2.fq  1SN_10A_R1.fq  1SN_28A_R2.fq\r\n",
      "1HL_16A_R1.fq  1HL_3A_R2.fq   1NF_22A_R1.fq  1SN_10A_R2.fq  1SN_29A_R1.fq\r\n",
      "1HL_16A_R2.fq  1HL_4A_R1.fq   1NF_22A_R2.fq  1SN_11A_R1.fq  1SN_29A_R2.fq\r\n",
      "1HL_17A_R1.fq  1HL_4A_R2.fq   1NF_23A_R1.fq  1SN_11A_R2.fq  1SN_2A_R1.fq\r\n",
      "1HL_17A_R2.fq  1HL_5A_R1.fq   1NF_23A_R2.fq  1SN_12A_R1.fq  1SN_2A_R2.fq\r\n",
      "1HL_19A_R1.fq  1HL_5A_R2.fq   1NF_24A_R1.fq  1SN_12A_R2.fq  1SN_30A_R1.fq\r\n",
      "1HL_19A_R2.fq  1HL_6A_R1.fq   1NF_24A_R2.fq  1SN_13A_R1.fq  1SN_30A_R2.fq\r\n",
      "1HL_1A_R1.fq   1HL_6A_R2.fq   1NF_25A_R1.fq  1SN_13A_R2.fq  1SN_31A_R1.fq\r\n",
      "1HL_1A_R2.fq   1HL_7A_R1.fq   1NF_25A_R2.fq  1SN_14A_R1.fq  1SN_31A_R2.fq\r\n",
      "1HL_20A_R1.fq  1HL_7A_R2.fq   1NF_26A_R1.fq  1SN_14A_R2.fq  1SN_32A_R1.fq\r\n",
      "1HL_20A_R2.fq  1HL_8A_R1.fq   1NF_26A_R2.fq  1SN_15A_R1.fq  1SN_32A_R2.fq\r\n",
      "1HL_21A_R1.fq  1HL_8A_R2.fq   1NF_27A_R1.fq  1SN_15A_R2.fq  1SN_3A_R1.fq\r\n",
      "1HL_21A_R2.fq  1HL_9A_R1.fq   1NF_27A_R2.fq  1SN_16A_R1.fq  1SN_3A_R2.fq\r\n",
      "1HL_22A_R1.fq  1HL_9A_R2.fq   1NF_28A_R1.fq  1SN_16A_R2.fq  1SN_4A_R1.fq\r\n",
      "1HL_22A_R2.fq  1NF_10A_R1.fq  1NF_28A_R2.fq  1SN_17A_R1.fq  1SN_4A_R2.fq\r\n",
      "1HL_23A_R1.fq  1NF_10A_R2.fq  1NF_29A_R1.fq  1SN_17A_R2.fq  1SN_5A_R1.fq\r\n",
      "1HL_23A_R2.fq  1NF_11A_R1.fq  1NF_29A_R2.fq  1SN_18A_R1.fq  1SN_5A_R2.fq\r\n",
      "1HL_24A_R1.fq  1NF_11A_R2.fq  1NF_2A_R1.fq   1SN_18A_R2.fq  1SN_6A_R1.fq\r\n",
      "1HL_24A_R2.fq  1NF_12A_R1.fq  1NF_2A_R2.fq   1SN_19A_R1.fq  1SN_6A_R2.fq\r\n",
      "1HL_25A_R1.fq  1NF_12A_R2.fq  1NF_30A_R1.fq  1SN_19A_R2.fq  1SN_7A_R1.fq\r\n",
      "1HL_25A_R2.fq  1NF_13A_R1.fq  1NF_30A_R2.fq  1SN_1A_R1.fq   1SN_7A_R2.fq\r\n",
      "1HL_26A_R1.fq  1NF_13A_R2.fq  1NF_31A_R1.fq  1SN_1A_R2.fq   1SN_8A_R1.fq\r\n",
      "1HL_26A_R2.fq  1NF_14A_R1.fq  1NF_31A_R2.fq  1SN_20A_R1.fq  1SN_8A_R2.fq\r\n",
      "1HL_27A_R1.fq  1NF_14A_R2.fq  1NF_32A_R1.fq  1SN_20A_R2.fq  1SN_9A_R1.fq\r\n",
      "1HL_27A_R2.fq  1NF_15A_R1.fq  1NF_32A_R2.fq  1SN_21A_R1.fq  1SN_9A_R2.fq\r\n",
      "1HL_28A_R1.fq  1NF_15A_R2.fq  1NF_33A_R1.fq  1SN_21A_R2.fq\r\n",
      "1HL_28A_R2.fq  1NF_16A_R1.fq  1NF_33A_R2.fq  1SN_22A_R1.fq\r\n",
      "1HL_29A_R1.fq  1NF_16A_R2.fq  1NF_4A_R1.fq   1SN_22A_R2.fq\r\n"
     ]
    }
   ],
   "source": [
    "ls /analysis/20160609_pyrad/fastq/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "%%bash\n",
    "time for gfile in /analysis/20160609_pyrad/fastq/*_R1.fq;\n",
    "    do /usr/local/bioinformatics/pear-0.9.6-bin-64/pear-0.9.6-bin-64 -f $gfile \\\n",
    "            -r ${gfile/_R1.fq/_R2.fq} \\\n",
    "            -o ${gfile/_R1.fq/} \\\n",
    "            -n 33 \\\n",
    "            -t 33 \\\n",
    "            -q 10 \\\n",
    "            -j 20  >> pear.log 2>&1;\n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": true
   },
   "source": [
    "#### Roadrunner locked up - rebooted"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Thu Jun 16 20:27:00 UTC 2016\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "date"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/analysis/20160609_pyrad\n"
     ]
    }
   ],
   "source": [
    "cd /analysis/20160609_pyrad/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "%%bash\n",
    "time for gfile in /analysis/20160609_pyrad/fastq/*_R1.fq;\n",
    "    do /usr/local/bioinformatics/pear-0.9.6-bin-64/pear-0.9.6-bin-64 -f $gfile \\\n",
    "            -r ${gfile/_R1.fq/_R2.fq} \\\n",
    "            -o ${gfile/_R1.fq/} \\\n",
    "            -n 33 \\\n",
    "            -t 33 \\\n",
    "            -q 10 \\\n",
    "            -j 20  >> pear.log 2>&1;\n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": true
   },
   "source": [
    "#### My computer crashed and roadrunner locked up - rebooted both"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/analysis/20160609_pyrad\n"
     ]
    }
   ],
   "source": [
    "cd /analysis/20160609_pyrad/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Check output files and pear.log to see if previous step finished"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "barcodes.txt  \u001b[0m\u001b[01;34mfastq\u001b[0m/  params.txt  pear.log  \u001b[01;34mstats\u001b[0m/\r\n"
     ]
    }
   ],
   "source": [
    "ls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/analysis/20160609_pyrad/fastq\n"
     ]
    }
   ],
   "source": [
    "cd fastq/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1HL_10A.assembled.fastq            1NF_11A_R1.fq\r\n",
      "1HL_10A.discarded.fastq            1NF_11A_R2.fq\r\n",
      "1HL_10A.unassembled.forward.fastq  1NF_12A.assembled.fastq\r\n",
      "1HL_10A.unassembled.reverse.fastq  1NF_12A.discarded.fastq\r\n",
      "1HL_10A_R1.fq                      1NF_12A.unassembled.forward.fastq\r\n",
      "1HL_10A_R2.fq                      1NF_12A.unassembled.reverse.fastq\r\n",
      "1HL_11A.assembled.fastq            1NF_12A_R1.fq\r\n",
      "1HL_11A.discarded.fastq            1NF_12A_R2.fq\r\n",
      "1HL_11A.unassembled.forward.fastq  1NF_13A.assembled.fastq\r\n",
      "1HL_11A.unassembled.reverse.fastq  1NF_13A.discarded.fastq\r\n",
      "1HL_11A_R1.fq                      1NF_13A.unassembled.forward.fastq\r\n",
      "1HL_11A_R2.fq                      1NF_13A.unassembled.reverse.fastq\r\n",
      "1HL_12A.assembled.fastq            1NF_13A_R1.fq\r\n",
      "1HL_12A.discarded.fastq            1NF_13A_R2.fq\r\n",
      "1HL_12A.unassembled.forward.fastq  1NF_14A.assembled.fastq\r\n",
      "1HL_12A.unassembled.reverse.fastq  1NF_14A.discarded.fastq\r\n",
      "1HL_12A_R1.fq                      1NF_14A.unassembled.forward.fastq\r\n",
      "1HL_12A_R2.fq                      1NF_14A.unassembled.reverse.fastq\r\n",
      "1HL_13A.assembled.fastq            1NF_14A_R1.fq\r\n",
      "1HL_13A.discarded.fastq            1NF_14A_R2.fq\r\n",
      "1HL_13A.unassembled.forward.fastq  1NF_15A.assembled.fastq\r\n",
      "1HL_13A.unassembled.reverse.fastq  1NF_15A.discarded.fastq\r\n",
      "1HL_13A_R1.fq                      1NF_15A.unassembled.forward.fastq\r\n",
      "1HL_13A_R2.fq                      1NF_15A.unassembled.reverse.fastq\r\n",
      "1HL_14A.assembled.fastq            1NF_15A_R1.fq\r\n",
      "1HL_14A.discarded.fastq            1NF_15A_R2.fq\r\n",
      "1HL_14A.unassembled.forward.fastq  1NF_16A.assembled.fastq\r\n",
      "1HL_14A.unassembled.reverse.fastq  1NF_16A.discarded.fastq\r\n",
      "1HL_14A_R1.fq                      1NF_16A.unassembled.forward.fastq\r\n",
      "1HL_14A_R2.fq                      1NF_16A.unassembled.reverse.fastq\r\n",
      "1HL_15A.assembled.fastq            1NF_16A_R1.fq\r\n",
      "1HL_15A.discarded.fastq            1NF_16A_R2.fq\r\n",
      "1HL_15A.unassembled.forward.fastq  1NF_17A.assembled.fastq\r\n",
      "1HL_15A.unassembled.reverse.fastq  1NF_17A.discarded.fastq\r\n",
      "1HL_15A_R1.fq                      1NF_17A.unassembled.forward.fastq\r\n",
      "1HL_15A_R2.fq                      1NF_17A.unassembled.reverse.fastq\r\n",
      "1HL_16A.assembled.fastq            1NF_17A_R1.fq\r\n",
      "1HL_16A.discarded.fastq            1NF_17A_R2.fq\r\n",
      "1HL_16A.unassembled.forward.fastq  1NF_18A.assembled.fastq\r\n",
      "1HL_16A.unassembled.reverse.fastq  1NF_18A.discarded.fastq\r\n",
      "1HL_16A_R1.fq                      1NF_18A.unassembled.forward.fastq\r\n",
      "1HL_16A_R2.fq                      1NF_18A.unassembled.reverse.fastq\r\n",
      "1HL_17A.assembled.fastq            1NF_18A_R1.fq\r\n",
      "1HL_17A.discarded.fastq            1NF_18A_R2.fq\r\n",
      "1HL_17A.unassembled.forward.fastq  1NF_19A.assembled.fastq\r\n",
      "1HL_17A.unassembled.reverse.fastq  1NF_19A.discarded.fastq\r\n",
      "1HL_17A_R1.fq                      1NF_19A.unassembled.forward.fastq\r\n",
      "1HL_17A_R2.fq                      1NF_19A.unassembled.reverse.fastq\r\n",
      "1HL_19A.assembled.fastq            1NF_19A_R1.fq\r\n",
      "1HL_19A.discarded.fastq            1NF_19A_R2.fq\r\n",
      "1HL_19A.unassembled.forward.fastq  1NF_1A.assembled.fastq\r\n",
      "1HL_19A.unassembled.reverse.fastq  1NF_1A.discarded.fastq\r\n",
      "1HL_19A_R1.fq                      1NF_1A.unassembled.forward.fastq\r\n",
      "1HL_19A_R2.fq                      1NF_1A.unassembled.reverse.fastq\r\n",
      "1HL_1A.assembled.fastq             1NF_1A_R1.fq\r\n",
      "1HL_1A.discarded.fastq             1NF_1A_R2.fq\r\n",
      "1HL_1A.unassembled.forward.fastq   1NF_20A.assembled.fastq\r\n",
      "1HL_1A.unassembled.reverse.fastq   1NF_20A.discarded.fastq\r\n",
      "1HL_1A_R1.fq                       1NF_20A.unassembled.forward.fastq\r\n",
      "1HL_1A_R2.fq                       1NF_20A.unassembled.reverse.fastq\r\n",
      "1HL_20A.assembled.fastq            1NF_20A_R1.fq\r\n",
      "1HL_20A.discarded.fastq            1NF_20A_R2.fq\r\n",
      "1HL_20A.unassembled.forward.fastq  1NF_21A.assembled.fastq\r\n",
      "1HL_20A.unassembled.reverse.fastq  1NF_21A.discarded.fastq\r\n",
      "1HL_20A_R1.fq                      1NF_21A.unassembled.forward.fastq\r\n",
      "1HL_20A_R2.fq                      1NF_21A.unassembled.reverse.fastq\r\n",
      "1HL_21A.assembled.fastq            1NF_21A_R1.fq\r\n",
      "1HL_21A.discarded.fastq            1NF_21A_R2.fq\r\n",
      "1HL_21A.unassembled.forward.fastq  1NF_22A.assembled.fastq\r\n",
      "1HL_21A.unassembled.reverse.fastq  1NF_22A.discarded.fastq\r\n",
      "1HL_21A_R1.fq                      1NF_22A.unassembled.forward.fastq\r\n",
      "1HL_21A_R2.fq                      1NF_22A.unassembled.reverse.fastq\r\n",
      "1HL_22A.assembled.fastq            1NF_22A_R1.fq\r\n",
      "1HL_22A.discarded.fastq            1NF_22A_R2.fq\r\n",
      "1HL_22A.unassembled.forward.fastq  1NF_23A.assembled.fastq\r\n",
      "1HL_22A.unassembled.reverse.fastq  1NF_23A.discarded.fastq\r\n",
      "1HL_22A_R1.fq                      1NF_23A.unassembled.forward.fastq\r\n",
      "1HL_22A_R2.fq                      1NF_23A.unassembled.reverse.fastq\r\n",
      "1HL_23A.assembled.fastq            1NF_23A_R1.fq\r\n",
      "1HL_23A.discarded.fastq            1NF_23A_R2.fq\r\n",
      "1HL_23A.unassembled.forward.fastq  1NF_24A.assembled.fastq\r\n",
      "1HL_23A.unassembled.reverse.fastq  1NF_24A.discarded.fastq\r\n",
      "1HL_23A_R1.fq                      1NF_24A.unassembled.forward.fastq\r\n",
      "1HL_23A_R2.fq                      1NF_24A.unassembled.reverse.fastq\r\n",
      "1HL_24A.assembled.fastq            1NF_24A_R1.fq\r\n",
      "1HL_24A.discarded.fastq            1NF_24A_R2.fq\r\n",
      "1HL_24A.unassembled.forward.fastq  1NF_25A.assembled.fastq\r\n",
      "1HL_24A.unassembled.reverse.fastq  1NF_25A.discarded.fastq\r\n",
      "1HL_24A_R1.fq                      1NF_25A.unassembled.forward.fastq\r\n",
      "1HL_24A_R2.fq                      1NF_25A.unassembled.reverse.fastq\r\n",
      "1HL_25A.assembled.fastq            1NF_25A_R1.fq\r\n",
      "1HL_25A.discarded.fastq            1NF_25A_R2.fq\r\n",
      "1HL_25A.unassembled.forward.fastq  1NF_26A.assembled.fastq\r\n",
      "1HL_25A.unassembled.reverse.fastq  1NF_26A.discarded.fastq\r\n",
      "1HL_25A_R1.fq                      1NF_26A.unassembled.forward.fastq\r\n",
      "1HL_25A_R2.fq                      1NF_26A.unassembled.reverse.fastq\r\n",
      "1HL_26A.assembled.fastq            1NF_26A_R1.fq\r\n",
      "1HL_26A.discarded.fastq            1NF_26A_R2.fq\r\n",
      "1HL_26A.unassembled.forward.fastq  1NF_27A.assembled.fastq\r\n",
      "1HL_26A.unassembled.reverse.fastq  1NF_27A.discarded.fastq\r\n",
      "1HL_26A_R1.fq                      1NF_27A.unassembled.forward.fastq\r\n",
      "1HL_26A_R2.fq                      1NF_27A.unassembled.reverse.fastq\r\n",
      "1HL_27A.assembled.fastq            1NF_27A_R1.fq\r\n",
      "1HL_27A.discarded.fastq            1NF_27A_R2.fq\r\n",
      "1HL_27A.unassembled.forward.fastq  1NF_28A.assembled.fastq\r\n",
      "1HL_27A.unassembled.reverse.fastq  1NF_28A.discarded.fastq\r\n",
      "1HL_27A_R1.fq                      1NF_28A.unassembled.forward.fastq\r\n",
      "1HL_27A_R2.fq                      1NF_28A.unassembled.reverse.fastq\r\n",
      "1HL_28A.assembled.fastq            1NF_28A_R1.fq\r\n",
      "1HL_28A.discarded.fastq            1NF_28A_R2.fq\r\n",
      "1HL_28A.unassembled.forward.fastq  1NF_29A.assembled.fastq\r\n",
      "1HL_28A.unassembled.reverse.fastq  1NF_29A.discarded.fastq\r\n",
      "1HL_28A_R1.fq                      1NF_29A.unassembled.forward.fastq\r\n",
      "1HL_28A_R2.fq                      1NF_29A.unassembled.reverse.fastq\r\n",
      "1HL_29A.assembled.fastq            1NF_29A_R1.fq\r\n",
      "1HL_29A.discarded.fastq            1NF_29A_R2.fq\r\n",
      "1HL_29A.unassembled.forward.fastq  1NF_2A_R1.fq\r\n",
      "1HL_29A.unassembled.reverse.fastq  1NF_2A_R2.fq\r\n",
      "1HL_29A_R1.fq                      1NF_30A_R1.fq\r\n",
      "1HL_29A_R2.fq                      1NF_30A_R2.fq\r\n",
      "1HL_2A.assembled.fastq             1NF_31A_R1.fq\r\n",
      "1HL_2A.discarded.fastq             1NF_31A_R2.fq\r\n",
      "1HL_2A.unassembled.forward.fastq   1NF_32A_R1.fq\r\n",
      "1HL_2A.unassembled.reverse.fastq   1NF_32A_R2.fq\r\n",
      "1HL_2A_R1.fq                       1NF_33A_R1.fq\r\n",
      "1HL_2A_R2.fq                       1NF_33A_R2.fq\r\n",
      "1HL_31A.assembled.fastq            1NF_4A_R1.fq\r\n",
      "1HL_31A.discarded.fastq            1NF_4A_R2.fq\r\n",
      "1HL_31A.unassembled.forward.fastq  1NF_5A_R1.fq\r\n",
      "1HL_31A.unassembled.reverse.fastq  1NF_5A_R2.fq\r\n",
      "1HL_31A_R1.fq                      1NF_6A_R1.fq\r\n",
      "1HL_31A_R2.fq                      1NF_6A_R2.fq\r\n",
      "1HL_33A.assembled.fastq            1NF_7A_R1.fq\r\n",
      "1HL_33A.discarded.fastq            1NF_7A_R2.fq\r\n",
      "1HL_33A.unassembled.forward.fastq  1NF_8A_R1.fq\r\n",
      "1HL_33A.unassembled.reverse.fastq  1NF_8A_R2.fq\r\n",
      "1HL_33A_R1.fq                      1NF_9A_R1.fq\r\n",
      "1HL_33A_R2.fq                      1NF_9A_R2.fq\r\n",
      "1HL_34A.assembled.fastq            1SN_10A_R1.fq\r\n",
      "1HL_34A.discarded.fastq            1SN_10A_R2.fq\r\n",
      "1HL_34A.unassembled.forward.fastq  1SN_11A_R1.fq\r\n",
      "1HL_34A.unassembled.reverse.fastq  1SN_11A_R2.fq\r\n",
      "1HL_34A_R1.fq                      1SN_12A_R1.fq\r\n",
      "1HL_34A_R2.fq                      1SN_12A_R2.fq\r\n",
      "1HL_35A.assembled.fastq            1SN_13A_R1.fq\r\n",
      "1HL_35A.discarded.fastq            1SN_13A_R2.fq\r\n",
      "1HL_35A.unassembled.forward.fastq  1SN_14A_R1.fq\r\n",
      "1HL_35A.unassembled.reverse.fastq  1SN_14A_R2.fq\r\n",
      "1HL_35A_R1.fq                      1SN_15A_R1.fq\r\n",
      "1HL_35A_R2.fq                      1SN_15A_R2.fq\r\n",
      "1HL_3A.assembled.fastq             1SN_16A_R1.fq\r\n",
      "1HL_3A.discarded.fastq             1SN_16A_R2.fq\r\n",
      "1HL_3A.unassembled.forward.fastq   1SN_17A_R1.fq\r\n",
      "1HL_3A.unassembled.reverse.fastq   1SN_17A_R2.fq\r\n",
      "1HL_3A_R1.fq                       1SN_18A_R1.fq\r\n",
      "1HL_3A_R2.fq                       1SN_18A_R2.fq\r\n",
      "1HL_4A.assembled.fastq             1SN_19A_R1.fq\r\n",
      "1HL_4A.discarded.fastq             1SN_19A_R2.fq\r\n",
      "1HL_4A.unassembled.forward.fastq   1SN_1A_R1.fq\r\n",
      "1HL_4A.unassembled.reverse.fastq   1SN_1A_R2.fq\r\n",
      "1HL_4A_R1.fq                       1SN_20A_R1.fq\r\n",
      "1HL_4A_R2.fq                       1SN_20A_R2.fq\r\n",
      "1HL_5A.assembled.fastq             1SN_21A_R1.fq\r\n",
      "1HL_5A.discarded.fastq             1SN_21A_R2.fq\r\n",
      "1HL_5A.unassembled.forward.fastq   1SN_22A_R1.fq\r\n",
      "1HL_5A.unassembled.reverse.fastq   1SN_22A_R2.fq\r\n",
      "1HL_5A_R1.fq                       1SN_23A_R1.fq\r\n",
      "1HL_5A_R2.fq                       1SN_23A_R2.fq\r\n",
      "1HL_6A.assembled.fastq             1SN_24A_R1.fq\r\n",
      "1HL_6A.discarded.fastq             1SN_24A_R2.fq\r\n",
      "1HL_6A.unassembled.forward.fastq   1SN_25A_R1.fq\r\n",
      "1HL_6A.unassembled.reverse.fastq   1SN_25A_R2.fq\r\n",
      "1HL_6A_R1.fq                       1SN_26A_R1.fq\r\n",
      "1HL_6A_R2.fq                       1SN_26A_R2.fq\r\n",
      "1HL_7A.assembled.fastq             1SN_27A_R1.fq\r\n",
      "1HL_7A.discarded.fastq             1SN_27A_R2.fq\r\n",
      "1HL_7A.unassembled.forward.fastq   1SN_28A_R1.fq\r\n",
      "1HL_7A.unassembled.reverse.fastq   1SN_28A_R2.fq\r\n",
      "1HL_7A_R1.fq                       1SN_29A_R1.fq\r\n",
      "1HL_7A_R2.fq                       1SN_29A_R2.fq\r\n",
      "1HL_8A.assembled.fastq             1SN_2A_R1.fq\r\n",
      "1HL_8A.discarded.fastq             1SN_2A_R2.fq\r\n",
      "1HL_8A.unassembled.forward.fastq   1SN_30A_R1.fq\r\n",
      "1HL_8A.unassembled.reverse.fastq   1SN_30A_R2.fq\r\n",
      "1HL_8A_R1.fq                       1SN_31A_R1.fq\r\n",
      "1HL_8A_R2.fq                       1SN_31A_R2.fq\r\n",
      "1HL_9A.assembled.fastq             1SN_32A_R1.fq\r\n",
      "1HL_9A.discarded.fastq             1SN_32A_R2.fq\r\n",
      "1HL_9A.unassembled.forward.fastq   1SN_3A_R1.fq\r\n",
      "1HL_9A.unassembled.reverse.fastq   1SN_3A_R2.fq\r\n",
      "1HL_9A_R1.fq                       1SN_4A_R1.fq\r\n",
      "1HL_9A_R2.fq                       1SN_4A_R2.fq\r\n",
      "1NF_10A.assembled.fastq            1SN_5A_R1.fq\r\n",
      "1NF_10A.discarded.fastq            1SN_5A_R2.fq\r\n",
      "1NF_10A.unassembled.forward.fastq  1SN_6A_R1.fq\r\n",
      "1NF_10A.unassembled.reverse.fastq  1SN_6A_R2.fq\r\n",
      "1NF_10A_R1.fq                      1SN_7A_R1.fq\r\n",
      "1NF_10A_R2.fq                      1SN_7A_R2.fq\r\n",
      "1NF_11A.assembled.fastq            1SN_8A_R1.fq\r\n",
      "1NF_11A.discarded.fastq            1SN_8A_R2.fq\r\n",
      "1NF_11A.unassembled.forward.fastq  1SN_9A_R1.fq\r\n",
      "1NF_11A.unassembled.reverse.fastq  1SN_9A_R2.fq\r\n"
     ]
    }
   ],
   "source": [
    "ls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/analysis/20160609_pyrad\n"
     ]
    }
   ],
   "source": [
    "cd .."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "barcodes.txt  \u001b[0m\u001b[01;34mfastq\u001b[0m/  params.txt  pear.log  \u001b[01;34mstats\u001b[0m/\r\n"
     ]
    }
   ],
   "source": [
    "ls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "bash: line 2: pear: command not found\n",
      "bash: line 2: pear: command not found\n",
      "bash: line 2: pear: command not found\n",
      "bash: line 2: pear: command not found\n",
      "bash: line 2: pear: command not found\n",
      "bash: line 2: pear: command not found\n",
      "bash: line 2: pear: command not found\n",
      "bash: line 2: pear: command not found\n",
      "bash: line 2: pear: command not found\n",
      "bash: line 2: pear: command not found\n",
      "bash: line 2: pear: command not found\n",
      "bash: line 2: pear: command not found\n",
      "bash: line 2: pear: command not found\n",
      "bash: line 2: pear: command not found\n",
      "bash: line 2: pear: command not found\n",
      "bash: line 2: pear: command not found\n",
      "bash: line 2: pear: command not found\n",
      "bash: line 2: pear: command not found\n",
      "bash: line 2: pear: command not found\n",
      "bash: line 2: pear: command not found\n",
      "bash: line 2: pear: command not found\n",
      "bash: line 2: pear: command not found\n",
      "bash: line 2: pear: command not found\n",
      "bash: line 2: pear: command not found\n",
      "bash: line 2: pear: command not found\n",
      "bash: line 2: pear: command not found\n",
      "bash: line 2: pear: command not found\n",
      "bash: line 2: pear: command not found\n",
      "bash: line 2: pear: command not found\n",
      "bash: line 2: pear: command not found\n",
      "bash: line 2: pear: command not found\n",
      "bash: line 2: pear: command not found\n",
      "bash: line 2: pear: command not found\n",
      "bash: line 2: pear: command not found\n",
      "bash: line 2: pear: command not found\n",
      "bash: line 2: pear: command not found\n",
      "bash: line 2: pear: command not found\n",
      "bash: line 2: pear: command not found\n",
      "bash: line 2: pear: command not found\n",
      "bash: line 2: pear: command not found\n",
      "bash: line 2: pear: command not found\n",
      "bash: line 2: pear: command not found\n",
      "bash: line 2: pear: command not found\n",
      "bash: line 2: pear: command not found\n",
      "bash: line 2: pear: command not found\n",
      "bash: line 2: pear: command not found\n",
      "bash: line 2: pear: command not found\n",
      "bash: line 2: pear: command not found\n",
      "bash: line 2: pear: command not found\n",
      "bash: line 2: pear: command not found\n",
      "bash: line 2: pear: command not found\n",
      "bash: line 2: pear: command not found\n",
      "bash: line 2: pear: command not found\n",
      "bash: line 2: pear: command not found\n",
      "bash: line 2: pear: command not found\n",
      "bash: line 2: pear: command not found\n",
      "bash: line 2: pear: command not found\n",
      "bash: line 2: pear: command not found\n",
      "bash: line 2: pear: command not found\n",
      "bash: line 2: pear: command not found\n",
      "bash: line 2: pear: command not found\n",
      "bash: line 2: pear: command not found\n",
      "bash: line 2: pear: command not found\n",
      "bash: line 2: pear: command not found\n",
      "bash: line 2: pear: command not found\n",
      "bash: line 2: pear: command not found\n",
      "bash: line 2: pear: command not found\n",
      "bash: line 2: pear: command not found\n",
      "bash: line 2: pear: command not found\n",
      "bash: line 2: pear: command not found\n",
      "bash: line 2: pear: command not found\n",
      "bash: line 2: pear: command not found\n",
      "bash: line 2: pear: command not found\n",
      "bash: line 2: pear: command not found\n",
      "bash: line 2: pear: command not found\n",
      "bash: line 2: pear: command not found\n",
      "bash: line 2: pear: command not found\n",
      "bash: line 2: pear: command not found\n",
      "bash: line 2: pear: command not found\n",
      "bash: line 2: pear: command not found\n",
      "bash: line 2: pear: command not found\n",
      "bash: line 2: pear: command not found\n",
      "bash: line 2: pear: command not found\n",
      "bash: line 2: pear: command not found\n",
      "bash: line 2: pear: command not found\n",
      "bash: line 2: pear: command not found\n",
      "bash: line 2: pear: command not found\n",
      "bash: line 2: pear: command not found\n",
      "bash: line 2: pear: command not found\n",
      "bash: line 2: pear: command not found\n",
      "bash: line 2: pear: command not found\n",
      "bash: line 2: pear: command not found\n",
      "bash: line 2: pear: command not found\n",
      "bash: line 2: pear: command not found\n",
      "bash: line 2: pear: command not found\n",
      "bash: line 2: pear: command not found\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64/: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64/: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64/: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64/: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64/: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64/: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64/: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64/: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64/: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64/: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64/: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64/: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64/: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64/: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64/: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64/: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64/: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64/: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64/: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64/: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64/: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64/: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64/: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64/: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64/: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64/: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64/: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64/: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64/: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64/: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64/: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64/: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64/: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64/: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64/: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64/: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64/: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64/: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64/: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64/: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64/: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64/: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64/: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64/: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64/: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64/: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64/: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64/: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64/: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64/: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64/: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64/: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64/: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64/: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64/: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64/: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64/: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64/: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64/: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64/: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64/: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64/: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64/: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64/: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64/: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64/: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64/: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64/: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64/: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64/: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64/: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64/: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64/: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64/: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64/: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64/: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64/: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64/: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64/: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64/: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64/: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64/: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64/: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64/: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64/: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64/: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64/: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64/: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64/: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64/: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64/: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64/: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64/: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64/: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64/: Is a directory\n",
      "bash: line 2: /usr/local/bioinformatics/pear-0.9.6-bin-64/: Is a directory\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: /analysis/20160609_pyrad/fastq/1HL_10A_R1.fq\n",
      "Reverse reads file.................: /analysis/20160609_pyrad/fastq/1HL_10A_R2.fq\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 33\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 10\n",
      "Minimum read size after trimming...: 33\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 20\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.284260\n",
      "  C: 0.220915\n",
      "  G: 0.223558\n",
      "  T: 0.271267\n",
      "  976333 uncalled bases\n",
      "Assemblying reads: 0%\r",
      "Assemblying reads: 4%\r",
      "Assemblying reads: 7%\r",
      "Assemblying reads: 10%\r",
      "Assemblying reads: 12%\r",
      "Assemblying reads: 15%\r",
      "Assemblying reads: 18%\r",
      "Assemblying reads: 21%\r",
      "Assemblying reads: 23%\r",
      "Assemblying reads: 26%\r",
      "Assemblying reads: 29%\r",
      "Assemblying reads: 32%\r",
      "Assemblying reads: 35%\r",
      "Assemblying reads: 37%\r",
      "Assemblying reads: 40%\r",
      "Assemblying reads: 43%\r",
      "Assemblying reads: 46%\r",
      "Assemblying reads: 49%\r",
      "Assemblying reads: 51%\r",
      "Assemblying reads: 54%\r",
      "Assemblying reads: 57%\r",
      "Assemblying reads: 60%\r",
      "Assemblying reads: 62%\r",
      "Assemblying reads: 65%\r",
      "Assemblying reads: 68%\r",
      "Assemblying reads: 71%\r",
      "Assemblying reads: 74%\r",
      "Assemblying reads: 76%\r",
      "Assemblying reads: 79%\r",
      "Assemblying reads: 82%\r",
      "Assemblying reads: 85%\r",
      "Assemblying reads: 88%\r",
      "Assemblying reads: 90%\r",
      "Assemblying reads: 93%\r",
      "Assemblying reads: 96%\r",
      "Assemblying reads: 99%\r",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 2,890,999 / 4,151,970 (69.630%)\n",
      "Discarded reads ...................: 1,260,971 / 4,151,970 (30.370%)\n",
      "Not assembled reads ...............: 0 / 4,151,970 (0.000%)\n",
      "Assembled reads file...............: /analysis/20160609_pyrad/fastq/1HL_10A.assembled.fastq\n",
      "Discarded reads file...............: /analysis/20160609_pyrad/fastq/1HL_10A.discarded.fastq\n",
      "Unassembled forward reads file.....: /analysis/20160609_pyrad/fastq/1HL_10A.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: /analysis/20160609_pyrad/fastq/1HL_10A.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: /analysis/20160609_pyrad/fastq/1HL_11A_R1.fq\n",
      "Reverse reads file.................: /analysis/20160609_pyrad/fastq/1HL_11A_R2.fq\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 33\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 10\n",
      "Minimum read size after trimming...: 33\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 20\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.284750\n",
      "  C: 0.219094\n",
      "  G: 0.222236\n",
      "  T: 0.273920\n",
      "  1052017 uncalled bases\n",
      "Assemblying reads: 0%\r",
      "Assemblying reads: 4%\r",
      "Assemblying reads: 6%\r",
      "Assemblying reads: 9%\r",
      "Assemblying reads: 11%\r",
      "Assemblying reads: 14%\r",
      "Assemblying reads: 16%\r",
      "Assemblying reads: 19%\r",
      "Assemblying reads: 22%\r",
      "Assemblying reads: 24%\r",
      "Assemblying reads: 27%\r",
      "Assemblying reads: 29%\r",
      "Assemblying reads: 32%\r",
      "Assemblying reads: 34%\r",
      "Assemblying reads: 37%\r",
      "Assemblying reads: 40%\r",
      "Assemblying reads: 42%\r",
      "Assemblying reads: 45%\r",
      "Assemblying reads: 47%\r",
      "Assemblying reads: 50%\r",
      "Assemblying reads: 52%\r",
      "Assemblying reads: 55%\r",
      "Assemblying reads: 57%\r",
      "Assemblying reads: 60%\r",
      "Assemblying reads: 63%\r",
      "Assemblying reads: 65%\r",
      "Assemblying reads: 68%\r",
      "Assemblying reads: 70%\r",
      "Assemblying reads: 73%\r",
      "Assemblying reads: 75%\r",
      "Assemblying reads: 78%\r",
      "Assemblying reads: 81%\r",
      "Assemblying reads: 83%\r",
      "Assemblying reads: 86%\r",
      "Assemblying reads: 88%\r",
      "Assemblying reads: 91%\r",
      "Assemblying reads: 93%\r",
      "Assemblying reads: 96%\r",
      "Assemblying reads: 99%\r",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 3,088,425 / 4,508,779 (68.498%)\n",
      "Discarded reads ...................: 1,420,354 / 4,508,779 (31.502%)\n",
      "Not assembled reads ...............: 0 / 4,508,779 (0.000%)\n",
      "Assembled reads file...............: /analysis/20160609_pyrad/fastq/1HL_11A.assembled.fastq\n",
      "Discarded reads file...............: /analysis/20160609_pyrad/fastq/1HL_11A.discarded.fastq\n",
      "Unassembled forward reads file.....: /analysis/20160609_pyrad/fastq/1HL_11A.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: /analysis/20160609_pyrad/fastq/1HL_11A.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: /analysis/20160609_pyrad/fastq/1HL_12A_R1.fq\n",
      "Reverse reads file.................: /analysis/20160609_pyrad/fastq/1HL_12A_R2.fq\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 33\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 10\n",
      "Minimum read size after trimming...: 33\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 20\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.287139\n",
      "  C: 0.219617\n",
      "  G: 0.224007\n",
      "  T: 0.269237\n",
      "  760356 uncalled bases\n",
      "Assemblying reads: 0%\r",
      "Assemblying reads: 5%\r",
      "Assemblying reads: 9%\r",
      "Assemblying reads: 12%\r",
      "Assemblying reads: 16%\r",
      "Assemblying reads: 19%\r",
      "Assemblying reads: 23%\r",
      "Assemblying reads: 26%\r",
      "Assemblying reads: 30%\r",
      "Assemblying reads: 33%\r",
      "Assemblying reads: 37%\r",
      "Assemblying reads: 40%\r",
      "Assemblying reads: 44%\r",
      "Assemblying reads: 47%\r",
      "Assemblying reads: 51%\r",
      "Assemblying reads: 54%\r",
      "Assemblying reads: 58%\r",
      "Assemblying reads: 61%\r",
      "Assemblying reads: 65%\r",
      "Assemblying reads: 68%\r",
      "Assemblying reads: 72%\r",
      "Assemblying reads: 75%\r",
      "Assemblying reads: 79%\r",
      "Assemblying reads: 82%\r",
      "Assemblying reads: 86%\r",
      "Assemblying reads: 89%\r",
      "Assemblying reads: 93%\r",
      "Assemblying reads: 96%\r",
      "Assemblying reads: 100%\r",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 2,323,231 / 3,302,719 (70.343%)\n",
      "Discarded reads ...................: 979,488 / 3,302,719 (29.657%)\n",
      "Not assembled reads ...............: 0 / 3,302,719 (0.000%)\n",
      "Assembled reads file...............: /analysis/20160609_pyrad/fastq/1HL_12A.assembled.fastq\n",
      "Discarded reads file...............: /analysis/20160609_pyrad/fastq/1HL_12A.discarded.fastq\n",
      "Unassembled forward reads file.....: /analysis/20160609_pyrad/fastq/1HL_12A.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: /analysis/20160609_pyrad/fastq/1HL_12A.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: /analysis/20160609_pyrad/fastq/1HL_13A_R1.fq\n",
      "Reverse reads file.................: /analysis/20160609_pyrad/fastq/1HL_13A_R2.fq\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 33\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 10\n",
      "Minimum read size after trimming...: 33\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 20\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.281475\n",
      "  C: 0.225744\n",
      "  G: 0.229937\n",
      "  T: 0.262844\n",
      "  872487 uncalled bases\n",
      "Assemblying reads: 0%\r",
      "Assemblying reads: 4%\r",
      "Assemblying reads: 7%\r",
      "Assemblying reads: 10%\r",
      "Assemblying reads: 14%\r",
      "Assemblying reads: 17%\r",
      "Assemblying reads: 20%\r",
      "Assemblying reads: 23%\r",
      "Assemblying reads: 26%\r",
      "Assemblying reads: 29%\r",
      "Assemblying reads: 32%\r",
      "Assemblying reads: 35%\r",
      "Assemblying reads: 38%\r",
      "Assemblying reads: 41%\r",
      "Assemblying reads: 44%\r",
      "Assemblying reads: 47%\r",
      "Assemblying reads: 50%\r",
      "Assemblying reads: 53%\r",
      "Assemblying reads: 56%\r",
      "Assemblying reads: 60%\r",
      "Assemblying reads: 63%\r",
      "Assemblying reads: 66%\r",
      "Assemblying reads: 69%\r",
      "Assemblying reads: 72%\r",
      "Assemblying reads: 75%\r",
      "Assemblying reads: 78%\r",
      "Assemblying reads: 81%\r",
      "Assemblying reads: 84%\r",
      "Assemblying reads: 87%\r",
      "Assemblying reads: 90%\r",
      "Assemblying reads: 93%\r",
      "Assemblying reads: 96%\r",
      "Assemblying reads: 99%\r",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 2,701,305 / 3,778,167 (71.498%)\n",
      "Discarded reads ...................: 1,076,862 / 3,778,167 (28.502%)\n",
      "Not assembled reads ...............: 0 / 3,778,167 (0.000%)\n",
      "Assembled reads file...............: /analysis/20160609_pyrad/fastq/1HL_13A.assembled.fastq\n",
      "Discarded reads file...............: /analysis/20160609_pyrad/fastq/1HL_13A.discarded.fastq\n",
      "Unassembled forward reads file.....: /analysis/20160609_pyrad/fastq/1HL_13A.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: /analysis/20160609_pyrad/fastq/1HL_13A.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: /analysis/20160609_pyrad/fastq/1HL_14A_R1.fq\n",
      "Reverse reads file.................: /analysis/20160609_pyrad/fastq/1HL_14A_R2.fq\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 33\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 10\n",
      "Minimum read size after trimming...: 33\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 20\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.281942\n",
      "  C: 0.219923\n",
      "  G: 0.228922\n",
      "  T: 0.269212\n",
      "  620505 uncalled bases\n",
      "Assemblying reads: 0%\r",
      "Assemblying reads: 6%\r",
      "Assemblying reads: 11%\r",
      "Assemblying reads: 15%\r",
      "Assemblying reads: 19%\r",
      "Assemblying reads: 24%\r",
      "Assemblying reads: 28%\r",
      "Assemblying reads: 32%\r",
      "Assemblying reads: 37%\r",
      "Assemblying reads: 41%\r",
      "Assemblying reads: 45%\r",
      "Assemblying reads: 49%\r",
      "Assemblying reads: 54%\r",
      "Assemblying reads: 58%\r",
      "Assemblying reads: 62%\r",
      "Assemblying reads: 67%\r",
      "Assemblying reads: 71%\r",
      "Assemblying reads: 75%\r",
      "Assemblying reads: 80%\r",
      "Assemblying reads: 84%\r",
      "Assemblying reads: 88%\r",
      "Assemblying reads: 92%\r",
      "Assemblying reads: 97%\r",
      "Assemblying reads: 100%\r",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 1,865,526 / 2,687,944 (69.403%)\n",
      "Discarded reads ...................: 822,418 / 2,687,944 (30.597%)\n",
      "Not assembled reads ...............: 0 / 2,687,944 (0.000%)\n",
      "Assembled reads file...............: /analysis/20160609_pyrad/fastq/1HL_14A.assembled.fastq\n",
      "Discarded reads file...............: /analysis/20160609_pyrad/fastq/1HL_14A.discarded.fastq\n",
      "Unassembled forward reads file.....: /analysis/20160609_pyrad/fastq/1HL_14A.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: /analysis/20160609_pyrad/fastq/1HL_14A.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: /analysis/20160609_pyrad/fastq/1HL_15A_R1.fq\n",
      "Reverse reads file.................: /analysis/20160609_pyrad/fastq/1HL_15A_R2.fq\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 33\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 10\n",
      "Minimum read size after trimming...: 33\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 20\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.286995\n",
      "  C: 0.218292\n",
      "  G: 0.222091\n",
      "  T: 0.272623\n",
      "  629863 uncalled bases\n",
      "Assemblying reads: 0%\r",
      "Assemblying reads: 6%\r",
      "Assemblying reads: 10%\r",
      "Assemblying reads: 15%\r",
      "Assemblying reads: 19%\r",
      "Assemblying reads: 23%\r",
      "Assemblying reads: 27%\r",
      "Assemblying reads: 32%\r",
      "Assemblying reads: 36%\r",
      "Assemblying reads: 40%\r",
      "Assemblying reads: 44%\r",
      "Assemblying reads: 48%\r",
      "Assemblying reads: 53%\r",
      "Assemblying reads: 57%\r",
      "Assemblying reads: 61%\r",
      "Assemblying reads: 65%\r",
      "Assemblying reads: 69%\r",
      "Assemblying reads: 74%\r",
      "Assemblying reads: 78%\r",
      "Assemblying reads: 82%\r",
      "Assemblying reads: 86%\r",
      "Assemblying reads: 91%\r",
      "Assemblying reads: 95%\r",
      "Assemblying reads: 99%\r",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 1,837,684 / 2,745,386 (66.937%)\n",
      "Discarded reads ...................: 907,702 / 2,745,386 (33.063%)\n",
      "Not assembled reads ...............: 0 / 2,745,386 (0.000%)\n",
      "Assembled reads file...............: /analysis/20160609_pyrad/fastq/1HL_15A.assembled.fastq\n",
      "Discarded reads file...............: /analysis/20160609_pyrad/fastq/1HL_15A.discarded.fastq\n",
      "Unassembled forward reads file.....: /analysis/20160609_pyrad/fastq/1HL_15A.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: /analysis/20160609_pyrad/fastq/1HL_15A.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: /analysis/20160609_pyrad/fastq/1HL_16A_R1.fq\n",
      "Reverse reads file.................: /analysis/20160609_pyrad/fastq/1HL_16A_R2.fq\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 33\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 10\n",
      "Minimum read size after trimming...: 33\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 20\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.284772\n",
      "  C: 0.220373\n",
      "  G: 0.225556\n",
      "  T: 0.269300\n",
      "  704839 uncalled bases\n",
      "Assemblying reads: 0%\r",
      "Assemblying reads: 6%\r",
      "Assemblying reads: 9%\r",
      "Assemblying reads: 13%\r",
      "Assemblying reads: 17%\r",
      "Assemblying reads: 21%\r",
      "Assemblying reads: 25%\r",
      "Assemblying reads: 28%\r",
      "Assemblying reads: 32%\r",
      "Assemblying reads: 36%\r",
      "Assemblying reads: 40%\r",
      "Assemblying reads: 43%\r",
      "Assemblying reads: 47%\r",
      "Assemblying reads: 51%\r",
      "Assemblying reads: 55%\r",
      "Assemblying reads: 59%\r",
      "Assemblying reads: 62%\r",
      "Assemblying reads: 66%\r",
      "Assemblying reads: 70%\r",
      "Assemblying reads: 74%\r",
      "Assemblying reads: 78%\r",
      "Assemblying reads: 81%\r",
      "Assemblying reads: 85%\r",
      "Assemblying reads: 89%\r",
      "Assemblying reads: 93%\r",
      "Assemblying reads: 97%\r",
      "Assemblying reads: 100%\r",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 2,100,838 / 3,052,493 (68.824%)\n",
      "Discarded reads ...................: 951,655 / 3,052,493 (31.176%)\n",
      "Not assembled reads ...............: 0 / 3,052,493 (0.000%)\n",
      "Assembled reads file...............: /analysis/20160609_pyrad/fastq/1HL_16A.assembled.fastq\n",
      "Discarded reads file...............: /analysis/20160609_pyrad/fastq/1HL_16A.discarded.fastq\n",
      "Unassembled forward reads file.....: /analysis/20160609_pyrad/fastq/1HL_16A.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: /analysis/20160609_pyrad/fastq/1HL_16A.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: /analysis/20160609_pyrad/fastq/1HL_17A_R1.fq\n",
      "Reverse reads file.................: /analysis/20160609_pyrad/fastq/1HL_17A_R2.fq\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 33\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 10\n",
      "Minimum read size after trimming...: 33\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 20\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.285335\n",
      "  C: 0.219349\n",
      "  G: 0.220786\n",
      "  T: 0.274530\n",
      "  1060536 uncalled bases\n",
      "Assemblying reads: 0%\r",
      "Assemblying reads: 4%\r",
      "Assemblying reads: 6%\r",
      "Assemblying reads: 9%\r",
      "Assemblying reads: 11%\r",
      "Assemblying reads: 14%\r",
      "Assemblying reads: 16%\r",
      "Assemblying reads: 19%\r",
      "Assemblying reads: 21%\r",
      "Assemblying reads: 24%\r",
      "Assemblying reads: 26%\r",
      "Assemblying reads: 29%\r",
      "Assemblying reads: 31%\r",
      "Assemblying reads: 34%\r",
      "Assemblying reads: 36%\r",
      "Assemblying reads: 39%\r",
      "Assemblying reads: 41%\r",
      "Assemblying reads: 44%\r",
      "Assemblying reads: 46%\r",
      "Assemblying reads: 49%\r",
      "Assemblying reads: 51%\r",
      "Assemblying reads: 54%\r",
      "Assemblying reads: 57%\r",
      "Assemblying reads: 59%\r",
      "Assemblying reads: 62%\r",
      "Assemblying reads: 64%\r",
      "Assemblying reads: 67%\r",
      "Assemblying reads: 69%\r",
      "Assemblying reads: 72%\r",
      "Assemblying reads: 74%\r",
      "Assemblying reads: 77%\r",
      "Assemblying reads: 79%\r",
      "Assemblying reads: 82%\r",
      "Assemblying reads: 84%\r",
      "Assemblying reads: 87%\r",
      "Assemblying reads: 89%\r",
      "Assemblying reads: 92%\r",
      "Assemblying reads: 94%\r",
      "Assemblying reads: 97%\r",
      "Assemblying reads: 99%\r",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 3,106,139 / 4,587,720 (67.706%)\n",
      "Discarded reads ...................: 1,481,581 / 4,587,720 (32.294%)\n",
      "Not assembled reads ...............: 0 / 4,587,720 (0.000%)\n",
      "Assembled reads file...............: /analysis/20160609_pyrad/fastq/1HL_17A.assembled.fastq\n",
      "Discarded reads file...............: /analysis/20160609_pyrad/fastq/1HL_17A.discarded.fastq\n",
      "Unassembled forward reads file.....: /analysis/20160609_pyrad/fastq/1HL_17A.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: /analysis/20160609_pyrad/fastq/1HL_17A.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: /analysis/20160609_pyrad/fastq/1HL_19A_R1.fq\n",
      "Reverse reads file.................: /analysis/20160609_pyrad/fastq/1HL_19A_R2.fq\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 33\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 10\n",
      "Minimum read size after trimming...: 33\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 20\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.286462\n",
      "  C: 0.221234\n",
      "  G: 0.223039\n",
      "  T: 0.269265\n",
      "  1059693 uncalled bases\n",
      "Assemblying reads: 0%\r",
      "Assemblying reads: 4%\r",
      "Assemblying reads: 6%\r",
      "Assemblying reads: 9%\r",
      "Assemblying reads: 11%\r",
      "Assemblying reads: 14%\r",
      "Assemblying reads: 16%\r",
      "Assemblying reads: 19%\r",
      "Assemblying reads: 21%\r",
      "Assemblying reads: 24%\r",
      "Assemblying reads: 26%\r",
      "Assemblying reads: 29%\r",
      "Assemblying reads: 31%\r",
      "Assemblying reads: 34%\r",
      "Assemblying reads: 36%\r",
      "Assemblying reads: 39%\r",
      "Assemblying reads: 41%\r",
      "Assemblying reads: 44%\r",
      "Assemblying reads: 46%\r",
      "Assemblying reads: 49%\r",
      "Assemblying reads: 51%\r",
      "Assemblying reads: 54%\r",
      "Assemblying reads: 56%\r",
      "Assemblying reads: 59%\r",
      "Assemblying reads: 61%\r",
      "Assemblying reads: 64%\r",
      "Assemblying reads: 66%\r",
      "Assemblying reads: 69%\r",
      "Assemblying reads: 71%\r",
      "Assemblying reads: 74%\r",
      "Assemblying reads: 76%\r",
      "Assemblying reads: 79%\r",
      "Assemblying reads: 81%\r",
      "Assemblying reads: 84%\r",
      "Assemblying reads: 86%\r",
      "Assemblying reads: 89%\r",
      "Assemblying reads: 91%\r",
      "Assemblying reads: 94%\r",
      "Assemblying reads: 96%\r",
      "Assemblying reads: 99%\r",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 3,174,884 / 4,607,650 (68.905%)\n",
      "Discarded reads ...................: 1,432,766 / 4,607,650 (31.095%)\n",
      "Not assembled reads ...............: 0 / 4,607,650 (0.000%)\n",
      "Assembled reads file...............: /analysis/20160609_pyrad/fastq/1HL_19A.assembled.fastq\n",
      "Discarded reads file...............: /analysis/20160609_pyrad/fastq/1HL_19A.discarded.fastq\n",
      "Unassembled forward reads file.....: /analysis/20160609_pyrad/fastq/1HL_19A.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: /analysis/20160609_pyrad/fastq/1HL_19A.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: /analysis/20160609_pyrad/fastq/1HL_1A_R1.fq\n",
      "Reverse reads file.................: /analysis/20160609_pyrad/fastq/1HL_1A_R2.fq\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 33\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 10\n",
      "Minimum read size after trimming...: 33\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 20\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.285947\n",
      "  C: 0.218539\n",
      "  G: 0.224484\n",
      "  T: 0.271030\n",
      "  691996 uncalled bases\n",
      "Assemblying reads: 0%\r",
      "Assemblying reads: 6%\r",
      "Assemblying reads: 9%\r",
      "Assemblying reads: 13%\r",
      "Assemblying reads: 17%\r",
      "Assemblying reads: 21%\r",
      "Assemblying reads: 25%\r",
      "Assemblying reads: 29%\r",
      "Assemblying reads: 33%\r",
      "Assemblying reads: 36%\r",
      "Assemblying reads: 40%\r",
      "Assemblying reads: 44%\r",
      "Assemblying reads: 48%\r",
      "Assemblying reads: 52%\r",
      "Assemblying reads: 56%\r",
      "Assemblying reads: 59%\r",
      "Assemblying reads: 63%\r",
      "Assemblying reads: 67%\r",
      "Assemblying reads: 71%\r",
      "Assemblying reads: 75%\r",
      "Assemblying reads: 79%\r",
      "Assemblying reads: 83%\r",
      "Assemblying reads: 86%\r",
      "Assemblying reads: 90%\r",
      "Assemblying reads: 94%\r",
      "Assemblying reads: 98%\r",
      "Assemblying reads: 100%\r",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 2,098,741 / 3,010,262 (69.720%)\n",
      "Discarded reads ...................: 911,521 / 3,010,262 (30.280%)\n",
      "Not assembled reads ...............: 0 / 3,010,262 (0.000%)\n",
      "Assembled reads file...............: /analysis/20160609_pyrad/fastq/1HL_1A.assembled.fastq\n",
      "Discarded reads file...............: /analysis/20160609_pyrad/fastq/1HL_1A.discarded.fastq\n",
      "Unassembled forward reads file.....: /analysis/20160609_pyrad/fastq/1HL_1A.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: /analysis/20160609_pyrad/fastq/1HL_1A.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: /analysis/20160609_pyrad/fastq/1HL_20A_R1.fq\n",
      "Reverse reads file.................: /analysis/20160609_pyrad/fastq/1HL_20A_R2.fq\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 33\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 10\n",
      "Minimum read size after trimming...: 33\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 20\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.287080\n",
      "  C: 0.218017\n",
      "  G: 0.225527\n",
      "  T: 0.269376\n",
      "  840036 uncalled bases\n",
      "Assemblying reads: 0%\r",
      "Assemblying reads: 5%\r",
      "Assemblying reads: 8%\r",
      "Assemblying reads: 11%\r",
      "Assemblying reads: 14%\r",
      "Assemblying reads: 17%\r",
      "Assemblying reads: 20%\r",
      "Assemblying reads: 23%\r",
      "Assemblying reads: 26%\r",
      "Assemblying reads: 30%\r",
      "Assemblying reads: 33%\r",
      "Assemblying reads: 36%\r",
      "Assemblying reads: 39%\r",
      "Assemblying reads: 42%\r",
      "Assemblying reads: 45%\r",
      "Assemblying reads: 48%\r",
      "Assemblying reads: 52%\r",
      "Assemblying reads: 55%\r",
      "Assemblying reads: 58%\r",
      "Assemblying reads: 61%\r",
      "Assemblying reads: 64%\r",
      "Assemblying reads: 67%\r",
      "Assemblying reads: 70%\r",
      "Assemblying reads: 73%\r",
      "Assemblying reads: 77%\r",
      "Assemblying reads: 80%\r",
      "Assemblying reads: 83%\r",
      "Assemblying reads: 86%\r",
      "Assemblying reads: 89%\r",
      "Assemblying reads: 92%\r",
      "Assemblying reads: 95%\r",
      "Assemblying reads: 99%\r",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 2,516,582 / 3,692,276 (68.158%)\n",
      "Discarded reads ...................: 1,175,694 / 3,692,276 (31.842%)\n",
      "Not assembled reads ...............: 0 / 3,692,276 (0.000%)\n",
      "Assembled reads file...............: /analysis/20160609_pyrad/fastq/1HL_20A.assembled.fastq\n",
      "Discarded reads file...............: /analysis/20160609_pyrad/fastq/1HL_20A.discarded.fastq\n",
      "Unassembled forward reads file.....: /analysis/20160609_pyrad/fastq/1HL_20A.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: /analysis/20160609_pyrad/fastq/1HL_20A.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: /analysis/20160609_pyrad/fastq/1HL_21A_R1.fq\n",
      "Reverse reads file.................: /analysis/20160609_pyrad/fastq/1HL_21A_R2.fq\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 33\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 10\n",
      "Minimum read size after trimming...: 33\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 20\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.284969\n",
      "  C: 0.220369\n",
      "  G: 0.224433\n",
      "  T: 0.270228\n",
      "  751300 uncalled bases\n",
      "Assemblying reads: 0%\r",
      "Assemblying reads: 5%\r",
      "Assemblying reads: 9%\r",
      "Assemblying reads: 12%\r",
      "Assemblying reads: 16%\r",
      "Assemblying reads: 19%\r",
      "Assemblying reads: 23%\r",
      "Assemblying reads: 26%\r",
      "Assemblying reads: 30%\r",
      "Assemblying reads: 33%\r",
      "Assemblying reads: 37%\r",
      "Assemblying reads: 40%\r",
      "Assemblying reads: 44%\r",
      "Assemblying reads: 48%\r",
      "Assemblying reads: 51%\r",
      "Assemblying reads: 55%\r",
      "Assemblying reads: 58%\r",
      "Assemblying reads: 62%\r",
      "Assemblying reads: 65%\r",
      "Assemblying reads: 69%\r",
      "Assemblying reads: 72%\r",
      "Assemblying reads: 76%\r",
      "Assemblying reads: 79%\r",
      "Assemblying reads: 83%\r",
      "Assemblying reads: 86%\r",
      "Assemblying reads: 90%\r",
      "Assemblying reads: 93%\r",
      "Assemblying reads: 97%\r",
      "Assemblying reads: 100%\r",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 2,203,056 / 3,278,190 (67.203%)\n",
      "Discarded reads ...................: 1,075,134 / 3,278,190 (32.797%)\n",
      "Not assembled reads ...............: 0 / 3,278,190 (0.000%)\n",
      "Assembled reads file...............: /analysis/20160609_pyrad/fastq/1HL_21A.assembled.fastq\n",
      "Discarded reads file...............: /analysis/20160609_pyrad/fastq/1HL_21A.discarded.fastq\n",
      "Unassembled forward reads file.....: /analysis/20160609_pyrad/fastq/1HL_21A.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: /analysis/20160609_pyrad/fastq/1HL_21A.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: /analysis/20160609_pyrad/fastq/1HL_22A_R1.fq\n",
      "Reverse reads file.................: /analysis/20160609_pyrad/fastq/1HL_22A_R2.fq\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 33\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 10\n",
      "Minimum read size after trimming...: 33\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 20\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.286768\n",
      "  C: 0.218420\n",
      "  G: 0.226207\n",
      "  T: 0.268605\n",
      "  619188 uncalled bases\n",
      "Assemblying reads: 0%\r",
      "Assemblying reads: 6%\r",
      "Assemblying reads: 11%\r",
      "Assemblying reads: 15%\r",
      "Assemblying reads: 19%\r",
      "Assemblying reads: 24%\r",
      "Assemblying reads: 28%\r",
      "Assemblying reads: 32%\r",
      "Assemblying reads: 37%\r",
      "Assemblying reads: 41%\r",
      "Assemblying reads: 45%\r",
      "Assemblying reads: 50%\r",
      "Assemblying reads: 54%\r",
      "Assemblying reads: 59%\r",
      "Assemblying reads: 63%\r",
      "Assemblying reads: 67%\r",
      "Assemblying reads: 72%\r",
      "Assemblying reads: 76%\r",
      "Assemblying reads: 80%\r",
      "Assemblying reads: 85%\r",
      "Assemblying reads: 89%\r",
      "Assemblying reads: 93%\r",
      "Assemblying reads: 98%\r",
      "Assemblying reads: 100%\r",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 1,807,508 / 2,666,966 (67.774%)\n",
      "Discarded reads ...................: 859,458 / 2,666,966 (32.226%)\n",
      "Not assembled reads ...............: 0 / 2,666,966 (0.000%)\n",
      "Assembled reads file...............: /analysis/20160609_pyrad/fastq/1HL_22A.assembled.fastq\n",
      "Discarded reads file...............: /analysis/20160609_pyrad/fastq/1HL_22A.discarded.fastq\n",
      "Unassembled forward reads file.....: /analysis/20160609_pyrad/fastq/1HL_22A.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: /analysis/20160609_pyrad/fastq/1HL_22A.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: /analysis/20160609_pyrad/fastq/1HL_23A_R1.fq\n",
      "Reverse reads file.................: /analysis/20160609_pyrad/fastq/1HL_23A_R2.fq\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 33\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 10\n",
      "Minimum read size after trimming...: 33\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 20\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.283043\n",
      "  C: 0.218893\n",
      "  G: 0.223030\n",
      "  T: 0.275034\n",
      "  1053503 uncalled bases\n",
      "Assemblying reads: 0%\r",
      "Assemblying reads: 4%\r",
      "Assemblying reads: 6%\r",
      "Assemblying reads: 9%\r",
      "Assemblying reads: 11%\r",
      "Assemblying reads: 14%\r",
      "Assemblying reads: 16%\r",
      "Assemblying reads: 19%\r",
      "Assemblying reads: 21%\r",
      "Assemblying reads: 24%\r",
      "Assemblying reads: 26%\r",
      "Assemblying reads: 29%\r",
      "Assemblying reads: 31%\r",
      "Assemblying reads: 34%\r",
      "Assemblying reads: 36%\r",
      "Assemblying reads: 39%\r",
      "Assemblying reads: 42%\r",
      "Assemblying reads: 44%\r",
      "Assemblying reads: 47%\r",
      "Assemblying reads: 49%\r",
      "Assemblying reads: 52%\r",
      "Assemblying reads: 54%\r",
      "Assemblying reads: 57%\r",
      "Assemblying reads: 59%\r",
      "Assemblying reads: 62%\r",
      "Assemblying reads: 64%\r",
      "Assemblying reads: 67%\r",
      "Assemblying reads: 69%\r",
      "Assemblying reads: 72%\r",
      "Assemblying reads: 75%\r",
      "Assemblying reads: 77%\r",
      "Assemblying reads: 80%\r",
      "Assemblying reads: 82%\r",
      "Assemblying reads: 85%\r",
      "Assemblying reads: 87%\r",
      "Assemblying reads: 90%\r",
      "Assemblying reads: 92%\r",
      "Assemblying reads: 95%\r",
      "Assemblying reads: 97%\r",
      "Assemblying reads: 100%\r",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 3,117,475 / 4,566,989 (68.261%)\n",
      "Discarded reads ...................: 1,449,514 / 4,566,989 (31.739%)\n",
      "Not assembled reads ...............: 0 / 4,566,989 (0.000%)\n",
      "Assembled reads file...............: /analysis/20160609_pyrad/fastq/1HL_23A.assembled.fastq\n",
      "Discarded reads file...............: /analysis/20160609_pyrad/fastq/1HL_23A.discarded.fastq\n",
      "Unassembled forward reads file.....: /analysis/20160609_pyrad/fastq/1HL_23A.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: /analysis/20160609_pyrad/fastq/1HL_23A.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: /analysis/20160609_pyrad/fastq/1HL_24A_R1.fq\n",
      "Reverse reads file.................: /analysis/20160609_pyrad/fastq/1HL_24A_R2.fq\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 33\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 10\n",
      "Minimum read size after trimming...: 33\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 20\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.284089\n",
      "  C: 0.219803\n",
      "  G: 0.220925\n",
      "  T: 0.275182\n",
      "  786608 uncalled bases\n",
      "Assemblying reads: 0%\r",
      "Assemblying reads: 5%\r",
      "Assemblying reads: 8%\r",
      "Assemblying reads: 12%\r",
      "Assemblying reads: 15%\r",
      "Assemblying reads: 18%\r",
      "Assemblying reads: 22%\r",
      "Assemblying reads: 25%\r",
      "Assemblying reads: 29%\r",
      "Assemblying reads: 32%\r",
      "Assemblying reads: 35%\r",
      "Assemblying reads: 39%\r",
      "Assemblying reads: 42%\r",
      "Assemblying reads: 46%\r",
      "Assemblying reads: 49%\r",
      "Assemblying reads: 52%\r",
      "Assemblying reads: 56%\r",
      "Assemblying reads: 59%\r",
      "Assemblying reads: 62%\r",
      "Assemblying reads: 66%\r",
      "Assemblying reads: 69%\r",
      "Assemblying reads: 73%\r",
      "Assemblying reads: 76%\r",
      "Assemblying reads: 79%\r",
      "Assemblying reads: 83%\r",
      "Assemblying reads: 86%\r",
      "Assemblying reads: 90%\r",
      "Assemblying reads: 93%\r",
      "Assemblying reads: 96%\r",
      "Assemblying reads: 100%\r",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 2,290,319 / 3,418,136 (67.005%)\n",
      "Discarded reads ...................: 1,127,817 / 3,418,136 (32.995%)\n",
      "Not assembled reads ...............: 0 / 3,418,136 (0.000%)\n",
      "Assembled reads file...............: /analysis/20160609_pyrad/fastq/1HL_24A.assembled.fastq\n",
      "Discarded reads file...............: /analysis/20160609_pyrad/fastq/1HL_24A.discarded.fastq\n",
      "Unassembled forward reads file.....: /analysis/20160609_pyrad/fastq/1HL_24A.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: /analysis/20160609_pyrad/fastq/1HL_24A.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: /analysis/20160609_pyrad/fastq/1HL_25A_R1.fq\n",
      "Reverse reads file.................: /analysis/20160609_pyrad/fastq/1HL_25A_R2.fq\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 33\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 10\n",
      "Minimum read size after trimming...: 33\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 20\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.285912\n",
      "  C: 0.220616\n",
      "  G: 0.222195\n",
      "  T: 0.271276\n",
      "  871202 uncalled bases\n",
      "Assemblying reads: 0%\r",
      "Assemblying reads: 4%\r",
      "Assemblying reads: 7%\r",
      "Assemblying reads: 10%\r",
      "Assemblying reads: 14%\r",
      "Assemblying reads: 17%\r",
      "Assemblying reads: 20%\r",
      "Assemblying reads: 23%\r",
      "Assemblying reads: 26%\r",
      "Assemblying reads: 29%\r",
      "Assemblying reads: 32%\r",
      "Assemblying reads: 35%\r",
      "Assemblying reads: 38%\r",
      "Assemblying reads: 41%\r",
      "Assemblying reads: 44%\r",
      "Assemblying reads: 47%\r",
      "Assemblying reads: 50%\r",
      "Assemblying reads: 53%\r",
      "Assemblying reads: 56%\r",
      "Assemblying reads: 59%\r",
      "Assemblying reads: 62%\r",
      "Assemblying reads: 65%\r",
      "Assemblying reads: 68%\r",
      "Assemblying reads: 71%\r",
      "Assemblying reads: 75%\r",
      "Assemblying reads: 78%\r",
      "Assemblying reads: 81%\r",
      "Assemblying reads: 84%\r",
      "Assemblying reads: 87%\r",
      "Assemblying reads: 90%\r",
      "Assemblying reads: 93%\r",
      "Assemblying reads: 96%\r",
      "Assemblying reads: 99%\r",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 2,606,955 / 3,794,161 (68.710%)\n",
      "Discarded reads ...................: 1,187,206 / 3,794,161 (31.290%)\n",
      "Not assembled reads ...............: 0 / 3,794,161 (0.000%)\n",
      "Assembled reads file...............: /analysis/20160609_pyrad/fastq/1HL_25A.assembled.fastq\n",
      "Discarded reads file...............: /analysis/20160609_pyrad/fastq/1HL_25A.discarded.fastq\n",
      "Unassembled forward reads file.....: /analysis/20160609_pyrad/fastq/1HL_25A.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: /analysis/20160609_pyrad/fastq/1HL_25A.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: /analysis/20160609_pyrad/fastq/1HL_26A_R1.fq\n",
      "Reverse reads file.................: /analysis/20160609_pyrad/fastq/1HL_26A_R2.fq\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 33\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 10\n",
      "Minimum read size after trimming...: 33\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 20\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.283029\n",
      "  C: 0.223790\n",
      "  G: 0.227327\n",
      "  T: 0.265854\n",
      "  962198 uncalled bases\n",
      "Assemblying reads: 0%\r",
      "Assemblying reads: 4%\r",
      "Assemblying reads: 7%\r",
      "Assemblying reads: 9%\r",
      "Assemblying reads: 12%\r",
      "Assemblying reads: 15%\r",
      "Assemblying reads: 18%\r",
      "Assemblying reads: 21%\r",
      "Assemblying reads: 23%\r",
      "Assemblying reads: 26%\r",
      "Assemblying reads: 29%\r",
      "Assemblying reads: 32%\r",
      "Assemblying reads: 35%\r",
      "Assemblying reads: 37%\r",
      "Assemblying reads: 40%\r",
      "Assemblying reads: 43%\r",
      "Assemblying reads: 46%\r",
      "Assemblying reads: 48%\r",
      "Assemblying reads: 51%\r",
      "Assemblying reads: 54%\r",
      "Assemblying reads: 57%\r",
      "Assemblying reads: 60%\r",
      "Assemblying reads: 62%\r",
      "Assemblying reads: 65%\r",
      "Assemblying reads: 68%\r",
      "Assemblying reads: 71%\r",
      "Assemblying reads: 73%\r",
      "Assemblying reads: 76%\r",
      "Assemblying reads: 79%\r",
      "Assemblying reads: 82%\r",
      "Assemblying reads: 85%\r",
      "Assemblying reads: 87%\r",
      "Assemblying reads: 90%\r",
      "Assemblying reads: 93%\r",
      "Assemblying reads: 96%\r",
      "Assemblying reads: 98%\r",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 2,915,184 / 4,165,278 (69.988%)\n",
      "Discarded reads ...................: 1,250,094 / 4,165,278 (30.012%)\n",
      "Not assembled reads ...............: 0 / 4,165,278 (0.000%)\n",
      "Assembled reads file...............: /analysis/20160609_pyrad/fastq/1HL_26A.assembled.fastq\n",
      "Discarded reads file...............: /analysis/20160609_pyrad/fastq/1HL_26A.discarded.fastq\n",
      "Unassembled forward reads file.....: /analysis/20160609_pyrad/fastq/1HL_26A.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: /analysis/20160609_pyrad/fastq/1HL_26A.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: /analysis/20160609_pyrad/fastq/1HL_27A_R1.fq\n",
      "Reverse reads file.................: /analysis/20160609_pyrad/fastq/1HL_27A_R2.fq\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 33\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 10\n",
      "Minimum read size after trimming...: 33\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 20\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.281204\n",
      "  C: 0.223493\n",
      "  G: 0.225397\n",
      "  T: 0.269907\n",
      "  957521 uncalled bases\n",
      "Assemblying reads: 0%\r",
      "Assemblying reads: 4%\r",
      "Assemblying reads: 7%\r",
      "Assemblying reads: 10%\r",
      "Assemblying reads: 12%\r",
      "Assemblying reads: 15%\r",
      "Assemblying reads: 18%\r",
      "Assemblying reads: 21%\r",
      "Assemblying reads: 23%\r",
      "Assemblying reads: 26%\r",
      "Assemblying reads: 29%\r",
      "Assemblying reads: 32%\r",
      "Assemblying reads: 35%\r",
      "Assemblying reads: 37%\r",
      "Assemblying reads: 40%\r",
      "Assemblying reads: 43%\r",
      "Assemblying reads: 46%\r",
      "Assemblying reads: 49%\r",
      "Assemblying reads: 51%\r",
      "Assemblying reads: 54%\r",
      "Assemblying reads: 57%\r",
      "Assemblying reads: 60%\r",
      "Assemblying reads: 63%\r",
      "Assemblying reads: 65%\r",
      "Assemblying reads: 68%\r",
      "Assemblying reads: 71%\r",
      "Assemblying reads: 74%\r",
      "Assemblying reads: 77%\r",
      "Assemblying reads: 79%\r",
      "Assemblying reads: 82%\r",
      "Assemblying reads: 85%\r",
      "Assemblying reads: 88%\r",
      "Assemblying reads: 90%\r",
      "Assemblying reads: 93%\r",
      "Assemblying reads: 96%\r",
      "Assemblying reads: 99%\r",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 2,887,147 / 4,146,188 (69.634%)\n",
      "Discarded reads ...................: 1,259,041 / 4,146,188 (30.366%)\n",
      "Not assembled reads ...............: 0 / 4,146,188 (0.000%)\n",
      "Assembled reads file...............: /analysis/20160609_pyrad/fastq/1HL_27A.assembled.fastq\n",
      "Discarded reads file...............: /analysis/20160609_pyrad/fastq/1HL_27A.discarded.fastq\n",
      "Unassembled forward reads file.....: /analysis/20160609_pyrad/fastq/1HL_27A.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: /analysis/20160609_pyrad/fastq/1HL_27A.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: /analysis/20160609_pyrad/fastq/1HL_28A_R1.fq\n",
      "Reverse reads file.................: /analysis/20160609_pyrad/fastq/1HL_28A_R2.fq\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 33\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 10\n",
      "Minimum read size after trimming...: 33\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 20\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.285403\n",
      "  C: 0.219129\n",
      "  G: 0.223981\n",
      "  T: 0.271486\n",
      "  1082276 uncalled bases\n",
      "Assemblying reads: 0%\r",
      "Assemblying reads: 3%\r",
      "Assemblying reads: 6%\r",
      "Assemblying reads: 8%\r",
      "Assemblying reads: 11%\r",
      "Assemblying reads: 13%\r",
      "Assemblying reads: 16%\r",
      "Assemblying reads: 18%\r",
      "Assemblying reads: 21%\r",
      "Assemblying reads: 23%\r",
      "Assemblying reads: 25%\r",
      "Assemblying reads: 28%\r",
      "Assemblying reads: 30%\r",
      "Assemblying reads: 33%\r",
      "Assemblying reads: 35%\r",
      "Assemblying reads: 38%\r",
      "Assemblying reads: 40%\r",
      "Assemblying reads: 43%\r",
      "Assemblying reads: 45%\r",
      "Assemblying reads: 47%\r",
      "Assemblying reads: 50%\r",
      "Assemblying reads: 52%\r",
      "Assemblying reads: 55%\r",
      "Assemblying reads: 57%\r",
      "Assemblying reads: 60%\r",
      "Assemblying reads: 62%\r",
      "Assemblying reads: 65%\r",
      "Assemblying reads: 67%\r",
      "Assemblying reads: 69%\r",
      "Assemblying reads: 72%\r",
      "Assemblying reads: 74%\r",
      "Assemblying reads: 77%\r",
      "Assemblying reads: 79%\r",
      "Assemblying reads: 82%\r",
      "Assemblying reads: 84%\r",
      "Assemblying reads: 87%\r",
      "Assemblying reads: 89%\r",
      "Assemblying reads: 91%\r",
      "Assemblying reads: 94%\r",
      "Assemblying reads: 96%\r",
      "Assemblying reads: 99%\r",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 3,182,673 / 4,733,136 (67.242%)\n",
      "Discarded reads ...................: 1,550,463 / 4,733,136 (32.758%)\n",
      "Not assembled reads ...............: 0 / 4,733,136 (0.000%)\n",
      "Assembled reads file...............: /analysis/20160609_pyrad/fastq/1HL_28A.assembled.fastq\n",
      "Discarded reads file...............: /analysis/20160609_pyrad/fastq/1HL_28A.discarded.fastq\n",
      "Unassembled forward reads file.....: /analysis/20160609_pyrad/fastq/1HL_28A.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: /analysis/20160609_pyrad/fastq/1HL_28A.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: /analysis/20160609_pyrad/fastq/1HL_29A_R1.fq\n",
      "Reverse reads file.................: /analysis/20160609_pyrad/fastq/1HL_29A_R2.fq\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 33\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 10\n",
      "Minimum read size after trimming...: 33\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 20\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.282155\n",
      "  C: 0.226136\n",
      "  G: 0.227452\n",
      "  T: 0.264257\n",
      "  977328 uncalled bases\n",
      "Assemblying reads: 0%\r",
      "Assemblying reads: 4%\r",
      "Assemblying reads: 7%\r",
      "Assemblying reads: 9%\r",
      "Assemblying reads: 12%\r",
      "Assemblying reads: 15%\r",
      "Assemblying reads: 18%\r",
      "Assemblying reads: 20%\r",
      "Assemblying reads: 23%\r",
      "Assemblying reads: 26%\r",
      "Assemblying reads: 28%\r",
      "Assemblying reads: 31%\r",
      "Assemblying reads: 34%\r",
      "Assemblying reads: 37%\r",
      "Assemblying reads: 39%\r",
      "Assemblying reads: 42%\r",
      "Assemblying reads: 45%\r",
      "Assemblying reads: 48%\r",
      "Assemblying reads: 50%\r",
      "Assemblying reads: 53%\r",
      "Assemblying reads: 56%\r",
      "Assemblying reads: 58%\r",
      "Assemblying reads: 61%\r",
      "Assemblying reads: 64%\r",
      "Assemblying reads: 67%\r",
      "Assemblying reads: 69%\r",
      "Assemblying reads: 72%\r",
      "Assemblying reads: 75%\r",
      "Assemblying reads: 78%\r",
      "Assemblying reads: 80%\r",
      "Assemblying reads: 83%\r",
      "Assemblying reads: 86%\r",
      "Assemblying reads: 89%\r",
      "Assemblying reads: 91%\r",
      "Assemblying reads: 94%\r",
      "Assemblying reads: 97%\r",
      "Assemblying reads: 99%\r",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 3,004,016 / 4,238,592 (70.873%)\n",
      "Discarded reads ...................: 1,234,576 / 4,238,592 (29.127%)\n",
      "Not assembled reads ...............: 0 / 4,238,592 (0.000%)\n",
      "Assembled reads file...............: /analysis/20160609_pyrad/fastq/1HL_29A.assembled.fastq\n",
      "Discarded reads file...............: /analysis/20160609_pyrad/fastq/1HL_29A.discarded.fastq\n",
      "Unassembled forward reads file.....: /analysis/20160609_pyrad/fastq/1HL_29A.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: /analysis/20160609_pyrad/fastq/1HL_29A.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: /analysis/20160609_pyrad/fastq/1HL_2A_R1.fq\n",
      "Reverse reads file.................: /analysis/20160609_pyrad/fastq/1HL_2A_R2.fq\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 33\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 10\n",
      "Minimum read size after trimming...: 33\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 20\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.287959\n",
      "  C: 0.219214\n",
      "  G: 0.223470\n",
      "  T: 0.269357\n",
      "  830284 uncalled bases\n",
      "Assemblying reads: 0%\r",
      "Assemblying reads: 5%\r",
      "Assemblying reads: 8%\r",
      "Assemblying reads: 11%\r",
      "Assemblying reads: 14%\r",
      "Assemblying reads: 18%\r",
      "Assemblying reads: 21%\r",
      "Assemblying reads: 24%\r",
      "Assemblying reads: 27%\r",
      "Assemblying reads: 30%\r",
      "Assemblying reads: 34%\r",
      "Assemblying reads: 37%\r",
      "Assemblying reads: 40%\r",
      "Assemblying reads: 43%\r",
      "Assemblying reads: 47%\r",
      "Assemblying reads: 50%\r",
      "Assemblying reads: 53%\r",
      "Assemblying reads: 56%\r",
      "Assemblying reads: 59%\r",
      "Assemblying reads: 63%\r",
      "Assemblying reads: 66%\r",
      "Assemblying reads: 69%\r",
      "Assemblying reads: 72%\r",
      "Assemblying reads: 75%\r",
      "Assemblying reads: 79%\r",
      "Assemblying reads: 82%\r",
      "Assemblying reads: 85%\r",
      "Assemblying reads: 88%\r",
      "Assemblying reads: 92%\r",
      "Assemblying reads: 95%\r",
      "Assemblying reads: 98%\r",
      "Assemblying reads: 100%\r",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 2,525,685 / 3,592,790 (70.299%)\n",
      "Discarded reads ...................: 1,067,105 / 3,592,790 (29.701%)\n",
      "Not assembled reads ...............: 0 / 3,592,790 (0.000%)\n",
      "Assembled reads file...............: /analysis/20160609_pyrad/fastq/1HL_2A.assembled.fastq\n",
      "Discarded reads file...............: /analysis/20160609_pyrad/fastq/1HL_2A.discarded.fastq\n",
      "Unassembled forward reads file.....: /analysis/20160609_pyrad/fastq/1HL_2A.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: /analysis/20160609_pyrad/fastq/1HL_2A.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: /analysis/20160609_pyrad/fastq/1HL_31A_R1.fq\n",
      "Reverse reads file.................: /analysis/20160609_pyrad/fastq/1HL_31A_R2.fq\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 33\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 10\n",
      "Minimum read size after trimming...: 33\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 20\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.286352\n",
      "  C: 0.223404\n",
      "  G: 0.222343\n",
      "  T: 0.267902\n",
      "  726029 uncalled bases\n",
      "Assemblying reads: 0%\r",
      "Assemblying reads: 5%\r",
      "Assemblying reads: 9%\r",
      "Assemblying reads: 13%\r",
      "Assemblying reads: 16%\r",
      "Assemblying reads: 20%\r",
      "Assemblying reads: 24%\r",
      "Assemblying reads: 27%\r",
      "Assemblying reads: 31%\r",
      "Assemblying reads: 34%\r",
      "Assemblying reads: 38%\r",
      "Assemblying reads: 42%\r",
      "Assemblying reads: 45%\r",
      "Assemblying reads: 49%\r",
      "Assemblying reads: 53%\r",
      "Assemblying reads: 56%\r",
      "Assemblying reads: 60%\r",
      "Assemblying reads: 64%\r",
      "Assemblying reads: 67%\r",
      "Assemblying reads: 71%\r",
      "Assemblying reads: 75%\r",
      "Assemblying reads: 78%\r",
      "Assemblying reads: 82%\r",
      "Assemblying reads: 85%\r",
      "Assemblying reads: 89%\r",
      "Assemblying reads: 93%\r",
      "Assemblying reads: 96%\r",
      "Assemblying reads: 100%\r",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 2,203,255 / 3,177,985 (69.329%)\n",
      "Discarded reads ...................: 974,730 / 3,177,985 (30.671%)\n",
      "Not assembled reads ...............: 0 / 3,177,985 (0.000%)\n",
      "Assembled reads file...............: /analysis/20160609_pyrad/fastq/1HL_31A.assembled.fastq\n",
      "Discarded reads file...............: /analysis/20160609_pyrad/fastq/1HL_31A.discarded.fastq\n",
      "Unassembled forward reads file.....: /analysis/20160609_pyrad/fastq/1HL_31A.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: /analysis/20160609_pyrad/fastq/1HL_31A.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: /analysis/20160609_pyrad/fastq/1HL_33A_R1.fq\n",
      "Reverse reads file.................: /analysis/20160609_pyrad/fastq/1HL_33A_R2.fq\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 33\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 10\n",
      "Minimum read size after trimming...: 33\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 20\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.285419\n",
      "  C: 0.224124\n",
      "  G: 0.221250\n",
      "  T: 0.269207\n",
      "  789912 uncalled bases\n",
      "Assemblying reads: 0%\r",
      "Assemblying reads: 5%\r",
      "Assemblying reads: 8%\r",
      "Assemblying reads: 12%\r",
      "Assemblying reads: 15%\r",
      "Assemblying reads: 18%\r",
      "Assemblying reads: 22%\r",
      "Assemblying reads: 25%\r",
      "Assemblying reads: 28%\r",
      "Assemblying reads: 32%\r",
      "Assemblying reads: 35%\r",
      "Assemblying reads: 38%\r",
      "Assemblying reads: 42%\r",
      "Assemblying reads: 45%\r",
      "Assemblying reads: 48%\r",
      "Assemblying reads: 52%\r",
      "Assemblying reads: 55%\r",
      "Assemblying reads: 59%\r",
      "Assemblying reads: 62%\r",
      "Assemblying reads: 65%\r",
      "Assemblying reads: 69%\r",
      "Assemblying reads: 72%\r",
      "Assemblying reads: 75%\r",
      "Assemblying reads: 79%\r",
      "Assemblying reads: 82%\r",
      "Assemblying reads: 85%\r",
      "Assemblying reads: 89%\r",
      "Assemblying reads: 92%\r",
      "Assemblying reads: 95%\r",
      "Assemblying reads: 99%\r",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 2,357,587 / 3,450,088 (68.334%)\n",
      "Discarded reads ...................: 1,092,501 / 3,450,088 (31.666%)\n",
      "Not assembled reads ...............: 0 / 3,450,088 (0.000%)\n",
      "Assembled reads file...............: /analysis/20160609_pyrad/fastq/1HL_33A.assembled.fastq\n",
      "Discarded reads file...............: /analysis/20160609_pyrad/fastq/1HL_33A.discarded.fastq\n",
      "Unassembled forward reads file.....: /analysis/20160609_pyrad/fastq/1HL_33A.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: /analysis/20160609_pyrad/fastq/1HL_33A.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: /analysis/20160609_pyrad/fastq/1HL_34A_R1.fq\n",
      "Reverse reads file.................: /analysis/20160609_pyrad/fastq/1HL_34A_R2.fq\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 33\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 10\n",
      "Minimum read size after trimming...: 33\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 20\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.286763\n",
      "  C: 0.219421\n",
      "  G: 0.220701\n",
      "  T: 0.273115\n",
      "  844931 uncalled bases\n",
      "Assemblying reads: 0%\r",
      "Assemblying reads: 5%\r",
      "Assemblying reads: 8%\r",
      "Assemblying reads: 11%\r",
      "Assemblying reads: 14%\r",
      "Assemblying reads: 17%\r",
      "Assemblying reads: 20%\r",
      "Assemblying reads: 24%\r",
      "Assemblying reads: 27%\r",
      "Assemblying reads: 30%\r",
      "Assemblying reads: 33%\r",
      "Assemblying reads: 36%\r",
      "Assemblying reads: 40%\r",
      "Assemblying reads: 43%\r",
      "Assemblying reads: 46%\r",
      "Assemblying reads: 49%\r",
      "Assemblying reads: 52%\r",
      "Assemblying reads: 55%\r",
      "Assemblying reads: 59%\r",
      "Assemblying reads: 62%\r",
      "Assemblying reads: 65%\r",
      "Assemblying reads: 68%\r",
      "Assemblying reads: 71%\r",
      "Assemblying reads: 74%\r",
      "Assemblying reads: 78%\r",
      "Assemblying reads: 81%\r",
      "Assemblying reads: 84%\r",
      "Assemblying reads: 87%\r",
      "Assemblying reads: 90%\r",
      "Assemblying reads: 94%\r",
      "Assemblying reads: 97%\r",
      "Assemblying reads: 100%\r",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 2,461,367 / 3,643,848 (67.549%)\n",
      "Discarded reads ...................: 1,182,481 / 3,643,848 (32.451%)\n",
      "Not assembled reads ...............: 0 / 3,643,848 (0.000%)\n",
      "Assembled reads file...............: /analysis/20160609_pyrad/fastq/1HL_34A.assembled.fastq\n",
      "Discarded reads file...............: /analysis/20160609_pyrad/fastq/1HL_34A.discarded.fastq\n",
      "Unassembled forward reads file.....: /analysis/20160609_pyrad/fastq/1HL_34A.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: /analysis/20160609_pyrad/fastq/1HL_34A.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: /analysis/20160609_pyrad/fastq/1HL_35A_R1.fq\n",
      "Reverse reads file.................: /analysis/20160609_pyrad/fastq/1HL_35A_R2.fq\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 33\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 10\n",
      "Minimum read size after trimming...: 33\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 20\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.288171\n",
      "  C: 0.218204\n",
      "  G: 0.222090\n",
      "  T: 0.271534\n",
      "  757197 uncalled bases\n",
      "Assemblying reads: 0%\r",
      "Assemblying reads: 5%\r",
      "Assemblying reads: 9%\r",
      "Assemblying reads: 12%\r",
      "Assemblying reads: 16%\r",
      "Assemblying reads: 19%\r",
      "Assemblying reads: 23%\r",
      "Assemblying reads: 26%\r",
      "Assemblying reads: 30%\r",
      "Assemblying reads: 33%\r",
      "Assemblying reads: 37%\r",
      "Assemblying reads: 40%\r",
      "Assemblying reads: 44%\r",
      "Assemblying reads: 47%\r",
      "Assemblying reads: 51%\r",
      "Assemblying reads: 54%\r",
      "Assemblying reads: 58%\r",
      "Assemblying reads: 61%\r",
      "Assemblying reads: 65%\r",
      "Assemblying reads: 68%\r",
      "Assemblying reads: 72%\r",
      "Assemblying reads: 75%\r",
      "Assemblying reads: 79%\r",
      "Assemblying reads: 82%\r",
      "Assemblying reads: 86%\r",
      "Assemblying reads: 89%\r",
      "Assemblying reads: 93%\r",
      "Assemblying reads: 96%\r",
      "Assemblying reads: 100%\r",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 2,222,135 / 3,298,636 (67.365%)\n",
      "Discarded reads ...................: 1,076,501 / 3,298,636 (32.635%)\n",
      "Not assembled reads ...............: 0 / 3,298,636 (0.000%)\n",
      "Assembled reads file...............: /analysis/20160609_pyrad/fastq/1HL_35A.assembled.fastq\n",
      "Discarded reads file...............: /analysis/20160609_pyrad/fastq/1HL_35A.discarded.fastq\n",
      "Unassembled forward reads file.....: /analysis/20160609_pyrad/fastq/1HL_35A.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: /analysis/20160609_pyrad/fastq/1HL_35A.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: /analysis/20160609_pyrad/fastq/1HL_3A_R1.fq\n",
      "Reverse reads file.................: /analysis/20160609_pyrad/fastq/1HL_3A_R2.fq\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 33\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 10\n",
      "Minimum read size after trimming...: 33\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 20\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.288734\n",
      "  C: 0.216823\n",
      "  G: 0.222551\n",
      "  T: 0.271893\n",
      "  666878 uncalled bases\n",
      "Assemblying reads: 0%\r",
      "Assemblying reads: 6%\r",
      "Assemblying reads: 10%\r",
      "Assemblying reads: 14%\r",
      "Assemblying reads: 18%\r",
      "Assemblying reads: 22%\r",
      "Assemblying reads: 26%\r",
      "Assemblying reads: 30%\r",
      "Assemblying reads: 34%\r",
      "Assemblying reads: 38%\r",
      "Assemblying reads: 42%\r",
      "Assemblying reads: 46%\r",
      "Assemblying reads: 50%\r",
      "Assemblying reads: 54%\r",
      "Assemblying reads: 58%\r",
      "Assemblying reads: 62%\r",
      "Assemblying reads: 66%\r",
      "Assemblying reads: 70%\r",
      "Assemblying reads: 74%\r",
      "Assemblying reads: 78%\r",
      "Assemblying reads: 82%\r",
      "Assemblying reads: 86%\r",
      "Assemblying reads: 90%\r",
      "Assemblying reads: 94%\r",
      "Assemblying reads: 98%\r",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 1,994,744 / 2,885,740 (69.124%)\n",
      "Discarded reads ...................: 890,996 / 2,885,740 (30.876%)\n",
      "Not assembled reads ...............: 0 / 2,885,740 (0.000%)\n",
      "Assembled reads file...............: /analysis/20160609_pyrad/fastq/1HL_3A.assembled.fastq\n",
      "Discarded reads file...............: /analysis/20160609_pyrad/fastq/1HL_3A.discarded.fastq\n",
      "Unassembled forward reads file.....: /analysis/20160609_pyrad/fastq/1HL_3A.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: /analysis/20160609_pyrad/fastq/1HL_3A.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: /analysis/20160609_pyrad/fastq/1HL_4A_R1.fq\n",
      "Reverse reads file.................: /analysis/20160609_pyrad/fastq/1HL_4A_R2.fq\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 33\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 10\n",
      "Minimum read size after trimming...: 33\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 20\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.285469\n",
      "  C: 0.221915\n",
      "  G: 0.225571\n",
      "  T: 0.267044\n",
      "  822572 uncalled bases\n",
      "Assemblying reads: 0%\r",
      "Assemblying reads: 5%\r",
      "Assemblying reads: 8%\r",
      "Assemblying reads: 11%\r",
      "Assemblying reads: 15%\r",
      "Assemblying reads: 18%\r",
      "Assemblying reads: 21%\r",
      "Assemblying reads: 25%\r",
      "Assemblying reads: 28%\r",
      "Assemblying reads: 31%\r",
      "Assemblying reads: 34%\r",
      "Assemblying reads: 38%\r",
      "Assemblying reads: 41%\r",
      "Assemblying reads: 44%\r",
      "Assemblying reads: 48%\r",
      "Assemblying reads: 51%\r",
      "Assemblying reads: 54%\r",
      "Assemblying reads: 57%\r",
      "Assemblying reads: 61%\r",
      "Assemblying reads: 64%\r",
      "Assemblying reads: 67%\r",
      "Assemblying reads: 71%\r",
      "Assemblying reads: 74%\r",
      "Assemblying reads: 77%\r",
      "Assemblying reads: 81%\r",
      "Assemblying reads: 84%\r",
      "Assemblying reads: 87%\r",
      "Assemblying reads: 90%\r",
      "Assemblying reads: 94%\r",
      "Assemblying reads: 97%\r",
      "Assemblying reads: 100%\r",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 2,492,589 / 3,510,860 (70.997%)\n",
      "Discarded reads ...................: 1,018,271 / 3,510,860 (29.003%)\n",
      "Not assembled reads ...............: 0 / 3,510,860 (0.000%)\n",
      "Assembled reads file...............: /analysis/20160609_pyrad/fastq/1HL_4A.assembled.fastq\n",
      "Discarded reads file...............: /analysis/20160609_pyrad/fastq/1HL_4A.discarded.fastq\n",
      "Unassembled forward reads file.....: /analysis/20160609_pyrad/fastq/1HL_4A.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: /analysis/20160609_pyrad/fastq/1HL_4A.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: /analysis/20160609_pyrad/fastq/1HL_5A_R1.fq\n",
      "Reverse reads file.................: /analysis/20160609_pyrad/fastq/1HL_5A_R2.fq\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 33\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 10\n",
      "Minimum read size after trimming...: 33\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 20\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.284051\n",
      "  C: 0.220209\n",
      "  G: 0.226839\n",
      "  T: 0.268900\n",
      "  635240 uncalled bases\n",
      "Assemblying reads: 0%\r",
      "Assemblying reads: 6%\r",
      "Assemblying reads: 11%\r",
      "Assemblying reads: 15%\r",
      "Assemblying reads: 19%\r",
      "Assemblying reads: 23%\r",
      "Assemblying reads: 28%\r",
      "Assemblying reads: 32%\r",
      "Assemblying reads: 36%\r",
      "Assemblying reads: 40%\r",
      "Assemblying reads: 45%\r",
      "Assemblying reads: 49%\r",
      "Assemblying reads: 53%\r",
      "Assemblying reads: 57%\r",
      "Assemblying reads: 62%\r",
      "Assemblying reads: 66%\r",
      "Assemblying reads: 70%\r",
      "Assemblying reads: 74%\r",
      "Assemblying reads: 79%\r",
      "Assemblying reads: 83%\r",
      "Assemblying reads: 87%\r",
      "Assemblying reads: 91%\r",
      "Assemblying reads: 96%\r",
      "Assemblying reads: 100%\r",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 1,914,213 / 2,720,076 (70.374%)\n",
      "Discarded reads ...................: 805,863 / 2,720,076 (29.626%)\n",
      "Not assembled reads ...............: 0 / 2,720,076 (0.000%)\n",
      "Assembled reads file...............: /analysis/20160609_pyrad/fastq/1HL_5A.assembled.fastq\n",
      "Discarded reads file...............: /analysis/20160609_pyrad/fastq/1HL_5A.discarded.fastq\n",
      "Unassembled forward reads file.....: /analysis/20160609_pyrad/fastq/1HL_5A.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: /analysis/20160609_pyrad/fastq/1HL_5A.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: /analysis/20160609_pyrad/fastq/1HL_6A_R1.fq\n",
      "Reverse reads file.................: /analysis/20160609_pyrad/fastq/1HL_6A_R2.fq\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 33\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 10\n",
      "Minimum read size after trimming...: 33\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 20\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.285547\n",
      "  C: 0.219469\n",
      "  G: 0.224365\n",
      "  T: 0.270619\n",
      "  609029 uncalled bases\n",
      "Assemblying reads: 0%\r",
      "Assemblying reads: 6%\r",
      "Assemblying reads: 11%\r",
      "Assemblying reads: 15%\r",
      "Assemblying reads: 19%\r",
      "Assemblying reads: 24%\r",
      "Assemblying reads: 28%\r",
      "Assemblying reads: 33%\r",
      "Assemblying reads: 37%\r",
      "Assemblying reads: 41%\r",
      "Assemblying reads: 46%\r",
      "Assemblying reads: 50%\r",
      "Assemblying reads: 54%\r",
      "Assemblying reads: 59%\r",
      "Assemblying reads: 63%\r",
      "Assemblying reads: 67%\r",
      "Assemblying reads: 72%\r",
      "Assemblying reads: 76%\r",
      "Assemblying reads: 80%\r",
      "Assemblying reads: 85%\r",
      "Assemblying reads: 89%\r",
      "Assemblying reads: 93%\r",
      "Assemblying reads: 98%\r",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 1,778,884 / 2,661,086 (66.848%)\n",
      "Discarded reads ...................: 882,202 / 2,661,086 (33.152%)\n",
      "Not assembled reads ...............: 0 / 2,661,086 (0.000%)\n",
      "Assembled reads file...............: /analysis/20160609_pyrad/fastq/1HL_6A.assembled.fastq\n",
      "Discarded reads file...............: /analysis/20160609_pyrad/fastq/1HL_6A.discarded.fastq\n",
      "Unassembled forward reads file.....: /analysis/20160609_pyrad/fastq/1HL_6A.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: /analysis/20160609_pyrad/fastq/1HL_6A.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: /analysis/20160609_pyrad/fastq/1HL_7A_R1.fq\n",
      "Reverse reads file.................: /analysis/20160609_pyrad/fastq/1HL_7A_R2.fq\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 33\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 10\n",
      "Minimum read size after trimming...: 33\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 20\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.279246\n",
      "  C: 0.221508\n",
      "  G: 0.227431\n",
      "  T: 0.271815\n",
      "  851643 uncalled bases\n",
      "Assemblying reads: 0%\r",
      "Assemblying reads: 4%\r",
      "Assemblying reads: 8%\r",
      "Assemblying reads: 11%\r",
      "Assemblying reads: 14%\r",
      "Assemblying reads: 17%\r",
      "Assemblying reads: 20%\r",
      "Assemblying reads: 23%\r",
      "Assemblying reads: 26%\r",
      "Assemblying reads: 30%\r",
      "Assemblying reads: 33%\r",
      "Assemblying reads: 36%\r",
      "Assemblying reads: 39%\r",
      "Assemblying reads: 42%\r",
      "Assemblying reads: 45%\r",
      "Assemblying reads: 48%\r",
      "Assemblying reads: 52%\r",
      "Assemblying reads: 55%\r",
      "Assemblying reads: 58%\r",
      "Assemblying reads: 61%\r",
      "Assemblying reads: 64%\r",
      "Assemblying reads: 67%\r",
      "Assemblying reads: 70%\r",
      "Assemblying reads: 74%\r",
      "Assemblying reads: 77%\r",
      "Assemblying reads: 80%\r",
      "Assemblying reads: 83%\r",
      "Assemblying reads: 86%\r",
      "Assemblying reads: 89%\r",
      "Assemblying reads: 92%\r",
      "Assemblying reads: 95%\r",
      "Assemblying reads: 99%\r",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 2,555,275 / 3,689,392 (69.260%)\n",
      "Discarded reads ...................: 1,134,117 / 3,689,392 (30.740%)\n",
      "Not assembled reads ...............: 0 / 3,689,392 (0.000%)\n",
      "Assembled reads file...............: /analysis/20160609_pyrad/fastq/1HL_7A.assembled.fastq\n",
      "Discarded reads file...............: /analysis/20160609_pyrad/fastq/1HL_7A.discarded.fastq\n",
      "Unassembled forward reads file.....: /analysis/20160609_pyrad/fastq/1HL_7A.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: /analysis/20160609_pyrad/fastq/1HL_7A.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: /analysis/20160609_pyrad/fastq/1HL_8A_R1.fq\n",
      "Reverse reads file.................: /analysis/20160609_pyrad/fastq/1HL_8A_R2.fq\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 33\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 10\n",
      "Minimum read size after trimming...: 33\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 20\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.283546\n",
      "  C: 0.220649\n",
      "  G: 0.226514\n",
      "  T: 0.269290\n",
      "  788910 uncalled bases\n",
      "Assemblying reads: 0%\r",
      "Assemblying reads: 5%\r",
      "Assemblying reads: 8%\r",
      "Assemblying reads: 12%\r",
      "Assemblying reads: 15%\r",
      "Assemblying reads: 19%\r",
      "Assemblying reads: 22%\r",
      "Assemblying reads: 25%\r",
      "Assemblying reads: 29%\r",
      "Assemblying reads: 32%\r",
      "Assemblying reads: 36%\r",
      "Assemblying reads: 39%\r",
      "Assemblying reads: 42%\r",
      "Assemblying reads: 46%\r",
      "Assemblying reads: 49%\r",
      "Assemblying reads: 53%\r",
      "Assemblying reads: 56%\r",
      "Assemblying reads: 59%\r",
      "Assemblying reads: 63%\r",
      "Assemblying reads: 66%\r",
      "Assemblying reads: 70%\r",
      "Assemblying reads: 73%\r",
      "Assemblying reads: 76%\r",
      "Assemblying reads: 80%\r",
      "Assemblying reads: 83%\r",
      "Assemblying reads: 87%\r",
      "Assemblying reads: 90%\r",
      "Assemblying reads: 93%\r",
      "Assemblying reads: 97%\r",
      "Assemblying reads: 100%\r",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 2,330,048 / 3,403,545 (68.459%)\n",
      "Discarded reads ...................: 1,073,497 / 3,403,545 (31.541%)\n",
      "Not assembled reads ...............: 0 / 3,403,545 (0.000%)\n",
      "Assembled reads file...............: /analysis/20160609_pyrad/fastq/1HL_8A.assembled.fastq\n",
      "Discarded reads file...............: /analysis/20160609_pyrad/fastq/1HL_8A.discarded.fastq\n",
      "Unassembled forward reads file.....: /analysis/20160609_pyrad/fastq/1HL_8A.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: /analysis/20160609_pyrad/fastq/1HL_8A.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: /analysis/20160609_pyrad/fastq/1HL_9A_R1.fq\n",
      "Reverse reads file.................: /analysis/20160609_pyrad/fastq/1HL_9A_R2.fq\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 33\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 10\n",
      "Minimum read size after trimming...: 33\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 20\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.281821\n",
      "  C: 0.223687\n",
      "  G: 0.226621\n",
      "  T: 0.267870\n",
      "  871271 uncalled bases\n",
      "Assemblying reads: 0%\r",
      "Assemblying reads: 4%\r",
      "Assemblying reads: 7%\r",
      "Assemblying reads: 11%\r",
      "Assemblying reads: 14%\r",
      "Assemblying reads: 17%\r",
      "Assemblying reads: 20%\r",
      "Assemblying reads: 23%\r",
      "Assemblying reads: 26%\r",
      "Assemblying reads: 29%\r",
      "Assemblying reads: 32%\r",
      "Assemblying reads: 35%\r",
      "Assemblying reads: 38%\r",
      "Assemblying reads: 41%\r",
      "Assemblying reads: 44%\r",
      "Assemblying reads: 48%\r",
      "Assemblying reads: 51%\r",
      "Assemblying reads: 54%\r",
      "Assemblying reads: 57%\r",
      "Assemblying reads: 60%\r",
      "Assemblying reads: 63%\r",
      "Assemblying reads: 66%\r",
      "Assemblying reads: 69%\r",
      "Assemblying reads: 72%\r",
      "Assemblying reads: 75%\r",
      "Assemblying reads: 78%\r",
      "Assemblying reads: 81%\r",
      "Assemblying reads: 85%\r",
      "Assemblying reads: 88%\r",
      "Assemblying reads: 91%\r",
      "Assemblying reads: 94%\r",
      "Assemblying reads: 97%\r",
      "Assemblying reads: 100%\r",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 2,620,475 / 3,753,204 (69.820%)\n",
      "Discarded reads ...................: 1,132,729 / 3,753,204 (30.180%)\n",
      "Not assembled reads ...............: 0 / 3,753,204 (0.000%)\n",
      "Assembled reads file...............: /analysis/20160609_pyrad/fastq/1HL_9A.assembled.fastq\n",
      "Discarded reads file...............: /analysis/20160609_pyrad/fastq/1HL_9A.discarded.fastq\n",
      "Unassembled forward reads file.....: /analysis/20160609_pyrad/fastq/1HL_9A.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: /analysis/20160609_pyrad/fastq/1HL_9A.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: /analysis/20160609_pyrad/fastq/1NF_10A_R1.fq\n",
      "Reverse reads file.................: /analysis/20160609_pyrad/fastq/1NF_10A_R2.fq\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 33\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 10\n",
      "Minimum read size after trimming...: 33\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 20\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.284157\n",
      "  C: 0.220971\n",
      "  G: 0.225414\n",
      "  T: 0.269459\n",
      "  664722 uncalled bases\n",
      "Assemblying reads: 0%\r",
      "Assemblying reads: 6%\r",
      "Assemblying reads: 10%\r",
      "Assemblying reads: 14%\r",
      "Assemblying reads: 18%\r",
      "Assemblying reads: 22%\r",
      "Assemblying reads: 26%\r",
      "Assemblying reads: 30%\r",
      "Assemblying reads: 34%\r",
      "Assemblying reads: 38%\r",
      "Assemblying reads: 42%\r",
      "Assemblying reads: 47%\r",
      "Assemblying reads: 51%\r",
      "Assemblying reads: 55%\r",
      "Assemblying reads: 59%\r",
      "Assemblying reads: 63%\r",
      "Assemblying reads: 67%\r",
      "Assemblying reads: 71%\r",
      "Assemblying reads: 75%\r",
      "Assemblying reads: 79%\r",
      "Assemblying reads: 83%\r",
      "Assemblying reads: 87%\r",
      "Assemblying reads: 91%\r",
      "Assemblying reads: 95%\r",
      "Assemblying reads: 99%\r",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 1,955,600 / 2,848,504 (68.654%)\n",
      "Discarded reads ...................: 892,904 / 2,848,504 (31.346%)\n",
      "Not assembled reads ...............: 0 / 2,848,504 (0.000%)\n",
      "Assembled reads file...............: /analysis/20160609_pyrad/fastq/1NF_10A.assembled.fastq\n",
      "Discarded reads file...............: /analysis/20160609_pyrad/fastq/1NF_10A.discarded.fastq\n",
      "Unassembled forward reads file.....: /analysis/20160609_pyrad/fastq/1NF_10A.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: /analysis/20160609_pyrad/fastq/1NF_10A.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: /analysis/20160609_pyrad/fastq/1NF_11A_R1.fq\n",
      "Reverse reads file.................: /analysis/20160609_pyrad/fastq/1NF_11A_R2.fq\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 33\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 10\n",
      "Minimum read size after trimming...: 33\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 20\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.287054\n",
      "  C: 0.217908\n",
      "  G: 0.223626\n",
      "  T: 0.271412\n",
      "  677071 uncalled bases\n",
      "Assemblying reads: 0%\r",
      "Assemblying reads: 6%\r",
      "Assemblying reads: 10%\r",
      "Assemblying reads: 14%\r",
      "Assemblying reads: 18%\r",
      "Assemblying reads: 21%\r",
      "Assemblying reads: 25%\r",
      "Assemblying reads: 29%\r",
      "Assemblying reads: 33%\r",
      "Assemblying reads: 37%\r",
      "Assemblying reads: 41%\r",
      "Assemblying reads: 45%\r",
      "Assemblying reads: 49%\r",
      "Assemblying reads: 53%\r",
      "Assemblying reads: 57%\r",
      "Assemblying reads: 61%\r",
      "Assemblying reads: 65%\r",
      "Assemblying reads: 69%\r",
      "Assemblying reads: 73%\r",
      "Assemblying reads: 77%\r",
      "Assemblying reads: 81%\r",
      "Assemblying reads: 84%\r",
      "Assemblying reads: 88%\r",
      "Assemblying reads: 92%\r",
      "Assemblying reads: 96%\r",
      "Assemblying reads: 100%\r",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 2,004,118 / 2,939,324 (68.183%)\n",
      "Discarded reads ...................: 935,206 / 2,939,324 (31.817%)\n",
      "Not assembled reads ...............: 0 / 2,939,324 (0.000%)\n",
      "Assembled reads file...............: /analysis/20160609_pyrad/fastq/1NF_11A.assembled.fastq\n",
      "Discarded reads file...............: /analysis/20160609_pyrad/fastq/1NF_11A.discarded.fastq\n",
      "Unassembled forward reads file.....: /analysis/20160609_pyrad/fastq/1NF_11A.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: /analysis/20160609_pyrad/fastq/1NF_11A.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: /analysis/20160609_pyrad/fastq/1NF_12A_R1.fq\n",
      "Reverse reads file.................: /analysis/20160609_pyrad/fastq/1NF_12A_R2.fq\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 33\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 10\n",
      "Minimum read size after trimming...: 33\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 20\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.285198\n",
      "  C: 0.218229\n",
      "  G: 0.228122\n",
      "  T: 0.268451\n",
      "  673211 uncalled bases\n",
      "Assemblying reads: 0%\r",
      "Assemblying reads: 6%\r",
      "Assemblying reads: 10%\r",
      "Assemblying reads: 14%\r",
      "Assemblying reads: 18%\r",
      "Assemblying reads: 22%\r",
      "Assemblying reads: 25%\r",
      "Assemblying reads: 29%\r",
      "Assemblying reads: 33%\r",
      "Assemblying reads: 37%\r",
      "Assemblying reads: 41%\r",
      "Assemblying reads: 45%\r",
      "Assemblying reads: 49%\r",
      "Assemblying reads: 53%\r",
      "Assemblying reads: 57%\r",
      "Assemblying reads: 61%\r",
      "Assemblying reads: 65%\r",
      "Assemblying reads: 69%\r",
      "Assemblying reads: 73%\r",
      "Assemblying reads: 77%\r",
      "Assemblying reads: 81%\r",
      "Assemblying reads: 85%\r",
      "Assemblying reads: 89%\r",
      "Assemblying reads: 93%\r",
      "Assemblying reads: 97%\r",
      "Assemblying reads: 100%\r",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 2,051,515 / 2,930,997 (69.994%)\n",
      "Discarded reads ...................: 879,482 / 2,930,997 (30.006%)\n",
      "Not assembled reads ...............: 0 / 2,930,997 (0.000%)\n",
      "Assembled reads file...............: /analysis/20160609_pyrad/fastq/1NF_12A.assembled.fastq\n",
      "Discarded reads file...............: /analysis/20160609_pyrad/fastq/1NF_12A.discarded.fastq\n",
      "Unassembled forward reads file.....: /analysis/20160609_pyrad/fastq/1NF_12A.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: /analysis/20160609_pyrad/fastq/1NF_12A.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: /analysis/20160609_pyrad/fastq/1NF_13A_R1.fq\n",
      "Reverse reads file.................: /analysis/20160609_pyrad/fastq/1NF_13A_R2.fq\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 33\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 10\n",
      "Minimum read size after trimming...: 33\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 20\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.283198\n",
      "  C: 0.221826\n",
      "  G: 0.226473\n",
      "  T: 0.268503\n",
      "  1200405 uncalled bases\n",
      "Assemblying reads: 0%\r",
      "Assemblying reads: 3%\r",
      "Assemblying reads: 5%\r",
      "Assemblying reads: 7%\r",
      "Assemblying reads: 10%\r",
      "Assemblying reads: 12%\r",
      "Assemblying reads: 14%\r",
      "Assemblying reads: 16%\r",
      "Assemblying reads: 19%\r",
      "Assemblying reads: 21%\r",
      "Assemblying reads: 23%\r",
      "Assemblying reads: 25%\r",
      "Assemblying reads: 28%\r",
      "Assemblying reads: 30%\r",
      "Assemblying reads: 32%\r",
      "Assemblying reads: 34%\r",
      "Assemblying reads: 37%\r",
      "Assemblying reads: 39%\r",
      "Assemblying reads: 41%\r",
      "Assemblying reads: 43%\r",
      "Assemblying reads: 45%\r",
      "Assemblying reads: 48%\r",
      "Assemblying reads: 50%\r",
      "Assemblying reads: 52%\r",
      "Assemblying reads: 54%\r",
      "Assemblying reads: 57%\r",
      "Assemblying reads: 59%\r",
      "Assemblying reads: 61%\r",
      "Assemblying reads: 63%\r",
      "Assemblying reads: 66%\r",
      "Assemblying reads: 68%\r",
      "Assemblying reads: 70%\r",
      "Assemblying reads: 72%\r",
      "Assemblying reads: 74%\r",
      "Assemblying reads: 77%\r",
      "Assemblying reads: 79%\r",
      "Assemblying reads: 81%\r",
      "Assemblying reads: 83%\r",
      "Assemblying reads: 86%\r",
      "Assemblying reads: 88%\r",
      "Assemblying reads: 90%\r",
      "Assemblying reads: 92%\r",
      "Assemblying reads: 95%\r",
      "Assemblying reads: 97%\r",
      "Assemblying reads: 99%\r",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 3,671,926 / 5,183,236 (70.842%)\n",
      "Discarded reads ...................: 1,511,310 / 5,183,236 (29.158%)\n",
      "Not assembled reads ...............: 0 / 5,183,236 (0.000%)\n",
      "Assembled reads file...............: /analysis/20160609_pyrad/fastq/1NF_13A.assembled.fastq\n",
      "Discarded reads file...............: /analysis/20160609_pyrad/fastq/1NF_13A.discarded.fastq\n",
      "Unassembled forward reads file.....: /analysis/20160609_pyrad/fastq/1NF_13A.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: /analysis/20160609_pyrad/fastq/1NF_13A.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: /analysis/20160609_pyrad/fastq/1NF_14A_R1.fq\n",
      "Reverse reads file.................: /analysis/20160609_pyrad/fastq/1NF_14A_R2.fq\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 33\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 10\n",
      "Minimum read size after trimming...: 33\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 20\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.288088\n",
      "  C: 0.217760\n",
      "  G: 0.223964\n",
      "  T: 0.270188\n",
      "  801159 uncalled bases\n",
      "Assemblying reads: 0%\r",
      "Assemblying reads: 5%\r",
      "Assemblying reads: 8%\r",
      "Assemblying reads: 11%\r",
      "Assemblying reads: 15%\r",
      "Assemblying reads: 18%\r",
      "Assemblying reads: 21%\r",
      "Assemblying reads: 25%\r",
      "Assemblying reads: 28%\r",
      "Assemblying reads: 31%\r",
      "Assemblying reads: 35%\r",
      "Assemblying reads: 38%\r",
      "Assemblying reads: 41%\r",
      "Assemblying reads: 45%\r",
      "Assemblying reads: 48%\r",
      "Assemblying reads: 52%\r",
      "Assemblying reads: 55%\r",
      "Assemblying reads: 58%\r",
      "Assemblying reads: 62%\r",
      "Assemblying reads: 65%\r",
      "Assemblying reads: 68%\r",
      "Assemblying reads: 72%\r",
      "Assemblying reads: 75%\r",
      "Assemblying reads: 78%\r",
      "Assemblying reads: 82%\r",
      "Assemblying reads: 85%\r",
      "Assemblying reads: 88%\r",
      "Assemblying reads: 92%\r",
      "Assemblying reads: 95%\r",
      "Assemblying reads: 98%\r",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 2,333,572 / 3,467,126 (67.306%)\n",
      "Discarded reads ...................: 1,133,554 / 3,467,126 (32.694%)\n",
      "Not assembled reads ...............: 0 / 3,467,126 (0.000%)\n",
      "Assembled reads file...............: /analysis/20160609_pyrad/fastq/1NF_14A.assembled.fastq\n",
      "Discarded reads file...............: /analysis/20160609_pyrad/fastq/1NF_14A.discarded.fastq\n",
      "Unassembled forward reads file.....: /analysis/20160609_pyrad/fastq/1NF_14A.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: /analysis/20160609_pyrad/fastq/1NF_14A.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: /analysis/20160609_pyrad/fastq/1NF_15A_R1.fq\n",
      "Reverse reads file.................: /analysis/20160609_pyrad/fastq/1NF_15A_R2.fq\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 33\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 10\n",
      "Minimum read size after trimming...: 33\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 20\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.285599\n",
      "  C: 0.217128\n",
      "  G: 0.227087\n",
      "  T: 0.270186\n",
      "  777701 uncalled bases\n",
      "Assemblying reads: 0%\r",
      "Assemblying reads: 5%\r",
      "Assemblying reads: 8%\r",
      "Assemblying reads: 12%\r",
      "Assemblying reads: 15%\r",
      "Assemblying reads: 19%\r",
      "Assemblying reads: 22%\r",
      "Assemblying reads: 26%\r",
      "Assemblying reads: 29%\r",
      "Assemblying reads: 33%\r",
      "Assemblying reads: 36%\r",
      "Assemblying reads: 40%\r",
      "Assemblying reads: 43%\r",
      "Assemblying reads: 46%\r",
      "Assemblying reads: 50%\r",
      "Assemblying reads: 53%\r",
      "Assemblying reads: 57%\r",
      "Assemblying reads: 60%\r",
      "Assemblying reads: 64%\r",
      "Assemblying reads: 67%\r",
      "Assemblying reads: 71%\r",
      "Assemblying reads: 74%\r",
      "Assemblying reads: 78%\r",
      "Assemblying reads: 81%\r",
      "Assemblying reads: 85%\r",
      "Assemblying reads: 88%\r",
      "Assemblying reads: 91%\r",
      "Assemblying reads: 95%\r",
      "Assemblying reads: 98%\r",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 2,296,320 / 3,344,566 (68.658%)\n",
      "Discarded reads ...................: 1,048,246 / 3,344,566 (31.342%)\n",
      "Not assembled reads ...............: 0 / 3,344,566 (0.000%)\n",
      "Assembled reads file...............: /analysis/20160609_pyrad/fastq/1NF_15A.assembled.fastq\n",
      "Discarded reads file...............: /analysis/20160609_pyrad/fastq/1NF_15A.discarded.fastq\n",
      "Unassembled forward reads file.....: /analysis/20160609_pyrad/fastq/1NF_15A.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: /analysis/20160609_pyrad/fastq/1NF_15A.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: /analysis/20160609_pyrad/fastq/1NF_16A_R1.fq\n",
      "Reverse reads file.................: /analysis/20160609_pyrad/fastq/1NF_16A_R2.fq\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 33\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 10\n",
      "Minimum read size after trimming...: 33\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 20\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.281531\n",
      "  C: 0.223127\n",
      "  G: 0.231810\n",
      "  T: 0.263532\n",
      "  881970 uncalled bases\n",
      "Assemblying reads: 0%\r",
      "Assemblying reads: 4%\r",
      "Assemblying reads: 7%\r",
      "Assemblying reads: 10%\r",
      "Assemblying reads: 13%\r",
      "Assemblying reads: 16%\r",
      "Assemblying reads: 19%\r",
      "Assemblying reads: 22%\r",
      "Assemblying reads: 26%\r",
      "Assemblying reads: 29%\r",
      "Assemblying reads: 32%\r",
      "Assemblying reads: 35%\r",
      "Assemblying reads: 38%\r",
      "Assemblying reads: 41%\r",
      "Assemblying reads: 44%\r",
      "Assemblying reads: 47%\r",
      "Assemblying reads: 50%\r",
      "Assemblying reads: 53%\r",
      "Assemblying reads: 56%\r",
      "Assemblying reads: 59%\r",
      "Assemblying reads: 62%\r",
      "Assemblying reads: 65%\r",
      "Assemblying reads: 68%\r",
      "Assemblying reads: 71%\r",
      "Assemblying reads: 74%\r",
      "Assemblying reads: 77%\r",
      "Assemblying reads: 80%\r",
      "Assemblying reads: 83%\r",
      "Assemblying reads: 86%\r",
      "Assemblying reads: 89%\r",
      "Assemblying reads: 92%\r",
      "Assemblying reads: 95%\r",
      "Assemblying reads: 98%\r",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 2,701,311 / 3,813,301 (70.839%)\n",
      "Discarded reads ...................: 1,111,990 / 3,813,301 (29.161%)\n",
      "Not assembled reads ...............: 0 / 3,813,301 (0.000%)\n",
      "Assembled reads file...............: /analysis/20160609_pyrad/fastq/1NF_16A.assembled.fastq\n",
      "Discarded reads file...............: /analysis/20160609_pyrad/fastq/1NF_16A.discarded.fastq\n",
      "Unassembled forward reads file.....: /analysis/20160609_pyrad/fastq/1NF_16A.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: /analysis/20160609_pyrad/fastq/1NF_16A.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: /analysis/20160609_pyrad/fastq/1NF_17A_R1.fq\n",
      "Reverse reads file.................: /analysis/20160609_pyrad/fastq/1NF_17A_R2.fq\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 33\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 10\n",
      "Minimum read size after trimming...: 33\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 20\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.284847\n",
      "  C: 0.215946\n",
      "  G: 0.223492\n",
      "  T: 0.275716\n",
      "  964953 uncalled bases\n",
      "Assemblying reads: 0%\r",
      "Assemblying reads: 4%\r",
      "Assemblying reads: 7%\r",
      "Assemblying reads: 10%\r",
      "Assemblying reads: 12%\r",
      "Assemblying reads: 15%\r",
      "Assemblying reads: 18%\r",
      "Assemblying reads: 21%\r",
      "Assemblying reads: 24%\r",
      "Assemblying reads: 26%\r",
      "Assemblying reads: 29%\r",
      "Assemblying reads: 32%\r",
      "Assemblying reads: 35%\r",
      "Assemblying reads: 38%\r",
      "Assemblying reads: 40%\r",
      "Assemblying reads: 43%\r",
      "Assemblying reads: 46%\r",
      "Assemblying reads: 49%\r",
      "Assemblying reads: 52%\r",
      "Assemblying reads: 54%\r",
      "Assemblying reads: 57%\r",
      "Assemblying reads: 60%\r",
      "Assemblying reads: 63%\r",
      "Assemblying reads: 66%\r",
      "Assemblying reads: 69%\r",
      "Assemblying reads: 71%\r",
      "Assemblying reads: 74%\r",
      "Assemblying reads: 77%\r",
      "Assemblying reads: 80%\r",
      "Assemblying reads: 83%\r",
      "Assemblying reads: 85%\r",
      "Assemblying reads: 88%\r",
      "Assemblying reads: 91%\r",
      "Assemblying reads: 94%\r",
      "Assemblying reads: 97%\r",
      "Assemblying reads: 99%\r",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 2,846,779 / 4,120,982 (69.080%)\n",
      "Discarded reads ...................: 1,274,203 / 4,120,982 (30.920%)\n",
      "Not assembled reads ...............: 0 / 4,120,982 (0.000%)\n",
      "Assembled reads file...............: /analysis/20160609_pyrad/fastq/1NF_17A.assembled.fastq\n",
      "Discarded reads file...............: /analysis/20160609_pyrad/fastq/1NF_17A.discarded.fastq\n",
      "Unassembled forward reads file.....: /analysis/20160609_pyrad/fastq/1NF_17A.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: /analysis/20160609_pyrad/fastq/1NF_17A.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: /analysis/20160609_pyrad/fastq/1NF_18A_R1.fq\n",
      "Reverse reads file.................: /analysis/20160609_pyrad/fastq/1NF_18A_R2.fq\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 33\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 10\n",
      "Minimum read size after trimming...: 33\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 20\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.291065\n",
      "  C: 0.214441\n",
      "  G: 0.221778\n",
      "  T: 0.272716\n",
      "  504128 uncalled bases\n",
      "Assemblying reads: 0%\r",
      "Assemblying reads: 8%\r",
      "Assemblying reads: 13%\r",
      "Assemblying reads: 18%\r",
      "Assemblying reads: 24%\r",
      "Assemblying reads: 29%\r",
      "Assemblying reads: 34%\r",
      "Assemblying reads: 39%\r",
      "Assemblying reads: 45%\r",
      "Assemblying reads: 50%\r",
      "Assemblying reads: 55%\r",
      "Assemblying reads: 61%\r",
      "Assemblying reads: 66%\r",
      "Assemblying reads: 71%\r",
      "Assemblying reads: 76%\r",
      "Assemblying reads: 82%\r",
      "Assemblying reads: 87%\r",
      "Assemblying reads: 92%\r",
      "Assemblying reads: 97%\r",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 1,469,499 / 2,195,389 (66.936%)\n",
      "Discarded reads ...................: 725,890 / 2,195,389 (33.064%)\n",
      "Not assembled reads ...............: 0 / 2,195,389 (0.000%)\n",
      "Assembled reads file...............: /analysis/20160609_pyrad/fastq/1NF_18A.assembled.fastq\n",
      "Discarded reads file...............: /analysis/20160609_pyrad/fastq/1NF_18A.discarded.fastq\n",
      "Unassembled forward reads file.....: /analysis/20160609_pyrad/fastq/1NF_18A.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: /analysis/20160609_pyrad/fastq/1NF_18A.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: /analysis/20160609_pyrad/fastq/1NF_19A_R1.fq\n",
      "Reverse reads file.................: /analysis/20160609_pyrad/fastq/1NF_19A_R2.fq\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 33\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 10\n",
      "Minimum read size after trimming...: 33\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 20\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.284788\n",
      "  C: 0.218296\n",
      "  G: 0.226489\n",
      "  T: 0.270428\n",
      "  617454 uncalled bases\n",
      "Assemblying reads: 0%\r",
      "Assemblying reads: 6%\r",
      "Assemblying reads: 11%\r",
      "Assemblying reads: 15%\r",
      "Assemblying reads: 20%\r",
      "Assemblying reads: 24%\r",
      "Assemblying reads: 28%\r",
      "Assemblying reads: 33%\r",
      "Assemblying reads: 37%\r",
      "Assemblying reads: 41%\r",
      "Assemblying reads: 46%\r",
      "Assemblying reads: 50%\r",
      "Assemblying reads: 55%\r",
      "Assemblying reads: 59%\r",
      "Assemblying reads: 63%\r",
      "Assemblying reads: 68%\r",
      "Assemblying reads: 72%\r",
      "Assemblying reads: 76%\r",
      "Assemblying reads: 81%\r",
      "Assemblying reads: 85%\r",
      "Assemblying reads: 90%\r",
      "Assemblying reads: 94%\r",
      "Assemblying reads: 98%\r",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 1,775,039 / 2,644,544 (67.121%)\n",
      "Discarded reads ...................: 869,505 / 2,644,544 (32.879%)\n",
      "Not assembled reads ...............: 0 / 2,644,544 (0.000%)\n",
      "Assembled reads file...............: /analysis/20160609_pyrad/fastq/1NF_19A.assembled.fastq\n",
      "Discarded reads file...............: /analysis/20160609_pyrad/fastq/1NF_19A.discarded.fastq\n",
      "Unassembled forward reads file.....: /analysis/20160609_pyrad/fastq/1NF_19A.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: /analysis/20160609_pyrad/fastq/1NF_19A.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: /analysis/20160609_pyrad/fastq/1NF_1A_R1.fq\n",
      "Reverse reads file.................: /analysis/20160609_pyrad/fastq/1NF_1A_R2.fq\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 33\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 10\n",
      "Minimum read size after trimming...: 33\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 20\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.287105\n",
      "  C: 0.217074\n",
      "  G: 0.226171\n",
      "  T: 0.269650\n",
      "  404635 uncalled bases\n",
      "Assemblying reads: 0%\r",
      "Assemblying reads: 10%\r",
      "Assemblying reads: 17%\r",
      "Assemblying reads: 23%\r",
      "Assemblying reads: 30%\r",
      "Assemblying reads: 36%\r",
      "Assemblying reads: 43%\r",
      "Assemblying reads: 50%\r",
      "Assemblying reads: 56%\r",
      "Assemblying reads: 63%\r",
      "Assemblying reads: 70%\r",
      "Assemblying reads: 76%\r",
      "Assemblying reads: 83%\r",
      "Assemblying reads: 90%\r",
      "Assemblying reads: 96%\r",
      "Assemblying reads: 100%\r",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 1,157,346 / 1,743,417 (66.384%)\n",
      "Discarded reads ...................: 586,071 / 1,743,417 (33.616%)\n",
      "Not assembled reads ...............: 0 / 1,743,417 (0.000%)\n",
      "Assembled reads file...............: /analysis/20160609_pyrad/fastq/1NF_1A.assembled.fastq\n",
      "Discarded reads file...............: /analysis/20160609_pyrad/fastq/1NF_1A.discarded.fastq\n",
      "Unassembled forward reads file.....: /analysis/20160609_pyrad/fastq/1NF_1A.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: /analysis/20160609_pyrad/fastq/1NF_1A.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: /analysis/20160609_pyrad/fastq/1NF_20A_R1.fq\n",
      "Reverse reads file.................: /analysis/20160609_pyrad/fastq/1NF_20A_R2.fq\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 33\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 10\n",
      "Minimum read size after trimming...: 33\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 20\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.288912\n",
      "  C: 0.218564\n",
      "  G: 0.223118\n",
      "  T: 0.269406\n",
      "  752785 uncalled bases\n",
      "Assemblying reads: 0%\r",
      "Assemblying reads: 5%\r",
      "Assemblying reads: 9%\r",
      "Assemblying reads: 12%\r",
      "Assemblying reads: 16%\r",
      "Assemblying reads: 19%\r",
      "Assemblying reads: 23%\r",
      "Assemblying reads: 26%\r",
      "Assemblying reads: 30%\r",
      "Assemblying reads: 33%\r",
      "Assemblying reads: 37%\r",
      "Assemblying reads: 40%\r",
      "Assemblying reads: 44%\r",
      "Assemblying reads: 48%\r",
      "Assemblying reads: 51%\r",
      "Assemblying reads: 55%\r",
      "Assemblying reads: 58%\r",
      "Assemblying reads: 62%\r",
      "Assemblying reads: 65%\r",
      "Assemblying reads: 69%\r",
      "Assemblying reads: 72%\r",
      "Assemblying reads: 76%\r",
      "Assemblying reads: 79%\r",
      "Assemblying reads: 83%\r",
      "Assemblying reads: 86%\r",
      "Assemblying reads: 90%\r",
      "Assemblying reads: 93%\r",
      "Assemblying reads: 97%\r",
      "Assemblying reads: 100%\r",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 2,261,913 / 3,273,230 (69.103%)\n",
      "Discarded reads ...................: 1,011,317 / 3,273,230 (30.897%)\n",
      "Not assembled reads ...............: 0 / 3,273,230 (0.000%)\n",
      "Assembled reads file...............: /analysis/20160609_pyrad/fastq/1NF_20A.assembled.fastq\n",
      "Discarded reads file...............: /analysis/20160609_pyrad/fastq/1NF_20A.discarded.fastq\n",
      "Unassembled forward reads file.....: /analysis/20160609_pyrad/fastq/1NF_20A.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: /analysis/20160609_pyrad/fastq/1NF_20A.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: /analysis/20160609_pyrad/fastq/1NF_21A_R1.fq\n",
      "Reverse reads file.................: /analysis/20160609_pyrad/fastq/1NF_21A_R2.fq\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 33\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 10\n",
      "Minimum read size after trimming...: 33\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 20\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.283222\n",
      "  C: 0.221884\n",
      "  G: 0.228245\n",
      "  T: 0.266648\n",
      "  931251 uncalled bases\n",
      "Assemblying reads: 0%\r",
      "Assemblying reads: 4%\r",
      "Assemblying reads: 7%\r",
      "Assemblying reads: 10%\r",
      "Assemblying reads: 13%\r",
      "Assemblying reads: 15%\r",
      "Assemblying reads: 18%\r",
      "Assemblying reads: 21%\r",
      "Assemblying reads: 24%\r",
      "Assemblying reads: 27%\r",
      "Assemblying reads: 30%\r",
      "Assemblying reads: 32%\r",
      "Assemblying reads: 35%\r",
      "Assemblying reads: 38%\r",
      "Assemblying reads: 41%\r",
      "Assemblying reads: 44%\r",
      "Assemblying reads: 47%\r",
      "Assemblying reads: 49%\r",
      "Assemblying reads: 52%\r",
      "Assemblying reads: 55%\r",
      "Assemblying reads: 58%\r",
      "Assemblying reads: 61%\r",
      "Assemblying reads: 64%\r",
      "Assemblying reads: 66%\r",
      "Assemblying reads: 69%\r",
      "Assemblying reads: 72%\r",
      "Assemblying reads: 75%\r",
      "Assemblying reads: 78%\r",
      "Assemblying reads: 81%\r",
      "Assemblying reads: 84%\r",
      "Assemblying reads: 86%\r",
      "Assemblying reads: 89%\r",
      "Assemblying reads: 92%\r",
      "Assemblying reads: 95%\r",
      "Assemblying reads: 98%\r",
      "Assemblying reads: 100%\r",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 2,903,774 / 4,075,131 (71.256%)\n",
      "Discarded reads ...................: 1,171,357 / 4,075,131 (28.744%)\n",
      "Not assembled reads ...............: 0 / 4,075,131 (0.000%)\n",
      "Assembled reads file...............: /analysis/20160609_pyrad/fastq/1NF_21A.assembled.fastq\n",
      "Discarded reads file...............: /analysis/20160609_pyrad/fastq/1NF_21A.discarded.fastq\n",
      "Unassembled forward reads file.....: /analysis/20160609_pyrad/fastq/1NF_21A.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: /analysis/20160609_pyrad/fastq/1NF_21A.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: /analysis/20160609_pyrad/fastq/1NF_22A_R1.fq\n",
      "Reverse reads file.................: /analysis/20160609_pyrad/fastq/1NF_22A_R2.fq\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 33\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 10\n",
      "Minimum read size after trimming...: 33\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 20\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.284947\n",
      "  C: 0.219154\n",
      "  G: 0.226847\n",
      "  T: 0.269051\n",
      "  879359 uncalled bases\n",
      "Assemblying reads: 0%\r",
      "Assemblying reads: 4%\r",
      "Assemblying reads: 7%\r",
      "Assemblying reads: 10%\r",
      "Assemblying reads: 13%\r",
      "Assemblying reads: 16%\r",
      "Assemblying reads: 19%\r",
      "Assemblying reads: 22%\r",
      "Assemblying reads: 25%\r",
      "Assemblying reads: 28%\r",
      "Assemblying reads: 31%\r",
      "Assemblying reads: 35%\r",
      "Assemblying reads: 38%\r",
      "Assemblying reads: 41%\r",
      "Assemblying reads: 44%\r",
      "Assemblying reads: 47%\r",
      "Assemblying reads: 50%\r",
      "Assemblying reads: 53%\r",
      "Assemblying reads: 56%\r",
      "Assemblying reads: 59%\r",
      "Assemblying reads: 62%\r",
      "Assemblying reads: 65%\r",
      "Assemblying reads: 68%\r",
      "Assemblying reads: 71%\r",
      "Assemblying reads: 74%\r",
      "Assemblying reads: 77%\r",
      "Assemblying reads: 80%\r",
      "Assemblying reads: 83%\r",
      "Assemblying reads: 86%\r",
      "Assemblying reads: 89%\r",
      "Assemblying reads: 92%\r",
      "Assemblying reads: 95%\r",
      "Assemblying reads: 98%\r",
      "Assemblying reads: 100%\r",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 2,674,529 / 3,827,463 (69.877%)\n",
      "Discarded reads ...................: 1,152,934 / 3,827,463 (30.123%)\n",
      "Not assembled reads ...............: 0 / 3,827,463 (0.000%)\n",
      "Assembled reads file...............: /analysis/20160609_pyrad/fastq/1NF_22A.assembled.fastq\n",
      "Discarded reads file...............: /analysis/20160609_pyrad/fastq/1NF_22A.discarded.fastq\n",
      "Unassembled forward reads file.....: /analysis/20160609_pyrad/fastq/1NF_22A.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: /analysis/20160609_pyrad/fastq/1NF_22A.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: /analysis/20160609_pyrad/fastq/1NF_23A_R1.fq\n",
      "Reverse reads file.................: /analysis/20160609_pyrad/fastq/1NF_23A_R2.fq\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 33\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 10\n",
      "Minimum read size after trimming...: 33\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 20\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.284968\n",
      "  C: 0.220204\n",
      "  G: 0.227276\n",
      "  T: 0.267552\n",
      "  1015053 uncalled bases\n",
      "Assemblying reads: 0%\r",
      "Assemblying reads: 4%\r",
      "Assemblying reads: 6%\r",
      "Assemblying reads: 9%\r",
      "Assemblying reads: 12%\r",
      "Assemblying reads: 14%\r",
      "Assemblying reads: 17%\r",
      "Assemblying reads: 20%\r",
      "Assemblying reads: 22%\r",
      "Assemblying reads: 25%\r",
      "Assemblying reads: 28%\r",
      "Assemblying reads: 30%\r",
      "Assemblying reads: 33%\r",
      "Assemblying reads: 35%\r",
      "Assemblying reads: 38%\r",
      "Assemblying reads: 41%\r",
      "Assemblying reads: 43%\r",
      "Assemblying reads: 46%\r",
      "Assemblying reads: 49%\r",
      "Assemblying reads: 51%\r",
      "Assemblying reads: 54%\r",
      "Assemblying reads: 57%\r",
      "Assemblying reads: 59%\r",
      "Assemblying reads: 62%\r",
      "Assemblying reads: 65%\r",
      "Assemblying reads: 67%\r",
      "Assemblying reads: 70%\r",
      "Assemblying reads: 73%\r",
      "Assemblying reads: 75%\r",
      "Assemblying reads: 78%\r",
      "Assemblying reads: 80%\r",
      "Assemblying reads: 83%\r",
      "Assemblying reads: 86%\r",
      "Assemblying reads: 88%\r",
      "Assemblying reads: 91%\r",
      "Assemblying reads: 94%\r",
      "Assemblying reads: 96%\r",
      "Assemblying reads: 99%\r",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 3,106,447 / 4,372,177 (71.050%)\n",
      "Discarded reads ...................: 1,265,730 / 4,372,177 (28.950%)\n",
      "Not assembled reads ...............: 0 / 4,372,177 (0.000%)\n",
      "Assembled reads file...............: /analysis/20160609_pyrad/fastq/1NF_23A.assembled.fastq\n",
      "Discarded reads file...............: /analysis/20160609_pyrad/fastq/1NF_23A.discarded.fastq\n",
      "Unassembled forward reads file.....: /analysis/20160609_pyrad/fastq/1NF_23A.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: /analysis/20160609_pyrad/fastq/1NF_23A.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: /analysis/20160609_pyrad/fastq/1NF_24A_R1.fq\n",
      "Reverse reads file.................: /analysis/20160609_pyrad/fastq/1NF_24A_R2.fq\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 33\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 10\n",
      "Minimum read size after trimming...: 33\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 20\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.288096\n",
      "  C: 0.220296\n",
      "  G: 0.221235\n",
      "  T: 0.270373\n",
      "  889909 uncalled bases\n",
      "Assemblying reads: 0%\r",
      "Assemblying reads: 4%\r",
      "Assemblying reads: 7%\r",
      "Assemblying reads: 10%\r",
      "Assemblying reads: 13%\r",
      "Assemblying reads: 16%\r",
      "Assemblying reads: 19%\r",
      "Assemblying reads: 22%\r",
      "Assemblying reads: 25%\r",
      "Assemblying reads: 28%\r",
      "Assemblying reads: 31%\r",
      "Assemblying reads: 34%\r",
      "Assemblying reads: 37%\r",
      "Assemblying reads: 40%\r",
      "Assemblying reads: 43%\r",
      "Assemblying reads: 46%\r",
      "Assemblying reads: 49%\r",
      "Assemblying reads: 52%\r",
      "Assemblying reads: 55%\r",
      "Assemblying reads: 58%\r",
      "Assemblying reads: 61%\r",
      "Assemblying reads: 64%\r",
      "Assemblying reads: 67%\r",
      "Assemblying reads: 70%\r",
      "Assemblying reads: 73%\r",
      "Assemblying reads: 76%\r",
      "Assemblying reads: 79%\r",
      "Assemblying reads: 82%\r",
      "Assemblying reads: 85%\r",
      "Assemblying reads: 88%\r",
      "Assemblying reads: 91%\r",
      "Assemblying reads: 94%\r",
      "Assemblying reads: 97%\r",
      "Assemblying reads: 99%\r",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 2,650,885 / 3,886,155 (68.214%)\n",
      "Discarded reads ...................: 1,235,270 / 3,886,155 (31.786%)\n",
      "Not assembled reads ...............: 0 / 3,886,155 (0.000%)\n",
      "Assembled reads file...............: /analysis/20160609_pyrad/fastq/1NF_24A.assembled.fastq\n",
      "Discarded reads file...............: /analysis/20160609_pyrad/fastq/1NF_24A.discarded.fastq\n",
      "Unassembled forward reads file.....: /analysis/20160609_pyrad/fastq/1NF_24A.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: /analysis/20160609_pyrad/fastq/1NF_24A.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: /analysis/20160609_pyrad/fastq/1NF_25A_R1.fq\n",
      "Reverse reads file.................: /analysis/20160609_pyrad/fastq/1NF_25A_R2.fq\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 33\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 10\n",
      "Minimum read size after trimming...: 33\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 20\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.287368\n",
      "  C: 0.218703\n",
      "  G: 0.220460\n",
      "  T: 0.273469\n",
      "  638537 uncalled bases\n",
      "Assemblying reads: 0%\r",
      "Assemblying reads: 6%\r",
      "Assemblying reads: 10%\r",
      "Assemblying reads: 14%\r",
      "Assemblying reads: 18%\r",
      "Assemblying reads: 22%\r",
      "Assemblying reads: 27%\r",
      "Assemblying reads: 31%\r",
      "Assemblying reads: 35%\r",
      "Assemblying reads: 39%\r",
      "Assemblying reads: 43%\r",
      "Assemblying reads: 47%\r",
      "Assemblying reads: 51%\r",
      "Assemblying reads: 55%\r",
      "Assemblying reads: 60%\r",
      "Assemblying reads: 64%\r",
      "Assemblying reads: 68%\r",
      "Assemblying reads: 72%\r",
      "Assemblying reads: 76%\r",
      "Assemblying reads: 80%\r",
      "Assemblying reads: 84%\r",
      "Assemblying reads: 88%\r",
      "Assemblying reads: 93%\r",
      "Assemblying reads: 97%\r",
      "Assemblying reads: 100%\r",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 1,879,642 / 2,807,991 (66.939%)\n",
      "Discarded reads ...................: 928,349 / 2,807,991 (33.061%)\n",
      "Not assembled reads ...............: 0 / 2,807,991 (0.000%)\n",
      "Assembled reads file...............: /analysis/20160609_pyrad/fastq/1NF_25A.assembled.fastq\n",
      "Discarded reads file...............: /analysis/20160609_pyrad/fastq/1NF_25A.discarded.fastq\n",
      "Unassembled forward reads file.....: /analysis/20160609_pyrad/fastq/1NF_25A.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: /analysis/20160609_pyrad/fastq/1NF_25A.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: /analysis/20160609_pyrad/fastq/1NF_26A_R1.fq\n",
      "Reverse reads file.................: /analysis/20160609_pyrad/fastq/1NF_26A_R2.fq\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 33\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 10\n",
      "Minimum read size after trimming...: 33\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 20\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.288072\n",
      "  C: 0.217058\n",
      "  G: 0.221301\n",
      "  T: 0.273569\n",
      "  887799 uncalled bases\n",
      "Assemblying reads: 0%\r",
      "Assemblying reads: 4%\r",
      "Assemblying reads: 7%\r",
      "Assemblying reads: 10%\r",
      "Assemblying reads: 13%\r",
      "Assemblying reads: 17%\r",
      "Assemblying reads: 20%\r",
      "Assemblying reads: 23%\r",
      "Assemblying reads: 26%\r",
      "Assemblying reads: 29%\r",
      "Assemblying reads: 32%\r",
      "Assemblying reads: 35%\r",
      "Assemblying reads: 38%\r",
      "Assemblying reads: 41%\r",
      "Assemblying reads: 44%\r",
      "Assemblying reads: 47%\r",
      "Assemblying reads: 50%\r",
      "Assemblying reads: 53%\r",
      "Assemblying reads: 56%\r",
      "Assemblying reads: 59%\r",
      "Assemblying reads: 62%\r",
      "Assemblying reads: 65%\r",
      "Assemblying reads: 68%\r",
      "Assemblying reads: 71%\r",
      "Assemblying reads: 74%\r",
      "Assemblying reads: 77%\r",
      "Assemblying reads: 81%\r",
      "Assemblying reads: 84%\r",
      "Assemblying reads: 87%\r",
      "Assemblying reads: 90%\r",
      "Assemblying reads: 93%\r",
      "Assemblying reads: 96%\r",
      "Assemblying reads: 99%\r",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 2,616,210 / 3,796,337 (68.914%)\n",
      "Discarded reads ...................: 1,180,127 / 3,796,337 (31.086%)\n",
      "Not assembled reads ...............: 0 / 3,796,337 (0.000%)\n",
      "Assembled reads file...............: /analysis/20160609_pyrad/fastq/1NF_26A.assembled.fastq\n",
      "Discarded reads file...............: /analysis/20160609_pyrad/fastq/1NF_26A.discarded.fastq\n",
      "Unassembled forward reads file.....: /analysis/20160609_pyrad/fastq/1NF_26A.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: /analysis/20160609_pyrad/fastq/1NF_26A.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: /analysis/20160609_pyrad/fastq/1NF_27A_R1.fq\n",
      "Reverse reads file.................: /analysis/20160609_pyrad/fastq/1NF_27A_R2.fq\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 33\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 10\n",
      "Minimum read size after trimming...: 33\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 20\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.285131\n",
      "  C: 0.219412\n",
      "  G: 0.228628\n",
      "  T: 0.266829\n",
      "  707858 uncalled bases\n",
      "Assemblying reads: 0%\r",
      "Assemblying reads: 5%\r",
      "Assemblying reads: 9%\r",
      "Assemblying reads: 13%\r",
      "Assemblying reads: 17%\r",
      "Assemblying reads: 20%\r",
      "Assemblying reads: 24%\r",
      "Assemblying reads: 28%\r",
      "Assemblying reads: 32%\r",
      "Assemblying reads: 35%\r",
      "Assemblying reads: 39%\r",
      "Assemblying reads: 43%\r",
      "Assemblying reads: 47%\r",
      "Assemblying reads: 50%\r",
      "Assemblying reads: 54%\r",
      "Assemblying reads: 58%\r",
      "Assemblying reads: 62%\r",
      "Assemblying reads: 65%\r",
      "Assemblying reads: 69%\r",
      "Assemblying reads: 73%\r",
      "Assemblying reads: 77%\r",
      "Assemblying reads: 80%\r",
      "Assemblying reads: 84%\r",
      "Assemblying reads: 88%\r",
      "Assemblying reads: 92%\r",
      "Assemblying reads: 95%\r",
      "Assemblying reads: 99%\r",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 2,158,729 / 3,089,535 (69.872%)\n",
      "Discarded reads ...................: 930,806 / 3,089,535 (30.128%)\n",
      "Not assembled reads ...............: 0 / 3,089,535 (0.000%)\n",
      "Assembled reads file...............: /analysis/20160609_pyrad/fastq/1NF_27A.assembled.fastq\n",
      "Discarded reads file...............: /analysis/20160609_pyrad/fastq/1NF_27A.discarded.fastq\n",
      "Unassembled forward reads file.....: /analysis/20160609_pyrad/fastq/1NF_27A.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: /analysis/20160609_pyrad/fastq/1NF_27A.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: /analysis/20160609_pyrad/fastq/1NF_28A_R1.fq\n",
      "Reverse reads file.................: /analysis/20160609_pyrad/fastq/1NF_28A_R2.fq\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 33\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 10\n",
      "Minimum read size after trimming...: 33\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 20\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.282908\n",
      "  C: 0.220026\n",
      "  G: 0.230713\n",
      "  T: 0.266354\n",
      "  740492 uncalled bases\n",
      "Assemblying reads: 0%\r",
      "Assemblying reads: 5%\r",
      "Assemblying reads: 9%\r",
      "Assemblying reads: 12%\r",
      "Assemblying reads: 16%\r",
      "Assemblying reads: 20%\r",
      "Assemblying reads: 23%\r",
      "Assemblying reads: 27%\r",
      "Assemblying reads: 31%\r",
      "Assemblying reads: 34%\r",
      "Assemblying reads: 38%\r",
      "Assemblying reads: 41%\r",
      "Assemblying reads: 45%\r",
      "Assemblying reads: 49%\r",
      "Assemblying reads: 52%\r",
      "Assemblying reads: 56%\r",
      "Assemblying reads: 60%\r",
      "Assemblying reads: 63%\r",
      "Assemblying reads: 67%\r",
      "Assemblying reads: 70%\r",
      "Assemblying reads: 74%\r",
      "Assemblying reads: 78%\r",
      "Assemblying reads: 81%\r",
      "Assemblying reads: 85%\r",
      "Assemblying reads: 89%\r",
      "Assemblying reads: 92%\r",
      "Assemblying reads: 96%\r",
      "Assemblying reads: 99%\r",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 2,221,247 / 3,194,615 (69.531%)\n",
      "Discarded reads ...................: 973,368 / 3,194,615 (30.469%)\n",
      "Not assembled reads ...............: 0 / 3,194,615 (0.000%)\n",
      "Assembled reads file...............: /analysis/20160609_pyrad/fastq/1NF_28A.assembled.fastq\n",
      "Discarded reads file...............: /analysis/20160609_pyrad/fastq/1NF_28A.discarded.fastq\n",
      "Unassembled forward reads file.....: /analysis/20160609_pyrad/fastq/1NF_28A.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: /analysis/20160609_pyrad/fastq/1NF_28A.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: /analysis/20160609_pyrad/fastq/1NF_29A_R1.fq\n",
      "Reverse reads file.................: /analysis/20160609_pyrad/fastq/1NF_29A_R2.fq\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 33\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 10\n",
      "Minimum read size after trimming...: 33\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 20\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....:  ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: /analysis/20160609_pyrad/fastq/1HL_10A_R1.fq\n",
      "Reverse reads file.................: /analysis/20160609_pyrad/fastq/1HL_10A_R2.fq\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 33\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 10\n",
      "Minimum read size after trimming...: 33\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 20\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.284260\n",
      "  C: 0.220915\n",
      "  G: 0.223558\n",
      "  T: 0.271267\n",
      "  976333 uncalled bases\n",
      "Assemblying reads: 0%\r",
      "Assemblying reads: 4%\r",
      "Assemblying reads: 7%\r",
      "Assemblying reads: 10%\r",
      "Assemblying reads: 12%\r",
      "Assemblying reads: 15%\r",
      "Assemblying reads: 18%\r",
      "Assemblying reads: 21%\r",
      "Assemblying reads: 23%\r",
      "Assemblying reads: 26%\r",
      "Assemblying reads: 29%\r",
      "Assemblying reads: 32%\r",
      "Assemblying reads: 35%\r",
      "Assemblying reads: 37%\r",
      "Assemblying reads: 40%\r",
      "Assemblying reads: 43%\r",
      "Assemblying reads: 46%\r",
      "Assemblying reads: 49%\r",
      "Assemblying reads: 51%\r",
      "Assemblying reads: 54%\r",
      "Assemblying reads: 57%\r",
      "Assemblying reads: 60%\r",
      "Assemblying reads: 62%\r",
      "Assemblying reads: 65%\r",
      "Assemblying reads: 68%\r",
      "Assemblying reads: 71%\r",
      "Assemblying reads: 74%\r",
      "Assemblying reads: 76%\r",
      "Assemblying reads: 79%\r",
      "Assemblying reads: 82%\r",
      "Assemblying reads: 85%\r",
      "Assemblying reads: 88%\r",
      "Assemblying reads: 90%\r",
      "Assemblying reads: 93%\r",
      "Assemblying reads: 96%\r",
      "Assemblying reads: 99%\r",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 2,890,999 / 4,151,970 (69.630%)\n",
      "Discarded reads ...................: 1,260,971 / 4,151,970 (30.370%)\n",
      "Not assembled reads ...............: 0 / 4,151,970 (0.000%)\n",
      "Assembled reads file...............: /analysis/20160609_pyrad/fastq/1HL_10A.assembled.fastq\n",
      "Discarded reads file...............: /analysis/20160609_pyrad/fastq/1HL_10A.discarded.fastq\n",
      "Unassembled forward reads file.....: /analysis/20160609_pyrad/fastq/1HL_10A.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: /analysis/20160609_pyrad/fastq/1HL_10A.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: /analysis/20160609_pyrad/fastq/1HL_11A_R1.fq\n",
      "Reverse reads file.................: /analysis/20160609_pyrad/fastq/1HL_11A_R2.fq\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 33\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 10\n",
      "Minimum read size after trimming...: 33\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 20\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.284750\n",
      "  C: 0.219094\n",
      "  G: 0.222236\n",
      "  T: 0.273920\n",
      "  1052017 uncalled bases\n",
      "Assemblying reads: 0%\r",
      "Assemblying reads: 4%\r",
      "Assemblying reads: 6%\r",
      "Assemblying reads: 9%\r",
      "Assemblying reads: 11%\r",
      "Assemblying reads: 14%\r",
      "Assemblying reads: 16%\r",
      "Assemblying reads: 19%\r",
      "Assemblying reads: 22%\r",
      "Assemblying reads: 24%\r",
      "Assemblying reads: 27%\r",
      "Assemblying reads: 29%\r",
      "Assemblying reads: 32%\r",
      "Assemblying reads: 34%\r",
      "Assemblying reads: 37%\r",
      "Assemblying reads: 40%\r",
      "Assemblying reads: 42%\r",
      "Assemblying reads: 45%\r",
      "Assemblying reads: 47%\r",
      "Assemblying reads: 50%\r",
      "Assemblying reads: 52%\r",
      "Assemblying reads: 55%\r",
      "Assemblying reads: 57%\r",
      "Assemblying reads: 60%\r",
      "Assemblying reads: 63%\r",
      "Assemblying reads: 65%\r",
      "Assemblying reads: 68%\r",
      "Assemblying reads: 70%\r",
      "Assemblying reads: 73%\r",
      "Assemblying reads: 75%\r",
      "Assemblying reads: 78%\r",
      "Assemblying reads: 81%\r",
      "Assemblying reads: 83%\r",
      "Assemblying reads: 86%\r",
      "Assemblying reads: 88%\r",
      "Assemblying reads: 91%\r",
      "Assemblying reads: 93%\r",
      "Assemblying reads: 96%\r",
      "Assemblying reads: 99%\r",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 3,088,425 / 4,508,779 (68.498%)\n",
      "Discarded reads ...................: 1,420,354 / 4,508,779 (31.502%)\n",
      "Not assembled reads ...............: 0 / 4,508,779 (0.000%)\n",
      "Assembled reads file...............: /analysis/20160609_pyrad/fastq/1HL_11A.assembled.fastq\n",
      "Discarded reads file...............: /analysis/20160609_pyrad/fastq/1HL_11A.discarded.fastq\n",
      "Unassembled forward reads file.....: /analysis/20160609_pyrad/fastq/1HL_11A.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: /analysis/20160609_pyrad/fastq/1HL_11A.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: /analysis/20160609_pyrad/fastq/1HL_12A_R1.fq\n",
      "Reverse reads file.................: /analysis/20160609_pyrad/fastq/1HL_12A_R2.fq\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 33\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 10\n",
      "Minimum read size after trimming...: 33\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 20\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.287139\n",
      "  C: 0.219617\n",
      "  G: 0.224007\n",
      "  T: 0.269237\n",
      "  760356 uncalled bases\n",
      "Assemblying reads: 0%\r",
      "Assemblying reads: 5%\r",
      "Assemblying reads: 9%\r",
      "Assemblying reads: 12%\r",
      "Assemblying reads: 16%\r",
      "Assemblying reads: 19%\r",
      "Assemblying reads: 23%\r",
      "Assemblying reads: 26%\r",
      "Assemblying reads: 30%\r",
      "Assemblying reads: 33%\r",
      "Assemblying reads: 37%\r",
      "Assemblying reads: 40%\r",
      "Assemblying reads: 44%\r",
      "Assemblying reads: 47%\r",
      "Assemblying reads: 51%\r",
      "Assemblying reads: 54%\r",
      "Assemblying reads: 58%\r",
      "Assemblying reads: 61%\r",
      "Assemblying reads: 65%\r",
      "Assemblying reads: 68%\r",
      "Assemblying reads: 72%\r",
      "Assemblying reads: 75%\r",
      "Assemblying reads: 79%\r",
      "Assemblying reads: 82%\r",
      "Assemblying reads: 86%\r",
      "Assemblying reads: 89%\r",
      "Assemblying reads: 93%\r",
      "Assemblying reads: 96%\r",
      "Assemblying reads: 100%\r",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 2,323,231 / 3,302,719 (70.343%)\n",
      "Discarded reads ...................: 979,488 / 3,302,719 (29.657%)\n",
      "Not assembled reads ...............: 0 / 3,302,719 (0.000%)\n",
      "Assembled reads file...............: /analysis/20160609_pyrad/fastq/1HL_12A.assembled.fastq\n",
      "Discarded reads file...............: /analysis/20160609_pyrad/fastq/1HL_12A.discarded.fastq\n",
      "Unassembled forward reads file.....: /analysis/20160609_pyrad/fastq/1HL_12A.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: /analysis/20160609_pyrad/fastq/1HL_12A.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: /analysis/20160609_pyrad/fastq/1HL_13A_R1.fq\n",
      "Reverse reads file.................: /analysis/20160609_pyrad/fastq/1HL_13A_R2.fq\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 33\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 10\n",
      "Minimum read size after trimming...: 33\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 20\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.281475\n",
      "  C: 0.225744\n",
      "  G: 0.229937\n",
      "  T: 0.262844\n",
      "  872487 uncalled bases\n",
      "Assemblying reads: 0%\r",
      "Assemblying reads: 4%\r",
      "Assemblying reads: 7%\r",
      "Assemblying reads: 10%\r",
      "Assemblying reads: 14%\r",
      "Assemblying reads: 17%\r",
      "Assemblying reads: 20%\r",
      "Assemblying reads: 23%\r",
      "Assemblying reads: 26%\r",
      "Assemblying reads: 29%\r",
      "Assemblying reads: 32%\r",
      "Assemblying reads: 35%\r",
      "Assemblying reads: 38%\r",
      "Assemblying reads: 41%\r",
      "Assemblying reads: 44%\r",
      "Assemblying reads: 47%\r",
      "Assemblying reads: 50%\r",
      "Assemblying reads: 53%\r",
      "Assemblying reads: 56%\r",
      "Assemblying reads: 60%\r",
      "Assemblying reads: 63%\r",
      "Assemblying reads: 66%\r",
      "Assemblying reads: 69%\r",
      "Assemblying reads: 72%\r",
      "Assemblying reads: 75%\r",
      "Assemblying reads: 78%\r",
      "Assemblying reads: 81%\r",
      "Assemblying reads: 84%\r",
      "Assemblying reads: 87%\r",
      "Assemblying reads: 90%\r",
      "Assemblying reads: 93%\r",
      "Assemblying reads: 96%\r",
      "Assemblying reads: 99%\r",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 2,701,305 / 3,778,167 (71.498%)\n",
      "Discarded reads ...................: 1,076,862 / 3,778,167 (28.502%)\n",
      "Not assembled reads ...............: 0 / 3,778,167 (0.000%)\n",
      "Assembled reads file...............: /analysis/20160609_pyrad/fastq/1HL_13A.assembled.fastq\n",
      "Discarded reads file...............: /analysis/20160609_pyrad/fastq/1HL_13A.discarded.fastq\n",
      "Unassembled forward reads file.....: /analysis/20160609_pyrad/fastq/1HL_13A.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: /analysis/20160609_pyrad/fastq/1HL_13A.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: /analysis/20160609_pyrad/fastq/1HL_14A_R1.fq\n",
      "Reverse reads file.................: /analysis/20160609_pyrad/fastq/1HL_14A_R2.fq\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 33\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 10\n",
      "Minimum read size after trimming...: 33\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 20\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.281942\n",
      "  C: 0.219923\n",
      "  G: 0.228922\n",
      "  T: 0.269212\n",
      "  620505 uncalled bases\n",
      "Assemblying reads: 0%\r",
      "Assemblying reads: 6%\r",
      "Assemblying reads: 11%\r",
      "Assemblying reads: 15%\r",
      "Assemblying reads: 19%\r",
      "Assemblying reads: 24%\r",
      "Assemblying reads: 28%\r",
      "Assemblying reads: 32%\r",
      "Assemblying reads: 37%\r",
      "Assemblying reads: 41%\r",
      "Assemblying reads: 45%\r",
      "Assemblying reads: 49%\r",
      "Assemblying reads: 54%\r",
      "Assemblying reads: 58%\r",
      "Assemblying reads: 62%\r",
      "Assemblying reads: 67%\r",
      "Assemblying reads: 71%\r",
      "Assemblying reads: 75%\r",
      "Assemblying reads: 80%\r",
      "Assemblying reads: 84%\r",
      "Assemblying reads: 88%\r",
      "Assemblying reads: 92%\r",
      "Assemblying reads: 97%\r",
      "Assemblying reads: 100%\r",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 1,865,526 / 2,687,944 (69.403%)\n",
      "Discarded reads ...................: 822,418 / 2,687,944 (30.597%)\n",
      "Not assembled reads ...............: 0 / 2,687,944 (0.000%)\n",
      "Assembled reads file...............: /analysis/20160609_pyrad/fastq/1HL_14A.assembled.fastq\n",
      "Discarded reads file...............: /analysis/20160609_pyrad/fastq/1HL_14A.discarded.fastq\n",
      "Unassembled forward reads file.....: /analysis/20160609_pyrad/fastq/1HL_14A.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: /analysis/20160609_pyrad/fastq/1HL_14A.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: /analysis/20160609_pyrad/fastq/1HL_15A_R1.fq\n",
      "Reverse reads file.................: /analysis/20160609_pyrad/fastq/1HL_15A_R2.fq\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 33\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 10\n",
      "Minimum read size after trimming...: 33\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 20\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.286995\n",
      "  C: 0.218292\n",
      "  G: 0.222091\n",
      "  T: 0.272623\n",
      "  629863 uncalled bases\n",
      "Assemblying reads: 0%\r",
      "Assemblying reads: 6%\r",
      "Assemblying reads: 10%\r",
      "Assemblying reads: 15%\r",
      "Assemblying reads: 19%\r",
      "Assemblying reads: 23%\r",
      "Assemblying reads: 27%\r",
      "Assemblying reads: 32%\r",
      "Assemblying reads: 36%\r",
      "Assemblying reads: 40%\r",
      "Assemblying reads: 44%\r",
      "Assemblying reads: 48%\r",
      "Assemblying reads: 53%\r",
      "Assemblying reads: 57%\r",
      "Assemblying reads: 61%\r",
      "Assemblying reads: 65%\r",
      "Assemblying reads: 69%\r",
      "Assemblying reads: 74%\r",
      "Assemblying reads: 78%\r",
      "Assemblying reads: 82%\r",
      "Assemblying reads: 86%\r",
      "Assemblying reads: 91%\r",
      "Assemblying reads: 95%\r",
      "Assemblying reads: 99%\r",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 1,837,684 / 2,745,386 (66.937%)\n",
      "Discarded reads ...................: 907,702 / 2,745,386 (33.063%)\n",
      "Not assembled reads ...............: 0 / 2,745,386 (0.000%)\n",
      "Assembled reads file...............: /analysis/20160609_pyrad/fastq/1HL_15A.assembled.fastq\n",
      "Discarded reads file...............: /analysis/20160609_pyrad/fastq/1HL_15A.discarded.fastq\n",
      "Unassembled forward reads file.....: /analysis/20160609_pyrad/fastq/1HL_15A.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: /analysis/20160609_pyrad/fastq/1HL_15A.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: /analysis/20160609_pyrad/fastq/1HL_16A_R1.fq\n",
      "Reverse reads file.................: /analysis/20160609_pyrad/fastq/1HL_16A_R2.fq\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 33\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 10\n",
      "Minimum read size after trimming...: 33\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 20\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.284772\n",
      "  C: 0.220373\n",
      "  G: 0.225556\n",
      "  T: 0.269300\n",
      "  704839 uncalled bases\n",
      "Assemblying reads: 0%\r",
      "Assemblying reads: 6%\r",
      "Assemblying reads: 9%\r",
      "Assemblying reads: 13%\r",
      "Assemblying reads: 17%\r",
      "Assemblying reads: 21%\r",
      "Assemblying reads: 25%\r",
      "Assemblying reads: 28%\r",
      "Assemblying reads: 32%\r",
      "Assemblying reads: 36%\r",
      "Assemblying reads: 40%\r",
      "Assemblying reads: 43%\r",
      "Assemblying reads: 47%\r",
      "Assemblying reads: 51%\r",
      "Assemblying reads: 55%\r",
      "Assemblying reads: 59%\r",
      "Assemblying reads: 62%\r",
      "Assemblying reads: 66%\r",
      "Assemblying reads: 70%\r",
      "Assemblying reads: 74%\r",
      "Assemblying reads: 78%\r",
      "Assemblying reads: 81%\r",
      "Assemblying reads: 85%\r",
      "Assemblying reads: 89%\r",
      "Assemblying reads: 93%\r",
      "Assemblying reads: 97%\r",
      "Assemblying reads: 100%\r",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 2,100,838 / 3,052,493 (68.824%)\n",
      "Discarded reads ...................: 951,655 / 3,052,493 (31.176%)\n",
      "Not assembled reads ...............: 0 / 3,052,493 (0.000%)\n",
      "Assembled reads file...............: /analysis/20160609_pyrad/fastq/1HL_16A.assembled.fastq\n",
      "Discarded reads file...............: /analysis/20160609_pyrad/fastq/1HL_16A.discarded.fastq\n",
      "Unassembled forward reads file.....: /analysis/20160609_pyrad/fastq/1HL_16A.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: /analysis/20160609_pyrad/fastq/1HL_16A.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: /analysis/20160609_pyrad/fastq/1HL_17A_R1.fq\n",
      "Reverse reads file.................: /analysis/20160609_pyrad/fastq/1HL_17A_R2.fq\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 33\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 10\n",
      "Minimum read size after trimming...: 33\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 20\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.285335\n",
      "  C: 0.219349\n",
      "  G: 0.220786\n",
      "  T: 0.274530\n",
      "  1060536 uncalled bases\n",
      "Assemblying reads: 0%\r",
      "Assemblying reads: 4%\r",
      "Assemblying reads: 6%\r",
      "Assemblying reads: 9%\r",
      "Assemblying reads: 11%\r",
      "Assemblying reads: 14%\r",
      "Assemblying reads: 16%\r",
      "Assemblying reads: 19%\r",
      "Assemblying reads: 21%\r",
      "Assemblying reads: 24%\r",
      "Assemblying reads: 26%\r",
      "Assemblying reads: 29%\r",
      "Assemblying reads: 31%\r",
      "Assemblying reads: 34%\r",
      "Assemblying reads: 36%\r",
      "Assemblying reads: 39%\r",
      "Assemblying reads: 41%\r",
      "Assemblying reads: 44%\r",
      "Assemblying reads: 46%\r",
      "Assemblying reads: 49%\r",
      "Assemblying reads: 51%\r",
      "Assemblying reads: 54%\r",
      "Assemblying reads: 57%\r",
      "Assemblying reads: 59%\r",
      "Assemblying reads: 62%\r",
      "Assemblying reads: 64%\r",
      "Assemblying reads: 67%\r",
      "Assemblying reads: 69%\r",
      "Assemblying reads: 72%\r",
      "Assemblying reads: 74%\r",
      "Assemblying reads: 77%\r",
      "Assemblying reads: 79%\r",
      "Assemblying reads: 82%\r",
      "Assemblying reads: 84%\r",
      "Assemblying reads: 87%\r",
      "Assemblying reads: 89%\r",
      "Assemblying reads: 92%\r",
      "Assemblying reads: 94%\r",
      "Assemblying reads: 97%\r",
      "Assemblying reads: 99%\r",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 3,106,139 / 4,587,720 (67.706%)\n",
      "Discarded reads ...................: 1,481,581 / 4,587,720 (32.294%)\n",
      "Not assembled reads ...............: 0 / 4,587,720 (0.000%)\n",
      "Assembled reads file...............: /analysis/20160609_pyrad/fastq/1HL_17A.assembled.fastq\n",
      "Discarded reads file...............: /analysis/20160609_pyrad/fastq/1HL_17A.discarded.fastq\n",
      "Unassembled forward reads file.....: /analysis/20160609_pyrad/fastq/1HL_17A.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: /analysis/20160609_pyrad/fastq/1HL_17A.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: /analysis/20160609_pyrad/fastq/1HL_19A_R1.fq\n",
      "Reverse reads file.................: /analysis/20160609_pyrad/fastq/1HL_19A_R2.fq\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 33\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 10\n",
      "Minimum read size after trimming...: 33\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 20\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.286462\n",
      "  C: 0.221234\n",
      "  G: 0.223039\n",
      "  T: 0.269265\n",
      "  1059693 uncalled bases\n",
      "Assemblying reads: 0%\r",
      "Assemblying reads: 4%\r",
      "Assemblying reads: 6%\r",
      "Assemblying reads: 9%\r",
      "Assemblying reads: 11%\r",
      "Assemblying reads: 14%\r",
      "Assemblying reads: 16%\r",
      "Assemblying reads: 19%\r",
      "Assemblying reads: 21%\r",
      "Assemblying reads: 24%\r",
      "Assemblying reads: 26%\r",
      "Assemblying reads: 29%\r",
      "Assemblying reads: 31%\r",
      "Assemblying reads: 34%\r",
      "Assemblying reads: 36%\r",
      "Assemblying reads: 39%\r",
      "Assemblying reads: 41%\r",
      "Assemblying reads: 44%\r",
      "Assemblying reads: 46%\r",
      "Assemblying reads: 49%\r",
      "Assemblying reads: 51%\r",
      "Assemblying reads: 54%\r",
      "Assemblying reads: 56%\r",
      "Assemblying reads: 59%\r",
      "Assemblying reads: 61%\r",
      "Assemblying reads: 64%\r",
      "Assemblying reads: 66%\r",
      "Assemblying reads: 69%\r",
      "Assemblying reads: 71%\r",
      "Assemblying reads: 74%\r",
      "Assemblying reads: 76%\r",
      "Assemblying reads: 79%\r",
      "Assemblying reads: 81%\r",
      "Assemblying reads: 84%\r",
      "Assemblying reads: 86%\r",
      "Assemblying reads: 89%\r",
      "Assemblying reads: 91%\r",
      "Assemblying reads: 94%\r",
      "Assemblying reads: 96%\r",
      "Assemblying reads: 99%\r",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 3,174,884 / 4,607,650 (68.905%)\n",
      "Discarded reads ...................: 1,432,766 / 4,607,650 (31.095%)\n",
      "Not assembled reads ...............: 0 / 4,607,650 (0.000%)\n",
      "Assembled reads file...............: /analysis/20160609_pyrad/fastq/1HL_19A.assembled.fastq\n",
      "Discarded reads file...............: /analysis/20160609_pyrad/fastq/1HL_19A.discarded.fastq\n",
      "Unassembled forward reads file.....: /analysis/20160609_pyrad/fastq/1HL_19A.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: /analysis/20160609_pyrad/fastq/1HL_19A.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: /analysis/20160609_pyrad/fastq/1HL_1A_R1.fq\n",
      "Reverse reads file.................: /analysis/20160609_pyrad/fastq/1HL_1A_R2.fq\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 33\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 10\n",
      "Minimum read size after trimming...: 33\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 20\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.285947\n",
      "  C: 0.218539\n",
      "  G: 0.224484\n",
      "  T: 0.271030\n",
      "  691996 uncalled bases\n",
      "Assemblying reads: 0%\r",
      "Assemblying reads: 6%\r",
      "Assemblying reads: 9%\r",
      "Assemblying reads: 13%\r",
      "Assemblying reads: 17%\r",
      "Assemblying reads: 21%\r",
      "Assemblying reads: 25%\r",
      "Assemblying reads: 29%\r",
      "Assemblying reads: 33%\r",
      "Assemblying reads: 36%\r",
      "Assemblying reads: 40%\r",
      "Assemblying reads: 44%\r",
      "Assemblying reads: 48%\r",
      "Assemblying reads: 52%\r",
      "Assemblying reads: 56%\r",
      "Assemblying reads: 59%\r",
      "Assemblying reads: 63%\r",
      "Assemblying reads: 67%\r",
      "Assemblying reads: 71%\r",
      "Assemblying reads: 75%\r",
      "Assemblying reads: 79%\r",
      "Assemblying reads: 83%\r",
      "Assemblying reads: 86%\r",
      "Assemblying reads: 90%\r",
      "Assemblying reads: 94%\r",
      "Assemblying reads: 98%\r",
      "Assemblying reads: 100%\r",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 2,098,741 / 3,010,262 (69.720%)\n",
      "Discarded reads ...................: 911,521 / 3,010,262 (30.280%)\n",
      "Not assembled reads ...............: 0 / 3,010,262 (0.000%)\n",
      "Assembled reads file...............: /analysis/20160609_pyrad/fastq/1HL_1A.assembled.fastq\n",
      "Discarded reads file...............: /analysis/20160609_pyrad/fastq/1HL_1A.discarded.fastq\n",
      "Unassembled forward reads file.....: /analysis/20160609_pyrad/fastq/1HL_1A.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: /analysis/20160609_pyrad/fastq/1HL_1A.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: /analysis/20160609_pyrad/fastq/1HL_20A_R1.fq\n",
      "Reverse reads file.................: /analysis/20160609_pyrad/fastq/1HL_20A_R2.fq\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 33\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 10\n",
      "Minimum read size after trimming...: 33\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 20\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.287080\n",
      "  C: 0.218017\n",
      "  G: 0.225527\n",
      "  T: 0.269376\n",
      "  840036 uncalled bases\n",
      "Assemblying reads: 0%\r",
      "Assemblying reads: 5%\r",
      "Assemblying reads: 8%\r",
      "Assemblying reads: 11%\r",
      "Assemblying reads: 14%\r",
      "Assemblying reads: 17%\r",
      "Assemblying reads: 20%\r",
      "Assemblying reads: 23%\r",
      "Assemblying reads: 26%\r",
      "Assemblying reads: 30%\r",
      "Assemblying reads: 33%\r",
      "Assemblying reads: 36%\r",
      "Assemblying reads: 39%\r",
      "Assemblying reads: 42%\r",
      "Assemblying reads: 45%\r",
      "Assemblying reads: 48%\r",
      "Assemblying reads: 52%\r",
      "Assemblying reads: 55%\r",
      "Assemblying reads: 58%\r",
      "Assemblying reads: 61%\r",
      "Assemblying reads: 64%\r",
      "Assemblying reads: 67%\r",
      "Assemblying reads: 70%\r",
      "Assemblying reads: 73%\r",
      "Assemblying reads: 77%\r",
      "Assemblying reads: 80%\r",
      "Assemblying reads: 83%\r",
      "Assemblying reads: 86%\r",
      "Assemblying reads: 89%\r",
      "Assemblying reads: 92%\r",
      "Assemblying reads: 95%\r",
      "Assemblying reads: 99%\r",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 2,516,582 / 3,692,276 (68.158%)\n",
      "Discarded reads ...................: 1,175,694 / 3,692,276 (31.842%)\n",
      "Not assembled reads ...............: 0 / 3,692,276 (0.000%)\n",
      "Assembled reads file...............: /analysis/20160609_pyrad/fastq/1HL_20A.assembled.fastq\n",
      "Discarded reads file...............: /analysis/20160609_pyrad/fastq/1HL_20A.discarded.fastq\n",
      "Unassembled forward reads file.....: /analysis/20160609_pyrad/fastq/1HL_20A.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: /analysis/20160609_pyrad/fastq/1HL_20A.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: /analysis/20160609_pyrad/fastq/1HL_21A_R1.fq\n",
      "Reverse reads file.................: /analysis/20160609_pyrad/fastq/1HL_21A_R2.fq\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 33\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 10\n",
      "Minimum read size after trimming...: 33\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 20\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.284969\n",
      "  C: 0.220369\n",
      "  G: 0.224433\n",
      "  T: 0.270228\n",
      "  751300 uncalled bases\n",
      "Assemblying reads: 0%\r",
      "Assemblying reads: 5%\r",
      "Assemblying reads: 9%\r",
      "Assemblying reads: 12%\r",
      "Assemblying reads: 16%\r",
      "Assemblying reads: 19%\r",
      "Assemblying reads: 23%\r",
      "Assemblying reads: 26%\r",
      "Assemblying reads: 30%\r",
      "Assemblying reads: 33%\r",
      "Assemblying reads: 37%\r",
      "Assemblying reads: 40%\r",
      "Assemblying reads: 44%\r",
      "Assemblying reads: 48%\r",
      "Assemblying reads: 51%\r",
      "Assemblying reads: 55%\r",
      "Assemblying reads: 58%\r",
      "Assemblying reads: 62%\r",
      "Assemblying reads: 65%\r",
      "Assemblying reads: 69%\r",
      "Assemblying reads: 72%\r",
      "Assemblying reads: 76%\r",
      "Assemblying reads: 79%\r",
      "Assemblying reads: 83%\r",
      "Assemblying reads: 86%\r",
      "Assemblying reads: 90%\r",
      "Assemblying reads: 93%\r",
      "Assemblying reads: 97%\r",
      "Assemblying reads: 100%\r",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 2,203,056 / 3,278,190 (67.203%)\n",
      "Discarded reads ...................: 1,075,134 / 3,278,190 (32.797%)\n",
      "Not assembled reads ...............: 0 / 3,278,190 (0.000%)\n",
      "Assembled reads file...............: /analysis/20160609_pyrad/fastq/1HL_21A.assembled.fastq\n",
      "Discarded reads file...............: /analysis/20160609_pyrad/fastq/1HL_21A.discarded.fastq\n",
      "Unassembled forward reads file.....: /analysis/20160609_pyrad/fastq/1HL_21A.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: /analysis/20160609_pyrad/fastq/1HL_21A.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: /analysis/20160609_pyrad/fastq/1HL_22A_R1.fq\n",
      "Reverse reads file.................: /analysis/20160609_pyrad/fastq/1HL_22A_R2.fq\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 33\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 10\n",
      "Minimum read size after trimming...: 33\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 20\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.286768\n",
      "  C: 0.218420\n",
      "  G: 0.226207\n",
      "  T: 0.268605\n",
      "  619188 uncalled bases\n",
      "Assemblying reads: 0%\r",
      "Assemblying reads: 6%\r",
      "Assemblying reads: 11%\r",
      "Assemblying reads: 15%\r",
      "Assemblying reads: 19%\r",
      "Assemblying reads: 24%\r",
      "Assemblying reads: 28%\r",
      "Assemblying reads: 32%\r",
      "Assemblying reads: 37%\r",
      "Assemblying reads: 41%\r",
      "Assemblying reads: 45%\r",
      "Assemblying reads: 50%\r",
      "Assemblying reads: 54%\r",
      "Assemblying reads: 59%\r",
      "Assemblying reads: 63%\r",
      "Assemblying reads: 67%\r",
      "Assemblying reads: 72%\r",
      "Assemblying reads: 76%\r",
      "Assemblying reads: 80%\r",
      "Assemblying reads: 85%\r",
      "Assemblying reads: 89%\r",
      "Assemblying reads: 93%\r",
      "Assemblying reads: 98%\r",
      "Assemblying reads: 100%\r",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 1,807,508 / 2,666,966 (67.774%)\n",
      "Discarded reads ...................: 859,458 / 2,666,966 (32.226%)\n",
      "Not assembled reads ...............: 0 / 2,666,966 (0.000%)\n",
      "Assembled reads file...............: /analysis/20160609_pyrad/fastq/1HL_22A.assembled.fastq\n",
      "Discarded reads file...............: /analysis/20160609_pyrad/fastq/1HL_22A.discarded.fastq\n",
      "Unassembled forward reads file.....: /analysis/20160609_pyrad/fastq/1HL_22A.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: /analysis/20160609_pyrad/fastq/1HL_22A.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: /analysis/20160609_pyrad/fastq/1HL_23A_R1.fq\n",
      "Reverse reads file.................: /analysis/20160609_pyrad/fastq/1HL_23A_R2.fq\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 33\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 10\n",
      "Minimum read size after trimming...: 33\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 20\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.283043\n",
      "  C: 0.218893\n",
      "  G: 0.223030\n",
      "  T: 0.275034\n",
      "  1053503 uncalled bases\n",
      "Assemblying reads: 0%\r",
      "Assemblying reads: 4%\r",
      "Assemblying reads: 6%\r",
      "Assemblying reads: 9%\r",
      "Assemblying reads: 11%\r",
      "Assemblying reads: 14%\r",
      "Assemblying reads: 16%\r",
      "Assemblying reads: 19%\r",
      "Assemblying reads: 21%\r",
      "Assemblying reads: 24%\r",
      "Assemblying reads: 26%\r",
      "Assemblying reads: 29%\r",
      "Assemblying reads: 31%\r",
      "Assemblying reads: 34%\r",
      "Assemblying reads: 36%\r",
      "Assemblying reads: 39%\r",
      "Assemblying reads: 42%\r",
      "Assemblying reads: 44%\r",
      "Assemblying reads: 47%\r",
      "Assemblying reads: 49%\r",
      "Assemblying reads: 52%\r",
      "Assemblying reads: 54%\r",
      "Assemblying reads: 57%\r",
      "Assemblying reads: 59%\r",
      "Assemblying reads: 62%\r",
      "Assemblying reads: 64%\r",
      "Assemblying reads: 67%\r",
      "Assemblying reads: 69%\r",
      "Assemblying reads: 72%\r",
      "Assemblying reads: 75%\r",
      "Assemblying reads: 77%\r",
      "Assemblying reads: 80%\r",
      "Assemblying reads: 82%\r",
      "Assemblying reads: 85%\r",
      "Assemblying reads: 87%\r",
      "Assemblying reads: 90%\r",
      "Assemblying reads: 92%\r",
      "Assemblying reads: 95%\r",
      "Assemblying reads: 97%\r",
      "Assemblying reads: 100%\r",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 3,117,475 / 4,566,989 (68.261%)\n",
      "Discarded reads ...................: 1,449,514 / 4,566,989 (31.739%)\n",
      "Not assembled reads ...............: 0 / 4,566,989 (0.000%)\n",
      "Assembled reads file...............: /analysis/20160609_pyrad/fastq/1HL_23A.assembled.fastq\n",
      "Discarded reads file...............: /analysis/20160609_pyrad/fastq/1HL_23A.discarded.fastq\n",
      "Unassembled forward reads file.....: /analysis/20160609_pyrad/fastq/1HL_23A.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: /analysis/20160609_pyrad/fastq/1HL_23A.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: /analysis/20160609_pyrad/fastq/1HL_24A_R1.fq\n",
      "Reverse reads file.................: /analysis/20160609_pyrad/fastq/1HL_24A_R2.fq\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 33\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 10\n",
      "Minimum read size after trimming...: 33\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 20\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.284089\n",
      "  C: 0.219803\n",
      "  G: 0.220925\n",
      "  T: 0.275182\n",
      "  786608 uncalled bases\n",
      "Assemblying reads: 0%\r",
      "Assemblying reads: 5%\r",
      "Assemblying reads: 8%\r",
      "Assemblying reads: 12%\r",
      "Assemblying reads: 15%\r",
      "Assemblying reads: 18%\r",
      "Assemblying reads: 22%\r",
      "Assemblying reads: 25%\r",
      "Assemblying reads: 29%\r",
      "Assemblying reads: 32%\r",
      "Assemblying reads: 35%\r",
      "Assemblying reads: 39%\r",
      "Assemblying reads: 42%\r",
      "Assemblying reads: 46%\r",
      "Assemblying reads: 49%\r",
      "Assemblying reads: 52%\r",
      "Assemblying reads: 56%\r",
      "Assemblying reads: 59%\r",
      "Assemblying reads: 62%\r",
      "Assemblying reads: 66%\r",
      "Assemblying reads: 69%\r",
      "Assemblying reads: 73%\r",
      "Assemblying reads: 76%\r",
      "Assemblying reads: 79%\r",
      "Assemblying reads: 83%\r",
      "Assemblying reads: 86%\r",
      "Assemblying reads: 90%\r",
      "Assemblying reads: 93%\r",
      "Assemblying reads: 96%\r",
      "Assemblying reads: 100%\r",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 2,290,319 / 3,418,136 (67.005%)\n",
      "Discarded reads ...................: 1,127,817 / 3,418,136 (32.995%)\n",
      "Not assembled reads ...............: 0 / 3,418,136 (0.000%)\n",
      "Assembled reads file...............: /analysis/20160609_pyrad/fastq/1HL_24A.assembled.fastq\n",
      "Discarded reads file...............: /analysis/20160609_pyrad/fastq/1HL_24A.discarded.fastq\n",
      "Unassembled forward reads file.....: /analysis/20160609_pyrad/fastq/1HL_24A.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: /analysis/20160609_pyrad/fastq/1HL_24A.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: /analysis/20160609_pyrad/fastq/1HL_25A_R1.fq\n",
      "Reverse reads file.................: /analysis/20160609_pyrad/fastq/1HL_25A_R2.fq\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 33\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 10\n",
      "Minimum read size after trimming...: 33\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 20\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.285912\n",
      "  C: 0.220616\n",
      "  G: 0.222195\n",
      "  T: 0.271276\n",
      "  871202 uncalled bases\n",
      "Assemblying reads: 0%\r",
      "Assemblying reads: 4%\r",
      "Assemblying reads: 7%\r",
      "Assemblying reads: 10%\r",
      "Assemblying reads: 14%\r",
      "Assemblying reads: 17%\r",
      "Assemblying reads: 20%\r",
      "Assemblying reads: 23%\r",
      "Assemblying reads: 26%\r",
      "Assemblying reads: 29%\r",
      "Assemblying reads: 32%\r",
      "Assemblying reads: 35%\r",
      "Assemblying reads: 38%\r",
      "Assemblying reads: 41%\r",
      "Assemblying reads: 44%\r",
      "Assemblying reads: 47%\r",
      "Assemblying reads: 50%\r",
      "Assemblying reads: 53%\r",
      "Assemblying reads: 56%\r",
      "Assemblying reads: 59%\r",
      "Assemblying reads: 62%\r",
      "Assemblying reads: 65%\r",
      "Assemblying reads: 68%\r",
      "Assemblying reads: 71%\r",
      "Assemblying reads: 75%\r",
      "Assemblying reads: 78%\r",
      "Assemblying reads: 81%\r",
      "Assemblying reads: 84%\r",
      "Assemblying reads: 87%\r",
      "Assemblying reads: 90%\r",
      "Assemblying reads: 93%\r",
      "Assemblying reads: 96%\r",
      "Assemblying reads: 99%\r",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 2,606,955 / 3,794,161 (68.710%)\n",
      "Discarded reads ...................: 1,187,206 / 3,794,161 (31.290%)\n",
      "Not assembled reads ...............: 0 / 3,794,161 (0.000%)\n",
      "Assembled reads file...............: /analysis/20160609_pyrad/fastq/1HL_25A.assembled.fastq\n",
      "Discarded reads file...............: /analysis/20160609_pyrad/fastq/1HL_25A.discarded.fastq\n",
      "Unassembled forward reads file.....: /analysis/20160609_pyrad/fastq/1HL_25A.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: /analysis/20160609_pyrad/fastq/1HL_25A.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: /analysis/20160609_pyrad/fastq/1HL_26A_R1.fq\n",
      "Reverse reads file.................: /analysis/20160609_pyrad/fastq/1HL_26A_R2.fq\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 33\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 10\n",
      "Minimum read size after trimming...: 33\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 20\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.283029\n",
      "  C: 0.223790\n",
      "  G: 0.227327\n",
      "  T: 0.265854\n",
      "  962198 uncalled bases\n",
      "Assemblying reads: 0%\r",
      "Assemblying reads: 4%\r",
      "Assemblying reads: 7%\r",
      "Assemblying reads: 9%\r",
      "Assemblying reads: 12%\r",
      "Assemblying reads: 15%\r",
      "Assemblying reads: 18%\r",
      "Assemblying reads: 21%\r",
      "Assemblying reads: 23%\r",
      "Assemblying reads: 26%\r",
      "Assemblying reads: 29%\r",
      "Assemblying reads: 32%\r",
      "Assemblying reads: 35%\r",
      "Assemblying reads: 37%\r",
      "Assemblying reads: 40%\r",
      "Assemblying reads: 43%\r",
      "Assemblying reads: 46%\r",
      "Assemblying reads: 48%\r",
      "Assemblying reads: 51%\r",
      "Assemblying reads: 54%\r",
      "Assemblying reads: 57%\r",
      "Assemblying reads: 60%\r",
      "Assemblying reads: 62%\r",
      "Assemblying reads: 65%\r",
      "Assemblying reads: 68%\r",
      "Assemblying reads: 71%\r",
      "Assemblying reads: 73%\r",
      "Assemblying reads: 76%\r",
      "Assemblying reads: 79%\r",
      "Assemblying reads: 82%\r",
      "Assemblying reads: 85%\r",
      "Assemblying reads: 87%\r",
      "Assemblying reads: 90%\r",
      "Assemblying reads: 93%\r",
      "Assemblying reads: 96%\r",
      "Assemblying reads: 98%\r",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 2,915,184 / 4,165,278 (69.988%)\n",
      "Discarded reads ...................: 1,250,094 / 4,165,278 (30.012%)\n",
      "Not assembled reads ...............: 0 / 4,165,278 (0.000%)\n",
      "Assembled reads file...............: /analysis/20160609_pyrad/fastq/1HL_26A.assembled.fastq\n",
      "Discarded reads file...............: /analysis/20160609_pyrad/fastq/1HL_26A.discarded.fastq\n",
      "Unassembled forward reads file.....: /analysis/20160609_pyrad/fastq/1HL_26A.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: /analysis/20160609_pyrad/fastq/1HL_26A.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: /analysis/20160609_pyrad/fastq/1HL_27A_R1.fq\n",
      "Reverse reads file.................: /analysis/20160609_pyrad/fastq/1HL_27A_R2.fq\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 33\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 10\n",
      "Minimum read size after trimming...: 33\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 20\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.281204\n",
      "  C: 0.223493\n",
      "  G: 0.225397\n",
      "  T: 0.269907\n",
      "  957521 uncalled bases\n",
      "Assemblying reads: 0%\r",
      "Assemblying reads: 4%\r",
      "Assemblying reads: 7%\r",
      "Assemblying reads: 10%\r",
      "Assemblying reads: 12%\r",
      "Assemblying reads: 15%\r",
      "Assemblying reads: 18%\r",
      "Assemblying reads: 21%\r",
      "Assemblying reads: 23%\r",
      "Assemblying reads: 26%\r",
      "Assemblying reads: 29%\r",
      "Assemblying reads: 32%\r",
      "Assemblying reads: 35%\r",
      "Assemblying reads: 37%\r",
      "Assemblying reads: 40%\r",
      "Assemblying reads: 43%\r",
      "Assemblying reads: 46%\r",
      "Assemblying reads: 49%\r",
      "Assemblying reads: 51%\r",
      "Assemblying reads: 54%\r",
      "Assemblying reads: 57%\r",
      "Assemblying reads: 60%\r",
      "Assemblying reads: 63%\r",
      "Assemblying reads: 65%\r",
      "Assemblying reads: 68%\r",
      "Assemblying reads: 71%\r",
      "Assemblying reads: 74%\r",
      "Assemblying reads: 77%\r",
      "Assemblying reads: 79%\r",
      "Assemblying reads: 82%\r",
      "Assemblying reads: 85%\r",
      "Assemblying reads: 88%\r",
      "Assemblying reads: 90%\r",
      "Assemblying reads: 93%\r",
      "Assemblying reads: 96%\r",
      "Assemblying reads: 99%\r",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 2,887,147 / 4,146,188 (69.634%)\n",
      "Discarded reads ...................: 1,259,041 / 4,146,188 (30.366%)\n",
      "Not assembled reads ...............: 0 / 4,146,188 (0.000%)\n",
      "Assembled reads file...............: /analysis/20160609_pyrad/fastq/1HL_27A.assembled.fastq\n",
      "Discarded reads file...............: /analysis/20160609_pyrad/fastq/1HL_27A.discarded.fastq\n",
      "Unassembled forward reads file.....: /analysis/20160609_pyrad/fastq/1HL_27A.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: /analysis/20160609_pyrad/fastq/1HL_27A.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: /analysis/20160609_pyrad/fastq/1HL_28A_R1.fq\n",
      "Reverse reads file.................: /analysis/20160609_pyrad/fastq/1HL_28A_R2.fq\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 33\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 10\n",
      "Minimum read size after trimming...: 33\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 20\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.285403\n",
      "  C: 0.219129\n",
      "  G: 0.223981\n",
      "  T: 0.271486\n",
      "  1082276 uncalled bases\n",
      "Assemblying reads: 0%\r",
      "Assemblying reads: 3%\r",
      "Assemblying reads: 6%\r",
      "Assemblying reads: 8%\r",
      "Assemblying reads: 11%\r",
      "Assemblying reads: 13%\r",
      "Assemblying reads: 16%\r",
      "Assemblying reads: 18%\r",
      "Assemblying reads: 21%\r",
      "Assemblying reads: 23%\r",
      "Assemblying reads: 25%\r",
      "Assemblying reads: 28%\r",
      "Assemblying reads: 30%\r",
      "Assemblying reads: 33%\r",
      "Assemblying reads: 35%\r",
      "Assemblying reads: 38%\r",
      "Assemblying reads: 40%\r",
      "Assemblying reads: 43%\r",
      "Assemblying reads: 45%\r",
      "Assemblying reads: 47%\r",
      "Assemblying reads: 50%\r",
      "Assemblying reads: 52%\r",
      "Assemblying reads: 55%\r",
      "Assemblying reads: 57%\r",
      "Assemblying reads: 60%\r",
      "Assemblying reads: 62%\r",
      "Assemblying reads: 65%\r",
      "Assemblying reads: 67%\r",
      "Assemblying reads: 69%\r",
      "Assemblying reads: 72%\r",
      "Assemblying reads: 74%\r",
      "Assemblying reads: 77%\r",
      "Assemblying reads: 79%\r",
      "Assemblying reads: 82%\r",
      "Assemblying reads: 84%\r",
      "Assemblying reads: 87%\r",
      "Assemblying reads: 89%\r",
      "Assemblying reads: 91%\r",
      "Assemblying reads: 94%\r",
      "Assemblying reads: 96%\r",
      "Assemblying reads: 99%\r",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 3,182,673 / 4,733,136 (67.242%)\n",
      "Discarded reads ...................: 1,550,463 / 4,733,136 (32.758%)\n",
      "Not assembled reads ...............: 0 / 4,733,136 (0.000%)\n",
      "Assembled reads file...............: /analysis/20160609_pyrad/fastq/1HL_28A.assembled.fastq\n",
      "Discarded reads file...............: /analysis/20160609_pyrad/fastq/1HL_28A.discarded.fastq\n",
      "Unassembled forward reads file.....: /analysis/20160609_pyrad/fastq/1HL_28A.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: /analysis/20160609_pyrad/fastq/1HL_28A.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: /analysis/20160609_pyrad/fastq/1HL_29A_R1.fq\n",
      "Reverse reads file.................: /analysis/20160609_pyrad/fastq/1HL_29A_R2.fq\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 33\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 10\n",
      "Minimum read size after trimming...: 33\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 20\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.282155\n",
      "  C: 0.226136\n",
      "  G: 0.227452\n",
      "  T: 0.264257\n",
      "  977328 uncalled bases\n",
      "Assemblying reads: 0%\r",
      "Assemblying reads: 4%\r",
      "Assemblying reads: 7%\r",
      "Assemblying reads: 9%\r",
      "Assemblying reads: 12%\r",
      "Assemblying reads: 15%\r",
      "Assemblying reads: 18%\r",
      "Assemblying reads: 20%\r",
      "Assemblying reads: 23%\r",
      "Assemblying reads: 26%\r",
      "Assemblying reads: 28%\r",
      "Assemblying reads: 31%\r",
      "Assemblying reads: 34%\r",
      "Assemblying reads: 37%\r",
      "Assemblying reads: 39%\r",
      "Assemblying reads: 42%\r",
      "Assemblying reads: 45%\r",
      "Assemblying reads: 48%\r",
      "Assemblying reads: 50%\r",
      "Assemblying reads: 53%\r",
      "Assemblying reads: 56%\r",
      "Assemblying reads: 58%\r",
      "Assemblying reads: 61%\r",
      "Assemblying reads: 64%\r",
      "Assemblying reads: 67%\r",
      "Assemblying reads: 69%\r",
      "Assemblying reads: 72%\r",
      "Assemblying reads: 75%\r",
      "Assemblying reads: 78%\r",
      "Assemblying reads: 80%\r",
      "Assemblying reads: 83%\r",
      "Assemblying reads: 86%\r",
      "Assemblying reads: 89%\r",
      "Assemblying reads: 91%\r",
      "Assemblying reads: 94%\r",
      "Assemblying reads: 97%\r",
      "Assemblying reads: 99%\r",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 3,004,016 / 4,238,592 (70.873%)\n",
      "Discarded reads ...................: 1,234,576 / 4,238,592 (29.127%)\n",
      "Not assembled reads ...............: 0 / 4,238,592 (0.000%)\n",
      "Assembled reads file...............: /analysis/20160609_pyrad/fastq/1HL_29A.assembled.fastq\n",
      "Discarded reads file...............: /analysis/20160609_pyrad/fastq/1HL_29A.discarded.fastq\n",
      "Unassembled forward reads file.....: /analysis/20160609_pyrad/fastq/1HL_29A.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: /analysis/20160609_pyrad/fastq/1HL_29A.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: /analysis/20160609_pyrad/fastq/1HL_2A_R1.fq\n",
      "Reverse reads file.................: /analysis/20160609_pyrad/fastq/1HL_2A_R2.fq\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 33\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 10\n",
      "Minimum read size after trimming...: 33\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 20\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.287959\n",
      "  C: 0.219214\n",
      "  G: 0.223470\n",
      "  T: 0.269357\n",
      "  830284 uncalled bases\n",
      "Assemblying reads: 0%\r",
      "Assemblying reads: 5%\r",
      "Assemblying reads: 8%\r",
      "Assemblying reads: 11%\r",
      "Assemblying reads: 14%\r",
      "Assemblying reads: 18%\r",
      "Assemblying reads: 21%\r",
      "Assemblying reads: 24%\r",
      "Assemblying reads: 27%\r",
      "Assemblying reads: 30%\r",
      "Assemblying reads: 34%\r",
      "Assemblying reads: 37%\r",
      "Assemblying reads: 40%\r",
      "Assemblying reads: 43%\r",
      "Assemblying reads: 47%\r",
      "Assemblying reads: 50%\r",
      "Assemblying reads: 53%\r",
      "Assemblying reads: 56%\r",
      "Assemblying reads: 59%\r",
      "Assemblying reads: 63%\r",
      "Assemblying reads: 66%\r",
      "Assemblying reads: 69%\r",
      "Assemblying reads: 72%\r",
      "Assemblying reads: 75%\r",
      "Assemblying reads: 79%\r",
      "Assemblying reads: 82%\r",
      "Assemblying reads: 85%\r",
      "Assemblying reads: 88%\r",
      "Assemblying reads: 92%\r",
      "Assemblying reads: 95%\r",
      "Assemblying reads: 98%\r",
      "Assemblying reads: 100%\r",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 2,525,685 / 3,592,790 (70.299%)\n",
      "Discarded reads ...................: 1,067,105 / 3,592,790 (29.701%)\n",
      "Not assembled reads ...............: 0 / 3,592,790 (0.000%)\n",
      "Assembled reads file...............: /analysis/20160609_pyrad/fastq/1HL_2A.assembled.fastq\n",
      "Discarded reads file...............: /analysis/20160609_pyrad/fastq/1HL_2A.discarded.fastq\n",
      "Unassembled forward reads file.....: /analysis/20160609_pyrad/fastq/1HL_2A.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: /analysis/20160609_pyrad/fastq/1HL_2A.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: /analysis/20160609_pyrad/fastq/1HL_31A_R1.fq\n",
      "Reverse reads file.................: /analysis/20160609_pyrad/fastq/1HL_31A_R2.fq\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 33\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 10\n",
      "Minimum read size after trimming...: 33\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 20\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.286352\n",
      "  C: 0.223404\n",
      "  G: 0.222343\n",
      "  T: 0.267902\n",
      "  726029 uncalled bases\n",
      "Assemblying reads: 0%\r",
      "Assemblying reads: 5%\r",
      "Assemblying reads: 9%\r",
      "Assemblying reads: 13%\r",
      "Assemblying reads: 16%\r",
      "Assemblying reads: 20%\r",
      "Assemblying reads: 24%\r",
      "Assemblying reads: 27%\r",
      "Assemblying reads: 31%\r",
      "Assemblying reads: 34%\r",
      "Assemblying reads: 38%\r",
      "Assemblying reads: 42%\r",
      "Assemblying reads: 45%\r",
      "Assemblying reads: 49%\r",
      "Assemblying reads: 53%\r",
      "Assemblying reads: 56%\r",
      "Assemblying reads: 60%\r",
      "Assemblying reads: 64%\r",
      "Assemblying reads: 67%\r",
      "Assemblying reads: 71%\r",
      "Assemblying reads: 75%\r",
      "Assemblying reads: 78%\r",
      "Assemblying reads: 82%\r",
      "Assemblying reads: 85%\r",
      "Assemblying reads: 89%\r",
      "Assemblying reads: 93%\r",
      "Assemblying reads: 96%\r",
      "Assemblying reads: 100%\r",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 2,203,255 / 3,177,985 (69.329%)\n",
      "Discarded reads ...................: 974,730 / 3,177,985 (30.671%)\n",
      "Not assembled reads ...............: 0 / 3,177,985 (0.000%)\n",
      "Assembled reads file...............: /analysis/20160609_pyrad/fastq/1HL_31A.assembled.fastq\n",
      "Discarded reads file...............: /analysis/20160609_pyrad/fastq/1HL_31A.discarded.fastq\n",
      "Unassembled forward reads file.....: /analysis/20160609_pyrad/fastq/1HL_31A.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: /analysis/20160609_pyrad/fastq/1HL_31A.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: /analysis/20160609_pyrad/fastq/1HL_33A_R1.fq\n",
      "Reverse reads file.................: /analysis/20160609_pyrad/fastq/1HL_33A_R2.fq\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 33\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 10\n",
      "Minimum read size after trimming...: 33\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 20\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.285419\n",
      "  C: 0.224124\n",
      "  G: 0.221250\n",
      "  T: 0.269207\n",
      "  789912 uncalled bases\n",
      "Assemblying reads: 0%\r",
      "Assemblying reads: 5%\r",
      "Assemblying reads: 8%\r",
      "Assemblying reads: 12%\r",
      "Assemblying reads: 15%\r",
      "Assemblying reads: 18%\r",
      "Assemblying reads: 22%\r",
      "Assemblying reads: 25%\r",
      "Assemblying reads: 28%\r",
      "Assemblying reads: 32%\r",
      "Assemblying reads: 35%\r",
      "Assemblying reads: 38%\r",
      "Assemblying reads: 42%\r",
      "Assemblying reads: 45%\r",
      "Assemblying reads: 48%\r",
      "Assemblying reads: 52%\r",
      "Assemblying reads: 55%\r",
      "Assemblying reads: 59%\r",
      "Assemblying reads: 62%\r",
      "Assemblying reads: 65%\r",
      "Assemblying reads: 69%\r",
      "Assemblying reads: 72%\r",
      "Assemblying reads: 75%\r",
      "Assemblying reads: 79%\r",
      "Assemblying reads: 82%\r",
      "Assemblying reads: 85%\r",
      "Assemblying reads: 89%\r",
      "Assemblying reads: 92%\r",
      "Assemblying reads: 95%\r",
      "Assemblying reads: 99%\r",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 2,357,587 / 3,450,088 (68.334%)\n",
      "Discarded reads ...................: 1,092,501 / 3,450,088 (31.666%)\n",
      "Not assembled reads ...............: 0 / 3,450,088 (0.000%)\n",
      "Assembled reads file...............: /analysis/20160609_pyrad/fastq/1HL_33A.assembled.fastq\n",
      "Discarded reads file...............: /analysis/20160609_pyrad/fastq/1HL_33A.discarded.fastq\n",
      "Unassembled forward reads file.....: /analysis/20160609_pyrad/fastq/1HL_33A.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: /analysis/20160609_pyrad/fastq/1HL_33A.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: /analysis/20160609_pyrad/fastq/1HL_34A_R1.fq\n",
      "Reverse reads file.................: /analysis/20160609_pyrad/fastq/1HL_34A_R2.fq\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 33\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 10\n",
      "Minimum read size after trimming...: 33\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 20\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.286763\n",
      "  C: 0.219421\n",
      "  G: 0.220701\n",
      "  T: 0.273115\n",
      "  844931 uncalled bases\n",
      "Assemblying reads: 0%\r",
      "Assemblying reads: 5%\r",
      "Assemblying reads: 8%\r",
      "Assemblying reads: 11%\r",
      "Assemblying reads: 14%\r",
      "Assemblying reads: 17%\r",
      "Assemblying reads: 20%\r",
      "Assemblying reads: 24%\r",
      "Assemblying reads: 27%\r",
      "Assemblying reads: 30%\r",
      "Assemblying reads: 33%\r",
      "Assemblying reads: 36%\r",
      "Assemblying reads: 40%\r",
      "Assemblying reads: 43%\r",
      "Assemblying reads: 46%\r",
      "Assemblying reads: 49%\r",
      "Assemblying reads: 52%\r",
      "Assemblying reads: 55%\r",
      "Assemblying reads: 59%\r",
      "Assemblying reads: 62%\r",
      "Assemblying reads: 65%\r",
      "Assemblying reads: 68%\r",
      "Assemblying reads: 71%\r",
      "Assemblying reads: 74%\r",
      "Assemblying reads: 78%\r",
      "Assemblying reads: 81%\r",
      "Assemblying reads: 84%\r",
      "Assemblying reads: 87%\r",
      "Assemblying reads: 90%\r",
      "Assemblying reads: 94%\r",
      "Assemblying reads: 97%\r",
      "Assemblying reads: 100%\r",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 2,461,367 / 3,643,848 (67.549%)\n",
      "Discarded reads ...................: 1,182,481 / 3,643,848 (32.451%)\n",
      "Not assembled reads ...............: 0 / 3,643,848 (0.000%)\n",
      "Assembled reads file...............: /analysis/20160609_pyrad/fastq/1HL_34A.assembled.fastq\n",
      "Discarded reads file...............: /analysis/20160609_pyrad/fastq/1HL_34A.discarded.fastq\n",
      "Unassembled forward reads file.....: /analysis/20160609_pyrad/fastq/1HL_34A.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: /analysis/20160609_pyrad/fastq/1HL_34A.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: /analysis/20160609_pyrad/fastq/1HL_35A_R1.fq\n",
      "Reverse reads file.................: /analysis/20160609_pyrad/fastq/1HL_35A_R2.fq\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 33\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 10\n",
      "Minimum read size after trimming...: 33\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 20\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.288171\n",
      "  C: 0.218204\n",
      "  G: 0.222090\n",
      "  T: 0.271534\n",
      "  757197 uncalled bases\n",
      "Assemblying reads: 0%\r",
      "Assemblying reads: 5%\r",
      "Assemblying reads: 9%\r",
      "Assemblying reads: 12%\r",
      "Assemblying reads: 16%\r",
      "Assemblying reads: 19%\r",
      "Assemblying reads: 23%\r",
      "Assemblying reads: 26%\r",
      "Assemblying reads: 30%\r",
      "Assemblying reads: 33%\r",
      "Assemblying reads: 37%\r",
      "Assemblying reads: 40%\r",
      "Assemblying reads: 44%\r",
      "Assemblying reads: 47%\r",
      "Assemblying reads: 51%\r",
      "Assemblying reads: 54%\r",
      "Assemblying reads: 58%\r",
      "Assemblying reads: 61%\r",
      "Assemblying reads: 65%\r",
      "Assemblying reads: 68%\r",
      "Assemblying reads: 72%\r",
      "Assemblying reads: 75%\r",
      "Assemblying reads: 79%\r",
      "Assemblying reads: 82%\r",
      "Assemblying reads: 86%\r",
      "Assemblying reads: 89%\r",
      "Assemblying reads: 93%\r",
      "Assemblying reads: 96%\r",
      "Assemblying reads: 100%\r",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 2,222,135 / 3,298,636 (67.365%)\n",
      "Discarded reads ...................: 1,076,501 / 3,298,636 (32.635%)\n",
      "Not assembled reads ...............: 0 / 3,298,636 (0.000%)\n",
      "Assembled reads file...............: /analysis/20160609_pyrad/fastq/1HL_35A.assembled.fastq\n",
      "Discarded reads file...............: /analysis/20160609_pyrad/fastq/1HL_35A.discarded.fastq\n",
      "Unassembled forward reads file.....: /analysis/20160609_pyrad/fastq/1HL_35A.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: /analysis/20160609_pyrad/fastq/1HL_35A.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: /analysis/20160609_pyrad/fastq/1HL_3A_R1.fq\n",
      "Reverse reads file.................: /analysis/20160609_pyrad/fastq/1HL_3A_R2.fq\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 33\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 10\n",
      "Minimum read size after trimming...: 33\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 20\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.288734\n",
      "  C: 0.216823\n",
      "  G: 0.222551\n",
      "  T: 0.271893\n",
      "  666878 uncalled bases\n",
      "Assemblying reads: 0%\r",
      "Assemblying reads: 6%\r",
      "Assemblying reads: 10%\r",
      "Assemblying reads: 14%\r",
      "Assemblying reads: 18%\r",
      "Assemblying reads: 22%\r",
      "Assemblying reads: 26%\r",
      "Assemblying reads: 30%\r",
      "Assemblying reads: 34%\r",
      "Assemblying reads: 38%\r",
      "Assemblying reads: 42%\r",
      "Assemblying reads: 46%\r",
      "Assemblying reads: 50%\r",
      "Assemblying reads: 54%\r",
      "Assemblying reads: 58%\r",
      "Assemblying reads: 62%\r",
      "Assemblying reads: 66%\r",
      "Assemblying reads: 70%\r",
      "Assemblying reads: 74%\r",
      "Assemblying reads: 78%\r",
      "Assemblying reads: 82%\r",
      "Assemblying reads: 86%\r",
      "Assemblying reads: 90%\r",
      "Assemblying reads: 94%\r",
      "Assemblying reads: 98%\r",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 1,994,744 / 2,885,740 (69.124%)\n",
      "Discarded reads ...................: 890,996 / 2,885,740 (30.876%)\n",
      "Not assembled reads ...............: 0 / 2,885,740 (0.000%)\n",
      "Assembled reads file...............: /analysis/20160609_pyrad/fastq/1HL_3A.assembled.fastq\n",
      "Discarded reads file...............: /analysis/20160609_pyrad/fastq/1HL_3A.discarded.fastq\n",
      "Unassembled forward reads file.....: /analysis/20160609_pyrad/fastq/1HL_3A.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: /analysis/20160609_pyrad/fastq/1HL_3A.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: /analysis/20160609_pyrad/fastq/1HL_4A_R1.fq\n",
      "Reverse reads file.................: /analysis/20160609_pyrad/fastq/1HL_4A_R2.fq\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 33\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 10\n",
      "Minimum read size after trimming...: 33\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 20\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.285469\n",
      "  C: 0.221915\n",
      "  G: 0.225571\n",
      "  T: 0.267044\n",
      "  822572 uncalled bases\n",
      "Assemblying reads: 0%\r",
      "Assemblying reads: 5%\r",
      "Assemblying reads: 8%\r",
      "Assemblying reads: 11%\r",
      "Assemblying reads: 15%\r",
      "Assemblying reads: 18%\r",
      "Assemblying reads: 21%\r",
      "Assemblying reads: 25%\r",
      "Assemblying reads: 28%\r",
      "Assemblying reads: 31%\r",
      "Assemblying reads: 34%\r",
      "Assemblying reads: 38%\r",
      "Assemblying reads: 41%\r",
      "Assemblying reads: 44%\r",
      "Assemblying reads: 48%\r",
      "Assemblying reads: 51%\r",
      "Assemblying reads: 54%\r",
      "Assemblying reads: 57%\r",
      "Assemblying reads: 61%\r",
      "Assemblying reads: 64%\r",
      "Assemblying reads: 67%\r",
      "Assemblying reads: 71%\r",
      "Assemblying reads: 74%\r",
      "Assemblying reads: 77%\r",
      "Assemblying reads: 81%\r",
      "Assemblying reads: 84%\r",
      "Assemblying reads: 87%\r",
      "Assemblying reads: 90%\r",
      "Assemblying reads: 94%\r",
      "Assemblying reads: 97%\r",
      "Assemblying reads: 100%\r",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 2,492,589 / 3,510,860 (70.997%)\n",
      "Discarded reads ...................: 1,018,271 / 3,510,860 (29.003%)\n",
      "Not assembled reads ...............: 0 / 3,510,860 (0.000%)\n",
      "Assembled reads file...............: /analysis/20160609_pyrad/fastq/1HL_4A.assembled.fastq\n",
      "Discarded reads file...............: /analysis/20160609_pyrad/fastq/1HL_4A.discarded.fastq\n",
      "Unassembled forward reads file.....: /analysis/20160609_pyrad/fastq/1HL_4A.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: /analysis/20160609_pyrad/fastq/1HL_4A.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: /analysis/20160609_pyrad/fastq/1HL_5A_R1.fq\n",
      "Reverse reads file.................: /analysis/20160609_pyrad/fastq/1HL_5A_R2.fq\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 33\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 10\n",
      "Minimum read size after trimming...: 33\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 20\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.284051\n",
      "  C: 0.220209\n",
      "  G: 0.226839\n",
      "  T: 0.268900\n",
      "  635240 uncalled bases\n",
      "Assemblying reads: 0%\r",
      "Assemblying reads: 6%\r",
      "Assemblying reads: 11%\r",
      "Assemblying reads: 15%\r",
      "Assemblying reads: 19%\r",
      "Assemblying reads: 23%\r",
      "Assemblying reads: 28%\r",
      "Assemblying reads: 32%\r",
      "Assemblying reads: 36%\r",
      "Assemblying reads: 40%\r",
      "Assemblying reads: 45%\r",
      "Assemblying reads: 49%\r",
      "Assemblying reads: 53%\r",
      "Assemblying reads: 57%\r",
      "Assemblying reads: 62%\r",
      "Assemblying reads: 66%\r",
      "Assemblying reads: 70%\r",
      "Assemblying reads: 74%\r",
      "Assemblying reads: 79%\r",
      "Assemblying reads: 83%\r",
      "Assemblying reads: 87%\r",
      "Assemblying reads: 91%\r",
      "Assemblying reads: 96%\r",
      "Assemblying reads: 100%\r",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 1,914,213 / 2,720,076 (70.374%)\n",
      "Discarded reads ...................: 805,863 / 2,720,076 (29.626%)\n",
      "Not assembled reads ...............: 0 / 2,720,076 (0.000%)\n",
      "Assembled reads file...............: /analysis/20160609_pyrad/fastq/1HL_5A.assembled.fastq\n",
      "Discarded reads file...............: /analysis/20160609_pyrad/fastq/1HL_5A.discarded.fastq\n",
      "Unassembled forward reads file.....: /analysis/20160609_pyrad/fastq/1HL_5A.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: /analysis/20160609_pyrad/fastq/1HL_5A.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: /analysis/20160609_pyrad/fastq/1HL_6A_R1.fq\n",
      "Reverse reads file.................: /analysis/20160609_pyrad/fastq/1HL_6A_R2.fq\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 33\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 10\n",
      "Minimum read size after trimming...: 33\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 20\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.285547\n",
      "  C: 0.219469\n",
      "  G: 0.224365\n",
      "  T: 0.270619\n",
      "  609029 uncalled bases\n",
      "Assemblying reads: 0%\r",
      "Assemblying reads: 6%\r",
      "Assemblying reads: 11%\r",
      "Assemblying reads: 15%\r",
      "Assemblying reads: 19%\r",
      "Assemblying reads: 24%\r",
      "Assemblying reads: 28%\r",
      "Assemblying reads: 33%\r",
      "Assemblying reads: 37%\r",
      "Assemblying reads: 41%\r",
      "Assemblying reads: 46%\r",
      "Assemblying reads: 50%\r",
      "Assemblying reads: 54%\r",
      "Assemblying reads: 59%\r",
      "Assemblying reads: 63%\r",
      "Assemblying reads: 67%\r",
      "Assemblying reads: 72%\r",
      "Assemblying reads: 76%\r",
      "Assemblying reads: 80%\r",
      "Assemblying reads: 85%\r",
      "Assemblying reads: 89%\r",
      "Assemblying reads: 93%\r",
      "Assemblying reads: 98%\r",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 1,778,884 / 2,661,086 (66.848%)\n",
      "Discarded reads ...................: 882,202 / 2,661,086 (33.152%)\n",
      "Not assembled reads ...............: 0 / 2,661,086 (0.000%)\n",
      "Assembled reads file...............: /analysis/20160609_pyrad/fastq/1HL_6A.assembled.fastq\n",
      "Discarded reads file...............: /analysis/20160609_pyrad/fastq/1HL_6A.discarded.fastq\n",
      "Unassembled forward reads file.....: /analysis/20160609_pyrad/fastq/1HL_6A.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: /analysis/20160609_pyrad/fastq/1HL_6A.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: /analysis/20160609_pyrad/fastq/1HL_7A_R1.fq\n",
      "Reverse reads file.................: /analysis/20160609_pyrad/fastq/1HL_7A_R2.fq\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 33\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 10\n",
      "Minimum read size after trimming...: 33\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 20\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.279246\n",
      "  C: 0.221508\n",
      "  G: 0.227431\n",
      "  T: 0.271815\n",
      "  851643 uncalled bases\n",
      "Assemblying reads: 0%\r",
      "Assemblying reads: 4%\r",
      "Assemblying reads: 8%\r",
      "Assemblying reads: 11%\r",
      "Assemblying reads: 14%\r",
      "Assemblying reads: 17%\r",
      "Assemblying reads: 20%\r",
      "Assemblying reads: 23%\r",
      "Assemblying reads: 26%\r",
      "Assemblying reads: 30%\r",
      "Assemblying reads: 33%\r",
      "Assemblying reads: 36%\r",
      "Assemblying reads: 39%\r",
      "Assemblying reads: 42%\r",
      "Assemblying reads: 45%\r",
      "Assemblying reads: 48%\r",
      "Assemblying reads: 52%\r",
      "Assemblying reads: 55%\r",
      "Assemblying reads: 58%\r",
      "Assemblying reads: 61%\r",
      "Assemblying reads: 64%\r",
      "Assemblying reads: 67%\r",
      "Assemblying reads: 70%\r",
      "Assemblying reads: 74%\r",
      "Assemblying reads: 77%\r",
      "Assemblying reads: 80%\r",
      "Assemblying reads: 83%\r",
      "Assemblying reads: 86%\r",
      "Assemblying reads: 89%\r",
      "Assemblying reads: 92%\r",
      "Assemblying reads: 95%\r",
      "Assemblying reads: 99%\r",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 2,555,275 / 3,689,392 (69.260%)\n",
      "Discarded reads ...................: 1,134,117 / 3,689,392 (30.740%)\n",
      "Not assembled reads ...............: 0 / 3,689,392 (0.000%)\n",
      "Assembled reads file...............: /analysis/20160609_pyrad/fastq/1HL_7A.assembled.fastq\n",
      "Discarded reads file...............: /analysis/20160609_pyrad/fastq/1HL_7A.discarded.fastq\n",
      "Unassembled forward reads file.....: /analysis/20160609_pyrad/fastq/1HL_7A.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: /analysis/20160609_pyrad/fastq/1HL_7A.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: /analysis/20160609_pyrad/fastq/1HL_8A_R1.fq\n",
      "Reverse reads file.................: /analysis/20160609_pyrad/fastq/1HL_8A_R2.fq\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 33\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 10\n",
      "Minimum read size after trimming...: 33\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 20\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.283546\n",
      "  C: 0.220649\n",
      "  G: 0.226514\n",
      "  T: 0.269290\n",
      "  788910 uncalled bases\n",
      "Assemblying reads: 0%\r",
      "Assemblying reads: 5%\r",
      "Assemblying reads: 8%\r",
      "Assemblying reads: 12%\r",
      "Assemblying reads: 15%\r",
      "Assemblying reads: 19%\r",
      "Assemblying reads: 22%\r",
      "Assemblying reads: 25%\r",
      "Assemblying reads: 29%\r",
      "Assemblying reads: 32%\r",
      "Assemblying reads: 36%\r",
      "Assemblying reads: 39%\r",
      "Assemblying reads: 42%\r",
      "Assemblying reads: 46%\r",
      "Assemblying reads: 49%\r",
      "Assemblying reads: 53%\r",
      "Assemblying reads: 56%\r",
      "Assemblying reads: 59%\r",
      "Assemblying reads: 63%\r",
      "Assemblying reads: 66%\r",
      "Assemblying reads: 70%\r",
      "Assemblying reads: 73%\r",
      "Assemblying reads: 76%\r",
      "Assemblying reads: 80%\r",
      "Assemblying reads: 83%\r",
      "Assemblying reads: 87%\r",
      "Assemblying reads: 90%\r",
      "Assemblying reads: 93%\r",
      "Assemblying reads: 97%\r",
      "Assemblying reads: 100%\r",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 2,330,048 / 3,403,545 (68.459%)\n",
      "Discarded reads ...................: 1,073,497 / 3,403,545 (31.541%)\n",
      "Not assembled reads ...............: 0 / 3,403,545 (0.000%)\n",
      "Assembled reads file...............: /analysis/20160609_pyrad/fastq/1HL_8A.assembled.fastq\n",
      "Discarded reads file...............: /analysis/20160609_pyrad/fastq/1HL_8A.discarded.fastq\n",
      "Unassembled forward reads file.....: /analysis/20160609_pyrad/fastq/1HL_8A.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: /analysis/20160609_pyrad/fastq/1HL_8A.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: /analysis/20160609_pyrad/fastq/1HL_9A_R1.fq\n",
      "Reverse reads file.................: /analysis/20160609_pyrad/fastq/1HL_9A_R2.fq\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 33\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 10\n",
      "Minimum read size after trimming...: 33\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 20\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.281821\n",
      "  C: 0.223687\n",
      "  G: 0.226621\n",
      "  T: 0.267870\n",
      "  871271 uncalled bases\n",
      "Assemblying reads: 0%\r",
      "Assemblying reads: 4%\r",
      "Assemblying reads: 7%\r",
      "Assemblying reads: 11%\r",
      "Assemblying reads: 14%\r",
      "Assemblying reads: 17%\r",
      "Assemblying reads: 20%\r",
      "Assemblying reads: 23%\r",
      "Assemblying reads: 26%\r",
      "Assemblying reads: 29%\r",
      "Assemblying reads: 32%\r",
      "Assemblying reads: 35%\r",
      "Assemblying reads: 38%\r",
      "Assemblying reads: 41%\r",
      "Assemblying reads: 44%\r",
      "Assemblying reads: 48%\r",
      "Assemblying reads: 51%\r",
      "Assemblying reads: 54%\r",
      "Assemblying reads: 57%\r",
      "Assemblying reads: 60%\r",
      "Assemblying reads: 63%\r",
      "Assemblying reads: 66%\r",
      "Assemblying reads: 69%\r",
      "Assemblying reads: 72%\r",
      "Assemblying reads: 75%\r",
      "Assemblying reads: 78%\r",
      "Assemblying reads: 81%\r",
      "Assemblying reads: 85%\r",
      "Assemblying reads: 88%\r",
      "Assemblying reads: 91%\r",
      "Assemblying reads: 94%\r",
      "Assemblying reads: 97%\r",
      "Assemblying reads: 100%\r",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 2,620,475 / 3,753,204 (69.820%)\n",
      "Discarded reads ...................: 1,132,729 / 3,753,204 (30.180%)\n",
      "Not assembled reads ...............: 0 / 3,753,204 (0.000%)\n",
      "Assembled reads file...............: /analysis/20160609_pyrad/fastq/1HL_9A.assembled.fastq\n",
      "Discarded reads file...............: /analysis/20160609_pyrad/fastq/1HL_9A.discarded.fastq\n",
      "Unassembled forward reads file.....: /analysis/20160609_pyrad/fastq/1HL_9A.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: /analysis/20160609_pyrad/fastq/1HL_9A.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: /analysis/20160609_pyrad/fastq/1NF_10A_R1.fq\n",
      "Reverse reads file.................: /analysis/20160609_pyrad/fastq/1NF_10A_R2.fq\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 33\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 10\n",
      "Minimum read size after trimming...: 33\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 20\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.284157\n",
      "  C: 0.220971\n",
      "  G: 0.225414\n",
      "  T: 0.269459\n",
      "  664722 uncalled bases\n",
      "Assemblying reads: 0%\r",
      "Assemblying reads: 6%\r",
      "Assemblying reads: 10%\r",
      "Assemblying reads: 14%\r",
      "Assemblying reads: 18%\r",
      "Assemblying reads: 22%\r",
      "Assemblying reads: 26%\r",
      "Assemblying reads: 30%\r",
      "Assemblying reads: 34%\r",
      "Assemblying reads: 38%\r",
      "Assemblying reads: 42%\r",
      "Assemblying reads: 47%\r",
      "Assemblying reads: 51%\r",
      "Assemblying reads: 55%\r",
      "Assemblying reads: 59%\r",
      "Assemblying reads: 63%\r",
      "Assemblying reads: 67%\r",
      "Assemblying reads: 71%\r",
      "Assemblying reads: 75%\r",
      "Assemblying reads: 79%\r",
      "Assemblying reads: 83%\r",
      "Assemblying reads: 87%\r",
      "Assemblying reads: 91%\r",
      "Assemblying reads: 95%\r",
      "Assemblying reads: 99%\r",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 1,955,600 / 2,848,504 (68.654%)\n",
      "Discarded reads ...................: 892,904 / 2,848,504 (31.346%)\n",
      "Not assembled reads ...............: 0 / 2,848,504 (0.000%)\n",
      "Assembled reads file...............: /analysis/20160609_pyrad/fastq/1NF_10A.assembled.fastq\n",
      "Discarded reads file...............: /analysis/20160609_pyrad/fastq/1NF_10A.discarded.fastq\n",
      "Unassembled forward reads file.....: /analysis/20160609_pyrad/fastq/1NF_10A.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: /analysis/20160609_pyrad/fastq/1NF_10A.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: /analysis/20160609_pyrad/fastq/1NF_11A_R1.fq\n",
      "Reverse reads file.................: /analysis/20160609_pyrad/fastq/1NF_11A_R2.fq\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 33\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 10\n",
      "Minimum read size after trimming...: 33\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 20\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.287054\n",
      "  C: 0.217908\n",
      "  G: 0.223626\n",
      "  T: 0.271412\n",
      "  677071 uncalled bases\n",
      "Assemblying reads: 0%\r",
      "Assemblying reads: 6%\r",
      "Assemblying reads: 10%\r",
      "Assemblying reads: 14%\r",
      "Assemblying reads: 18%\r",
      "Assemblying reads: 21%\r",
      "Assemblying reads: 25%\r",
      "Assemblying reads: 29%\r",
      "Assemblying reads: 33%\r",
      "Assemblying reads: 37%\r",
      "Assemblying reads: 41%\r",
      "Assemblying reads: 45%\r",
      "Assemblying reads: 49%\r",
      "Assemblying reads: 53%\r",
      "Assemblying reads: 57%\r",
      "Assemblying reads: 61%\r",
      "Assemblying reads: 65%\r",
      "Assemblying reads: 69%\r",
      "Assemblying reads: 73%\r",
      "Assemblying reads: 77%\r",
      "Assemblying reads: 81%\r",
      "Assemblying reads: 84%\r",
      "Assemblying reads: 88%\r",
      "Assemblying reads: 92%\r",
      "Assemblying reads: 96%\r",
      "Assemblying reads: 100%\r",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 2,004,118 / 2,939,324 (68.183%)\n",
      "Discarded reads ...................: 935,206 / 2,939,324 (31.817%)\n",
      "Not assembled reads ...............: 0 / 2,939,324 (0.000%)\n",
      "Assembled reads file...............: /analysis/20160609_pyrad/fastq/1NF_11A.assembled.fastq\n",
      "Discarded reads file...............: /analysis/20160609_pyrad/fastq/1NF_11A.discarded.fastq\n",
      "Unassembled forward reads file.....: /analysis/20160609_pyrad/fastq/1NF_11A.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: /analysis/20160609_pyrad/fastq/1NF_11A.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: /analysis/20160609_pyrad/fastq/1NF_12A_R1.fq\n",
      "Reverse reads file.................: /analysis/20160609_pyrad/fastq/1NF_12A_R2.fq\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 33\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 10\n",
      "Minimum read size after trimming...: 33\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 20\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.285198\n",
      "  C: 0.218229\n",
      "  G: 0.228122\n",
      "  T: 0.268451\n",
      "  673211 uncalled bases\n",
      "Assemblying reads: 0%\r",
      "Assemblying reads: 6%\r",
      "Assemblying reads: 10%\r",
      "Assemblying reads: 14%\r",
      "Assemblying reads: 18%\r",
      "Assemblying reads: 22%\r",
      "Assemblying reads: 25%\r",
      "Assemblying reads: 29%\r",
      "Assemblying reads: 33%\r",
      "Assemblying reads: 37%\r",
      "Assemblying reads: 41%\r",
      "Assemblying reads: 45%\r",
      "Assemblying reads: 49%\r",
      "Assemblying reads: 53%\r",
      "Assemblying reads: 57%\r",
      "Assemblying reads: 61%\r",
      "Assemblying reads: 65%\r",
      "Assemblying reads: 69%\r",
      "Assemblying reads: 73%\r",
      "Assemblying reads: 77%\r",
      "Assemblying reads: 81%\r",
      "Assemblying reads: 85%\r",
      "Assemblying reads: 89%\r",
      "Assemblying reads: 93%\r",
      "Assemblying reads: 97%\r",
      "Assemblying reads: 100%\r",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 2,051,515 / 2,930,997 (69.994%)\n",
      "Discarded reads ...................: 879,482 / 2,930,997 (30.006%)\n",
      "Not assembled reads ...............: 0 / 2,930,997 (0.000%)\n",
      "Assembled reads file...............: /analysis/20160609_pyrad/fastq/1NF_12A.assembled.fastq\n",
      "Discarded reads file...............: /analysis/20160609_pyrad/fastq/1NF_12A.discarded.fastq\n",
      "Unassembled forward reads file.....: /analysis/20160609_pyrad/fastq/1NF_12A.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: /analysis/20160609_pyrad/fastq/1NF_12A.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: /analysis/20160609_pyrad/fastq/1NF_13A_R1.fq\n",
      "Reverse reads file.................: /analysis/20160609_pyrad/fastq/1NF_13A_R2.fq\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 33\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 10\n",
      "Minimum read size after trimming...: 33\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 20\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.283198\n",
      "  C: 0.221826\n",
      "  G: 0.226473\n",
      "  T: 0.268503\n",
      "  1200405 uncalled bases\n",
      "Assemblying reads: 0%\r",
      "Assemblying reads: 3%\r",
      "Assemblying reads: 5%\r",
      "Assemblying reads: 7%\r",
      "Assemblying reads: 10%\r",
      "Assemblying reads: 12%\r",
      "Assemblying reads: 14%\r",
      "Assemblying reads: 16%\r",
      "Assemblying reads: 19%\r",
      "Assemblying reads: 21%\r",
      "Assemblying reads: 23%\r",
      "Assemblying reads: 25%\r",
      "Assemblying reads: 28%\r",
      "Assemblying reads: 30%\r",
      "Assemblying reads: 32%\r",
      "Assemblying reads: 34%\r",
      "Assemblying reads: 37%\r",
      "Assemblying reads: 39%\r",
      "Assemblying reads: 41%\r",
      "Assemblying reads: 43%\r",
      "Assemblying reads: 45%\r",
      "Assemblying reads: 48%\r",
      "Assemblying reads: 50%\r",
      "Assemblying reads: 52%\r",
      "Assemblying reads: 54%\r",
      "Assemblying reads: 57%\r",
      "Assemblying reads: 59%\r",
      "Assemblying reads: 61%\r",
      "Assemblying reads: 63%\r",
      "Assemblying reads: 66%\r",
      "Assemblying reads: 68%\r",
      "Assemblying reads: 70%\r",
      "Assemblying reads: 72%\r",
      "Assemblying reads: 74%\r",
      "Assemblying reads: 77%\r",
      "Assemblying reads: 79%\r",
      "Assemblying reads: 81%\r",
      "Assemblying reads: 83%\r",
      "Assemblying reads: 86%\r",
      "Assemblying reads: 88%\r",
      "Assemblying reads: 90%\r",
      "Assemblying reads: 92%\r",
      "Assemblying reads: 95%\r",
      "Assemblying reads: 97%\r",
      "Assemblying reads: 99%\r",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 3,671,926 / 5,183,236 (70.842%)\n",
      "Discarded reads ...................: 1,511,310 / 5,183,236 (29.158%)\n",
      "Not assembled reads ...............: 0 / 5,183,236 (0.000%)\n",
      "Assembled reads file...............: /analysis/20160609_pyrad/fastq/1NF_13A.assembled.fastq\n",
      "Discarded reads file...............: /analysis/20160609_pyrad/fastq/1NF_13A.discarded.fastq\n",
      "Unassembled forward reads file.....: /analysis/20160609_pyrad/fastq/1NF_13A.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: /analysis/20160609_pyrad/fastq/1NF_13A.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: /analysis/20160609_pyrad/fastq/1NF_14A_R1.fq\n",
      "Reverse reads file.................: /analysis/20160609_pyrad/fastq/1NF_14A_R2.fq\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 33\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 10\n",
      "Minimum read size after trimming...: 33\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 20\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.288088\n",
      "  C: 0.217760\n",
      "  G: 0.223964\n",
      "  T: 0.270188\n",
      "  801159 uncalled bases\n",
      "Assemblying reads: 0%\r",
      "Assemblying reads: 5%\r",
      "Assemblying reads: 8%\r",
      "Assemblying reads: 11%\r",
      "Assemblying reads: 15%\r",
      "Assemblying reads: 18%\r",
      "Assemblying reads: 21%\r",
      "Assemblying reads: 25%\r",
      "Assemblying reads: 28%\r",
      "Assemblying reads: 31%\r",
      "Assemblying reads: 35%\r",
      "Assemblying reads: 38%\r",
      "Assemblying reads: 41%\r",
      "Assemblying reads: 45%\r",
      "Assemblying reads: 48%\r",
      "Assemblying reads: 52%\r",
      "Assemblying reads: 55%\r",
      "Assemblying reads: 58%\r",
      "Assemblying reads: 62%\r",
      "Assemblying reads: 65%\r",
      "Assemblying reads: 68%\r",
      "Assemblying reads: 72%\r",
      "Assemblying reads: 75%\r",
      "Assemblying reads: 78%\r",
      "Assemblying reads: 82%\r",
      "Assemblying reads: 85%\r",
      "Assemblying reads: 88%\r",
      "Assemblying reads: 92%\r",
      "Assemblying reads: 95%\r",
      "Assemblying reads: 98%\r",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 2,333,572 / 3,467,126 (67.306%)\n",
      "Discarded reads ...................: 1,133,554 / 3,467,126 (32.694%)\n",
      "Not assembled reads ...............: 0 / 3,467,126 (0.000%)\n",
      "Assembled reads file...............: /analysis/20160609_pyrad/fastq/1NF_14A.assembled.fastq\n",
      "Discarded reads file...............: /analysis/20160609_pyrad/fastq/1NF_14A.discarded.fastq\n",
      "Unassembled forward reads file.....: /analysis/20160609_pyrad/fastq/1NF_14A.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: /analysis/20160609_pyrad/fastq/1NF_14A.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: /analysis/20160609_pyrad/fastq/1NF_15A_R1.fq\n",
      "Reverse reads file.................: /analysis/20160609_pyrad/fastq/1NF_15A_R2.fq\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 33\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 10\n",
      "Minimum read size after trimming...: 33\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 20\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.285599\n",
      "  C: 0.217128\n",
      "  G: 0.227087\n",
      "  T: 0.270186\n",
      "  777701 uncalled bases\n",
      "Assemblying reads: 0%\r",
      "Assemblying reads: 5%\r",
      "Assemblying reads: 8%\r",
      "Assemblying reads: 12%\r",
      "Assemblying reads: 15%\r",
      "Assemblying reads: 19%\r",
      "Assemblying reads: 22%\r",
      "Assemblying reads: 26%\r",
      "Assemblying reads: 29%\r",
      "Assemblying reads: 33%\r",
      "Assemblying reads: 36%\r",
      "Assemblying reads: 40%\r",
      "Assemblying reads: 43%\r",
      "Assemblying reads: 46%\r",
      "Assemblying reads: 50%\r",
      "Assemblying reads: 53%\r",
      "Assemblying reads: 57%\r",
      "Assemblying reads: 60%\r",
      "Assemblying reads: 64%\r",
      "Assemblying reads: 67%\r",
      "Assemblying reads: 71%\r",
      "Assemblying reads: 74%\r",
      "Assemblying reads: 78%\r",
      "Assemblying reads: 81%\r",
      "Assemblying reads: 85%\r",
      "Assemblying reads: 88%\r",
      "Assemblying reads: 91%\r",
      "Assemblying reads: 95%\r",
      "Assemblying reads: 98%\r",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 2,296,320 / 3,344,566 (68.658%)\n",
      "Discarded reads ...................: 1,048,246 / 3,344,566 (31.342%)\n",
      "Not assembled reads ...............: 0 / 3,344,566 (0.000%)\n",
      "Assembled reads file...............: /analysis/20160609_pyrad/fastq/1NF_15A.assembled.fastq\n",
      "Discarded reads file...............: /analysis/20160609_pyrad/fastq/1NF_15A.discarded.fastq\n",
      "Unassembled forward reads file.....: /analysis/20160609_pyrad/fastq/1NF_15A.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: /analysis/20160609_pyrad/fastq/1NF_15A.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: /analysis/20160609_pyrad/fastq/1NF_16A_R1.fq\n",
      "Reverse reads file.................: /analysis/20160609_pyrad/fastq/1NF_16A_R2.fq\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 33\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 10\n",
      "Minimum read size after trimming...: 33\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 20\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.281531\n",
      "  C: 0.223127\n",
      "  G: 0.231810\n",
      "  T: 0.263532\n",
      "  881970 uncalled bases\n",
      "Assemblying reads: 0%\r",
      "Assemblying reads: 4%\r",
      "Assemblying reads: 7%\r",
      "Assemblying reads: 10%\r",
      "Assemblying reads: 13%\r",
      "Assemblying reads: 16%\r",
      "Assemblying reads: 19%\r",
      "Assemblying reads: 22%\r",
      "Assemblying reads: 26%\r",
      "Assemblying reads: 29%\r",
      "Assemblying reads: 32%\r",
      "Assemblying reads: 35%\r",
      "Assemblying reads: 38%\r",
      "Assemblying reads: 41%\r",
      "Assemblying reads: 44%\r",
      "Assemblying reads: 47%\r",
      "Assemblying reads: 50%\r",
      "Assemblying reads: 53%\r",
      "Assemblying reads: 56%\r",
      "Assemblying reads: 59%\r",
      "Assemblying reads: 62%\r",
      "Assemblying reads: 65%\r",
      "Assemblying reads: 68%\r",
      "Assemblying reads: 71%\r",
      "Assemblying reads: 74%\r",
      "Assemblying reads: 77%\r",
      "Assemblying reads: 80%\r",
      "Assemblying reads: 83%\r",
      "Assemblying reads: 86%\r",
      "Assemblying reads: 89%\r",
      "Assemblying reads: 92%\r",
      "Assemblying reads: 95%\r",
      "Assemblying reads: 98%\r",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 2,701,311 / 3,813,301 (70.839%)\n",
      "Discarded reads ...................: 1,111,990 / 3,813,301 (29.161%)\n",
      "Not assembled reads ...............: 0 / 3,813,301 (0.000%)\n",
      "Assembled reads file...............: /analysis/20160609_pyrad/fastq/1NF_16A.assembled.fastq\n",
      "Discarded reads file...............: /analysis/20160609_pyrad/fastq/1NF_16A.discarded.fastq\n",
      "Unassembled forward reads file.....: /analysis/20160609_pyrad/fastq/1NF_16A.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: /analysis/20160609_pyrad/fastq/1NF_16A.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: /analysis/20160609_pyrad/fastq/1NF_17A_R1.fq\n",
      "Reverse reads file.................: /analysis/20160609_pyrad/fastq/1NF_17A_R2.fq\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 33\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 10\n",
      "Minimum read size after trimming...: 33\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 20\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.284847\n",
      "  C: 0.215946\n",
      "  G: 0.223492\n",
      "  T: 0.275716\n",
      "  964953 uncalled bases\n",
      "Assemblying reads: 0%\r",
      "Assemblying reads: 4%\r",
      "Assemblying reads: 7%\r",
      "Assemblying reads: 10%\r",
      "Assemblying reads: 12%\r",
      "Assemblying reads: 15%\r",
      "Assemblying reads: 18%\r",
      "Assemblying reads: 21%\r",
      "Assemblying reads: 24%\r",
      "Assemblying reads: 26%\r",
      "Assemblying reads: 29%\r",
      "Assemblying reads: 32%\r",
      "Assemblying reads: 35%\r",
      "Assemblying reads: 38%\r",
      "Assemblying reads: 40%\r",
      "Assemblying reads: 43%\r",
      "Assemblying reads: 46%\r",
      "Assemblying reads: 49%\r",
      "Assemblying reads: 52%\r",
      "Assemblying reads: 54%\r",
      "Assemblying reads: 57%\r",
      "Assemblying reads: 60%\r",
      "Assemblying reads: 63%\r",
      "Assemblying reads: 66%\r",
      "Assemblying reads: 69%\r",
      "Assemblying reads: 71%\r",
      "Assemblying reads: 74%\r",
      "Assemblying reads: 77%\r",
      "Assemblying reads: 80%\r",
      "Assemblying reads: 83%\r",
      "Assemblying reads: 85%\r",
      "Assemblying reads: 88%\r",
      "Assemblying reads: 91%\r",
      "Assemblying reads: 94%\r",
      "Assemblying reads: 97%\r",
      "Assemblying reads: 99%\r",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 2,846,779 / 4,120,982 (69.080%)\n",
      "Discarded reads ...................: 1,274,203 / 4,120,982 (30.920%)\n",
      "Not assembled reads ...............: 0 / 4,120,982 (0.000%)\n",
      "Assembled reads file...............: /analysis/20160609_pyrad/fastq/1NF_17A.assembled.fastq\n",
      "Discarded reads file...............: /analysis/20160609_pyrad/fastq/1NF_17A.discarded.fastq\n",
      "Unassembled forward reads file.....: /analysis/20160609_pyrad/fastq/1NF_17A.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: /analysis/20160609_pyrad/fastq/1NF_17A.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: /analysis/20160609_pyrad/fastq/1NF_18A_R1.fq\n",
      "Reverse reads file.................: /analysis/20160609_pyrad/fastq/1NF_18A_R2.fq\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 33\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 10\n",
      "Minimum read size after trimming...: 33\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 20\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.291065\n",
      "  C: 0.214441\n",
      "  G: 0.221778\n",
      "  T: 0.272716\n",
      "  504128 uncalled bases\n",
      "Assemblying reads: 0%\r",
      "Assemblying reads: 8%\r",
      "Assemblying reads: 13%\r",
      "Assemblying reads: 18%\r",
      "Assemblying reads: 24%\r",
      "Assemblying reads: 29%\r",
      "Assemblying reads: 34%\r",
      "Assemblying reads: 39%\r",
      "Assemblying reads: 45%\r",
      "Assemblying reads: 50%\r",
      "Assemblying reads: 55%\r",
      "Assemblying reads: 61%\r",
      "Assemblying reads: 66%\r",
      "Assemblying reads: 71%\r",
      "Assemblying reads: 76%\r",
      "Assemblying reads: 82%\r",
      "Assemblying reads: 87%\r",
      "Assemblying reads: 92%\r",
      "Assemblying reads: 97%\r",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 1,469,499 / 2,195,389 (66.936%)\n",
      "Discarded reads ...................: 725,890 / 2,195,389 (33.064%)\n",
      "Not assembled reads ...............: 0 / 2,195,389 (0.000%)\n",
      "Assembled reads file...............: /analysis/20160609_pyrad/fastq/1NF_18A.assembled.fastq\n",
      "Discarded reads file...............: /analysis/20160609_pyrad/fastq/1NF_18A.discarded.fastq\n",
      "Unassembled forward reads file.....: /analysis/20160609_pyrad/fastq/1NF_18A.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: /analysis/20160609_pyrad/fastq/1NF_18A.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: /analysis/20160609_pyrad/fastq/1NF_19A_R1.fq\n",
      "Reverse reads file.................: /analysis/20160609_pyrad/fastq/1NF_19A_R2.fq\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 33\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 10\n",
      "Minimum read size after trimming...: 33\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 20\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.284788\n",
      "  C: 0.218296\n",
      "  G: 0.226489\n",
      "  T: 0.270428\n",
      "  617454 uncalled bases\n",
      "Assemblying reads: 0%\r",
      "Assemblying reads: 6%\r",
      "Assemblying reads: 11%\r",
      "Assemblying reads: 15%\r",
      "Assemblying reads: 20%\r",
      "Assemblying reads: 24%\r",
      "Assemblying reads: 28%\r",
      "Assemblying reads: 33%\r",
      "Assemblying reads: 37%\r",
      "Assemblying reads: 41%\r",
      "Assemblying reads: 46%\r",
      "Assemblying reads: 50%\r",
      "Assemblying reads: 55%\r",
      "Assemblying reads: 59%\r",
      "Assemblying reads: 63%\r",
      "Assemblying reads: 68%\r",
      "Assemblying reads: 72%\r",
      "Assemblying reads: 76%\r",
      "Assemblying reads: 81%\r",
      "Assemblying reads: 85%\r",
      "Assemblying reads: 90%\r",
      "Assemblying reads: 94%\r",
      "Assemblying reads: 98%\r",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 1,775,039 / 2,644,544 (67.121%)\n",
      "Discarded reads ...................: 869,505 / 2,644,544 (32.879%)\n",
      "Not assembled reads ...............: 0 / 2,644,544 (0.000%)\n",
      "Assembled reads file...............: /analysis/20160609_pyrad/fastq/1NF_19A.assembled.fastq\n",
      "Discarded reads file...............: /analysis/20160609_pyrad/fastq/1NF_19A.discarded.fastq\n",
      "Unassembled forward reads file.....: /analysis/20160609_pyrad/fastq/1NF_19A.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: /analysis/20160609_pyrad/fastq/1NF_19A.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: /analysis/20160609_pyrad/fastq/1NF_1A_R1.fq\n",
      "Reverse reads file.................: /analysis/20160609_pyrad/fastq/1NF_1A_R2.fq\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 33\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 10\n",
      "Minimum read size after trimming...: 33\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 20\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.287105\n",
      "  C: 0.217074\n",
      "  G: 0.226171\n",
      "  T: 0.269650\n",
      "  404635 uncalled bases\n",
      "Assemblying reads: 0%\r",
      "Assemblying reads: 10%\r",
      "Assemblying reads: 17%\r",
      "Assemblying reads: 23%\r",
      "Assemblying reads: 30%\r",
      "Assemblying reads: 36%\r",
      "Assemblying reads: 43%\r",
      "Assemblying reads: 50%\r",
      "Assemblying reads: 56%\r",
      "Assemblying reads: 63%\r",
      "Assemblying reads: 70%\r",
      "Assemblying reads: 76%\r",
      "Assemblying reads: 83%\r",
      "Assemblying reads: 90%\r",
      "Assemblying reads: 96%\r",
      "Assemblying reads: 100%\r",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 1,157,346 / 1,743,417 (66.384%)\n",
      "Discarded reads ...................: 586,071 / 1,743,417 (33.616%)\n",
      "Not assembled reads ...............: 0 / 1,743,417 (0.000%)\n",
      "Assembled reads file...............: /analysis/20160609_pyrad/fastq/1NF_1A.assembled.fastq\n",
      "Discarded reads file...............: /analysis/20160609_pyrad/fastq/1NF_1A.discarded.fastq\n",
      "Unassembled forward reads file.....: /analysis/20160609_pyrad/fastq/1NF_1A.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: /analysis/20160609_pyrad/fastq/1NF_1A.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: /analysis/20160609_pyrad/fastq/1NF_20A_R1.fq\n",
      "Reverse reads file.................: /analysis/20160609_pyrad/fastq/1NF_20A_R2.fq\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 33\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 10\n",
      "Minimum read size after trimming...: 33\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 20\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.288912\n",
      "  C: 0.218564\n",
      "  G: 0.223118\n",
      "  T: 0.269406\n",
      "  752785 uncalled bases\n",
      "Assemblying reads: 0%\r",
      "Assemblying reads: 5%\r",
      "Assemblying reads: 9%\r",
      "Assemblying reads: 12%\r",
      "Assemblying reads: 16%\r",
      "Assemblying reads: 19%\r",
      "Assemblying reads: 23%\r",
      "Assemblying reads: 26%\r",
      "Assemblying reads: 30%\r",
      "Assemblying reads: 33%\r",
      "Assemblying reads: 37%\r",
      "Assemblying reads: 40%\r",
      "Assemblying reads: 44%\r",
      "Assemblying reads: 48%\r",
      "Assemblying reads: 51%\r",
      "Assemblying reads: 55%\r",
      "Assemblying reads: 58%\r",
      "Assemblying reads: 62%\r",
      "Assemblying reads: 65%\r",
      "Assemblying reads: 69%\r",
      "Assemblying reads: 72%\r",
      "Assemblying reads: 76%\r",
      "Assemblying reads: 79%\r",
      "Assemblying reads: 83%\r",
      "Assemblying reads: 86%\r",
      "Assemblying reads: 90%\r",
      "Assemblying reads: 93%\r",
      "Assemblying reads: 97%\r",
      "Assemblying reads: 100%\r",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 2,261,913 / 3,273,230 (69.103%)\n",
      "Discarded reads ...................: 1,011,317 / 3,273,230 (30.897%)\n",
      "Not assembled reads ...............: 0 / 3,273,230 (0.000%)\n",
      "Assembled reads file...............: /analysis/20160609_pyrad/fastq/1NF_20A.assembled.fastq\n",
      "Discarded reads file...............: /analysis/20160609_pyrad/fastq/1NF_20A.discarded.fastq\n",
      "Unassembled forward reads file.....: /analysis/20160609_pyrad/fastq/1NF_20A.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: /analysis/20160609_pyrad/fastq/1NF_20A.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: /analysis/20160609_pyrad/fastq/1NF_21A_R1.fq\n",
      "Reverse reads file.................: /analysis/20160609_pyrad/fastq/1NF_21A_R2.fq\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 33\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 10\n",
      "Minimum read size after trimming...: 33\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 20\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.283222\n",
      "  C: 0.221884\n",
      "  G: 0.228245\n",
      "  T: 0.266648\n",
      "  931251 uncalled bases\n",
      "Assemblying reads: 0%\r",
      "Assemblying reads: 4%\r",
      "Assemblying reads: 7%\r",
      "Assemblying reads: 10%\r",
      "Assemblying reads: 13%\r",
      "Assemblying reads: 15%\r",
      "Assemblying reads: 18%\r",
      "Assemblying reads: 21%\r",
      "Assemblying reads: 24%\r",
      "Assemblying reads: 27%\r",
      "Assemblying reads: 30%\r",
      "Assemblying reads: 32%\r",
      "Assemblying reads: 35%\r",
      "Assemblying reads: 38%\r",
      "Assemblying reads: 41%\r",
      "Assemblying reads: 44%\r",
      "Assemblying reads: 47%\r",
      "Assemblying reads: 49%\r",
      "Assemblying reads: 52%\r",
      "Assemblying reads: 55%\r",
      "Assemblying reads: 58%\r",
      "Assemblying reads: 61%\r",
      "Assemblying reads: 64%\r",
      "Assemblying reads: 66%\r",
      "Assemblying reads: 69%\r",
      "Assemblying reads: 72%\r",
      "Assemblying reads: 75%\r",
      "Assemblying reads: 78%\r",
      "Assemblying reads: 81%\r",
      "Assemblying reads: 84%\r",
      "Assemblying reads: 86%\r",
      "Assemblying reads: 89%\r",
      "Assemblying reads: 92%\r",
      "Assemblying reads: 95%\r",
      "Assemblying reads: 98%\r",
      "Assemblying reads: 100%\r",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 2,903,774 / 4,075,131 (71.256%)\n",
      "Discarded reads ...................: 1,171,357 / 4,075,131 (28.744%)\n",
      "Not assembled reads ...............: 0 / 4,075,131 (0.000%)\n",
      "Assembled reads file...............: /analysis/20160609_pyrad/fastq/1NF_21A.assembled.fastq\n",
      "Discarded reads file...............: /analysis/20160609_pyrad/fastq/1NF_21A.discarded.fastq\n",
      "Unassembled forward reads file.....: /analysis/20160609_pyrad/fastq/1NF_21A.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: /analysis/20160609_pyrad/fastq/1NF_21A.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: /analysis/20160609_pyrad/fastq/1NF_22A_R1.fq\n",
      "Reverse reads file.................: /analysis/20160609_pyrad/fastq/1NF_22A_R2.fq\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 33\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 10\n",
      "Minimum read size after trimming...: 33\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 20\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.284947\n",
      "  C: 0.219154\n",
      "  G: 0.226847\n",
      "  T: 0.269051\n",
      "  879359 uncalled bases\n",
      "Assemblying reads: 0%\r",
      "Assemblying reads: 4%\r",
      "Assemblying reads: 7%\r",
      "Assemblying reads: 10%\r",
      "Assemblying reads: 13%\r",
      "Assemblying reads: 16%\r",
      "Assemblying reads: 19%\r",
      "Assemblying reads: 22%\r",
      "Assemblying reads: 25%\r",
      "Assemblying reads: 28%\r",
      "Assemblying reads: 31%\r",
      "Assemblying reads: 35%\r",
      "Assemblying reads: 38%\r",
      "Assemblying reads: 41%\r",
      "Assemblying reads: 44%\r",
      "Assemblying reads: 47%\r",
      "Assemblying reads: 50%\r",
      "Assemblying reads: 53%\r",
      "Assemblying reads: 56%\r",
      "Assemblying reads: 59%\r",
      "Assemblying reads: 62%\r",
      "Assemblying reads: 65%\r",
      "Assemblying reads: 68%\r",
      "Assemblying reads: 71%\r",
      "Assemblying reads: 74%\r",
      "Assemblying reads: 77%\r",
      "Assemblying reads: 80%\r",
      "Assemblying reads: 83%\r",
      "Assemblying reads: 86%\r",
      "Assemblying reads: 89%\r",
      "Assemblying reads: 92%\r",
      "Assemblying reads: 95%\r",
      "Assemblying reads: 98%\r",
      "Assemblying reads: 100%\r",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 2,674,529 / 3,827,463 (69.877%)\n",
      "Discarded reads ...................: 1,152,934 / 3,827,463 (30.123%)\n",
      "Not assembled reads ...............: 0 / 3,827,463 (0.000%)\n",
      "Assembled reads file...............: /analysis/20160609_pyrad/fastq/1NF_22A.assembled.fastq\n",
      "Discarded reads file...............: /analysis/20160609_pyrad/fastq/1NF_22A.discarded.fastq\n",
      "Unassembled forward reads file.....: /analysis/20160609_pyrad/fastq/1NF_22A.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: /analysis/20160609_pyrad/fastq/1NF_22A.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: /analysis/20160609_pyrad/fastq/1NF_23A_R1.fq\n",
      "Reverse reads file.................: /analysis/20160609_pyrad/fastq/1NF_23A_R2.fq\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 33\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 10\n",
      "Minimum read size after trimming...: 33\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 20\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.284968\n",
      "  C: 0.220204\n",
      "  G: 0.227276\n",
      "  T: 0.267552\n",
      "  1015053 uncalled bases\n",
      "Assemblying reads: 0%\r",
      "Assemblying reads: 4%\r",
      "Assemblying reads: 6%\r",
      "Assemblying reads: 9%\r",
      "Assemblying reads: 12%\r",
      "Assemblying reads: 14%\r",
      "Assemblying reads: 17%\r",
      "Assemblying reads: 20%\r",
      "Assemblying reads: 22%\r",
      "Assemblying reads: 25%\r",
      "Assemblying reads: 28%\r",
      "Assemblying reads: 30%\r",
      "Assemblying reads: 33%\r",
      "Assemblying reads: 35%\r",
      "Assemblying reads: 38%\r",
      "Assemblying reads: 41%\r",
      "Assemblying reads: 43%\r",
      "Assemblying reads: 46%\r",
      "Assemblying reads: 49%\r",
      "Assemblying reads: 51%\r",
      "Assemblying reads: 54%\r",
      "Assemblying reads: 57%\r",
      "Assemblying reads: 59%\r",
      "Assemblying reads: 62%\r",
      "Assemblying reads: 65%\r",
      "Assemblying reads: 67%\r",
      "Assemblying reads: 70%\r",
      "Assemblying reads: 73%\r",
      "Assemblying reads: 75%\r",
      "Assemblying reads: 78%\r",
      "Assemblying reads: 80%\r",
      "Assemblying reads: 83%\r",
      "Assemblying reads: 86%\r",
      "Assemblying reads: 88%\r",
      "Assemblying reads: 91%\r",
      "Assemblying reads: 94%\r",
      "Assemblying reads: 96%\r",
      "Assemblying reads: 99%\r",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 3,106,447 / 4,372,177 (71.050%)\n",
      "Discarded reads ...................: 1,265,730 / 4,372,177 (28.950%)\n",
      "Not assembled reads ...............: 0 / 4,372,177 (0.000%)\n",
      "Assembled reads file...............: /analysis/20160609_pyrad/fastq/1NF_23A.assembled.fastq\n",
      "Discarded reads file...............: /analysis/20160609_pyrad/fastq/1NF_23A.discarded.fastq\n",
      "Unassembled forward reads file.....: /analysis/20160609_pyrad/fastq/1NF_23A.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: /analysis/20160609_pyrad/fastq/1NF_23A.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: /analysis/20160609_pyrad/fastq/1NF_24A_R1.fq\n",
      "Reverse reads file.................: /analysis/20160609_pyrad/fastq/1NF_24A_R2.fq\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 33\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 10\n",
      "Minimum read size after trimming...: 33\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 20\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.288096\n",
      "  C: 0.220296\n",
      "  G: 0.221235\n",
      "  T: 0.270373\n",
      "  889909 uncalled bases\n",
      "Assemblying reads: 0%\r",
      "Assemblying reads: 4%\r",
      "Assemblying reads: 7%\r",
      "Assemblying reads: 10%\r",
      "Assemblying reads: 13%\r",
      "Assemblying reads: 16%\r",
      "Assemblying reads: 19%\r",
      "Assemblying reads: 22%\r",
      "Assemblying reads: 25%\r",
      "Assemblying reads: 28%\r",
      "Assemblying reads: 31%\r",
      "Assemblying reads: 34%\r",
      "Assemblying reads: 37%\r",
      "Assemblying reads: 40%\r",
      "Assemblying reads: 43%\r",
      "Assemblying reads: 46%\r",
      "Assemblying reads: 49%\r",
      "Assemblying reads: 52%\r",
      "Assemblying reads: 55%\r",
      "Assemblying reads: 58%\r",
      "Assemblying reads: 61%\r",
      "Assemblying reads: 64%\r",
      "Assemblying reads: 67%\r",
      "Assemblying reads: 70%\r",
      "Assemblying reads: 73%\r",
      "Assemblying reads: 76%\r",
      "Assemblying reads: 79%\r",
      "Assemblying reads: 82%\r",
      "Assemblying reads: 85%\r",
      "Assemblying reads: 88%\r",
      "Assemblying reads: 91%\r",
      "Assemblying reads: 94%\r",
      "Assemblying reads: 97%\r",
      "Assemblying reads: 99%\r",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 2,650,885 / 3,886,155 (68.214%)\n",
      "Discarded reads ...................: 1,235,270 / 3,886,155 (31.786%)\n",
      "Not assembled reads ...............: 0 / 3,886,155 (0.000%)\n",
      "Assembled reads file...............: /analysis/20160609_pyrad/fastq/1NF_24A.assembled.fastq\n",
      "Discarded reads file...............: /analysis/20160609_pyrad/fastq/1NF_24A.discarded.fastq\n",
      "Unassembled forward reads file.....: /analysis/20160609_pyrad/fastq/1NF_24A.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: /analysis/20160609_pyrad/fastq/1NF_24A.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: /analysis/20160609_pyrad/fastq/1NF_25A_R1.fq\n",
      "Reverse reads file.................: /analysis/20160609_pyrad/fastq/1NF_25A_R2.fq\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 33\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 10\n",
      "Minimum read size after trimming...: 33\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 20\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.287368\n",
      "  C: 0.218703\n",
      "  G: 0.220460\n",
      "  T: 0.273469\n",
      "  638537 uncalled bases\n",
      "Assemblying reads: 0%\r",
      "Assemblying reads: 6%\r",
      "Assemblying reads: 10%\r",
      "Assemblying reads: 14%\r",
      "Assemblying reads: 18%\r",
      "Assemblying reads: 22%\r",
      "Assemblying reads: 27%\r",
      "Assemblying reads: 31%\r",
      "Assemblying reads: 35%\r",
      "Assemblying reads: 39%\r",
      "Assemblying reads: 43%\r",
      "Assemblying reads: 47%\r",
      "Assemblying reads: 51%\r",
      "Assemblying reads: 55%\r",
      "Assemblying reads: 60%\r",
      "Assemblying reads: 64%\r",
      "Assemblying reads: 68%\r",
      "Assemblying reads: 72%\r",
      "Assemblying reads: 76%\r",
      "Assemblying reads: 80%\r",
      "Assemblying reads: 84%\r",
      "Assemblying reads: 88%\r",
      "Assemblying reads: 93%\r",
      "Assemblying reads: 97%\r",
      "Assemblying reads: 100%\r",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 1,879,642 / 2,807,991 (66.939%)\n",
      "Discarded reads ...................: 928,349 / 2,807,991 (33.061%)\n",
      "Not assembled reads ...............: 0 / 2,807,991 (0.000%)\n",
      "Assembled reads file...............: /analysis/20160609_pyrad/fastq/1NF_25A.assembled.fastq\n",
      "Discarded reads file...............: /analysis/20160609_pyrad/fastq/1NF_25A.discarded.fastq\n",
      "Unassembled forward reads file.....: /analysis/20160609_pyrad/fastq/1NF_25A.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: /analysis/20160609_pyrad/fastq/1NF_25A.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: /analysis/20160609_pyrad/fastq/1NF_26A_R1.fq\n",
      "Reverse reads file.................: /analysis/20160609_pyrad/fastq/1NF_26A_R2.fq\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 33\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 10\n",
      "Minimum read size after trimming...: 33\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 20\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.288072\n",
      "  C: 0.217058\n",
      "  G: 0.221301\n",
      "  T: 0.273569\n",
      "  887799 uncalled bases\n",
      "Assemblying reads: 0%\r",
      "Assemblying reads: 4%\r",
      "Assemblying reads: 7%\r",
      "Assemblying reads: 10%\r",
      "Assemblying reads: 13%\r",
      "Assemblying reads: 17%\r",
      "Assemblying reads: 20%\r",
      "Assemblying reads: 23%\r",
      "Assemblying reads: 26%\r",
      "Assemblying reads: 29%\r",
      "Assemblying reads: 32%\r",
      "Assemblying reads: 35%\r",
      "Assemblying reads: 38%\r",
      "Assemblying reads: 41%\r",
      "Assemblying reads: 44%\r",
      "Assemblying reads: 47%\r",
      "Assemblying reads: 50%\r",
      "Assemblying reads: 53%\r",
      "Assemblying reads: 56%\r",
      "Assemblying reads: 59%\r",
      "Assemblying reads: 62%\r",
      "Assemblying reads: 65%\r",
      "Assemblying reads: 68%\r",
      "Assemblying reads: 71%\r",
      "Assemblying reads: 74%\r",
      "Assemblying reads: 77%\r",
      "Assemblying reads: 81%\r",
      "Assemblying reads: 84%\r",
      "Assemblying reads: 87%\r",
      "Assemblying reads: 90%\r",
      "Assemblying reads: 93%\r",
      "Assemblying reads: 96%\r",
      "Assemblying reads: 99%\r",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 2,616,210 / 3,796,337 (68.914%)\n",
      "Discarded reads ...................: 1,180,127 / 3,796,337 (31.086%)\n",
      "Not assembled reads ...............: 0 / 3,796,337 (0.000%)\n",
      "Assembled reads file...............: /analysis/20160609_pyrad/fastq/1NF_26A.assembled.fastq\n",
      "Discarded reads file...............: /analysis/20160609_pyrad/fastq/1NF_26A.discarded.fastq\n",
      "Unassembled forward reads file.....: /analysis/20160609_pyrad/fastq/1NF_26A.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: /analysis/20160609_pyrad/fastq/1NF_26A.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: /analysis/20160609_pyrad/fastq/1NF_27A_R1.fq\n",
      "Reverse reads file.................: /analysis/20160609_pyrad/fastq/1NF_27A_R2.fq\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 33\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 10\n",
      "Minimum read size after trimming...: 33\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 20\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.285131\n",
      "  C: 0.219412\n",
      "  G: 0.228628\n",
      "  T: 0.266829\n",
      "  707858 uncalled bases\n",
      "Assemblying reads: 0%\r",
      "Assemblying reads: 5%\r",
      "Assemblying reads: 9%\r",
      "Assemblying reads: 13%\r",
      "Assemblying reads: 17%\r",
      "Assemblying reads: 20%\r",
      "Assemblying reads: 24%\r",
      "Assemblying reads: 28%\r",
      "Assemblying reads: 32%\r",
      "Assemblying reads: 35%\r",
      "Assemblying reads: 39%\r",
      "Assemblying reads: 43%\r",
      "Assemblying reads: 47%\r",
      "Assemblying reads: 50%\r",
      "Assemblying reads: 54%\r",
      "Assemblying reads: 58%\r",
      "Assemblying reads: 62%\r",
      "Assemblying reads: 65%\r",
      "Assemblying reads: 69%\r",
      "Assemblying reads: 73%\r",
      "Assemblying reads: 77%\r",
      "Assemblying reads: 80%\r",
      "Assemblying reads: 84%\r",
      "Assemblying reads: 88%\r",
      "Assemblying reads: 92%\r",
      "Assemblying reads: 95%\r",
      "Assemblying reads: 99%\r",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 2,158,729 / 3,089,535 (69.872%)\n",
      "Discarded reads ...................: 930,806 / 3,089,535 (30.128%)\n",
      "Not assembled reads ...............: 0 / 3,089,535 (0.000%)\n",
      "Assembled reads file...............: /analysis/20160609_pyrad/fastq/1NF_27A.assembled.fastq\n",
      "Discarded reads file...............: /analysis/20160609_pyrad/fastq/1NF_27A.discarded.fastq\n",
      "Unassembled forward reads file.....: /analysis/20160609_pyrad/fastq/1NF_27A.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: /analysis/20160609_pyrad/fastq/1NF_27A.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: /analysis/20160609_pyrad/fastq/1NF_28A_R1.fq\n",
      "Reverse reads file.................: /analysis/20160609_pyrad/fastq/1NF_28A_R2.fq\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 33\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 10\n",
      "Minimum read size after trimming...: 33\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 20\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: DONE\n",
      "  A: 0.282908\n",
      "  C: 0.220026\n",
      "  G: 0.230713\n",
      "  T: 0.266354\n",
      "  740492 uncalled bases\n",
      "Assemblying reads: 0%\r",
      "Assemblying reads: 5%\r",
      "Assemblying reads: 9%\r",
      "Assemblying reads: 12%\r",
      "Assemblying reads: 16%\r",
      "Assemblying reads: 20%\r",
      "Assemblying reads: 23%\r",
      "Assemblying reads: 27%\r",
      "Assemblying reads: 31%\r",
      "Assemblying reads: 34%\r",
      "Assemblying reads: 38%\r",
      "Assemblying reads: 41%\r",
      "Assemblying reads: 45%\r",
      "Assemblying reads: 49%\r",
      "Assemblying reads: 52%\r",
      "Assemblying reads: 56%\r",
      "Assemblying reads: 60%\r",
      "Assemblying reads: 63%\r",
      "Assemblying reads: 67%\r",
      "Assemblying reads: 70%\r",
      "Assemblying reads: 74%\r",
      "Assemblying reads: 78%\r",
      "Assemblying reads: 81%\r",
      "Assemblying reads: 85%\r",
      "Assemblying reads: 89%\r",
      "Assemblying reads: 92%\r",
      "Assemblying reads: 96%\r",
      "Assemblying reads: 99%\r",
      "Assemblying reads: 100%\n",
      "\n",
      "Assembled reads ...................: 2,221,247 / 3,194,615 (69.531%)\n",
      "Discarded reads ...................: 973,368 / 3,194,615 (30.469%)\n",
      "Not assembled reads ...............: 0 / 3,194,615 (0.000%)\n",
      "Assembled reads file...............: /analysis/20160609_pyrad/fastq/1NF_28A.assembled.fastq\n",
      "Discarded reads file...............: /analysis/20160609_pyrad/fastq/1NF_28A.discarded.fastq\n",
      "Unassembled forward reads file.....: /analysis/20160609_pyrad/fastq/1NF_28A.unassembled.forward.fastq\n",
      "Unassembled reverse reads file.....: /analysis/20160609_pyrad/fastq/1NF_28A.unassembled.reverse.fastq\n",
      " ____  _____    _    ____ \n",
      "|  _ \\| ____|  / \\  |  _ \\\n",
      "| |_) |  _|   / _ \\ | |_) |\n",
      "|  __/| |___ / ___ \\|  _ <\n",
      "|_|   |_____/_/   \\_\\_| \\_\\\n",
      "\n",
      "PEAR v0.9.6 [January 15, 2015]\n",
      "\n",
      "Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR\n",
      "Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593\n",
      "\n",
      "Forward reads file.................: /analysis/20160609_pyrad/fastq/1NF_29A_R1.fq\n",
      "Reverse reads file.................: /analysis/20160609_pyrad/fastq/1NF_29A_R2.fq\n",
      "PHRED..............................: 33\n",
      "Using empirical frequencies........: YES\n",
      "Statistical method.................: OES\n",
      "Maximum assembly length............: 999999\n",
      "Minimum assembly length............: 33\n",
      "p-value............................: 0.010000\n",
      "Quality score threshold (trimming).: 10\n",
      "Minimum read size after trimming...: 33\n",
      "Maximal ratio of uncalled bases....: 1.000000\n",
      "Minimum overlap....................: 10\n",
      "Scoring method.....................: Scaled score\n",
      "Threads............................: 20\n",
      "\n",
      "Allocating memory..................: 200,000,000 bytes\n",
      "Computing empirical frequencies....: "
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "cat pear.log"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Previous PEAR command did not finish - will re-run"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Wed Jun 29 18:10:08 UTC 2016\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "date"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "%%bash\n",
    "#Rename previous pear.log so new log info isn't appended to previous file\n",
    "mv pear.log pear.log.20160609"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "barcodes.txt  \u001b[0m\u001b[01;34mfastq\u001b[0m/  params.txt  pear.log.20160609  \u001b[01;34mstats\u001b[0m/\r\n"
     ]
    }
   ],
   "source": [
    "ls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "%%bash\n",
    "time for gfile in /analysis/20160609_pyrad/fastq/*_R1.fq;\n",
    "    do /usr/local/bioinformatics/pear-0.9.6-bin-64/pear-0.9.6-bin-64 -f $gfile \\\n",
    "            -r ${gfile/_R1.fq/_R2.fq} \\\n",
    "            -o ${gfile/_R1.fq/} \\\n",
    "            -n 33 \\\n",
    "            -t 33 \\\n",
    "            -q 10 \\\n",
    "            -j 20  >> pear.log 2>&1;\n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": true
   },
   "source": [
    "#### Roadrunner crashed again. Rebooted. Disabled screen saver (read this might be an issue somwhere on the web...)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Re-running PEAR"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/analysis/20160609_pyrad\n"
     ]
    }
   ],
   "source": [
    "cd /analysis/20160609_pyrad/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "barcodes.txt  \u001b[0m\u001b[01;34mfastq\u001b[0m/  params.txt  pear.log  pear.log.20160609  \u001b[01;34mstats\u001b[0m/\r\n"
     ]
    }
   ],
   "source": [
    "ls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "rm pear.log"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "barcodes.txt  \u001b[0m\u001b[01;34mfastq\u001b[0m/  params.txt  pear.log.20160609  \u001b[01;34mstats\u001b[0m/\r\n"
     ]
    }
   ],
   "source": [
    "ls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "%%bash\n",
    "time for gfile in /analysis/20160609_pyrad/fastq/*_R1.fq;\n",
    "    do /usr/local/bioinformatics/pear-0.9.6-bin-64/pear-0.9.6-bin-64 -f $gfile \\\n",
    "            -r ${gfile/_R1.fq/_R2.fq} \\\n",
    "            -o ${gfile/_R1.fq/} \\\n",
    "            -n 33 \\\n",
    "            -t 33 \\\n",
    "            -q 10 \\\n",
    "            -j 20  >> pear.log 2>&1;\n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Computer kept crashing when running this remotely, in Docker container. Now running locally"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Thu Jun 30 15:23:02 PDT 2016\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "date"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Software:\n",
      "\n",
      "    System Software Overview:\n",
      "\n",
      "      System Version: OS X 10.11.5 (15F34)\n",
      "      Kernel Version: Darwin 15.5.0\n",
      "      Boot Volume: boot_drive\n",
      "      Boot Mode: Normal\n",
      "      Computer Name: roadrunner\n",
      "      User Name: sam (sam)\n",
      "      Secure Virtual Memory: Enabled\n",
      "      System Integrity Protection: Enabled\n",
      "      Time since boot: 35 minutes\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "system_profiler SPSoftwareDataType"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "      Model Name: Xserve\n",
      "      Model Identifier: Xserve3,1\n",
      "      Processor Name: Quad-Core Intel Xeon\n",
      "      Processor Speed: 2.26 GHz\n",
      "      Number of Processors: 2\n",
      "      Total Number of Cores: 8\n",
      "      L2 Cache (per Core): 256 KB\n",
      "      L3 Cache (per Processor): 8 MB\n",
      "      Memory: 48 GB\n",
      "      Processor Interconnect Speed: 5.86 GT/s\n",
      "      Boot ROM Version: XS31.0081.B04\n",
      "      SMC Version (system): 1.43f4\n",
      "      LOM Revision: 1.1.8\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "#Uses grep to exclude lines that display serial number and hardware UUID\n",
    "system_profiler SPHardwareDataType | grep -v [SH][ea]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/sam/analysis/20160609_pyrad\n"
     ]
    }
   ],
   "source": [
    "cd /Users/sam/analysis/20160609_pyrad/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "barcodes.txt       params.txt         pear.log.20160609\r\n",
      "\u001b[34mfastq\u001b[m\u001b[m/             pear.log           \u001b[34mstats\u001b[m\u001b[m/\r\n"
     ]
    }
   ],
   "source": [
    "ls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "%%bash\n",
    "mv pear.log pear.log.20160629"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "barcodes.txt       params.txt         pear.log.20160629\r\n",
      "\u001b[34mfastq\u001b[m\u001b[m/             pear.log.20160609  \u001b[34mstats\u001b[m\u001b[m/\r\n"
     ]
    }
   ],
   "source": [
    "ls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "%%bash\n",
    "time for gfile in /Users/sam/analysis/20160609_pyrad/fastq/*_R1.fq;\n",
    "    do pear -f $gfile \\\n",
    "            -r ${gfile/_R1.fq/_R2.fq} \\\n",
    "            -o ${gfile/_R1.fq/} \\\n",
    "            -n 33 \\\n",
    "            -t 33 \\\n",
    "            -q 10 \\\n",
    "            -j 20  >> pear.log 2>&1;\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
