{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Running in Docker container on Swoose\n",
    "\n",
    "Started Docker container with the following command:\n",
    "\n",
    "```docker run -p 8888:8888 -v /home/sam/data/pacbio_oly/:/home/data -v /mnt/owl/:/home/owl -it bioinformatics/bioinformatics:v0 /bin/bash```\n",
    "\n",
    "The command allows ```/home/sam/data/``` and ```mnt/owl``` (our server Owl) to be accessible to the Docker container.\n",
    "\n",
    "Once  access to Jupyter Notebook over port 8888 and makes my Jupyter Notebook GitHub repo and my data files the container was started, started Jupyter Notebook with the following command inside the Docker container:\n",
    "\n",
    "```jupyter notebook --allow-root```\n",
    "\n",
    "This is configured in the Docker container to launch a Jupyter Notebook without a browser on port 8888.\n",
    "The Docker container is running on an image created from this [Dockerfile (Git commit 4f0d2ff](https://github.com/RobertsLab/code/commit/4f0d2ffb3e1596bd3a267095a01d0b6fcabb2e73)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TODAY'S DATE\n",
      "Wed Oct 18 14:28:20 UTC 2017\n",
      "------------\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "bash: line 15: unexpected EOF while looking for matching `\"'\n",
      "bash: line 17: syntax error: unexpected end of file\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "echo \"TODAY'S DATE\"\n",
    "date\n",
    "echo \"------------\"\n",
    "echo \"\"\n",
    "echo \"HOSTNAME: $hostname\"\"\n",
    "echo \"\"\n",
    "echo \"------------\"\n",
    "echo \"Computer Specs:\"\n",
    "echo \"\"\n",
    "lscpu\n",
    "echo \"\"\n",
    "echo \"------------\"\n",
    "echo \"\"\n",
    "echo \"Memory Specs\"\n",
    "echo \"\"\n",
    "free -mh"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TODAY'S DATE\n",
      "Wed Oct 18 14:28:44 UTC 2017\n",
      "------------\n",
      "\n",
      "HOSTNAME: \n",
      "\n",
      "------------\n",
      "Computer Specs:\n",
      "\n",
      "Architecture:          x86_64\n",
      "CPU op-mode(s):        32-bit, 64-bit\n",
      "Byte Order:            Little Endian\n",
      "CPU(s):                24\n",
      "On-line CPU(s) list:   0-23\n",
      "Thread(s) per core:    2\n",
      "Core(s) per socket:    6\n",
      "Socket(s):             2\n",
      "NUMA node(s):          1\n",
      "Vendor ID:             GenuineIntel\n",
      "CPU family:            6\n",
      "Model:                 44\n",
      "Model name:            Intel(R) Xeon(R) CPU           X5670  @ 2.93GHz\n",
      "Stepping:              2\n",
      "CPU MHz:               2925.850\n",
      "BogoMIPS:              5851.96\n",
      "Virtualization:        VT-x\n",
      "L1d cache:             32K\n",
      "L1i cache:             32K\n",
      "L2 cache:              256K\n",
      "L3 cache:              12288K\n",
      "NUMA node0 CPU(s):     0-23\n",
      "\n",
      "------------\n",
      "\n",
      "Memory Specs\n",
      "\n",
      "             total       used       free     shared    buffers     cached\n",
      "Mem:           70G       5.7G        65G       120M       596M       1.6G\n",
      "-/+ buffers/cache:       3.6G        67G\n",
      "Swap:         4.7G         0B       4.7G\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "echo \"TODAY'S DATE\"\n",
    "date\n",
    "echo \"------------\"\n",
    "echo \"\"\n",
    "echo \"HOSTNAME: $hostname\"\n",
    "echo \"\"\n",
    "echo \"------------\"\n",
    "echo \"Computer Specs:\"\n",
    "echo \"\"\n",
    "lscpu\n",
    "echo \"\"\n",
    "echo \"------------\"\n",
    "echo \"\"\n",
    "echo \"Memory Specs\"\n",
    "echo \"\"\n",
    "free -mh"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "473812ccc24f\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "hostname"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/data\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "pwd"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Assemble PacBio data with [Canu v1.6](https://github.com/marbl/canu/releases)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/usr/local/bioinformatics/canu-1.6/Linux-amd64/bin/canu\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "which canu"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "mkdir 20171018_oly_pacbio_canu"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "ls 20171018_oly_pacbio_canu/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "ls 20171018_oly_pacbio_canu/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "m130619_081336_42134_c100525122550000001823081109281326_s1_p0.fastq\n",
      "m170211_224036_42134_c101073082550000001823236402101737_s1_X0_filtered_subreads.fastq.gz\n",
      "m170301_100013_42134_c101174162550000001823269408211761_s1_p0_filtered_subreads.fastq.gz\n",
      "m170301_162825_42134_c101174162550000001823269408211762_s1_p0_filtered_subreads.fastq.gz\n",
      "m170301_225711_42134_c101174162550000001823269408211763_s1_p0_filtered_subreads.fastq.gz\n",
      "m170308_163922_42134_c101174252550000001823269408211742_s1_p0_filtered_subreads.fastq.gz\n",
      "m170308_230815_42134_c101174252550000001823269408211743_s1_p0_filtered_subreads.fastq.gz\n",
      "m170315_001112_42134_c101169372550000001823273008151717_s1_p0_filtered_subreads.fastq.gz\n",
      "m170315_063041_42134_c101169382550000001823273008151700_s1_p0_filtered_subreads.fastq.gz\n",
      "m170315_124938_42134_c101169382550000001823273008151701_s1_p0_filtered_subreads.fastq.gz\n",
      "m170315_190851_42134_c101169382550000001823273008151702_s1_p0_filtered_subreads.fastq.gz\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "ls m*"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "usage:   canu [-version] [-citation] \\\n",
      "              [-correct | -trim | -assemble | -trim-assemble] \\\n",
      "              [-s <assembly-specifications-file>] \\\n",
      "               -p <assembly-prefix> \\\n",
      "               -d <assembly-directory> \\\n",
      "               genomeSize=<number>[g|m|k] \\\n",
      "              [other-options] \\\n",
      "              [-pacbio-raw |\n",
      "               -pacbio-corrected |\n",
      "               -nanopore-raw |\n",
      "               -nanopore-corrected] file1 file2 ...\n",
      "\n",
      "example: canu -d run1 -p godzilla genomeSize=1g -nanopore-raw reads/*.fasta.gz \n",
      "\n",
      "\n",
      "  To restrict canu to only a specific stage, use:\n",
      "    -correct       - generate corrected reads\n",
      "    -trim          - generate trimmed reads\n",
      "    -assemble      - generate an assembly\n",
      "    -trim-assemble - generate trimmed reads and then assemble them\n",
      "\n",
      "  The assembly is computed in the -d <assembly-directory>, with output files named\n",
      "  using the -p <assembly-prefix>.  This directory is created if needed.  It is not\n",
      "  possible to run multiple assemblies in the same directory.\n",
      "\n",
      "  The genome size should be your best guess of the haploid genome size of what is being\n",
      "  assembled.  It is used primarily to estimate coverage in reads, NOT as the desired\n",
      "  assembly size.  Fractional values are allowed: '4.7m' equals '4700k' equals '4700000'\n",
      "\n",
      "  Some common options:\n",
      "    useGrid=string\n",
      "      - Run under grid control (true), locally (false), or set up for grid control\n",
      "        but don't submit any jobs (remote)\n",
      "    rawErrorRate=fraction-error\n",
      "      - The allowed difference in an overlap between two raw uncorrected reads.  For lower\n",
      "        quality reads, use a higher number.  The defaults are 0.300 for PacBio reads and\n",
      "        0.500 for Nanopore reads.\n",
      "    correctedErrorRate=fraction-error\n",
      "      - The allowed difference in an overlap between two corrected reads.  Assemblies of\n",
      "        low coverage or data with biological differences will benefit from a slight increase\n",
      "        in this.  Defaults are 0.045 for PacBio reads and 0.144 for Nanopore reads.\n",
      "    gridOptions=string\n",
      "      - Pass string to the command used to submit jobs to the grid.  Can be used to set\n",
      "        maximum run time limits.  Should NOT be used to set memory limits; Canu will do\n",
      "        that for you.\n",
      "    minReadLength=number\n",
      "      - Ignore reads shorter than 'number' bases long.  Default: 1000.\n",
      "    minOverlapLength=number\n",
      "      - Ignore read-to-read overlaps shorter than 'number' bases long.  Default: 500.\n",
      "  A full list of options can be printed with '-options'.  All options can be supplied in\n",
      "  an optional sepc file with the -s option.\n",
      "\n",
      "  Reads can be either FASTA or FASTQ format, uncompressed, or compressed with gz, bz2 or xz.\n",
      "  Reads are specified by the technology they were generated with:\n",
      "    -pacbio-raw         <files>\n",
      "    -pacbio-corrected   <files>\n",
      "    -nanopore-raw       <files>\n",
      "    -nanopore-corrected <files>\n",
      "\n",
      "Complete documentation at http://canu.readthedocs.org/en/latest/\n",
      "\n",
      "ERROR:  mhap overlapper requires java version at least 1.8.0; you have 1.7.0_151 (from 'java').\n",
      "ERROR:  'java -showversion' reports:\n",
      "ERROR:    'java version \"1.7.0_151\"'\n",
      "ERROR:    'OpenJDK Runtime Environment (IcedTea 2.6.11) (7u151-2.6.11-1~deb8u1)'\n",
      "ERROR:    'OpenJDK 64-Bit Server VM (build 24.151-b01, mixed mode)'\n",
      "ERROR:    ''\n",
      "ERROR:    'Usage: java [-options] class [args...]'\n",
      "ERROR:    '           (to execute a class)'\n",
      "ERROR:    '   or  java [-options] -jar jarfile [args...]'\n",
      "ERROR:    '           (to execute a jar file)'\n",
      "ERROR:    'where options include:'\n",
      "ERROR:    '    -d32\t  use a 32-bit data model if available'\n",
      "ERROR:    '    -d64\t  use a 64-bit data model if available'\n",
      "ERROR:    '    -server\t  to select the \"server\" VM'\n",
      "ERROR:    '    -zero\t  to select the \"zero\" VM'\n",
      "ERROR:    '    -jamvm\t  to select the \"jamvm\" VM'\n",
      "ERROR:    '    -avian\t  to select the \"avian\" VM'\n",
      "ERROR:    '    -dcevm\t  to select the \"dcevm\" VM'\n",
      "ERROR:    '                  The default VM is server,'\n",
      "ERROR:    '                  because you are running on a server-class machine.'\n",
      "ERROR:    ''\n",
      "ERROR:    ''\n",
      "ERROR:    '    -cp <class search path of directories and zip/jar files>'\n",
      "ERROR:    '    -classpath <class search path of directories and zip/jar files>'\n",
      "ERROR:    '                  A : separated list of directories, JAR archives,'\n",
      "ERROR:    '                  and ZIP archives to search for class files.'\n",
      "ERROR:    '    -D<name>=<value>'\n",
      "ERROR:    '                  set a system property'\n",
      "ERROR:    '    -verbose:[class|gc|jni]'\n",
      "ERROR:    '                  enable verbose output'\n",
      "ERROR:    '    -version      print product version and exit'\n",
      "ERROR:    '    -version:<value>'\n",
      "ERROR:    '                  require the specified version to run'\n",
      "ERROR:    '    -showversion  print product version and continue'\n",
      "ERROR:    '    -jre-restrict-search | -no-jre-restrict-search'\n",
      "ERROR:    '                  include/exclude user private JREs in the version search'\n",
      "ERROR:    '    -? -help      print this help message'\n",
      "ERROR:    '    -X            print help on non-standard options'\n",
      "ERROR:    '    -ea[:<packagename>...|:<classname>]'\n",
      "ERROR:    '    -enableassertions[:<packagename>...|:<classname>]'\n",
      "ERROR:    '                  enable assertions with specified granularity'\n",
      "ERROR:    '    -da[:<packagename>...|:<classname>]'\n",
      "ERROR:    '    -disableassertions[:<packagename>...|:<classname>]'\n",
      "ERROR:    '                  disable assertions with specified granularity'\n",
      "ERROR:    '    -esa | -enablesystemassertions'\n",
      "ERROR:    '                  enable system assertions'\n",
      "ERROR:    '    -dsa | -disablesystemassertions'\n",
      "ERROR:    '                  disable system assertions'\n",
      "ERROR:    '    -agentlib:<libname>[=<options>]'\n",
      "ERROR:    '                  load native agent library <libname>, e.g. -agentlib:hprof'\n",
      "ERROR:    '                  see also, -agentlib:jdwp=help and -agentlib:hprof=help'\n",
      "ERROR:    '    -agentpath:<pathname>[=<options>]'\n",
      "ERROR:    '                  load native agent library by full pathname'\n",
      "ERROR:    '    -javaagent:<jarpath>[=<options>]'\n",
      "ERROR:    '                  load Java programming language agent, see java.lang.instrument'\n",
      "ERROR:    '    -splash:<imagepath>'\n",
      "ERROR:    '                  show splash screen with specified image'\n",
      "ERROR:    'See http://www.oracle.com/technetwork/java/javase/documentation/index.html for more details.'\n",
      "ERROR:  Failed to run gnuplot from 'gnuplot'.ERROR:  Set option gnuplot=<path-to-gnuplot> or gnuplotTested=true to skip this test and not generate plots.\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "-- Canu 1.6\n",
      "--\n",
      "-- CITATIONS\n",
      "--\n",
      "-- Koren S, Walenz BP, Berlin K, Miller JR, Phillippy AM.\n",
      "-- Canu: scalable and accurate long-read assembly via adaptive k-mer weighting and repeat separation.\n",
      "-- Genome Res. 2017 May;27(5):722-736.\n",
      "-- http://doi.org/10.1101/gr.215087.116\n",
      "-- \n",
      "-- Read and contig alignments during correction, consensus and GFA building use:\n",
      "--   Šošic M, Šikic M.\n",
      "--   Edlib: a C/C ++ library for fast, exact sequence alignment using edit distance.\n",
      "--   Bioinformatics. 2017 May 1;33(9):1394-1395.\n",
      "--   http://doi.org/10.1093/bioinformatics/btw753\n",
      "-- \n",
      "-- Overlaps are generated using:\n",
      "--   Berlin K, et al.\n",
      "--   Assembling large genomes with single-molecule sequencing and locality-sensitive hashing.\n",
      "--   Nat Biotechnol. 2015 Jun;33(6):623-30.\n",
      "--   http://doi.org/10.1038/nbt.3238\n",
      "-- \n",
      "--   Myers EW, et al.\n",
      "--   A Whole-Genome Assembly of Drosophila.\n",
      "--   Science. 2000 Mar 24;287(5461):2196-204.\n",
      "--   http://doi.org/10.1126/science.287.5461.2196\n",
      "-- \n",
      "--   Li H.\n",
      "--   Minimap and miniasm: fast mapping and de novo assembly for noisy long sequences.\n",
      "--   Bioinformatics. 2016 Jul 15;32(14):2103-10.\n",
      "--   http://doi.org/10.1093/bioinformatics/btw152\n",
      "-- \n",
      "-- Corrected read consensus sequences are generated using an algorithm derived from FALCON-sense:\n",
      "--   Chin CS, et al.\n",
      "--   Phased diploid genome assembly with single-molecule real-time sequencing.\n",
      "--   Nat Methods. 2016 Dec;13(12):1050-1054.\n",
      "--   http://doi.org/10.1038/nmeth.4035\n",
      "-- \n",
      "-- Contig consensus sequences are generated using an algorithm derived from pbdagcon:\n",
      "--   Chin CS, et al.\n",
      "--   Nonhybrid, finished microbial genome assemblies from long-read SMRT sequencing data.\n",
      "--   Nat Methods. 2013 Jun;10(6):563-9\n",
      "--   http://doi.org/10.1038/nmeth.2474\n",
      "-- \n",
      "-- CONFIGURE CANU\n",
      "--\n",
      "\n",
      "real\t0m2.252s\n",
      "user\t0m0.304s\n",
      "sys\t0m0.076s\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "time canu \\\n",
    "useGrid=false \\\n",
    "-p 20171009_oly_pacbio \\\n",
    "-d /home/data/20171018_oly_pacbio_canu/ \\\n",
    "genomeSize=1.9g \\\n",
    "correctedErrorRate=0.075 \\\n",
    "-pacbio-raw m*"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Hmmmm, java version isn't current. In theory, the Dockerfile should pull the most current version. Will fix..."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "java version \"1.7.0_151\"\n",
      "OpenJDK Runtime Environment (IcedTea 2.6.11) (7u151-2.6.11-1~deb8u1)\n",
      "OpenJDK 64-Bit Server VM (build 24.151-b01, mixed mode)\n",
      "\n",
      "Usage: java [-options] class [args...]\n",
      "           (to execute a class)\n",
      "   or  java [-options] -jar jarfile [args...]\n",
      "           (to execute a jar file)\n",
      "where options include:\n",
      "    -d32\t  use a 32-bit data model if available\n",
      "    -d64\t  use a 64-bit data model if available\n",
      "    -server\t  to select the \"server\" VM\n",
      "    -zero\t  to select the \"zero\" VM\n",
      "    -jamvm\t  to select the \"jamvm\" VM\n",
      "    -avian\t  to select the \"avian\" VM\n",
      "    -dcevm\t  to select the \"dcevm\" VM\n",
      "                  The default VM is server,\n",
      "                  because you are running on a server-class machine.\n",
      "\n",
      "\n",
      "    -cp <class search path of directories and zip/jar files>\n",
      "    -classpath <class search path of directories and zip/jar files>\n",
      "                  A : separated list of directories, JAR archives,\n",
      "                  and ZIP archives to search for class files.\n",
      "    -D<name>=<value>\n",
      "                  set a system property\n",
      "    -verbose:[class|gc|jni]\n",
      "                  enable verbose output\n",
      "    -version      print product version and exit\n",
      "    -version:<value>\n",
      "                  require the specified version to run\n",
      "    -showversion  print product version and continue\n",
      "    -jre-restrict-search | -no-jre-restrict-search\n",
      "                  include/exclude user private JREs in the version search\n",
      "    -? -help      print this help message\n",
      "    -X            print help on non-standard options\n",
      "    -ea[:<packagename>...|:<classname>]\n",
      "    -enableassertions[:<packagename>...|:<classname>]\n",
      "                  enable assertions with specified granularity\n",
      "    -da[:<packagename>...|:<classname>]\n",
      "    -disableassertions[:<packagename>...|:<classname>]\n",
      "                  disable assertions with specified granularity\n",
      "    -esa | -enablesystemassertions\n",
      "                  enable system assertions\n",
      "    -dsa | -disablesystemassertions\n",
      "                  disable system assertions\n",
      "    -agentlib:<libname>[=<options>]\n",
      "                  load native agent library <libname>, e.g. -agentlib:hprof\n",
      "                  see also, -agentlib:jdwp=help and -agentlib:hprof=help\n",
      "    -agentpath:<pathname>[=<options>]\n",
      "                  load native agent library by full pathname\n",
      "    -javaagent:<jarpath>[=<options>]\n",
      "                  load Java programming language agent, see java.lang.instrument\n",
      "    -splash:<imagepath>\n",
      "                  show splash screen with specified image\n",
      "See http://www.oracle.com/technetwork/java/javase/documentation/index.html for more details.\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "java -showversion"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Wed Oct 18 16:48:11 UTC 2017\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "date"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "bash: line 1: fg: no job control\n",
      "openjdk version \"1.8.0_131\"\n",
      "OpenJDK Runtime Environment (build 1.8.0_131-8u131-b11-1~bpo8+1-b11)\n",
      "OpenJDK 64-Bit Server VM (build 25.131-b11, mixed mode)\n",
      "\n",
      "Usage: java [-options] class [args...]\n",
      "           (to execute a class)\n",
      "   or  java [-options] -jar jarfile [args...]\n",
      "           (to execute a jar file)\n",
      "where options include:\n",
      "    -d32\t  use a 32-bit data model if available\n",
      "    -d64\t  use a 64-bit data model if available\n",
      "    -server\t  to select the \"server\" VM\n",
      "    -zero\t  to select the \"zero\" VM\n",
      "    -dcevm\t  to select the \"dcevm\" VM\n",
      "                  The default VM is server,\n",
      "                  because you are running on a server-class machine.\n",
      "\n",
      "\n",
      "    -cp <class search path of directories and zip/jar files>\n",
      "    -classpath <class search path of directories and zip/jar files>\n",
      "                  A : separated list of directories, JAR archives,\n",
      "                  and ZIP archives to search for class files.\n",
      "    -D<name>=<value>\n",
      "                  set a system property\n",
      "    -verbose:[class|gc|jni]\n",
      "                  enable verbose output\n",
      "    -version      print product version and exit\n",
      "    -version:<value>\n",
      "                  Warning: this feature is deprecated and will be removed\n",
      "                  in a future release.\n",
      "                  require the specified version to run\n",
      "    -showversion  print product version and continue\n",
      "    -jre-restrict-search | -no-jre-restrict-search\n",
      "                  Warning: this feature is deprecated and will be removed\n",
      "                  in a future release.\n",
      "                  include/exclude user private JREs in the version search\n",
      "    -? -help      print this help message\n",
      "    -X            print help on non-standard options\n",
      "    -ea[:<packagename>...|:<classname>]\n",
      "    -enableassertions[:<packagename>...|:<classname>]\n",
      "                  enable assertions with specified granularity\n",
      "    -da[:<packagename>...|:<classname>]\n",
      "    -disableassertions[:<packagename>...|:<classname>]\n",
      "                  disable assertions with specified granularity\n",
      "    -esa | -enablesystemassertions\n",
      "                  enable system assertions\n",
      "    -dsa | -disablesystemassertions\n",
      "                  disable system assertions\n",
      "    -agentlib:<libname>[=<options>]\n",
      "                  load native agent library <libname>, e.g. -agentlib:hprof\n",
      "                  see also, -agentlib:jdwp=help and -agentlib:hprof=help\n",
      "    -agentpath:<pathname>[=<options>]\n",
      "                  load native agent library by full pathname\n",
      "    -javaagent:<jarpath>[=<options>]\n",
      "                  load Java programming language agent, see java.lang.instrument\n",
      "    -splash:<imagepath>\n",
      "                  show splash screen with specified image\n",
      "See http://www.oracle.com/technetwork/java/javase/documentation/index.html for more details.\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "%%bash\n",
    "java -showversion"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "usage:   canu [-version] [-citation] \\\n",
      "              [-correct | -trim | -assemble | -trim-assemble] \\\n",
      "              [-s <assembly-specifications-file>] \\\n",
      "               -p <assembly-prefix> \\\n",
      "               -d <assembly-directory> \\\n",
      "               genomeSize=<number>[g|m|k] \\\n",
      "              [other-options] \\\n",
      "              [-pacbio-raw |\n",
      "               -pacbio-corrected |\n",
      "               -nanopore-raw |\n",
      "               -nanopore-corrected] file1 file2 ...\n",
      "\n",
      "example: canu -d run1 -p godzilla genomeSize=1g -nanopore-raw reads/*.fasta.gz \n",
      "\n",
      "\n",
      "  To restrict canu to only a specific stage, use:\n",
      "    -correct       - generate corrected reads\n",
      "    -trim          - generate trimmed reads\n",
      "    -assemble      - generate an assembly\n",
      "    -trim-assemble - generate trimmed reads and then assemble them\n",
      "\n",
      "  The assembly is computed in the -d <assembly-directory>, with output files named\n",
      "  using the -p <assembly-prefix>.  This directory is created if needed.  It is not\n",
      "  possible to run multiple assemblies in the same directory.\n",
      "\n",
      "  The genome size should be your best guess of the haploid genome size of what is being\n",
      "  assembled.  It is used primarily to estimate coverage in reads, NOT as the desired\n",
      "  assembly size.  Fractional values are allowed: '4.7m' equals '4700k' equals '4700000'\n",
      "\n",
      "  Some common options:\n",
      "    useGrid=string\n",
      "      - Run under grid control (true), locally (false), or set up for grid control\n",
      "        but don't submit any jobs (remote)\n",
      "    rawErrorRate=fraction-error\n",
      "      - The allowed difference in an overlap between two raw uncorrected reads.  For lower\n",
      "        quality reads, use a higher number.  The defaults are 0.300 for PacBio reads and\n",
      "        0.500 for Nanopore reads.\n",
      "    correctedErrorRate=fraction-error\n",
      "      - The allowed difference in an overlap between two corrected reads.  Assemblies of\n",
      "        low coverage or data with biological differences will benefit from a slight increase\n",
      "        in this.  Defaults are 0.045 for PacBio reads and 0.144 for Nanopore reads.\n",
      "    gridOptions=string\n",
      "      - Pass string to the command used to submit jobs to the grid.  Can be used to set\n",
      "        maximum run time limits.  Should NOT be used to set memory limits; Canu will do\n",
      "        that for you.\n",
      "    minReadLength=number\n",
      "      - Ignore reads shorter than 'number' bases long.  Default: 1000.\n",
      "    minOverlapLength=number\n",
      "      - Ignore read-to-read overlaps shorter than 'number' bases long.  Default: 500.\n",
      "  A full list of options can be printed with '-options'.  All options can be supplied in\n",
      "  an optional sepc file with the -s option.\n",
      "\n",
      "  Reads can be either FASTA or FASTQ format, uncompressed, or compressed with gz, bz2 or xz.\n",
      "  Reads are specified by the technology they were generated with:\n",
      "    -pacbio-raw         <files>\n",
      "    -pacbio-corrected   <files>\n",
      "    -nanopore-raw       <files>\n",
      "    -nanopore-corrected <files>\n",
      "\n",
      "Complete documentation at http://canu.readthedocs.org/en/latest/\n",
      "\n",
      "ERROR:  Failed to run gnuplot from 'gnuplot'.ERROR:  Set option gnuplot=<path-to-gnuplot> or gnuplotTested=true to skip this test and not generate plots.\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "-- Canu 1.6\n",
      "--\n",
      "-- CITATIONS\n",
      "--\n",
      "-- Koren S, Walenz BP, Berlin K, Miller JR, Phillippy AM.\n",
      "-- Canu: scalable and accurate long-read assembly via adaptive k-mer weighting and repeat separation.\n",
      "-- Genome Res. 2017 May;27(5):722-736.\n",
      "-- http://doi.org/10.1101/gr.215087.116\n",
      "-- \n",
      "-- Read and contig alignments during correction, consensus and GFA building use:\n",
      "--   Šošic M, Šikic M.\n",
      "--   Edlib: a C/C ++ library for fast, exact sequence alignment using edit distance.\n",
      "--   Bioinformatics. 2017 May 1;33(9):1394-1395.\n",
      "--   http://doi.org/10.1093/bioinformatics/btw753\n",
      "-- \n",
      "-- Overlaps are generated using:\n",
      "--   Berlin K, et al.\n",
      "--   Assembling large genomes with single-molecule sequencing and locality-sensitive hashing.\n",
      "--   Nat Biotechnol. 2015 Jun;33(6):623-30.\n",
      "--   http://doi.org/10.1038/nbt.3238\n",
      "-- \n",
      "--   Myers EW, et al.\n",
      "--   A Whole-Genome Assembly of Drosophila.\n",
      "--   Science. 2000 Mar 24;287(5461):2196-204.\n",
      "--   http://doi.org/10.1126/science.287.5461.2196\n",
      "-- \n",
      "--   Li H.\n",
      "--   Minimap and miniasm: fast mapping and de novo assembly for noisy long sequences.\n",
      "--   Bioinformatics. 2016 Jul 15;32(14):2103-10.\n",
      "--   http://doi.org/10.1093/bioinformatics/btw152\n",
      "-- \n",
      "-- Corrected read consensus sequences are generated using an algorithm derived from FALCON-sense:\n",
      "--   Chin CS, et al.\n",
      "--   Phased diploid genome assembly with single-molecule real-time sequencing.\n",
      "--   Nat Methods. 2016 Dec;13(12):1050-1054.\n",
      "--   http://doi.org/10.1038/nmeth.4035\n",
      "-- \n",
      "-- Contig consensus sequences are generated using an algorithm derived from pbdagcon:\n",
      "--   Chin CS, et al.\n",
      "--   Nonhybrid, finished microbial genome assemblies from long-read SMRT sequencing data.\n",
      "--   Nat Methods. 2013 Jun;10(6):563-9\n",
      "--   http://doi.org/10.1038/nmeth.2474\n",
      "-- \n",
      "-- CONFIGURE CANU\n",
      "--\n",
      "-- Detected Java(TM) Runtime Environment '1.8.0_131' (from 'java').\n",
      "\n",
      "real\t0m0.234s\n",
      "user\t0m0.184s\n",
      "sys\t0m0.064s\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "time canu \\\n",
    "useGrid=false \\\n",
    "-p 20171009_oly_pacbio \\\n",
    "-d /home/data/20171018_oly_pacbio_canu/ \\\n",
    "genomeSize=1.9g \\\n",
    "correctedErrorRate=0.075 \\\n",
    "-pacbio-raw m*"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Error message says it can't find gnuplot. Will update Docker file to install gnuplot. BRB..."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Wed Oct 18 17:05:55 UTC 2017\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "date"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "usage:   canu [-version] [-citation] \\\n",
      "              [-correct | -trim | -assemble | -trim-assemble] \\\n",
      "              [-s <assembly-specifications-file>] \\\n",
      "               -p <assembly-prefix> \\\n",
      "               -d <assembly-directory> \\\n",
      "               genomeSize=<number>[g|m|k] \\\n",
      "              [other-options] \\\n",
      "              [-pacbio-raw |\n",
      "               -pacbio-corrected |\n",
      "               -nanopore-raw |\n",
      "               -nanopore-corrected] file1 file2 ...\n",
      "\n",
      "example: canu -d run1 -p godzilla genomeSize=1g -nanopore-raw reads/*.fasta.gz \n",
      "\n",
      "\n",
      "  To restrict canu to only a specific stage, use:\n",
      "    -correct       - generate corrected reads\n",
      "    -trim          - generate trimmed reads\n",
      "    -assemble      - generate an assembly\n",
      "    -trim-assemble - generate trimmed reads and then assemble them\n",
      "\n",
      "  The assembly is computed in the -d <assembly-directory>, with output files named\n",
      "  using the -p <assembly-prefix>.  This directory is created if needed.  It is not\n",
      "  possible to run multiple assemblies in the same directory.\n",
      "\n",
      "  The genome size should be your best guess of the haploid genome size of what is being\n",
      "  assembled.  It is used primarily to estimate coverage in reads, NOT as the desired\n",
      "  assembly size.  Fractional values are allowed: '4.7m' equals '4700k' equals '4700000'\n",
      "\n",
      "  Some common options:\n",
      "    useGrid=string\n",
      "      - Run under grid control (true), locally (false), or set up for grid control\n",
      "        but don't submit any jobs (remote)\n",
      "    rawErrorRate=fraction-error\n",
      "      - The allowed difference in an overlap between two raw uncorrected reads.  For lower\n",
      "        quality reads, use a higher number.  The defaults are 0.300 for PacBio reads and\n",
      "        0.500 for Nanopore reads.\n",
      "    correctedErrorRate=fraction-error\n",
      "      - The allowed difference in an overlap between two corrected reads.  Assemblies of\n",
      "        low coverage or data with biological differences will benefit from a slight increase\n",
      "        in this.  Defaults are 0.045 for PacBio reads and 0.144 for Nanopore reads.\n",
      "    gridOptions=string\n",
      "      - Pass string to the command used to submit jobs to the grid.  Can be used to set\n",
      "        maximum run time limits.  Should NOT be used to set memory limits; Canu will do\n",
      "        that for you.\n",
      "    minReadLength=number\n",
      "      - Ignore reads shorter than 'number' bases long.  Default: 1000.\n",
      "    minOverlapLength=number\n",
      "      - Ignore read-to-read overlaps shorter than 'number' bases long.  Default: 500.\n",
      "  A full list of options can be printed with '-options'.  All options can be supplied in\n",
      "  an optional sepc file with the -s option.\n",
      "\n",
      "  Reads can be either FASTA or FASTQ format, uncompressed, or compressed with gz, bz2 or xz.\n",
      "  Reads are specified by the technology they were generated with:\n",
      "    -pacbio-raw         <files>\n",
      "    -pacbio-corrected   <files>\n",
      "    -nanopore-raw       <files>\n",
      "    -nanopore-corrected <files>\n",
      "\n",
      "Complete documentation at http://canu.readthedocs.org/en/latest/\n",
      "\n",
      "ERROR:  ovsMethod=parallel requires useGrid=true or useGrid=remote.  Set ovsMethod=sequential if no grid is available\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "-- Canu 1.6\n",
      "--\n",
      "-- CITATIONS\n",
      "--\n",
      "-- Koren S, Walenz BP, Berlin K, Miller JR, Phillippy AM.\n",
      "-- Canu: scalable and accurate long-read assembly via adaptive k-mer weighting and repeat separation.\n",
      "-- Genome Res. 2017 May;27(5):722-736.\n",
      "-- http://doi.org/10.1101/gr.215087.116\n",
      "-- \n",
      "-- Read and contig alignments during correction, consensus and GFA building use:\n",
      "--   Šošic M, Šikic M.\n",
      "--   Edlib: a C/C ++ library for fast, exact sequence alignment using edit distance.\n",
      "--   Bioinformatics. 2017 May 1;33(9):1394-1395.\n",
      "--   http://doi.org/10.1093/bioinformatics/btw753\n",
      "-- \n",
      "-- Overlaps are generated using:\n",
      "--   Berlin K, et al.\n",
      "--   Assembling large genomes with single-molecule sequencing and locality-sensitive hashing.\n",
      "--   Nat Biotechnol. 2015 Jun;33(6):623-30.\n",
      "--   http://doi.org/10.1038/nbt.3238\n",
      "-- \n",
      "--   Myers EW, et al.\n",
      "--   A Whole-Genome Assembly of Drosophila.\n",
      "--   Science. 2000 Mar 24;287(5461):2196-204.\n",
      "--   http://doi.org/10.1126/science.287.5461.2196\n",
      "-- \n",
      "--   Li H.\n",
      "--   Minimap and miniasm: fast mapping and de novo assembly for noisy long sequences.\n",
      "--   Bioinformatics. 2016 Jul 15;32(14):2103-10.\n",
      "--   http://doi.org/10.1093/bioinformatics/btw152\n",
      "-- \n",
      "-- Corrected read consensus sequences are generated using an algorithm derived from FALCON-sense:\n",
      "--   Chin CS, et al.\n",
      "--   Phased diploid genome assembly with single-molecule real-time sequencing.\n",
      "--   Nat Methods. 2016 Dec;13(12):1050-1054.\n",
      "--   http://doi.org/10.1038/nmeth.4035\n",
      "-- \n",
      "-- Contig consensus sequences are generated using an algorithm derived from pbdagcon:\n",
      "--   Chin CS, et al.\n",
      "--   Nonhybrid, finished microbial genome assemblies from long-read SMRT sequencing data.\n",
      "--   Nat Methods. 2013 Jun;10(6):563-9\n",
      "--   http://doi.org/10.1038/nmeth.2474\n",
      "-- \n",
      "-- CONFIGURE CANU\n",
      "--\n",
      "-- Detected Java(TM) Runtime Environment '1.8.0_131' (from 'java').\n",
      "-- Detected gnuplot version '5.2 patchlevel 0' (from 'gnuplot') and image format 'svg'.\n",
      "-- Detected 24 CPUs and 71 gigabytes of memory.\n",
      "-- No grid engine detected, grid disabled.\n",
      "--\n",
      "--                            (tag)Concurrency\n",
      "--                     (tag)Threads          |\n",
      "--            (tag)Memory         |          |\n",
      "--        (tag)         |         |          |  algorithm\n",
      "--        -------  ------  --------   --------  -----------------------------\n",
      "-- Local: meryl     71 GB   24 CPUs x   1 job   (k-mer counting)\n",
      "-- Local: cormhap   32 GB   12 CPUs x   2 jobs  (overlap detection with mhap)\n",
      "-- Local: obtovl    12 GB    8 CPUs x   3 jobs  (overlap detection)\n",
      "-- Local: utgovl    12 GB    8 CPUs x   3 jobs  (overlap detection)\n",
      "-- Local: cor       11 GB    4 CPUs x   6 jobs  (read correction)\n",
      "-- Local: ovb        2 GB    1 CPU  x  24 jobs  (overlap store bucketizer)\n",
      "-- Local: ovs       32 GB    1 CPU  x  24 jobs  (overlap store sorting)\n",
      "-- Local: red        8 GB    8 CPUs x   3 jobs  (read error detection)\n",
      "-- Local: oea        2 GB    1 CPU  x  24 jobs  (overlap error adjustment)\n",
      "-- Local: bat       35 GB   12 CPUs x   2 jobs  (contig construction)\n",
      "-- Local: cns       71 GB    8 CPUs x   3 jobs  (consensus)\n",
      "-- Local: gfa       16 GB   12 CPUs x   2 jobs  (GFA alignment and processing)\n",
      "--\n",
      "-- Found PacBio uncorrected reads in the input files.\n",
      "\n",
      "real\t0m0.297s\n",
      "user\t0m0.232s\n",
      "sys\t0m0.056s\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "time canu \\\n",
    "useGrid=false \\\n",
    "-p 20171009_oly_pacbio \\\n",
    "-d /home/data/20171018_oly_pacbio_canu/ \\\n",
    "genomeSize=1.9g \\\n",
    "correctedErrorRate=0.075 \\\n",
    "-pacbio-raw m*"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Error message says I need to change ```ovsMethod=``` to sequentila if no grid available. Figured that'd be adjusted when I specified ```useGrid=false```... Will change command."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Code overview:\n",
    "\n",
    "- ```useGrid=false```: Needed to disable Canu's parallel computing scheduler thingy (on by default)\n",
    "- ```ovsMethod=sequential```: Same as above.\n",
    "- ```-p```: The assembly prefix string. Screwed this up; leftover from copy/paste of earlier attempt.\n",
    "- ```-d```: Destination folder for output.\n",
    "- ```genomeSize=```: Needs estimated genome size. Used 1.9g [based on BGI k-mer estimate](http://owl.fish.washington.edu/O_lurida_genome_assemblies_BGI/20160512/20160512_F15FTSUSAT0327_genome_survey.pdf).\n",
    "- ```correctedErrorRate=```: Expecting low coverage, so increased this from the default (4.5% to 7.5%)."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Process is interrupted.\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "time canu \\\n",
    "useGrid=false \\\n",
    "ovsMethod=sequential \\\n",
    "-p 20171009_oly_pacbio \\\n",
    "-d /home/data/20171018_oly_pacbio_canu/ \\\n",
    "genomeSize=1.9g \\\n",
    "correctedErrorRate=0.075 \\\n",
    "-pacbio-raw m*"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "ename": "OSError",
     "evalue": "\"sh\" shell not found",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mOSError\u001b[0m                                   Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-4-3c08e02d176d>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mget_ipython\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mmagic\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34mu'ls -l'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;32m/root/miniconda2/lib/python2.7/site-packages/IPython/core/interactiveshell.pyc\u001b[0m in \u001b[0;36mmagic\u001b[0;34m(self, arg_s)\u001b[0m\n\u001b[1;32m   2158\u001b[0m         \u001b[0mmagic_name\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0m_\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmagic_arg_s\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0marg_s\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mpartition\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m' '\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2159\u001b[0m         \u001b[0mmagic_name\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mmagic_name\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mlstrip\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mprefilter\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mESC_MAGIC\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 2160\u001b[0;31m         \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mrun_line_magic\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mmagic_name\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmagic_arg_s\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   2161\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2162\u001b[0m     \u001b[0;31m#-------------------------------------------------------------------------\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/root/miniconda2/lib/python2.7/site-packages/IPython/core/interactiveshell.pyc\u001b[0m in \u001b[0;36mrun_line_magic\u001b[0;34m(self, magic_name, line)\u001b[0m\n\u001b[1;32m   2079\u001b[0m                 \u001b[0mkwargs\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'local_ns'\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0msys\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_getframe\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mstack_depth\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mf_locals\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2080\u001b[0m             \u001b[0;32mwith\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mbuiltin_trap\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 2081\u001b[0;31m                 \u001b[0mresult\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mfn\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   2082\u001b[0m             \u001b[0;32mreturn\u001b[0m \u001b[0mresult\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2083\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/root/miniconda2/lib/python2.7/site-packages/IPython/core/alias.pyc\u001b[0m in \u001b[0;36m__call__\u001b[0;34m(self, rest)\u001b[0m\n\u001b[1;32m    185\u001b[0m             \u001b[0mcmd\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m'%s %s'\u001b[0m \u001b[0;34m%\u001b[0m \u001b[0;34m(\u001b[0m\u001b[0mcmd\u001b[0m \u001b[0;34m%\u001b[0m \u001b[0mtuple\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0mnargs\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m' '\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mjoin\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mnargs\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    186\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 187\u001b[0;31m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mshell\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msystem\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcmd\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    188\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    189\u001b[0m \u001b[0;31m#-----------------------------------------------------------------------------\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/root/miniconda2/lib/python2.7/site-packages/IPython/core/interactiveshell.pyc\u001b[0m in \u001b[0;36msystem_piped\u001b[0;34m(self, cmd)\u001b[0m\n\u001b[1;32m   2209\u001b[0m         \u001b[0;31m# a non-None value would trigger :func:`sys.displayhook` calls.\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2210\u001b[0m         \u001b[0;31m# Instead, we store the exit_code in user_ns.\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 2211\u001b[0;31m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0muser_ns\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'_exit_code'\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0msystem\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mvar_expand\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcmd\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdepth\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;36m1\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   2212\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2213\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0msystem_raw\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcmd\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/root/miniconda2/lib/python2.7/site-packages/IPython/utils/_process_posix.pyc\u001b[0m in \u001b[0;36msystem\u001b[0;34m(self, cmd)\u001b[0m\n\u001b[1;32m    155\u001b[0m                 \u001b[0mchild\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mpexpect\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mspawnb\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msh\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0margs\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'-c'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcmd\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;31m# Pexpect-U\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    156\u001b[0m             \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 157\u001b[0;31m                 \u001b[0mchild\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mpexpect\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mspawn\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msh\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0margs\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'-c'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcmd\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m  \u001b[0;31m# Vanilla Pexpect\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    158\u001b[0m             \u001b[0mflush\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0msys\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mstdout\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mflush\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    159\u001b[0m             \u001b[0;32mwhile\u001b[0m \u001b[0mTrue\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/root/miniconda2/lib/python2.7/site-packages/IPython/utils/_process_posix.pyc\u001b[0m in \u001b[0;36msh\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m     64\u001b[0m             \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_sh\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mpexpect\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mwhich\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'sh'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     65\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_sh\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0mNone\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 66\u001b[0;31m                 \u001b[0;32mraise\u001b[0m \u001b[0mOSError\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'\"sh\" shell not found'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     67\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     68\u001b[0m         \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_sh\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mOSError\u001b[0m: \"sh\" shell not found"
     ]
    }
   ],
   "source": [
    "ls -l"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Couldn't find program: 'bash'\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "ls -l"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This was crazy! It turns out the entire working directory's contents were deleted!!! Very scary. Going to re-copy the FASTQ files and try again..."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Wed Oct 18 22:36:30 UTC 2017\n",
      "a56217b83023\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "date\n",
    "hostname"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "total 11249952\r\n",
      "-rw-r--r-- 1 root root      52177 Oct 18 22:09 20171018_docker_oly_canu.ipynb\r\n",
      "-rw-rw-r-- 1 1000 1000  389254888 Oct 18 22:25 m130619_081336_42134_c100525122550000001823081109281326_s1_p0.fastq\r\n",
      "-rwxrwxr-x 1 1000 1000 1243634577 Oct 18 22:25 \u001b[0m\u001b[01;32mm170211_224036_42134_c101073082550000001823236402101737_s1_X0_filtered_subreads.fastq.gz\u001b[0m\u001b[K*\r\n",
      "-rwxrwxr-x 1 1000 1000 1388585547 Oct 18 22:26 \u001b[01;32mm170301_100013_42134_c101174162550000001823269408211761_s1_p0_filtered_subreads.fastq.gz\u001b[0m\u001b[K*\r\n",
      "-rwxrwxr-x 1 1000 1000 1258960377 Oct 18 22:26 \u001b[01;32mm170301_162825_42134_c101174162550000001823269408211762_s1_p0_filtered_subreads.fastq.gz\u001b[0m\u001b[K*\r\n",
      "-rwxrwxr-x 1 1000 1000 1384102715 Oct 18 22:26 \u001b[01;32mm170301_225711_42134_c101174162550000001823269408211763_s1_p0_filtered_subreads.fastq.gz\u001b[0m\u001b[K*\r\n",
      "-rwxrwxr-x 1 1000 1000 1305285460 Oct 18 22:27 \u001b[01;32mm170308_163922_42134_c101174252550000001823269408211742_s1_p0_filtered_subreads.fastq.gz\u001b[0m\u001b[K*\r\n",
      "-rwxrwxr-x 1 1000 1000 1321354256 Oct 18 22:27 \u001b[01;32mm170308_230815_42134_c101174252550000001823269408211743_s1_p0_filtered_subreads.fastq.gz\u001b[0m\u001b[K*\r\n",
      "-rwxrwxr-x 1 1000 1000  910321803 Oct 18 22:27 \u001b[01;32mm170315_001112_42134_c101169372550000001823273008151717_s1_p0_filtered_subreads.fastq.gz\u001b[0m\u001b[K*\r\n",
      "-rwxrwxr-x 1 1000 1000  798040545 Oct 18 22:28 \u001b[01;32mm170315_063041_42134_c101169382550000001823273008151700_s1_p0_filtered_subreads.fastq.gz\u001b[0m\u001b[K*\r\n",
      "-rwxrwxr-x 1 1000 1000  724173972 Oct 18 22:28 \u001b[01;32mm170315_124938_42134_c101169382550000001823273008151701_s1_p0_filtered_subreads.fastq.gz\u001b[0m\u001b[K*\r\n",
      "-rwxrwxr-x 1 1000 1000  796112682 Oct 18 22:28 \u001b[01;32mm170315_190851_42134_c101169382550000001823273008151702_s1_p0_filtered_subreads.fastq.gz\u001b[0m\u001b[K*\r\n"
     ]
    }
   ],
   "source": [
    "ls -l"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "mkdir /home/data/20171018_oly_pacbio_canu"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "20171018_docker_oly_canu.ipynb\r\n",
      "\u001b[0m\u001b[01;34m20171018_oly_pacbio_canu\u001b[0m/\r\n",
      "m130619_081336_42134_c100525122550000001823081109281326_s1_p0.fastq\r\n",
      "\u001b[01;32mm170211_224036_42134_c101073082550000001823236402101737_s1_X0_filtered_subreads.fastq.gz\u001b[0m\u001b[K*\r\n",
      "\u001b[01;32mm170301_100013_42134_c101174162550000001823269408211761_s1_p0_filtered_subreads.fastq.gz\u001b[0m\u001b[K*\r\n",
      "\u001b[01;32mm170301_162825_42134_c101174162550000001823269408211762_s1_p0_filtered_subreads.fastq.gz\u001b[0m\u001b[K*\r\n",
      "\u001b[01;32mm170301_225711_42134_c101174162550000001823269408211763_s1_p0_filtered_subreads.fastq.gz\u001b[0m\u001b[K*\r\n",
      "\u001b[01;32mm170308_163922_42134_c101174252550000001823269408211742_s1_p0_filtered_subreads.fastq.gz\u001b[0m\u001b[K*\r\n",
      "\u001b[01;32mm170308_230815_42134_c101174252550000001823269408211743_s1_p0_filtered_subreads.fastq.gz\u001b[0m\u001b[K*\r\n",
      "\u001b[01;32mm170315_001112_42134_c101169372550000001823273008151717_s1_p0_filtered_subreads.fastq.gz\u001b[0m\u001b[K*\r\n",
      "\u001b[01;32mm170315_063041_42134_c101169382550000001823273008151700_s1_p0_filtered_subreads.fastq.gz\u001b[0m\u001b[K*\r\n",
      "\u001b[01;32mm170315_124938_42134_c101169382550000001823273008151701_s1_p0_filtered_subreads.fastq.gz\u001b[0m\u001b[K*\r\n",
      "\u001b[01;32mm170315_190851_42134_c101169382550000001823273008151702_s1_p0_filtered_subreads.fastq.gz\u001b[0m\u001b[K*\r\n"
     ]
    }
   ],
   "source": [
    "ls -F"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "-- Canu 1.6\n",
      "--\n",
      "-- CITATIONS\n",
      "--\n",
      "-- Koren S, Walenz BP, Berlin K, Miller JR, Phillippy AM.\n",
      "-- Canu: scalable and accurate long-read assembly via adaptive k-mer weighting and repeat separation.\n",
      "-- Genome Res. 2017 May;27(5):722-736.\n",
      "-- http://doi.org/10.1101/gr.215087.116\n",
      "-- \n",
      "-- Read and contig alignments during correction, consensus and GFA building use:\n",
      "--   Šošic M, Šikic M.\n",
      "--   Edlib: a C/C ++ library for fast, exact sequence alignment using edit distance.\n",
      "--   Bioinformatics. 2017 May 1;33(9):1394-1395.\n",
      "--   http://doi.org/10.1093/bioinformatics/btw753\n",
      "-- \n",
      "-- Overlaps are generated using:\n",
      "--   Berlin K, et al.\n",
      "--   Assembling large genomes with single-molecule sequencing and locality-sensitive hashing.\n",
      "--   Nat Biotechnol. 2015 Jun;33(6):623-30.\n",
      "--   http://doi.org/10.1038/nbt.3238\n",
      "-- \n",
      "--   Myers EW, et al.\n",
      "--   A Whole-Genome Assembly of Drosophila.\n",
      "--   Science. 2000 Mar 24;287(5461):2196-204.\n",
      "--   http://doi.org/10.1126/science.287.5461.2196\n",
      "-- \n",
      "--   Li H.\n",
      "--   Minimap and miniasm: fast mapping and de novo assembly for noisy long sequences.\n",
      "--   Bioinformatics. 2016 Jul 15;32(14):2103-10.\n",
      "--   http://doi.org/10.1093/bioinformatics/btw152\n",
      "-- \n",
      "-- Corrected read consensus sequences are generated using an algorithm derived from FALCON-sense:\n",
      "--   Chin CS, et al.\n",
      "--   Phased diploid genome assembly with single-molecule real-time sequencing.\n",
      "--   Nat Methods. 2016 Dec;13(12):1050-1054.\n",
      "--   http://doi.org/10.1038/nmeth.4035\n",
      "-- \n",
      "-- Contig consensus sequences are generated using an algorithm derived from pbdagcon:\n",
      "--   Chin CS, et al.\n",
      "--   Nonhybrid, finished microbial genome assemblies from long-read SMRT sequencing data.\n",
      "--   Nat Methods. 2013 Jun;10(6):563-9\n",
      "--   http://doi.org/10.1038/nmeth.2474\n",
      "-- \n",
      "-- CONFIGURE CANU\n",
      "--\n",
      "-- Detected Java(TM) Runtime Environment '1.8.0_131' (from 'java').\n",
      "-- Detected gnuplot version '5.2 patchlevel 0' (from 'gnuplot') and image format 'svg'.\n",
      "-- Detected 24 CPUs and 71 gigabytes of memory.\n",
      "-- No grid engine detected, grid disabled.\n",
      "--\n",
      "--                            (tag)Concurrency\n",
      "--                     (tag)Threads          |\n",
      "--            (tag)Memory         |          |\n",
      "--        (tag)         |         |          |  algorithm\n",
      "--        -------  ------  --------   --------  -----------------------------\n",
      "-- Local: meryl     71 GB   24 CPUs x   1 job   (k-mer counting)\n",
      "-- Local: cormhap   32 GB   12 CPUs x   2 jobs  (overlap detection with mhap)\n",
      "-- Local: obtovl    12 GB    8 CPUs x   3 jobs  (overlap detection)\n",
      "-- Local: utgovl    12 GB    8 CPUs x   3 jobs  (overlap detection)\n",
      "-- Local: cor       11 GB    4 CPUs x   6 jobs  (read correction)\n",
      "-- Local: ovb        2 GB    1 CPU  x  24 jobs  (overlap store bucketizer)\n",
      "-- Local: ovs       32 GB    1 CPU  x  24 jobs  (overlap store sorting)\n",
      "-- Local: red        8 GB    8 CPUs x   3 jobs  (read error detection)\n",
      "-- Local: oea        2 GB    1 CPU  x  24 jobs  (overlap error adjustment)\n",
      "-- Local: bat       35 GB   12 CPUs x   2 jobs  (contig construction)\n",
      "-- Local: cns       71 GB    8 CPUs x   3 jobs  (consensus)\n",
      "-- Local: gfa       16 GB   12 CPUs x   2 jobs  (GFA alignment and processing)\n",
      "--\n",
      "-- Found PacBio uncorrected reads in the input files.\n",
      "--\n",
      "-- Generating assembly '20171018_oly_pacbio' in '/home/data/20171018_oly_pacbio_canu'\n",
      "--\n",
      "-- Parameters:\n",
      "--\n",
      "--  genomeSize        1900000000\n",
      "--\n",
      "--  Overlap Generation Limits:\n",
      "--    corOvlErrorRate 0.2400 ( 24.00%)\n",
      "--    obtOvlErrorRate 0.0750 (  7.50%)\n",
      "--    utgOvlErrorRate 0.0750 (  7.50%)\n",
      "--\n",
      "--  Overlap Processing Limits:\n",
      "--    corErrorRate    0.3000 ( 30.00%)\n",
      "--    obtErrorRate    0.0750 (  7.50%)\n",
      "--    utgErrorRate    0.0750 (  7.50%)\n",
      "--    cnsErrorRate    0.0750 (  7.50%)\n",
      "--\n",
      "--\n",
      "-- BEGIN CORRECTION\n",
      "--\n",
      "----------------------------------------\n",
      "-- Starting command on Wed Oct 18 22:37:54 2017 with 849.001 GB free disk space\n",
      "\n",
      "    cd correction\n",
      "    /usr/local/bioinformatics/canu-1.6/Linux-amd64/bin/gatekeeperCreate \\\n",
      "      -minlength 1000 \\\n",
      "      -o ./20171018_oly_pacbio.gkpStore.BUILDING \\\n",
      "      ./20171018_oly_pacbio.gkpStore.gkp \\\n",
      "    > ./20171018_oly_pacbio.gkpStore.BUILDING.err 2>&1\n",
      "\n",
      "-- Finished on Wed Oct 18 22:41:44 2017 (230 seconds) with 845.607 GB free disk space\n",
      "----------------------------------------\n",
      "--\n",
      "-- In gatekeeper store 'correction/20171018_oly_pacbio.gkpStore':\n",
      "--   Found 1738852 reads.\n",
      "--   Found 12725659496 bases (6.69 times coverage).\n",
      "--\n",
      "--   Read length histogram (one '*' equals 3616.75 reads):\n",
      "--        0    999      0 \n",
      "--     1000   1999 132777 ************************************\n",
      "--     2000   2999 143827 ***************************************\n",
      "--     3000   3999 138184 **************************************\n",
      "--     4000   4999 130729 ************************************\n",
      "--     5000   5999 168299 **********************************************\n",
      "--     6000   6999 253173 **********************************************************************\n",
      "--     7000   7999 189248 ****************************************************\n",
      "--     8000   8999 135856 *************************************\n",
      "--     9000   9999  98310 ***************************\n",
      "--    10000  10999  74004 ********************\n",
      "--    11000  11999  56563 ***************\n",
      "--    12000  12999  44400 ************\n",
      "--    13000  13999  35095 *********\n",
      "--    14000  14999  27878 *******\n",
      "--    15000  15999  21885 ******\n",
      "--    16000  16999  17635 ****\n",
      "--    17000  17999  14000 ***\n",
      "--    18000  18999  11258 ***\n",
      "--    19000  19999   9067 **\n",
      "--    20000  20999   7244 **\n",
      "--    21000  21999   5867 *\n",
      "--    22000  22999   4521 *\n",
      "--    23000  23999   3708 *\n",
      "--    24000  24999   2892 \n",
      "--    25000  25999   2245 \n",
      "--    26000  26999   1894 \n",
      "--    27000  27999   1551 \n",
      "--    28000  28999   1257 \n",
      "--    29000  29999   1005 \n",
      "--    30000  30999    794 \n",
      "--    31000  31999    704 \n",
      "--    32000  32999    542 \n",
      "--    33000  33999    422 \n",
      "--    34000  34999    338 \n",
      "--    35000  35999    273 \n",
      "--    36000  36999    258 \n",
      "--    37000  37999    213 \n",
      "--    38000  38999    165 \n",
      "--    39000  39999    140 \n",
      "--    40000  40999    127 \n",
      "--    41000  41999     97 \n",
      "--    42000  42999     76 \n",
      "--    43000  43999     59 \n",
      "--    44000  44999     49 \n",
      "--    45000  45999     42 \n",
      "--    46000  46999     36 \n",
      "--    47000  47999     29 \n",
      "--    48000  48999     27 \n",
      "--    49000  49999     21 \n",
      "--    50000  50999     11 \n",
      "--    51000  51999     10 \n",
      "--    52000  52999      8 \n",
      "--    53000  53999      7 \n",
      "--    54000  54999      4 \n",
      "--    55000  55999      6 \n",
      "--    56000  56999      5 \n",
      "--    57000  57999      3 \n",
      "--    58000  58999      6 \n",
      "--    59000  59999      0 \n",
      "--    60000  60999      3 \n",
      "--    61000  61999      0 \n",
      "--    62000  62999      1 \n",
      "--    63000  63999      2 \n",
      "--    64000  64999      0 \n",
      "--    65000  65999      1 \n",
      "--    66000  66999      0 \n",
      "--    67000  67999      0 \n",
      "--    68000  68999      1 \n",
      "-- Finished stage 'cor-gatekeeper', reset canuIteration.\n",
      "-- Finished stage 'merylConfigure', reset canuIteration.\n",
      "--\n",
      "-- Running jobs.  First attempt out of 2.\n",
      "----------------------------------------\n",
      "-- Starting 'meryl' concurrent execution on Wed Oct 18 22:42:07 2017 with 845.554 GB free disk space (1 processes; 1 concurrently)\n",
      "\n",
      "    cd correction/0-mercounts\n",
      "    ./meryl.sh 1 > ./meryl.000001.out 2>&1\n",
      "\n",
      "-- Finished on Wed Oct 18 23:15:25 2017 (1998 seconds) with 837.703 GB free disk space\n",
      "----------------------------------------\n",
      "-- Meryl finished successfully.\n",
      "-- Finished stage 'merylCheck', reset canuIteration.\n",
      "--\n",
      "--  16-mers                                                                                           Fraction\n",
      "--    Occurrences   NumMers                                                                         Unique Total\n",
      "--       1-     1 385488671 ****************************************************************       0.2091 0.0304\n",
      "--       2-     2 324564045 ******************************************************                 0.3852 0.0815\n",
      "--       3-     4 418987507 ********************************************************************** 0.5166 0.1387\n",
      "--       5-     7 303655603 **************************************************                     0.6831 0.2457\n",
      "--       8-    11 175681884 *****************************                                          0.8092 0.3707\n",
      "--      12-    16  96333136 ****************                                                       0.8866 0.4859\n",
      "--      17-    22  53526300 ********                                                               0.9312 0.5811\n",
      "--      23-    29  30754739 *****                                                                  0.9569 0.6563\n",
      "--      30-    37  18389991 ***                                                                    0.9721 0.7149\n",
      "--      38-    46  11388024 *                                                                      0.9813 0.7606\n",
      "--      47-    56   7298685 *                                                                      0.9871 0.7964\n",
      "--      57-    67   4830492                                                                        0.9909 0.8248\n",
      "--      68-    79   3287983                                                                        0.9934 0.8476\n",
      "--      80-    92   2276997                                                                        0.9951 0.8660\n",
      "--      93-   106   1616821                                                                        0.9963 0.8810\n",
      "--     107-   121   1168529                                                                        0.9971 0.8934\n",
      "--     122-   137    861132                                                                        0.9977 0.9037\n",
      "--     138-   154    644741                                                                        0.9982 0.9123\n",
      "--     155-   172    494164                                                                        0.9985 0.9196\n",
      "--     173-   191    385159                                                                        0.9988 0.9259\n",
      "--     192-   211    303250                                                                        0.9990 0.9313\n",
      "--     212-   232    241440                                                                        0.9992 0.9361\n",
      "--     233-   254    192941                                                                        0.9993 0.9403\n",
      "--     255-   277    156500                                                                        0.9994 0.9439\n",
      "--     278-   301    127545                                                                        0.9995 0.9472\n",
      "--     302-   326    105904                                                                        0.9996 0.9501\n",
      "--     327-   352     87888                                                                        0.9996 0.9526\n",
      "--     353-   379     72868                                                                        0.9997 0.9550\n",
      "--     380-   407     61693                                                                        0.9997 0.9571\n",
      "--     408-   436     52543                                                                        0.9997 0.9590\n",
      "--     437-   466     44553                                                                        0.9998 0.9607\n",
      "--     467-   497     38393                                                                        0.9998 0.9623\n",
      "--     498-   529     33193                                                                        0.9998 0.9637\n",
      "--     530-   562     29162                                                                        0.9998 0.9651\n",
      "--     563-   596     25892                                                                        0.9998 0.9663\n",
      "--     597-   631     22484                                                                        0.9999 0.9675\n",
      "--     632-   667     20132                                                                        0.9999 0.9686\n",
      "--     668-   704     17937                                                                        0.9999 0.9696\n",
      "--     705-   742     15848                                                                        0.9999 0.9706\n",
      "--     743-   781     14105                                                                        0.9999 0.9715\n",
      "--     782-   821     12734                                                                        0.9999 0.9723\n",
      "--\n",
      "--     4763892 (max occurrences)\n",
      "-- 12314088045 (total mers, non-unique)\n",
      "--  1457976283 (distinct mers, non-unique)\n",
      "--   385488671 (unique mers)\n",
      "-- For mhap overlapping, set repeat k-mer threshold to 126995.\n",
      "--\n",
      "-- Found 12699576716 16-mers; 1843464954 distinct and 385488671 unique.  Largest count 4763892.\n",
      "-- Finished stage 'cor-meryl', reset canuIteration.\n",
      "--\n",
      "-- OVERLAPPER (mhap) (correction)\n",
      "--\n",
      "-- Set corMhapSensitivity=high based on read coverage of 6.\n",
      "--\n",
      "-- PARAMETERS: hashes=768, minMatches=2, threshold=0.73\n",
      "--\n",
      "-- Given 32 GB, can fit 48000 reads per block.\n",
      "-- For 38 blocks, set stride to 9 blocks.\n",
      "-- Logging partitioning to 'correction/1-overlapper/partitioning.log'.\n",
      "-- Configured 37 mhap precompute jobs.\n",
      "-- Configured 91 mhap overlap jobs.\n",
      "-- Finished stage 'cor-mhapConfigure', reset canuIteration.\n",
      "--\n",
      "-- Running jobs.  First attempt out of 2.\n",
      "----------------------------------------\n",
      "-- Starting 'cormhap' concurrent execution on Wed Oct 18 23:16:40 2017 with 845.581 GB free disk space (37 processes; 2 concurrently)\n",
      "\n",
      "    cd correction/1-overlapper\n",
      "    ./precompute.sh 1 > ./precompute.000001.out 2>&1\n",
      "    ./precompute.sh 2 > ./precompute.000002.out 2>&1\n",
      "    ./precompute.sh 3 > ./precompute.000003.out 2>&1\n",
      "    ./precompute.sh 4 > ./precompute.000004.out 2>&1\n",
      "    ./precompute.sh 5 > ./precompute.000005.out 2>&1\n",
      "    ./precompute.sh 6 > ./precompute.000006.out 2>&1\n",
      "    ./precompute.sh 7 > ./precompute.000007.out 2>&1\n",
      "    ./precompute.sh 8 > ./precompute.000008.out 2>&1\n",
      "    ./precompute.sh 9 > ./precompute.000009.out 2>&1\n",
      "    ./precompute.sh 10 > ./precompute.000010.out 2>&1\n",
      "    ./precompute.sh 11 > ./precompute.000011.out 2>&1\n",
      "    ./precompute.sh 12 > ./precompute.000012.out 2>&1\n",
      "    ./precompute.sh 13 > ./precompute.000013.out 2>&1\n",
      "    ./precompute.sh 14 > ./precompute.000014.out 2>&1\n",
      "    ./precompute.sh 15 > ./precompute.000015.out 2>&1\n",
      "    ./precompute.sh 16 > ./precompute.000016.out 2>&1\n",
      "    ./precompute.sh 17 > ./precompute.000017.out 2>&1\n",
      "    ./precompute.sh 18 > ./precompute.000018.out 2>&1\n",
      "    ./precompute.sh 19 > ./precompute.000019.out 2>&1\n",
      "    ./precompute.sh 20 > ./precompute.000020.out 2>&1\n",
      "    ./precompute.sh 21 > ./precompute.000021.out 2>&1\n",
      "    ./precompute.sh 22 > ./precompute.000022.out 2>&1\n",
      "    ./precompute.sh 23 > ./precompute.000023.out 2>&1\n",
      "    ./precompute.sh 24 > ./precompute.000024.out 2>&1\n",
      "    ./precompute.sh 25 > ./precompute.000025.out 2>&1\n",
      "    ./precompute.sh 26 > ./precompute.000026.out 2>&1\n",
      "    ./precompute.sh 27 > ./precompute.000027.out 2>&1\n",
      "    ./precompute.sh 28 > ./precompute.000028.out 2>&1\n",
      "    ./precompute.sh 29 > ./precompute.000029.out 2>&1\n",
      "    ./precompute.sh 30 > ./precompute.000030.out 2>&1\n",
      "    ./precompute.sh 31 > ./precompute.000031.out 2>&1\n",
      "    ./precompute.sh 32 > ./precompute.000032.out 2>&1\n",
      "    ./precompute.sh 33 > ./precompute.000033.out 2>&1\n",
      "    ./precompute.sh 34 > ./precompute.000034.out 2>&1\n",
      "    ./precompute.sh 35 > ./precompute.000035.out 2>&1\n",
      "    ./precompute.sh 36 > ./precompute.000036.out 2>&1\n",
      "    ./precompute.sh 37 > ./precompute.000037.out 2>&1\n",
      "\n",
      "-- Finished on Thu Oct 19 08:16:20 2017 (32380 seconds) with 796 GB free disk space\n",
      "----------------------------------------\n",
      "-- All 37 mhap precompute jobs finished successfully.\n",
      "-- Finished stage 'cor-mhapPrecomputeCheck', reset canuIteration.\n",
      "--\n",
      "-- Running jobs.  First attempt out of 2.\n",
      "----------------------------------------\n",
      "-- Starting 'cormhap' concurrent execution on Thu Oct 19 08:16:20 2017 with 796 GB free disk space (91 processes; 2 concurrently)\n",
      "\n",
      "    cd correction/1-overlapper\n",
      "    ./mhap.sh 1 > ./mhap.000001.out 2>&1\n",
      "    ./mhap.sh 2 > ./mhap.000002.out 2>&1\n",
      "    ./mhap.sh 3 > ./mhap.000003.out 2>&1\n",
      "    ./mhap.sh 4 > ./mhap.000004.out 2>&1\n",
      "    ./mhap.sh 5 > ./mhap.000005.out 2>&1\n",
      "    ./mhap.sh 6 > ./mhap.000006.out 2>&1\n",
      "    ./mhap.sh 7 > ./mhap.000007.out 2>&1\n",
      "    ./mhap.sh 8 > ./mhap.000008.out 2>&1\n",
      "    ./mhap.sh 9 > ./mhap.000009.out 2>&1\n",
      "    ./mhap.sh 10 > ./mhap.000010.out 2>&1\n",
      "    ./mhap.sh 11 > ./mhap.000011.out 2>&1\n",
      "    ./mhap.sh 12 > ./mhap.000012.out 2>&1\n",
      "    ./mhap.sh 13 > ./mhap.000013.out 2>&1\n",
      "    ./mhap.sh 14 > ./mhap.000014.out 2>&1\n",
      "    ./mhap.sh 15 > ./mhap.000015.out 2>&1\n",
      "    ./mhap.sh 16 > ./mhap.000016.out 2>&1\n",
      "    ./mhap.sh 17 > ./mhap.000017.out 2>&1\n",
      "    ./mhap.sh 18 > ./mhap.000018.out 2>&1\n",
      "    ./mhap.sh 19 > ./mhap.000019.out 2>&1\n",
      "    ./mhap.sh 20 > ./mhap.000020.out 2>&1\n",
      "    ./mhap.sh 21 > ./mhap.000021.out 2>&1\n",
      "    ./mhap.sh 22 > ./mhap.000022.out 2>&1\n",
      "    ./mhap.sh 23 > ./mhap.000023.out 2>&1\n",
      "    ./mhap.sh 24 > ./mhap.000024.out 2>&1\n",
      "    ./mhap.sh 25 > ./mhap.000025.out 2>&1\n",
      "    ./mhap.sh 26 > ./mhap.000026.out 2>&1\n",
      "    ./mhap.sh 27 > ./mhap.000027.out 2>&1\n",
      "    ./mhap.sh 28 > ./mhap.000028.out 2>&1\n",
      "    ./mhap.sh 29 > ./mhap.000029.out 2>&1\n",
      "    ./mhap.sh 30 > ./mhap.000030.out 2>&1\n",
      "    ./mhap.sh 31 > ./mhap.000031.out 2>&1\n",
      "    ./mhap.sh 32 > ./mhap.000032.out 2>&1\n",
      "    ./mhap.sh 33 > ./mhap.000033.out 2>&1\n",
      "    ./mhap.sh 34 > ./mhap.000034.out 2>&1\n",
      "    ./mhap.sh 35 > ./mhap.000035.out 2>&1\n",
      "    ./mhap.sh 36 > ./mhap.000036.out 2>&1\n",
      "    ./mhap.sh 37 > ./mhap.000037.out 2>&1\n",
      "    ./mhap.sh 38 > ./mhap.000038.out 2>&1\n",
      "    ./mhap.sh 39 > ./mhap.000039.out 2>&1\n",
      "    ./mhap.sh 40 > ./mhap.000040.out 2>&1\n",
      "    ./mhap.sh 41 > ./mhap.000041.out 2>&1\n",
      "    ./mhap.sh 42 > ./mhap.000042.out 2>&1\n",
      "    ./mhap.sh 43 > ./mhap.000043.out 2>&1\n",
      "    ./mhap.sh 44 > ./mhap.000044.out 2>&1\n",
      "    ./mhap.sh 45 > ./mhap.000045.out 2>&1\n",
      "    ./mhap.sh 46 > ./mhap.000046.out 2>&1\n",
      "    ./mhap.sh 47 > ./mhap.000047.out 2>&1\n",
      "    ./mhap.sh 48 > ./mhap.000048.out 2>&1\n",
      "    ./mhap.sh 49 > ./mhap.000049.out 2>&1\n",
      "    ./mhap.sh 50 > ./mhap.000050.out 2>&1\n",
      "    ./mhap.sh 51 > ./mhap.000051.out 2>&1\n",
      "    ./mhap.sh 52 > ./mhap.000052.out 2>&1\n",
      "    ./mhap.sh 53 > ./mhap.000053.out 2>&1\n",
      "    ./mhap.sh 54 > ./mhap.000054.out 2>&1\n",
      "    ./mhap.sh 55 > ./mhap.000055.out 2>&1\n",
      "    ./mhap.sh 56 > ./mhap.000056.out 2>&1\n",
      "    ./mhap.sh 57 > ./mhap.000057.out 2>&1\n",
      "    ./mhap.sh 58 > ./mhap.000058.out 2>&1\n",
      "    ./mhap.sh 59 > ./mhap.000059.out 2>&1\n",
      "    ./mhap.sh 60 > ./mhap.000060.out 2>&1\n",
      "    ./mhap.sh 61 > ./mhap.000061.out 2>&1\n",
      "    ./mhap.sh 62 > ./mhap.000062.out 2>&1\n",
      "    ./mhap.sh 63 > ./mhap.000063.out 2>&1\n",
      "    ./mhap.sh 64 > ./mhap.000064.out 2>&1\n",
      "    ./mhap.sh 65 > ./mhap.000065.out 2>&1\n",
      "    ./mhap.sh 66 > ./mhap.000066.out 2>&1\n",
      "    ./mhap.sh 67 > ./mhap.000067.out 2>&1\n",
      "    ./mhap.sh 68 > ./mhap.000068.out 2>&1\n",
      "    ./mhap.sh 69 > ./mhap.000069.out 2>&1\n",
      "    ./mhap.sh 70 > ./mhap.000070.out 2>&1\n",
      "    ./mhap.sh 71 > ./mhap.000071.out 2>&1\n",
      "    ./mhap.sh 72 > ./mhap.000072.out 2>&1\n",
      "    ./mhap.sh 73 > ./mhap.000073.out 2>&1\n",
      "    ./mhap.sh 74 > ./mhap.000074.out 2>&1\n",
      "    ./mhap.sh 75 > ./mhap.000075.out 2>&1\n",
      "    ./mhap.sh 76 > ./mhap.000076.out 2>&1\n",
      "    ./mhap.sh 77 > ./mhap.000077.out 2>&1\n",
      "    ./mhap.sh 78 > ./mhap.000078.out 2>&1\n",
      "    ./mhap.sh 79 > ./mhap.000079.out 2>&1\n",
      "    ./mhap.sh 80 > ./mhap.000080.out 2>&1\n",
      "    ./mhap.sh 81 > ./mhap.000081.out 2>&1\n",
      "    ./mhap.sh 82 > ./mhap.000082.out 2>&1\n",
      "    ./mhap.sh 83 > ./mhap.000083.out 2>&1\n",
      "    ./mhap.sh 84 > ./mhap.000084.out 2>&1\n",
      "    ./mhap.sh 85 > ./mhap.000085.out 2>&1\n",
      "    ./mhap.sh 86 > ./mhap.000086.out 2>&1\n",
      "    ./mhap.sh 87 > ./mhap.000087.out 2>&1\n",
      "    ./mhap.sh 88 > ./mhap.000088.out 2>&1\n",
      "    ./mhap.sh 89 > ./mhap.000089.out 2>&1\n",
      "    ./mhap.sh 90 > ./mhap.000090.out 2>&1\n",
      "    ./mhap.sh 91 > ./mhap.000091.out 2>&1\n",
      "\n",
      "-- Finished on Fri Oct 20 03:15:18 2017 (68338 seconds) with 709.861 GB free disk space\n",
      "----------------------------------------\n",
      "-- Found 91 mhap overlap output files.\n",
      "-- Finished stage 'cor-mhapCheck', reset canuIteration.\n",
      "----------------------------------------\n",
      "-- Starting command on Fri Oct 20 03:15:20 2017 with 709.861 GB free disk space\n",
      "\n",
      "    cd correction\n",
      "    /usr/local/bioinformatics/canu-1.6/Linux-amd64/bin/ovStoreBuild \\\n",
      "     -O ./20171018_oly_pacbio.ovlStore.BUILDING \\\n",
      "     -G ./20171018_oly_pacbio.gkpStore \\\n",
      "     -M 4-32 \\\n",
      "     -L ./1-overlapper/ovljob.files \\\n",
      "     > ./20171018_oly_pacbio.ovlStore.err 2>&1\n",
      "\n",
      "-- Finished on Fri Oct 20 06:04:35 2017 (10155 seconds) with 540.286 GB free disk space\n",
      "----------------------------------------\n",
      "-- Checking store.\n",
      "----------------------------------------\n",
      "-- Starting command on Fri Oct 20 06:04:35 2017 with 540.286 GB free disk space\n",
      "\n",
      "    cd correction\n",
      "    /usr/local/bioinformatics/canu-1.6/Linux-amd64/bin/ovStoreDump \\\n",
      "     -G ./20171018_oly_pacbio.gkpStore \\\n",
      "     -O ./20171018_oly_pacbio.ovlStore \\\n",
      "     -d -counts \\\n",
      "     > ./20171018_oly_pacbio.ovlStore/counts.dat 2> ./20171018_oly_pacbio.ovlStore/counts.err\n",
      "\n",
      "-- Finished on Fri Oct 20 06:04:38 2017 (3 seconds) with 540.267 GB free disk space\n",
      "----------------------------------------\n",
      "--\n",
      "-- Overlap store 'correction/20171018_oly_pacbio.ovlStore' successfully constructed.\n",
      "-- Found 9101620838 overlaps for 1737978 reads; 874 reads have no overlaps.\n",
      "--\n",
      "FILE correction/1-overlapper/results/000001.ovb\n",
      "FILE correction/1-overlapper/results/000002.ovb\n",
      "FILE correction/1-overlapper/results/000003.ovb\n",
      "FILE correction/1-overlapper/results/000004.ovb\n",
      "FILE correction/1-overlapper/results/000005.ovb\n",
      "FILE correction/1-overlapper/results/000006.ovb\n",
      "FILE correction/1-overlapper/results/000007.ovb\n",
      "FILE correction/1-overlapper/results/000008.ovb\n",
      "FILE correction/1-overlapper/results/000009.ovb\n",
      "FILE correction/1-overlapper/results/000010.ovb\n",
      "FILE correction/1-overlapper/results/000011.ovb\n",
      "FILE correction/1-overlapper/results/000012.ovb\n",
      "FILE correction/1-overlapper/results/000013.ovb\n",
      "FILE correction/1-overlapper/results/000014.ovb\n",
      "FILE correction/1-overlapper/results/000015.ovb\n",
      "FILE correction/1-overlapper/results/000016.ovb\n",
      "FILE correction/1-overlapper/results/000017.ovb\n",
      "FILE correction/1-overlapper/results/000018.ovb\n",
      "FILE correction/1-overlapper/results/000019.ovb\n",
      "FILE correction/1-overlapper/results/000020.ovb\n",
      "FILE correction/1-overlapper/results/000021.ovb\n",
      "FILE correction/1-overlapper/results/000022.ovb\n",
      "FILE correction/1-overlapper/results/000023.ovb\n",
      "FILE correction/1-overlapper/results/000024.ovb\n",
      "FILE correction/1-overlapper/results/000025.ovb\n",
      "FILE correction/1-overlapper/results/000026.ovb\n",
      "FILE correction/1-overlapper/results/000027.ovb\n",
      "FILE correction/1-overlapper/results/000028.ovb\n",
      "FILE correction/1-overlapper/results/000029.ovb\n",
      "FILE correction/1-overlapper/results/000030.ovb\n",
      "FILE correction/1-overlapper/results/000031.ovb\n",
      "FILE correction/1-overlapper/results/000032.ovb\n",
      "FILE correction/1-overlapper/results/000033.ovb\n",
      "FILE correction/1-overlapper/results/000034.ovb\n",
      "FILE correction/1-overlapper/results/000035.ovb\n",
      "FILE correction/1-overlapper/results/000036.ovb\n",
      "FILE correction/1-overlapper/results/000037.ovb\n",
      "FILE correction/1-overlapper/results/000038.ovb\n",
      "FILE correction/1-overlapper/results/000039.ovb\n",
      "FILE correction/1-overlapper/results/000040.ovb\n",
      "FILE correction/1-overlapper/results/000041.ovb\n",
      "FILE correction/1-overlapper/results/000042.ovb\n",
      "FILE correction/1-overlapper/results/000043.ovb\n",
      "FILE correction/1-overlapper/results/000044.ovb\n",
      "FILE correction/1-overlapper/results/000045.ovb\n",
      "FILE correction/1-overlapper/results/000046.ovb\n",
      "FILE correction/1-overlapper/results/000047.ovb\n",
      "FILE correction/1-overlapper/results/000048.ovb\n",
      "FILE correction/1-overlapper/results/000049.ovb\n",
      "FILE correction/1-overlapper/results/000050.ovb\n",
      "FILE correction/1-overlapper/results/000051.ovb\n",
      "FILE correction/1-overlapper/results/000052.ovb\n",
      "FILE correction/1-overlapper/results/000053.ovb\n",
      "FILE correction/1-overlapper/results/000054.ovb\n",
      "FILE correction/1-overlapper/results/000055.ovb\n",
      "FILE correction/1-overlapper/results/000056.ovb\n",
      "FILE correction/1-overlapper/results/000057.ovb\n",
      "FILE correction/1-overlapper/results/000058.ovb\n",
      "FILE correction/1-overlapper/results/000059.ovb\n",
      "FILE correction/1-overlapper/results/000060.ovb\n",
      "FILE correction/1-overlapper/results/000061.ovb\n",
      "FILE correction/1-overlapper/results/000062.ovb\n",
      "FILE correction/1-overlapper/results/000063.ovb\n",
      "FILE correction/1-overlapper/results/000064.ovb\n",
      "FILE correction/1-overlapper/results/000065.ovb\n",
      "FILE correction/1-overlapper/results/000066.ovb\n",
      "FILE correction/1-overlapper/results/000067.ovb\n",
      "FILE correction/1-overlapper/results/000068.ovb\n",
      "FILE correction/1-overlapper/results/000069.ovb\n",
      "FILE correction/1-overlapper/results/000070.ovb\n",
      "FILE correction/1-overlapper/results/000071.ovb\n",
      "FILE correction/1-overlapper/results/000072.ovb\n",
      "FILE correction/1-overlapper/results/000073.ovb\n",
      "FILE correction/1-overlapper/results/000074.ovb\n",
      "FILE correction/1-overlapper/results/000075.ovb\n",
      "FILE correction/1-overlapper/results/000076.ovb\n",
      "FILE correction/1-overlapper/results/000077.ovb\n",
      "FILE correction/1-overlapper/results/000078.ovb\n",
      "FILE correction/1-overlapper/results/000079.ovb\n",
      "FILE correction/1-overlapper/results/000080.ovb\n",
      "FILE correction/1-overlapper/results/000081.ovb\n",
      "FILE correction/1-overlapper/results/000082.ovb\n",
      "FILE correction/1-overlapper/results/000083.ovb\n",
      "FILE correction/1-overlapper/results/000084.ovb\n",
      "FILE correction/1-overlapper/results/000085.ovb\n",
      "FILE correction/1-overlapper/results/000086.ovb\n",
      "FILE correction/1-overlapper/results/000087.ovb\n",
      "FILE correction/1-overlapper/results/000088.ovb\n",
      "FILE correction/1-overlapper/results/000089.ovb\n",
      "FILE correction/1-overlapper/results/000090.ovb\n",
      "FILE correction/1-overlapper/results/000091.ovb\n",
      "FILE correction/1-overlapper/results/000001.counts\n",
      "FILE correction/1-overlapper/results/000002.counts\n",
      "FILE correction/1-overlapper/results/000003.counts\n",
      "FILE correction/1-overlapper/results/000004.counts\n",
      "FILE correction/1-overlapper/results/000005.counts\n",
      "FILE correction/1-overlapper/results/000006.counts\n",
      "FILE correction/1-overlapper/results/000007.counts\n",
      "FILE correction/1-overlapper/results/000008.counts\n",
      "FILE correction/1-overlapper/results/000009.counts\n",
      "FILE correction/1-overlapper/results/000010.counts\n",
      "FILE correction/1-overlapper/results/000011.counts\n",
      "FILE correction/1-overlapper/results/000012.counts\n",
      "FILE correction/1-overlapper/results/000013.counts\n",
      "FILE correction/1-overlapper/results/000014.counts\n",
      "FILE correction/1-overlapper/results/000015.counts\n",
      "FILE correction/1-overlapper/results/000016.counts\n",
      "FILE correction/1-overlapper/results/000017.counts\n",
      "FILE correction/1-overlapper/results/000018.counts\n",
      "FILE correction/1-overlapper/results/000019.counts\n",
      "FILE correction/1-overlapper/results/000020.counts\n",
      "FILE correction/1-overlapper/results/000021.counts\n",
      "FILE correction/1-overlapper/results/000022.counts\n",
      "FILE correction/1-overlapper/results/000023.counts\n",
      "FILE correction/1-overlapper/results/000024.counts\n",
      "FILE correction/1-overlapper/results/000025.counts\n",
      "FILE correction/1-overlapper/results/000026.counts\n",
      "FILE correction/1-overlapper/results/000027.counts\n",
      "FILE correction/1-overlapper/results/000028.counts\n",
      "FILE correction/1-overlapper/results/000029.counts\n",
      "FILE correction/1-overlapper/results/000030.counts\n",
      "FILE correction/1-overlapper/results/000031.counts\n",
      "FILE correction/1-overlapper/results/000032.counts\n",
      "FILE correction/1-overlapper/results/000033.counts\n",
      "FILE correction/1-overlapper/results/000034.counts\n",
      "FILE correction/1-overlapper/results/000035.counts\n",
      "FILE correction/1-overlapper/results/000036.counts\n",
      "FILE correction/1-overlapper/results/000037.counts\n",
      "FILE correction/1-overlapper/results/000038.counts\n",
      "FILE correction/1-overlapper/results/000039.counts\n",
      "FILE correction/1-overlapper/results/000040.counts\n",
      "FILE correction/1-overlapper/results/000041.counts\n",
      "FILE correction/1-overlapper/results/000042.counts\n",
      "FILE correction/1-overlapper/results/000043.counts\n",
      "FILE correction/1-overlapper/results/000044.counts\n",
      "FILE correction/1-overlapper/results/000045.counts\n",
      "FILE correction/1-overlapper/results/000046.counts\n",
      "FILE correction/1-overlapper/results/000047.counts\n",
      "FILE correction/1-overlapper/results/000048.counts\n",
      "FILE correction/1-overlapper/results/000049.counts\n",
      "FILE correction/1-overlapper/results/000050.counts\n",
      "FILE correction/1-overlapper/results/000051.counts\n",
      "FILE correction/1-overlapper/results/000052.counts\n",
      "FILE correction/1-overlapper/results/000053.counts\n",
      "FILE correction/1-overlapper/results/000054.counts\n",
      "FILE correction/1-overlapper/results/000055.counts\n",
      "FILE correction/1-overlapper/results/000056.counts\n",
      "FILE correction/1-overlapper/results/000057.counts\n",
      "FILE correction/1-overlapper/results/000058.counts\n",
      "FILE correction/1-overlapper/results/000059.counts\n",
      "FILE correction/1-overlapper/results/000060.counts\n",
      "FILE correction/1-overlapper/results/000061.counts\n",
      "FILE correction/1-overlapper/results/000062.counts\n",
      "FILE correction/1-overlapper/results/000063.counts\n",
      "FILE correction/1-overlapper/results/000064.counts\n",
      "FILE correction/1-overlapper/results/000065.counts\n",
      "FILE correction/1-overlapper/results/000066.counts\n",
      "FILE correction/1-overlapper/results/000067.counts\n",
      "FILE correction/1-overlapper/results/000068.counts\n",
      "FILE correction/1-overlapper/results/000069.counts\n",
      "FILE correction/1-overlapper/results/000070.counts\n",
      "FILE correction/1-overlapper/results/000071.counts\n",
      "FILE correction/1-overlapper/results/000072.counts\n",
      "FILE correction/1-overlapper/results/000073.counts\n",
      "FILE correction/1-overlapper/results/000074.counts\n",
      "FILE correction/1-overlapper/results/000075.counts\n",
      "FILE correction/1-overlapper/results/000076.counts\n",
      "FILE correction/1-overlapper/results/000077.counts\n",
      "FILE correction/1-overlapper/results/000078.counts\n",
      "FILE correction/1-overlapper/results/000079.counts\n",
      "FILE correction/1-overlapper/results/000080.counts\n",
      "FILE correction/1-overlapper/results/000081.counts\n",
      "FILE correction/1-overlapper/results/000082.counts\n",
      "FILE correction/1-overlapper/results/000083.counts\n",
      "FILE correction/1-overlapper/results/000084.counts\n",
      "FILE correction/1-overlapper/results/000085.counts\n",
      "FILE correction/1-overlapper/results/000086.counts\n",
      "FILE correction/1-overlapper/results/000087.counts\n",
      "FILE correction/1-overlapper/results/000088.counts\n",
      "FILE correction/1-overlapper/results/000089.counts\n",
      "FILE correction/1-overlapper/results/000090.counts\n",
      "FILE correction/1-overlapper/results/000091.counts\n",
      "DIRECTORY correction/1-overlapper/queries\n",
      "DIRECTORY correction/1-overlapper/queries/000004\n",
      "FILE correction/1-overlapper/queries/000004/000034.dat\n",
      "FILE correction/1-overlapper/queries/000004/000035.dat\n",
      "FILE correction/1-overlapper/queries/000004/000036.dat\n",
      "FILE correction/1-overlapper/queries/000004/000032.dat\n",
      "FILE correction/1-overlapper/queries/000004/000030.dat\n",
      "FILE correction/1-overlapper/queries/000004/000029.dat\n",
      "FILE correction/1-overlapper/queries/000004/000031.dat\n",
      "FILE correction/1-overlapper/queries/000004/000033.dat\n",
      "FILE correction/1-overlapper/queries/000004/000037.dat\n",
      "DIRECTORY correction/1-overlapper/queries/000059\n",
      "FILE correction/1-overlapper/queries/000059/000034.dat\n",
      "FILE correction/1-overlapper/queries/000059/000035.dat\n",
      "FILE correction/1-overlapper/queries/000059/000028.dat\n",
      "FILE correction/1-overlapper/queries/000059/000032.dat\n",
      "FILE correction/1-overlapper/queries/000059/000030.dat\n",
      "FILE correction/1-overlapper/queries/000059/000029.dat\n",
      "FILE correction/1-overlapper/queries/000059/000031.dat\n",
      "FILE correction/1-overlapper/queries/000059/000033.dat\n",
      "FILE correction/1-overlapper/queries/000059/000027.dat\n",
      "DIRECTORY correction/1-overlapper/queries/000022\n",
      "FILE correction/1-overlapper/queries/000022/000020.dat\n",
      "FILE correction/1-overlapper/queries/000022/000018.dat\n",
      "FILE correction/1-overlapper/queries/000022/000024.dat\n",
      "FILE correction/1-overlapper/queries/000022/000023.dat\n",
      "FILE correction/1-overlapper/queries/000022/000017.dat\n",
      "FILE correction/1-overlapper/queries/000022/000016.dat\n",
      "FILE correction/1-overlapper/queries/000022/000021.dat\n",
      "FILE correction/1-overlapper/queries/000022/000022.dat\n",
      "FILE correction/1-overlapper/queries/000022/000019.dat\n",
      "DIRECTORY correction/1-overlapper/queries/000015\n",
      "FILE correction/1-overlapper/queries/000015/000028.dat\n",
      "FILE correction/1-overlapper/queries/000015/000024.dat\n",
      "FILE correction/1-overlapper/queries/000015/000023.dat\n",
      "FILE correction/1-overlapper/queries/000015/000030.dat\n",
      "FILE correction/1-overlapper/queries/000015/000026.dat\n",
      "FILE correction/1-overlapper/queries/000015/000029.dat\n",
      "FILE correction/1-overlapper/queries/000015/000025.dat\n",
      "FILE correction/1-overlapper/queries/000015/000031.dat\n",
      "FILE correction/1-overlapper/queries/000015/000027.dat\n",
      "DIRECTORY correction/1-overlapper/queries/000090\n",
      "FILE correction/1-overlapper/queries/000090/000037.dat\n",
      "DIRECTORY correction/1-overlapper/queries/000013\n",
      "FILE correction/1-overlapper/queries/000013/000006.dat\n",
      "FILE correction/1-overlapper/queries/000013/000013.dat\n",
      "FILE correction/1-overlapper/queries/000013/000010.dat\n",
      "FILE correction/1-overlapper/queries/000013/000012.dat\n",
      "FILE correction/1-overlapper/queries/000013/000009.dat\n",
      "FILE correction/1-overlapper/queries/000013/000008.dat\n",
      "FILE correction/1-overlapper/queries/000013/000005.dat\n",
      "FILE correction/1-overlapper/queries/000013/000007.dat\n",
      "FILE correction/1-overlapper/queries/000013/000011.dat\n",
      "DIRECTORY correction/1-overlapper/queries/000080\n",
      "FILE correction/1-overlapper/queries/000080/000034.dat\n",
      "FILE correction/1-overlapper/queries/000080/000035.dat\n",
      "FILE correction/1-overlapper/queries/000080/000028.dat\n",
      "FILE correction/1-overlapper/queries/000080/000036.dat\n",
      "FILE correction/1-overlapper/queries/000080/000032.dat\n",
      "FILE correction/1-overlapper/queries/000080/000030.dat\n",
      "FILE correction/1-overlapper/queries/000080/000029.dat\n",
      "FILE correction/1-overlapper/queries/000080/000031.dat\n",
      "FILE correction/1-overlapper/queries/000080/000033.dat\n",
      "DIRECTORY correction/1-overlapper/queries/000058\n",
      "FILE correction/1-overlapper/queries/000058/000020.dat\n",
      "FILE correction/1-overlapper/queries/000058/000018.dat\n",
      "FILE correction/1-overlapper/queries/000058/000024.dat\n",
      "FILE correction/1-overlapper/queries/000058/000023.dat\n",
      "FILE correction/1-overlapper/queries/000058/000026.dat\n",
      "FILE correction/1-overlapper/queries/000058/000025.dat\n",
      "FILE correction/1-overlapper/queries/000058/000021.dat\n",
      "FILE correction/1-overlapper/queries/000058/000022.dat\n",
      "FILE correction/1-overlapper/queries/000058/000019.dat\n",
      "DIRECTORY correction/1-overlapper/queries/000057\n",
      "FILE correction/1-overlapper/queries/000057/000035.dat\n",
      "FILE correction/1-overlapper/queries/000057/000036.dat\n",
      "FILE correction/1-overlapper/queries/000057/000037.dat\n",
      "DIRECTORY correction/1-overlapper/queries/000027\n",
      "FILE correction/1-overlapper/queries/000027/000034.dat\n",
      "FILE correction/1-overlapper/queries/000027/000028.dat\n",
      "FILE correction/1-overlapper/queries/000027/000032.dat\n",
      "FILE correction/1-overlapper/queries/000027/000030.dat\n",
      "FILE correction/1-overlapper/queries/000027/000026.dat\n",
      "FILE correction/1-overlapper/queries/000027/000029.dat\n",
      "FILE correction/1-overlapper/queries/000027/000031.dat\n",
      "FILE correction/1-overlapper/queries/000027/000033.dat\n",
      "FILE correction/1-overlapper/queries/000027/000027.dat\n",
      "DIRECTORY correction/1-overlapper/queries/000084\n",
      "FILE correction/1-overlapper/queries/000084/000034.dat\n",
      "FILE correction/1-overlapper/queries/000084/000035.dat\n",
      "FILE correction/1-overlapper/queries/000084/000036.dat\n",
      "FILE correction/1-overlapper/queries/000084/000032.dat\n",
      "FILE correction/1-overlapper/queries/000084/000031.dat\n",
      "FILE correction/1-overlapper/queries/000084/000033.dat\n",
      "FILE correction/1-overlapper/queries/000084/000037.dat\n",
      "DIRECTORY correction/1-overlapper/queries/000070\n",
      "FILE correction/1-overlapper/queries/000070/000028.dat\n",
      "FILE correction/1-overlapper/queries/000070/000024.dat\n",
      "FILE correction/1-overlapper/queries/000070/000023.dat\n",
      "FILE correction/1-overlapper/queries/000070/000030.dat\n",
      "FILE correction/1-overlapper/queries/000070/000026.dat\n",
      "FILE correction/1-overlapper/queries/000070/000029.dat\n",
      "FILE correction/1-overlapper/queries/000070/000025.dat\n",
      "FILE correction/1-overlapper/queries/000070/000031.dat\n",
      "FILE correction/1-overlapper/queries/000070/000027.dat\n",
      "DIRECTORY correction/1-overlapper/queries/000068\n",
      "FILE correction/1-overlapper/queries/000068/000028.dat\n",
      "FILE correction/1-overlapper/queries/000068/000024.dat\n",
      "FILE correction/1-overlapper/queries/000068/000023.dat\n",
      "FILE correction/1-overlapper/queries/000068/000030.dat\n",
      "FILE correction/1-overlapper/queries/000068/000026.dat\n",
      "FILE correction/1-overlapper/queries/000068/000029.dat\n",
      "FILE correction/1-overlapper/queries/000068/000025.dat\n",
      "FILE correction/1-overlapper/queries/000068/000022.dat\n",
      "FILE correction/1-overlapper/queries/000068/000027.dat\n",
      "DIRECTORY correction/1-overlapper/queries/000071\n",
      "FILE correction/1-overlapper/queries/000071/000034.dat\n",
      "FILE correction/1-overlapper/queries/000071/000035.dat\n",
      "FILE correction/1-overlapper/queries/000071/000036.dat\n",
      "FILE correction/1-overlapper/queries/000071/000032.dat\n",
      "FILE correction/1-overlapper/queries/000071/000033.dat\n",
      "FILE correction/1-overlapper/queries/000071/000037.dat\n",
      "DIRECTORY correction/1-overlapper/queries/000024\n",
      "FILE correction/1-overlapper/queries/000024/000034.dat\n",
      "FILE correction/1-overlapper/queries/000024/000035.dat\n",
      "FILE correction/1-overlapper/queries/000024/000036.dat\n",
      "FILE correction/1-overlapper/queries/000024/000037.dat\n",
      "DIRECTORY correction/1-overlapper/queries/000033\n",
      "FILE correction/1-overlapper/queries/000033/000014.dat\n",
      "FILE correction/1-overlapper/queries/000033/000013.dat\n",
      "FILE correction/1-overlapper/queries/000033/000018.dat\n",
      "FILE correction/1-overlapper/queries/000033/000010.dat\n",
      "FILE correction/1-overlapper/queries/000033/000017.dat\n",
      "FILE correction/1-overlapper/queries/000033/000012.dat\n",
      "FILE correction/1-overlapper/queries/000033/000015.dat\n",
      "FILE correction/1-overlapper/queries/000033/000016.dat\n",
      "FILE correction/1-overlapper/queries/000033/000011.dat\n",
      "DIRECTORY correction/1-overlapper/queries/000019\n",
      "FILE correction/1-overlapper/queries/000019/000028.dat\n",
      "FILE correction/1-overlapper/queries/000019/000024.dat\n",
      "FILE correction/1-overlapper/queries/000019/000032.dat\n",
      "FILE correction/1-overlapper/queries/000019/000030.dat\n",
      "FILE correction/1-overlapper/queries/000019/000026.dat\n",
      "FILE correction/1-overlapper/queries/000019/000029.dat\n",
      "FILE correction/1-overlapper/queries/000019/000025.dat\n",
      "FILE correction/1-overlapper/queries/000019/000031.dat\n",
      "FILE correction/1-overlapper/queries/000019/000027.dat\n",
      "DIRECTORY correction/1-overlapper/queries/000040\n",
      "FILE correction/1-overlapper/queries/000040/000020.dat\n",
      "FILE correction/1-overlapper/queries/000040/000014.dat\n",
      "FILE correction/1-overlapper/queries/000040/000013.dat\n",
      "FILE correction/1-overlapper/queries/000040/000018.dat\n",
      "FILE correction/1-overlapper/queries/000040/000017.dat\n",
      "FILE correction/1-overlapper/queries/000040/000012.dat\n",
      "FILE correction/1-overlapper/queries/000040/000015.dat\n",
      "FILE correction/1-overlapper/queries/000040/000016.dat\n",
      "FILE correction/1-overlapper/queries/000040/000019.dat\n",
      "DIRECTORY correction/1-overlapper/queries/000028\n",
      "FILE correction/1-overlapper/queries/000028/000035.dat\n",
      "FILE correction/1-overlapper/queries/000028/000036.dat\n",
      "FILE correction/1-overlapper/queries/000028/000037.dat\n",
      "DIRECTORY correction/1-overlapper/queries/000075\n",
      "FILE correction/1-overlapper/queries/000075/000034.dat\n",
      "FILE correction/1-overlapper/queries/000075/000035.dat\n",
      "FILE correction/1-overlapper/queries/000075/000036.dat\n",
      "FILE correction/1-overlapper/queries/000075/000037.dat\n",
      "DIRECTORY correction/1-overlapper/queries/000060\n",
      "FILE correction/1-overlapper/queries/000060/000036.dat\n",
      "FILE correction/1-overlapper/queries/000060/000037.dat\n",
      "DIRECTORY correction/1-overlapper/queries/000047\n",
      "FILE correction/1-overlapper/queries/000047/000028.dat\n",
      "FILE correction/1-overlapper/queries/000047/000024.dat\n",
      "FILE correction/1-overlapper/queries/000047/000023.dat\n",
      "FILE correction/1-overlapper/queries/000047/000030.dat\n",
      "FILE correction/1-overlapper/queries/000047/000026.dat\n",
      "FILE correction/1-overlapper/queries/000047/000029.dat\n",
      "FILE correction/1-overlapper/queries/000047/000025.dat\n",
      "FILE correction/1-overlapper/queries/000047/000031.dat\n",
      "FILE correction/1-overlapper/queries/000047/000027.dat\n",
      "DIRECTORY correction/1-overlapper/queries/000056\n",
      "FILE correction/1-overlapper/queries/000056/000034.dat\n",
      "FILE correction/1-overlapper/queries/000056/000028.dat\n",
      "FILE correction/1-overlapper/queries/000056/000032.dat\n",
      "FILE correction/1-overlapper/queries/000056/000030.dat\n",
      "FILE correction/1-overlapper/queries/000056/000026.dat\n",
      "FILE correction/1-overlapper/queries/000056/000029.dat\n",
      "FILE correction/1-overlapper/queries/000056/000031.dat\n",
      "FILE correction/1-overlapper/queries/000056/000033.dat\n",
      "FILE correction/1-overlapper/queries/000056/000027.dat\n",
      "DIRECTORY correction/1-overlapper/queries/000064\n",
      "FILE correction/1-overlapper/queries/000064/000020.dat\n",
      "FILE correction/1-overlapper/queries/000064/000028.dat\n",
      "FILE correction/1-overlapper/queries/000064/000024.dat\n",
      "FILE correction/1-overlapper/queries/000064/000023.dat\n",
      "FILE correction/1-overlapper/queries/000064/000026.dat\n",
      "FILE correction/1-overlapper/queries/000064/000025.dat\n",
      "FILE correction/1-overlapper/queries/000064/000021.dat\n",
      "FILE correction/1-overlapper/queries/000064/000022.dat\n",
      "FILE correction/1-overlapper/queries/000064/000027.dat\n",
      "DIRECTORY correction/1-overlapper/queries/000009\n",
      "FILE correction/1-overlapper/queries/000009/000006.dat\n",
      "FILE correction/1-overlapper/queries/000009/000010.dat\n",
      "FILE correction/1-overlapper/queries/000009/000012.dat\n",
      "FILE correction/1-overlapper/queries/000009/000004.dat\n",
      "FILE correction/1-overlapper/queries/000009/000009.dat\n",
      "FILE correction/1-overlapper/queries/000009/000008.dat\n",
      "FILE correction/1-overlapper/queries/000009/000005.dat\n",
      "FILE correction/1-overlapper/queries/000009/000007.dat\n",
      "FILE correction/1-overlapper/queries/000009/000011.dat\n",
      "DIRECTORY correction/1-overlapper/queries/000003\n",
      "FILE correction/1-overlapper/queries/000003/000020.dat\n",
      "FILE correction/1-overlapper/queries/000003/000028.dat\n",
      "FILE correction/1-overlapper/queries/000003/000024.dat\n",
      "FILE correction/1-overlapper/queries/000003/000023.dat\n",
      "FILE correction/1-overlapper/queries/000003/000026.dat\n",
      "FILE correction/1-overlapper/queries/000003/000025.dat\n",
      "FILE correction/1-overlapper/queries/000003/000021.dat\n",
      "FILE correction/1-overlapper/queries/000003/000022.dat\n",
      "FILE correction/1-overlapper/queries/000003/000027.dat\n",
      "DIRECTORY correction/1-overlapper/queries/000087\n",
      "FILE correction/1-overlapper/queries/000087/000034.dat\n",
      "FILE correction/1-overlapper/queries/000087/000035.dat\n",
      "FILE correction/1-overlapper/queries/000087/000036.dat\n",
      "FILE correction/1-overlapper/queries/000087/000037.dat\n",
      "DIRECTORY correction/1-overlapper/queries/000016\n",
      "FILE correction/1-overlapper/queries/000016/000034.dat\n",
      "FILE correction/1-overlapper/queries/000016/000035.dat\n",
      "FILE correction/1-overlapper/queries/000016/000036.dat\n",
      "FILE correction/1-overlapper/queries/000016/000032.dat\n",
      "FILE correction/1-overlapper/queries/000016/000033.dat\n",
      "FILE correction/1-overlapper/queries/000016/000037.dat\n",
      "DIRECTORY correction/1-overlapper/queries/000036\n",
      "FILE correction/1-overlapper/queries/000036/000037.dat\n",
      "DIRECTORY correction/1-overlapper/queries/000010\n",
      "FILE correction/1-overlapper/queries/000010/000020.dat\n",
      "FILE correction/1-overlapper/queries/000010/000014.dat\n",
      "FILE correction/1-overlapper/queries/000010/000013.dat\n",
      "FILE correction/1-overlapper/queries/000010/000018.dat\n",
      "FILE correction/1-overlapper/queries/000010/000017.dat\n",
      "FILE correction/1-overlapper/queries/000010/000015.dat\n",
      "FILE correction/1-overlapper/queries/000010/000016.dat\n",
      "FILE correction/1-overlapper/queries/000010/000021.dat\n",
      "FILE correction/1-overlapper/queries/000010/000019.dat\n",
      "DIRECTORY correction/1-overlapper/queries/000020\n",
      "FILE correction/1-overlapper/queries/000020/000034.dat\n",
      "FILE correction/1-overlapper/queries/000020/000035.dat\n",
      "FILE correction/1-overlapper/queries/000020/000036.dat\n",
      "FILE correction/1-overlapper/queries/000020/000033.dat\n",
      "FILE correction/1-overlapper/queries/000020/000037.dat\n",
      "DIRECTORY correction/1-overlapper/queries/000086\n",
      "FILE correction/1-overlapper/queries/000086/000034.dat\n",
      "FILE correction/1-overlapper/queries/000086/000035.dat\n",
      "FILE correction/1-overlapper/queries/000086/000036.dat\n",
      "FILE correction/1-overlapper/queries/000086/000033.dat\n",
      "FILE correction/1-overlapper/queries/000086/000037.dat\n",
      "DIRECTORY correction/1-overlapper/queries/000025\n",
      "FILE correction/1-overlapper/queries/000025/000014.dat\n",
      "FILE correction/1-overlapper/queries/000025/000013.dat\n",
      "FILE correction/1-overlapper/queries/000025/000010.dat\n",
      "FILE correction/1-overlapper/queries/000025/000012.dat\n",
      "FILE correction/1-overlapper/queries/000025/000015.dat\n",
      "FILE correction/1-overlapper/queries/000025/000016.dat\n",
      "FILE correction/1-overlapper/queries/000025/000009.dat\n",
      "FILE correction/1-overlapper/queries/000025/000008.dat\n",
      "FILE correction/1-overlapper/queries/000025/000011.dat\n",
      "DIRECTORY correction/1-overlapper/queries/000049\n",
      "FILE correction/1-overlapper/queries/000049/000020.dat\n",
      "FILE correction/1-overlapper/queries/000049/000018.dat\n",
      "FILE correction/1-overlapper/queries/000049/000023.dat\n",
      "FILE correction/1-overlapper/queries/000049/000017.dat\n",
      "FILE correction/1-overlapper/queries/000049/000015.dat\n",
      "FILE correction/1-overlapper/queries/000049/000016.dat\n",
      "FILE correction/1-overlapper/queries/000049/000021.dat\n",
      "FILE correction/1-overlapper/queries/000049/000022.dat\n",
      "FILE correction/1-overlapper/queries/000049/000019.dat\n",
      "DIRECTORY correction/1-overlapper/queries/000046\n",
      "FILE correction/1-overlapper/queries/000046/000020.dat\n",
      "FILE correction/1-overlapper/queries/000046/000014.dat\n",
      "FILE correction/1-overlapper/queries/000046/000018.dat\n",
      "FILE correction/1-overlapper/queries/000046/000017.dat\n",
      "FILE correction/1-overlapper/queries/000046/000015.dat\n",
      "FILE correction/1-overlapper/queries/000046/000016.dat\n",
      "FILE correction/1-overlapper/queries/000046/000021.dat\n",
      "FILE correction/1-overlapper/queries/000046/000022.dat\n",
      "FILE correction/1-overlapper/queries/000046/000019.dat\n",
      "DIRECTORY correction/1-overlapper/queries/000076\n",
      "FILE correction/1-overlapper/queries/000076/000034.dat\n",
      "FILE correction/1-overlapper/queries/000076/000028.dat\n",
      "FILE correction/1-overlapper/queries/000076/000032.dat\n",
      "FILE correction/1-overlapper/queries/000076/000030.dat\n",
      "FILE correction/1-overlapper/queries/000076/000026.dat\n",
      "FILE correction/1-overlapper/queries/000076/000029.dat\n",
      "FILE correction/1-overlapper/queries/000076/000031.dat\n",
      "FILE correction/1-overlapper/queries/000076/000033.dat\n",
      "FILE correction/1-overlapper/queries/000076/000027.dat\n",
      "DIRECTORY correction/1-overlapper/queries/000038\n",
      "FILE correction/1-overlapper/queries/000038/000020.dat\n",
      "FILE correction/1-overlapper/queries/000038/000028.dat\n",
      "FILE correction/1-overlapper/queries/000038/000024.dat\n",
      "FILE correction/1-overlapper/queries/000038/000023.dat\n",
      "FILE correction/1-overlapper/queries/000038/000026.dat\n",
      "FILE correction/1-overlapper/queries/000038/000025.dat\n",
      "FILE correction/1-overlapper/queries/000038/000021.dat\n",
      "FILE correction/1-overlapper/queries/000038/000022.dat\n",
      "FILE correction/1-overlapper/queries/000038/000027.dat\n",
      "DIRECTORY correction/1-overlapper/queries/000031\n",
      "FILE correction/1-overlapper/queries/000031/000034.dat\n",
      "FILE correction/1-overlapper/queries/000031/000035.dat\n",
      "FILE correction/1-overlapper/queries/000031/000028.dat\n",
      "FILE correction/1-overlapper/queries/000031/000032.dat\n",
      "FILE correction/1-overlapper/queries/000031/000030.dat\n",
      "FILE correction/1-overlapper/queries/000031/000029.dat\n",
      "FILE correction/1-overlapper/queries/000031/000031.dat\n",
      "FILE correction/1-overlapper/queries/000031/000033.dat\n",
      "FILE correction/1-overlapper/queries/000031/000027.dat\n",
      "DIRECTORY correction/1-overlapper/queries/000066\n",
      "FILE correction/1-overlapper/queries/000066/000028.dat\n",
      "FILE correction/1-overlapper/queries/000066/000024.dat\n",
      "FILE correction/1-overlapper/queries/000066/000023.dat\n",
      "FILE correction/1-overlapper/queries/000066/000026.dat\n",
      "FILE correction/1-overlapper/queries/000066/000029.dat\n",
      "FILE correction/1-overlapper/queries/000066/000025.dat\n",
      "FILE correction/1-overlapper/queries/000066/000021.dat\n",
      "FILE correction/1-overlapper/queries/000066/000022.dat\n",
      "FILE correction/1-overlapper/queries/000066/000027.dat\n",
      "DIRECTORY correction/1-overlapper/queries/000008\n",
      "FILE correction/1-overlapper/queries/000008/000034.dat\n",
      "FILE correction/1-overlapper/queries/000008/000035.dat\n",
      "FILE correction/1-overlapper/queries/000008/000036.dat\n",
      "FILE correction/1-overlapper/queries/000008/000032.dat\n",
      "FILE correction/1-overlapper/queries/000008/000030.dat\n",
      "FILE correction/1-overlapper/queries/000008/000031.dat\n",
      "FILE correction/1-overlapper/queries/000008/000033.dat\n",
      "FILE correction/1-overlapper/queries/000008/000037.dat\n",
      "DIRECTORY correction/1-overlapper/queries/000083\n",
      "FILE correction/1-overlapper/queries/000083/000034.dat\n",
      "FILE correction/1-overlapper/queries/000083/000035.dat\n",
      "FILE correction/1-overlapper/queries/000083/000036.dat\n",
      "FILE correction/1-overlapper/queries/000083/000032.dat\n",
      "FILE correction/1-overlapper/queries/000083/000030.dat\n",
      "FILE correction/1-overlapper/queries/000083/000031.dat\n",
      "FILE correction/1-overlapper/queries/000083/000033.dat\n",
      "FILE correction/1-overlapper/queries/000083/000037.dat\n",
      "DIRECTORY correction/1-overlapper/queries/000053\n",
      "FILE correction/1-overlapper/queries/000053/000028.dat\n",
      "FILE correction/1-overlapper/queries/000053/000032.dat\n",
      "FILE correction/1-overlapper/queries/000053/000030.dat\n",
      "FILE correction/1-overlapper/queries/000053/000026.dat\n",
      "FILE correction/1-overlapper/queries/000053/000029.dat\n",
      "FILE correction/1-overlapper/queries/000053/000025.dat\n",
      "FILE correction/1-overlapper/queries/000053/000031.dat\n",
      "FILE correction/1-overlapper/queries/000053/000033.dat\n",
      "FILE correction/1-overlapper/queries/000053/000027.dat\n",
      "DIRECTORY correction/1-overlapper/queries/000030\n",
      "FILE correction/1-overlapper/queries/000030/000020.dat\n",
      "FILE correction/1-overlapper/queries/000030/000018.dat\n",
      "FILE correction/1-overlapper/queries/000030/000024.dat\n",
      "FILE correction/1-overlapper/queries/000030/000023.dat\n",
      "FILE correction/1-overlapper/queries/000030/000026.dat\n",
      "FILE correction/1-overlapper/queries/000030/000025.dat\n",
      "FILE correction/1-overlapper/queries/000030/000021.dat\n",
      "FILE correction/1-overlapper/queries/000030/000022.dat\n",
      "FILE correction/1-overlapper/queries/000030/000019.dat\n",
      "DIRECTORY correction/1-overlapper/queries/000085\n",
      "FILE correction/1-overlapper/queries/000085/000034.dat\n",
      "FILE correction/1-overlapper/queries/000085/000035.dat\n",
      "FILE correction/1-overlapper/queries/000085/000036.dat\n",
      "FILE correction/1-overlapper/queries/000085/000032.dat\n",
      "FILE correction/1-overlapper/queries/000085/000033.dat\n",
      "FILE correction/1-overlapper/queries/000085/000037.dat\n",
      "DIRECTORY correction/1-overlapper/queries/000055\n",
      "FILE correction/1-overlapper/queries/000055/000020.dat\n",
      "FILE correction/1-overlapper/queries/000055/000018.dat\n",
      "FILE correction/1-overlapper/queries/000055/000024.dat\n",
      "FILE correction/1-overlapper/queries/000055/000023.dat\n",
      "FILE correction/1-overlapper/queries/000055/000017.dat\n",
      "FILE correction/1-overlapper/queries/000055/000025.dat\n",
      "FILE correction/1-overlapper/queries/000055/000021.dat\n",
      "FILE correction/1-overlapper/queries/000055/000022.dat\n",
      "FILE correction/1-overlapper/queries/000055/000019.dat\n",
      "DIRECTORY correction/1-overlapper/queries/000001\n",
      "FILE correction/1-overlapper/queries/000001/000002.dat\n",
      "FILE correction/1-overlapper/queries/000001/000006.dat\n",
      "FILE correction/1-overlapper/queries/000001/000010.dat\n",
      "FILE correction/1-overlapper/queries/000001/000003.dat\n",
      "FILE correction/1-overlapper/queries/000001/000004.dat\n",
      "FILE correction/1-overlapper/queries/000001/000009.dat\n",
      "FILE correction/1-overlapper/queries/000001/000008.dat\n",
      "FILE correction/1-overlapper/queries/000001/000005.dat\n",
      "FILE correction/1-overlapper/queries/000001/000007.dat\n",
      "DIRECTORY correction/1-overlapper/queries/000078\n",
      "FILE correction/1-overlapper/queries/000078/000034.dat\n",
      "FILE correction/1-overlapper/queries/000078/000035.dat\n",
      "FILE correction/1-overlapper/queries/000078/000028.dat\n",
      "FILE correction/1-overlapper/queries/000078/000032.dat\n",
      "FILE correction/1-overlapper/queries/000078/000030.dat\n",
      "FILE correction/1-overlapper/queries/000078/000029.dat\n",
      "FILE correction/1-overlapper/queries/000078/000031.dat\n",
      "FILE correction/1-overlapper/queries/000078/000033.dat\n",
      "FILE correction/1-overlapper/queries/000078/000027.dat\n",
      "DIRECTORY correction/1-overlapper/queries/000091\n",
      "DIRECTORY correction/1-overlapper/queries/000005\n",
      "FILE correction/1-overlapper/queries/000005/000006.dat\n",
      "FILE correction/1-overlapper/queries/000005/000010.dat\n",
      "FILE correction/1-overlapper/queries/000005/000003.dat\n",
      "FILE correction/1-overlapper/queries/000005/000004.dat\n",
      "FILE correction/1-overlapper/queries/000005/000009.dat\n",
      "FILE correction/1-overlapper/queries/000005/000008.dat\n",
      "FILE correction/1-overlapper/queries/000005/000005.dat\n",
      "FILE correction/1-overlapper/queries/000005/000007.dat\n",
      "FILE correction/1-overlapper/queries/000005/000011.dat\n",
      "DIRECTORY correction/1-overlapper/queries/000035\n",
      "FILE correction/1-overlapper/queries/000035/000034.dat\n",
      "FILE correction/1-overlapper/queries/000035/000035.dat\n",
      "FILE correction/1-overlapper/queries/000035/000028.dat\n",
      "FILE correction/1-overlapper/queries/000035/000036.dat\n",
      "FILE correction/1-overlapper/queries/000035/000032.dat\n",
      "FILE correction/1-overlapper/queries/000035/000030.dat\n",
      "FILE correction/1-overlapper/queries/000035/000029.dat\n",
      "FILE correction/1-overlapper/queries/000035/000031.dat\n",
      "FILE correction/1-overlapper/queries/000035/000033.dat\n",
      "DIRECTORY correction/1-overlapper/queries/000034\n",
      "FILE correction/1-overlapper/queries/000034/000020.dat\n",
      "FILE correction/1-overlapper/queries/000034/000024.dat\n",
      "FILE correction/1-overlapper/queries/000034/000023.dat\n",
      "FILE correction/1-overlapper/queries/000034/000026.dat\n",
      "FILE correction/1-overlapper/queries/000034/000025.dat\n",
      "FILE correction/1-overlapper/queries/000034/000021.dat\n",
      "FILE correction/1-overlapper/queries/000034/000022.dat\n",
      "FILE correction/1-overlapper/queries/000034/000019.dat\n",
      "FILE correction/1-overlapper/queries/000034/000027.dat\n",
      "DIRECTORY correction/1-overlapper/queries/000029\n",
      "FILE correction/1-overlapper/queries/000029/000014.dat\n",
      "FILE correction/1-overlapper/queries/000029/000013.dat\n",
      "FILE correction/1-overlapper/queries/000029/000010.dat\n",
      "FILE correction/1-overlapper/queries/000029/000017.dat\n",
      "FILE correction/1-overlapper/queries/000029/000012.dat\n",
      "FILE correction/1-overlapper/queries/000029/000015.dat\n",
      "FILE correction/1-overlapper/queries/000029/000016.dat\n",
      "FILE correction/1-overlapper/queries/000029/000009.dat\n",
      "FILE correction/1-overlapper/queries/000029/000011.dat\n",
      "DIRECTORY correction/1-overlapper/queries/000007\n",
      "FILE correction/1-overlapper/queries/000007/000028.dat\n",
      "FILE correction/1-overlapper/queries/000007/000024.dat\n",
      "FILE correction/1-overlapper/queries/000007/000023.dat\n",
      "FILE correction/1-overlapper/queries/000007/000026.dat\n",
      "FILE correction/1-overlapper/queries/000007/000029.dat\n",
      "FILE correction/1-overlapper/queries/000007/000025.dat\n",
      "FILE correction/1-overlapper/queries/000007/000021.dat\n",
      "FILE correction/1-overlapper/queries/000007/000022.dat\n",
      "FILE correction/1-overlapper/queries/000007/000027.dat\n",
      "DIRECTORY correction/1-overlapper/queries/000050\n",
      "FILE correction/1-overlapper/queries/000050/000028.dat\n",
      "FILE correction/1-overlapper/queries/000050/000024.dat\n",
      "FILE correction/1-overlapper/queries/000050/000032.dat\n",
      "FILE correction/1-overlapper/queries/000050/000030.dat\n",
      "FILE correction/1-overlapper/queries/000050/000026.dat\n",
      "FILE correction/1-overlapper/queries/000050/000029.dat\n",
      "FILE correction/1-overlapper/queries/000050/000025.dat\n",
      "FILE correction/1-overlapper/queries/000050/000031.dat\n",
      "FILE correction/1-overlapper/queries/000050/000027.dat\n",
      "DIRECTORY correction/1-overlapper/queries/000062\n",
      "FILE correction/1-overlapper/queries/000062/000034.dat\n",
      "FILE correction/1-overlapper/queries/000062/000035.dat\n",
      "FILE correction/1-overlapper/queries/000062/000028.dat\n",
      "FILE correction/1-overlapper/queries/000062/000036.dat\n",
      "FILE correction/1-overlapper/queries/000062/000032.dat\n",
      "FILE correction/1-overlapper/queries/000062/000030.dat\n",
      "FILE correction/1-overlapper/queries/000062/000029.dat\n",
      "FILE correction/1-overlapper/queries/000062/000031.dat\n",
      "FILE correction/1-overlapper/queries/000062/000033.dat\n",
      "DIRECTORY correction/1-overlapper/queries/000018\n",
      "FILE correction/1-overlapper/queries/000018/000020.dat\n",
      "FILE correction/1-overlapper/queries/000018/000018.dat\n",
      "FILE correction/1-overlapper/queries/000018/000023.dat\n",
      "FILE correction/1-overlapper/queries/000018/000017.dat\n",
      "FILE correction/1-overlapper/queries/000018/000015.dat\n",
      "FILE correction/1-overlapper/queries/000018/000016.dat\n",
      "FILE correction/1-overlapper/queries/000018/000021.dat\n",
      "FILE correction/1-overlapper/queries/000018/000022.dat\n",
      "FILE correction/1-overlapper/queries/000018/000019.dat\n",
      "DIRECTORY correction/1-overlapper/queries/000045\n",
      "FILE correction/1-overlapper/queries/000045/000034.dat\n",
      "FILE correction/1-overlapper/queries/000045/000035.dat\n",
      "FILE correction/1-overlapper/queries/000045/000036.dat\n",
      "FILE correction/1-overlapper/queries/000045/000032.dat\n",
      "FILE correction/1-overlapper/queries/000045/000031.dat\n",
      "FILE correction/1-overlapper/queries/000045/000033.dat\n",
      "FILE correction/1-overlapper/queries/000045/000037.dat\n",
      "DIRECTORY correction/1-overlapper/queries/000065\n",
      "FILE correction/1-overlapper/queries/000065/000034.dat\n",
      "FILE correction/1-overlapper/queries/000065/000035.dat\n",
      "FILE correction/1-overlapper/queries/000065/000036.dat\n",
      "FILE correction/1-overlapper/queries/000065/000032.dat\n",
      "FILE correction/1-overlapper/queries/000065/000030.dat\n",
      "FILE correction/1-overlapper/queries/000065/000029.dat\n",
      "FILE correction/1-overlapper/queries/000065/000031.dat\n",
      "FILE correction/1-overlapper/queries/000065/000033.dat\n",
      "FILE correction/1-overlapper/queries/000065/000037.dat\n",
      "DIRECTORY correction/1-overlapper/queries/000061\n",
      "FILE correction/1-overlapper/queries/000061/000020.dat\n",
      "FILE correction/1-overlapper/queries/000061/000024.dat\n",
      "FILE correction/1-overlapper/queries/000061/000023.dat\n",
      "FILE correction/1-overlapper/queries/000061/000026.dat\n",
      "FILE correction/1-overlapper/queries/000061/000025.dat\n",
      "FILE correction/1-overlapper/queries/000061/000021.dat\n",
      "FILE correction/1-overlapper/queries/000061/000022.dat\n",
      "FILE correction/1-overlapper/queries/000061/000019.dat\n",
      "FILE correction/1-overlapper/queries/000061/000027.dat\n",
      "DIRECTORY correction/1-overlapper/queries/000011\n",
      "FILE correction/1-overlapper/queries/000011/000028.dat\n",
      "FILE correction/1-overlapper/queries/000011/000024.dat\n",
      "FILE correction/1-overlapper/queries/000011/000023.dat\n",
      "FILE correction/1-overlapper/queries/000011/000030.dat\n",
      "FILE correction/1-overlapper/queries/000011/000026.dat\n",
      "FILE correction/1-overlapper/queries/000011/000029.dat\n",
      "FILE correction/1-overlapper/queries/000011/000025.dat\n",
      "FILE correction/1-overlapper/queries/000011/000022.dat\n",
      "FILE correction/1-overlapper/queries/000011/000027.dat\n",
      "DIRECTORY correction/1-overlapper/queries/000044\n",
      "FILE correction/1-overlapper/queries/000044/000028.dat\n",
      "FILE correction/1-overlapper/queries/000044/000024.dat\n",
      "FILE correction/1-overlapper/queries/000044/000023.dat\n",
      "FILE correction/1-overlapper/queries/000044/000030.dat\n",
      "FILE correction/1-overlapper/queries/000044/000026.dat\n",
      "FILE correction/1-overlapper/queries/000044/000029.dat\n",
      "FILE correction/1-overlapper/queries/000044/000025.dat\n",
      "FILE correction/1-overlapper/queries/000044/000022.dat\n",
      "FILE correction/1-overlapper/queries/000044/000027.dat\n",
      "DIRECTORY correction/1-overlapper/queries/000037\n",
      "FILE correction/1-overlapper/queries/000037/000014.dat\n",
      "FILE correction/1-overlapper/queries/000037/000013.dat\n",
      "FILE correction/1-overlapper/queries/000037/000018.dat\n",
      "FILE correction/1-overlapper/queries/000037/000017.dat\n",
      "FILE correction/1-overlapper/queries/000037/000012.dat\n",
      "FILE correction/1-overlapper/queries/000037/000015.dat\n",
      "FILE correction/1-overlapper/queries/000037/000016.dat\n",
      "FILE correction/1-overlapper/queries/000037/000019.dat\n",
      "FILE correction/1-overlapper/queries/000037/000011.dat\n",
      "DIRECTORY correction/1-overlapper/queries/000041\n",
      "FILE correction/1-overlapper/queries/000041/000028.dat\n",
      "FILE correction/1-overlapper/queries/000041/000024.dat\n",
      "FILE correction/1-overlapper/queries/000041/000023.dat\n",
      "FILE correction/1-overlapper/queries/000041/000026.dat\n",
      "FILE correction/1-overlapper/queries/000041/000029.dat\n",
      "FILE correction/1-overlapper/queries/000041/000025.dat\n",
      "FILE correction/1-overlapper/queries/000041/000021.dat\n",
      "FILE correction/1-overlapper/queries/000041/000022.dat\n",
      "FILE correction/1-overlapper/queries/000041/000027.dat\n",
      "DIRECTORY correction/1-overlapper/queries/000026\n",
      "FILE correction/1-overlapper/queries/000026/000020.dat\n",
      "FILE correction/1-overlapper/queries/000026/000018.dat\n",
      "FILE correction/1-overlapper/queries/000026/000024.dat\n",
      "FILE correction/1-overlapper/queries/000026/000023.dat\n",
      "FILE correction/1-overlapper/queries/000026/000017.dat\n",
      "FILE correction/1-overlapper/queries/000026/000025.dat\n",
      "FILE correction/1-overlapper/queries/000026/000021.dat\n",
      "FILE correction/1-overlapper/queries/000026/000022.dat\n",
      "FILE correction/1-overlapper/queries/000026/000019.dat\n",
      "DIRECTORY correction/1-overlapper/queries/000063\n",
      "FILE correction/1-overlapper/queries/000063/000037.dat\n",
      "DIRECTORY correction/1-overlapper/queries/000042\n",
      "FILE correction/1-overlapper/queries/000042/000034.dat\n",
      "FILE correction/1-overlapper/queries/000042/000035.dat\n",
      "FILE correction/1-overlapper/queries/000042/000036.dat\n",
      "FILE correction/1-overlapper/queries/000042/000032.dat\n",
      "FILE correction/1-overlapper/queries/000042/000030.dat\n",
      "FILE correction/1-overlapper/queries/000042/000031.dat\n",
      "FILE correction/1-overlapper/queries/000042/000033.dat\n",
      "FILE correction/1-overlapper/queries/000042/000037.dat\n",
      "DIRECTORY correction/1-overlapper/queries/000069\n",
      "FILE correction/1-overlapper/queries/000069/000034.dat\n",
      "FILE correction/1-overlapper/queries/000069/000035.dat\n",
      "FILE correction/1-overlapper/queries/000069/000036.dat\n",
      "FILE correction/1-overlapper/queries/000069/000032.dat\n",
      "FILE correction/1-overlapper/queries/000069/000031.dat\n",
      "FILE correction/1-overlapper/queries/000069/000033.dat\n",
      "FILE correction/1-overlapper/queries/000069/000037.dat\n",
      "DIRECTORY correction/1-overlapper/queries/000048\n",
      "FILE correction/1-overlapper/queries/000048/000034.dat\n",
      "FILE correction/1-overlapper/queries/000048/000035.dat\n",
      "FILE correction/1-overlapper/queries/000048/000036.dat\n",
      "FILE correction/1-overlapper/queries/000048/000032.dat\n",
      "FILE correction/1-overlapper/queries/000048/000033.dat\n",
      "FILE correction/1-overlapper/queries/000048/000037.dat\n",
      "DIRECTORY correction/1-overlapper/queries/000002\n",
      "FILE correction/1-overlapper/queries/000002/000014.dat\n",
      "FILE correction/1-overlapper/queries/000002/000013.dat\n",
      "FILE correction/1-overlapper/queries/000002/000018.dat\n",
      "FILE correction/1-overlapper/queries/000002/000017.dat\n",
      "FILE correction/1-overlapper/queries/000002/000012.dat\n",
      "FILE correction/1-overlapper/queries/000002/000015.dat\n",
      "FILE correction/1-overlapper/queries/000002/000016.dat\n",
      "FILE correction/1-overlapper/queries/000002/000019.dat\n",
      "FILE correction/1-overlapper/queries/000002/000011.dat\n",
      "DIRECTORY correction/1-overlapper/queries/000072\n",
      "FILE correction/1-overlapper/queries/000072/000028.dat\n",
      "FILE correction/1-overlapper/queries/000072/000024.dat\n",
      "FILE correction/1-overlapper/queries/000072/000032.dat\n",
      "FILE correction/1-overlapper/queries/000072/000030.dat\n",
      "FILE correction/1-overlapper/queries/000072/000026.dat\n",
      "FILE correction/1-overlapper/queries/000072/000029.dat\n",
      "FILE correction/1-overlapper/queries/000072/000025.dat\n",
      "FILE correction/1-overlapper/queries/000072/000031.dat\n",
      "FILE correction/1-overlapper/queries/000072/000027.dat\n",
      "DIRECTORY correction/1-overlapper/queries/000021\n",
      "FILE correction/1-overlapper/queries/000021/000014.dat\n",
      "FILE correction/1-overlapper/queries/000021/000013.dat\n",
      "FILE correction/1-overlapper/queries/000021/000010.dat\n",
      "FILE correction/1-overlapper/queries/000021/000012.dat\n",
      "FILE correction/1-overlapper/queries/000021/000015.dat\n",
      "FILE correction/1-overlapper/queries/000021/000009.dat\n",
      "FILE correction/1-overlapper/queries/000021/000008.dat\n",
      "FILE correction/1-overlapper/queries/000021/000007.dat\n",
      "FILE correction/1-overlapper/queries/000021/000011.dat\n",
      "DIRECTORY correction/1-overlapper/queries/000088\n",
      "FILE correction/1-overlapper/queries/000088/000035.dat\n",
      "FILE correction/1-overlapper/queries/000088/000036.dat\n",
      "FILE correction/1-overlapper/queries/000088/000037.dat\n",
      "DIRECTORY correction/1-overlapper/queries/000077\n",
      "FILE correction/1-overlapper/queries/000077/000035.dat\n",
      "FILE correction/1-overlapper/queries/000077/000036.dat\n",
      "FILE correction/1-overlapper/queries/000077/000037.dat\n",
      "DIRECTORY correction/1-overlapper/queries/000043\n",
      "FILE correction/1-overlapper/queries/000043/000020.dat\n",
      "FILE correction/1-overlapper/queries/000043/000014.dat\n",
      "FILE correction/1-overlapper/queries/000043/000013.dat\n",
      "FILE correction/1-overlapper/queries/000043/000018.dat\n",
      "FILE correction/1-overlapper/queries/000043/000017.dat\n",
      "FILE correction/1-overlapper/queries/000043/000015.dat\n",
      "FILE correction/1-overlapper/queries/000043/000016.dat\n",
      "FILE correction/1-overlapper/queries/000043/000021.dat\n",
      "FILE correction/1-overlapper/queries/000043/000019.dat\n",
      "DIRECTORY correction/1-overlapper/queries/000051\n",
      "FILE correction/1-overlapper/queries/000051/000034.dat\n",
      "FILE correction/1-overlapper/queries/000051/000035.dat\n",
      "FILE correction/1-overlapper/queries/000051/000036.dat\n",
      "FILE correction/1-overlapper/queries/000051/000033.dat\n",
      "FILE correction/1-overlapper/queries/000051/000037.dat\n",
      "DIRECTORY correction/1-overlapper/queries/000006\n",
      "FILE correction/1-overlapper/queries/000006/000020.dat\n",
      "FILE correction/1-overlapper/queries/000006/000014.dat\n",
      "FILE correction/1-overlapper/queries/000006/000013.dat\n",
      "FILE correction/1-overlapper/queries/000006/000018.dat\n",
      "FILE correction/1-overlapper/queries/000006/000017.dat\n",
      "FILE correction/1-overlapper/queries/000006/000012.dat\n",
      "FILE correction/1-overlapper/queries/000006/000015.dat\n",
      "FILE correction/1-overlapper/queries/000006/000016.dat\n",
      "FILE correction/1-overlapper/queries/000006/000019.dat\n",
      "DIRECTORY correction/1-overlapper/queries/000089\n",
      "FILE correction/1-overlapper/queries/000089/000036.dat\n",
      "FILE correction/1-overlapper/queries/000089/000037.dat\n",
      "DIRECTORY correction/1-overlapper/queries/000032\n",
      "FILE correction/1-overlapper/queries/000032/000036.dat\n",
      "FILE correction/1-overlapper/queries/000032/000037.dat\n",
      "DIRECTORY correction/1-overlapper/queries/000073\n",
      "FILE correction/1-overlapper/queries/000073/000034.dat\n",
      "FILE correction/1-overlapper/queries/000073/000035.dat\n",
      "FILE correction/1-overlapper/queries/000073/000036.dat\n",
      "FILE correction/1-overlapper/queries/000073/000033.dat\n",
      "FILE correction/1-overlapper/queries/000073/000037.dat\n",
      "DIRECTORY correction/1-overlapper/queries/000082\n",
      "FILE correction/1-overlapper/queries/000082/000034.dat\n",
      "FILE correction/1-overlapper/queries/000082/000035.dat\n",
      "FILE correction/1-overlapper/queries/000082/000036.dat\n",
      "FILE correction/1-overlapper/queries/000082/000032.dat\n",
      "FILE correction/1-overlapper/queries/000082/000030.dat\n",
      "FILE correction/1-overlapper/queries/000082/000029.dat\n",
      "FILE correction/1-overlapper/queries/000082/000031.dat\n",
      "FILE correction/1-overlapper/queries/000082/000033.dat\n",
      "FILE correction/1-overlapper/queries/000082/000037.dat\n",
      "DIRECTORY correction/1-overlapper/queries/000023\n",
      "FILE correction/1-overlapper/queries/000023/000028.dat\n",
      "FILE correction/1-overlapper/queries/000023/000032.dat\n",
      "FILE correction/1-overlapper/queries/000023/000030.dat\n",
      "FILE correction/1-overlapper/queries/000023/000026.dat\n",
      "FILE correction/1-overlapper/queries/000023/000029.dat\n",
      "FILE correction/1-overlapper/queries/000023/000025.dat\n",
      "FILE correction/1-overlapper/queries/000023/000031.dat\n",
      "FILE correction/1-overlapper/queries/000023/000033.dat\n",
      "FILE correction/1-overlapper/queries/000023/000027.dat\n",
      "DIRECTORY correction/1-overlapper/queries/000079\n",
      "FILE correction/1-overlapper/queries/000079/000036.dat\n",
      "FILE correction/1-overlapper/queries/000079/000037.dat\n",
      "DIRECTORY correction/1-overlapper/queries/000017\n",
      "FILE correction/1-overlapper/queries/000017/000006.dat\n",
      "FILE correction/1-overlapper/queries/000017/000014.dat\n",
      "FILE correction/1-overlapper/queries/000017/000013.dat\n",
      "FILE correction/1-overlapper/queries/000017/000010.dat\n",
      "FILE correction/1-overlapper/queries/000017/000012.dat\n",
      "FILE correction/1-overlapper/queries/000017/000009.dat\n",
      "FILE correction/1-overlapper/queries/000017/000008.dat\n",
      "FILE correction/1-overlapper/queries/000017/000007.dat\n",
      "FILE correction/1-overlapper/queries/000017/000011.dat\n",
      "DIRECTORY correction/1-overlapper/queries/000012\n",
      "FILE correction/1-overlapper/queries/000012/000034.dat\n",
      "FILE correction/1-overlapper/queries/000012/000035.dat\n",
      "FILE correction/1-overlapper/queries/000012/000036.dat\n",
      "FILE correction/1-overlapper/queries/000012/000032.dat\n",
      "FILE correction/1-overlapper/queries/000012/000031.dat\n",
      "FILE correction/1-overlapper/queries/000012/000033.dat\n",
      "FILE correction/1-overlapper/queries/000012/000037.dat\n",
      "DIRECTORY correction/1-overlapper/queries/000067\n",
      "FILE correction/1-overlapper/queries/000067/000034.dat\n",
      "FILE correction/1-overlapper/queries/000067/000035.dat\n",
      "FILE correction/1-overlapper/queries/000067/000036.dat\n",
      "FILE correction/1-overlapper/queries/000067/000032.dat\n",
      "FILE correction/1-overlapper/queries/000067/000030.dat\n",
      "FILE correction/1-overlapper/queries/000067/000031.dat\n",
      "FILE correction/1-overlapper/queries/000067/000033.dat\n",
      "FILE correction/1-overlapper/queries/000067/000037.dat\n",
      "DIRECTORY correction/1-overlapper/queries/000054\n",
      "FILE correction/1-overlapper/queries/000054/000034.dat\n",
      "FILE correction/1-overlapper/queries/000054/000035.dat\n",
      "FILE correction/1-overlapper/queries/000054/000036.dat\n",
      "FILE correction/1-overlapper/queries/000054/000037.dat\n",
      "DIRECTORY correction/1-overlapper/queries/000014\n",
      "FILE correction/1-overlapper/queries/000014/000020.dat\n",
      "FILE correction/1-overlapper/queries/000014/000014.dat\n",
      "FILE correction/1-overlapper/queries/000014/000018.dat\n",
      "FILE correction/1-overlapper/queries/000014/000017.dat\n",
      "FILE correction/1-overlapper/queries/000014/000015.dat\n",
      "FILE correction/1-overlapper/queries/000014/000016.dat\n",
      "FILE correction/1-overlapper/queries/000014/000021.dat\n",
      "FILE correction/1-overlapper/queries/000014/000022.dat\n",
      "FILE correction/1-overlapper/queries/000014/000019.dat\n",
      "DIRECTORY correction/1-overlapper/queries/000052\n",
      "FILE correction/1-overlapper/queries/000052/000020.dat\n",
      "FILE correction/1-overlapper/queries/000052/000018.dat\n",
      "FILE correction/1-overlapper/queries/000052/000024.dat\n",
      "FILE correction/1-overlapper/queries/000052/000023.dat\n",
      "FILE correction/1-overlapper/queries/000052/000017.dat\n",
      "FILE correction/1-overlapper/queries/000052/000016.dat\n",
      "FILE correction/1-overlapper/queries/000052/000021.dat\n",
      "FILE correction/1-overlapper/queries/000052/000022.dat\n",
      "FILE correction/1-overlapper/queries/000052/000019.dat\n",
      "DIRECTORY correction/1-overlapper/queries/000074\n",
      "FILE correction/1-overlapper/queries/000074/000028.dat\n",
      "FILE correction/1-overlapper/queries/000074/000032.dat\n",
      "FILE correction/1-overlapper/queries/000074/000030.dat\n",
      "FILE correction/1-overlapper/queries/000074/000026.dat\n",
      "FILE correction/1-overlapper/queries/000074/000029.dat\n",
      "FILE correction/1-overlapper/queries/000074/000025.dat\n",
      "FILE correction/1-overlapper/queries/000074/000031.dat\n",
      "FILE correction/1-overlapper/queries/000074/000033.dat\n",
      "FILE correction/1-overlapper/queries/000074/000027.dat\n",
      "DIRECTORY correction/1-overlapper/queries/000039\n",
      "FILE correction/1-overlapper/queries/000039/000034.dat\n",
      "FILE correction/1-overlapper/queries/000039/000035.dat\n",
      "FILE correction/1-overlapper/queries/000039/000036.dat\n",
      "FILE correction/1-overlapper/queries/000039/000032.dat\n",
      "FILE correction/1-overlapper/queries/000039/000030.dat\n",
      "FILE correction/1-overlapper/queries/000039/000029.dat\n",
      "FILE correction/1-overlapper/queries/000039/000031.dat\n",
      "FILE correction/1-overlapper/queries/000039/000033.dat\n",
      "FILE correction/1-overlapper/queries/000039/000037.dat\n",
      "DIRECTORY correction/1-overlapper/queries/000081\n",
      "FILE correction/1-overlapper/queries/000081/000037.dat\n",
      "FILE correction/1-overlapper/blocks/000001.dat\n",
      "FILE correction/1-overlapper/blocks/000002.dat\n",
      "FILE correction/1-overlapper/blocks/000003.dat\n",
      "FILE correction/1-overlapper/blocks/000004.dat\n",
      "FILE correction/1-overlapper/blocks/000005.dat\n",
      "FILE correction/1-overlapper/blocks/000006.dat\n",
      "FILE correction/1-overlapper/blocks/000007.dat\n",
      "FILE correction/1-overlapper/blocks/000008.dat\n",
      "FILE correction/1-overlapper/blocks/000009.dat\n",
      "FILE correction/1-overlapper/blocks/000010.dat\n",
      "FILE correction/1-overlapper/blocks/000011.dat\n",
      "FILE correction/1-overlapper/blocks/000012.dat\n",
      "FILE correction/1-overlapper/blocks/000013.dat\n",
      "FILE correction/1-overlapper/blocks/000014.dat\n",
      "FILE correction/1-overlapper/blocks/000015.dat\n",
      "FILE correction/1-overlapper/blocks/000016.dat\n",
      "FILE correction/1-overlapper/blocks/000017.dat\n",
      "FILE correction/1-overlapper/blocks/000018.dat\n",
      "FILE correction/1-overlapper/blocks/000019.dat\n",
      "FILE correction/1-overlapper/blocks/000020.dat\n",
      "FILE correction/1-overlapper/blocks/000021.dat\n",
      "FILE correction/1-overlapper/blocks/000022.dat\n",
      "FILE correction/1-overlapper/blocks/000023.dat\n",
      "FILE correction/1-overlapper/blocks/000024.dat\n",
      "FILE correction/1-overlapper/blocks/000025.dat\n",
      "FILE correction/1-overlapper/blocks/000026.dat\n",
      "FILE correction/1-overlapper/blocks/000027.dat\n",
      "FILE correction/1-overlapper/blocks/000028.dat\n",
      "FILE correction/1-overlapper/blocks/000029.dat\n",
      "FILE correction/1-overlapper/blocks/000030.dat\n",
      "FILE correction/1-overlapper/blocks/000031.dat\n",
      "FILE correction/1-overlapper/blocks/000032.dat\n",
      "FILE correction/1-overlapper/blocks/000033.dat\n",
      "FILE correction/1-overlapper/blocks/000034.dat\n",
      "FILE correction/1-overlapper/blocks/000035.dat\n",
      "FILE correction/1-overlapper/blocks/000036.dat\n",
      "FILE correction/1-overlapper/blocks/000037.dat\n",
      "REMOVE TREE correction/1-overlapper/queries/000052\n",
      "REMOVE TREE correction/1-overlapper/queries/000030\n",
      "REMOVE TREE correction/1-overlapper/queries/000041\n",
      "REMOVE TREE correction/1-overlapper/queries/000069\n",
      "REMOVE TREE correction/1-overlapper/queries/000048\n",
      "REMOVE TREE correction/1-overlapper/queries/000079\n",
      "REMOVE TREE correction/1-overlapper/queries/000013\n",
      "REMOVE TREE correction/1-overlapper/queries/000033\n",
      "REMOVE TREE correction/1-overlapper/queries/000047\n",
      "REMOVE TREE correction/1-overlapper/queries/000020\n",
      "REMOVE TREE correction/1-overlapper/queries/000068\n",
      "REMOVE TREE correction/1-overlapper/queries/000059\n",
      "REMOVE TREE correction/1-overlapper/queries/000053\n",
      "REMOVE TREE correction/1-overlapper/queries/000034\n",
      "REMOVE TREE correction/1-overlapper/queries/000024\n",
      "REMOVE TREE correction/1-overlapper/queries/000012\n",
      "REMOVE TREE correction/1-overlapper/queries/000074\n",
      "REMOVE TREE correction/1-overlapper/queries/000032\n",
      "REMOVE TREE correction/1-overlapper/queries/000040\n",
      "REMOVE TREE correction/1-overlapper/queries/000028\n",
      "REMOVE TREE correction/1-overlapper/queries/000015\n",
      "REMOVE TREE correction/1-overlapper/queries/000014\n",
      "REMOVE TREE correction/1-overlapper/queries/000085\n",
      "REMOVE TREE correction/1-overlapper/queries/000004\n",
      "REMOVE TREE correction/1-overlapper/queries/000011\n",
      "REMOVE TREE correction/1-overlapper/queries/000060\n",
      "REMOVE TREE correction/1-overlapper/queries/000027\n",
      "REMOVE TREE correction/1-overlapper/queries/000086\n",
      "REMOVE TREE correction/1-overlapper/queries/000022\n",
      "REMOVE TREE correction/1-overlapper/queries/000090\n",
      "REMOVE TREE correction/1-overlapper/queries/000081\n",
      "REMOVE TREE correction/1-overlapper/queries/000018\n",
      "REMOVE TREE correction/1-overlapper/queries/000076\n",
      "REMOVE TREE correction/1-overlapper/queries/000083\n",
      "REMOVE TREE correction/1-overlapper/queries/000038\n",
      "REMOVE TREE correction/1-overlapper/queries/000078\n",
      "REMOVE TREE correction/1-overlapper/queries/000062\n",
      "REMOVE TREE correction/1-overlapper/queries/000089\n",
      "REMOVE TREE correction/1-overlapper/queries/000023\n",
      "REMOVE TREE correction/1-overlapper/queries/000016\n",
      "REMOVE TREE correction/1-overlapper/queries/000036\n",
      "REMOVE TREE correction/1-overlapper/queries/000026\n",
      "REMOVE TREE correction/1-overlapper/queries/000003\n",
      "REMOVE TREE correction/1-overlapper/queries/000046\n",
      "REMOVE TREE correction/1-overlapper/queries/000009\n",
      "REMOVE TREE correction/1-overlapper/queries/000055\n",
      "REMOVE TREE correction/1-overlapper/queries/000002\n",
      "REMOVE TREE correction/1-overlapper/queries/000056\n",
      "REMOVE TREE correction/1-overlapper/queries/000072\n",
      "REMOVE TREE correction/1-overlapper/queries/000067\n",
      "REMOVE TREE correction/1-overlapper/queries/000082\n",
      "REMOVE TREE correction/1-overlapper/queries/000005\n",
      "REMOVE TREE correction/1-overlapper/queries/000019\n",
      "REMOVE TREE correction/1-overlapper/queries/000029\n",
      "REMOVE TREE correction/1-overlapper/queries/000064\n",
      "REMOVE TREE correction/1-overlapper/queries/000077\n",
      "REMOVE TREE correction/1-overlapper/queries/000075\n",
      "REMOVE TREE correction/1-overlapper/queries/000063\n",
      "REMOVE TREE correction/1-overlapper/queries/000049\n",
      "REMOVE TREE correction/1-overlapper/queries/000044\n",
      "REMOVE TREE correction/1-overlapper/queries/000091\n",
      "REMOVE TREE correction/1-overlapper/queries/000031\n",
      "REMOVE TREE correction/1-overlapper/queries/000066\n",
      "REMOVE TREE correction/1-overlapper/queries/000070\n",
      "REMOVE TREE correction/1-overlapper/queries/000054\n",
      "REMOVE TREE correction/1-overlapper/queries/000035\n",
      "REMOVE TREE correction/1-overlapper/queries/000043\n",
      "REMOVE TREE correction/1-overlapper/queries/000006\n",
      "REMOVE TREE correction/1-overlapper/queries/000045\n",
      "REMOVE TREE correction/1-overlapper/queries/000071\n",
      "REMOVE TREE correction/1-overlapper/queries/000073\n",
      "REMOVE TREE correction/1-overlapper/queries/000050\n",
      "REMOVE TREE correction/1-overlapper/queries/000080\n",
      "REMOVE TREE correction/1-overlapper/queries/000017\n",
      "REMOVE TREE correction/1-overlapper/queries/000039\n",
      "REMOVE TREE correction/1-overlapper/queries/000021\n",
      "REMOVE TREE correction/1-overlapper/queries/000037\n",
      "REMOVE TREE correction/1-overlapper/queries/000061\n",
      "REMOVE TREE correction/1-overlapper/queries/000010\n",
      "REMOVE TREE correction/1-overlapper/queries/000051\n",
      "REMOVE TREE correction/1-overlapper/queries/000042\n",
      "REMOVE TREE correction/1-overlapper/queries/000065\n",
      "REMOVE TREE correction/1-overlapper/queries\n",
      "REMOVE TREE correction/1-overlapper/queries/000008\n",
      "REMOVE TREE correction/1-overlapper/queries/000058\n",
      "REMOVE TREE correction/1-overlapper/queries/000001\n",
      "REMOVE TREE correction/1-overlapper/queries/000057\n",
      "REMOVE TREE correction/1-overlapper/queries/000025\n",
      "REMOVE TREE correction/1-overlapper/queries/000007\n",
      "REMOVE TREE correction/1-overlapper/queries/000088\n",
      "REMOVE TREE correction/1-overlapper/queries/000084\n",
      "REMOVE TREE correction/1-overlapper/queries/000087\n",
      "--\n",
      "-- Purged 1008.114 GB in 885 overlap output files.\n",
      "-- Overlap store 'correction/20171018_oly_pacbio.ovlStore' statistics not available (skipped in correction and trimming stages).\n",
      "--\n",
      "-- Finished stage 'cor-createOverlapStore', reset canuIteration.\n",
      "-- Set corMinCoverage=0 based on read coverage of 6.\n",
      "-- Computing global filter scores 'correction/2-correction/20171018_oly_pacbio.globalScores'.\n",
      "----------------------------------------\n",
      "-- Starting command on Fri Oct 20 06:05:11 2017 with 674.787 GB free disk space\n",
      "\n",
      "    cd correction/2-correction\n",
      "    /usr/local/bioinformatics/canu-1.6/Linux-amd64/bin/filterCorrectionOverlaps \\\n",
      "      -G ../20171018_oly_pacbio.gkpStore \\\n",
      "      -O ../20171018_oly_pacbio.ovlStore \\\n",
      "      -S ./20171018_oly_pacbio.globalScores.WORKING \\\n",
      "      -c 40 \\\n",
      "      -l 0 \\\n",
      "    > ./20171018_oly_pacbio.globalScores.err 2>&1\n",
      "\n",
      "-- Finished on Fri Oct 20 06:37:26 2017 (1935 seconds) with 674.603 GB free disk space\n",
      "----------------------------------------\n",
      "--  PARAMETERS:\n",
      "--  ----------\n",
      "--  \n",
      "--       40 (expected coverage)\n",
      "--        0 (don't use overlaps shorter than this)\n",
      "--    0.000 (don't use overlaps with erate less than this)\n",
      "--    1.000 (don't use overlaps with erate more than this)\n",
      "--  \n",
      "--  OVERLAPS:\n",
      "--  --------\n",
      "--  \n",
      "--  IGNORED:\n",
      "--  \n",
      "--             0 (< 0.0000 fraction error)\n",
      "--             0 (> 0.4095 fraction error)\n",
      "--             0 (< 0 bases long)\n",
      "--             0 (> 2097151 bases long)\n",
      "--  \n",
      "--  FILTERED:\n",
      "--  \n",
      "--    9033508064 (too many overlaps, discard these shortest ones)\n",
      "--  \n",
      "--  EVIDENCE:\n",
      "--  \n",
      "--      68112774 (longest overlaps)\n",
      "--  \n",
      "--  TOTAL:\n",
      "--  \n",
      "--    9101620838 (all overlaps)\n",
      "--  \n",
      "--  READS:\n",
      "--  -----\n",
      "--  \n",
      "--           874 (no overlaps)\n",
      "--        102785 (no overlaps filtered)\n",
      "--        111724 (<  50% overlaps filtered)\n",
      "--        303069 (<  80% overlaps filtered)\n",
      "--        629330 (<  95% overlaps filtered)\n",
      "--       1635193 (< 100% overlaps filtered)\n",
      "--  \n",
      "-- Computing expected corrected read lengths 'correction/2-correction/20171018_oly_pacbio.estimate.log'.\n",
      "----------------------------------------\n",
      "-- Starting command on Fri Oct 20 06:37:29 2017 with 674.603 GB free disk space\n",
      "\n",
      "    cd correction/2-correction\n",
      "    /usr/local/bioinformatics/canu-1.6/Linux-amd64/bin/generateCorrectionLayouts \\\n",
      "      -G ../20171018_oly_pacbio.gkpStore \\\n",
      "      -O ../20171018_oly_pacbio.ovlStore \\\n",
      "      -S ./20171018_oly_pacbio.globalScores \\\n",
      "      -c 0 \\\n",
      "      -C 80 \\\n",
      "      -p ./20171018_oly_pacbio.estimate.WORKING\n",
      "\n",
      "-- Finished on Fri Oct 20 07:08:45 2017 (1876 seconds) with 674.569 GB free disk space\n",
      "----------------------------------------\n",
      "-- Sorting reads by expected corrected length.\n",
      "-- Sorting reads by uncorrected length.\n",
      "-- Loading expected corrected read lengths.\n",
      "-- Picking longest corrected reads.\n",
      "-- Writing longest corrected reads to 'correction/2-correction/20171018_oly_pacbio.readsToCorrect'.\n",
      "-- Summarizing filter.\n",
      "--\n",
      "-- Reads to be corrected:\n",
      "--   1737978 reads longer than 0 bp\n",
      "--   12703460519 bp\n",
      "-- Expected corrected reads:\n",
      "--   1737978 reads\n",
      "--   10398007480 bp\n",
      "--   0 bp minimum length\n",
      "--   5983 bp mean length\n",
      "--   14867 bp n50 length\n",
      "-- Using overlaps no worse than 0.3 fraction error for correcting reads (from corErrorRate parameter).\n",
      "-- Finished stage 'cor-buildCorrectionLayouts', reset canuIteration.\n",
      "--\n",
      "-- Running jobs.  First attempt out of 2.\n",
      "----------------------------------------\n",
      "-- Starting 'cor' concurrent execution on Fri Oct 20 07:10:15 2017 with 673.78 GB free disk space (116 processes; 6 concurrently)\n",
      "\n",
      "    cd correction/2-correction\n",
      "    ./correctReads.sh 1 > ./correctReads.000001.out 2>&1\n",
      "    ./correctReads.sh 2 > ./correctReads.000002.out 2>&1\n",
      "    ./correctReads.sh 3 > ./correctReads.000003.out 2>&1\n",
      "    ./correctReads.sh 4 > ./correctReads.000004.out 2>&1\n",
      "    ./correctReads.sh 5 > ./correctReads.000005.out 2>&1\n",
      "    ./correctReads.sh 6 > ./correctReads.000006.out 2>&1\n",
      "    ./correctReads.sh 7 > ./correctReads.000007.out 2>&1\n",
      "    ./correctReads.sh 8 > ./correctReads.000008.out 2>&1\n",
      "    ./correctReads.sh 9 > ./correctReads.000009.out 2>&1\n",
      "    ./correctReads.sh 10 > ./correctReads.000010.out 2>&1\n",
      "    ./correctReads.sh 11 > ./correctReads.000011.out 2>&1\n",
      "    ./correctReads.sh 12 > ./correctReads.000012.out 2>&1\n",
      "    ./correctReads.sh 13 > ./correctReads.000013.out 2>&1\n",
      "    ./correctReads.sh 14 > ./correctReads.000014.out 2>&1\n",
      "    ./correctReads.sh 15 > ./correctReads.000015.out 2>&1\n",
      "    ./correctReads.sh 16 > ./correctReads.000016.out 2>&1\n",
      "    ./correctReads.sh 17 > ./correctReads.000017.out 2>&1\n",
      "    ./correctReads.sh 18 > ./correctReads.000018.out 2>&1\n",
      "    ./correctReads.sh 19 > ./correctReads.000019.out 2>&1\n",
      "    ./correctReads.sh 20 > ./correctReads.000020.out 2>&1\n",
      "    ./correctReads.sh 21 > ./correctReads.000021.out 2>&1\n",
      "    ./correctReads.sh 22 > ./correctReads.000022.out 2>&1\n",
      "    ./correctReads.sh 23 > ./correctReads.000023.out 2>&1\n",
      "    ./correctReads.sh 24 > ./correctReads.000024.out 2>&1\n",
      "    ./correctReads.sh 25 > ./correctReads.000025.out 2>&1\n",
      "    ./correctReads.sh 26 > ./correctReads.000026.out 2>&1\n",
      "    ./correctReads.sh 27 > ./correctReads.000027.out 2>&1\n",
      "    ./correctReads.sh 28 > ./correctReads.000028.out 2>&1\n",
      "    ./correctReads.sh 29 > ./correctReads.000029.out 2>&1\n",
      "    ./correctReads.sh 30 > ./correctReads.000030.out 2>&1\n",
      "    ./correctReads.sh 31 > ./correctReads.000031.out 2>&1\n",
      "    ./correctReads.sh 32 > ./correctReads.000032.out 2>&1\n",
      "    ./correctReads.sh 33 > ./correctReads.000033.out 2>&1\n",
      "    ./correctReads.sh 34 > ./correctReads.000034.out 2>&1\n",
      "    ./correctReads.sh 35 > ./correctReads.000035.out 2>&1\n",
      "    ./correctReads.sh 36 > ./correctReads.000036.out 2>&1\n",
      "    ./correctReads.sh 37 > ./correctReads.000037.out 2>&1\n",
      "    ./correctReads.sh 38 > ./correctReads.000038.out 2>&1\n",
      "    ./correctReads.sh 39 > ./correctReads.000039.out 2>&1\n",
      "    ./correctReads.sh 40 > ./correctReads.000040.out 2>&1\n",
      "    ./correctReads.sh 41 > ./correctReads.000041.out 2>&1\n",
      "    ./correctReads.sh 42 > ./correctReads.000042.out 2>&1\n",
      "    ./correctReads.sh 43 > ./correctReads.000043.out 2>&1\n",
      "    ./correctReads.sh 44 > ./correctReads.000044.out 2>&1\n",
      "    ./correctReads.sh 45 > ./correctReads.000045.out 2>&1\n",
      "    ./correctReads.sh 46 > ./correctReads.000046.out 2>&1\n",
      "    ./correctReads.sh 47 > ./correctReads.000047.out 2>&1\n",
      "    ./correctReads.sh 48 > ./correctReads.000048.out 2>&1\n",
      "    ./correctReads.sh 49 > ./correctReads.000049.out 2>&1\n",
      "    ./correctReads.sh 50 > ./correctReads.000050.out 2>&1\n",
      "    ./correctReads.sh 51 > ./correctReads.000051.out 2>&1\n",
      "    ./correctReads.sh 52 > ./correctReads.000052.out 2>&1\n",
      "    ./correctReads.sh 53 > ./correctReads.000053.out 2>&1\n",
      "    ./correctReads.sh 54 > ./correctReads.000054.out 2>&1\n",
      "    ./correctReads.sh 55 > ./correctReads.000055.out 2>&1\n",
      "    ./correctReads.sh 56 > ./correctReads.000056.out 2>&1\n",
      "    ./correctReads.sh 57 > ./correctReads.000057.out 2>&1\n",
      "    ./correctReads.sh 58 > ./correctReads.000058.out 2>&1\n",
      "    ./correctReads.sh 59 > ./correctReads.000059.out 2>&1\n",
      "    ./correctReads.sh 60 > ./correctReads.000060.out 2>&1\n",
      "    ./correctReads.sh 61 > ./correctReads.000061.out 2>&1\n",
      "    ./correctReads.sh 62 > ./correctReads.000062.out 2>&1\n",
      "    ./correctReads.sh 63 > ./correctReads.000063.out 2>&1\n",
      "    ./correctReads.sh 64 > ./correctReads.000064.out 2>&1\n",
      "    ./correctReads.sh 65 > ./correctReads.000065.out 2>&1\n",
      "    ./correctReads.sh 66 > ./correctReads.000066.out 2>&1\n",
      "    ./correctReads.sh 67 > ./correctReads.000067.out 2>&1\n",
      "    ./correctReads.sh 68 > ./correctReads.000068.out 2>&1\n",
      "    ./correctReads.sh 69 > ./correctReads.000069.out 2>&1\n",
      "    ./correctReads.sh 70 > ./correctReads.000070.out 2>&1\n",
      "    ./correctReads.sh 71 > ./correctReads.000071.out 2>&1\n",
      "    ./correctReads.sh 72 > ./correctReads.000072.out 2>&1\n",
      "    ./correctReads.sh 73 > ./correctReads.000073.out 2>&1\n",
      "    ./correctReads.sh 74 > ./correctReads.000074.out 2>&1\n",
      "    ./correctReads.sh 75 > ./correctReads.000075.out 2>&1\n",
      "    ./correctReads.sh 76 > ./correctReads.000076.out 2>&1\n",
      "    ./correctReads.sh 77 > ./correctReads.000077.out 2>&1\n",
      "    ./correctReads.sh 78 > ./correctReads.000078.out 2>&1\n",
      "    ./correctReads.sh 79 > ./correctReads.000079.out 2>&1\n",
      "    ./correctReads.sh 80 > ./correctReads.000080.out 2>&1\n",
      "    ./correctReads.sh 81 > ./correctReads.000081.out 2>&1\n",
      "    ./correctReads.sh 82 > ./correctReads.000082.out 2>&1\n",
      "    ./correctReads.sh 83 > ./correctReads.000083.out 2>&1\n",
      "    ./correctReads.sh 84 > ./correctReads.000084.out 2>&1\n",
      "    ./correctReads.sh 85 > ./correctReads.000085.out 2>&1\n",
      "    ./correctReads.sh 86 > ./correctReads.000086.out 2>&1\n",
      "    ./correctReads.sh 87 > ./correctReads.000087.out 2>&1\n",
      "    ./correctReads.sh 88 > ./correctReads.000088.out 2>&1\n",
      "    ./correctReads.sh 89 > ./correctReads.000089.out 2>&1\n",
      "    ./correctReads.sh 90 > ./correctReads.000090.out 2>&1\n",
      "    ./correctReads.sh 91 > ./correctReads.000091.out 2>&1\n",
      "    ./correctReads.sh 92 > ./correctReads.000092.out 2>&1\n",
      "    ./correctReads.sh 93 > ./correctReads.000093.out 2>&1\n",
      "    ./correctReads.sh 94 > ./correctReads.000094.out 2>&1\n",
      "    ./correctReads.sh 95 > ./correctReads.000095.out 2>&1\n",
      "    ./correctReads.sh 96 > ./correctReads.000096.out 2>&1\n",
      "    ./correctReads.sh 97 > ./correctReads.000097.out 2>&1\n",
      "    ./correctReads.sh 98 > ./correctReads.000098.out 2>&1\n",
      "    ./correctReads.sh 99 > ./correctReads.000099.out 2>&1\n",
      "    ./correctReads.sh 100 > ./correctReads.000100.out 2>&1\n",
      "    ./correctReads.sh 101 > ./correctReads.000101.out 2>&1\n",
      "    ./correctReads.sh 102 > ./correctReads.000102.out 2>&1\n",
      "    ./correctReads.sh 103 > ./correctReads.000103.out 2>&1\n",
      "    ./correctReads.sh 104 > ./correctReads.000104.out 2>&1\n",
      "    ./correctReads.sh 105 > ./correctReads.000105.out 2>&1\n",
      "    ./correctReads.sh 106 > ./correctReads.000106.out 2>&1\n",
      "    ./correctReads.sh 107 > ./correctReads.000107.out 2>&1\n",
      "    ./correctReads.sh 108 > ./correctReads.000108.out 2>&1\n",
      "    ./correctReads.sh 109 > ./correctReads.000109.out 2>&1\n",
      "    ./correctReads.sh 110 > ./correctReads.000110.out 2>&1\n",
      "    ./correctReads.sh 111 > ./correctReads.000111.out 2>&1\n",
      "    ./correctReads.sh 112 > ./correctReads.000112.out 2>&1\n",
      "    ./correctReads.sh 113 > ./correctReads.000113.out 2>&1\n",
      "    ./correctReads.sh 114 > ./correctReads.000114.out 2>&1\n",
      "    ./correctReads.sh 115 > ./correctReads.000115.out 2>&1\n",
      "    ./correctReads.sh 116 > ./correctReads.000116.out 2>&1\n",
      "\n",
      "-- Finished on Fri Oct 20 11:44:54 2017 (16479 seconds) with 662.711 GB free disk space\n",
      "----------------------------------------\n",
      "-- Found 116 read correction output files.\n",
      "-- Finished stage 'cor-generateCorrectedReads', reset canuIteration.\n",
      "-- Found 116 read correction output files.\n",
      "-- Finished stage 'cor-generateCorrectedReads', reset canuIteration.\n",
      "-- Concatenating correctReads output.\n",
      "-- Analyzing correctReads output.\n",
      "--\n",
      "-- Reads to be corrected:\n",
      "--   1737978 reads longer than 0 bp\n",
      "--   12703460519 bp\n",
      "-- Expected corrected reads:\n",
      "--   1737978 reads\n",
      "--   10398007480 bp\n",
      "--   0 bp minimum length\n",
      "--   5983 bp mean length\n",
      "--   14867 bp n50 length\n",
      "-- Actual corrected reads:\n",
      "--   1665075 reads\n",
      "--   11661226162 bp\n",
      "--   72903 reads with 0 corrected blocks\n",
      "--   1665075 reads with 1 corrected block\n",
      "--\n",
      "-- Purging correctReads output after merging to final output file.\n",
      "-- Purged 116 .dump.success sentinels.\n",
      "-- Purged 116 .fasta outputs.\n",
      "-- Purged 116 .err outputs.\n",
      "-- Purged 116 .out job log outputs.\n",
      "-- Finished stage 'cor-dumpCorrectedReads', reset canuIteration.\n",
      "--\n",
      "-- Corrected reads saved in '20171018_oly_pacbio.correctedReads.fasta.gz'.\n",
      "--\n",
      "--\n",
      "-- BEGIN TRIMMING\n",
      "--\n",
      "----------------------------------------\n",
      "-- Starting command on Fri Oct 20 11:52:06 2017 with 837.718 GB free disk space\n",
      "\n",
      "    cd trimming\n",
      "    /usr/local/bioinformatics/canu-1.6/Linux-amd64/bin/gatekeeperCreate \\\n",
      "      -minlength 1000 \\\n",
      "      -o ./20171018_oly_pacbio.gkpStore.BUILDING \\\n",
      "      ./20171018_oly_pacbio.gkpStore.gkp \\\n",
      "    > ./20171018_oly_pacbio.gkpStore.BUILDING.err 2>&1\n",
      "\n",
      "-- Finished on Fri Oct 20 11:55:10 2017 (184 seconds) with 834.594 GB free disk space\n",
      "----------------------------------------\n",
      "--\n",
      "-- In gatekeeper store 'trimming/20171018_oly_pacbio.gkpStore':\n",
      "--   Found 1665075 reads.\n",
      "--   Found 11661226162 bases (6.13 times coverage).\n",
      "--\n",
      "--   Read length histogram (one '*' equals 3544.54 reads):\n",
      "--        0    999      0 \n",
      "--     1000   1999 132249 *************************************\n",
      "--     2000   2999 142073 ****************************************\n",
      "--     3000   3999 138655 ***************************************\n",
      "--     4000   4999 131179 *************************************\n",
      "--     5000   5999 188994 *****************************************************\n",
      "--     6000   6999 248118 **********************************************************************\n",
      "--     7000   7999 175810 *************************************************\n",
      "--     8000   8999 124541 ***********************************\n",
      "--     9000   9999  89076 *************************\n",
      "--    10000  10999  66103 ******************\n",
      "--    11000  11999  49617 *************\n",
      "--    12000  12999  38761 **********\n",
      "--    13000  13999  30214 ********\n",
      "--    14000  14999  23492 ******\n",
      "--    15000  15999  18097 *****\n",
      "--    16000  16999  14393 ****\n",
      "--    17000  17999  11353 ***\n",
      "--    18000  18999   8829 **\n",
      "--    19000  19999   7023 *\n",
      "--    20000  20999   5576 *\n",
      "--    21000  21999   4398 *\n",
      "--    22000  22999   3378 \n",
      "--    23000  23999   2685 \n",
      "--    24000  24999   2079 \n",
      "--    25000  25999   1628 \n",
      "--    26000  26999   1308 \n",
      "--    27000  27999   1075 \n",
      "--    28000  28999    838 \n",
      "--    29000  29999    704 \n",
      "--    30000  30999    543 \n",
      "--    31000  31999    456 \n",
      "--    32000  32999    365 \n",
      "--    33000  33999    275 \n",
      "--    34000  34999    201 \n",
      "--    35000  35999    169 \n",
      "--    36000  36999    153 \n",
      "--    37000  37999    124 \n",
      "--    38000  38999     98 \n",
      "--    39000  39999     90 \n",
      "--    40000  40999     75 \n",
      "--    41000  41999     61 \n",
      "--    42000  42999     44 \n",
      "--    43000  43999     39 \n",
      "--    44000  44999     31 \n",
      "--    45000  45999     19 \n",
      "--    46000  46999     16 \n",
      "--    47000  47999     13 \n",
      "--    48000  48999     18 \n",
      "--    49000  49999      8 \n",
      "--    50000  50999      5 \n",
      "--    51000  51999      2 \n",
      "--    52000  52999      6 \n",
      "--    53000  53999      6 \n",
      "--    54000  54999      2 \n",
      "--    55000  55999      4 \n",
      "--    56000  56999      2 \n",
      "--    57000  57999      2 \n",
      "--    58000  58999      1 \n",
      "--    59000  59999      0 \n",
      "--    60000  60999      0 \n",
      "--    61000  61999      0 \n",
      "--    62000  62999      0 \n",
      "--    63000  63999      1 \n",
      "-- Finished stage 'obt-gatekeeper', reset canuIteration.\n",
      "-- Finished stage 'merylConfigure', reset canuIteration.\n",
      "--\n",
      "-- Running jobs.  First attempt out of 2.\n",
      "----------------------------------------\n",
      "-- Starting 'meryl' concurrent execution on Fri Oct 20 11:55:42 2017 with 834.544 GB free disk space (1 processes; 1 concurrently)\n",
      "\n",
      "    cd trimming/0-mercounts\n",
      "    ./meryl.sh 1 > ./meryl.000001.out 2>&1\n",
      "\n",
      "-- Finished on Fri Oct 20 12:46:02 2017 (3020 seconds) with 790.643 GB free disk space\n",
      "----------------------------------------\n",
      "-- Meryl finished successfully.\n",
      "-- Finished stage 'merylCheck', reset canuIteration.\n",
      "--\n",
      "--  22-mers                                                                                           Fraction\n",
      "--    Occurrences   NumMers                                                                         Unique Total\n",
      "--       1-     1 7725326153 *******************************************************************--> 0.9280 0.6645\n",
      "--       2-     2 252935243 ********************************************************************** 0.9584 0.7080\n",
      "--       3-     4 154513749 ******************************************                             0.9701 0.7331\n",
      "--       5-     7  92684893 *************************                                              0.9817 0.7699\n",
      "--       8-    11  51684289 **************                                                         0.9902 0.8115\n",
      "--      12-    16  23204817 ******                                                                 0.9951 0.8472\n",
      "--      17-    22   9293416 **                                                                     0.9974 0.8706\n",
      "--      23-    29   4292264 *                                                                      0.9983 0.8840\n",
      "--      30-    37   2636301                                                                        0.9988 0.8928\n",
      "--      38-    46   1865506                                                                        0.9991 0.9001\n",
      "--      47-    56   1362364                                                                        0.9993 0.9066\n",
      "--      57-    67    985672                                                                        0.9994 0.9125\n",
      "--      68-    79    724229                                                                        0.9995 0.9176\n",
      "--      80-    92    554444                                                                        0.9996 0.9221\n",
      "--      93-   106    426303                                                                        0.9997 0.9261\n",
      "--     107-   121    335555                                                                        0.9997 0.9297\n",
      "--     122-   137    271808                                                                        0.9998 0.9330\n",
      "--     138-   154    225883                                                                        0.9998 0.9359\n",
      "--     155-   172    188849                                                                        0.9998 0.9388\n",
      "--     173-   191    152527                                                                        0.9999 0.9414\n",
      "--     192-   211    122845                                                                        0.9999 0.9437\n",
      "--     212-   232    101214                                                                        0.9999 0.9459\n",
      "--     233-   254     84206                                                                        0.9999 0.9478\n",
      "--     255-   277     71725                                                                        0.9999 0.9495\n",
      "--     278-   301     61832                                                                        0.9999 0.9512\n",
      "--     302-   326     53875                                                                        0.9999 0.9527\n",
      "--     327-   352     46016                                                                        0.9999 0.9541\n",
      "--     353-   379     41374                                                                        0.9999 0.9555\n",
      "--     380-   407     36957                                                                        0.9999 0.9568\n",
      "--     408-   436     32650                                                                        1.0000 0.9580\n",
      "--     437-   466     29408                                                                        1.0000 0.9592\n",
      "--     467-   497     27261                                                                        1.0000 0.9603\n",
      "--     498-   529     25266                                                                        1.0000 0.9615\n",
      "--     530-   562     23524                                                                        1.0000 0.9626\n",
      "--     563-   596     21810                                                                        1.0000 0.9637\n",
      "--     597-   631     19971                                                                        1.0000 0.9648\n",
      "--     632-   667     17723                                                                        1.0000 0.9658\n",
      "--     668-   704     16416                                                                        1.0000 0.9668\n",
      "--     705-   742     14590                                                                        1.0000 0.9678\n",
      "--     743-   781     12486                                                                        1.0000 0.9687\n",
      "--     782-   821     10855                                                                        1.0000 0.9695\n",
      "--\n",
      "--     2538138 (max occurrences)\n",
      "--  3900933434 (total mers, non-unique)\n",
      "--   599354004 (distinct mers, non-unique)\n",
      "--  7725326153 (unique mers)\n",
      "----------------------------------------\n",
      "-- Starting command on Fri Oct 20 12:46:04 2017 with 790.643 GB free disk space\n",
      "\n",
      "    cd trimming/0-mercounts\n",
      "    /usr/local/bioinformatics/canu-1.6/Linux-amd64/bin/meryl \\\n",
      "      -Dt \\\n",
      "      -n 642 \\\n",
      "      -s ./20171018_oly_pacbio.ms22 \\\n",
      "    > ./20171018_oly_pacbio.ms22.frequentMers.fasta \\\n",
      "    2> ./20171018_oly_pacbio.ms22.frequentMers.fasta.err\n",
      "\n",
      "-- Finished on Fri Oct 20 12:50:41 2017 (277 seconds) with 790.638 GB free disk space\n",
      "----------------------------------------\n",
      "-- Reset obtOvlMerThreshold from auto to 642.\n",
      "--\n",
      "-- Found 11626259587 22-mers; 8324680157 distinct and 7725326153 unique.  Largest count 2538138.\n",
      "-- Finished stage 'obt-meryl', reset canuIteration.\n",
      "--\n",
      "-- OVERLAPPER (normal) (trimming) erate=0.075\n",
      "--\n",
      "----------------------------------------\n",
      "-- Starting command on Fri Oct 20 12:50:46 2017 with 834.537 GB free disk space\n",
      "\n",
      "    cd trimming/1-overlapper\n",
      "    /usr/local/bioinformatics/canu-1.6/Linux-amd64/bin/overlapInCorePartition \\\n",
      "     -g  ../20171018_oly_pacbio.gkpStore \\\n",
      "     -bl 100000000 \\\n",
      "     -bs 0 \\\n",
      "     -rs 2000000 \\\n",
      "     -rl 0 \\\n",
      "     -ol 500 \\\n",
      "     -o  ./20171018_oly_pacbio.partition \\\n",
      "    > ./20171018_oly_pacbio.partition.err 2>&1\n",
      "\n",
      "-- Finished on Fri Oct 20 12:50:46 2017 (lickety-split) with 834.536 GB free disk space\n",
      "----------------------------------------\n",
      "--\n",
      "-- Configured 117 overlapInCore jobs.\n",
      "-- Finished stage 'obt-overlapConfigure', reset canuIteration.\n",
      "--\n",
      "-- Running jobs.  First attempt out of 2.\n",
      "----------------------------------------\n",
      "-- Starting 'obtovl' concurrent execution on Fri Oct 20 12:50:46 2017 with 834.536 GB free disk space (117 processes; 3 concurrently)\n",
      "\n",
      "    cd trimming/1-overlapper\n",
      "    ./overlap.sh 1 > ./overlap.000001.out 2>&1\n",
      "    ./overlap.sh 2 > ./overlap.000002.out 2>&1\n",
      "    ./overlap.sh 3 > ./overlap.000003.out 2>&1\n",
      "    ./overlap.sh 4 > ./overlap.000004.out 2>&1\n",
      "    ./overlap.sh 5 > ./overlap.000005.out 2>&1\n",
      "    ./overlap.sh 6 > ./overlap.000006.out 2>&1\n",
      "    ./overlap.sh 7 > ./overlap.000007.out 2>&1\n",
      "    ./overlap.sh 8 > ./overlap.000008.out 2>&1\n",
      "    ./overlap.sh 9 > ./overlap.000009.out 2>&1\n",
      "    ./overlap.sh 10 > ./overlap.000010.out 2>&1\n",
      "    ./overlap.sh 11 > ./overlap.000011.out 2>&1\n",
      "    ./overlap.sh 12 > ./overlap.000012.out 2>&1\n",
      "    ./overlap.sh 13 > ./overlap.000013.out 2>&1\n",
      "    ./overlap.sh 14 > ./overlap.000014.out 2>&1\n",
      "    ./overlap.sh 15 > ./overlap.000015.out 2>&1\n",
      "    ./overlap.sh 16 > ./overlap.000016.out 2>&1\n",
      "    ./overlap.sh 17 > ./overlap.000017.out 2>&1\n",
      "    ./overlap.sh 18 > ./overlap.000018.out 2>&1\n",
      "    ./overlap.sh 19 > ./overlap.000019.out 2>&1\n",
      "    ./overlap.sh 20 > ./overlap.000020.out 2>&1\n",
      "    ./overlap.sh 21 > ./overlap.000021.out 2>&1\n",
      "    ./overlap.sh 22 > ./overlap.000022.out 2>&1\n",
      "    ./overlap.sh 23 > ./overlap.000023.out 2>&1\n",
      "    ./overlap.sh 24 > ./overlap.000024.out 2>&1\n",
      "    ./overlap.sh 25 > ./overlap.000025.out 2>&1\n",
      "    ./overlap.sh 26 > ./overlap.000026.out 2>&1\n",
      "    ./overlap.sh 27 > ./overlap.000027.out 2>&1\n",
      "    ./overlap.sh 28 > ./overlap.000028.out 2>&1\n",
      "    ./overlap.sh 29 > ./overlap.000029.out 2>&1\n",
      "    ./overlap.sh 30 > ./overlap.000030.out 2>&1\n",
      "    ./overlap.sh 31 > ./overlap.000031.out 2>&1\n",
      "    ./overlap.sh 32 > ./overlap.000032.out 2>&1\n",
      "    ./overlap.sh 33 > ./overlap.000033.out 2>&1\n",
      "    ./overlap.sh 34 > ./overlap.000034.out 2>&1\n",
      "    ./overlap.sh 35 > ./overlap.000035.out 2>&1\n",
      "    ./overlap.sh 36 > ./overlap.000036.out 2>&1\n",
      "    ./overlap.sh 37 > ./overlap.000037.out 2>&1\n",
      "    ./overlap.sh 38 > ./overlap.000038.out 2>&1\n",
      "    ./overlap.sh 39 > ./overlap.000039.out 2>&1\n",
      "    ./overlap.sh 40 > ./overlap.000040.out 2>&1\n",
      "    ./overlap.sh 41 > ./overlap.000041.out 2>&1\n",
      "    ./overlap.sh 42 > ./overlap.000042.out 2>&1\n",
      "    ./overlap.sh 43 > ./overlap.000043.out 2>&1\n",
      "    ./overlap.sh 44 > ./overlap.000044.out 2>&1\n",
      "    ./overlap.sh 45 > ./overlap.000045.out 2>&1\n",
      "    ./overlap.sh 46 > ./overlap.000046.out 2>&1\n",
      "    ./overlap.sh 47 > ./overlap.000047.out 2>&1\n",
      "    ./overlap.sh 48 > ./overlap.000048.out 2>&1\n",
      "    ./overlap.sh 49 > ./overlap.000049.out 2>&1\n",
      "    ./overlap.sh 50 > ./overlap.000050.out 2>&1\n",
      "    ./overlap.sh 51 > ./overlap.000051.out 2>&1\n",
      "    ./overlap.sh 52 > ./overlap.000052.out 2>&1\n",
      "    ./overlap.sh 53 > ./overlap.000053.out 2>&1\n",
      "    ./overlap.sh 54 > ./overlap.000054.out 2>&1\n",
      "    ./overlap.sh 55 > ./overlap.000055.out 2>&1\n",
      "    ./overlap.sh 56 > ./overlap.000056.out 2>&1\n",
      "    ./overlap.sh 57 > ./overlap.000057.out 2>&1\n",
      "    ./overlap.sh 58 > ./overlap.000058.out 2>&1\n",
      "    ./overlap.sh 59 > ./overlap.000059.out 2>&1\n",
      "    ./overlap.sh 60 > ./overlap.000060.out 2>&1\n",
      "    ./overlap.sh 61 > ./overlap.000061.out 2>&1\n",
      "    ./overlap.sh 62 > ./overlap.000062.out 2>&1\n",
      "    ./overlap.sh 63 > ./overlap.000063.out 2>&1\n",
      "    ./overlap.sh 64 > ./overlap.000064.out 2>&1\n",
      "    ./overlap.sh 65 > ./overlap.000065.out 2>&1\n",
      "    ./overlap.sh 66 > ./overlap.000066.out 2>&1\n",
      "    ./overlap.sh 67 > ./overlap.000067.out 2>&1\n",
      "    ./overlap.sh 68 > ./overlap.000068.out 2>&1\n",
      "    ./overlap.sh 69 > ./overlap.000069.out 2>&1\n",
      "    ./overlap.sh 70 > ./overlap.000070.out 2>&1\n",
      "    ./overlap.sh 71 > ./overlap.000071.out 2>&1\n",
      "    ./overlap.sh 72 > ./overlap.000072.out 2>&1\n",
      "    ./overlap.sh 73 > ./overlap.000073.out 2>&1\n",
      "    ./overlap.sh 74 > ./overlap.000074.out 2>&1\n",
      "    ./overlap.sh 75 > ./overlap.000075.out 2>&1\n",
      "    ./overlap.sh 76 > ./overlap.000076.out 2>&1\n",
      "    ./overlap.sh 77 > ./overlap.000077.out 2>&1\n",
      "    ./overlap.sh 78 > ./overlap.000078.out 2>&1\n",
      "    ./overlap.sh 79 > ./overlap.000079.out 2>&1\n",
      "    ./overlap.sh 80 > ./overlap.000080.out 2>&1\n",
      "    ./overlap.sh 81 > ./overlap.000081.out 2>&1\n",
      "    ./overlap.sh 82 > ./overlap.000082.out 2>&1\n",
      "    ./overlap.sh 83 > ./overlap.000083.out 2>&1\n",
      "    ./overlap.sh 84 > ./overlap.000084.out 2>&1\n",
      "    ./overlap.sh 85 > ./overlap.000085.out 2>&1\n",
      "    ./overlap.sh 86 > ./overlap.000086.out 2>&1\n",
      "    ./overlap.sh 87 > ./overlap.000087.out 2>&1\n",
      "    ./overlap.sh 88 > ./overlap.000088.out 2>&1\n",
      "    ./overlap.sh 89 > ./overlap.000089.out 2>&1\n",
      "    ./overlap.sh 90 > ./overlap.000090.out 2>&1\n",
      "    ./overlap.sh 91 > ./overlap.000091.out 2>&1\n",
      "    ./overlap.sh 92 > ./overlap.000092.out 2>&1\n",
      "    ./overlap.sh 93 > ./overlap.000093.out 2>&1\n",
      "    ./overlap.sh 94 > ./overlap.000094.out 2>&1\n",
      "    ./overlap.sh 95 > ./overlap.000095.out 2>&1\n",
      "    ./overlap.sh 96 > ./overlap.000096.out 2>&1\n",
      "    ./overlap.sh 97 > ./overlap.000097.out 2>&1\n",
      "    ./overlap.sh 98 > ./overlap.000098.out 2>&1\n",
      "    ./overlap.sh 99 > ./overlap.000099.out 2>&1\n",
      "    ./overlap.sh 100 > ./overlap.000100.out 2>&1\n",
      "    ./overlap.sh 101 > ./overlap.000101.out 2>&1\n",
      "    ./overlap.sh 102 > ./overlap.000102.out 2>&1\n",
      "    ./overlap.sh 103 > ./overlap.000103.out 2>&1\n",
      "    ./overlap.sh 104 > ./overlap.000104.out 2>&1\n",
      "    ./overlap.sh 105 > ./overlap.000105.out 2>&1\n",
      "    ./overlap.sh 106 > ./overlap.000106.out 2>&1\n",
      "    ./overlap.sh 107 > ./overlap.000107.out 2>&1\n",
      "    ./overlap.sh 108 > ./overlap.000108.out 2>&1\n",
      "    ./overlap.sh 109 > ./overlap.000109.out 2>&1\n",
      "    ./overlap.sh 110 > ./overlap.000110.out 2>&1\n",
      "    ./overlap.sh 111 > ./overlap.000111.out 2>&1\n",
      "    ./overlap.sh 112 > ./overlap.000112.out 2>&1\n",
      "    ./overlap.sh 113 > ./overlap.000113.out 2>&1\n",
      "    ./overlap.sh 114 > ./overlap.000114.out 2>&1\n",
      "    ./overlap.sh 115 > ./overlap.000115.out 2>&1\n",
      "    ./overlap.sh 116 > ./overlap.000116.out 2>&1\n",
      "    ./overlap.sh 117 > ./overlap.000117.out 2>&1\n",
      "\n",
      "-- Finished on Fri Oct 20 19:54:56 2017 (25450 seconds) with 833.147 GB free disk space\n",
      "----------------------------------------\n",
      "-- Found 117 overlapInCore output files.\n",
      "--\n",
      "-- overlapInCore compute 'trimming/1-overlapper':\n",
      "--   kmer hits\n",
      "--     with no overlap       8395434118  47.1623932 +- 54925643.046\n",
      "--     with an overlap         53892335  .247863248 +- 416346.085\n",
      "--\n",
      "--   overlaps                  53890954  .444444444 +- 416335.571\n",
      "--     contained                      0           0 +- 0\n",
      "--     dovetail                       0           0 +- 0\n",
      "--\n",
      "--   overlaps rejected\n",
      "--     multiple per pair              0           0 +- 0\n",
      "--     bad short window               0           0 +- 0\n",
      "--     bad long window                0           0 +- 0\n",
      "-- Finished stage 'obt-overlapCheck', reset canuIteration.\n",
      "----------------------------------------\n",
      "-- Starting command on Fri Oct 20 19:54:56 2017 with 833.147 GB free disk space\n",
      "\n",
      "    cd trimming\n",
      "    /usr/local/bioinformatics/canu-1.6/Linux-amd64/bin/ovStoreBuild \\\n",
      "     -O ./20171018_oly_pacbio.ovlStore.BUILDING \\\n",
      "     -G ./20171018_oly_pacbio.gkpStore \\\n",
      "     -M 4-32 \\\n",
      "     -L ./1-overlapper/ovljob.files \\\n",
      "     > ./20171018_oly_pacbio.ovlStore.err 2>&1\n",
      "\n",
      "-- Finished on Fri Oct 20 19:55:45 2017 (49 seconds) with 831.101 GB free disk space\n",
      "----------------------------------------\n",
      "-- Checking store.\n",
      "----------------------------------------\n",
      "-- Starting command on Fri Oct 20 19:55:46 2017 with 831.101 GB free disk space\n",
      "\n",
      "    cd trimming\n",
      "    /usr/local/bioinformatics/canu-1.6/Linux-amd64/bin/ovStoreDump \\\n",
      "     -G ./20171018_oly_pacbio.gkpStore \\\n",
      "     -O ./20171018_oly_pacbio.ovlStore \\\n",
      "     -d -counts \\\n",
      "     > ./20171018_oly_pacbio.ovlStore/counts.dat 2> ./20171018_oly_pacbio.ovlStore/counts.err\n",
      "\n",
      "-- Finished on Fri Oct 20 19:55:46 2017 (1 second) with 831.086 GB free disk space\n",
      "----------------------------------------\n",
      "--\n",
      "-- Overlap store 'trimming/20171018_oly_pacbio.ovlStore' successfully constructed.\n",
      "-- Found 107784670 overlaps for 797225 reads; 867847 reads have no overlaps.\n",
      "--\n",
      "FILE trimming/1-overlapper/001/000001.ovb\n",
      "FILE trimming/1-overlapper/001/000002.ovb\n",
      "FILE trimming/1-overlapper/001/000003.ovb\n",
      "FILE trimming/1-overlapper/001/000004.ovb\n",
      "FILE trimming/1-overlapper/001/000005.ovb\n",
      "FILE trimming/1-overlapper/001/000006.ovb\n",
      "FILE trimming/1-overlapper/001/000007.ovb\n",
      "FILE trimming/1-overlapper/001/000008.ovb\n",
      "FILE trimming/1-overlapper/001/000009.ovb\n",
      "FILE trimming/1-overlapper/001/000010.ovb\n",
      "FILE trimming/1-overlapper/001/000011.ovb\n",
      "FILE trimming/1-overlapper/001/000012.ovb\n",
      "FILE trimming/1-overlapper/001/000013.ovb\n",
      "FILE trimming/1-overlapper/001/000014.ovb\n",
      "FILE trimming/1-overlapper/001/000015.ovb\n",
      "FILE trimming/1-overlapper/001/000016.ovb\n",
      "FILE trimming/1-overlapper/001/000017.ovb\n",
      "FILE trimming/1-overlapper/001/000018.ovb\n",
      "FILE trimming/1-overlapper/001/000019.ovb\n",
      "FILE trimming/1-overlapper/001/000020.ovb\n",
      "FILE trimming/1-overlapper/001/000021.ovb\n",
      "FILE trimming/1-overlapper/001/000022.ovb\n",
      "FILE trimming/1-overlapper/001/000023.ovb\n",
      "FILE trimming/1-overlapper/001/000024.ovb\n",
      "FILE trimming/1-overlapper/001/000025.ovb\n",
      "FILE trimming/1-overlapper/001/000026.ovb\n",
      "FILE trimming/1-overlapper/001/000027.ovb\n",
      "FILE trimming/1-overlapper/001/000028.ovb\n",
      "FILE trimming/1-overlapper/001/000029.ovb\n",
      "FILE trimming/1-overlapper/001/000030.ovb\n",
      "FILE trimming/1-overlapper/001/000031.ovb\n",
      "FILE trimming/1-overlapper/001/000032.ovb\n",
      "FILE trimming/1-overlapper/001/000033.ovb\n",
      "FILE trimming/1-overlapper/001/000034.ovb\n",
      "FILE trimming/1-overlapper/001/000035.ovb\n",
      "FILE trimming/1-overlapper/001/000036.ovb\n",
      "FILE trimming/1-overlapper/001/000037.ovb\n",
      "FILE trimming/1-overlapper/001/000038.ovb\n",
      "FILE trimming/1-overlapper/001/000039.ovb\n",
      "FILE trimming/1-overlapper/001/000040.ovb\n",
      "FILE trimming/1-overlapper/001/000041.ovb\n",
      "FILE trimming/1-overlapper/001/000042.ovb\n",
      "FILE trimming/1-overlapper/001/000043.ovb\n",
      "FILE trimming/1-overlapper/001/000044.ovb\n",
      "FILE trimming/1-overlapper/001/000045.ovb\n",
      "FILE trimming/1-overlapper/001/000046.ovb\n",
      "FILE trimming/1-overlapper/001/000047.ovb\n",
      "FILE trimming/1-overlapper/001/000048.ovb\n",
      "FILE trimming/1-overlapper/001/000049.ovb\n",
      "FILE trimming/1-overlapper/001/000050.ovb\n",
      "FILE trimming/1-overlapper/001/000051.ovb\n",
      "FILE trimming/1-overlapper/001/000052.ovb\n",
      "FILE trimming/1-overlapper/001/000053.ovb\n",
      "FILE trimming/1-overlapper/001/000054.ovb\n",
      "FILE trimming/1-overlapper/001/000055.ovb\n",
      "FILE trimming/1-overlapper/001/000056.ovb\n",
      "FILE trimming/1-overlapper/001/000057.ovb\n",
      "FILE trimming/1-overlapper/001/000058.ovb\n",
      "FILE trimming/1-overlapper/001/000059.ovb\n",
      "FILE trimming/1-overlapper/001/000060.ovb\n",
      "FILE trimming/1-overlapper/001/000061.ovb\n",
      "FILE trimming/1-overlapper/001/000062.ovb\n",
      "FILE trimming/1-overlapper/001/000063.ovb\n",
      "FILE trimming/1-overlapper/001/000064.ovb\n",
      "FILE trimming/1-overlapper/001/000065.ovb\n",
      "FILE trimming/1-overlapper/001/000066.ovb\n",
      "FILE trimming/1-overlapper/001/000067.ovb\n",
      "FILE trimming/1-overlapper/001/000068.ovb\n",
      "FILE trimming/1-overlapper/001/000069.ovb\n",
      "FILE trimming/1-overlapper/001/000070.ovb\n",
      "FILE trimming/1-overlapper/001/000071.ovb\n",
      "FILE trimming/1-overlapper/001/000072.ovb\n",
      "FILE trimming/1-overlapper/001/000073.ovb\n",
      "FILE trimming/1-overlapper/001/000074.ovb\n",
      "FILE trimming/1-overlapper/001/000075.ovb\n",
      "FILE trimming/1-overlapper/001/000076.ovb\n",
      "FILE trimming/1-overlapper/001/000077.ovb\n",
      "FILE trimming/1-overlapper/001/000078.ovb\n",
      "FILE trimming/1-overlapper/001/000079.ovb\n",
      "FILE trimming/1-overlapper/001/000080.ovb\n",
      "FILE trimming/1-overlapper/001/000081.ovb\n",
      "FILE trimming/1-overlapper/001/000082.ovb\n",
      "FILE trimming/1-overlapper/001/000083.ovb\n",
      "FILE trimming/1-overlapper/001/000084.ovb\n",
      "FILE trimming/1-overlapper/001/000085.ovb\n",
      "FILE trimming/1-overlapper/001/000086.ovb\n",
      "FILE trimming/1-overlapper/001/000087.ovb\n",
      "FILE trimming/1-overlapper/001/000088.ovb\n",
      "FILE trimming/1-overlapper/001/000089.ovb\n",
      "FILE trimming/1-overlapper/001/000090.ovb\n",
      "FILE trimming/1-overlapper/001/000091.ovb\n",
      "FILE trimming/1-overlapper/001/000092.ovb\n",
      "FILE trimming/1-overlapper/001/000093.ovb\n",
      "FILE trimming/1-overlapper/001/000094.ovb\n",
      "FILE trimming/1-overlapper/001/000095.ovb\n",
      "FILE trimming/1-overlapper/001/000096.ovb\n",
      "FILE trimming/1-overlapper/001/000097.ovb\n",
      "FILE trimming/1-overlapper/001/000098.ovb\n",
      "FILE trimming/1-overlapper/001/000099.ovb\n",
      "FILE trimming/1-overlapper/001/000100.ovb\n",
      "FILE trimming/1-overlapper/001/000101.ovb\n",
      "FILE trimming/1-overlapper/001/000102.ovb\n",
      "FILE trimming/1-overlapper/001/000103.ovb\n",
      "FILE trimming/1-overlapper/001/000104.ovb\n",
      "FILE trimming/1-overlapper/001/000105.ovb\n",
      "FILE trimming/1-overlapper/001/000106.ovb\n",
      "FILE trimming/1-overlapper/001/000107.ovb\n",
      "FILE trimming/1-overlapper/001/000108.ovb\n",
      "FILE trimming/1-overlapper/001/000109.ovb\n",
      "FILE trimming/1-overlapper/001/000110.ovb\n",
      "FILE trimming/1-overlapper/001/000111.ovb\n",
      "FILE trimming/1-overlapper/001/000112.ovb\n",
      "FILE trimming/1-overlapper/001/000113.ovb\n",
      "FILE trimming/1-overlapper/001/000114.ovb\n",
      "FILE trimming/1-overlapper/001/000115.ovb\n",
      "FILE trimming/1-overlapper/001/000116.ovb\n",
      "FILE trimming/1-overlapper/001/000117.ovb\n",
      "FILE trimming/1-overlapper/001/000001.stats\n",
      "FILE trimming/1-overlapper/001/000001.counts\n",
      "FILE trimming/1-overlapper/001/000002.stats\n",
      "FILE trimming/1-overlapper/001/000002.counts\n",
      "FILE trimming/1-overlapper/001/000003.stats\n",
      "FILE trimming/1-overlapper/001/000003.counts\n",
      "FILE trimming/1-overlapper/001/000004.stats\n",
      "FILE trimming/1-overlapper/001/000004.counts\n",
      "FILE trimming/1-overlapper/001/000005.stats\n",
      "FILE trimming/1-overlapper/001/000005.counts\n",
      "FILE trimming/1-overlapper/001/000006.stats\n",
      "FILE trimming/1-overlapper/001/000006.counts\n",
      "FILE trimming/1-overlapper/001/000007.stats\n",
      "FILE trimming/1-overlapper/001/000007.counts\n",
      "FILE trimming/1-overlapper/001/000008.stats\n",
      "FILE trimming/1-overlapper/001/000008.counts\n",
      "FILE trimming/1-overlapper/001/000009.stats\n",
      "FILE trimming/1-overlapper/001/000009.counts\n",
      "FILE trimming/1-overlapper/001/000010.stats\n",
      "FILE trimming/1-overlapper/001/000010.counts\n",
      "FILE trimming/1-overlapper/001/000011.stats\n",
      "FILE trimming/1-overlapper/001/000011.counts\n",
      "FILE trimming/1-overlapper/001/000012.stats\n",
      "FILE trimming/1-overlapper/001/000012.counts\n",
      "FILE trimming/1-overlapper/001/000013.stats\n",
      "FILE trimming/1-overlapper/001/000013.counts\n",
      "FILE trimming/1-overlapper/001/000014.stats\n",
      "FILE trimming/1-overlapper/001/000014.counts\n",
      "FILE trimming/1-overlapper/001/000015.stats\n",
      "FILE trimming/1-overlapper/001/000015.counts\n",
      "FILE trimming/1-overlapper/001/000016.stats\n",
      "FILE trimming/1-overlapper/001/000016.counts\n",
      "FILE trimming/1-overlapper/001/000017.stats\n",
      "FILE trimming/1-overlapper/001/000017.counts\n",
      "FILE trimming/1-overlapper/001/000018.stats\n",
      "FILE trimming/1-overlapper/001/000018.counts\n",
      "FILE trimming/1-overlapper/001/000019.stats\n",
      "FILE trimming/1-overlapper/001/000019.counts\n",
      "FILE trimming/1-overlapper/001/000020.stats\n",
      "FILE trimming/1-overlapper/001/000020.counts\n",
      "FILE trimming/1-overlapper/001/000021.stats\n",
      "FILE trimming/1-overlapper/001/000021.counts\n",
      "FILE trimming/1-overlapper/001/000022.stats\n",
      "FILE trimming/1-overlapper/001/000022.counts\n",
      "FILE trimming/1-overlapper/001/000023.stats\n",
      "FILE trimming/1-overlapper/001/000023.counts\n",
      "FILE trimming/1-overlapper/001/000024.stats\n",
      "FILE trimming/1-overlapper/001/000024.counts\n",
      "FILE trimming/1-overlapper/001/000025.stats\n",
      "FILE trimming/1-overlapper/001/000025.counts\n",
      "FILE trimming/1-overlapper/001/000026.stats\n",
      "FILE trimming/1-overlapper/001/000026.counts\n",
      "FILE trimming/1-overlapper/001/000027.stats\n",
      "FILE trimming/1-overlapper/001/000027.counts\n",
      "FILE trimming/1-overlapper/001/000028.stats\n",
      "FILE trimming/1-overlapper/001/000028.counts\n",
      "FILE trimming/1-overlapper/001/000029.stats\n",
      "FILE trimming/1-overlapper/001/000029.counts\n",
      "FILE trimming/1-overlapper/001/000030.stats\n",
      "FILE trimming/1-overlapper/001/000030.counts\n",
      "FILE trimming/1-overlapper/001/000031.stats\n",
      "FILE trimming/1-overlapper/001/000031.counts\n",
      "FILE trimming/1-overlapper/001/000032.stats\n",
      "FILE trimming/1-overlapper/001/000032.counts\n",
      "FILE trimming/1-overlapper/001/000033.stats\n",
      "FILE trimming/1-overlapper/001/000033.counts\n",
      "FILE trimming/1-overlapper/001/000034.stats\n",
      "FILE trimming/1-overlapper/001/000034.counts\n",
      "FILE trimming/1-overlapper/001/000035.stats\n",
      "FILE trimming/1-overlapper/001/000035.counts\n",
      "FILE trimming/1-overlapper/001/000036.stats\n",
      "FILE trimming/1-overlapper/001/000036.counts\n",
      "FILE trimming/1-overlapper/001/000037.stats\n",
      "FILE trimming/1-overlapper/001/000037.counts\n",
      "FILE trimming/1-overlapper/001/000038.stats\n",
      "FILE trimming/1-overlapper/001/000038.counts\n",
      "FILE trimming/1-overlapper/001/000039.stats\n",
      "FILE trimming/1-overlapper/001/000039.counts\n",
      "FILE trimming/1-overlapper/001/000040.stats\n",
      "FILE trimming/1-overlapper/001/000040.counts\n",
      "FILE trimming/1-overlapper/001/000041.stats\n",
      "FILE trimming/1-overlapper/001/000041.counts\n",
      "FILE trimming/1-overlapper/001/000042.stats\n",
      "FILE trimming/1-overlapper/001/000042.counts\n",
      "FILE trimming/1-overlapper/001/000043.stats\n",
      "FILE trimming/1-overlapper/001/000043.counts\n",
      "FILE trimming/1-overlapper/001/000044.stats\n",
      "FILE trimming/1-overlapper/001/000044.counts\n",
      "FILE trimming/1-overlapper/001/000045.stats\n",
      "FILE trimming/1-overlapper/001/000045.counts\n",
      "FILE trimming/1-overlapper/001/000046.stats\n",
      "FILE trimming/1-overlapper/001/000046.counts\n",
      "FILE trimming/1-overlapper/001/000047.stats\n",
      "FILE trimming/1-overlapper/001/000047.counts\n",
      "FILE trimming/1-overlapper/001/000048.stats\n",
      "FILE trimming/1-overlapper/001/000048.counts\n",
      "FILE trimming/1-overlapper/001/000049.stats\n",
      "FILE trimming/1-overlapper/001/000049.counts\n",
      "FILE trimming/1-overlapper/001/000050.stats\n",
      "FILE trimming/1-overlapper/001/000050.counts\n",
      "FILE trimming/1-overlapper/001/000051.stats\n",
      "FILE trimming/1-overlapper/001/000051.counts\n",
      "FILE trimming/1-overlapper/001/000052.stats\n",
      "FILE trimming/1-overlapper/001/000052.counts\n",
      "FILE trimming/1-overlapper/001/000053.stats\n",
      "FILE trimming/1-overlapper/001/000053.counts\n",
      "FILE trimming/1-overlapper/001/000054.stats\n",
      "FILE trimming/1-overlapper/001/000054.counts\n",
      "FILE trimming/1-overlapper/001/000055.stats\n",
      "FILE trimming/1-overlapper/001/000055.counts\n",
      "FILE trimming/1-overlapper/001/000056.stats\n",
      "FILE trimming/1-overlapper/001/000056.counts\n",
      "FILE trimming/1-overlapper/001/000057.stats\n",
      "FILE trimming/1-overlapper/001/000057.counts\n",
      "FILE trimming/1-overlapper/001/000058.stats\n",
      "FILE trimming/1-overlapper/001/000058.counts\n",
      "FILE trimming/1-overlapper/001/000059.stats\n",
      "FILE trimming/1-overlapper/001/000059.counts\n",
      "FILE trimming/1-overlapper/001/000060.stats\n",
      "FILE trimming/1-overlapper/001/000060.counts\n",
      "FILE trimming/1-overlapper/001/000061.stats\n",
      "FILE trimming/1-overlapper/001/000061.counts\n",
      "FILE trimming/1-overlapper/001/000062.stats\n",
      "FILE trimming/1-overlapper/001/000062.counts\n",
      "FILE trimming/1-overlapper/001/000063.stats\n",
      "FILE trimming/1-overlapper/001/000063.counts\n",
      "FILE trimming/1-overlapper/001/000064.stats\n",
      "FILE trimming/1-overlapper/001/000064.counts\n",
      "FILE trimming/1-overlapper/001/000065.stats\n",
      "FILE trimming/1-overlapper/001/000065.counts\n",
      "FILE trimming/1-overlapper/001/000066.stats\n",
      "FILE trimming/1-overlapper/001/000066.counts\n",
      "FILE trimming/1-overlapper/001/000067.stats\n",
      "FILE trimming/1-overlapper/001/000067.counts\n",
      "FILE trimming/1-overlapper/001/000068.stats\n",
      "FILE trimming/1-overlapper/001/000068.counts\n",
      "FILE trimming/1-overlapper/001/000069.stats\n",
      "FILE trimming/1-overlapper/001/000069.counts\n",
      "FILE trimming/1-overlapper/001/000070.stats\n",
      "FILE trimming/1-overlapper/001/000070.counts\n",
      "FILE trimming/1-overlapper/001/000071.stats\n",
      "FILE trimming/1-overlapper/001/000071.counts\n",
      "FILE trimming/1-overlapper/001/000072.stats\n",
      "FILE trimming/1-overlapper/001/000072.counts\n",
      "FILE trimming/1-overlapper/001/000073.stats\n",
      "FILE trimming/1-overlapper/001/000073.counts\n",
      "FILE trimming/1-overlapper/001/000074.stats\n",
      "FILE trimming/1-overlapper/001/000074.counts\n",
      "FILE trimming/1-overlapper/001/000075.stats\n",
      "FILE trimming/1-overlapper/001/000075.counts\n",
      "FILE trimming/1-overlapper/001/000076.stats\n",
      "FILE trimming/1-overlapper/001/000076.counts\n",
      "FILE trimming/1-overlapper/001/000077.stats\n",
      "FILE trimming/1-overlapper/001/000077.counts\n",
      "FILE trimming/1-overlapper/001/000078.stats\n",
      "FILE trimming/1-overlapper/001/000078.counts\n",
      "FILE trimming/1-overlapper/001/000079.stats\n",
      "FILE trimming/1-overlapper/001/000079.counts\n",
      "FILE trimming/1-overlapper/001/000080.stats\n",
      "FILE trimming/1-overlapper/001/000080.counts\n",
      "FILE trimming/1-overlapper/001/000081.stats\n",
      "FILE trimming/1-overlapper/001/000081.counts\n",
      "FILE trimming/1-overlapper/001/000082.stats\n",
      "FILE trimming/1-overlapper/001/000082.counts\n",
      "FILE trimming/1-overlapper/001/000083.stats\n",
      "FILE trimming/1-overlapper/001/000083.counts\n",
      "FILE trimming/1-overlapper/001/000084.stats\n",
      "FILE trimming/1-overlapper/001/000084.counts\n",
      "FILE trimming/1-overlapper/001/000085.stats\n",
      "FILE trimming/1-overlapper/001/000085.counts\n",
      "FILE trimming/1-overlapper/001/000086.stats\n",
      "FILE trimming/1-overlapper/001/000086.counts\n",
      "FILE trimming/1-overlapper/001/000087.stats\n",
      "FILE trimming/1-overlapper/001/000087.counts\n",
      "FILE trimming/1-overlapper/001/000088.stats\n",
      "FILE trimming/1-overlapper/001/000088.counts\n",
      "FILE trimming/1-overlapper/001/000089.stats\n",
      "FILE trimming/1-overlapper/001/000089.counts\n",
      "FILE trimming/1-overlapper/001/000090.stats\n",
      "FILE trimming/1-overlapper/001/000090.counts\n",
      "FILE trimming/1-overlapper/001/000091.stats\n",
      "FILE trimming/1-overlapper/001/000091.counts\n",
      "FILE trimming/1-overlapper/001/000092.stats\n",
      "FILE trimming/1-overlapper/001/000092.counts\n",
      "FILE trimming/1-overlapper/001/000093.stats\n",
      "FILE trimming/1-overlapper/001/000093.counts\n",
      "FILE trimming/1-overlapper/001/000094.stats\n",
      "FILE trimming/1-overlapper/001/000094.counts\n",
      "FILE trimming/1-overlapper/001/000095.stats\n",
      "FILE trimming/1-overlapper/001/000095.counts\n",
      "FILE trimming/1-overlapper/001/000096.stats\n",
      "FILE trimming/1-overlapper/001/000096.counts\n",
      "FILE trimming/1-overlapper/001/000097.stats\n",
      "FILE trimming/1-overlapper/001/000097.counts\n",
      "FILE trimming/1-overlapper/001/000098.stats\n",
      "FILE trimming/1-overlapper/001/000098.counts\n",
      "FILE trimming/1-overlapper/001/000099.stats\n",
      "FILE trimming/1-overlapper/001/000099.counts\n",
      "FILE trimming/1-overlapper/001/000100.stats\n",
      "FILE trimming/1-overlapper/001/000100.counts\n",
      "FILE trimming/1-overlapper/001/000101.stats\n",
      "FILE trimming/1-overlapper/001/000101.counts\n",
      "FILE trimming/1-overlapper/001/000102.stats\n",
      "FILE trimming/1-overlapper/001/000102.counts\n",
      "FILE trimming/1-overlapper/001/000103.stats\n",
      "FILE trimming/1-overlapper/001/000103.counts\n",
      "FILE trimming/1-overlapper/001/000104.stats\n",
      "FILE trimming/1-overlapper/001/000104.counts\n",
      "FILE trimming/1-overlapper/001/000105.stats\n",
      "FILE trimming/1-overlapper/001/000105.counts\n",
      "FILE trimming/1-overlapper/001/000106.stats\n",
      "FILE trimming/1-overlapper/001/000106.counts\n",
      "FILE trimming/1-overlapper/001/000107.stats\n",
      "FILE trimming/1-overlapper/001/000107.counts\n",
      "FILE trimming/1-overlapper/001/000108.stats\n",
      "FILE trimming/1-overlapper/001/000108.counts\n",
      "FILE trimming/1-overlapper/001/000109.stats\n",
      "FILE trimming/1-overlapper/001/000109.counts\n",
      "FILE trimming/1-overlapper/001/000110.stats\n",
      "FILE trimming/1-overlapper/001/000110.counts\n",
      "FILE trimming/1-overlapper/001/000111.stats\n",
      "FILE trimming/1-overlapper/001/000111.counts\n",
      "FILE trimming/1-overlapper/001/000112.stats\n",
      "FILE trimming/1-overlapper/001/000112.counts\n",
      "FILE trimming/1-overlapper/001/000113.stats\n",
      "FILE trimming/1-overlapper/001/000113.counts\n",
      "FILE trimming/1-overlapper/001/000114.stats\n",
      "FILE trimming/1-overlapper/001/000114.counts\n",
      "FILE trimming/1-overlapper/001/000115.stats\n",
      "FILE trimming/1-overlapper/001/000115.counts\n",
      "FILE trimming/1-overlapper/001/000116.stats\n",
      "FILE trimming/1-overlapper/001/000116.counts\n",
      "FILE trimming/1-overlapper/001/000117.stats\n",
      "FILE trimming/1-overlapper/001/000117.counts\n",
      "--\n",
      "-- Purged 1.387 GB in 351 overlap output files.\n",
      "-- Overlap store 'trimming/20171018_oly_pacbio.ovlStore' statistics not available (skipped in correction and trimming stages).\n",
      "--\n",
      "-- Finished stage 'obt-createOverlapStore', reset canuIteration.\n",
      "----------------------------------------\n",
      "-- Starting command on Fri Oct 20 19:55:49 2017 with 832.49 GB free disk space\n",
      "\n",
      "    cd trimming/3-overlapbasedtrimming\n",
      "    /usr/local/bioinformatics/canu-1.6/Linux-amd64/bin/trimReads \\\n",
      "      -G  ../20171018_oly_pacbio.gkpStore \\\n",
      "      -O  ../20171018_oly_pacbio.ovlStore \\\n",
      "      -Co ./20171018_oly_pacbio.1.trimReads.clear \\\n",
      "      -e  0.075 \\\n",
      "      -minlength 1000 \\\n",
      "      -ol 1 \\\n",
      "      -oc 1 \\\n",
      "      -o  ./20171018_oly_pacbio.1.trimReads \\\n",
      "    >     ./20171018_oly_pacbio.1.trimReads.err 2>&1\n",
      "\n",
      "-- Finished on Fri Oct 20 19:56:34 2017 (45 seconds) with 832.388 GB free disk space\n",
      "----------------------------------------\n",
      "--  PARAMETERS:\n",
      "--  ----------\n",
      "--     1000    (reads trimmed below this many bases are deleted)\n",
      "--   0.0750    (use overlaps at or below this fraction error)\n",
      "--        1    (break region if overlap is less than this long, for 'largest covered' algorithm)\n",
      "--        1    (break region if overlap coverage is less than this many read, for 'largest covered' algorithm)\n",
      "--  \n",
      "--  INPUT READS:\n",
      "--  -----------\n",
      "--  1665075 reads  11661226162 bases (reads processed)\n",
      "--       0 reads            0 bases (reads not processed, previously deleted)\n",
      "--       0 reads            0 bases (reads not processed, in a library where trimming isn't allowed)\n",
      "--  \n",
      "--  OUTPUT READS:\n",
      "--  ------------\n",
      "--  712481 reads   3458657537 bases (trimmed reads output)\n",
      "--   10785 reads     59293896 bases (reads with no change, kept as is)\n",
      "--  867850 reads   5755379456 bases (reads with no overlaps, deleted)\n",
      "--   73959 reads    499189010 bases (reads with short trimmed length, deleted)\n",
      "--  \n",
      "--  TRIMMING DETAILS:\n",
      "--  ----------------\n",
      "--  639125 reads   1149862364 bases (bases trimmed from the 5' end of a read)\n",
      "--  686805 reads    738843899 bases (bases trimmed from the 3' end of a read)\n",
      "-- Finished stage 'obt-trimReads', reset canuIteration.\n",
      "----------------------------------------\n",
      "-- Starting command on Fri Oct 20 19:56:34 2017 with 832.388 GB free disk space\n",
      "\n",
      "    cd trimming/3-overlapbasedtrimming\n",
      "    /usr/local/bioinformatics/canu-1.6/Linux-amd64/bin/splitReads \\\n",
      "      -G  ../20171018_oly_pacbio.gkpStore \\\n",
      "      -O  ../20171018_oly_pacbio.ovlStore \\\n",
      "      -Ci ./20171018_oly_pacbio.1.trimReads.clear \\\n",
      "      -Co ./20171018_oly_pacbio.2.splitReads.clear \\\n",
      "      -e  0.075 \\\n",
      "      -minlength 1000 \\\n",
      "      -o  ./20171018_oly_pacbio.2.splitReads \\\n",
      "    >     ./20171018_oly_pacbio.2.splitReads.err 2>&1\n",
      "\n",
      "-- Finished on Fri Oct 20 19:57:06 2017 (32 seconds) with 832.375 GB free disk space\n",
      "----------------------------------------\n",
      "--  PARAMETERS:\n",
      "--  ----------\n",
      "--     1000    (reads trimmed below this many bases are deleted)\n",
      "--   0.0750    (use overlaps at or below this fraction error)\n",
      "--  INPUT READS:\n",
      "--  -----------\n",
      "--  723266 reads   5406657696 bases (reads processed)\n",
      "--  941809 reads   6254568466 bases (reads not processed, previously deleted)\n",
      "--       0 reads            0 bases (reads not processed, in a library where trimming isn't allowed)\n",
      "--  \n",
      "--  PROCESSED:\n",
      "--  --------\n",
      "--       0 reads            0 bases (no overlaps)\n",
      "--    3211 reads     24048944 bases (no coverage after adjusting for trimming done already)\n",
      "--       0 reads            0 bases (processed for chimera)\n",
      "--       0 reads            0 bases (processed for spur)\n",
      "--  720055 reads   5382608752 bases (processed for subreads)\n",
      "--  \n",
      "--  READS WITH SIGNALS:\n",
      "--  ------------------\n",
      "--       0 reads            0 signals (number of 5' spur signal)\n",
      "--       0 reads            0 signals (number of 3' spur signal)\n",
      "--       0 reads            0 signals (number of chimera signal)\n",
      "--    1289 reads         1310 signals (number of subread signal)\n",
      "--  \n",
      "--  SIGNALS:\n",
      "--  -------\n",
      "--       0 reads            0 bases (size of 5' spur signal)\n",
      "--       0 reads            0 bases (size of 3' spur signal)\n",
      "--       0 reads            0 bases (size of chimera signal)\n",
      "--    1310 reads       418184 bases (size of subread signal)\n",
      "--  \n",
      "--  TRIMMING:\n",
      "--  --------\n",
      "--     655 reads      2190728 bases (trimmed from the 5' end of the read)\n",
      "--     637 reads      2255145 bases (trimmed from the 3' end of the read)\n",
      "-- Finished stage 'obt-splitReads', reset canuIteration.\n",
      "----------------------------------------\n",
      "-- Starting command on Fri Oct 20 19:57:07 2017 with 832.375 GB free disk space\n",
      "\n",
      "    cd trimming\n",
      "    /usr/local/bioinformatics/canu-1.6/Linux-amd64/bin/gatekeeperDumpFASTQ -fasta -nolibname \\\n",
      "      -G ./20171018_oly_pacbio.gkpStore \\\n",
      "      -c ./3-overlapbasedtrimming/20171018_oly_pacbio.2.splitReads.clear \\\n",
      "      -o ../20171018_oly_pacbio.trimmedReads.gz \\\n",
      "    >    ../20171018_oly_pacbio.trimmedReads.err 2>&1\n",
      "\n",
      "-- Finished on Fri Oct 20 19:58:19 2017 (72 seconds) with 831.247 GB free disk space\n",
      "----------------------------------------\n",
      "-- Finished stage 'obt-dumpReads', reset canuIteration.\n",
      "--\n",
      "-- Trimmed reads saved in 'trimming/20171018_oly_pacbio.trimmedReads.fasta.gz'\n",
      "--\n",
      "--\n",
      "-- BEGIN ASSEMBLY\n",
      "--\n",
      "----------------------------------------\n",
      "-- Starting command on Fri Oct 20 19:58:20 2017 with 833.292 GB free disk space\n",
      "\n",
      "    cd unitigging\n",
      "    /usr/local/bioinformatics/canu-1.6/Linux-amd64/bin/gatekeeperCreate \\\n",
      "      -minlength 1000 \\\n",
      "      -o ./20171018_oly_pacbio.gkpStore.BUILDING \\\n",
      "      ./20171018_oly_pacbio.gkpStore.gkp \\\n",
      "    > ./20171018_oly_pacbio.gkpStore.BUILDING.err 2>&1\n",
      "\n",
      "-- Finished on Fri Oct 20 19:59:18 2017 (58 seconds) with 832.265 GB free disk space\n",
      "----------------------------------------\n",
      "--\n",
      "-- In gatekeeper store 'unitigging/20171018_oly_pacbio.gkpStore':\n",
      "--   Found 723260 reads.\n",
      "--   Found 3513500524 bases (1.84 times coverage).\n",
      "--\n",
      "--   Read length histogram (one '*' equals 1701.81 reads):\n",
      "--        0    999      0 \n",
      "--     1000   1999 112782 ******************************************************************\n",
      "--     2000   2999  91499 *****************************************************\n",
      "--     3000   3999  83298 ************************************************\n",
      "--     4000   4999  78185 *********************************************\n",
      "--     5000   5999 119127 **********************************************************************\n",
      "--     6000   6999 114487 *******************************************************************\n",
      "--     7000   7999  57597 *********************************\n",
      "--     8000   8999  29853 *****************\n",
      "--     9000   9999  16050 *********\n",
      "--    10000  10999   8856 *****\n",
      "--    11000  11999   4946 **\n",
      "--    12000  12999   2980 *\n",
      "--    13000  13999   1555 \n",
      "--    14000  14999    900 \n",
      "--    15000  15999    476 \n",
      "--    16000  16999    284 \n",
      "--    17000  17999    178 \n",
      "--    18000  18999     79 \n",
      "--    19000  19999     60 \n",
      "--    20000  20999     29 \n",
      "--    21000  21999     13 \n",
      "--    22000  22999      9 \n",
      "--    23000  23999      2 \n",
      "--    24000  24999      9 \n",
      "--    25000  25999      3 \n",
      "--    26000  26999      2 \n",
      "--    27000  27999      0 \n",
      "--    28000  28999      0 \n",
      "--    29000  29999      0 \n",
      "--    30000  30999      0 \n",
      "--    31000  31999      0 \n",
      "--    32000  32999      1 \n",
      "-- Finished stage 'utg-gatekeeper', reset canuIteration.\n",
      "-- Finished stage 'merylConfigure', reset canuIteration.\n",
      "--\n",
      "-- Running jobs.  First attempt out of 2.\n",
      "----------------------------------------\n",
      "-- Starting 'meryl' concurrent execution on Fri Oct 20 19:59:28 2017 with 832.244 GB free disk space (1 processes; 1 concurrently)\n",
      "\n",
      "    cd unitigging/0-mercounts\n",
      "    ./meryl.sh 1 > ./meryl.000001.out 2>&1\n",
      "\n",
      "-- Finished on Fri Oct 20 20:05:41 2017 (373 seconds) with 826.267 GB free disk space\n",
      "----------------------------------------\n",
      "-- Meryl finished successfully.\n",
      "-- Finished stage 'merylCheck', reset canuIteration.\n",
      "--\n",
      "--  22-mers                                                                                           Fraction\n",
      "--    Occurrences   NumMers                                                                         Unique Total\n",
      "--       1-     1 725696616 *******************************************************************--> 0.6568 0.2074\n",
      "--       2-     2 123437132 ********************************************************************** 0.7685 0.2780\n",
      "--       3-     4 104020258 **********************************************************             0.8253 0.3319\n",
      "--       5-     7  73227892 *****************************************                              0.8905 0.4230\n",
      "--       8-    11  42101119 ***********************                                                0.9422 0.5347\n",
      "--      12-    16  18520505 **********                                                             0.9720 0.6308\n",
      "--      17-    22   6981818 ***                                                                    0.9854 0.6923\n",
      "--      23-    29   3099691 *                                                                      0.9906 0.7255\n",
      "--      30-    37   1925672 *                                                                      0.9932 0.7467\n",
      "--      38-    46   1384781                                                                        0.9948 0.7644\n",
      "--      47-    56   1004329                                                                        0.9960 0.7805\n",
      "--      57-    67    724229                                                                        0.9969 0.7949\n",
      "--      68-    79    531154                                                                        0.9975 0.8073\n",
      "--      80-    92    405947                                                                        0.9980 0.8183\n",
      "--      93-   106    312986                                                                        0.9983 0.8281\n",
      "--     107-   121    247423                                                                        0.9986 0.8368\n",
      "--     122-   137    200424                                                                        0.9988 0.8448\n",
      "--     138-   154    168840                                                                        0.9990 0.8521\n",
      "--     155-   172    138460                                                                        0.9991 0.8591\n",
      "--     173-   191    106811                                                                        0.9993 0.8655\n",
      "--     192-   211     85471                                                                        0.9994 0.8710\n",
      "--     212-   232     67771                                                                        0.9994 0.8758\n",
      "--     233-   254     58104                                                                        0.9995 0.8801\n",
      "--     255-   277     49886                                                                        0.9995 0.8841\n",
      "--     278-   301     43767                                                                        0.9996 0.8879\n",
      "--     302-   326     37472                                                                        0.9996 0.8915\n",
      "--     327-   352     33145                                                                        0.9997 0.8948\n",
      "--     353-   379     29659                                                                        0.9997 0.8980\n",
      "--     380-   407     26130                                                                        0.9997 0.9011\n",
      "--     408-   436     23742                                                                        0.9997 0.9041\n",
      "--     437-   466     22165                                                                        0.9998 0.9069\n",
      "--     467-   497     20683                                                                        0.9998 0.9098\n",
      "--     498-   529     19528                                                                        0.9998 0.9126\n",
      "--     530-   562     18025                                                                        0.9998 0.9155\n",
      "--     563-   596     16088                                                                        0.9998 0.9183\n",
      "--     597-   631     14130                                                                        0.9999 0.9209\n",
      "--     632-   667     12734                                                                        0.9999 0.9234\n",
      "--     668-   704     10497                                                                        0.9999 0.9258\n",
      "--     705-   742      8933                                                                        0.9999 0.9278\n",
      "--     743-   781      7716                                                                        0.9999 0.9297\n",
      "--     782-   821      6982                                                                        0.9999 0.9313\n",
      "--\n",
      "--     1180657 (max occurrences)\n",
      "--  2772615448 (total mers, non-unique)\n",
      "--   379253455 (distinct mers, non-unique)\n",
      "--   725696616 (unique mers)\n",
      "----------------------------------------\n",
      "-- Starting command on Fri Oct 20 20:05:41 2017 with 826.267 GB free disk space\n",
      "\n",
      "    cd unitigging/0-mercounts\n",
      "    /usr/local/bioinformatics/canu-1.6/Linux-amd64/bin/meryl \\\n",
      "      -Dt \\\n",
      "      -n 574 \\\n",
      "      -s ./20171018_oly_pacbio.ms22 \\\n",
      "    > ./20171018_oly_pacbio.ms22.frequentMers.fasta \\\n",
      "    2> ./20171018_oly_pacbio.ms22.frequentMers.fasta.err\n",
      "\n",
      "-- Finished on Fri Oct 20 20:06:11 2017 (30 seconds) with 826.263 GB free disk space\n",
      "----------------------------------------\n",
      "-- Reset utgOvlMerThreshold from auto to 574.\n",
      "--\n",
      "-- Found 3498312064 22-mers; 1104950071 distinct and 725696616 unique.  Largest count 1180657.\n",
      "-- Finished stage 'utg-meryl', reset canuIteration.\n",
      "--\n",
      "-- OVERLAPPER (normal) (assembly) erate=0.075\n",
      "--\n",
      "----------------------------------------\n",
      "-- Starting command on Fri Oct 20 20:06:11 2017 with 832.239 GB free disk space\n",
      "\n",
      "    cd unitigging/1-overlapper\n",
      "    /usr/local/bioinformatics/canu-1.6/Linux-amd64/bin/overlapInCorePartition \\\n",
      "     -g  ../20171018_oly_pacbio.gkpStore \\\n",
      "     -bl 100000000 \\\n",
      "     -bs 0 \\\n",
      "     -rs 2000000 \\\n",
      "     -rl 0 \\\n",
      "     -ol 500 \\\n",
      "     -o  ./20171018_oly_pacbio.partition \\\n",
      "    > ./20171018_oly_pacbio.partition.err 2>&1\n",
      "\n",
      "-- Finished on Fri Oct 20 20:06:11 2017 (lickety-split) with 832.239 GB free disk space\n",
      "----------------------------------------\n",
      "--\n",
      "-- Configured 36 overlapInCore jobs.\n",
      "-- Finished stage 'utg-overlapConfigure', reset canuIteration.\n",
      "--\n",
      "-- Running jobs.  First attempt out of 2.\n",
      "----------------------------------------\n",
      "-- Starting 'utgovl' concurrent execution on Fri Oct 20 20:06:11 2017 with 832.239 GB free disk space (36 processes; 3 concurrently)\n",
      "\n",
      "    cd unitigging/1-overlapper\n",
      "    ./overlap.sh 1 > ./overlap.000001.out 2>&1\n",
      "    ./overlap.sh 2 > ./overlap.000002.out 2>&1\n",
      "    ./overlap.sh 3 > ./overlap.000003.out 2>&1\n",
      "    ./overlap.sh 4 > ./overlap.000004.out 2>&1\n",
      "    ./overlap.sh 5 > ./overlap.000005.out 2>&1\n",
      "    ./overlap.sh 6 > ./overlap.000006.out 2>&1\n",
      "    ./overlap.sh 7 > ./overlap.000007.out 2>&1\n",
      "    ./overlap.sh 8 > ./overlap.000008.out 2>&1\n",
      "    ./overlap.sh 9 > ./overlap.000009.out 2>&1\n",
      "    ./overlap.sh 10 > ./overlap.000010.out 2>&1\n",
      "    ./overlap.sh 11 > ./overlap.000011.out 2>&1\n",
      "    ./overlap.sh 12 > ./overlap.000012.out 2>&1\n",
      "    ./overlap.sh 13 > ./overlap.000013.out 2>&1\n",
      "    ./overlap.sh 14 > ./overlap.000014.out 2>&1\n",
      "    ./overlap.sh 15 > ./overlap.000015.out 2>&1\n",
      "    ./overlap.sh 16 > ./overlap.000016.out 2>&1\n",
      "    ./overlap.sh 17 > ./overlap.000017.out 2>&1\n",
      "    ./overlap.sh 18 > ./overlap.000018.out 2>&1\n",
      "    ./overlap.sh 19 > ./overlap.000019.out 2>&1\n",
      "    ./overlap.sh 20 > ./overlap.000020.out 2>&1\n",
      "    ./overlap.sh 21 > ./overlap.000021.out 2>&1\n",
      "    ./overlap.sh 22 > ./overlap.000022.out 2>&1\n",
      "    ./overlap.sh 23 > ./overlap.000023.out 2>&1\n",
      "    ./overlap.sh 24 > ./overlap.000024.out 2>&1\n",
      "    ./overlap.sh 25 > ./overlap.000025.out 2>&1\n",
      "    ./overlap.sh 26 > ./overlap.000026.out 2>&1\n",
      "    ./overlap.sh 27 > ./overlap.000027.out 2>&1\n",
      "    ./overlap.sh 28 > ./overlap.000028.out 2>&1\n",
      "    ./overlap.sh 29 > ./overlap.000029.out 2>&1\n",
      "    ./overlap.sh 30 > ./overlap.000030.out 2>&1\n",
      "    ./overlap.sh 31 > ./overlap.000031.out 2>&1\n",
      "    ./overlap.sh 32 > ./overlap.000032.out 2>&1\n",
      "    ./overlap.sh 33 > ./overlap.000033.out 2>&1\n",
      "    ./overlap.sh 34 > ./overlap.000034.out 2>&1\n",
      "    ./overlap.sh 35 > ./overlap.000035.out 2>&1\n",
      "    ./overlap.sh 36 > ./overlap.000036.out 2>&1\n",
      "\n",
      "-- Finished on Sat Oct 21 01:09:56 2017 (18225 seconds) with 831.975 GB free disk space\n",
      "----------------------------------------\n",
      "-- Found 36 overlapInCore output files.\n",
      "--\n",
      "-- overlapInCore compute 'unitigging/1-overlapper':\n",
      "--   kmer hits\n",
      "--     with no overlap       3213673599  11.0833333 +- 54804988.866\n",
      "--     with an overlap         12904018  .944444444 +- 232589.324\n",
      "--\n",
      "--   overlaps                  12904018  .944444444 +- 232589.324\n",
      "--     contained                5611059   155862.75 +- 109113.414\n",
      "--     dovetail                 7292959  .194444444 +- 125630.562\n",
      "--\n",
      "--   overlaps rejected\n",
      "--     multiple per pair              0           0 +- 0\n",
      "--     bad short window               0           0 +- 0\n",
      "--     bad long window                0           0 +- 0\n",
      "-- Finished stage 'utg-overlapCheck', reset canuIteration.\n",
      "----------------------------------------\n",
      "-- Starting command on Sat Oct 21 01:09:56 2017 with 831.975 GB free disk space\n",
      "\n",
      "    cd unitigging\n",
      "    /usr/local/bioinformatics/canu-1.6/Linux-amd64/bin/ovStoreBuild \\\n",
      "     -O ./20171018_oly_pacbio.ovlStore.BUILDING \\\n",
      "     -G ./20171018_oly_pacbio.gkpStore \\\n",
      "     -M 4-32 \\\n",
      "     -L ./1-overlapper/ovljob.files \\\n",
      "     > ./20171018_oly_pacbio.ovlStore.err 2>&1\n",
      "\n",
      "-- Finished on Sat Oct 21 01:10:08 2017 (12 seconds) with 831.478 GB free disk space\n",
      "----------------------------------------\n",
      "-- Checking store.\n",
      "----------------------------------------\n",
      "-- Starting command on Sat Oct 21 01:10:08 2017 with 831.478 GB free disk space\n",
      "\n",
      "    cd unitigging\n",
      "    /usr/local/bioinformatics/canu-1.6/Linux-amd64/bin/ovStoreDump \\\n",
      "     -G ./20171018_oly_pacbio.gkpStore \\\n",
      "     -O ./20171018_oly_pacbio.ovlStore \\\n",
      "     -d -counts \\\n",
      "     > ./20171018_oly_pacbio.ovlStore/counts.dat 2> ./20171018_oly_pacbio.ovlStore/counts.err\n",
      "\n",
      "-- Finished on Sat Oct 21 01:10:08 2017 (lickety-split) with 831.471 GB free disk space\n",
      "----------------------------------------\n",
      "--\n",
      "-- Overlap store 'unitigging/20171018_oly_pacbio.ovlStore' successfully constructed.\n",
      "-- Found 25808036 overlaps for 718569 reads; 4691 reads have no overlaps.\n",
      "--\n",
      "FILE unitigging/1-overlapper/001/000001.ovb\n",
      "FILE unitigging/1-overlapper/001/000002.ovb\n",
      "FILE unitigging/1-overlapper/001/000003.ovb\n",
      "FILE unitigging/1-overlapper/001/000004.ovb\n",
      "FILE unitigging/1-overlapper/001/000005.ovb\n",
      "FILE unitigging/1-overlapper/001/000006.ovb\n",
      "FILE unitigging/1-overlapper/001/000007.ovb\n",
      "FILE unitigging/1-overlapper/001/000008.ovb\n",
      "FILE unitigging/1-overlapper/001/000009.ovb\n",
      "FILE unitigging/1-overlapper/001/000010.ovb\n",
      "FILE unitigging/1-overlapper/001/000011.ovb\n",
      "FILE unitigging/1-overlapper/001/000012.ovb\n",
      "FILE unitigging/1-overlapper/001/000013.ovb\n",
      "FILE unitigging/1-overlapper/001/000014.ovb\n",
      "FILE unitigging/1-overlapper/001/000015.ovb\n",
      "FILE unitigging/1-overlapper/001/000016.ovb\n",
      "FILE unitigging/1-overlapper/001/000017.ovb\n",
      "FILE unitigging/1-overlapper/001/000018.ovb\n",
      "FILE unitigging/1-overlapper/001/000019.ovb\n",
      "FILE unitigging/1-overlapper/001/000020.ovb\n",
      "FILE unitigging/1-overlapper/001/000021.ovb\n",
      "FILE unitigging/1-overlapper/001/000022.ovb\n",
      "FILE unitigging/1-overlapper/001/000023.ovb\n",
      "FILE unitigging/1-overlapper/001/000024.ovb\n",
      "FILE unitigging/1-overlapper/001/000025.ovb\n",
      "FILE unitigging/1-overlapper/001/000026.ovb\n",
      "FILE unitigging/1-overlapper/001/000027.ovb\n",
      "FILE unitigging/1-overlapper/001/000028.ovb\n",
      "FILE unitigging/1-overlapper/001/000029.ovb\n",
      "FILE unitigging/1-overlapper/001/000030.ovb\n",
      "FILE unitigging/1-overlapper/001/000031.ovb\n",
      "FILE unitigging/1-overlapper/001/000032.ovb\n",
      "FILE unitigging/1-overlapper/001/000033.ovb\n",
      "FILE unitigging/1-overlapper/001/000034.ovb\n",
      "FILE unitigging/1-overlapper/001/000035.ovb\n",
      "FILE unitigging/1-overlapper/001/000036.ovb\n",
      "FILE unitigging/1-overlapper/001/000001.stats\n",
      "FILE unitigging/1-overlapper/001/000001.counts\n",
      "FILE unitigging/1-overlapper/001/000002.stats\n",
      "FILE unitigging/1-overlapper/001/000002.counts\n",
      "FILE unitigging/1-overlapper/001/000003.stats\n",
      "FILE unitigging/1-overlapper/001/000003.counts\n",
      "FILE unitigging/1-overlapper/001/000004.stats\n",
      "FILE unitigging/1-overlapper/001/000004.counts\n",
      "FILE unitigging/1-overlapper/001/000005.stats\n",
      "FILE unitigging/1-overlapper/001/000005.counts\n",
      "FILE unitigging/1-overlapper/001/000006.stats\n",
      "FILE unitigging/1-overlapper/001/000006.counts\n",
      "FILE unitigging/1-overlapper/001/000007.stats\n",
      "FILE unitigging/1-overlapper/001/000007.counts\n",
      "FILE unitigging/1-overlapper/001/000008.stats\n",
      "FILE unitigging/1-overlapper/001/000008.counts\n",
      "FILE unitigging/1-overlapper/001/000009.stats\n",
      "FILE unitigging/1-overlapper/001/000009.counts\n",
      "FILE unitigging/1-overlapper/001/000010.stats\n",
      "FILE unitigging/1-overlapper/001/000010.counts\n",
      "FILE unitigging/1-overlapper/001/000011.stats\n",
      "FILE unitigging/1-overlapper/001/000011.counts\n",
      "FILE unitigging/1-overlapper/001/000012.stats\n",
      "FILE unitigging/1-overlapper/001/000012.counts\n",
      "FILE unitigging/1-overlapper/001/000013.stats\n",
      "FILE unitigging/1-overlapper/001/000013.counts\n",
      "FILE unitigging/1-overlapper/001/000014.stats\n",
      "FILE unitigging/1-overlapper/001/000014.counts\n",
      "FILE unitigging/1-overlapper/001/000015.stats\n",
      "FILE unitigging/1-overlapper/001/000015.counts\n",
      "FILE unitigging/1-overlapper/001/000016.stats\n",
      "FILE unitigging/1-overlapper/001/000016.counts\n",
      "FILE unitigging/1-overlapper/001/000017.stats\n",
      "FILE unitigging/1-overlapper/001/000017.counts\n",
      "FILE unitigging/1-overlapper/001/000018.stats\n",
      "FILE unitigging/1-overlapper/001/000018.counts\n",
      "FILE unitigging/1-overlapper/001/000019.stats\n",
      "FILE unitigging/1-overlapper/001/000019.counts\n",
      "FILE unitigging/1-overlapper/001/000020.stats\n",
      "FILE unitigging/1-overlapper/001/000020.counts\n",
      "FILE unitigging/1-overlapper/001/000021.stats\n",
      "FILE unitigging/1-overlapper/001/000021.counts\n",
      "FILE unitigging/1-overlapper/001/000022.stats\n",
      "FILE unitigging/1-overlapper/001/000022.counts\n",
      "FILE unitigging/1-overlapper/001/000023.stats\n",
      "FILE unitigging/1-overlapper/001/000023.counts\n",
      "FILE unitigging/1-overlapper/001/000024.stats\n",
      "FILE unitigging/1-overlapper/001/000024.counts\n",
      "FILE unitigging/1-overlapper/001/000025.stats\n",
      "FILE unitigging/1-overlapper/001/000025.counts\n",
      "FILE unitigging/1-overlapper/001/000026.stats\n",
      "FILE unitigging/1-overlapper/001/000026.counts\n",
      "FILE unitigging/1-overlapper/001/000027.stats\n",
      "FILE unitigging/1-overlapper/001/000027.counts\n",
      "FILE unitigging/1-overlapper/001/000028.stats\n",
      "FILE unitigging/1-overlapper/001/000028.counts\n",
      "FILE unitigging/1-overlapper/001/000029.stats\n",
      "FILE unitigging/1-overlapper/001/000029.counts\n",
      "FILE unitigging/1-overlapper/001/000030.stats\n",
      "FILE unitigging/1-overlapper/001/000030.counts\n",
      "FILE unitigging/1-overlapper/001/000031.stats\n",
      "FILE unitigging/1-overlapper/001/000031.counts\n",
      "FILE unitigging/1-overlapper/001/000032.stats\n",
      "FILE unitigging/1-overlapper/001/000032.counts\n",
      "FILE unitigging/1-overlapper/001/000033.stats\n",
      "FILE unitigging/1-overlapper/001/000033.counts\n",
      "FILE unitigging/1-overlapper/001/000034.stats\n",
      "FILE unitigging/1-overlapper/001/000034.counts\n",
      "FILE unitigging/1-overlapper/001/000035.stats\n",
      "FILE unitigging/1-overlapper/001/000035.counts\n",
      "FILE unitigging/1-overlapper/001/000036.stats\n",
      "FILE unitigging/1-overlapper/001/000036.counts\n",
      "--\n",
      "-- Purged 0.264 GB in 108 overlap output files.\n",
      "----------------------------------------\n",
      "-- Starting command on Sat Oct 21 01:10:09 2017 with 831.743 GB free disk space\n",
      "\n",
      "    cd unitigging\n",
      "    /usr/local/bioinformatics/canu-1.6/Linux-amd64/bin/ovStoreStats \\\n",
      "     -G ./20171018_oly_pacbio.gkpStore \\\n",
      "     -O ./20171018_oly_pacbio.ovlStore \\\n",
      "     -o ./20171018_oly_pacbio.ovlStore \\\n",
      "     > ./20171018_oly_pacbio.ovlStore.summary.err 2>&1\n",
      "\n",
      "-- Finished on Sat Oct 21 01:10:15 2017 (6 seconds) with 831.727 GB free disk space\n",
      "----------------------------------------\n",
      "--\n",
      "-- Overlap store 'unitigging/20171018_oly_pacbio.ovlStore' contains:\n",
      "--\n",
      "--   category            reads     %          read length        feature size or coverage  analysis\n",
      "--   ----------------  -------  -------  ----------------------  ------------------------  --------------------\n",
      "--   middle-missing       1699    0.23     5836.53 +- 3147.72        626.11 +- 728.33     (bad trimming)\n",
      "--   middle-hump         18543    2.56     5119.56 +- 2621.57        111.13 +- 322.64     (bad trimming)\n",
      "--   no-5-prime          38530    5.33     5943.75 +- 2512.95         78.06 +- 283.14     (bad trimming)\n",
      "--   no-3-prime          33037    4.57     5855.59 +- 2636.73         88.36 +- 302.87     (bad trimming)\n",
      "--   \n",
      "--   low-coverage       266450   36.84     4000.00 +- 2281.65          4.18 +- 1.93       (easy to assemble, potential for lower quality consensus)\n",
      "--   unique              63758    8.82     3557.46 +- 1987.19         19.17 +- 8.73       (easy to assemble, perfect, yay)\n",
      "--   repeat-cont         21849    3.02     3844.08 +- 2434.09        384.74 +- 454.87     (potential for consensus errors, no impact on assembly)\n",
      "--   repeat-dove           102    0.01    10318.17 +- 3376.82        164.14 +- 117.96     (hard to assemble, likely won't assemble correctly or even at all)\n",
      "--   \n",
      "--   span-repeat        100087   13.84     6237.15 +- 2208.44       2815.30 +- 2079.79    (read spans a large repeat, usually easy to assemble)\n",
      "--   uniq-repeat-cont   142305   19.68     5250.49 +- 2081.27                             (should be uniquely placed, low potential for consensus errors, no impact on assembly)\n",
      "--   uniq-repeat-dove    30539    4.22     7130.44 +- 2324.91                             (will end contigs, potential to misassemble)\n",
      "--   uniq-anchor          1670    0.23     7121.67 +- 2410.15       2457.85 +- 1879.88    (repeat read, with unique section, probable bad read)\n",
      "-- Finished stage 'utg-createOverlapStore', reset canuIteration.\n",
      "\n",
      "Configure RED for 8gb memory with batches of at most 5000 reads and (unlimited) bases.\n",
      "\n",
      "RED job   1 from read         1 to read      5000 -   0.179 GB for    5000 reads -   0.154 GB for  12699528 olaps -   0.002 GB for evidence\n",
      "RED job   2 from read      5001 to read     10000 -   0.232 GB for    5000 reads -   0.200 GB for  16490428 olaps -   0.002 GB for evidence\n",
      "RED job   3 from read     10001 to read     15000 -   0.365 GB for    5000 reads -   0.314 GB for  25957103 olaps -   0.002 GB for evidence\n",
      "RED job   4 from read     15001 to read     20000 -   0.370 GB for    5000 reads -   0.319 GB for  26335141 olaps -   0.002 GB for evidence\n",
      "RED job   5 from read     20001 to read     25000 -   0.375 GB for    5000 reads -   0.323 GB for  26707638 olaps -   0.002 GB for evidence\n",
      "RED job   6 from read     25001 to read     30000 -   0.364 GB for    5000 reads -   0.313 GB for  25863790 olaps -   0.002 GB for evidence\n",
      "RED job   7 from read     30001 to read     35000 -   0.366 GB for    5000 reads -   0.315 GB for  26036510 olaps -   0.002 GB for evidence\n",
      "RED job   8 from read     35001 to read     40000 -   0.365 GB for    5000 reads -   0.315 GB for  26016912 olaps -   0.002 GB for evidence\n",
      "RED job   9 from read     40001 to read     45000 -   0.360 GB for    5000 reads -   0.310 GB for  25611209 olaps -   0.002 GB for evidence\n",
      "RED job  10 from read     45001 to read     50000 -   0.359 GB for    5000 reads -   0.310 GB for  25569349 olaps -   0.002 GB for evidence\n",
      "RED job  11 from read     50001 to read     55000 -   0.362 GB for    5000 reads -   0.312 GB for  25744876 olaps -   0.002 GB for evidence\n",
      "RED job  12 from read     55001 to read     60000 -   0.363 GB for    5000 reads -   0.313 GB for  25848749 olaps -   0.002 GB for evidence\n",
      "RED job  13 from read     60001 to read     65000 -   0.362 GB for    5000 reads -   0.312 GB for  25734564 olaps -   0.002 GB for evidence\n",
      "RED job  14 from read     65001 to read     70000 -   0.362 GB for    5000 reads -   0.312 GB for  25780279 olaps -   0.002 GB for evidence\n",
      "RED job  15 from read     70001 to read     75000 -   0.359 GB for    5000 reads -   0.310 GB for  25581342 olaps -   0.002 GB for evidence\n",
      "RED job  16 from read     75001 to read     80000 -   0.356 GB for    5000 reads -   0.306 GB for  25298494 olaps -   0.002 GB for evidence\n",
      "RED job  17 from read     80001 to read     85000 -   0.356 GB for    5000 reads -   0.306 GB for  25267326 olaps -   0.003 GB for evidence\n",
      "RED job  18 from read     85001 to read     90000 -   0.362 GB for    5000 reads -   0.312 GB for  25766329 olaps -   0.002 GB for evidence\n",
      "RED job  19 from read     90001 to read     95000 -   0.367 GB for    5000 reads -   0.317 GB for  26148506 olaps -   0.002 GB for evidence\n",
      "RED job  20 from read     95001 to read    100000 -   0.358 GB for    5000 reads -   0.308 GB for  25433634 olaps -   0.003 GB for evidence\n",
      "RED job  21 from read    100001 to read    105000 -   0.364 GB for    5000 reads -   0.314 GB for  25948473 olaps -   0.002 GB for evidence\n",
      "RED job  22 from read    105001 to read    110000 -   0.363 GB for    5000 reads -   0.312 GB for  25750752 olaps -   0.003 GB for evidence\n",
      "RED job  23 from read    110001 to read    115000 -   0.364 GB for    5000 reads -   0.313 GB for  25856651 olaps -   0.002 GB for evidence\n",
      "RED job  24 from read    115001 to read    120000 -   0.367 GB for    5000 reads -   0.316 GB for  26093606 olaps -   0.002 GB for evidence\n",
      "RED job  25 from read    120001 to read    125000 -   0.345 GB for    5000 reads -   0.297 GB for  24504092 olaps -   0.002 GB for evidence\n",
      "RED job  26 from read    125001 to read    130000 -   0.323 GB for    5000 reads -   0.278 GB for  22991050 olaps -   0.001 GB for evidence\n",
      "RED job  27 from read    130001 to read    135000 -   0.313 GB for    5000 reads -   0.269 GB for  22205184 olaps -   0.002 GB for evidence\n",
      "RED job  28 from read    135001 to read    140000 -   0.317 GB for    5000 reads -   0.273 GB for  22569315 olaps -   0.002 GB for evidence\n",
      "RED job  29 from read    140001 to read    145000 -   0.313 GB for    5000 reads -   0.269 GB for  22252079 olaps -   0.002 GB for evidence\n",
      "RED job  30 from read    145001 to read    150000 -   0.311 GB for    5000 reads -   0.268 GB for  22172846 olaps -   0.001 GB for evidence\n",
      "RED job  31 from read    150001 to read    155000 -   0.314 GB for    5000 reads -   0.271 GB for  22357589 olaps -   0.001 GB for evidence\n",
      "RED job  32 from read    155001 to read    160000 -   0.314 GB for    5000 reads -   0.271 GB for  22386372 olaps -   0.001 GB for evidence\n",
      "RED job  33 from read    160001 to read    165000 -   0.320 GB for    5000 reads -   0.276 GB for  22831650 olaps -   0.001 GB for evidence\n",
      "RED job  34 from read    165001 to read    170000 -   0.314 GB for    5000 reads -   0.271 GB for  22355267 olaps -   0.002 GB for evidence\n",
      "RED job  35 from read    170001 to read    175000 -   0.319 GB for    5000 reads -   0.275 GB for  22742124 olaps -   0.001 GB for evidence\n",
      "RED job  36 from read    175001 to read    180000 -   0.330 GB for    5000 reads -   0.284 GB for  23450790 olaps -   0.002 GB for evidence\n",
      "RED job  37 from read    180001 to read    185000 -   0.323 GB for    5000 reads -   0.279 GB for  23012499 olaps -   0.001 GB for evidence\n",
      "RED job  38 from read    185001 to read    190000 -   0.327 GB for    5000 reads -   0.282 GB for  23323168 olaps -   0.002 GB for evidence\n",
      "RED job  39 from read    190001 to read    195000 -   0.315 GB for    5000 reads -   0.272 GB for  22428042 olaps -   0.002 GB for evidence\n",
      "RED job  40 from read    195001 to read    200000 -   0.321 GB for    5000 reads -   0.277 GB for  22898421 olaps -   0.002 GB for evidence\n",
      "RED job  41 from read    200001 to read    205000 -   0.321 GB for    5000 reads -   0.277 GB for  22845916 olaps -   0.002 GB for evidence\n",
      "RED job  42 from read    205001 to read    210000 -   0.319 GB for    5000 reads -   0.275 GB for  22677202 olaps -   0.002 GB for evidence\n",
      "RED job  43 from read    210001 to read    215000 -   0.318 GB for    5000 reads -   0.274 GB for  22613114 olaps -   0.002 GB for evidence\n",
      "RED job  44 from read    215001 to read    220000 -   0.326 GB for    5000 reads -   0.281 GB for  23223569 olaps -   0.002 GB for evidence\n",
      "RED job  45 from read    220001 to read    225000 -   0.315 GB for    5000 reads -   0.271 GB for  22391128 olaps -   0.002 GB for evidence\n",
      "RED job  46 from read    225001 to read    230000 -   0.324 GB for    5000 reads -   0.279 GB for  23054719 olaps -   0.002 GB for evidence\n",
      "RED job  47 from read    230001 to read    235000 -   0.319 GB for    5000 reads -   0.275 GB for  22703059 olaps -   0.002 GB for evidence\n",
      "RED job  48 from read    235001 to read    240000 -   0.314 GB for    5000 reads -   0.271 GB for  22382874 olaps -   0.001 GB for evidence\n",
      "RED job  49 from read    240001 to read    245000 -   0.316 GB for    5000 reads -   0.272 GB for  22489149 olaps -   0.002 GB for evidence\n",
      "RED job  50 from read    245001 to read    250000 -   0.314 GB for    5000 reads -   0.270 GB for  22328431 olaps -   0.002 GB for evidence\n",
      "RED job  51 from read    250001 to read    255000 -   0.320 GB for    5000 reads -   0.276 GB for  22762815 olaps -   0.002 GB for evidence\n",
      "RED job  52 from read    255001 to read    260000 -   0.315 GB for    5000 reads -   0.271 GB for  22371562 olaps -   0.002 GB for evidence\n",
      "RED job  53 from read    260001 to read    265000 -   0.316 GB for    5000 reads -   0.272 GB for  22495407 olaps -   0.001 GB for evidence\n",
      "RED job  54 from read    265001 to read    270000 -   0.316 GB for    5000 reads -   0.273 GB for  22530392 olaps -   0.002 GB for evidence\n",
      "RED job  55 from read    270001 to read    275000 -   0.314 GB for    5000 reads -   0.271 GB for  22391897 olaps -   0.002 GB for evidence\n",
      "RED job  56 from read    275001 to read    280000 -   0.311 GB for    5000 reads -   0.268 GB for  22154065 olaps -   0.002 GB for evidence\n",
      "RED job  57 from read    280001 to read    285000 -   0.312 GB for    5000 reads -   0.269 GB for  22237041 olaps -   0.002 GB for evidence\n",
      "RED job  58 from read    285001 to read    290000 -   0.325 GB for    5000 reads -   0.280 GB for  23167594 olaps -   0.001 GB for evidence\n",
      "RED job  59 from read    290001 to read    295000 -   0.329 GB for    5000 reads -   0.284 GB for  23423581 olaps -   0.002 GB for evidence\n",
      "RED job  60 from read    295001 to read    300000 -   0.325 GB for    5000 reads -   0.280 GB for  23111858 olaps -   0.002 GB for evidence\n",
      "RED job  61 from read    300001 to read    305000 -   0.322 GB for    5000 reads -   0.277 GB for  22918482 olaps -   0.002 GB for evidence\n",
      "RED job  62 from read    305001 to read    310000 -   0.316 GB for    5000 reads -   0.273 GB for  22526904 olaps -   0.002 GB for evidence\n",
      "RED job  63 from read    310001 to read    315000 -   0.325 GB for    5000 reads -   0.280 GB for  23148371 olaps -   0.002 GB for evidence\n",
      "RED job  64 from read    315001 to read    320000 -   0.329 GB for    5000 reads -   0.284 GB for  23435958 olaps -   0.002 GB for evidence\n",
      "RED job  65 from read    320001 to read    325000 -   0.317 GB for    5000 reads -   0.274 GB for  22595084 olaps -   0.002 GB for evidence\n",
      "RED job  66 from read    325001 to read    330000 -   0.316 GB for    5000 reads -   0.272 GB for  22489182 olaps -   0.001 GB for evidence\n",
      "RED job  67 from read    330001 to read    335000 -   0.318 GB for    5000 reads -   0.274 GB for  22655119 olaps -   0.002 GB for evidence\n",
      "RED job  68 from read    335001 to read    340000 -   0.313 GB for    5000 reads -   0.270 GB for  22307862 olaps -   0.001 GB for evidence\n",
      "RED job  69 from read    340001 to read    345000 -   0.317 GB for    5000 reads -   0.273 GB for  22572975 olaps -   0.002 GB for evidence\n",
      "RED job  70 from read    345001 to read    350000 -   0.321 GB for    5000 reads -   0.277 GB for  22869877 olaps -   0.002 GB for evidence\n",
      "RED job  71 from read    350001 to read    355000 -   0.319 GB for    5000 reads -   0.275 GB for  22747650 olaps -   0.002 GB for evidence\n",
      "RED job  72 from read    355001 to read    360000 -   0.323 GB for    5000 reads -   0.279 GB for  23021710 olaps -   0.002 GB for evidence\n",
      "RED job  73 from read    360001 to read    365000 -   0.321 GB for    5000 reads -   0.276 GB for  22814404 olaps -   0.002 GB for evidence\n",
      "RED job  74 from read    365001 to read    370000 -   0.336 GB for    5000 reads -   0.289 GB for  23887432 olaps -   0.002 GB for evidence\n",
      "RED job  75 from read    370001 to read    375000 -   0.324 GB for    5000 reads -   0.280 GB for  23096541 olaps -   0.001 GB for evidence\n",
      "RED job  76 from read    375001 to read    380000 -   0.322 GB for    5000 reads -   0.278 GB for  22926314 olaps -   0.002 GB for evidence\n",
      "RED job  77 from read    380001 to read    385000 -   0.318 GB for    5000 reads -   0.274 GB for  22635464 olaps -   0.002 GB for evidence\n",
      "RED job  78 from read    385001 to read    390000 -   0.318 GB for    5000 reads -   0.274 GB for  22604104 olaps -   0.002 GB for evidence\n",
      "RED job  79 from read    390001 to read    395000 -   0.319 GB for    5000 reads -   0.275 GB for  22743884 olaps -   0.001 GB for evidence\n",
      "RED job  80 from read    395001 to read    400000 -   0.319 GB for    5000 reads -   0.275 GB for  22718821 olaps -   0.001 GB for evidence\n",
      "RED job  81 from read    400001 to read    405000 -   0.320 GB for    5000 reads -   0.275 GB for  22739842 olaps -   0.002 GB for evidence\n",
      "RED job  82 from read    405001 to read    410000 -   0.319 GB for    5000 reads -   0.276 GB for  22765841 olaps -   0.001 GB for evidence\n",
      "RED job  83 from read    410001 to read    415000 -   0.318 GB for    5000 reads -   0.274 GB for  22617799 olaps -   0.002 GB for evidence\n",
      "RED job  84 from read    415001 to read    420000 -   0.323 GB for    5000 reads -   0.278 GB for  22960144 olaps -   0.002 GB for evidence\n",
      "RED job  85 from read    420001 to read    425000 -   0.320 GB for    5000 reads -   0.275 GB for  22751340 olaps -   0.002 GB for evidence\n",
      "RED job  86 from read    425001 to read    430000 -   0.319 GB for    5000 reads -   0.275 GB for  22708614 olaps -   0.002 GB for evidence\n",
      "RED job  87 from read    430001 to read    435000 -   0.324 GB for    5000 reads -   0.280 GB for  23091800 olaps -   0.001 GB for evidence\n",
      "RED job  88 from read    435001 to read    440000 -   0.334 GB for    5000 reads -   0.287 GB for  23740279 olaps -   0.002 GB for evidence\n",
      "RED job  89 from read    440001 to read    445000 -   0.364 GB for    5000 reads -   0.314 GB for  25913215 olaps -   0.002 GB for evidence\n",
      "RED job  90 from read    445001 to read    450000 -   0.351 GB for    5000 reads -   0.302 GB for  24962345 olaps -   0.002 GB for evidence\n",
      "RED job  91 from read    450001 to read    455000 -   0.359 GB for    5000 reads -   0.310 GB for  25574367 olaps -   0.002 GB for evidence\n",
      "RED job  92 from read    455001 to read    460000 -   0.354 GB for    5000 reads -   0.305 GB for  25211327 olaps -   0.002 GB for evidence\n",
      "RED job  93 from read    460001 to read    465000 -   0.356 GB for    5000 reads -   0.306 GB for  25309280 olaps -   0.003 GB for evidence\n",
      "RED job  94 from read    465001 to read    470000 -   0.363 GB for    5000 reads -   0.313 GB for  25864570 olaps -   0.002 GB for evidence\n",
      "RED job  95 from read    470001 to read    475000 -   0.363 GB for    5000 reads -   0.313 GB for  25842910 olaps -   0.002 GB for evidence\n",
      "RED job  96 from read    475001 to read    480000 -   0.361 GB for    5000 reads -   0.311 GB for  25680078 olaps -   0.002 GB for evidence\n",
      "RED job  97 from read    480001 to read    485000 -   0.356 GB for    5000 reads -   0.307 GB for  25385509 olaps -   0.002 GB for evidence\n",
      "RED job  98 from read    485001 to read    490000 -   0.357 GB for    5000 reads -   0.308 GB for  25421336 olaps -   0.002 GB for evidence\n",
      "RED job  99 from read    490001 to read    495000 -   0.367 GB for    5000 reads -   0.317 GB for  26151253 olaps -   0.001 GB for evidence\n",
      "RED job 100 from read    495001 to read    500000 -   0.358 GB for    5000 reads -   0.309 GB for  25531031 olaps -   0.002 GB for evidence\n",
      "RED job 101 from read    500001 to read    505000 -   0.361 GB for    5000 reads -   0.311 GB for  25721368 olaps -   0.002 GB for evidence\n",
      "RED job 102 from read    505001 to read    510000 -   0.360 GB for    5000 reads -   0.310 GB for  25624760 olaps -   0.002 GB for evidence\n",
      "RED job 103 from read    510001 to read    515000 -   0.362 GB for    5000 reads -   0.311 GB for  25716013 olaps -   0.002 GB for evidence\n",
      "RED job 104 from read    515001 to read    520000 -   0.371 GB for    5000 reads -   0.320 GB for  26412419 olaps -   0.002 GB for evidence\n",
      "RED job 105 from read    520001 to read    525000 -   0.367 GB for    5000 reads -   0.316 GB for  26085074 olaps -   0.002 GB for evidence\n",
      "RED job 106 from read    525001 to read    530000 -   0.368 GB for    5000 reads -   0.317 GB for  26218930 olaps -   0.002 GB for evidence\n",
      "RED job 107 from read    530001 to read    535000 -   0.367 GB for    5000 reads -   0.316 GB for  26138395 olaps -   0.002 GB for evidence\n",
      "RED job 108 from read    535001 to read    540000 -   0.368 GB for    5000 reads -   0.318 GB for  26230674 olaps -   0.002 GB for evidence\n",
      "RED job 109 from read    540001 to read    545000 -   0.367 GB for    5000 reads -   0.315 GB for  26019450 olaps -   0.004 GB for evidence\n",
      "RED job 110 from read    545001 to read    550000 -   0.357 GB for    5000 reads -   0.307 GB for  25380106 olaps -   0.002 GB for evidence\n",
      "RED job 111 from read    550001 to read    555000 -   0.363 GB for    5000 reads -   0.312 GB for  25766550 olaps -   0.003 GB for evidence\n",
      "RED job 112 from read    555001 to read    560000 -   0.364 GB for    5000 reads -   0.313 GB for  25870029 olaps -   0.002 GB for evidence\n",
      "RED job 113 from read    560001 to read    565000 -   0.369 GB for    5000 reads -   0.318 GB for  26230125 olaps -   0.003 GB for evidence\n",
      "RED job 114 from read    565001 to read    570000 -   0.364 GB for    5000 reads -   0.314 GB for  25906075 olaps -   0.002 GB for evidence\n",
      "RED job 115 from read    570001 to read    575000 -   0.365 GB for    5000 reads -   0.315 GB for  25992499 olaps -   0.002 GB for evidence\n",
      "RED job 116 from read    575001 to read    580000 -   0.361 GB for    5000 reads -   0.311 GB for  25698446 olaps -   0.002 GB for evidence\n",
      "RED job 117 from read    580001 to read    585000 -   0.369 GB for    5000 reads -   0.317 GB for  26202365 olaps -   0.003 GB for evidence\n",
      "RED job 118 from read    585001 to read    590000 -   0.363 GB for    5000 reads -   0.312 GB for  25780416 olaps -   0.003 GB for evidence\n",
      "RED job 119 from read    590001 to read    595000 -   0.364 GB for    5000 reads -   0.313 GB for  25841536 olaps -   0.003 GB for evidence\n",
      "RED job 120 from read    595001 to read    600000 -   0.363 GB for    5000 reads -   0.313 GB for  25860290 olaps -   0.002 GB for evidence\n",
      "RED job 121 from read    600001 to read    605000 -   0.368 GB for    5000 reads -   0.317 GB for  26209600 olaps -   0.002 GB for evidence\n",
      "RED job 122 from read    605001 to read    610000 -   0.365 GB for    5000 reads -   0.315 GB for  25989460 olaps -   0.002 GB for evidence\n",
      "RED job 123 from read    610001 to read    615000 -   0.367 GB for    5000 reads -   0.316 GB for  26137820 olaps -   0.002 GB for evidence\n",
      "RED job 124 from read    615001 to read    620000 -   0.363 GB for    5000 reads -   0.313 GB for  25838450 olaps -   0.002 GB for evidence\n",
      "RED job 125 from read    620001 to read    625000 -   0.354 GB for    5000 reads -   0.304 GB for  25143547 olaps -   0.003 GB for evidence\n",
      "RED job 126 from read    625001 to read    630000 -   0.370 GB for    5000 reads -   0.318 GB for  26289543 olaps -   0.003 GB for evidence\n",
      "RED job 127 from read    630001 to read    635000 -   0.364 GB for    5000 reads -   0.313 GB for  25824204 olaps -   0.003 GB for evidence\n",
      "RED job 128 from read    635001 to read    640000 -   0.360 GB for    5000 reads -   0.310 GB for  25567252 olaps -   0.003 GB for evidence\n",
      "RED job 129 from read    640001 to read    645000 -   0.362 GB for    5000 reads -   0.311 GB for  25720497 olaps -   0.003 GB for evidence\n",
      "RED job 130 from read    645001 to read    650000 -   0.365 GB for    5000 reads -   0.314 GB for  25967823 olaps -   0.002 GB for evidence\n",
      "RED job 131 from read    650001 to read    655000 -   0.364 GB for    5000 reads -   0.313 GB for  25813801 olaps -   0.003 GB for evidence\n",
      "RED job 132 from read    655001 to read    660000 -   0.361 GB for    5000 reads -   0.311 GB for  25684159 olaps -   0.002 GB for evidence\n",
      "RED job 133 from read    660001 to read    665000 -   0.363 GB for    5000 reads -   0.314 GB for  25896972 olaps -   0.002 GB for evidence\n",
      "RED job 134 from read    665001 to read    670000 -   0.359 GB for    5000 reads -   0.310 GB for  25566000 olaps -   0.002 GB for evidence\n",
      "RED job 135 from read    670001 to read    675000 -   0.357 GB for    5000 reads -   0.308 GB for  25471331 olaps -   0.002 GB for evidence\n",
      "RED job 136 from read    675001 to read    680000 -   0.354 GB for    5000 reads -   0.304 GB for  25126490 olaps -   0.003 GB for evidence\n",
      "RED job 137 from read    680001 to read    685000 -   0.358 GB for    5000 reads -   0.308 GB for  25475469 olaps -   0.002 GB for evidence\n",
      "RED job 138 from read    685001 to read    690000 -   0.353 GB for    5000 reads -   0.303 GB for  25012500 olaps -   0.003 GB for evidence\n",
      "RED job 139 from read    690001 to read    695000 -   0.355 GB for    5000 reads -   0.305 GB for  25215268 olaps -   0.003 GB for evidence\n",
      "RED job 140 from read    695001 to read    700000 -   0.365 GB for    5000 reads -   0.314 GB for  25963312 olaps -   0.002 GB for evidence\n",
      "RED job 141 from read    700001 to read    705000 -   0.363 GB for    5000 reads -   0.312 GB for  25805994 olaps -   0.002 GB for evidence\n",
      "RED job 142 from read    705001 to read    710000 -   0.352 GB for    5000 reads -   0.302 GB for  24982725 olaps -   0.003 GB for evidence\n",
      "RED job 143 from read    710001 to read    715000 -   0.359 GB for    5000 reads -   0.309 GB for  25552488 olaps -   0.002 GB for evidence\n",
      "RED job 144 from read    715001 to read    720000 -   0.358 GB for    5000 reads -   0.309 GB for  25495995 olaps -   0.002 GB for evidence\n",
      "RED job 145 from read    720001 to read    723260 -   0.235 GB for    3260 reads -   0.203 GB for  16738236 olaps -   0.001 GB for evidence\n",
      "-- Finished stage 'readErrorDetectionConfigure', reset canuIteration.\n",
      "--\n",
      "-- Running jobs.  First attempt out of 2.\n",
      "----------------------------------------\n",
      "-- Starting 'red' concurrent execution on Sat Oct 21 01:10:20 2017 with 831.727 GB free disk space (145 processes; 3 concurrently)\n",
      "\n",
      "    cd unitigging/3-overlapErrorAdjustment\n",
      "    ./red.sh 1 > ./red.000001.out 2>&1\n",
      "    ./red.sh 2 > ./red.000002.out 2>&1\n",
      "    ./red.sh 3 > ./red.000003.out 2>&1\n",
      "    ./red.sh 4 > ./red.000004.out 2>&1\n",
      "    ./red.sh 5 > ./red.000005.out 2>&1\n",
      "    ./red.sh 6 > ./red.000006.out 2>&1\n",
      "    ./red.sh 7 > ./red.000007.out 2>&1\n",
      "    ./red.sh 8 > ./red.000008.out 2>&1\n",
      "    ./red.sh 9 > ./red.000009.out 2>&1\n",
      "    ./red.sh 10 > ./red.000010.out 2>&1\n",
      "    ./red.sh 11 > ./red.000011.out 2>&1\n",
      "    ./red.sh 12 > ./red.000012.out 2>&1\n",
      "    ./red.sh 13 > ./red.000013.out 2>&1\n",
      "    ./red.sh 14 > ./red.000014.out 2>&1\n",
      "    ./red.sh 15 > ./red.000015.out 2>&1\n",
      "    ./red.sh 16 > ./red.000016.out 2>&1\n",
      "    ./red.sh 17 > ./red.000017.out 2>&1\n",
      "    ./red.sh 18 > ./red.000018.out 2>&1\n",
      "    ./red.sh 19 > ./red.000019.out 2>&1\n",
      "    ./red.sh 20 > ./red.000020.out 2>&1\n",
      "    ./red.sh 21 > ./red.000021.out 2>&1\n",
      "    ./red.sh 22 > ./red.000022.out 2>&1\n",
      "    ./red.sh 23 > ./red.000023.out 2>&1\n",
      "    ./red.sh 24 > ./red.000024.out 2>&1\n",
      "    ./red.sh 25 > ./red.000025.out 2>&1\n",
      "    ./red.sh 26 > ./red.000026.out 2>&1\n",
      "    ./red.sh 27 > ./red.000027.out 2>&1\n",
      "    ./red.sh 28 > ./red.000028.out 2>&1\n",
      "    ./red.sh 29 > ./red.000029.out 2>&1\n",
      "    ./red.sh 30 > ./red.000030.out 2>&1\n",
      "    ./red.sh 31 > ./red.000031.out 2>&1\n",
      "    ./red.sh 32 > ./red.000032.out 2>&1\n",
      "    ./red.sh 33 > ./red.000033.out 2>&1\n",
      "    ./red.sh 34 > ./red.000034.out 2>&1\n",
      "    ./red.sh 35 > ./red.000035.out 2>&1\n",
      "    ./red.sh 36 > ./red.000036.out 2>&1\n",
      "    ./red.sh 37 > ./red.000037.out 2>&1\n",
      "    ./red.sh 38 > ./red.000038.out 2>&1\n",
      "    ./red.sh 39 > ./red.000039.out 2>&1\n",
      "    ./red.sh 40 > ./red.000040.out 2>&1\n",
      "    ./red.sh 41 > ./red.000041.out 2>&1\n",
      "    ./red.sh 42 > ./red.000042.out 2>&1\n",
      "    ./red.sh 43 > ./red.000043.out 2>&1\n",
      "    ./red.sh 44 > ./red.000044.out 2>&1\n",
      "    ./red.sh 45 > ./red.000045.out 2>&1\n",
      "    ./red.sh 46 > ./red.000046.out 2>&1\n",
      "    ./red.sh 47 > ./red.000047.out 2>&1\n",
      "    ./red.sh 48 > ./red.000048.out 2>&1\n",
      "    ./red.sh 49 > ./red.000049.out 2>&1\n",
      "    ./red.sh 50 > ./red.000050.out 2>&1\n",
      "    ./red.sh 51 > ./red.000051.out 2>&1\n",
      "    ./red.sh 52 > ./red.000052.out 2>&1\n",
      "    ./red.sh 53 > ./red.000053.out 2>&1\n",
      "    ./red.sh 54 > ./red.000054.out 2>&1\n",
      "    ./red.sh 55 > ./red.000055.out 2>&1\n",
      "    ./red.sh 56 > ./red.000056.out 2>&1\n",
      "    ./red.sh 57 > ./red.000057.out 2>&1\n",
      "    ./red.sh 58 > ./red.000058.out 2>&1\n",
      "    ./red.sh 59 > ./red.000059.out 2>&1\n",
      "    ./red.sh 60 > ./red.000060.out 2>&1\n",
      "    ./red.sh 61 > ./red.000061.out 2>&1\n",
      "    ./red.sh 62 > ./red.000062.out 2>&1\n",
      "    ./red.sh 63 > ./red.000063.out 2>&1\n",
      "    ./red.sh 64 > ./red.000064.out 2>&1\n",
      "    ./red.sh 65 > ./red.000065.out 2>&1\n",
      "    ./red.sh 66 > ./red.000066.out 2>&1\n",
      "    ./red.sh 67 > ./red.000067.out 2>&1\n",
      "    ./red.sh 68 > ./red.000068.out 2>&1\n",
      "    ./red.sh 69 > ./red.000069.out 2>&1\n",
      "    ./red.sh 70 > ./red.000070.out 2>&1\n",
      "    ./red.sh 71 > ./red.000071.out 2>&1\n",
      "    ./red.sh 72 > ./red.000072.out 2>&1\n",
      "    ./red.sh 73 > ./red.000073.out 2>&1\n",
      "    ./red.sh 74 > ./red.000074.out 2>&1\n",
      "    ./red.sh 75 > ./red.000075.out 2>&1\n",
      "    ./red.sh 76 > ./red.000076.out 2>&1\n",
      "    ./red.sh 77 > ./red.000077.out 2>&1\n",
      "    ./red.sh 78 > ./red.000078.out 2>&1\n",
      "    ./red.sh 79 > ./red.000079.out 2>&1\n",
      "    ./red.sh 80 > ./red.000080.out 2>&1\n",
      "    ./red.sh 81 > ./red.000081.out 2>&1\n",
      "    ./red.sh 82 > ./red.000082.out 2>&1\n",
      "    ./red.sh 83 > ./red.000083.out 2>&1\n",
      "    ./red.sh 84 > ./red.000084.out 2>&1\n",
      "    ./red.sh 85 > ./red.000085.out 2>&1\n",
      "    ./red.sh 86 > ./red.000086.out 2>&1\n",
      "    ./red.sh 87 > ./red.000087.out 2>&1\n",
      "    ./red.sh 88 > ./red.000088.out 2>&1\n",
      "    ./red.sh 89 > ./red.000089.out 2>&1\n",
      "    ./red.sh 90 > ./red.000090.out 2>&1\n",
      "    ./red.sh 91 > ./red.000091.out 2>&1\n",
      "    ./red.sh 92 > ./red.000092.out 2>&1\n",
      "    ./red.sh 93 > ./red.000093.out 2>&1\n",
      "    ./red.sh 94 > ./red.000094.out 2>&1\n",
      "    ./red.sh 95 > ./red.000095.out 2>&1\n",
      "    ./red.sh 96 > ./red.000096.out 2>&1\n",
      "    ./red.sh 97 > ./red.000097.out 2>&1\n",
      "    ./red.sh 98 > ./red.000098.out 2>&1\n",
      "    ./red.sh 99 > ./red.000099.out 2>&1\n",
      "    ./red.sh 100 > ./red.000100.out 2>&1\n",
      "    ./red.sh 101 > ./red.000101.out 2>&1\n",
      "    ./red.sh 102 > ./red.000102.out 2>&1\n",
      "    ./red.sh 103 > ./red.000103.out 2>&1\n",
      "    ./red.sh 104 > ./red.000104.out 2>&1\n",
      "    ./red.sh 105 > ./red.000105.out 2>&1\n",
      "    ./red.sh 106 > ./red.000106.out 2>&1\n",
      "    ./red.sh 107 > ./red.000107.out 2>&1\n",
      "    ./red.sh 108 > ./red.000108.out 2>&1\n",
      "    ./red.sh 109 > ./red.000109.out 2>&1\n",
      "    ./red.sh 110 > ./red.000110.out 2>&1\n",
      "    ./red.sh 111 > ./red.000111.out 2>&1\n",
      "    ./red.sh 112 > ./red.000112.out 2>&1\n",
      "    ./red.sh 113 > ./red.000113.out 2>&1\n",
      "    ./red.sh 114 > ./red.000114.out 2>&1\n",
      "    ./red.sh 115 > ./red.000115.out 2>&1\n",
      "    ./red.sh 116 > ./red.000116.out 2>&1\n",
      "    ./red.sh 117 > ./red.000117.out 2>&1\n",
      "    ./red.sh 118 > ./red.000118.out 2>&1\n",
      "    ./red.sh 119 > ./red.000119.out 2>&1\n",
      "    ./red.sh 120 > ./red.000120.out 2>&1\n",
      "    ./red.sh 121 > ./red.000121.out 2>&1\n",
      "    ./red.sh 122 > ./red.000122.out 2>&1\n",
      "    ./red.sh 123 > ./red.000123.out 2>&1\n",
      "    ./red.sh 124 > ./red.000124.out 2>&1\n",
      "    ./red.sh 125 > ./red.000125.out 2>&1\n",
      "    ./red.sh 126 > ./red.000126.out 2>&1\n",
      "    ./red.sh 127 > ./red.000127.out 2>&1\n",
      "    ./red.sh 128 > ./red.000128.out 2>&1\n",
      "    ./red.sh 129 > ./red.000129.out 2>&1\n",
      "    ./red.sh 130 > ./red.000130.out 2>&1\n",
      "    ./red.sh 131 > ./red.000131.out 2>&1\n",
      "    ./red.sh 132 > ./red.000132.out 2>&1\n",
      "    ./red.sh 133 > ./red.000133.out 2>&1\n",
      "    ./red.sh 134 > ./red.000134.out 2>&1\n",
      "    ./red.sh 135 > ./red.000135.out 2>&1\n",
      "    ./red.sh 136 > ./red.000136.out 2>&1\n",
      "    ./red.sh 137 > ./red.000137.out 2>&1\n",
      "    ./red.sh 138 > ./red.000138.out 2>&1\n",
      "    ./red.sh 139 > ./red.000139.out 2>&1\n",
      "    ./red.sh 140 > ./red.000140.out 2>&1\n",
      "    ./red.sh 141 > ./red.000141.out 2>&1\n",
      "    ./red.sh 142 > ./red.000142.out 2>&1\n",
      "    ./red.sh 143 > ./red.000143.out 2>&1\n",
      "    ./red.sh 144 > ./red.000144.out 2>&1\n",
      "    ./red.sh 145 > ./red.000145.out 2>&1\n",
      "\n",
      "-- Finished on Sat Oct 21 01:15:26 2017 (306 seconds) with 831.56 GB free disk space\n",
      "----------------------------------------\n",
      "-- Found 145 read error detection output files.\n",
      "-- Finished stage 'readErrorDetectionCheck', reset canuIteration.\n",
      "\n",
      "Configure OEA for 2gb memory with batches of at most 25000 reads and (unlimited) bases.\n",
      "Configured 29 jobs.\n",
      "\n",
      "OEA job   1 from read         1 to read     25000 - 103.2 bases + 448.8 adjusts +  0.8 reads + 26.9 olaps +  8.0 fseq/rseq + 32.0 fadj/radj + 32.0 work + 256.0 misc = 907.6 MB\n",
      "OEA job   2 from read     25001 to read     50000 - 123.1 bases + 448.8 adjusts +  0.8 reads + 26.6 olaps +  8.0 fseq/rseq + 32.0 fadj/radj + 32.0 work + 256.0 misc = 927.3 MB\n",
      "OEA job   3 from read     50001 to read     75000 - 122.7 bases + 448.8 adjusts +  0.8 reads + 27.3 olaps +  8.0 fseq/rseq + 32.0 fadj/radj + 32.0 work + 256.0 misc = 927.6 MB\n",
      "OEA job   4 from read     75001 to read    100000 - 122.0 bases + 448.8 adjusts +  0.8 reads + 31.9 olaps +  8.0 fseq/rseq + 32.0 fadj/radj + 32.0 work + 256.0 misc = 931.4 MB\n",
      "OEA job   5 from read    100001 to read    125000 - 122.2 bases + 448.8 adjusts +  0.8 reads + 31.4 olaps +  8.0 fseq/rseq + 32.0 fadj/radj + 32.0 work + 256.0 misc = 931.2 MB\n",
      "OEA job   6 from read    125001 to read    150000 - 107.0 bases + 448.8 adjusts +  0.8 reads + 23.8 olaps +  8.0 fseq/rseq + 32.0 fadj/radj + 32.0 work + 256.0 misc = 908.3 MB\n",
      "OEA job   7 from read    150001 to read    175000 - 107.5 bases + 448.8 adjusts +  0.8 reads + 20.0 olaps +  8.0 fseq/rseq + 32.0 fadj/radj + 32.0 work + 256.0 misc = 905.1 MB\n",
      "OEA job   8 from read    175001 to read    200000 - 109.8 bases + 448.8 adjusts +  0.8 reads + 24.5 olaps +  8.0 fseq/rseq + 32.0 fadj/radj + 32.0 work + 256.0 misc = 911.8 MB\n",
      "OEA job   9 from read    200001 to read    225000 - 108.5 bases + 448.8 adjusts +  0.8 reads + 24.6 olaps +  8.0 fseq/rseq + 32.0 fadj/radj + 32.0 work + 256.0 misc = 910.6 MB\n",
      "OEA job  10 from read    225001 to read    250000 - 107.7 bases + 448.8 adjusts +  0.8 reads + 23.5 olaps +  8.0 fseq/rseq + 32.0 fadj/radj + 32.0 work + 256.0 misc = 908.8 MB\n",
      "OEA job  11 from read    250001 to read    275000 - 107.3 bases + 448.8 adjusts +  0.8 reads + 23.3 olaps +  8.0 fseq/rseq + 32.0 fadj/radj + 32.0 work + 256.0 misc = 908.2 MB\n",
      "OEA job  12 from read    275001 to read    300000 - 108.8 bases + 448.8 adjusts +  0.8 reads + 23.7 olaps +  8.0 fseq/rseq + 32.0 fadj/radj + 32.0 work + 256.0 misc = 910.0 MB\n",
      "OEA job  13 from read    300001 to read    325000 - 109.3 bases + 448.8 adjusts +  0.8 reads + 22.9 olaps +  8.0 fseq/rseq + 32.0 fadj/radj + 32.0 work + 256.0 misc = 909.7 MB\n",
      "OEA job  14 from read    325001 to read    350000 - 107.7 bases + 448.8 adjusts +  0.8 reads + 22.2 olaps +  8.0 fseq/rseq + 32.0 fadj/radj + 32.0 work + 256.0 misc = 907.5 MB\n",
      "OEA job  15 from read    350001 to read    375000 - 110.2 bases + 448.8 adjusts +  0.8 reads + 26.1 olaps +  8.0 fseq/rseq + 32.0 fadj/radj + 32.0 work + 256.0 misc = 913.8 MB\n",
      "OEA job  16 from read    375001 to read    400000 - 108.4 bases + 448.8 adjusts +  0.8 reads + 21.9 olaps +  8.0 fseq/rseq + 32.0 fadj/radj + 32.0 work + 256.0 misc = 907.8 MB\n",
      "OEA job  17 from read    400001 to read    425000 - 108.6 bases + 448.8 adjusts +  0.8 reads + 23.4 olaps +  8.0 fseq/rseq + 32.0 fadj/radj + 32.0 work + 256.0 misc = 909.5 MB\n",
      "OEA job  18 from read    425001 to read    450000 - 114.8 bases + 448.8 adjusts +  0.8 reads + 25.6 olaps +  8.0 fseq/rseq + 32.0 fadj/radj + 32.0 work + 256.0 misc = 918.0 MB\n",
      "OEA job  19 from read    450001 to read    475000 - 121.9 bases + 448.8 adjusts +  0.8 reads + 28.9 olaps +  8.0 fseq/rseq + 32.0 fadj/radj + 32.0 work + 256.0 misc = 928.4 MB\n",
      "OEA job  20 from read    475001 to read    500000 - 122.2 bases + 448.8 adjusts +  0.8 reads + 25.9 olaps +  8.0 fseq/rseq + 32.0 fadj/radj + 32.0 work + 256.0 misc = 925.7 MB\n",
      "OEA job  21 from read    500001 to read    525000 - 123.6 bases + 448.8 adjusts +  0.8 reads + 27.3 olaps +  8.0 fseq/rseq + 32.0 fadj/radj + 32.0 work + 256.0 misc = 928.5 MB\n",
      "OEA job  22 from read    525001 to read    550000 - 124.0 bases + 448.8 adjusts +  0.8 reads + 31.3 olaps +  8.0 fseq/rseq + 32.0 fadj/radj + 32.0 work + 256.0 misc = 932.8 MB\n",
      "OEA job  23 from read    550001 to read    575000 - 123.8 bases + 448.8 adjusts +  0.8 reads + 31.8 olaps +  8.0 fseq/rseq + 32.0 fadj/radj + 32.0 work + 256.0 misc = 933.1 MB\n",
      "OEA job  24 from read    575001 to read    600000 - 123.4 bases + 448.8 adjusts +  0.8 reads + 31.9 olaps +  8.0 fseq/rseq + 32.0 fadj/radj + 32.0 work + 256.0 misc = 932.9 MB\n",
      "OEA job  25 from read    600001 to read    625000 - 123.3 bases + 448.8 adjusts +  0.8 reads + 30.2 olaps +  8.0 fseq/rseq + 32.0 fadj/radj + 32.0 work + 256.0 misc = 931.1 MB\n",
      "OEA job  26 from read    625001 to read    650000 - 123.4 bases + 448.8 adjusts +  0.8 reads + 37.5 olaps +  8.0 fseq/rseq + 32.0 fadj/radj + 32.0 work + 256.0 misc = 938.5 MB\n",
      "OEA job  27 from read    650001 to read    675000 - 122.5 bases + 448.8 adjusts +  0.8 reads + 28.2 olaps +  8.0 fseq/rseq + 32.0 fadj/radj + 32.0 work + 256.0 misc = 928.2 MB\n",
      "OEA job  28 from read    675001 to read    700000 - 120.9 bases + 448.8 adjusts +  0.8 reads + 38.7 olaps +  8.0 fseq/rseq + 32.0 fadj/radj + 32.0 work + 256.0 misc = 937.2 MB\n",
      "OEA job  29 from read    700001 to read    723260 - 113.1 bases + 448.8 adjusts +  0.7 reads + 26.4 olaps +  8.0 fseq/rseq + 32.0 fadj/radj + 32.0 work + 256.0 misc = 917.0 MB\n",
      "\n",
      "-- Finished stage 'overlapErrorAdjustmentConfigure', reset canuIteration.\n",
      "--\n",
      "-- Running jobs.  First attempt out of 2.\n",
      "----------------------------------------\n",
      "-- Starting 'oea' concurrent execution on Sat Oct 21 01:15:32 2017 with 831.56 GB free disk space (29 processes; 24 concurrently)\n",
      "\n",
      "    cd unitigging/3-overlapErrorAdjustment\n",
      "    ./oea.sh 1 > ./oea.000001.out 2>&1\n",
      "    ./oea.sh 2 > ./oea.000002.out 2>&1\n",
      "    ./oea.sh 3 > ./oea.000003.out 2>&1\n",
      "    ./oea.sh 4 > ./oea.000004.out 2>&1\n",
      "    ./oea.sh 5 > ./oea.000005.out 2>&1\n",
      "    ./oea.sh 6 > ./oea.000006.out 2>&1\n",
      "    ./oea.sh 7 > ./oea.000007.out 2>&1\n",
      "    ./oea.sh 8 > ./oea.000008.out 2>&1\n",
      "    ./oea.sh 9 > ./oea.000009.out 2>&1\n",
      "    ./oea.sh 10 > ./oea.000010.out 2>&1\n",
      "    ./oea.sh 11 > ./oea.000011.out 2>&1\n",
      "    ./oea.sh 12 > ./oea.000012.out 2>&1\n",
      "    ./oea.sh 13 > ./oea.000013.out 2>&1\n",
      "    ./oea.sh 14 > ./oea.000014.out 2>&1\n",
      "    ./oea.sh 15 > ./oea.000015.out 2>&1\n",
      "    ./oea.sh 16 > ./oea.000016.out 2>&1\n",
      "    ./oea.sh 17 > ./oea.000017.out 2>&1\n",
      "    ./oea.sh 18 > ./oea.000018.out 2>&1\n",
      "    ./oea.sh 19 > ./oea.000019.out 2>&1\n",
      "    ./oea.sh 20 > ./oea.000020.out 2>&1\n",
      "    ./oea.sh 21 > ./oea.000021.out 2>&1\n",
      "    ./oea.sh 22 > ./oea.000022.out 2>&1\n",
      "    ./oea.sh 23 > ./oea.000023.out 2>&1\n",
      "    ./oea.sh 24 > ./oea.000024.out 2>&1\n",
      "    ./oea.sh 25 > ./oea.000025.out 2>&1\n",
      "    ./oea.sh 26 > ./oea.000026.out 2>&1\n",
      "    ./oea.sh 27 > ./oea.000027.out 2>&1\n",
      "    ./oea.sh 28 > ./oea.000028.out 2>&1\n",
      "    ./oea.sh 29 > ./oea.000029.out 2>&1\n",
      "\n",
      "-- Finished on Sat Oct 21 01:19:43 2017 (251 seconds) with 831.51 GB free disk space\n",
      "----------------------------------------\n",
      "-- Found 29 overlap error adjustment output files.\n",
      "-- Finished stage 'overlapErrorAdjustmentCheck', reset canuIteration.\n",
      "----------------------------------------\n",
      "-- Starting command on Sat Oct 21 01:19:43 2017 with 831.51 GB free disk space\n",
      "\n",
      "    cd unitigging/3-overlapErrorAdjustment\n",
      "    /usr/local/bioinformatics/canu-1.6/Linux-amd64/bin/ovStoreBuild \\\n",
      "      -G ../20171018_oly_pacbio.gkpStore \\\n",
      "      -O ../20171018_oly_pacbio.ovlStore \\\n",
      "      -evalues \\\n",
      "      -L ./oea.files \\\n",
      "    > ./oea.apply.err 2>&1\n",
      "\n",
      "-- Finished on Sat Oct 21 01:19:43 2017 (lickety-split) with 831.462 GB free disk space\n",
      "----------------------------------------\n",
      "-- No report available.\n",
      "-- Finished stage 'updateOverlapStore', reset canuIteration.\n",
      "-- Finished stage 'unitig', reset canuIteration.\n",
      "--\n",
      "-- Running jobs.  First attempt out of 2.\n",
      "----------------------------------------\n",
      "-- Starting 'bat' concurrent execution on Sat Oct 21 01:19:43 2017 with 831.462 GB free disk space (1 processes; 2 concurrently)\n",
      "\n",
      "    cd unitigging/4-unitigger\n",
      "    ./unitigger.sh 1 > ./unitigger.000001.out 2>&1\n",
      "\n",
      "-- Finished on Sat Oct 21 01:19:57 2017 (14 seconds) with 831.201 GB free disk space\n",
      "----------------------------------------\n",
      "-- Unitigger finished successfully.\n",
      "-- Found, in version 1, after unitig construction:\n",
      "--   contigs:      11246 sequences, total length 132675816 bp (including 326 repeats of total length 2418909 bp).\n",
      "--   bubbles:      0 sequences, total length 0 bp.\n",
      "--   unassembled:  479693 sequences, total length 2277551160 bp.\n",
      "--\n",
      "-- Contig sizes based on genome size --\n",
      "--            NG (bp)  LG (contigs)    sum (bp)\n",
      "--         ----------  ------------  ----------\n",
      "--\n",
      "-- Finished stage 'unitigCheck', reset canuIteration.\n",
      "----------------------------------------\n",
      "-- Starting command on Sat Oct 21 01:19:58 2017 with 831.201 GB free disk space\n",
      "\n",
      "    cd unitigging\n",
      "    /usr/local/bioinformatics/canu-1.6/Linux-amd64/bin/gatekeeperPartition \\\n",
      "      -G ./20171018_oly_pacbio.gkpStore \\\n",
      "      -T ./20171018_oly_pacbio.ctgStore 1 \\\n",
      "      -b 25000 \\\n",
      "      -p 256 \\\n",
      "    > ./20171018_oly_pacbio.ctgStore/partitionedReads.log 2>&1\n",
      "\n",
      "-- Finished on Sat Oct 21 01:19:59 2017 (1 second) with 830.25 GB free disk space\n",
      "----------------------------------------\n",
      "----------------------------------------\n",
      "-- Starting command on Sat Oct 21 01:20:00 2017 with 830.25 GB free disk space\n",
      "\n",
      "    cd unitigging\n",
      "    /usr/local/bioinformatics/canu-1.6/Linux-amd64/bin/gatekeeperPartition \\\n",
      "      -G ./20171018_oly_pacbio.gkpStore \\\n",
      "      -T ./20171018_oly_pacbio.utgStore 1 \\\n",
      "      -b 25000 \\\n",
      "      -p 256 \\\n",
      "    > ./20171018_oly_pacbio.utgStore/partitionedReads.log 2>&1\n",
      "\n",
      "-- Finished on Sat Oct 21 01:20:00 2017 (1 second) with 830.005 GB free disk space\n",
      "----------------------------------------\n",
      "-- Using slow alignment for consensus (iteration '0').\n",
      "-- Configured 30 contig and 7 unitig consensus jobs.\n",
      "-- Finished stage 'consensusConfigure', reset canuIteration.\n",
      "--\n",
      "-- Running jobs.  First attempt out of 2.\n",
      "----------------------------------------\n",
      "-- Starting 'cns' concurrent execution on Sat Oct 21 01:20:01 2017 with 830.005 GB free disk space (37 processes; 3 concurrently)\n",
      "\n",
      "    cd unitigging/5-consensus\n",
      "    ./consensus.sh 1 > ./consensus.000001.out 2>&1\n",
      "    ./consensus.sh 2 > ./consensus.000002.out 2>&1\n",
      "    ./consensus.sh 3 > ./consensus.000003.out 2>&1\n",
      "    ./consensus.sh 4 > ./consensus.000004.out 2>&1\n",
      "    ./consensus.sh 5 > ./consensus.000005.out 2>&1\n",
      "    ./consensus.sh 6 > ./consensus.000006.out 2>&1\n",
      "    ./consensus.sh 7 > ./consensus.000007.out 2>&1\n",
      "    ./consensus.sh 8 > ./consensus.000008.out 2>&1\n",
      "    ./consensus.sh 9 > ./consensus.000009.out 2>&1\n",
      "    ./consensus.sh 10 > ./consensus.000010.out 2>&1\n",
      "    ./consensus.sh 11 > ./consensus.000011.out 2>&1\n",
      "    ./consensus.sh 12 > ./consensus.000012.out 2>&1\n",
      "    ./consensus.sh 13 > ./consensus.000013.out 2>&1\n",
      "    ./consensus.sh 14 > ./consensus.000014.out 2>&1\n",
      "    ./consensus.sh 15 > ./consensus.000015.out 2>&1\n",
      "    ./consensus.sh 16 > ./consensus.000016.out 2>&1\n",
      "    ./consensus.sh 17 > ./consensus.000017.out 2>&1\n",
      "    ./consensus.sh 18 > ./consensus.000018.out 2>&1\n",
      "    ./consensus.sh 19 > ./consensus.000019.out 2>&1\n",
      "    ./consensus.sh 20 > ./consensus.000020.out 2>&1\n",
      "    ./consensus.sh 21 > ./consensus.000021.out 2>&1\n",
      "    ./consensus.sh 22 > ./consensus.000022.out 2>&1\n",
      "    ./consensus.sh 23 > ./consensus.000023.out 2>&1\n",
      "    ./consensus.sh 24 > ./consensus.000024.out 2>&1\n",
      "    ./consensus.sh 25 > ./consensus.000025.out 2>&1\n",
      "    ./consensus.sh 26 > ./consensus.000026.out 2>&1\n",
      "    ./consensus.sh 27 > ./consensus.000027.out 2>&1\n",
      "    ./consensus.sh 28 > ./consensus.000028.out 2>&1\n",
      "    ./consensus.sh 29 > ./consensus.000029.out 2>&1\n",
      "    ./consensus.sh 30 > ./consensus.000030.out 2>&1\n",
      "    ./consensus.sh 31 > ./consensus.000031.out 2>&1\n",
      "    ./consensus.sh 32 > ./consensus.000032.out 2>&1\n",
      "    ./consensus.sh 33 > ./consensus.000033.out 2>&1\n",
      "    ./consensus.sh 34 > ./consensus.000034.out 2>&1\n",
      "    ./consensus.sh 35 > ./consensus.000035.out 2>&1\n",
      "    ./consensus.sh 36 > ./consensus.000036.out 2>&1\n",
      "    ./consensus.sh 37 > ./consensus.000037.out 2>&1\n",
      "\n",
      "-- Finished on Sat Oct 21 01:46:57 2017 (1616 seconds) with 825.166 GB free disk space\n",
      "----------------------------------------\n",
      "-- Using slow alignment for consensus (iteration '1').\n",
      "-- Configured 30 contig and 7 unitig consensus jobs.\n",
      "-- All 37 consensus jobs finished successfully.\n",
      "-- Finished stage 'consensusCheck', reset canuIteration.\n",
      "-- Using slow alignment for consensus (iteration '0').\n",
      "-- Configured 30 contig and 7 unitig consensus jobs.\n",
      "----------------------------------------\n",
      "-- Starting command on Sat Oct 21 01:46:58 2017 with 825.166 GB free disk space\n",
      "\n",
      "    cd unitigging\n",
      "    /usr/local/bioinformatics/canu-1.6/Linux-amd64/bin/tgStoreLoad \\\n",
      "      -G ./20171018_oly_pacbio.gkpStore \\\n",
      "      -T ./20171018_oly_pacbio.ctgStore 2 \\\n",
      "      -L ./5-consensus/ctgcns.files \\\n",
      "    > ./5-consensus/ctgcns.files.ctgStoreLoad.err 2>&1\n",
      "\n",
      "-- Finished on Sat Oct 21 01:47:04 2017 (6 seconds) with 820.591 GB free disk space\n",
      "----------------------------------------\n",
      "----------------------------------------\n",
      "-- Starting command on Sat Oct 21 01:47:05 2017 with 820.595 GB free disk space\n",
      "\n",
      "    cd unitigging\n",
      "    /usr/local/bioinformatics/canu-1.6/Linux-amd64/bin/tgStoreLoad \\\n",
      "      -G ./20171018_oly_pacbio.gkpStore \\\n",
      "      -T ./20171018_oly_pacbio.utgStore 2 \\\n",
      "      -L ./5-consensus/utgcns.files \\\n",
      "    > ./5-consensus/utgcns.files.utgStoreLoad.err 2>&1\n",
      "\n",
      "-- Finished on Sat Oct 21 01:47:05 2017 (1 second) with 820.306 GB free disk space\n",
      "----------------------------------------\n",
      "-- Purging consensus output after loading to ctgStore and/or utgStore.\n",
      "-- Purged 37 .cns outputs.\n",
      "----------------------------------------\n",
      "-- Starting command on Sat Oct 21 01:47:06 2017 with 826.331 GB free disk space\n",
      "\n",
      "    cd unitigging\n",
      "    /usr/local/bioinformatics/canu-1.6/Linux-amd64/bin/tgStoreDump \\\n",
      "      -G ./20171018_oly_pacbio.gkpStore \\\n",
      "      -T ./20171018_oly_pacbio.ctgStore 2 \\\n",
      "      -sizes -s 1900000000 \\\n",
      "    > ./20171018_oly_pacbio.ctgStore/seqDB.v002.sizes.txt\n",
      "\n",
      "-- Finished on Sat Oct 21 01:47:18 2017 (12 seconds) with 826.331 GB free disk space\n",
      "----------------------------------------\n",
      "-- Found, in version 2, after consensus generation:\n",
      "--   contigs:      11246 sequences, total length 132146852 bp (including 326 repeats of total length 2387102 bp).\n",
      "--   bubbles:      0 sequences, total length 0 bp.\n",
      "--   unassembled:  479693 sequences, total length 2277137864 bp.\n",
      "--\n",
      "-- Contig sizes based on genome size --\n",
      "--            NG (bp)  LG (contigs)    sum (bp)\n",
      "--         ----------  ------------  ----------\n",
      "--\n",
      "-- Finished stage 'consensusLoad', reset canuIteration.\n",
      "----------------------------------------\n",
      "-- Starting command on Sat Oct 21 01:47:18 2017 with 826.331 GB free disk space\n",
      "\n",
      "    cd unitigging\n",
      "    /usr/local/bioinformatics/canu-1.6/Linux-amd64/bin/tgStoreCoverageStat \\\n",
      "      -G ./20171018_oly_pacbio.gkpStore \\\n",
      "      -T ./20171018_oly_pacbio.ctgStore 2 \\\n",
      "      -s 1900000000 \\\n",
      "      -o ./20171018_oly_pacbio.ctgStore.coverageStat \\\n",
      "    > ./20171018_oly_pacbio.ctgStore.coverageStat.err 2>&1\n",
      "\n",
      "-- Finished on Sat Oct 21 01:47:23 2017 (5 seconds) with 826.311 GB free disk space\n",
      "----------------------------------------\n",
      "-- Finished stage 'consensusAnalyze', reset canuIteration.\n",
      "--\n",
      "-- Running jobs.  First attempt out of 2.\n",
      "----------------------------------------\n",
      "-- Starting 'gfa' concurrent execution on Sat Oct 21 01:47:33 2017 with 826.311 GB free disk space (1 processes; 2 concurrently)\n",
      "\n",
      "    cd unitigging/4-unitigger\n",
      "    ./alignGFA.sh 1 > ./alignGFA.000001.out 2>&1\n",
      "\n",
      "-- Finished on Sat Oct 21 01:48:22 2017 (49 seconds) with 826.3 GB free disk space\n",
      "----------------------------------------\n",
      "----------------------------------------\n",
      "-- Starting command on Sat Oct 21 01:48:22 2017 with 826.3 GB free disk space\n",
      "\n",
      "    cd .\n",
      "    /usr/local/bioinformatics/canu-1.6/Linux-amd64/bin/tgStoreDump \\\n",
      "      -G ./unitigging/20171018_oly_pacbio.gkpStore \\\n",
      "      -T ./unitigging/20171018_oly_pacbio.ctgStore 2 \\\n",
      "      -o ./20171018_oly_pacbio.contigs \\\n",
      "      -layout \\\n",
      "    > ./20171018_oly_pacbio.contigs.layout.err 2>&1\n",
      "\n",
      "-- Finished on Sat Oct 21 01:48:52 2017 (30 seconds) with 821.639 GB free disk space\n",
      "----------------------------------------\n",
      "----------------------------------------\n",
      "-- Starting command on Sat Oct 21 01:48:52 2017 with 821.639 GB free disk space\n",
      "\n",
      "    cd .\n",
      "    /usr/local/bioinformatics/canu-1.6/Linux-amd64/bin/tgStoreDump \\\n",
      "      -G ./unitigging/20171018_oly_pacbio.gkpStore \\\n",
      "      -T ./unitigging/20171018_oly_pacbio.utgStore 2 \\\n",
      "      -o ./20171018_oly_pacbio.unitigs \\\n",
      "      -layout \\\n",
      "    > ./20171018_oly_pacbio.unitigs.layout.err 2>&1\n",
      "\n",
      "-- Finished on Sat Oct 21 01:48:54 2017 (2 seconds) with 821.34 GB free disk space\n",
      "----------------------------------------\n",
      "----------------------------------------\n",
      "-- Starting command on Sat Oct 21 01:48:54 2017 with 821.34 GB free disk space\n",
      "\n",
      "    cd .\n",
      "    /usr/local/bioinformatics/canu-1.6/Linux-amd64/bin/tgStoreDump \\\n",
      "      -G ./unitigging/20171018_oly_pacbio.gkpStore \\\n",
      "      -T ./unitigging/20171018_oly_pacbio.ctgStore 2 \\\n",
      "      -consensus -fasta \\\n",
      "      -unassembled \\\n",
      "    > ./20171018_oly_pacbio.unassembled.fasta\n",
      "    2> ./20171018_oly_pacbio.unassembled.err\n",
      "\n",
      "-- Finished on Sat Oct 21 01:49:16 2017 (22 seconds) with 819.15 GB free disk space\n",
      "----------------------------------------\n",
      "----------------------------------------\n",
      "-- Starting command on Sat Oct 21 01:49:16 2017 with 819.15 GB free disk space\n",
      "\n",
      "    cd .\n",
      "    /usr/local/bioinformatics/canu-1.6/Linux-amd64/bin/tgStoreDump \\\n",
      "      -G ./unitigging/20171018_oly_pacbio.gkpStore \\\n",
      "      -T ./unitigging/20171018_oly_pacbio.ctgStore 2 \\\n",
      "      -consensus -fasta \\\n",
      "      -contigs \\\n",
      "    > ./20171018_oly_pacbio.contigs.fasta\n",
      "    2> ./20171018_oly_pacbio.contigs.err\n",
      "\n",
      "-- Finished on Sat Oct 21 01:49:28 2017 (12 seconds) with 819.025 GB free disk space\n",
      "----------------------------------------\n",
      "----------------------------------------\n",
      "-- Starting command on Sat Oct 21 01:49:28 2017 with 819.025 GB free disk space\n",
      "\n",
      "    cd .\n",
      "    /usr/local/bioinformatics/canu-1.6/Linux-amd64/bin/tgStoreDump \\\n",
      "      -G ./unitigging/20171018_oly_pacbio.gkpStore \\\n",
      "      -T ./unitigging/20171018_oly_pacbio.utgStore 2 \\\n",
      "      -consensus -fasta \\\n",
      "      -contigs \\\n",
      "    > ./20171018_oly_pacbio.unitigs.fasta\n",
      "    2> ./20171018_oly_pacbio.unitigs.err\n",
      "\n",
      "-- Finished on Sat Oct 21 01:49:29 2017 (1 second) with 818.883 GB free disk space\n",
      "----------------------------------------\n",
      "-- Finished stage 'generateOutputs', reset canuIteration.\n",
      "--\n",
      "-- Assembly '20171018_oly_pacbio' finished in '/home/data/20171018_oly_pacbio_canu'.\n",
      "--\n",
      "-- Summary saved in 'unitigging.html'.\n",
      "--\n",
      "-- Sequences saved:\n",
      "--   Contigs       -> '20171018_oly_pacbio.contigs.fasta'\n",
      "--   Unassembled   -> '20171018_oly_pacbio.unassembled.fasta'\n",
      "--   Unitigs       -> '20171018_oly_pacbio.unitigs.fasta'\n",
      "--\n",
      "-- Read layouts saved:\n",
      "--   Contigs       -> '20171018_oly_pacbio.contigs.layout'.\n",
      "--   Unitigs       -> '20171018_oly_pacbio.unitigs.layout'.\n",
      "--\n",
      "-- Graphs saved:\n",
      "--   Contigs       -> '20171018_oly_pacbio.contigs.gfa'.\n",
      "--   Unitigs       -> '20171018_oly_pacbio.unitigs.gfa'.\n",
      "--\n",
      "-- Bye.\n",
      "-- Finished stage 'outputSequence', reset canuIteration.\n",
      "--\n",
      "-- Bye.\n",
      "\n",
      "real\t3071m38.266s\n",
      "user\t53627m42.644s\n",
      "sys\t478m24.228s\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "time canu \\\n",
    "useGrid=false \\\n",
    "ovsMethod=sequential \\\n",
    "-p 20171018_oly_pacbio \\\n",
    "-d /home/data/20171018_oly_pacbio_canu/ \\\n",
    "genomeSize=1.9g \\\n",
    "correctedErrorRate=0.075 \\\n",
    "-pacbio-raw m*"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "total 13G\n",
      "-rw-r--r--  1 root root 129M Oct 21 01:49 20171018_oly_pacbio.contigs.fasta\n",
      "-rw-r--r--  1 root root 2.2M Oct 21 01:49 20171018_oly_pacbio.contigs.gfa\n",
      "-rw-r--r--  1 root root 4.7G Oct 21 01:48 20171018_oly_pacbio.contigs.layout\n",
      "-rw-r--r--  1 root root  21M Oct 21 01:48 20171018_oly_pacbio.contigs.layout.readToTig\n",
      "-rw-r--r--  1 root root  22M Oct 21 01:48 20171018_oly_pacbio.contigs.layout.tigInfo\n",
      "-rw-r--r--  1 root root 3.6G Oct 20 11:50 20171018_oly_pacbio.correctedReads.fasta.gz\n",
      "-rw-r--r--  1 root root  24M Oct 20 11:50 20171018_oly_pacbio.correctedReads.length\n",
      "-rw-r--r--  1 root root  27K Oct 21 01:49 20171018_oly_pacbio.report\n",
      "-rw-r--r--  1 root root 1.2G Oct 20 19:58 20171018_oly_pacbio.trimmedReads.fasta.gz\n",
      "-rw-r--r--  1 root root 2.2G Oct 21 01:49 20171018_oly_pacbio.unassembled.fasta\n",
      "-rw-r--r--  1 root root 504K Oct 21 01:49 20171018_oly_pacbio.unitigs.bed\n",
      "-rw-r--r--  1 root root 146M Oct 21 01:49 20171018_oly_pacbio.unitigs.fasta\n",
      "-rw-r--r--  1 root root 8.5M Oct 21 01:49 20171018_oly_pacbio.unitigs.gfa\n",
      "-rw-r--r--  1 root root 301M Oct 21 01:48 20171018_oly_pacbio.unitigs.layout\n",
      "-rw-r--r--  1 root root 5.0M Oct 21 01:48 20171018_oly_pacbio.unitigs.layout.readToTig\n",
      "-rw-r--r--  1 root root 640K Oct 21 01:48 20171018_oly_pacbio.unitigs.layout.tigInfo\n",
      "drwxr-xr-x  2 root root  60K Oct 21 01:49 canu-logs\n",
      "drwxr-xr-x  2 root root 4.0K Oct 18 22:37 canu-scripts\n",
      "drwxr-xr-x  6 root root 4.0K Oct 20 11:52 correction\n",
      "-rw-r--r--  1 root root  19K Oct 20 11:52 correction.html\n",
      "drwxr-xr-x  2 root root 4.0K Oct 20 11:52 correction.html.files\n",
      "drwxr-xr-x  6 root root 4.0K Oct 20 19:58 trimming\n",
      "-rw-r--r--  1 root root 7.3K Oct 20 19:58 trimming.html\n",
      "drwxr-xr-x  2 root root 4.0K Oct 20 12:50 trimming.html.files\n",
      "drwxr-xr-x 11 root root 4.0K Oct 21 01:47 unitigging\n",
      "-rw-r--r--  1 root root 7.4K Oct 21 01:49 unitigging.html\n",
      "drwxr-xr-x  2 root root 4.0K Oct 20 20:06 unitigging.html.files\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "ls -lh /home/data/20171018_oly_pacbio_canu/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Things look good! Will re-launch notebook in order to mount Owl for coping Canu output folder."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "UsageError: Cell magic `%%date` not found.\n"
     ]
    }
   ],
   "source": [
    "%%date"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Mon Oct 23 13:39:45 UTC 2017\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "date"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "real\t11m17.919s\n",
      "user\t0m0.180s\n",
      "sys\t0m43.088s\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "time cp -R 20171018_oly_pacbio_canu/ /home/owl/Athaliana/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "total 13G\n",
      "-rw-r--r--  1 1027 users 129M Oct 23 13:53 20171018_oly_pacbio.contigs.fasta\n",
      "-rw-r--r--  1 1027 users 2.2M Oct 23 13:53 20171018_oly_pacbio.contigs.gfa\n",
      "-rw-r--r--  1 1027 users 4.7G Oct 23 13:53 20171018_oly_pacbio.contigs.layout\n",
      "-rw-r--r--  1 1027 users  21M Oct 23 13:51 20171018_oly_pacbio.contigs.layout.readToTig\n",
      "-rw-r--r--  1 1027 users  22M Oct 23 13:51 20171018_oly_pacbio.contigs.layout.tigInfo\n",
      "-rw-r--r--  1 1027 users 3.6G Oct 23 13:46 20171018_oly_pacbio.correctedReads.fasta.gz\n",
      "-rw-r--r--  1 1027 users  24M Oct 23 13:45 20171018_oly_pacbio.correctedReads.length\n",
      "-rw-r--r--  1 1027 users  27K Oct 23 13:44 20171018_oly_pacbio.report\n",
      "-rw-r--r--  1 1027 users 1.2G Oct 23 13:50 20171018_oly_pacbio.trimmedReads.fasta.gz\n",
      "-rw-r--r--  1 1027 users 2.2G Oct 23 13:51 20171018_oly_pacbio.unassembled.fasta\n",
      "-rw-r--r--  1 1027 users 504K Oct 23 13:53 20171018_oly_pacbio.unitigs.bed\n",
      "-rw-r--r--  1 1027 users 146M Oct 23 13:53 20171018_oly_pacbio.unitigs.fasta\n",
      "-rw-r--r--  1 1027 users 8.5M Oct 23 13:53 20171018_oly_pacbio.unitigs.gfa\n",
      "-rw-r--r--  1 1027 users 301M Oct 23 13:53 20171018_oly_pacbio.unitigs.layout\n",
      "-rw-r--r--  1 1027 users 5.0M Oct 23 13:53 20171018_oly_pacbio.unitigs.layout.readToTig\n",
      "-rw-r--r--  1 1027 users 640K Oct 23 13:53 20171018_oly_pacbio.unitigs.layout.tigInfo\n",
      "drwxr-xr-x  2 root users    0 Oct 23 13:42 canu-logs\n",
      "drwxr-xr-x  2 root users    0 Oct 23 13:42 canu-scripts\n",
      "drwxr-xr-x  6 root users    0 Oct 23 13:43 correction\n",
      "-rw-r--r--  1 1027 users  19K Oct 23 13:45 correction.html\n",
      "drwxr-xr-x  2 root users    0 Oct 23 13:45 correction.html.files\n",
      "drwxr-xr-x  6 root users    0 Oct 23 13:47 trimming\n",
      "-rw-r--r--  1 1027 users 7.3K Oct 23 13:47 trimming.html\n",
      "drwxr-xr-x  2 root users    0 Oct 23 13:47 trimming.html.files\n",
      "drwxr-xr-x 11 root users    0 Oct 23 13:50 unitigging\n",
      "-rw-r--r--  1 1027 users 7.4K Oct 23 13:50 unitigging.html\n",
      "drwxr-xr-x  2 root users    0 Oct 23 13:50 unitigging.html.files\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "ls -lh /home/owl/Athaliana/20171018_oly_pacbio_canu/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This is the main file that is useful from this whole Canu assembly: [20171018_oly_pacbio.contigs.fasta](http://owl.fish.washington.edu/Athaliana/20171018_oly_pacbio_canu/20171018_oly_pacbio.contigs.fasta)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.5.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
