{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TODAY'S DATE:\n",
      "Wed May 23 09:58:07 PDT 2018\n",
      "------------\n",
      "\n",
      "Distributor ID:\tUbuntu\n",
      "Description:\tUbuntu 16.04.4 LTS\n",
      "Release:\t16.04\n",
      "Codename:\txenial\n",
      "\n",
      "------------\n",
      "HOSTNAME: \n",
      "roadrunner\n",
      "\n",
      "------------\n",
      "Computer Specs:\n",
      "\n",
      "Architecture:          x86_64\n",
      "CPU op-mode(s):        32-bit, 64-bit\n",
      "Byte Order:            Little Endian\n",
      "CPU(s):                16\n",
      "On-line CPU(s) list:   0-15\n",
      "Thread(s) per core:    2\n",
      "Core(s) per socket:    4\n",
      "Socket(s):             2\n",
      "NUMA node(s):          1\n",
      "Vendor ID:             GenuineIntel\n",
      "CPU family:            6\n",
      "Model:                 26\n",
      "Model name:            Intel(R) Xeon(R) CPU           E5520  @ 2.27GHz\n",
      "Stepping:              5\n",
      "CPU MHz:               2394.000\n",
      "CPU max MHz:           2394.0000\n",
      "CPU min MHz:           1596.0000\n",
      "BogoMIPS:              4521.78\n",
      "Virtualization:        VT-x\n",
      "L1d cache:             32K\n",
      "L1i cache:             32K\n",
      "L2 cache:              256K\n",
      "L3 cache:              8192K\n",
      "NUMA node0 CPU(s):     0-15\n",
      "Flags:                 fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx rdtscp lm constant_tsc arch_perfmon pebs bts nopl xtopology nonstop_tsc aperfmperf pni dtes64 monitor ds_cpl vmx est tm2 ssse3 cx16 xtpr pdcm dca sse4_1 sse4_2 popcnt lahf_lm kaiser tpr_shadow vnmi flexpriority ept vpid dtherm ida\n",
      "\n",
      "------------\n",
      "\n",
      "Memory Specs\n",
      "\n",
      "              total        used        free      shared  buff/cache   available\n",
      "Mem:            47G        419M         40G         50M        6.5G         46G\n",
      "Swap:           47G          0B         47G\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "No LSB modules are available.\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "echo \"TODAY'S DATE:\"\n",
    "date\n",
    "echo \"------------\"\n",
    "echo \"\"\n",
    "lsb_release -a #Display operating system info\n",
    "echo \"\"\n",
    "echo \"------------\"\n",
    "echo \"HOSTNAME: \"; hostname \n",
    "echo \"\"\n",
    "echo \"------------\"\n",
    "echo \"Computer Specs:\"\n",
    "echo \"\"\n",
    "lscpu\n",
    "echo \"\"\n",
    "echo \"------------\"\n",
    "echo \"\"\n",
    "echo \"Memory Specs\"\n",
    "echo \"\"\n",
    "free -mh"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "%%bash\n",
    "mkdir /home/sam/analyses/20180523_oly_repeatmasker"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cp: cannot stat '/mnt/owl/Athaliana/20171130_oly_pbjelly/jelly.out.fasta': No such file or directory\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "mkdir /home/sam/data/oly_assemblies\n",
    "cp /mnt/owl/Athaliana/20171130_oly_pbjelly/jelly.out.fasta /home/sam/data/oly_assemblies"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Forgot that I had restarted roadrunner - need to re-mount owl...\n",
    "\n",
    "Needs ```sudo``` so will re-run outside of notebook."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### Copy Olympia oyster assembly: [pbjelly_sjw_01](https://github.com/RobertsLab/project-olympia.oyster-genomic/wiki/Genome-Assemblies)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "real\t0m18.276s\n",
      "user\t0m0.000s\n",
      "sys\t0m1.368s\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "time \\\n",
    "cp /mnt/owl/Athaliana/20171130_oly_pbjelly/jelly.out.fasta /home/sam/data/oly_assemblies"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "total 1233740\n",
      "-rw-rw-r-- 1 sam sam 1263344875 May 23 10:13 jelly.out.fasta\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "ls -l /home/sam/data/oly_assemblies/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "%%bash\n",
    "mv /home/sam/analyses/20180523_oly_repeatmasker \\\n",
    "/home/sam/analyses/20180523_oly_repeatmasker_pbjelly_sjw_01-01"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### Run RepeatMasker with default setting and following options:\n",
    "\n",
    "- ```-par 15``` : Use 15 CPU threads\n",
    "\n",
    "- ```-gff``` : Create GFF output file (in addition to default files)\n",
    "\n",
    "- ```-excln``` : Adjusts output table calculations to exclude sequence runs of >=25Ns. Useful for draft genome assemblies.\n",
    "\n",
    "- ```1>``` : Send stdout to file instead of printing to notebook.\n",
    "\n",
    "- ```2>``` : Send stderr to file instead of printing to notebook.\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "real\t398m44.441s\n",
      "user\t4976m30.380s\n",
      "sys\t1006m29.860s\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "time \\\n",
    "/home/shared/RepeatMasker-4.0.7/RepeatMasker \\\n",
    "/home/sam/data/oly_assemblies/jelly.out.fasta \\\n",
    "-par 15 \\\n",
    "-gff \\\n",
    "-excln \\\n",
    "1> /home/sam/analyses/20180523_oly_repeatmasker_pbjelly_sjw_01-01/stdout.out \\\n",
    "2> /home/sam/analyses/20180523_oly_repeatmasker_pbjelly_sjw_01-01/stderr.err\n",
    "sed '/^Subject:/ s/ / repeatmasker JOB COMPLETE/' ~/.default-subject.mail | msmtp \"$EMAIL\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "------------\n",
      "STANDARD OUT\n",
      "------------\n",
      "cycle 4 .......................................................................................................................................................................................................................................................................................................................................................................................................................\n",
      "cycle 5 .......................................................................................................................................................................................................................................................................................................................................................................................................................\n",
      "cycle 6 .......................................................................................................................................................................................................................................................................................................................................................................................................................\n",
      "cycle 7 .......................................................................................................................................................................................................................................................................................................................................................................................................................\n",
      "cycle 8 ............................................................................................................................................................................................................................................................................................................................................................................................................\n",
      "cycle 9 ............................................................................................................................................................................................................................................................................................................................................................................................................\n",
      "cycle 10 ............................................................................................................................................................................................................................................................................................................................................................................................................\n",
      "Generating output... .........................................................................................................................................................................................................................................................................................................................................................................................................\n",
      "masking\n",
      "done\n",
      "\n",
      "\n",
      "------------\n",
      "STANDARD ERROR\n",
      "------------\n",
      "\n",
      "\n",
      "------------\n",
      "REPEATMASKER TABLE\n",
      "------------\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cat: /home/sam/analyses/20180523_oly_repeatmasker_pbjelly_sjw_01-01/jelly.out.fasta.tbl: No such file or directory\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "echo \"------------\"\n",
    "echo \"STANDARD OUT\"\n",
    "echo \"------------\"\n",
    "tail /home/sam/analyses/20180523_oly_repeatmasker_pbjelly_sjw_01-01/stdout.out\n",
    "\n",
    "echo \"\"\n",
    "echo \"\"\n",
    "\n",
    "echo \"------------\"\n",
    "echo \"STANDARD ERROR\"\n",
    "echo \"------------\"\n",
    "tail /home/sam/analyses/20180523_oly_repeatmasker_pbjelly_sjw_01-01/stderr.err\n",
    "\n",
    "echo \"\"\n",
    "echo \"\"\n",
    "\n",
    "echo \"------------\"\n",
    "echo \"REPEATMASKER TABLE\"\n",
    "echo \"------------\"\n",
    "cat /home/sam/analyses/20180523_oly_repeatmasker_pbjelly_sjw_01-01/jelly.out.fasta.tbl"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Oof. Ran this in the wrong directory AND started running repeatmasker after this. Can't tell if files are overwritten or what. Need to re-run. Grr."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "real\t408m50.418s\n",
      "user\t4841m17.924s\n",
      "sys\t1297m22.872s\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "cd /home/sam/analyses/20180523_oly_repeatmasker_pbjelly_sjw_01-01\n",
    "time \\\n",
    "/home/shared/RepeatMasker-4.0.7/RepeatMasker \\\n",
    "/home/sam/data/oly_assemblies/jelly.out.fasta \\\n",
    "-par 15 \\\n",
    "-gff \\\n",
    "-excln \\\n",
    "1> /home/sam/analyses/20180523_oly_repeatmasker_pbjelly_sjw_01-01/stdout.out \\\n",
    "2> /home/sam/analyses/20180523_oly_repeatmasker_pbjelly_sjw_01-01/stderr.err\n",
    "sed '/^Subject:/ s/ / repeatmasker JOB COMPLETE/' ~/.default-subject.mail | msmtp \"$EMAIL\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "------------\n",
      "STANDARD OUT\n",
      "------------\n",
      "cycle 4 .......................................................................................................................................................................................................................................................................................................................................................................................................................\n",
      "cycle 5 .......................................................................................................................................................................................................................................................................................................................................................................................................................\n",
      "cycle 6 .......................................................................................................................................................................................................................................................................................................................................................................................................................\n",
      "cycle 7 .......................................................................................................................................................................................................................................................................................................................................................................................................................\n",
      "cycle 8 ............................................................................................................................................................................................................................................................................................................................................................................................................\n",
      "cycle 9 ............................................................................................................................................................................................................................................................................................................................................................................................................\n",
      "cycle 10 ............................................................................................................................................................................................................................................................................................................................................................................................................\n",
      "Generating output... .........................................................................................................................................................................................................................................................................................................................................................................................................\n",
      "masking\n",
      "done\n",
      "\n",
      "\n",
      "------------\n",
      "STANDARD ERROR\n",
      "------------\n",
      "\n",
      "\n",
      "------------\n",
      "REPEATMASKER TABLE\n",
      "------------\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cat: /home/sam/analyses/20180523_oly_repeatmasker_pbjelly_sjw_01-01/jelly.out.fasta.tbl: No such file or directory\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "echo \"------------\"\n",
    "echo \"STANDARD OUT\"\n",
    "echo \"------------\"\n",
    "tail /home/sam/analyses/20180523_oly_repeatmasker_pbjelly_sjw_01-01/stdout.out\n",
    "\n",
    "echo \"\"\n",
    "echo \"\"\n",
    "\n",
    "echo \"------------\"\n",
    "echo \"STANDARD ERROR\"\n",
    "echo \"------------\"\n",
    "tail /home/sam/analyses/20180523_oly_repeatmasker_pbjelly_sjw_01-01/stderr.err\n",
    "\n",
    "echo \"\"\n",
    "echo \"\"\n",
    "\n",
    "echo \"------------\"\n",
    "echo \"REPEATMASKER TABLE\"\n",
    "echo \"------------\"\n",
    "cat /home/sam/analyses/20180523_oly_repeatmasker_pbjelly_sjw_01-01/jelly.out.fasta.tbl"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### Run RepeatMasker with _Crassostrea gigas_ species setting and following options:\n",
    "\n",
    "- ```-species \"crassostrea gigas\"``` : Sets species to Crassostrea gigas\n",
    "\n",
    "- ```-par 15``` : Use 15 CPU threads\n",
    "\n",
    "- ```-gff``` : Create GFF output file (in addition to default files)\n",
    "\n",
    "- ```-excln``` : Adjusts output table calculations to exclude sequence runs of >=25Ns. Useful for draft genome assemblies.\n",
    "\n",
    "- ```1>``` : Send stdout to file instead of printing to notebook.\n",
    "\n",
    "- ```2>``` : Send stderr to file instead of printing to notebook."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "mkdir: cannot create directory ‘/home/sam/analyses/20180523_oly_repeatmasker_pbjelly_sjw_01-02’: File exists\n",
      "\n",
      "real\t391m39.372s\n",
      "user\t4547m11.540s\n",
      "sys\t1003m40.176s\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "mkdir /home/sam/analyses/20180523_oly_repeatmasker_pbjelly_sjw_01-02\n",
    "cd /home/sam/analyses/20180523_oly_repeatmasker_pbjelly_sjw_01-02\n",
    "\n",
    "time \\\n",
    "/home/shared/RepeatMasker-4.0.7/RepeatMasker \\\n",
    "/home/sam/data/oly_assemblies/jelly.out.fasta \\\n",
    "-species \"crassostrea gigas\" \\\n",
    "-par 15 \\\n",
    "-gff \\\n",
    "-excln \\\n",
    "1> /home/sam/analyses/20180523_oly_repeatmasker_pbjelly_sjw_01-02/stdout.out \\\n",
    "2> /home/sam/analyses/20180523_oly_repeatmasker_pbjelly_sjw_01-02/stderr.err\n",
    "\n",
    "sed '/^Subject:/ s/ / repeatmasker JOB COMPLETE/' ~/.default-subject.mail | msmtp \"$EMAIL\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "------------\n",
      "STANDARD OUT\n",
      "------------\n",
      "cycle 4 .............................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................\n",
      "cycle 5 \n",
      "cycle 6 .............................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................\n",
      "cycle 7 .............................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................\n",
      "cycle 8 .............................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................\n",
      "cycle 9 .............................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................\n",
      "cycle 10 ............................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................\n",
      "Generating output... ....................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................\n",
      "masking\n",
      "done\n",
      "\n",
      "\n",
      "------------\n",
      "STANDARD ERROR\n",
      "------------\n",
      "\n",
      "\n",
      "------------\n",
      "REPEATMASKER C.GIGAS TABLE\n",
      "------------\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cat: /home/sam/analyses/20180523_oly_repeatmasker_pbjelly_sjw_01-02/jelly.out.fasta.tbl: No such file or directory\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "echo \"------------\"\n",
    "echo \"STANDARD OUT\"\n",
    "echo \"------------\"\n",
    "tail /home/sam/analyses/20180523_oly_repeatmasker_pbjelly_sjw_01-02/stdout.out\n",
    "\n",
    "echo \"\"\n",
    "echo \"\"\n",
    "\n",
    "echo \"------------\"\n",
    "echo \"STANDARD ERROR\"\n",
    "echo \"------------\"\n",
    "tail /home/sam/analyses/20180523_oly_repeatmasker_pbjelly_sjw_01-02/stderr.err\n",
    "\n",
    "echo \"\"\n",
    "echo \"\"\n",
    "\n",
    "echo \"------------\"\n",
    "echo \"REPEATMASKER C.GIGAS TABLE\"\n",
    "echo \"------------\"\n",
    "cat /home/sam/analyses/20180523_oly_repeatmasker_pbjelly_sjw_01-02/jelly.out.fasta.tbl"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### Run RepeatMasker with _Crassostrea virginica_ species setting and following options:\n",
    "\n",
    "- ```-species \"crassostrea virginica\"``` : Sets species to Crassostrea virginica\n",
    "\n",
    "- ```-par 15``` : Use 15 CPU threads\n",
    "\n",
    "- ```-gff``` : Create GFF output file (in addition to default files)\n",
    "\n",
    "- ```-excln``` : Adjusts output table calculations to exclude sequence runs of >=25Ns. Useful for draft genome assemblies.\n",
    "\n",
    "- ```1>``` : Send stdout to file instead of printing to notebook.\n",
    "\n",
    "- ```2>``` : Send stderr to file instead of printing to notebook."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "real\t301m41.551s\n",
      "user\t3123m55.312s\n",
      "sys\t1092m28.468s\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "mkdir /home/sam/analyses/20180523_oly_repeatmasker_pbjelly_sjw_01-03\n",
    "cd /home/sam/analyses/20180523_oly_repeatmasker_pbjelly_sjw_01-03\n",
    "time \\\n",
    "/home/shared/RepeatMasker-4.0.7/RepeatMasker \\\n",
    "/home/sam/data/oly_assemblies/jelly.out.fasta \\\n",
    "-species \"crassostrea virginica\" \\\n",
    "-par 15 \\\n",
    "-gff \\\n",
    "-excln \\\n",
    "1> /home/sam/analyses/20180523_oly_repeatmasker_pbjelly_sjw_01-03/stdout.out \\\n",
    "2> /home/sam/analyses/20180523_oly_repeatmasker_pbjelly_sjw_01-03/stderr.err\n",
    "\n",
    "sed '/^Subject:/ s/ / repeatmasker JOB COMPLETE/' ~/.default-subject.mail | msmtp \"$EMAIL\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "------------\n",
      "STANDARD OUT\n",
      "------------\n",
      "cycle 4 ....................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................\n",
      "cycle 5 \n",
      "cycle 6 ....................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................\n",
      "cycle 7 ....................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................\n",
      "cycle 8 ..................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................\n",
      "cycle 9 ..................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................\n",
      "cycle 10 ...................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................\n",
      "Generating output... ...............................................................................................................................................................................................................................................................................................................................................................................................................................................................................................\n",
      "masking\n",
      "done\n",
      "\n",
      "\n",
      "------------\n",
      "STANDARD ERROR\n",
      "------------\n",
      "\n",
      "\n",
      "------------\n",
      "REPEATMASKER C.VIRGINICA TABLE\n",
      "------------\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cat: /home/sam/analyses/20180523_oly_repeatmasker_pbjelly_sjw_01-03/jelly.out.fasta.tbl: No such file or directory\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "echo \"------------\"\n",
    "echo \"STANDARD OUT\"\n",
    "echo \"------------\"\n",
    "tail /home/sam/analyses/20180523_oly_repeatmasker_pbjelly_sjw_01-03/stdout.out\n",
    "\n",
    "echo \"\"\n",
    "echo \"\"\n",
    "\n",
    "echo \"------------\"\n",
    "echo \"STANDARD ERROR\"\n",
    "echo \"------------\"\n",
    "tail /home/sam/analyses/20180523_oly_repeatmasker_pbjelly_sjw_01-03/stderr.err\n",
    "\n",
    "echo \"\"\n",
    "echo \"\"\n",
    "\n",
    "echo \"------------\"\n",
    "echo \"REPEATMASKER C.VIRGINICA TABLE\"\n",
    "echo \"------------\"\n",
    "cat /home/sam/analyses/20180523_oly_repeatmasker_pbjelly_sjw_01-03/jelly.out.fasta.tbl"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### Run RepeatMasker with _Ostrea lurida_ species setting and following options:\n",
    "\n",
    "- ```-species \"ostrea lurida\"``` : Sets species to Ostrea lurida\n",
    "\n",
    "- ```-par 15``` : Use 15 CPU threads\n",
    "\n",
    "- ```-gff``` : Create GFF output file (in addition to default files)\n",
    "\n",
    "- ```-excln``` : Adjusts output table calculations to exclude sequence runs of >=25Ns. Useful for draft genome assemblies.\n",
    "\n",
    "- ```1>``` : Send stdout to file instead of printing to notebook.\n",
    "\n",
    "- ```2>``` : Send stderr to file instead of printing to notebook."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "real\t297m9.951s\n",
      "user\t3102m3.256s\n",
      "sys\t1096m47.180s\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "mkdir /home/sam/analyses/20180523_oly_repeatmasker_pbjelly_sjw_01-04\n",
    "cd /home/sam/analyses/20180523_oly_repeatmasker_pbjelly_sjw_01-04\n",
    "\n",
    "time \\\n",
    "/home/shared/RepeatMasker-4.0.7/RepeatMasker \\\n",
    "/home/sam/data/oly_assemblies/jelly.out.fasta \\\n",
    "-species \"ostrea lurida\" \\\n",
    "-par 15 \\\n",
    "-gff \\\n",
    "-excln \\\n",
    "1> /home/sam/analyses/20180523_oly_repeatmasker_pbjelly_sjw_01-04/stdout.out \\\n",
    "2> /home/sam/analyses/20180523_oly_repeatmasker_pbjelly_sjw_01-04/stderr.err\n",
    "\n",
    "sed '/^Subject:/ s/ / repeatmasker JOB COMPLETE/' ~/.default-subject.mail | msmtp \"$EMAIL\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "------------\n",
      "STANDARD OUT\n",
      "------------\n",
      "cycle 4 ..................................................................................................................................................................................................................................................................................................................................................................................................\n",
      "cycle 5 \n",
      "cycle 6 ..................................................................................................................................................................................................................................................................................................................................................................................................\n",
      "cycle 7 ..................................................................................................................................................................................................................................................................................................................................................................................................\n",
      "cycle 8 ...............................................................................................................................................................................................................................................................................................................................................................................\n",
      "cycle 9 ...............................................................................................................................................................................................................................................................................................................................................................................\n",
      "cycle 10 ...............................................................................................................................................................................................................................................................................................................................................................................\n",
      "Generating output... ...........................................................................................................................................................................................................................................................................................................................................................................\n",
      "masking\n",
      "done\n",
      "\n",
      "\n",
      "------------\n",
      "STANDARD ERROR\n",
      "------------\n",
      "\n",
      "\n",
      "------------\n",
      "REPEATMASKER O.LURIDA TABLE\n",
      "------------\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cat: /home/sam/analyses/20180523_oly_repeatmasker_pbjelly_sjw_01-04/jelly.out.fasta.tbl: No such file or directory\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "echo \"------------\"\n",
    "echo \"STANDARD OUT\"\n",
    "echo \"------------\"\n",
    "tail /home/sam/analyses/20180523_oly_repeatmasker_pbjelly_sjw_01-04/stdout.out\n",
    "\n",
    "echo \"\"\n",
    "echo \"\"\n",
    "\n",
    "echo \"------------\"\n",
    "echo \"STANDARD ERROR\"\n",
    "echo \"------------\"\n",
    "tail /home/sam/analyses/20180523_oly_repeatmasker_pbjelly_sjw_01-04/stderr.err\n",
    "\n",
    "echo \"\"\n",
    "echo \"\"\n",
    "\n",
    "echo \"------------\"\n",
    "echo \"REPEATMASKER O.LURIDA TABLE\"\n",
    "echo \"------------\"\n",
    "cat /home/sam/analyses/20180523_oly_repeatmasker_pbjelly_sjw_01-04/jelly.out.fasta.tbl"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### It turns out the reason none of the *.tbl files displayed is due to the location from RepeatMasker! It spits some of the output files to the same directory of the genome file and some to a subdirectory that it creates!"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "jelly.out.fasta\n",
      "jelly.out.fasta.preThuMay240214172018.RMoutput\n",
      "jelly.out.fasta.preThuMay240845562018.RMoutput\n",
      "jelly.out.fasta.preThuMay241347382018.RMoutput\n",
      "jelly.out.fasta.preThuMay241644392018.RMoutput\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "ls /home/sam/data/oly_assemblies/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### DEFAULT SETTINGS SUMMARY TABLE (_Homo sapiens_)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "==================================================\n",
      "file name: jelly.out.fasta          \n",
      "sequences:        696946\n",
      "total length: 1253001795 bp  (1172226648 bp excl N/X-runs)\n",
      "GC level:         36.51 %\n",
      "bases masked:   20002806 bp ( 1.71 %)\n",
      "==================================================\n",
      "               number of      length   percentage\n",
      "               elements*    occupied  of sequence\n",
      "--------------------------------------------------\n",
      "SINEs:            17794      1061170 bp    0.09 %\n",
      "      ALUs          363        31340 bp    0.00 %\n",
      "      MIRs         1166        92129 bp    0.01 %\n",
      "\n",
      "LINEs:             4456       888114 bp    0.08 %\n",
      "      LINE1         976       103929 bp    0.01 %\n",
      "      LINE2         813        82891 bp    0.01 %\n",
      "      L3/CR1        699        63627 bp    0.01 %\n",
      "\n",
      "LTR elements:      1187       199118 bp    0.02 %\n",
      "      ERVL          155        15828 bp    0.00 %\n",
      "      ERVL-MaLRs    200        20737 bp    0.00 %\n",
      "      ERV_classI    379        42833 bp    0.00 %\n",
      "      ERV_classII    66         6896 bp    0.00 %\n",
      "\n",
      "DNA elements:      2290       196866 bp    0.02 %\n",
      "     hAT-Charlie    190        15468 bp    0.00 %\n",
      "     TcMar-Tigger   732        37473 bp    0.00 %\n",
      "\n",
      "Unclassified:       101        12946 bp    0.00 %\n",
      "\n",
      "Total interspersed repeats:  2358214 bp    0.20 %\n",
      "\n",
      "\n",
      "Small RNA:         5954       433422 bp    0.04 %\n",
      "\n",
      "Satellites:         366        55705 bp    0.00 %\n",
      "Simple repeats:  310641     14322152 bp    1.22 %\n",
      "Low complexity:   47381      2844279 bp    0.24 %\n",
      "==================================================\n",
      "\n",
      "* most repeats fragmented by insertions or deletions\n",
      "  have been counted as one element\n",
      "  Runs of >=20 X/Ns in query were excluded in % calcs\n",
      "\n",
      "\n",
      "The query species was assumed to be homo sapiens  \n",
      "RepeatMasker Combined Database: Dfam_Consensus-20170127, RepBase-20170127\n",
      "        \n",
      "run with rmblastn version 2.6.0+\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "cat /home/sam/data/oly_assemblies/jelly.out.fasta.preThuMay240214172018.RMoutput/jelly.out.fasta.tbl"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### DEFAULT SETTINGS SUMMARY TABLE (_Crassostrea gigas_)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "==================================================\n",
      "file name: jelly.out.fasta          \n",
      "sequences:        696946\n",
      "total length: 1253001795 bp  (1172226648 bp excl N/X-runs)\n",
      "GC level:         36.51 %\n",
      "bases masked:  160759267 bp ( 13.71 %)\n",
      "==================================================\n",
      "               number of      length   percentage\n",
      "               elements*    occupied  of sequence\n",
      "--------------------------------------------------\n",
      "Retroelements       213132     69887654 bp    5.96 %\n",
      "   SINEs:             2374       311974 bp    0.03 %\n",
      "   Penelope         171792     57862186 bp    4.94 %\n",
      "   LINEs:           195605     63430615 bp    5.41 %\n",
      "    CRE/SLACS            0            0 bp    0.00 %\n",
      "     L2/CR1/Rex        731       357995 bp    0.03 %\n",
      "     R1/LOA/Jockey       0            0 bp    0.00 %\n",
      "     R2/R4/NeSL         13        11377 bp    0.00 %\n",
      "     RTE/Bov-B        8085      1948581 bp    0.17 %\n",
      "     L1/CIN4             0            0 bp    0.00 %\n",
      "   LTR elements:     15153      6145065 bp    0.52 %\n",
      "     BEL/Pao          2119       955773 bp    0.08 %\n",
      "     Ty1/Copia         101        75372 bp    0.01 %\n",
      "     Gypsy/DIRS1     11776      4815361 bp    0.41 %\n",
      "       Retroviral        0            0 bp    0.00 %\n",
      "\n",
      "DNA transposons     256292     35689117 bp    3.04 %\n",
      "   hobo-Activator    19847      2059651 bp    0.18 %\n",
      "   Tc1-IS630-Pogo    43269      6806311 bp    0.58 %\n",
      "   En-Spm                0            0 bp    0.00 %\n",
      "   MuDR-IS905            0            0 bp    0.00 %\n",
      "   PiggyBac           7935      1060296 bp    0.09 %\n",
      "   Tourist/Harbinger  9503       887332 bp    0.08 %\n",
      "   Other (Mirage,        0            0 bp    0.00 %\n",
      "    P-element, Transib)\n",
      "\n",
      "Rolling-circles          0            0 bp    0.00 %\n",
      "\n",
      "Unclassified:       174943     38299211 bp    3.27 %\n",
      "\n",
      "Total interspersed repeats:   143875982 bp   12.27 %\n",
      "\n",
      "\n",
      "Small RNA:             280        78768 bp    0.01 %\n",
      "\n",
      "Satellites:           7383      1362194 bp    0.12 %\n",
      "Simple repeats:     278809     12982714 bp    1.11 %\n",
      "Low complexity:      44078      2622506 bp    0.22 %\n",
      "==================================================\n",
      "\n",
      "* most repeats fragmented by insertions or deletions\n",
      "  have been counted as one element\n",
      "  Runs of >=20 X/Ns in query were excluded in % calcs\n",
      "\n",
      "\n",
      "The query species was assumed to be crassostrea gigas\n",
      "RepeatMasker Combined Database: Dfam_Consensus-20170127, RepBase-20170127\n",
      "        \n",
      "run with rmblastn version 2.6.0+\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "cat /home/sam/data/oly_assemblies/jelly.out.fasta.preThuMay240845562018.RMoutput/jelly.out.fasta.tbl"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### DEFAULT SETTINGS SUMMARY TABLE (_Crassostrea virginica_)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "==================================================\n",
      "file name: jelly.out.fasta          \n",
      "sequences:        696946\n",
      "total length: 1253001795 bp  (1172226648 bp excl N/X-runs)\n",
      "GC level:         36.51 %\n",
      "bases masked:   39598953 bp ( 3.38 %)\n",
      "==================================================\n",
      "               number of      length   percentage\n",
      "               elements*    occupied  of sequence\n",
      "--------------------------------------------------\n",
      "Retroelements        63882     10327611 bp    0.88 %\n",
      "   SINEs:            63882     10327611 bp    0.88 %\n",
      "   Penelope              0            0 bp    0.00 %\n",
      "   LINEs:                0            0 bp    0.00 %\n",
      "    CRE/SLACS            0            0 bp    0.00 %\n",
      "     L2/CR1/Rex          0            0 bp    0.00 %\n",
      "     R1/LOA/Jockey       0            0 bp    0.00 %\n",
      "     R2/R4/NeSL          0            0 bp    0.00 %\n",
      "     RTE/Bov-B           0            0 bp    0.00 %\n",
      "     L1/CIN4             0            0 bp    0.00 %\n",
      "   LTR elements:         0            0 bp    0.00 %\n",
      "     BEL/Pao             0            0 bp    0.00 %\n",
      "     Ty1/Copia           0            0 bp    0.00 %\n",
      "     Gypsy/DIRS1         0            0 bp    0.00 %\n",
      "       Retroviral        0            0 bp    0.00 %\n",
      "\n",
      "DNA transposons       9433      2307292 bp    0.20 %\n",
      "   hobo-Activator        0            0 bp    0.00 %\n",
      "   Tc1-IS630-Pogo        0            0 bp    0.00 %\n",
      "   En-Spm                0            0 bp    0.00 %\n",
      "   MuDR-IS905            0            0 bp    0.00 %\n",
      "   PiggyBac              0            0 bp    0.00 %\n",
      "   Tourist/Harbinger     0            0 bp    0.00 %\n",
      "   Other (Mirage,        0            0 bp    0.00 %\n",
      "    P-element, Transib)\n",
      "\n",
      "Rolling-circles          0            0 bp    0.00 %\n",
      "\n",
      "Unclassified:        51558      9836468 bp    0.84 %\n",
      "\n",
      "Total interspersed repeats:    22471371 bp    1.92 %\n",
      "\n",
      "\n",
      "Small RNA:           64164     10406776 bp    0.89 %\n",
      "\n",
      "Satellites:             10         5985 bp    0.00 %\n",
      "Simple repeats:     298612     14185090 bp    1.21 %\n",
      "Low complexity:      47510      2866522 bp    0.24 %\n",
      "==================================================\n",
      "\n",
      "* most repeats fragmented by insertions or deletions\n",
      "  have been counted as one element\n",
      "  Runs of >=20 X/Ns in query were excluded in % calcs\n",
      "\n",
      "\n",
      "The query species was assumed to be crassostrea virginica\n",
      "RepeatMasker Combined Database: Dfam_Consensus-20170127, RepBase-20170127\n",
      "        \n",
      "run with rmblastn version 2.6.0+\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "cat /home/sam/data/oly_assemblies/jelly.out.fasta.preThuMay241347382018.RMoutput/jelly.out.fasta.tbl"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### DEFAULT SETTINGS SUMMARY TABLE (_Ostrea lurida_)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "==================================================\n",
      "file name: jelly.out.fasta          \n",
      "sequences:        696946\n",
      "total length: 1253001795 bp  (1172226648 bp excl N/X-runs)\n",
      "GC level:         36.51 %\n",
      "bases masked:   17617763 bp ( 1.50 %)\n",
      "==================================================\n",
      "               number of      length   percentage\n",
      "               elements*    occupied  of sequence\n",
      "--------------------------------------------------\n",
      "Retroelements            0            0 bp    0.00 %\n",
      "   SINEs:                0            0 bp    0.00 %\n",
      "   Penelope              0            0 bp    0.00 %\n",
      "   LINEs:                0            0 bp    0.00 %\n",
      "    CRE/SLACS            0            0 bp    0.00 %\n",
      "     L2/CR1/Rex          0            0 bp    0.00 %\n",
      "     R1/LOA/Jockey       0            0 bp    0.00 %\n",
      "     R2/R4/NeSL          0            0 bp    0.00 %\n",
      "     RTE/Bov-B           0            0 bp    0.00 %\n",
      "     L1/CIN4             0            0 bp    0.00 %\n",
      "   LTR elements:         0            0 bp    0.00 %\n",
      "     BEL/Pao             0            0 bp    0.00 %\n",
      "     Ty1/Copia           0            0 bp    0.00 %\n",
      "     Gypsy/DIRS1         0            0 bp    0.00 %\n",
      "       Retroviral        0            0 bp    0.00 %\n",
      "\n",
      "DNA transposons          0            0 bp    0.00 %\n",
      "   hobo-Activator        0            0 bp    0.00 %\n",
      "   Tc1-IS630-Pogo        0            0 bp    0.00 %\n",
      "   En-Spm                0            0 bp    0.00 %\n",
      "   MuDR-IS905            0            0 bp    0.00 %\n",
      "   PiggyBac              0            0 bp    0.00 %\n",
      "   Tourist/Harbinger     0            0 bp    0.00 %\n",
      "   Other (Mirage,        0            0 bp    0.00 %\n",
      "    P-element, Transib)\n",
      "\n",
      "Rolling-circles          0            0 bp    0.00 %\n",
      "\n",
      "Unclassified:            3          189 bp    0.00 %\n",
      "\n",
      "Total interspersed repeats:         189 bp    0.00 %\n",
      "\n",
      "\n",
      "Small RNA:             282        79165 bp    0.01 %\n",
      "\n",
      "Satellites:             10         5985 bp    0.00 %\n",
      "Simple repeats:     313082     14662647 bp    1.25 %\n",
      "Low complexity:      47785      2878201 bp    0.25 %\n",
      "==================================================\n",
      "\n",
      "* most repeats fragmented by insertions or deletions\n",
      "  have been counted as one element\n",
      "  Runs of >=20 X/Ns in query were excluded in % calcs\n",
      "\n",
      "\n",
      "The query species was assumed to be ostrea lurida \n",
      "RepeatMasker Combined Database: Dfam_Consensus-20170127, RepBase-20170127\n",
      "        \n",
      "run with rmblastn version 2.6.0+\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "cat /home/sam/data/oly_assemblies/jelly.out.fasta.preThuMay241644392018.RMoutput/jelly.out.fasta.tbl"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cp: cannot create regular file '/mnt/owl/Athaliana/jelly.out.fasta.preThuMay240214172018.RMoutput/jelly.out.fasta.masked': Permission denied\n",
      "cp: cannot create regular file '/mnt/owl/Athaliana/jelly.out.fasta.preThuMay240214172018.RMoutput/jelly.out.fasta.tbl': Permission denied\n",
      "cp: cannot create regular file '/mnt/owl/Athaliana/jelly.out.fasta.preThuMay240214172018.RMoutput/jelly.out.fasta.out': Permission denied\n",
      "cp: cannot create regular file '/mnt/owl/Athaliana/jelly.out.fasta.preThuMay240214172018.RMoutput/readme.txt': Permission denied\n",
      "cp: cannot create regular file '/mnt/owl/Athaliana/jelly.out.fasta.preThuMay240214172018.RMoutput/jelly.out.fasta.cat.gz': Permission denied\n",
      "cp: cannot create regular file '/mnt/owl/Athaliana/jelly.out.fasta.preThuMay240214172018.RMoutput/jelly.out.fasta.out.gff': Permission denied\n",
      "cp: preserving times for '/mnt/owl/Athaliana/jelly.out.fasta.preThuMay240214172018.RMoutput': Operation not permitted\n",
      "cp: cannot create regular file '/mnt/owl/Athaliana/jelly.out.fasta.preThuMay240845562018.RMoutput/jelly.out.fasta.masked': Permission denied\n",
      "cp: cannot create regular file '/mnt/owl/Athaliana/jelly.out.fasta.preThuMay240845562018.RMoutput/jelly.out.fasta.out.gff': Permission denied\n",
      "cp: cannot create regular file '/mnt/owl/Athaliana/jelly.out.fasta.preThuMay240845562018.RMoutput/jelly.out.fasta.cat.gz': Permission denied\n",
      "cp: cannot create regular file '/mnt/owl/Athaliana/jelly.out.fasta.preThuMay240845562018.RMoutput/jelly.out.fasta.tbl': Permission denied\n",
      "cp: cannot create regular file '/mnt/owl/Athaliana/jelly.out.fasta.preThuMay240845562018.RMoutput/jelly.out.fasta.out': Permission denied\n",
      "cp: cannot create regular file '/mnt/owl/Athaliana/jelly.out.fasta.preThuMay240845562018.RMoutput/readme.txt': Permission denied\n",
      "cp: preserving times for '/mnt/owl/Athaliana/jelly.out.fasta.preThuMay240845562018.RMoutput': Operation not permitted\n",
      "cp: cannot create regular file '/mnt/owl/Athaliana/jelly.out.fasta.preThuMay241347382018.RMoutput/jelly.out.fasta.masked': Permission denied\n",
      "cp: cannot create regular file '/mnt/owl/Athaliana/jelly.out.fasta.preThuMay241347382018.RMoutput/jelly.out.fasta.out.gff': Permission denied\n",
      "cp: cannot create regular file '/mnt/owl/Athaliana/jelly.out.fasta.preThuMay241347382018.RMoutput/jelly.out.fasta.cat.gz': Permission denied\n",
      "cp: cannot create regular file '/mnt/owl/Athaliana/jelly.out.fasta.preThuMay241347382018.RMoutput/jelly.out.fasta.tbl': Permission denied\n",
      "cp: cannot create regular file '/mnt/owl/Athaliana/jelly.out.fasta.preThuMay241347382018.RMoutput/jelly.out.fasta.out': Permission denied\n",
      "cp: cannot create regular file '/mnt/owl/Athaliana/jelly.out.fasta.preThuMay241347382018.RMoutput/readme.txt': Permission denied\n",
      "cp: preserving times for '/mnt/owl/Athaliana/jelly.out.fasta.preThuMay241347382018.RMoutput': Operation not permitted\n",
      "cp: cannot create regular file '/mnt/owl/Athaliana/jelly.out.fasta.preThuMay241644392018.RMoutput/jelly.out.fasta.masked': Permission denied\n",
      "cp: cannot create regular file '/mnt/owl/Athaliana/jelly.out.fasta.preThuMay241644392018.RMoutput/jelly.out.fasta.out.gff': Permission denied\n",
      "cp: cannot create regular file '/mnt/owl/Athaliana/jelly.out.fasta.preThuMay241644392018.RMoutput/jelly.out.fasta.cat.gz': Permission denied\n",
      "cp: cannot create regular file '/mnt/owl/Athaliana/jelly.out.fasta.preThuMay241644392018.RMoutput/jelly.out.fasta.tbl': Permission denied\n",
      "cp: cannot create regular file '/mnt/owl/Athaliana/jelly.out.fasta.preThuMay241644392018.RMoutput/jelly.out.fasta.out': Permission denied\n",
      "cp: cannot create regular file '/mnt/owl/Athaliana/jelly.out.fasta.preThuMay241644392018.RMoutput/readme.txt': Permission denied\n",
      "cp: preserving times for '/mnt/owl/Athaliana/jelly.out.fasta.preThuMay241644392018.RMoutput': Operation not permitted\n",
      "ls: cannot open directory '/mnt/owl/Athaliana/jelly.out.fasta.preThuMay240214172018.RMoutput': Permission denied\n",
      "ls: cannot open directory '/mnt/owl/Athaliana/jelly.out.fasta.preThuMay240845562018.RMoutput': Permission denied\n",
      "ls: cannot open directory '/mnt/owl/Athaliana/jelly.out.fasta.preThuMay241347382018.RMoutput': Permission denied\n",
      "ls: cannot open directory '/mnt/owl/Athaliana/jelly.out.fasta.preThuMay241644392018.RMoutput': Permission denied\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "cp -pR /home/sam/data/oly_assemblies/jelly.out.fasta.* /mnt/owl/Athaliana/\n",
    "\n",
    "ls /mnt/owl/Athaliana/jelly.out.fasta.*"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Ugh. Ran copy command with ```sudo``` outside of notebok."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/mnt/owl/Athaliana/jelly.out.fasta.preThuMay240214172018.RMoutput:\n",
      "jelly.out.fasta.cat.gz\n",
      "jelly.out.fasta.masked\n",
      "jelly.out.fasta.out\n",
      "jelly.out.fasta.out.gff\n",
      "jelly.out.fasta.tbl\n",
      "readme.txt\n",
      "\n",
      "/mnt/owl/Athaliana/jelly.out.fasta.preThuMay240845562018.RMoutput:\n",
      "jelly.out.fasta.cat.gz\n",
      "jelly.out.fasta.masked\n",
      "jelly.out.fasta.out\n",
      "jelly.out.fasta.out.gff\n",
      "jelly.out.fasta.tbl\n",
      "readme.txt\n",
      "\n",
      "/mnt/owl/Athaliana/jelly.out.fasta.preThuMay241347382018.RMoutput:\n",
      "jelly.out.fasta.cat.gz\n",
      "jelly.out.fasta.masked\n",
      "jelly.out.fasta.out\n",
      "jelly.out.fasta.out.gff\n",
      "jelly.out.fasta.tbl\n",
      "readme.txt\n",
      "\n",
      "/mnt/owl/Athaliana/jelly.out.fasta.preThuMay241644392018.RMoutput:\n",
      "jelly.out.fasta.cat.gz\n",
      "jelly.out.fasta.masked\n",
      "jelly.out.fasta.out\n",
      "jelly.out.fasta.out.gff\n",
      "jelly.out.fasta.tbl\n",
      "readme.txt\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "ls /mnt/owl/Athaliana/jelly.out.fasta.*"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": true
   },
   "source": [
    "Copied folders to wrong locations. Re-doing. Will need to use sudo outside of notebook."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "%%bash\n",
    "cp -pR /home/sam/data/oly_assemblies/jelly.out.fasta.preThuMay240214172018.RMoutput \\\n",
    "/home/sam/analyses/20180523_oly_repeatmasker_pbjelly_sjw_01-01/\n",
    "\n",
    "cp -pR /home/sam/data/oly_assemblies/jelly.out.fasta.preThuMay240845562018.RMoutput \\\n",
    "/home/sam/analyses/20180523_oly_repeatmasker_pbjelly_sjw_01-02/\n",
    "\n",
    "cp -pR /home/sam/data/oly_assemblies/jelly.out.fasta.preThuMay241347382018.RMoutput \\\n",
    "/home/sam/analyses/20180523_oly_repeatmasker_pbjelly_sjw_01-03/\n",
    "\n",
    "cp -pR /home/sam/data/oly_assemblies/jelly.out.fasta.preThuMay241644392018.RMoutput \\\n",
    "/home/sam/analyses/20180523_oly_repeatmasker_pbjelly_sjw_01-04/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/mnt/owl/Athaliana/20180523_oly_repeatmasker_pbjelly_sjw_01-01:\n",
      "jelly.out.fasta.preThuMay240214172018.RMoutput\n",
      "readme.txt\n",
      "stderr.err\n",
      "stdout.out\n",
      "\n",
      "/mnt/owl/Athaliana/20180523_oly_repeatmasker_pbjelly_sjw_01-02:\n",
      "jelly.out.fasta.preThuMay240845562018.RMoutput\n",
      "stderr.err\n",
      "stdout.out\n",
      "\n",
      "/mnt/owl/Athaliana/20180523_oly_repeatmasker_pbjelly_sjw_01-03:\n",
      "jelly.out.fasta.preThuMay241347382018.RMoutput\n",
      "stderr.err\n",
      "stdout.out\n",
      "\n",
      "/mnt/owl/Athaliana/20180523_oly_repeatmasker_pbjelly_sjw_01-04:\n",
      "jelly.out.fasta.preThuMay241644392018.RMoutput\n",
      "stderr.err\n",
      "stdout.out\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "ls /mnt/owl/Athaliana/20180523_oly_repeatmasker_pbjelly_sjw_01-0*"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
