{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Create _P.generosa_ primary gene annotations mapping file\n",
    "\n",
    "See this [GitHub Issue(https://github.com/RobertsLab/resources/issues/1443).\n",
    "\n",
    "This notebook will utilize NCBI BLAST and DIAMOND BLAST annotations generated by our [GenSas _P.generosa_ genome annotation](https://robertslab.github.io/sams-notebook/2019/09/28/Genome-Annotation-Pgenerosa_v074-a4-Using-GenSAS.html).\n",
    "\n",
    "It will compare the two sets of SwissProt ID annotations (SPIDs) to determine lowest E-value and use that entry as the representative entry for a gene. It will then use that canonical list of SPIDs to pull gene names and gene ontology (GO) IDs from UniProt, and create a tab-deltimited annotation mapping file."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### List computer specs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TODAY'S DATE\n",
      "Tue 19 Apr 2022 01:18:10 PM PDT\n",
      "------------\n",
      "\n",
      "Distributor ID:\tUbuntu\n",
      "Description:\tUbuntu 20.04.4 LTS\n",
      "Release:\t20.04\n",
      "Codename:\tfocal\n",
      "\n",
      "------------\n",
      "HOSTNAME: \n",
      "computer\n",
      "\n",
      "------------\n",
      "Computer Specs:\n",
      "\n",
      "Architecture:                    x86_64\n",
      "CPU op-mode(s):                  32-bit, 64-bit\n",
      "Byte Order:                      Little Endian\n",
      "Address sizes:                   45 bits physical, 48 bits virtual\n",
      "CPU(s):                          2\n",
      "On-line CPU(s) list:             0,1\n",
      "Thread(s) per core:              1\n",
      "Core(s) per socket:              1\n",
      "Socket(s):                       2\n",
      "NUMA node(s):                    1\n",
      "Vendor ID:                       GenuineIntel\n",
      "CPU family:                      6\n",
      "Model:                           165\n",
      "Model name:                      Intel(R) Core(TM) i9-10885H CPU @ 2.40GHz\n",
      "Stepping:                        2\n",
      "CPU MHz:                         2400.009\n",
      "BogoMIPS:                        4800.01\n",
      "Hypervisor vendor:               VMware\n",
      "Virtualization type:             full\n",
      "L1d cache:                       64 KiB\n",
      "L1i cache:                       64 KiB\n",
      "L2 cache:                        512 KiB\n",
      "L3 cache:                        32 MiB\n",
      "NUMA node0 CPU(s):               0,1\n",
      "Vulnerability Itlb multihit:     KVM: Mitigation: VMX unsupported\n",
      "Vulnerability L1tf:              Mitigation; PTE Inversion\n",
      "Vulnerability Mds:               Vulnerable: Clear CPU buffers attempted, no microcode; SMT Host state unknown\n",
      "Vulnerability Meltdown:          Mitigation; PTI\n",
      "Vulnerability Spec store bypass: Mitigation; Speculative Store Bypass disabled via prctl and seccomp\n",
      "Vulnerability Spectre v1:        Mitigation; usercopy/swapgs barriers and __user pointer sanitization\n",
      "Vulnerability Spectre v2:        Mitigation; Retpolines, IBPB conditional, IBRS_FW, STIBP disabled, RSB filling\n",
      "Vulnerability Srbds:             Not affected\n",
      "Vulnerability Tsx async abort:   Not affected\n",
      "Flags:                           fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ss syscall nx pdpe1gb rdtscp lm constant_tsc arch_perfmon nopl xtopology tsc_reliable nonstop_tsc cpuid tsc_known_freq pni pclmulqdq ssse3 fma cx16 pcid sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand hypervisor lahf_lm abm 3dnowprefetch invpcid_single pti ssbd ibrs ibpb stibp fsgsbase tsc_adjust bmi1 avx2 smep bmi2 invpcid rdseed adx smap clflushopt xsaveopt xsavec xgetbv1 xsaves arat flush_l1d arch_capabilities\n",
      "\n",
      "------------\n",
      "\n",
      "Memory Specs\n",
      "\n",
      "              total        used        free      shared  buff/cache   available\n",
      "Mem:           54Gi       3.1Gi        47Gi       110Mi       3.7Gi        50Gi\n",
      "Swap:         2.0Gi          0B       2.0Gi\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "No LSB modules are available.\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "echo \"TODAY'S DATE\"\n",
    "date\n",
    "echo \"------------\"\n",
    "echo \"\"\n",
    "lsb_release -a\n",
    "echo \"\"\n",
    "echo \"------------\"\n",
    "echo \"HOSTNAME: \"\n",
    "hostname\n",
    "echo \"\"\n",
    "echo \"------------\"\n",
    "echo \"Computer Specs:\"\n",
    "echo \"\"\n",
    "lscpu\n",
    "echo \"\"\n",
    "echo \"------------\"\n",
    "echo \"\"\n",
    "echo \"Memory Specs\"\n",
    "echo \"\"\n",
    "free -mh"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Set variables\n",
    "- `%env` indicates a bash variable\n",
    "- without `%env` is Python variablec"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "env: data_dir=/home/sam/data/P_generosa/genomes\n",
      "env: analysis_dir=/home/sam/analyses/20220419-pgen-gene_annotation_mapping\n",
      "env: base_url=https://gannet.fish.washington.edu/Atumefaciens/20190928_Pgenerosa_v074.a4_gensas_annotation\n",
      "env: blast_annotations=Panopea-generosa-vv0.74.a4.5d951a9b74287-blast_functional.tab\n",
      "env: diamond_annotations=Panopea-generosa-vv0.74.a4.5d951bcf45b4b-diamond_functional.tab\n",
      "env: uniprot_output=20220419-pgen-uniprot_batch-results.txt\n",
      "env: spid_list=Panopea-generosa-v1.0.a4-blast-diamond-functional-SPIDs.txt\n",
      "env: genome_IDs_SPIDs=Panopea-generosa-v1.0.a4-blast-diamond-functional-genome_IDs-SPIDs.txt\n",
      "env: blast_diamond_cat=Panopea-generosa-v1.0.a4-blast-diamond-functional.tab\n",
      "env: blast_diamond_cat_best=Panopea-generosa-v1.0.a4-blast-diamond-functional_best.tab\n",
      "env: parsed_uniprot=20220419-pgen-accession-gene_name-gene_description-go_ids.tab\n",
      "env: joined_output=20220419-pgen-gene-accessions-gene_id-gene_name-gene_description-alt_gene_description-go_ids.tab\n",
      "env: uniprot_mapping_script=/home/sam/programs/uniprot_mapping.pl\n"
     ]
    }
   ],
   "source": [
    "######################################################################\n",
    "### Set directories\n",
    "%env data_dir=/home/sam/data/P_generosa/genomes\n",
    "%env analysis_dir=/home/sam/analyses/20220419-pgen-gene_annotation_mapping\n",
    "analysis_dir=\"/home/sam/analyses/20220419-pgen-gene_annotation_mapping\"\n",
    "\n",
    "#####################################################################\n",
    "### Input files\n",
    "%env base_url=https://gannet.fish.washington.edu/Atumefaciens/20190928_Pgenerosa_v074.a4_gensas_annotation\n",
    "%env blast_annotations=Panopea-generosa-vv0.74.a4.5d951a9b74287-blast_functional.tab\n",
    "%env diamond_annotations=Panopea-generosa-vv0.74.a4.5d951bcf45b4b-diamond_functional.tab\n",
    "\n",
    "######################################################################\n",
    "### Output files\n",
    "# UniProt batch output\n",
    "%env uniprot_output=20220419-pgen-uniprot_batch-results.txt\n",
    "\n",
    "# Gene name list for UniProt batch submission\n",
    "%env spid_list=Panopea-generosa-v1.0.a4-blast-diamond-functional-SPIDs.txt\n",
    "\n",
    "# Genome IDs and SPIDs\n",
    "%env genome_IDs_SPIDs=Panopea-generosa-v1.0.a4-blast-diamond-functional-genome_IDs-SPIDs.txt\n",
    "\n",
    "%env blast_diamond_cat=Panopea-generosa-v1.0.a4-blast-diamond-functional.tab\n",
    "\n",
    "%env blast_diamond_cat_best=Panopea-generosa-v1.0.a4-blast-diamond-functional_best.tab\n",
    "\n",
    "# Parsed UniProt\n",
    "%env parsed_uniprot=20220419-pgen-accession-gene_name-gene_description-go_ids.tab\n",
    "\n",
    "# Final output\n",
    "%env joined_output=20220419-pgen-gene-accessions-gene_id-gene_name-gene_description-alt_gene_description-go_ids.tab\n",
    "\n",
    "######################################################################\n",
    "\n",
    "### Programs\n",
    "\n",
    "# UniProt batch submission/retrieval script\n",
    "%env uniprot_mapping_script=/home/sam/programs/uniprot_mapping.pl"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Make input/output directories"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "# If directories don't exist, make them\n",
    "mkdir --parents \"${data_dir}\" \"${analysis_dir}\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Download and inspect annotation files\n",
    "\n",
    "`--quiet`: Prevents `wget` output from overwhelming Jupyter Notebook\n",
    "\n",
    "`--continue`: If download was previously initiated, will continue where leftoff and will not create a second file if one already exists."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "total 2.6G\n",
      "-rw-rw-r-- 1 sam sam 1.5M Oct  3  2019 Panopea-generosa-vv0.74.a4.5d951a9b74287-blast_functional.tab\n",
      "-rw-rw-r-- 1 sam sam 1.3M Oct  3  2019 Panopea-generosa-vv0.74.a4.5d951bcf45b4b-diamond_functional.tab\n",
      "-rwxr-xr-x 1 sam sam 914M Nov  5  2019 Panopea-generosa-v1.0.fasta\n",
      "-rw-rw-r-- 1 sam sam 454M Mar 19 07:58 Panopea-generosa-v1.0.a4.gff3\n",
      "-rw-r--r-- 1 sam sam 503M Mar 22 06:48 Panopea-generosa-v1.0.a4_biotype.gff\n",
      "-rw-r--r-- 1 sam sam 4.8M Mar 24 07:30 Panopea-generosa-v1.0.a4_biotype-trna_strand_converted-no_RNAmmer.bed\n",
      "-rw-rw-r-- 1 sam sam  658 Mar 25 06:11 Panopea-generosa-v1.0.fa.fai\n",
      "-rw-rw-r-- 1 sam sam 9.7M Mar 30 11:03 Panopea-generosa-v1.0.a4_biotype.gtf\n",
      "-rw-rw-r-- 1 sam sam 507M Mar 30 11:43 Panopea-generosa-v1.0.a4_biotype.bed\n",
      "-rw-rw-r-- 1 sam sam  378 Mar 30 13:20 Panopea-generosa-v1.0.fa.lengths\n",
      "-rw-rw-r-- 1 sam sam 9.7M Mar 30 13:34 Panopea-generosa-v1.0.a4_biotype.sorted.gtf\n",
      "-rw-rw-r-- 1 sam sam 996K Mar 30 13:34 Panopea-generosa-v1.0.a4_biotype_non-coding.bed\n",
      "drwxrwxr-x 2 sam sam 4.0K Mar 30 13:46 feelnc_codpot_out\n",
      "-rw-rw-r-- 1 sam sam  70M Mar 31 06:00 Panopea-generosa-v1.0.a4.gtf\n",
      "-rw-rw-r-- 1 sam sam 101K Apr  6 12:22 spids.txt\n",
      "-rw-rw-r-- 1 sam sam 138M Apr  6 12:23 uniprot_mapping-all.txt\n",
      "-rw-rw-r-- 1 sam sam 282K Apr  7 07:13 uniprot_mapping-all-AC_only.txt\n",
      "-rw-rw-r-- 1 sam sam   78 Apr  7 13:57 File1.txt\n",
      "-rw-rw-r-- 1 sam sam  357 Apr  7 13:57 File2.txt\n",
      "\n",
      "---------------------------------------------------------\n",
      "\n",
      "==> Panopea-generosa-vv0.74.a4.5d951a9b74287-blast_functional.tab <==\n",
      "#\n",
      "# Output is generated by GenSAS 7.x-5.0\n",
      "#\n",
      "#name     : mRNA\n",
      "#start    : Start of alignment in subject\n",
      "#end      : End of alignment in subject\n",
      "#m_start  : Start of alignment in query\n",
      "#m_end    : End of alignment in query\n",
      "#al       : Alignment length\n",
      "#score    : Row score of the match\n",
      "#evalue   : E value of the match\n",
      "#identity : Percentage of identical matches\n",
      "mame\tstart\tend\tscore\tAccession\tMatch ID\tm_start\tm_end\tE-value\tidentity\tal\n",
      "21910-PGEN_.00g000010.m01\t121\t229\t165\tQ86IC9\tsp|Q86IC9|CAMT1_DICDI\t11\t122\t8.93e-14\t35.652\t115\n",
      "21910-PGEN_.00g000020.m01\t147\t467\t968\tP04177\tsp|P04177|TY3H_RAT\t20\t339\t3.47e-127\t55.140\t321\n",
      "21910-PGEN_.00g000050.m01\t566\t722\t182\tQ8L840\tsp|Q8L840|RQL4A_ARATH\t2\t167\t2.67e-14\t35.119\t168\n",
      "21910-PGEN_.00g000080.m01\t268\t322\t152\tA1E2V0\tsp|A1E2V0|BIRC3_CANLF\t163\t220\t3.91e-10\t53.448\t58\n",
      "21910-PGEN_.00g000090.m01\t199\t327\t161\tP34456\tsp|P34456|YMD2_CAEEL\t7\t134\t7.52e-12\t26.357\t129\n",
      "21910-PGEN_.00g000210.m01\t18\t200\t263\tO00463\tsp|O00463|TRAF5_HUMAN\t5\t191\t2.24e-25\t34.921\t189\n",
      "21910-PGEN_.00g000230.m01\t48\t155\t287\tQ00945\tsp|Q00945|CONO_LYMST\t31\t134\t1.59e-32\t50.000\t108\n",
      "21910-PGEN_.00g000240.m01\t4\t605\t1091\tQ5SWK7\tsp|Q5SWK7|RN145_MOUSE\t13\t601\t2.65e-139\t39.607\t611\n",
      "21910-PGEN_.00g000280.m01\t4\t153\t210\tQ8ZXT3\tsp|Q8ZXT3|Y1111_PYRAE\t853\t1012\t1.10e-17\t38.750\t160\n",
      "21910-PGEN_.00g000300.m01\t159\t347\t480\tQ5REG4\tsp|Q5REG4|DTX3_PONAB\t1135\t1320\t1.20e-51\t50.794\t189\n",
      "21910-PGEN_.00g000300.m02\t159\t347\t480\tQ5REG4\tsp|Q5REG4|DTX3_PONAB\t1138\t1323\t1.18e-51\t50.794\t189\n",
      "21910-PGEN_.00g000380.m01\t381\t508\t205\tQ8QG60\tsp|Q8QG60|CRY2_CHICK\t2\t145\t4.92e-18\t36.111\t144\n",
      "\n",
      "==> Panopea-generosa-vv0.74.a4.5d951bcf45b4b-diamond_functional.tab <==\n",
      "#\n",
      "# Output is generated by GenSAS 7.x-5.0\n",
      "#\n",
      "#name     : mRNA\n",
      "#start    : Start of alignment in subject\n",
      "#end      : End of alignment in subject\n",
      "#m_start  : Start of alignment in query\n",
      "#m_end    : End of alignment in query\n",
      "#al       : Alignment length\n",
      "#score    : Row score of the match\n",
      "#evalue   : E value of the match\n",
      "#identity : Percentage of identical matches\n",
      "mame\tstart\tend\tscore\tAccession\tMatch ID\tm_start\tm_end\tE-value\tidentity\tal\n",
      "21910-PGEN_.00g000020.m01\t147\t467\t945\tP04177\tsp|P04177|TY3H_RAT\t20\t339\t7.9e-101\t55.1\t321\n",
      "21910-PGEN_.00g000050.m01\t566\t722\t180\tQ8L840\tsp|Q8L840|RQL4A_ARATH\t2\t167\t2.4e-12\t35.1\t168\n",
      "21910-PGEN_.00g000060.m01\t1957\t2106\t129\tQ61043\tsp|Q61043|NIN_MOUSE\t31\t184\t1.7e-06\t26.0\t154\n",
      "21910-PGEN_.00g000080.m01\t233\t304\t134\tQ24307\tsp|Q24307|DIAP2_DROME\t174\t255\t6.2e-07\t34.1\t82\n",
      "21910-PGEN_.00g000120.m01\t6\t49\t118\tP34457\tsp|P34457|YMD3_CAEEL\t90\t133\t3.2e-05\t47.7\t44\n",
      "21910-PGEN_.00g000230.m01\t49\t155\t216\tQ00945\tsp|Q00945|CONO_LYMST\t32\t134\t9.9e-17\t49.5\t107\n",
      "21910-PGEN_.00g000240.m01\t4\t585\t1144\tQ5SWK7\tsp|Q5SWK7|RN145_MOUSE\t13\t592\t1.2e-123\t40.1\t591\n",
      "21910-PGEN_.00g000280.m01\t433\t592\t230\tQ9WYX8\tsp|Q9WYX8|Y508_THEMA\t863\t1022\t2.2e-17\t40.2\t164\n",
      "21910-PGEN_.00g000300.m01\t161\t347\t474\tQ80V91\tsp|Q80V91|DTX3_MOUSE\t1137\t1320\t1.2e-45\t51.3\t187\n",
      "21910-PGEN_.00g000300.m02\t161\t347\t474\tQ80V91\tsp|Q80V91|DTX3_MOUSE\t1140\t1323\t1.2e-45\t51.3\t187\n",
      "21910-PGEN_.00g000380.m01\t381\t508\t201\tQ8QG60\tsp|Q8QG60|CRY2_CHICK\t2\t145\t7.0e-15\t35.4\t144\n",
      "21910-PGEN_.00g000440.m01\t234\t1796\t1606\tQ9H583\tsp|Q9H583|HEAT1_HUMAN\t1\t1575\t8.0e-177\t30.0\t1624\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "cd \"${data_dir}\"\n",
    "\n",
    "wget --quiet --continue ${base_url}/${blast_annotations}\n",
    "wget --quiet --continue ${base_url}/${diamond_annotations}\n",
    "\n",
    "ls -ltrh\n",
    "\n",
    "echo \"\"\n",
    "echo \"---------------------------------------------------------\"\n",
    "echo \"\"\n",
    "head -n 25 *.tab"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Count number of header lines (i.e. beginning with a `#`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Panopea-generosa-vv0.74.a4.5d951a9b74287-blast_functional.tab:12\n",
      "Panopea-generosa-vv0.74.a4.5d951bcf45b4b-diamond_functional.tab:12\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "cd \"${data_dir}\"\n",
    "\n",
    "grep -c \"^#\" \"${blast_annotations}\" \"${diamond_annotations}\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Concatenate annotation files\n",
    "\n",
    "Also modifies mRNA names to generate gene names instead.\n",
    "\n",
    "- `awk 'NR > 13'`: Skips first 13 header lines\n",
    "- `sort -k1,1 -k9,9`: Sorts on first field (mRNA name), then on 9th field (e-value)\n",
    "- `sed 's/^21910-//'`: Removes leading info from each mRNA name, at the beginning of each line (`^`)\n",
    "- `sed 's/.m0[0-9]//'`: Removes `.m0N` from each mRNA name.\n",
    "- `awk '!array[$1]++'`: `awk` array that only prints line if it's the first occurrence of gene name (first field; `$1` (i.e. no duplicates)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Line count:\n",
      "31216 Panopea-generosa-v1.0.a4-blast-diamond-functional.tab\n",
      "--------------------------------------------------\n",
      "\n",
      "Line count:\n",
      "14676 Panopea-generosa-v1.0.a4-blast-diamond-functional_best.tab\n",
      "--------------------------------------------------\n",
      "\n",
      "PGEN_.00g000010\t121\t229\t165\tQ86IC9\tsp|Q86IC9|CAMT1_DICDI\t11\t122\t8.93e-14\t35.652\t115\n",
      "PGEN_.00g000020\t147\t467\t968\tP04177\tsp|P04177|TY3H_RAT\t20\t339\t3.47e-127\t55.140\t321\n",
      "PGEN_.00g000050\t566\t722\t180\tQ8L840\tsp|Q8L840|RQL4A_ARATH\t2\t167\t2.4e-12\t35.1\t168\n",
      "PGEN_.00g000060\t1957\t2106\t129\tQ61043\tsp|Q61043|NIN_MOUSE\t31\t184\t1.7e-06\t26.0\t154\n",
      "PGEN_.00g000080\t268\t322\t152\tA1E2V0\tsp|A1E2V0|BIRC3_CANLF\t163\t220\t3.91e-10\t53.448\t58\n",
      "PGEN_.00g000090\t199\t327\t161\tP34456\tsp|P34456|YMD2_CAEEL\t7\t134\t7.52e-12\t26.357\t129\n",
      "PGEN_.00g000120\t6\t49\t118\tP34457\tsp|P34457|YMD3_CAEEL\t90\t133\t3.2e-05\t47.7\t44\n",
      "PGEN_.00g000210\t18\t200\t263\tO00463\tsp|O00463|TRAF5_HUMAN\t5\t191\t2.24e-25\t34.921\t189\n",
      "PGEN_.00g000230\t48\t155\t287\tQ00945\tsp|Q00945|CONO_LYMST\t31\t134\t1.59e-32\t50.000\t108\n",
      "PGEN_.00g000240\t4\t585\t1144\tQ5SWK7\tsp|Q5SWK7|RN145_MOUSE\t13\t592\t1.2e-123\t40.1\t591\n",
      "PGEN_.00g000280\t4\t153\t210\tQ8ZXT3\tsp|Q8ZXT3|Y1111_PYRAE\t853\t1012\t1.10e-17\t38.750\t160\n",
      "PGEN_.00g000300\t159\t347\t480\tQ5REG4\tsp|Q5REG4|DTX3_PONAB\t1135\t1320\t1.20e-51\t50.794\t189\n",
      "PGEN_.00g000380\t381\t508\t205\tQ8QG60\tsp|Q8QG60|CRY2_CHICK\t2\t145\t4.92e-18\t36.111\t144\n",
      "PGEN_.00g000440\t792\t1796\t1362\tQ9H583\tsp|Q9H583|HEAT1_HUMAN\t539\t1575\t7.24e-156\t34.692\t1055\n",
      "PGEN_.00g000450\t347\t753\t1038\tA0JMR6\tsp|A0JMR6|MYSM1_XENLA\t138\t535\t3.16e-130\t48.426\t413\n",
      "PGEN_.00g000460\t478\t1045\t859\tO88917\tsp|O88917|AGRL1_RAT\t150\t740\t2.17e-95\t33.443\t610\n",
      "PGEN_.00g000490\t189\t393\t152\tQ7D513\tsp|Q7D513|EGTB_MYCTO\t307\t565\t1.7e-08\t24.8\t266\n",
      "PGEN_.00g000520\t118\t339\t495\tQ92968\tsp|Q92968|PEX13_HUMAN\t134\t356\t1.34e-56\t49.565\t230\n",
      "PGEN_.00g000530\t1\t191\t593\tA6H769\tsp|A6H769|RS7_BOVIN\t1\t160\t2.3e-60\t68.1\t191\n",
      "PGEN_.00g000540\t1876\t2081\t565\tQ14676\tsp|Q14676|MDC1_HUMAN\t1857\t2060\t1.19e-56\t46.117\t206\n",
      "PGEN_.00g000560\t17\t114\t135\tQ54W11\tsp|Q54W11|MCFL_DICDI\t21\t113\t3.1e-07\t32.7\t98\n",
      "PGEN_.00g000600\t49\t474\t1301\tQ9FKK7\tsp|Q9FKK7|XYLA_ARATH\t215\t641\t1.35e-173\t55.245\t429\n",
      "PGEN_.00g000660\t5\t308\t899\tQ3ZCD7\tsp|Q3ZCD7|TECR_BOVIN\t2\t303\t1.4e-95\t55.7\t305\n",
      "PGEN_.00g000670\t60\t1237\t3493\tQ9D180\tsp|Q9D180|CFA57_MOUSE\t1\t1143\t0.0\t59.102\t1181\n",
      "PGEN_.00g000680\t17\t206\t520\tQ9D6Z0\tsp|Q9D6Z0|ALKB7_MOUSE\t89\t278\t1.11e-64\t48.947\t190\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "cd \"${data_dir}\"\n",
    "\n",
    "for file in ${blast_annotations} ${diamond_annotations}\n",
    "do\n",
    "    awk 'NR > 13' ${file}\n",
    "done \\\n",
    ">> \"${blast_diamond_cat}\"\n",
    "\n",
    "sort -k1,1 -k9,9 \"${blast_diamond_cat}\" \\\n",
    "| sed 's/^21910-//' \\\n",
    "| sed 's/.m0[0-9]//' \\\n",
    "| awk '!array[$1]++' \\\n",
    ">> \"${blast_diamond_cat_best}\"\n",
    "\n",
    "echo \"\"\n",
    "echo \"Line count:\"\n",
    "\n",
    "wc -l \"${blast_diamond_cat}\"\n",
    "\n",
    "echo \"--------------------------------------------------\"\n",
    "\n",
    "echo \"\"\n",
    "echo \"Line count:\"\n",
    "\n",
    "wc -l \"${blast_diamond_cat_best}\"\n",
    "\n",
    "echo \"--------------------------------------------------\"\n",
    "echo \"\"\n",
    "\n",
    "head -n 25 \"${blast_diamond_cat_best}\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Create list of genome IDs and SwissProt IDs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Line count:\n",
      "14676 Panopea-generosa-v1.0.a4-blast-diamond-functional-genome_IDs-SPIDs.txt\n",
      "--------------------------------------------------\n",
      "Q86IC9 \t PGEN_.00g000010\n",
      "P04177 \t PGEN_.00g000020\n",
      "Q8L840 \t PGEN_.00g000050\n",
      "Q61043 \t PGEN_.00g000060\n",
      "A1E2V0 \t PGEN_.00g000080\n",
      "P34456 \t PGEN_.00g000090\n",
      "P34457 \t PGEN_.00g000120\n",
      "O00463 \t PGEN_.00g000210\n",
      "Q00945 \t PGEN_.00g000230\n",
      "Q5SWK7 \t PGEN_.00g000240\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "cd \"${analysis_dir}\"\n",
    "\n",
    "awk '{print $5,\"\\t\",$1}' \"${data_dir}\"/\"${blast_diamond_cat_best}\" > \"$genome_IDs_SPIDs\"\n",
    "\n",
    "echo \"\"\n",
    "echo \"Line count:\"\n",
    "\n",
    "wc -l \"$genome_IDs_SPIDs\"\n",
    "\n",
    "echo \"--------------------------------------------------\"\n",
    "\n",
    "head \"$genome_IDs_SPIDs\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Create list os SwissProt IDs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Line count:\n",
      "14676 Panopea-generosa-v1.0.a4-blast-diamond-functional-SPIDs.txt\n",
      "--------------------------------------------------\n",
      "\n",
      "Q86IC9\n",
      "P04177\n",
      "Q8L840\n",
      "Q61043\n",
      "A1E2V0\n",
      "P34456\n",
      "P34457\n",
      "O00463\n",
      "Q00945\n",
      "Q5SWK7\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "cd \"${analysis_dir}\"\n",
    "\n",
    "awk '{print $5}' \"${data_dir}\"/\"${blast_diamond_cat_best}\" > \"${spid_list}\"\n",
    "\n",
    "echo \"\"\n",
    "echo \"Line count:\"\n",
    "\n",
    "wc -l \"${spid_list}\"\n",
    "\n",
    "echo \"--------------------------------------------------\"\n",
    "\n",
    "echo \"\"\n",
    "\n",
    "head \"${spid_list}\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Batch submission/retrieval to/from UniProt\n",
    "\n",
    "Perl script obtained from UniProt: https://www.uniprot.org/help/api_batch_retrieval\n",
    "\n",
    "Modified to accept file with list of IDs and to map SPID to UniProt Accession"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "use strict;\n",
      "use warnings;\n",
      "use LWP::UserAgent;\n",
      "\n",
      "my $list = $ARGV[0]; # File containg list of UniProt identifiers.\n",
      "\n",
      "my $base = 'https://www.uniprot.org';\n",
      "my $tool = 'uploadlists';\n",
      "\n",
      "my $contact = 'samwhite@uw.edu'; # Please set a contact email address here to help us debug in case of problems (see https://www.uniprot.org/help/privacy).\n",
      "my $agent = LWP::UserAgent->new(agent => \"libwww-perl $contact\");\n",
      "push @{$agent->requests_redirectable}, 'POST';\n",
      "\n",
      "my $response = $agent->post(\"$base/$tool/\",\n",
      "                            [ 'file' => [$list],\n",
      "                              'format' => 'txt',\n",
      "                              'from' => 'SWISSPROT',\n",
      "                              'to' => 'ACC',\n",
      "                            ],\n",
      "                            'Content_Type' => 'form-data');\n",
      "\n",
      "while (my $wait = $response->header('Retry-After')) {\n",
      "  print STDERR \"Waiting ($wait)...\\n\";\n",
      "  sleep $wait;\n",
      "  $response = $agent->get($response->base);\n",
      "}\n",
      "\n",
      "$response->is_success ?\n",
      "  print $response->content :\n",
      "  die 'Failed, got ' . $response->status_line .\n",
      "    ' for ' . $response->request->uri . \"\\n\";\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "# Print script for viewing\n",
    "cat \"${uniprot_mapping_script}\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "cd \"${analysis_dir}\"\n",
    "\n",
    "# Run UniProt Prel mapping script and time how long it takes\n",
    "time \\\n",
    "perl \"${uniprot_mapping_script}\" \"${spid_list}\" > \"${uniprot_output}\"\n",
    "\n",
    "ls -ltrh\n",
    "\n",
    "echo \"\"\n",
    "echo \"\"\n",
    "echo \"--------------------------------------------------\"\n",
    "echo \"\"\n",
    "echo \"Line count:\"\n",
    "\n",
    "wc -l \"${uniprot_output}\"\n",
    "\n",
    "echo \"--------------------------------------------------\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Check mapping output\n",
    "\n",
    "Counting Accession lines (beginning with `AC`) should show a _lower_ count than the number of SwissProt IDs submitted, as UniProt automatically removes duplicates upon submission."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "cd \"${analysis_dir}\"\n",
    "\n",
    "head -n 30 \"${uniprot_output}\"\n",
    "\n",
    "echo \"\"\n",
    "\n",
    "echo \"----------------------------------------------------\"\n",
    "\n",
    "echo \"\"\n",
    "\n",
    "echo \"Number of accessions:\"\n",
    "\n",
    "echo \"\"\n",
    "\n",
    "grep -c \"^AC\" \"${uniprot_output}\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Parse the stuff we want\n",
    "\n",
    "- UniProt accession\n",
    "\n",
    "- Gene name/abbreviation\n",
    "\n",
    "- Gene description\n",
    "\n",
    "- GO IDs"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Check DE descriptor lines to decide pattern matching\n",
    "\n",
    "Checks lines beginning with `DE` to identify values in the 2nd field with `Name` in them.\n",
    "\n",
    "Identifies unique values. This will determine how to parse properly after this."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "cd \"${analysis_dir}\"\n",
    "\n",
    "grep \"^DE\" \"${uniprot_output}\" | awk '$2 ~ /Name/ { print $2 }' | sort -u"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "cd \"${analysis_dir}\"\n",
    "\n",
    "# Loop through UniProt records\n",
    "time \\\n",
    "while read -r line\n",
    "do\n",
    "  # Get record line descriptor\n",
    "  descriptor=$(echo \"${line}\" | awk '{print $1}')\n",
    "\n",
    "  # Capture second field for evaluation\n",
    "  go_line=$(echo \"${line}\" | awk '{print $2}')\n",
    "\n",
    "  # Append GO IDs to array\n",
    "  if [[ \"${go_line}\" == \"GO;\" ]]; then\n",
    "    go_id=$(echo \"${line}\" | awk '{print $3}')\n",
    "    go_ids_array+=(\"${go_id}\")\n",
    "  elif [[ \"${go_line}\" == \"GeneID;\" ]]; then\n",
    "    # Uses sed to strip trailing semi-colon\n",
    "    gene_id=$(echo \"${line}\" | awk '{print $3}' | sed 's/;$//')\n",
    "  fi\n",
    "\n",
    "  # Get gene description\n",
    "  if [[ \"${descriptor}\" == \"DE\" ]] && [[ \"${go_line}\" == \"RecName:\" ]]; then\n",
    "    # Uses sed to strip trailing spaces at end of line and remove commas\n",
    "    gene_description=$(echo \"${line}\" | awk -F \"[={]\" '{print $2}' | sed 's/[[:blank:]]*$//' | sed 's/,//g' | sed 's/;$//')\n",
    "\n",
    "  # Get alternate name\n",
    "  elif [[ \"${descriptor}\" == \"DE\" ]] && [[ \"${go_line}\" == \"AltName:\" ]]; then\n",
    "    # Uses sed to strip trailing spaces at end of line and remove commas\n",
    "    alt_gene_description=$(echo \"${line}\" | awk -F \"[={]\" '{print $2}' | sed 's/[[:blank:]]*$//' | sed 's/,//g' | sed 's/;$//')\n",
    "\n",
    "  # Get gene name\n",
    "  elif [[ \"${descriptor}\" == \"GN\"  ]] && [[ $(echo \"${line}\" | awk -F \"=\" '{print $1}') == \"GN   Name\" ]]; then\n",
    "    # Uses sed to strip trailing spaces at end of line\n",
    "    gene=$(echo \"${line}\" | awk -F 'Name=|{|;' '{print $2}' | sed 's/[[:blank:]]*$//')\n",
    "\n",
    "  # Get UniProt accession\n",
    "  elif [[ \"${descriptor}\" == \"AC\" ]]; then\n",
    "    # awk removes \"AC\" denotation\n",
    "    # sed removes all spaces\n",
    "    # sed removes trailing semi-colon\n",
    "    # Uses array to handle accessions being on multiple lines of UniProt records file\n",
    "    accession=$(echo \"${line}\" | awk '{$1=\"\";print $0}' | sed 's/[[:space:]]*//g' | sed 's/;$//')\n",
    "    accessions_array+=(\"${accession}\")\n",
    "\n",
    "  # Identify beginning on new record\n",
    "  elif [[ \"${descriptor}\" == \"//\" ]]; then\n",
    "\n",
    "    # Prints other comma-separated variables, then GOID1;GOID2;GOIDn\n",
    "    # IFS prevents spaces from being added between GO IDs\n",
    "    # sed removes \";\" after final GO ID\n",
    "    (IFS=; printf \"%s\\t%s\\t%s\\t%s\\t%s\\t%s\\n\" \"${accessions_array[*]}\" \"${gene_id}\" \"${gene}\" \"${gene_description}\" \"${alt_gene_description}\" \"${go_ids_array[*]}\" | sed 's/;$//')\n",
    "\n",
    "    # Re-initialize variables\n",
    "    accession=\"\"  \n",
    "    accessions_array=()\n",
    "    descriptor=\"\"\n",
    "    gene=\"\"\n",
    "    gene_description=\"\"\n",
    "    gene_id=\"\"\n",
    "    go_id=\"\"\n",
    "    go_ids_array=()\n",
    "  fi\n",
    "\n",
    "\n",
    "done < \"${uniprot_output}\" >> \"${parsed_uniprot}\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "tags": []
   },
   "source": [
    "### Inspect parsed UniProt file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "cd \"${analysis_dir}\"\n",
    "\n",
    "wc -l \"${parsed_uniprot}\"\n",
    "\n",
    "echo \"\"\n",
    "echo \"------------------------------------------------------------------\"\n",
    "echo \"\"\n",
    "\n",
    "head -n 25 \"${parsed_uniprot}\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%html\n",
    "# Sets markdown table align left in subsequent cell\n",
    "<style>\n",
    "  table {margin-left: 0 !important;}\n",
    "</style>"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Combine with original list of genes and SPIDs\n",
    "\n",
    "Output format (tab-delimited):\n",
    "\n",
    "| gene_ID | SPIDs | UniProt_gene_ID | gene | gene_description | alternate_gene_description | GO_IDs |\n",
    "|---------|-------|-----------------|------|------------------|----------------------------|--------|\n",
    "\n",
    "\n",
    "Explanation:\n",
    "\n",
    "- `awk -v FS='[;[:space:]]+'`: Sets the Field Separator variable to handle `; ` in UniProt accessions. Allows for proper searching.\n",
    "\n",
    "- `FNR == NR`: Restricts next block (designated by `{}`) to work only on first input file.\n",
    "\n",
    "- `{array[$1]=$0; next}`: Adds the entire line (`$0`) of the first file to the array names `array` and then moves on to the next set of commands for the second input file.\n",
    "\n",
    "- `($1 in array)`: Looks for the value of the first column (`$1`, which is SPID) from the second file to see if there's a match from the array (which contains the line from the first file).\n",
    "\n",
    "- `{print $2,array[$1]}'`: If there's a match, print the second column (`$2`, which is gene ID) from the second file, followed by the line from the first file.\n",
    "\n",
    "- `\"${parsed_uniprot}\" \"${spid_list}\"`: The first and second input files.\n",
    "\n",
    "- `\"${joined_output}\"`: Result of the join."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "cd \"${analysis_dir}\"\n",
    "\n",
    "awk \\\n",
    "-v FS='[;[:space:]]+' \\\n",
    "'NR==FNR \\\n",
    "{array[$1]=$0; next} \\\n",
    "($1 in array) \\\n",
    "{print $2\"\\t\"array[$1]}' \\\n",
    "\"${parsed_uniprot}\" \"${genome_IDs_SPIDs}\" \\\n",
    "> \"${joined_output}\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Inspect final annotation file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "cd \"${analysis_dir}\"\n",
    "\n",
    "wc -l \"${joined_output}\"\n",
    "\n",
    "echo \"\"\n",
    "echo \"------------------------------------------------------------------\"\n",
    "echo \"\"\n",
    "\n",
    "head -n 25 \"${joined_output}\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
