---
title: "20240916-snam-c1q-primer-design"
author: "Sam White"
date: "2024-09-16"
output: 
  github_document:
    toc: true
    number_sections: true
  bookdown::html_document2:
    theme: cosmo
    toc: true
    toc_float: true
    number_sections: true
    code_folding: show
    code_download: true
  html_document:
    theme: cosmo
    toc: true
    toc_float: true
    number_sections: true
    code_folding: show
    code_download: true
bibliography: references.bib
link-citations: true
---

This notebook describes using [Primer3](https://github.com/primer3-org/primer3) [@untergasser2012; @koressaar2007] to reproducibly design primers for sequencing of the [lake trout (*S.namaycush*)](https://en.wikipedia.org/wiki/Lake_trout) (Wikipedia) C1q gene ([`LOC120027825`](https://github.com/RobertsLab/resources/issues/1941#issuecomment-2246548960) (GitHub Issue)). Additionally, this is primarily to confirm that we can successfully amplify the C1q gene, since the [bisulfite PCR was unsuccessful](https://robertslab.github.io/sams-notebook/posts/2024/2024-09-11-Bisulfite-PCR---Second-Primer-Annealing-Gradient-Test-with-Lake-Trout-Bisulfite-treated-DNA/) (Notebook).

This process also utilizes [pyfaidx](https://github.com/mdshw5/pyfaidx) [@shirley2015].

[EMBOSS PrimerSearch](https://emboss.sourceforge.net/apps/cvs/emboss/apps/primersearch.html) [@rice2000] will be utilized to assess primer specificity across the genome.

The notebook entry was rendered directly from the R Markdown file [`20240916-snam-c1q-primer-design.Rmd](https://github.com/RobertsLab/code/blob/master/r_projects/sam/20240916-snam-c1q-primer-design/code/20240916-snam-c1q-primer-design.Rmd) (GitHub).

```{r setup, include=FALSE}
knitr::opts_chunk$set(
  echo = TRUE,         # Display code chunks
  eval = FALSE,        # Evaluate code chunks
  warning = FALSE,     # Hide warnings
  message = FALSE,     # Hide messages
  comment = ""         # Prevents appending '##' to beginning of lines in code output
)
```



# CREATE BASH VARIABLES FILE

This allows usage of Bash variables across R Markdown chunks.

```{r save-bash-variables-to-rvars-file, engine='bash', eval=TRUE}
{
echo "#### Assign Variables ####"
echo ""

echo "# DATA DIRECTORIES"
echo 'export data_dir="../data/S_namaycush/genomes"'
echo 'export output_top="../output"'
echo 'export genome_fasta_splits_dir="${data_dir}/fasta_splits"'
echo ""

echo "# SEQUENCE"
echo 'export sequence_ID="LOC120027825"'
echo ""

echo "# SEQUENCE REGIONS"
echo 'export left_buffer="500"'
echo 'export right_buffer="500"'
echo ""

echo "# INPUT FILES"
echo 'export genome_fasta="GCF_016432855.1_SaNama_1.0_genomic.fna"'
echo 'export genome_gff="GCF_016432855.1_SaNama_1.0_genomic.gff"'
echo 'export ncbi_gff_gz="GCF_016432855.1_SaNama_1.0_genomic.gff.gz"'
echo 'export ncbi_fasta_gz="GCF_016432855.1_SaNama_1.0_genomic.fna.gz"'
echo 'export ncbi_md5sums="md5checksums.txt"'
echo 'export ncbi_url="https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/016/432/855/GCF_016432855.1_SaNama_1.0/"'
echo ""

echo "# OUTPUT FILES"
echo 'export c1q_fasta="${sequence_ID}.fasta"'
echo 'export c1q_buffer_fasta="${sequence_ID}-${left_buffer}bp.left-${right_buffer}bp.right.fasta"'
echo 'export c1q_faidx_region_file="${sequence_ID}"-region.txt'
echo 'export c1q_buffer_faidx_region_file="${sequence_ID}-${left_buffer}bp.left-${right_buffer}bp.right-region.txt"'
echo ""

echo "# SET CPUS"
echo 'export threads=40'
echo ""

echo "# PROGRAMS"
echo 'export pyfaidx=/home/shared/pyfaidx-0.8.1.1'
echo 'export primer3_dir="/home/shared/primer3-2.6.1/src"'
echo 'export primer3="${primer3_dir}/primer3_core"'
echo 'export primer3_config="${primer3_dir}/primer3_config"'
echo 'export primersearch="/home/shared/EMBOSS-6.6.0/emboss/primersearch"'



} > .bashvars

cat .bashvars
```

```{r save-buffer-values-to-R-vars, eval=TRUE, echo=FALSE, include=FALSE}
# Use system to get the value of left_buffer from Bash into an R variable
left_buffer <- system("bash -c 'source .bashvars && echo $left_buffer'", intern=TRUE)

right_buffer <- system("bash -c 'source .bashvars && echo $right_buffer'", intern=TRUE)

```

# DOWNLOAD NCBI GENOME FILES

## Download the actual files

```{bash download-ncbi-genome-files, engine='bash', eval=TRUE}
# Load bash variables into memory
source .bashvars

for file in ${ncbi_gff_gz} ${ncbi_fasta_gz} ${ncbi_md5sums}
do
  wget \
  --no-check-certificate \
  --continue \
  --quiet \
  --directory-prefix=${data_dir} \
  ${ncbi_url}${file}
done

ls -lh "${data_dir}"
```

## Check MD5 Checkums

```{bash verify-ncbi-checksums, engine='bash', eval=TRUE}
# Load bash variables into memory
source .bashvars

cd "${data_dir}"

for file in *.gz
do
  grep "${file}" ${ncbi_md5sums} | md5sum -c -
done

```

## Decompress NCBI files

```{bash decompress-files, engine='bash', eval=TRUE}
# Load bash variables into memory
source .bashvars

cd "${data_dir}"

for file in *.gz
do
  gunzip "${file}"
done

ls -lh
```

# EXTRACT C1Q GENE SEQUENCE

## Peek at GFF

```{bash peek-at-gff, engine='bash', eval=TRUE}
# Load bash variables into memory
source .bashvars

awk '$3=="gene"' "${data_dir}/${genome_gff}" | grep "${sequence_ID}" | column -t

```

## Format region and extract C1q sequence as FastA.

```{bash extract-c1q-region, engine='bash', eval=TRUE}
# Load bash variables into memory
source .bashvars

# Format region for use with pyfaidx
region=$(awk '$3=="gene"' "${data_dir}/${genome_gff}" | grep "${sequence_ID}" | awk '{print $1":"$4"-"$5}')

echo "${region}"
echo ""

# Extract region
${pyfaidx} "${data_dir}/${genome_fasta}" "${region}" | tee "${output_top}/${c1q_fasta}"

```

# EXTRACT C1Q WITH 5'/3' BUFFER REGIONS

Since we want to sequence the entirety of the C1q gene, we need to add some buffer sequence outside of the 5'/3' ends of the gene. I've arbitrarily established large buffer regions:

-   Left buffer: `r left_buffer`bp
-   Right buffer: `r right_buffer`bp

```{bash extract-buffered-c1q-fasta, engine='bash', eval=TRUE}
# Load bash variables into memory
source .bashvars

# Format region for use with pyfaidx
# Use awk to add/subtract desired buffer regions
region=$(awk '$3=="gene"' "${data_dir}/${genome_gff}" \
| grep "${sequence_ID}" \
| awk -v left_buffer="$left_buffer" -v right_buffer="$right_buffer" '{print $1":"$4 - left_buffer"-"$5 + right_buffer}')

echo "Region formatting: ${region}"
echo ""

# Extract region
${pyfaidx} "${data_dir}/${genome_fasta}" "${region}" | tee "${output_top}/${c1q_buffer_fasta}"

```

# PRIMER DESIGN USING [PRIMER3](https://github.com/primer3-org/primer3)

## Design primers

Quick explanation: [Primer3](https://github.com/primer3-org/primer3) requires a specially formatted input file. The file must be formatted similarly to this:

```         
SEQUENCE_ID=${seq_id}
SEQUENCE_TEMPLATE=${sequence}
PRIMER_TASK=generic
PRIMER_PICK_LEFT_PRIMER=3
PRIMER_PICK_RIGHT_PRIMER=3
PRIMER_OPT_SIZE=18
PRIMER_MIN_SIZE=15
PRIMER_MAX_SIZE=21
PRIMER_MAX_NS_ACCEPTED=1
PRIMER_PRODUCT_SIZE_RANGE=75-150
P3_FILE_FLAG=1
PRIMER_EXPLAIN_FLAG=1
=
```

Values after the `=` on each line can be changed to whatever values the user decides. The \${sequence} must be a nucletoide sequence on a single line, with no line breaks.

The code in the chunk below uses a heredoc to write this information to a file. Use of a heredoc allows the variables specified in the Primer3 config to expand to their actual values. Everything *between* the following two lines gets printed (via cat) as shown and then redirected to the indicated file (`primer3-params.txt`):

``` bash
cat << EOF > ${output_top}/primer3-params.txt
EOF
```

[Primer3](https://github.com/primer3-org/primer3) is run with the `--format_output` to make a nice, human-readable output format.


I've also set [Primer3](https://github.com/primer3-org/primer3) to look for sequencing primers and have defined the `SEQUENCE_TARGET`.

```{bash primer3-with-t, engine='bash', eval=TRUE}
# Load bash variables into memory
source .bashvars

# Get sequence only, by skipping the first record
# Remove newlines so sequence is on a single line
c1q_sequence=$(awk 'NR > 1' "${output_top}/${c1q_fasta}" | tr -d '\n')

# Determine length of C1q
c1q_length=${#c1q_sequence}

# Get sequence only, by skipping the first record
# Remove newlines so sequence is on a single line
c1q_buffer_sequence=$(awk 'NR > 1' "${output_top}/${c1q_buffer_fasta}" | tr -d '\n')

# Determine length of C1q with the 5'/3' buffer sequences included
c1q_buffer_length=${#c1q_buffer_sequence}

# Calculate combined length of 5'buffer region and C1q length
c1q_buffer_and_seq_length=$((left_buffer + c1q_length))


# Use heredoc to create Primer3 parameters file
cat << EOF > ${output_top}/primer3-params.txt
SEQUENCE_ID=${sequence_ID}
SEQUENCE_TEMPLATE=${c1q_buffer_sequence}
PRIMER_TASK=pick_sequencing_primers
SEQUENCE_TARGET=${left_buffer},${c1q_length}
PRIMER_MIN_TM=50
PRIMER_OPT_TM=60
PRIMER_PICK_LEFT_PRIMER=1
PRIMER_PICK_RIGHT_PRIMER=1
PRIMER_OPT_SIZE=20
PRIMER_MIN_SIZE=18
PRIMER_MAX_SIZE=25
PRIMER_MAX_NS_ACCEPTED=0
P3_FILE_FLAG=1
PRIMER_EXPLAIN_FLAG=1
PRIMER_THERMODYNAMIC_PARAMETERS_PATH=${primer3_config}
PRIMER_NUM_RETURN=10
PRIMER_PAIR_EXPLAIN=considered 0, ok 0
=
EOF


# Run Primer3
${primer3} \
--format_output \
--output="${output_top}/primer3-primers.txt" \
"${output_top}/primer3-params.txt"

# Run Primer3 with default output for parsable results
${primer3} \
--output="${output_top}/primer3-primers-default-format.txt" \
"${output_top}/primer3-params.txt"


# Pull out any primers falling outside of target
tail -n 30 "${output_top}/primer3-primers.txt" \
| head -n -12 \
| awk -v left_buffer="$left_buffer" -v target="$c1q_buffer_and_seq_length" '$3 <= left_buffer || $3 >= target {print $0}'


### Write the left and right primer start positions to files for later use in R chunk(s).
# Pull out left primer start position
left_primer_start=$(tail -n 30 "${output_top}/primer3-primers.txt" \
| head -n -12 \
| awk -v left_buffer="$left_buffer" -v target="$c1q_buffer_and_seq_length" '$3 <= left_buffer || $3 >= target {print $0}' \
| grep "PRIMER" \
| awk 'NR == 1 {print $3}')

# Pull out right primer start position
right_primer_start=$(tail -n 30 "${output_top}/primer3-primers.txt" \
| head -n -12 \
| awk -v left_buffer="$left_buffer" -v target="$c1q_buffer_and_seq_length" '$3 <= left_buffer || $3 >= target {print $0}' \
| grep "PRIMER" \
| awk 'NR == 2 {print $3}')

echo "${left_primer_start}" > "${output_top}/left-primer-start.txt"
echo "${right_primer_start}" > "${output_top}/right-primer-start.txt"

echo ""
echo ""
echo "----------------------------------------------------------------------------------"
echo ""
echo "Primer results file:"
echo ""
echo ""
# Print the full output file
cat "${output_top}/primer3-primers.txt"

echo ""
echo ""
echo "----------------------------------------------------------------------------------"
echo ""
echo "Primer default (i.e. computer-friendly) results file:"
echo ""
echo ""

# Print the full output file
cat "${output_top}/primer3-primers-default-format.txt"
```

Looks like we have one primer pair where both are outside of the C1q gene sequence.

## Amplicon length

```{r amplicon-length, eval=TRUE, echo=FALSE, include=FALSE}
# Load left and right primer lengths
left_primer_start <- as.numeric(readLines("../output/left-primer-start.txt"))

right_primer_start <- as.numeric(readLines("../output/right-primer-start.txt"))

# Calculate amplicon length.
amplicon_length <- right_primer_start - left_primer_start
```

The primer pair will produced an amplicon length of `r amplicon_length`bp.

Next, we need to test them against the entire genome to assess specificity.

# SPLIT GENOME

For some reason, I feel like the EMBOSS PrimerSearch tool will _not_ work on mulit-FastA files (or, it makes the results more difficult to decipher?), but I'm just going off of memory and repeating a process I've done previously.

This will split the genome multi-FastA file into separate FastA files.

## Split mulit-FastA file in to individual FastA files with PyFaidx

```{bash split-genome-fasta, engine='bash', eval=TRUE}
# Load bash variables into memory
source .bashvars

# Make directory if it doesn't exist
mkdir --parents ${genome_fasta_splits_dir}

cd ${genome_fasta_splits_dir}

# Count sequences in FastA
echo "-------------------------------------------------------------------"
echo "NUMBER OF SEQUENCES IN ORIGINAL FASTA"
grep -c "^>" ../${genome_fasta}
echo "-------------------------------------------------------------------"
echo ""
echo ""



# Split FastA
${pyfaidx} \
--split-files \
../${genome_fasta}

# Count number of individual FastA files
echo "-------------------------------------------------------------------"
echo "NUMBER OF INDIVIDUAL FASTA FILES"
ls -1 | wc -l
echo "-------------------------------------------------------------------"
```

# PRIMER SEARCH WITH [EMBOSS PRIMERSEARCH](https://emboss.sourceforge.net/apps/cvs/emboss/apps/primersearch.html)

This will run [EMBOSS PrimerSearch](https://emboss.sourceforge.net/apps/cvs/emboss/apps/primersearch.html) against the genome to assess primer specificity.

## Create EMBOSS PrimerSearch Primers File

Create a tab-delimited file to use with EMBOSS PrimerSearch.


```{bash create-emboss-primers-file, engine='bash', eval=TRUE}
# Load bash variables into memory
source .bashvars

# Get primer info from Primer3 default format output
seq_id=$(grep "SEQUENCE_ID=" "${output_top}/primer3-primers-default-format.txt" | sed 's/SEQUENCE_ID=//')

left_primer=$(grep "PRIMER_LEFT_0_SEQUENCE=" "${output_top}/primer3-primers-default-format.txt" | sed 's/PRIMER_LEFT_0_SEQUENCE=//' | tr '[:lower:]' '[:upper:]')

right_primer=$(grep "PRIMER_RIGHT_7_SEQUENCE=" "${output_top}/primer3-primers-default-format.txt" | sed 's/PRIMER_RIGHT_7_SEQUENCE=//' | tr '[:lower:]' '[:upper:]')

# Create EMBOSS primer file
printf "%s\t%s\t%s\t\n" "${seq_id}" "${left_primer}" "${right_primer}" | tee "${output_top}/emboss-primers.txt"
```

## Run EMBOSS PrimerSearch

This will run EMBOSS PrimerSearch and allow for a 10% mismatch in primer annealing sites (`${primersearch} -auto ${fasta} ${primers} 10`).

Afterwards, the resulting output files (`*.primersearch`) will be searched for the term `Amplimer`, indicating a PCR product would be produced. If no amplimer is identified in a `*.primersearch` file, then that file is deleted. This should leave just the results in which primer matches were identified.

```{bash emboss-primersearch, engine='bash', eval=TRUE}
# Load bash variables into memory
source .bashvars

cd ${genome_fasta_splits_dir}

primers="../../../../output/emboss-primers.txt"


time \
for fasta in *.fna
  do
  # Remove path from FastA filename
  fasta_no_path=$(echo ${fasta##*/})
  
  # Remove file extension from FastA filename
  fasta_no_ext=$(echo ${fasta_no_path%%.*})
  
  # Convert filename to lowercase
  # Will be used for output from EMBOSS PrimerSearch
  fasta_no_ext_lower=$(echo ${fasta_no_ext} | tr '[:upper:]' '[:lower:]')
  
  ###### Run EMBOSS PrimerSearch on all FastA files ########
  # Allows for a 10% mismatch
  ${primersearch} -auto ${fasta} ${primers} 10
  ##### END EMBOSS ##########
  
  # Find EMBOSS PrimerSearch output files with primer matches
  # Remove those without a match
  if ! grep --quiet "Amplimer" "${fasta_no_ext_lower}.primersearch"
    then rm ${fasta_no_ext_lower}.primersearch
  fi
done

```

## Check primer matches

This will print the contents of any remaining `*.primersearch` output files (i.e. sequences with primer matches)

```{bash inspect-EMBOSS-primer-matches, engine='bash', eval=TRUE}
# Load bash variables into memory
source .bashvars

cd ${genome_fasta_splits_dir}

# Check contents of files with matches
for file in *.primersearch
  do
  echo "FILE: ${file}"
  echo ""
  cat ${file}
  echo "----------------------------------"
  echo ""
done
```

# SUMMARY

EMBOSS PrimerSearch has identified only a single location in the genome where the forward and reverse primers match, as well as in the expected location on chromosome `NC_052339`.

# CITATIONS
