---
title: "cbai_deg_multiple_conditions"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## R Markdown

This is an R Markdown document. Markdown is a simple formatting syntax for authoring HTML, PDF, and MS Word documents. For more details on using R Markdown see <http://rmarkdown.rstudio.com>.

When you click the **Knit** button a document will be generated that includes both content as well as the output of any embedded R code chunks within the document. You can embed an R code chunk like this:

```{r libraries}
library("DESeq2")
library("tidyverse")
library("dplyr")
library("pheatmap")
library("RColorBrewer")
library("genefilter")
library("ggplot2")
library("gplots")
library("limma")
library("spdep") 
library("adegenet")
library("tximport")
```

```{r download files}
download.file(url = "https://gannet.fish.washington.edu/Atumefaciens/20200429_cbai_salmon_2020GW_transcript_abundances/72_quant/quant.sf",
destfile = "../data/72_quant.sf")

download.file(url = "https://gannet.fish.washington.edu/Atumefaciens/20200429_cbai_salmon_2020GW_transcript_abundances/73_quant/quant.sf",
destfile = "../data/73_quant.sf")

download.file(url = "https://gannet.fish.washington.edu/Atumefaciens/20200429_cbai_salmon_2020GW_transcript_abundances/113_quant/quant.sf",
destfile = "../data/113_quant.sf")

download.file(url = "https://gannet.fish.washington.edu/Atumefaciens/20200429_cbai_salmon_2020GW_transcript_abundances/118_quant/quant.sf",
destfile = "../data/118_quant.sf")

download.file(url = "https://gannet.fish.washington.edu/Atumefaciens/20200429_cbai_salmon_2020GW_transcript_abundances/127_quant/quant.sf",
destfile = "../data/127_quant.sf")

download.file(url = "https://gannet.fish.washington.edu/Atumefaciens/20200429_cbai_salmon_2020GW_transcript_abundances/132_quant/quant.sf",
destfile = "../data/132_quant.sf")

download.file(url = "https://gannet.fish.washington.edu/Atumefaciens/20200429_cbai_salmon_2020GW_transcript_abundances/151_quant/quant.sf",
destfile = "../data/151_quant.sf")

download.file(url = "https://gannet.fish.washington.edu/Atumefaciens/20200429_cbai_salmon_2020GW_transcript_abundances/173_quant/quant.sf",
destfile = "../data/173_quant.sf")

download.file(url = "https://gannet.fish.washington.edu/Atumefaciens/20200429_cbai_salmon_2020GW_transcript_abundances/178_quant/quant.sf",
destfile = "../data/178_quant.sf")

download.file(url = "https://gannet.fish.washington.edu/Atumefaciens/20200429_cbai_salmon_2020GW_transcript_abundances/221_quant/quant.sf",
destfile = "../data/221_quant.sf")

download.file(url = "https://gannet.fish.washington.edu/Atumefaciens/20200429_cbai_salmon_2020GW_transcript_abundances/222_quant/quant.sf",
destfile = "../data/222_quant.sf")

download.file(url = "https://gannet.fish.washington.edu/Atumefaciens/20200429_cbai_salmon_2020GW_transcript_abundances/254_quant/quant.sf",
destfile = "../data/254_quant.sf")

download.file(url = "https://gannet.fish.washington.edu/Atumefaciens/20200429_cbai_salmon_2020GW_transcript_abundances/272_quant/quant.sf",
destfile = "../data/272_quant.sf")

download.file(url = "https://gannet.fish.washington.edu/Atumefaciens/20200429_cbai_salmon_2020GW_transcript_abundances/280_quant/quant.sf",
destfile = "../data/280_quant.sf")

download.file(url = "https://gannet.fish.washington.edu/Atumefaciens/20200429_cbai_salmon_2020GW_transcript_abundances/294_quant/quant.sf",
destfile = "../data/294_quant.sf")

download.file(url = "https://gannet.fish.washington.edu/Atumefaciens/20200429_cbai_salmon_2020GW_transcript_abundances/334_quant/quant.sf",
destfile = "../data/334_quant.sf")

download.file(url = "https://gannet.fish.washington.edu/Atumefaciens/20200429_cbai_salmon_2020GW_transcript_abundances/349_quant/quant.sf",
destfile = "../data/349_quant.sf")

download.file(url = "https://gannet.fish.washington.edu/Atumefaciens/20200429_cbai_salmon_2020GW_transcript_abundances/359_quant/quant.sf",
destfile = "../data/359_quant.sf")

download.file(url = "https://gannet.fish.washington.edu/Atumefaciens/20200429_cbai_salmon_2020GW_transcript_abundances/425_quant/quant.sf",
destfile = "../data/425_quant.sf")

download.file(url = "https://gannet.fish.washington.edu/Atumefaciens/20200429_cbai_salmon_2020GW_transcript_abundances/427_quant/quant.sf",
destfile = "../data/427_quant.sf")

download.file(url = "https://gannet.fish.washington.edu/Atumefaciens/20200429_cbai_salmon_2020GW_transcript_abundances/445_quant/quant.sf",
destfile = "../data/445_quant.sf")

download.file(url = "https://gannet.fish.washington.edu/Atumefaciens/20200429_cbai_salmon_2020GW_transcript_abundances/463_quant/quant.sf",
destfile = "../data/463_quant.sf")

download.file(url = "https://gannet.fish.washington.edu/Atumefaciens/20200429_cbai_salmon_2020GW_transcript_abundances/481_quant/quant.sf",
destfile = "../data/481_quant.sf")

download.file(url = "https://gannet.fish.washington.edu/Atumefaciens/20200429_cbai_salmon_2020GW_transcript_abundances/485_quant/quant.sf",
destfile = "../data/485_quant.sf")

```

```{bash check downloaded files}
head ../data/72_quant.sf | column -t
```
```{r treatment info}
treatmentinfo <- read.csv("../data/cbai_2020gw_rnaseq_sample_info.csv", header = TRUE, sep = ",")
treatmentinfo
```



```{r import gene counts}
# Read in salmon gene count data
# First column has no label, so use "empty" row.names to assign Trinity gene IDs to row names
gcount <- read.csv("../data/salmon.gene.counts.matrix", header = TRUE, row.names = "", check.names = FALSE, sep = "\t")
head(gcount)

# Round numbers to integers
gcount <- gcount %>% 
  rownames_to_column('gene_id') %>% 
  mutate_if(is.numeric, round, 0) %>% 
  column_to_rownames('gene_id')

head(gcount)
```




```{r match sample IDs}
#Ensure all sample IDs in colData are also in CountData and match their orders
# Sort column names
gcount <- gcount[ , order(names(gcount))]

head(gcount)

# Remove "_quant" from column names
colnames(gcount) = gsub(pattern = "_quant", replacement = "", colnames(gcount))

head(gcount)

rownames(treatmentinfo) <- treatmentinfo$sample_id
colnames(gcount) <- treatmentinfo$sample_id

head(treatmentinfo)
head(gcount)

# Check row and column names match. Should return "TRUE".
all(rownames(treatmentinfo) %in% colnames(gcount))
all(rownames(treatmentinfo) == colnames(gcount))
```
```{r}
inf_status <- sort(unique(treatmentinfo$infection_status))
days <- sort(unique(treatmentinfo$day))
```

```{r}
treatmentinfo$infection_status <- factor(treatmentinfo$infection_status, levels = inf_status)
treatmentinfo$day <- factor(treatmentinfo$day) #set group values as factor variables

head(treatmentinfo)
```


```{r}
#Set DESeq2 design
gdds <- DESeqDataSetFromMatrix(countData = gcount,
                              colData = treatmentinfo,
                              design = ~infection_status + day + infection_status:day)
```

```{r}
SF.gdds <- estimateSizeFactors( gdds ) #estimate size factors to determine if we can use vst  to transform our data. Size factors should be less than for to use vst
print(sizeFactors(SF.gdds)) #View size factors
```

```{r}
gvst <- vst(gdds, blind = FALSE)

head (assay(gvst), 3)
```

```{r}
gsampleDists <- dist(t(assay(gvst))) #calculate distance matix
gsampleDistMatrix <- as.matrix(gsampleDists) #distance matrix
rownames(gsampleDistMatrix) <- colnames(gvst) #assign row names
colnames(gsampleDistMatrix) <- NULL #assign col names
colors <- colorRampPalette( rev(brewer.pal(9, "Blues")) )(255) #assign colors
pheatmap(gsampleDistMatrix, #plot matrix
         clustering_distance_rows=gsampleDists, #cluster rows
         clustering_distance_cols=gsampleDists, #cluster columns
         col=colors) #set colors
```

```{r}
gPCAdata <- plotPCA(gvst, intgroup = c("infection_status", "day"), returnData=TRUE)
percentVar <- round(100*attr(gPCAdata, "percentVar")) #plot PCA of samples with all data
ggplot(gPCAdata, aes(PC1, PC2, color=infection_status, shape=day)) + 
  geom_point(size=3) +
  xlab(paste0("PC1: ",percentVar[1],"% variance")) +
  ylab(paste0("PC2: ",percentVar[2],"% variance")) + 
  scale_color_manual(values = c(infected="red", uninfected="cadetblue4")) +
  coord_fixed()
```

```{r}
DEG.group <- DESeq(gdds, test = "LRT", reduced = ~infection_status + day) #run differential expression test by group using the LRT model
DEG.group.results <- results(DEG.group) #save DE results
resultsNames(DEG.group) #view DE results
```

```{r}
group.results.ordered <- order(DEG.group.results$pvalue) #Order p-values by smallest value first
summary(DEG.group.results) #view results summary
```

```{r}
sig.num <- sum(DEG.group.results$padj < 0.05, na.rm=TRUE) #How many adjusted p-values were less than 0.05?
sig.num
```

